<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-08-21" />

<title>Collapse / filtering traits</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Collapse / filtering traits</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-08-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreed4ec61e976a18365521d4949ee0b9e2f7d4a0edbtargetblankd4ec61ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/d4ec61e976a18365521d4949ee0b9e2f7d4a0edb" target="_blank">d4ec61e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreed4ec61e976a18365521d4949ee0b9e2f7d4a0edbtargetblankd4ec61ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/d4ec61e976a18365521d4949ee0b9e2f7d4a0edb" target="_blank">d4ec61e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/gwas_catalog/
    Ignored:    output/gwas_study_info_cohort_corrected.csv

Untracked files:
    Untracked:  data/.DS_Store
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   code/collapse_diseases.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/collapse_traits.Rmd</code>) and
HTML (<code>docs/collapse_traits.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/d4ec61e976a18365521d4949ee0b9e2f7d4a0edb/analysis/collapse_traits.Rmd" target="_blank">d4ec61e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-21
</td>
<td>
Adding investigation harmonizing mapped traits
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
library(data.table)
library(dplyr)
library(ggplot2)
library(stringr)</code></pre>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_study_info_cohort_corrected.csv&quot;))

# number of studies per mapped trait
n_studies_trait = gwas_study_info |&gt;
  dplyr::group_by(MAPPED_TRAIT, MAPPED_TRAIT_URI) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))

n_studies_trait |&gt;
  head()</code></pre>
<pre><code># A tibble: 6 × 3
# Groups:   MAPPED_TRAIT [6]
  MAPPED_TRAIT               MAPPED_TRAIT_URI                          n_studies
  &lt;chr&gt;                      &lt;chr&gt;                                         &lt;int&gt;
1 neuroimaging measurement   http://www.ebi.ac.uk/efo/EFO_0004346           9773
2 protein measurement        http://www.ebi.ac.uk/efo/EFO_0004747           8748
3 blood protein amount       http://purl.obolibrary.org/obo/OBA_VT000…      5336
4 metabolite measurement     http://www.ebi.ac.uk/efo/EFO_0004725           4559
5 blood metabolite level     http://purl.obolibrary.org/obo/OBA_20500…      2738
6 gut microbiome measurement http://www.ebi.ac.uk/efo/EFO_0007874           2363</code></pre>
<pre class="r"><code>n_studies_trait |&gt;
  slice_head(n = 20) |&gt;
  ggplot(aes(y = reorder(MAPPED_TRAIT, n_studies), x = n_studies)) + 
  geom_col() + 
  theme_bw() + 
  labs(y = &quot;Trait&quot;, x = &quot;Number of studies&quot;)</code></pre>
<p><img src="figure/collapse_traits.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="unmapped-badly-mapped-traits" class="section level1">
<h1>Unmapped / badly mapped traits:</h1>
<pre class="r"><code># number of traits only observed once in the catalog 
# likely badly mapped traits: 

n_studies_trait |&gt;
  dplyr::filter(n_studies == 1) |&gt;
  nrow()</code></pre>
<pre><code>[1] 11592</code></pre>
<pre class="r"><code>unique_mapped_traits = n_studies_trait |&gt;
  dplyr::filter(n_studies == 1) |&gt;
  dplyr::pull(MAPPED_TRAIT)

gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits) |&gt;
  group_by(PUBMED_ID) |&gt;
  summarise(n_unmapped_per_study = n()) |&gt;
  arrange(desc(n_unmapped_per_study)) |&gt;
  head(n = 10)</code></pre>
<pre><code># A tibble: 10 × 2
   PUBMED_ID n_unmapped_per_study
       &lt;int&gt;                &lt;int&gt;
 1  35870639                 3411
 2  38412862                 2821
 3  39789286                  612
 4  29875488                  388
 5  39528826                  384
 6  33634981                  197
 7  24816252                  173
 8  36482414                  121
 9  26068415                   99
10  37253714                   95</code></pre>
<pre class="r"><code># this study tests 
# 6,790 proteins or protein complexes
gwas_study_info |&gt;
  filter(PUBMED_ID == 35870639) |&gt;
  nrow()</code></pre>
<pre><code>[1] 6790</code></pre>
<pre class="r"><code># not all are unmapped 
gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits,
                PUBMED_ID == &quot;35870639&quot;) |&gt;
  nrow()# map to blood protein amount</code></pre>
<pre><code>[1] 3411</code></pre>
<pre class="r"><code># serum proteome - i.e. blood proteome 

gwas_study_info  = 
gwas_study_info |&gt;
  dplyr::rows_update(tibble(PUBMED_ID = 35870639, 
                            MAPPED_TRAIT = &quot;blood protein amount&quot;,
                            MAPPED_TRAIT_URI = &quot;http://purl.obolibrary.org/obo/OBA_VT0005416&quot;),
                     unmatched = &quot;ignore&quot;)</code></pre>
<pre class="r"><code>gwas_study_info |&gt;
  filter(PUBMED_ID == 38412862) |&gt;
  nrow()</code></pre>
<pre><code>[1] 2821</code></pre>
<pre class="r"><code># 2,821 ratios between protein levels
# i.e. each study is a association study between ratio between two proteins

# ? put under blood protein amount (even though it is a ratio between amounts not an amount...)
gwas_study_info  = 
gwas_study_info |&gt;
  dplyr::rows_update(tibble(PUBMED_ID = 3841286, 
                            MAPPED_TRAIT = &quot;blood protein amount&quot;,
                            MAPPED_TRAIT_URI = &quot;http://purl.obolibrary.org/obo/OBA_VT0005416&quot;),
                     unmatched = &quot;ignore&quot;)</code></pre>
<pre class="r"><code>gwas_study_info |&gt;
  filter(PUBMED_ID == 39789286) |&gt;
  nrow()</code></pre>
<pre><code>[1] 3049</code></pre>
<pre class="r"><code># not all are unmapped 
gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits,
                PUBMED_ID == 39789286) |&gt;
  nrow()</code></pre>
<pre><code>[1] 612</code></pre>
<pre class="r"><code>not_well_mapped_traits = 
gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits,
                PUBMED_ID == 39789286) |&gt;
  pull(MAPPED_TRAIT)

grepl(&quot;in blood&quot;, not_well_mapped_traits) |&gt; length()</code></pre>
<pre><code>[1] 612</code></pre>
<pre class="r"><code>grepl(&quot;level of&quot;, not_well_mapped_traits) |&gt; length()</code></pre>
<pre><code>[1] 612</code></pre>
<pre class="r"><code># All unmapped traits are protein levels in blood 

gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits,
                PUBMED_ID == 39789286) |&gt;
  dplyr::filter(grepl(&quot;in blood&quot;, MAPPED_TRAIT)) |&gt; 
  nrow()</code></pre>
<pre><code>[1] 612</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  rowwise() |&gt;
  mutate(MAPPED_TRAIT = ifelse(PUBMED_ID == 39789286 &amp;&amp; MAPPED_TRAIT %in% unique_mapped_traits,
                               &quot;blood protein amount&quot;,
                               MAPPED_TRAIT)
         ) |&gt;
    mutate(MAPPED_TRAIT_URI = ifelse(PUBMED_ID == 39789286 &amp;&amp; MAPPED_TRAIT %in% unique_mapped_traits,
                                       &quot;http://purl.obolibrary.org/obo/OBA_VT0005416&quot;,
                                     MAPPED_TRAIT_URI)
           ) |&gt;
  ungroup()</code></pre>
<pre class="r"><code>gwas_study_info |&gt;
  filter(PUBMED_ID == 29875488) |&gt;
  nrow()</code></pre>
<pre><code>[1] 3283</code></pre>
<pre class="r"><code>gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits,
                PUBMED_ID == 29875488) |&gt;
  nrow()</code></pre>
<pre><code>[1] 388</code></pre>
<pre class="r"><code>not_well_mapped_traits = 
gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits,
                PUBMED_ID == 29875488) |&gt;
  pull(MAPPED_TRAIT)

grepl(&quot;measurement&quot;, not_well_mapped_traits) |&gt; length()</code></pre>
<pre><code>[1] 388</code></pre>
<pre class="r"><code># study title is: Genomic atlas of the human plasma proteome.

# assume therefore, all these values are protein blood measurements

gwas_study_info = gwas_study_info |&gt;
  rowwise() |&gt;
  mutate(MAPPED_TRAIT = ifelse(PUBMED_ID == 29875488 &amp;&amp; MAPPED_TRAIT %in% unique_mapped_traits,
                               &quot;blood protein amount&quot;,
                               MAPPED_TRAIT)
         ) |&gt;
    mutate(MAPPED_TRAIT_URI = ifelse(PUBMED_ID == 29875488 &amp;&amp; MAPPED_TRAIT %in% unique_mapped_traits,
                               &quot;http://purl.obolibrary.org/obo/OBA_VT0005416&quot;,
                               MAPPED_TRAIT_URI)
           ) |&gt;
  ungroup()</code></pre>
<pre class="r"><code># PUBMED_ID: 39528826

# suspect: 
# metabolite measurement    http://www.ebi.ac.uk/efo/EFO_0004725

# Title:
# Publication: Genetic architecture of cerebrospinal fluid and brain metabolite levels and the genetic colocalization of metabolites with human traits.</code></pre>
<pre class="r"><code># PUBMED+ID: 33634981

# suspect: 
# metabolite measurement    http://www.ebi.ac.uk/efo/EFO_0004725

# Genome-wide association study of metabolites in patients with coronary artery disease identified novel metabolite quantitative trait loci.</code></pre>
<pre class="r"><code># 24816252

# suspect: 
# metabolite measurement    http://www.ebi.ac.uk/efo/EFO_0004725

# An atlas of genetic influences on human blood metabolites.</code></pre>
<pre class="r"><code># 36482414
# Publication: Comprehensive characterization of putative genetic influences on plasma metabolome in a pediatric cohort.</code></pre>
<pre class="r"><code>gwas_study_info |&gt;
  dplyr::filter(MAPPED_TRAIT %in% unique_mapped_traits) |&gt;
  group_by(PUBMED_ID, STUDY, COHORT) |&gt;
  summarise(n_unmapped_per_study = n()) |&gt;
  arrange(desc(n_unmapped_per_study))</code></pre>
<pre><code># A tibble: 1,550 × 4
# Groups:   PUBMED_ID, STUDY [1,513]
   PUBMED_ID STUDY                                   COHORT n_unmapped_per_study
       &lt;int&gt; &lt;chr&gt;                                   &lt;chr&gt;                 &lt;int&gt;
 1  38412862 Genetic associations with ratios betwe… &quot;UKBB&quot;                 2821
 2  39528826 Genetic architecture of cerebrospinal … &quot;Knig…                  381
 3  33634981 Genome-wide association study of metab… &quot;&quot;                      197
 4  24816252 An atlas of genetic influences on huma… &quot;Twin…                  127
 5  36482414 Comprehensive characterization of puta… &quot;othe…                  121
 6  37253714 Whole-Genome Sequencing Analysis of Hu… &quot;ARIC…                   95
 7  37794183 Rare variant associations with plasma … &quot;UKBB&quot;                   89
 8  34737426 A generalized linear mixed model assoc… &quot;UKBB&quot;                   81
 9  26068415 Genome-wide association study identifi… &quot;EGCU…                   76
10  35347128 Genome-wide association studies of met… &quot;METS…                   55
# ℹ 1,540 more rows</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] stringr_1.5.1     ggplot2_3.5.2     dplyr_1.1.4       data.table_1.17.8
[5] workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       jsonlite_2.0.0     compiler_4.3.1     renv_1.0.3        
 [5] promises_1.3.3     tidyselect_1.2.1   Rcpp_1.1.0         git2r_0.36.2      
 [9] callr_3.7.6        later_1.4.2        jquerylib_0.1.4    scales_1.4.0      
[13] yaml_2.3.10        fastmap_1.2.0      here_1.0.1         R6_2.6.1          
[17] labeling_0.4.3     generics_0.1.4     knitr_1.50         tibble_3.3.0      
[21] rprojroot_2.1.0    RColorBrewer_1.1-3 bslib_0.9.0        pillar_1.11.0     
[25] rlang_1.1.6        utf8_1.2.6         cachem_1.1.0       stringi_1.8.7     
[29] httpuv_1.6.16      xfun_0.52          getPass_0.2-4      fs_1.6.6          
[33] sass_0.4.10        cli_3.6.5          withr_3.0.2        magrittr_2.0.3    
[37] ps_1.9.1           grid_4.3.1         digest_0.6.37      processx_3.8.6    
[41] rstudioapi_0.17.1  lifecycle_1.0.4    vctrs_0.6.5        evaluate_1.0.4    
[45] glue_1.8.0         farver_2.1.2       whisker_0.4.1      rmarkdown_2.29    
[49] httr_1.4.7         tools_4.3.1        pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
