<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Disease investigated by ancestry</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Disease investigated by ancestry</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreefa9a4daca664acf271eb0b33e79dbcda8d80cf78targetblankfa9a4daa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/fa9a4daca664acf271eb0b33e79dbcda8d80cf78" target="_blank">fa9a4da</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreefa9a4daca664acf271eb0b33e79dbcda8d80cf78targetblankfa9a4daa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/fa9a4daca664acf271eb0b33e79dbcda8d80cf78" target="_blank">fa9a4da</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/gwas_catalog/
    Ignored:    output/gwas_study_info_cohort_corrected.csv

Untracked files:
    Untracked:  data/.DS_Store
    Untracked:  gwas_time.pdf
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   code/collapse_diseases.R

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/disease_inves_by_ancest.Rmd</code>) and HTML
(<code>docs/disease_inves_by_ancest.html</code>) files. If youâ€™ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/fa9a4daca664acf271eb0b33e79dbcda8d80cf78/analysis/disease_inves_by_ancest.Rmd" target="_blank">fa9a4da</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-21
</td>
<td>
Starting test of relationship between proportion european and total
sample size
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/f5087d26cfd6a40b0d36c0e613a92061d8e81f81/docs/disease_inves_by_ancest.html" target="_blank">f5087d2</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/72172e35a53558385850a7967b1d415b69be6834/analysis/disease_inves_by_ancest.Rmd" target="_blank">72172e3</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
<td>
Split page into disease by ancest
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/2fd5755c1e0ffdca6c453751d9ce6c164e192cf6/docs/disease_inves_by_ancest.html" target="_blank">2fd5755</a>
</td>
<td>
Isobel Beasley
</td>
<td>
2022-02-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/7347b5d16849bb0b60e4fc7f831d46fbbef3848f/analysis/disease_inves_by_ancest.Rmd" target="_blank">7347b5d</a>
</td>
<td>
Isobel Beasley
</td>
<td>
2022-02-16
</td>
<td>
Add initial plotting using gwas cat stats
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<pre class="r"><code>library(dplyr)
library(data.table)
library(ggplot2)
# 
# gwas_study_info = data.table::fread(&quot;data/gwas_catalog/gwas-catalog-v1.0.3-studies-r2022-02-02.tsv&quot;, 
#                                     sep = &quot;\t&quot;, 
#                                     quote = &quot;&quot;)
 
gwas_study_info &lt;- fread(here::here(&quot;output/gwas_study_info_cohort_corrected.csv&quot;))

gwas_ancest_info &lt;-  fread(here::here(&quot;data/gwas_catalog/gwas-catalog-v1.0.3.1-ancestries-r2025-07-21.tsv&quot;),
                         sep = &quot;\t&quot;, quote = &quot;&quot;)

gwas_study_info = gwas_study_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_ancest_info = gwas_ancest_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_ancest_info = gwas_ancest_info |&gt; arrange(DATE)

gwas_study_info = gwas_study_info |&gt; arrange(DATE)

# Set up custom theme for ggplots
custom_theme &lt;-
  list(
    theme_bw() +
      theme(
        panel.border = element_blank(),
        axis.line = element_line(),
        text = element_text(size = 16),
        legend.position = &quot;bottom&quot;,
        strip.background = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)
      )
  )</code></pre>
</div>
<div id="get-top-disease-traits" class="section level1">
<h1>Get top disease traits</h1>
<p>The traits with the most number of pubmed ids:</p>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::select(MAPPED_TRAIT, MAPPED_TRAIT_URI, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(MAPPED_TRAIT, MAPPED_TRAIT_URI) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;MAPPED_TRAIT&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>head(n_studies_trait)</code></pre>
<pre><code># A tibble: 6 Ã— 3
# Groups:   MAPPED_TRAIT [6]
  MAPPED_TRAIT                                     MAPPED_TRAIT_URI    n_studies
  &lt;chr&gt;                                            &lt;chr&gt;                   &lt;int&gt;
1 high density lipoprotein cholesterol measurement http://www.ebi.ac.â€¦       134
2 body mass index                                  http://www.ebi.ac.â€¦       133
3 triglyceride measurement                         http://www.ebi.ac.â€¦       129
4 low density lipoprotein cholesterol measurement  http://www.ebi.ac.â€¦       119
5 type 2 diabetes mellitus                         http://purl.obolibâ€¦       118
6 total cholesterol measurement                    http://www.ebi.ac.â€¦       103</code></pre>
<div id="what-are-the-ancestry-groups" class="section level3">
<h3>What are the ancestry groups?</h3>
<pre class="r"><code># code adapted from https://github.com/armartin/prs_disparities/blob/master/gwas_disparities_time.R

group_ancestry_fn = function(study_ancest){
  
   case_when(
     
  # European
  study_ancest %in% c(&#39;European&#39;) ~ &#39;European&#39;,
  
  # African
  study_ancest %in% c(&#39;Sub-Saharan African, African American or Afro-Caribbean&#39;,
                      &#39;African unspecified, African American or Afro-Caribbean&#39;,
                      &#39;African American or Afro-Caribbean, African unspecified&#39;,
                      &#39;Sub-Saharan African, African unspecified&#39;,
                      &#39;African-American or Afro-Caribbean&#39;,
                      &#39;Sub-Saharan African&#39;,
                      &#39;African American or Afro-Caribbean&#39;,
                      &#39;African unspecified&#39;) ~ &#39;African&#39;,
      
  # Asian  
  study_ancest %in% c(&#39;East Asian, Asian unspecified&#39;, 
                      &#39;South Asian, East Asian &#39;,
                      &#39;South Asian, South East Asian&#39;, 
                      &#39;South Asian, South East Asian, East Asian&#39;,
                      &#39;South East Asian, East Asian&#39;, 
                      &#39;South East Asian, South Asian, East Asian&#39;,
                      &#39;South Asian, South East Asian, East Asian, Asian unspecified&#39;,
                      &#39;South East Asian, East Asian, South Asian&#39;,
                      &#39;East Asian, South Asian, South East Asian&#39;,
                      &#39;East Asian, South East Asian, South Asian, Asian unspecified&#39;,
                      &#39;South Asian&#39;,
                      &#39;South East Asian&#39;,
                      &#39;South Asian, East Asian&#39;,
                      &#39;South Asian, Asian unspecified&#39;,
                      &#39;Central Asian, South Asian&#39;,
                      &#39;East Asian, South Asian&#39;,
                      &#39;Central Asian&#39;,
                      &#39;East Asian&#39;,
                      &#39;Asian unspecified&#39;) ~&#39;Asian&#39;,
  
  # Middle eastern    
  study_ancest == &#39;Greater Middle Eastern (Middle Eastern, North African or Persian)&#39; ~&#39;Middle Eastern&#39;,

  # Oceanic
  study_ancest %in% c(&#39;Aboriginal Australian&#39;, 
                      &#39;Oceanian&#39;) ~&#39;Oceanic&#39;,
  
  # Hispanic/Latin American
  study_ancest %in% &quot;Hispanic or Latin American&quot; ~&#39;Hispanic/Latin American&#39;,
  
  # Other 
  study_ancest %in% c(&#39;Other&#39;,
                      &#39;Other, NR&#39;, 
                      &#39;NR, Other&#39;,
                      &#39;Other admixed ancestry&#39;, 
                      &#39;Native American&#39;) ~ &quot;Other&quot;,
  
  # Not reported
  study_ancest %in% &quot;NR&quot; ~ &quot;Not reported&quot;,
  
  # Multiple    
  grepl(&quot;, &quot;, study_ancest) ~ &#39;Multiple&#39;,
      
  TRUE~study_ancest
    )
  
    
  
}</code></pre>
<pre class="r"><code>ancestry_colors &lt;- c(
  &quot;African&quot; = &quot;#984EA3&quot;,
  &quot;European&quot; = &quot;#E41A1C&quot;,
  &quot;Asian&quot; = &quot;#377EB8&quot;, #east asian
  # &quot;South Asian&quot; = &quot;#4DAF4A&quot;,
  &quot;Hispanic/Latin American&quot; = &quot;#FF7F00&quot;,
  &quot;Middle Eastern&quot; = &quot;#FFFF33&quot;,
  &quot;Oceanic&quot; = &quot;#A65628&quot;,
  &quot;Other&quot; = &quot;#F781BF&quot;,
  &quot;Multiple&quot; = &quot;#999999&quot;,
  &quot;Not reported&quot; = &quot;black&quot;
  
)

# Define the desired stacking order
ancestry_levels &lt;- c(
  &quot;European&quot;,
  # &quot;East Asian&quot;,
  # &quot;South Asian&quot;,
  &quot;Asian&quot;,
  &quot;African&quot;,
  &quot;Hispanic/Latin American&quot;,
  &quot;Middle Eastern&quot;,
  &quot;Oceanic&quot;,
  &quot;Other&quot;,
  &quot;Multiple&quot;,
  &quot;Not reported&quot;
)</code></pre>
</div>
<div id="apply-this-ancestry-mapping" class="section level3">
<h3>Apply this ancestry mapping</h3>
<pre class="r"><code>grouped_ancest = vector()

for(study_ancest in unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY)){
  
grouped_ancest[study_ancest] = group_ancestry_fn(study_ancest)

}

grouped_ancest_map = data.frame(ancestry_group = grouped_ancest,
                                BROAD_ANCESTRAL_CATEGORY = unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY)
                                )


print(head(grouped_ancest_map))</code></pre>
<pre><code>                                   ancestry_group
European                                 European
Asian unspecified                           Asian
Other                                       Other
African American or Afro-Caribbean        African
NR                                   Not reported
South East Asian                            Asian
                                             BROAD_ANCESTRAL_CATEGORY
European                                                     European
Asian unspecified                                   Asian unspecified
Other                                                           Other
African American or Afro-Caribbean African American or Afro-Caribbean
NR                                                                 NR
South East Asian                                     South East Asian</code></pre>
<pre class="r"><code>gwas_ancest_info = dplyr::left_join(
            gwas_ancest_info,
            grouped_ancest_map,
            by = &quot;BROAD_ANCESTRAL_CATEGORY&quot;)</code></pre>
<pre class="r"><code>print(&quot;Total numbers (in millions) per ancestry group&quot;)</code></pre>
<pre><code>[1] &quot;Total numbers (in millions) per ancestry group&quot;</code></pre>
<pre class="r"><code>gwas_ancest_info %&gt;% 
  group_by(ancestry_group) %&gt;% 
  summarise(n = sum(NUMBER_OF_INDIVIDUALS, na.rm = TRUE)/10^5)</code></pre>
<pre><code># A tibble: 9 Ã— 2
  ancestry_group                 n
  &lt;chr&gt;                      &lt;dbl&gt;
1 African                  4077.  
2 Asian                    3128.  
3 European                91218.  
4 Hispanic/Latin American  1787.  
5 Middle Eastern             44.7 
6 Multiple                 1862.  
7 Not reported             1488.  
8 Oceanic                     1.34
9 Other                      20.1 </code></pre>
<pre class="r"><code>gwas_ancest_info %&gt;% 
  filter(!is.na(NUMBER_OF_INDIVIDUALS)) |&gt;
  arrange(DATE) %&gt;%
  mutate(ancestry_group = factor(ancestry_group, levels = ancestry_levels)) %&gt;%
  group_by(ancestry_group) %&gt;% 
  mutate(ancest_cumsum = cumsum(as.numeric(NUMBER_OF_INDIVIDUALS))) %&gt;% 
  select(DATE, ancest_cumsum, ancestry_group, NUMBER_OF_INDIVIDUALS) |&gt;
  ggplot(aes(x=DATE, y=ancest_cumsum/(10^6), fill = ancestry_group)
         ) + 
  geom_area(position = &#39;stack&#39;) + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  theme_classic() + 
  labs(x = &quot;Year&quot;, y = &quot;Individuals in GWAS catalog (millons)&quot;) + 
  scale_fill_manual(values = ancestry_colors, name=&#39;Ancestry group&#39;)  </code></pre>
<p><img src="figure/disease_inves_by_ancest.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="repeat-for-traits-only" class="section level1">
<h1>Repeat for traits only</h1>
<pre class="r"><code>gwas_ancest_info = 
left_join(
gwas_ancest_info,
gwas_study_info |&gt; select(STUDY_ACCESSION, 
                          COHORT, 
                          MAPPED_TRAIT),
by = &quot;STUDY_ACCESSION&quot;
)</code></pre>
<pre class="r"><code>gwas_ancest_info_plot = 
gwas_ancest_info %&gt;% 
  filter(!is.na(NUMBER_OF_INDIVIDUALS)) |&gt;
  filter(MAPPED_TRAIT == &#39;high density lipoprotein cholesterol measurement&#39;)


print(&quot;Total numbers (in millions) per ancestry group&quot;)</code></pre>
<pre><code>[1] &quot;Total numbers (in millions) per ancestry group&quot;</code></pre>
<pre class="r"><code>gwas_ancest_info_plot %&gt;% 
  group_by(ancestry_group) %&gt;% 
  summarise(n = sum(NUMBER_OF_INDIVIDUALS, na.rm = TRUE)/10^5)</code></pre>
<pre><code># A tibble: 9 Ã— 2
  ancestry_group                 n
  &lt;chr&gt;                      &lt;dbl&gt;
1 African                  14.5   
2 Asian                    35.5   
3 European                214.    
4 Hispanic/Latin American   7.71  
5 Middle Eastern            0.172 
6 Multiple                 52.1   
7 Not reported              5.17  
8 Oceanic                   0.0965
9 Other                     0.0751</code></pre>
<pre class="r"><code>gwas_ancest_info_plot %&gt;% 
  group_by(ancestry_group) %&gt;% 
  mutate(ancestry_group = factor(ancestry_group, levels = ancestry_levels)) %&gt;%
  mutate(ancest_cumsum = cumsum(as.numeric(NUMBER_OF_INDIVIDUALS))) %&gt;% 
  ggplot(aes(x=DATE, y=ancest_cumsum/(10^6), fill = ancestry_group)) + 
  #geom_area() + 
  geom_area(position = &#39;stack&#39;) + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  theme_classic() + 
  labs(x = &quot;Year&quot;, y = &quot;Individuals in GWAS catalog (millons)&quot;) + 
  scale_fill_manual(values = ancestry_colors, name=&#39;Ancestry group&#39;) </code></pre>
<p><img src="figure/disease_inves_by_ancest.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="proportion-european-per-trait" class="section level1">
<h1>Proportion European per trait</h1>
<pre class="r"><code>euro_n = gwas_ancest_info |&gt;
         filter(ancestry_group == &quot;European&quot;) |&gt;
         pull(NUMBER_OF_INDIVIDUALS) |&gt;
         sum(na.rm = T)

total_n = gwas_ancest_info |&gt;
         pull(NUMBER_OF_INDIVIDUALS) |&gt;
         sum(na.rm = T)

100 * euro_n / total_n</code></pre>
<pre><code>[1] 88.02662</code></pre>
<pre class="r"><code>prop_euro = vector()
total_n_vec = vector()


for(trait in n_studies_trait$MAPPED_TRAIT[1:1000]){
  
  euro_n = gwas_ancest_info |&gt;
           filter(ancestry_group == &quot;European&quot;) |&gt;
           filter(MAPPED_TRAIT == trait) |&gt;
           pull(NUMBER_OF_INDIVIDUALS) |&gt;
           sum(na.rm = T)
  
  total_n = gwas_ancest_info |&gt;
            filter(MAPPED_TRAIT == trait) |&gt;
            pull(NUMBER_OF_INDIVIDUALS) |&gt;
            sum(na.rm = T)
  
  prop_euro[trait] = 100 * euro_n / total_n
  total_n_vec[trait] = total_n
  
  #print(prop_euro[trait])
  
}

prop_euro_df = data.frame(prop_euro = prop_euro, 
                          trait = names(prop_euro),
                          total_n = total_n_vec)

prop_euro_df = left_join(prop_euro_df,
                         n_studies_trait |&gt; rename(trait = MAPPED_TRAIT),
                         by = &quot;trait&quot;)

prop_euro_df$prop_euro |&gt; summary()</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  12.07   81.42   89.92   86.23   93.92  100.00 </code></pre>
<pre class="r"><code>prop_euro_df |&gt; slice_min(prop_euro, n = 5)</code></pre>
<pre><code>  prop_euro                               trait total_n
1  12.06720     adolescent idiopathic scoliosis  386693
2  16.09211      response to antipsychotic drug   14678
3  17.40991              dilated cardiomyopathy 7229158
4  26.17495              glucose tolerance test  796781
5  30.40375 chronic hepatitis C virus infection 1182900
                           MAPPED_TRAIT_URI n_studies
1      http://www.ebi.ac.uk/efo/EFO_0005423        10
2 http://purl.obolibrary.org/obo/GO_0097332        12
3      http://www.ebi.ac.uk/efo/EFO_0000407         8
4      http://www.ebi.ac.uk/efo/EFO_0004307         8
5      http://www.ebi.ac.uk/efo/EFO_0004220        14</code></pre>
<pre class="r"><code>prop_euro_df |&gt; slice_max(prop_euro, n =5 )</code></pre>
<pre><code>  prop_euro                                trait  total_n
1 100.00000                 testicular carcinoma  1796522
2 100.00000 self reported educational attainment  4908190
3  99.99236      base metabolic rate measurement  3807772
4  99.99201                           hair color  1063503
5  99.98250                      eye measurement 16269499
                      MAPPED_TRAIT_URI n_studies
1 http://www.ebi.ac.uk/efo/EFO_0005088        11
2 http://www.ebi.ac.uk/efo/EFO_0004784         9
3 http://www.ebi.ac.uk/efo/EFO_0007777         9
4 http://www.ebi.ac.uk/efo/EFO_0003924        10
5 http://www.ebi.ac.uk/efo/EFO_0004731        13</code></pre>
<pre class="r"><code>prop_euro_df |&gt; slice_max(total_n, n = 5)</code></pre>
<pre><code>  prop_euro                        trait   total_n
1  77.99807     neuroimaging measurement 219548076
2  97.75004             diet measurement  79737834
3  81.97671                     COVID-19  74845527
4  99.96221                 brain volume  64578173
5  99.77048 brain connectivity attribute  60026325
                              MAPPED_TRAIT_URI n_studies
1         http://www.ebi.ac.uk/efo/EFO_0004346        13
2         http://www.ebi.ac.uk/efo/EFO_0008111        17
3 http://purl.obolibrary.org/obo/MONDO_0100096        34
4   http://purl.obolibrary.org/obo/OBA_2050009         8
5   http://purl.obolibrary.org/obo/OBA_2045295         8</code></pre>
<pre class="r"><code>cor(prop_euro_df$prop_euro, prop_euro_df$total_n, 
    method = &quot;spearman&quot;)</code></pre>
<pre><code>[1] 0.08398827</code></pre>
<pre class="r"><code>prop_euro_df |&gt;
  ggplot(aes(x = total_n, y = prop_euro)) + 
  geom_point() + 
  theme_bw() + 
  labs(x = &quot;Total number of individuals (studied for this trait)&quot;,
       y = &quot;Proportion of European ancestry idividuals (studied for this trait)&quot;)</code></pre>
<p><img src="figure/disease_inves_by_ancest.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>prop_euro_df |&gt;
  ggplot(aes(x = total_n, y = n_studies)) + 
  geom_point() + 
  theme_bw() + 
  labs(x = &quot;Total number of individuals (studied for this trait)&quot;,
       y = &quot;Total number of unique PUBMED IDs for this trait&quot;)</code></pre>
<p><img src="figure/disease_inves_by_ancest.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>prop_euro_df |&gt;
  ggplot(aes(x = n_studies, y = prop_euro)) + 
  geom_point() + 
  theme_bw() + 
  labs(x = &quot;Total number of unique PUBMED IDs for this trait&quot;,
       y = &quot;Proportion of European ancestry idividuals (studied for this trait)&quot;)</code></pre>
<p><img src="figure/disease_inves_by_ancest.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="disease-statistics-cdc" class="section level1">
<h1>Disease statistics CDC</h1>
<pre class="r"><code># following steps from https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-019-0379-x/MediaObjects/41588_2019_379_MOESM1_ESM.pdf

# wonder cdc 
# https://wonder.cdc.gov/wonder/help/QuickStart.html#
# https://wonder.cdc.gov/controller/datarequest/D76;jsessionid=2A56E973A3DF13BC5DFDD9C43725

# https://cran.r-project.org/web/packages/whomds/index.html
# whomds: Calculate Results from WHO Model Disability Survey Data




cdc_stats = data.table::fread(&quot;data/cdc/Underlying Cause of Death, 1999-2020.txt&quot;,
                              drop = c(&quot;Notes&quot;, &quot;Race Code&quot;, &quot;Cause of death Code&quot;)) %&gt;% 
             filter(!if_any(everything(), ~.x == &quot;&quot;))

cdc_stats %&gt;% 
  group_by(Race) %&gt;% 
  slice_max(Deaths,n=10)


# wonder cdc 
# https://wonder.cdc.gov/wonder/help/QuickStart.html#
# https://wonder.cdc.gov/controller/datarequest/D76;jsessionid=2A56E973A3DF13BC5DFDD9C43725

# https://cran.r-project.org/web/packages/whomds/index.html
# whomds: Calculate Results from WHO Model Disability Survey Data</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] ggplot2_3.5.2     data.table_1.17.8 dplyr_1.1.4       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       jsonlite_2.0.0     compiler_4.3.1     renv_1.0.3        
 [5] promises_1.3.3     tidyselect_1.2.1   Rcpp_1.1.0         stringr_1.5.1     
 [9] git2r_0.36.2       callr_3.7.6        later_1.4.2        jquerylib_0.1.4   
[13] scales_1.4.0       yaml_2.3.10        fastmap_1.2.0      here_1.0.1        
[17] R6_2.6.1           labeling_0.4.3     generics_0.1.4     knitr_1.50        
[21] tibble_3.3.0       rprojroot_2.1.0    RColorBrewer_1.1-3 bslib_0.9.0       
[25] pillar_1.11.0      rlang_1.1.6        utf8_1.2.6         cachem_1.1.0      
[29] stringi_1.8.7      httpuv_1.6.16      xfun_0.52          getPass_0.2-4     
[33] fs_1.6.6           sass_0.4.10        cli_3.6.5          withr_3.0.2       
[37] magrittr_2.0.3     ps_1.9.1           grid_4.3.1         digest_0.6.37     
[41] processx_3.8.6     rstudioapi_0.17.1  lifecycle_1.0.4    vctrs_0.6.5       
[45] evaluate_1.0.4     glue_1.8.0         farver_2.1.2       whisker_0.4.1     
[49] rmarkdown_2.29     httr_1.4.7         tools_4.3.1        pkgconfig_2.0.3   
[53] htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
