<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Disease investigated by ancestry</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Disease investigated by ancestry</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree72172e35a53558385850a7967b1d415b69be6834targetblank72172e3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/72172e35a53558385850a7967b1d415b69be6834" target="_blank">72172e3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree72172e35a53558385850a7967b1d415b69be6834targetblank72172e3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/72172e35a53558385850a7967b1d415b69be6834" target="_blank">72172e3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/cdc/
    Ignored:    data/gwas_catalog/

Untracked files:
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   README.md
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/disease_inves_by_ancest.Rmd</code>) and HTML
(<code>docs/disease_inves_by_ancest.html</code>) files. If youâ€™ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/72172e35a53558385850a7967b1d415b69be6834/analysis/disease_inves_by_ancest.Rmd" target="_blank">72172e3</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
<td>
Split page into disease by ancest
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/2fd5755c1e0ffdca6c453751d9ce6c164e192cf6/docs/disease_inves_by_ancest.html" target="_blank">2fd5755</a>
</td>
<td>
Isobel Beasley
</td>
<td>
2022-02-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/7347b5d16849bb0b60e4fc7f831d46fbbef3848f/analysis/disease_inves_by_ancest.Rmd" target="_blank">7347b5d</a>
</td>
<td>
Isobel Beasley
</td>
<td>
2022-02-16
</td>
<td>
Add initial plotting using gwas cat stats
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<pre class="r"><code>library(dplyr)
library(ggplot2)

gwas_study_info = data.table::fread(&quot;data/gwas_catalog/gwas-catalog-v1.0.3-studies-r2022-02-02.tsv&quot;, 
                                    sep = &quot;\t&quot;, 
                                    quote = &quot;&quot;)

gwas_ancest_info = data.table::fread(&quot;./data/gwas_catalog/gwas_catalog-ancestry_r2022-02-02.tsv&quot;, 
                                     sep = &quot;\t&quot;, 
                                     quote = &quot;&quot;)

gwas_study_info = gwas_study_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_ancest_info = gwas_ancest_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

# Set up custom theme for ggplots
custom_theme &lt;-
  list(
    theme_bw() +
      theme(
        panel.border = element_blank(),
        axis.line = element_line(),
        text = element_text(size = 16),
        legend.position = &quot;bottom&quot;,
        strip.background = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)
      )
  )</code></pre>
</div>
<div id="plot-figure-martin-et-al.-2019-like" class="section level2">
<h2>Plot figure Martin et al.Â 2019 like</h2>
<div id="for-all-ancestries" class="section level3">
<h3>For all ancestries</h3>
<pre class="r"><code># code adapted from https://github.com/armartin/prs_disparities/blob/master/gwas_disparities_time.R

# https://github.com/armartin/prs_disparities/blob/master/gwas_sfs_pop.R
# gwas cat
# http://bioconductor.org/packages/release/bioc/html/gwascat.html


# following steps from https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-019-0379-x/MediaObjects/41588_2019_379_MOESM1_ESM.pdf

# wonder cdc 
# https://wonder.cdc.gov/wonder/help/QuickStart.html#
# https://wonder.cdc.gov/controller/datarequest/D76;jsessionid=2A56E973A3DF13BC5DFDD9C43725

# https://cran.r-project.org/web/packages/whomds/index.html
# whomds: Calculate Results from WHO Model Disability Survey Data


# Order GWAS catalog by date 
gwas_ancest_info &lt;- gwas_ancest_info %&gt;% arrange(DATE)

# calculate cumulative number of individuals
gwas_ancest_info = gwas_ancest_info %&gt;%  
                   mutate(cum_num = cumsum(ifelse(is.na(NUMBER_OF_INDIVDUALS), 0, NUMBER_OF_INDIVDUALS))
                          )


# plot cumulative numbers
gwas_ancest_info %&gt;%  
  group_by(DATE) %&gt;% 
  slice_max(NUMBER_OF_INDIVDUALS) %&gt;% 
  ggplot(aes(x=DATE,y=cum_num/1e6)) + 
  geom_line() +
  #geom_area(position = &#39;stack&#39;) + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  custom_theme + 
  labs(x = &quot;Year&quot;, y = &quot;Individuals in GWAS catalog (millons)&quot;)</code></pre>
<p><img src="figure/disease_inves_by_ancest.Rmd/plot_like_martin_et_al_2019_not_grouped-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_like_martin_et_al_2019_not_grouped-1">
Past versions of plot_like_martin_et_al_2019_not_grouped-1.png
</button>
</p>
<div id="fig-plot_like_martin_et_al_2019_not_grouped-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/2fd5755c1e0ffdca6c453751d9ce6c164e192cf6/docs/figure/disease_inves_by_ancest.Rmd/plot_like_martin_et_al_2019_not_grouped-1.png" target="_blank">2fd5755</a>
</td>
<td>
Isobel Beasley
</td>
<td>
2022-02-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="group-ancestry-into-broader-categories" class="section level3">
<h3>Group ancestry into broader categories</h3>
<pre class="r"><code>group_ancestry_fn = function(study_ancest){
  
    case_when(study_ancest %in% c(&#39;Sub-Saharan African, African American or Afro-Caribbean&#39;,
                           &#39;Sub-Saharan African, African unspecified&#39;,
                           &#39;African-American or Afro-Caribbean&#39;) ~ &#39;African&#39;,
      #grouped_ancest = append(grouped_ancest,&#39;African&#39;)
      
      
    study_ancest %in% c(&#39;East Asian, Asian unspecified&#39;, 
                                  &#39;South Asian, East Asian &#39;,
                                  &#39;South Asian, South East Asian&#39;, 
                                  &#39;South Asian, South East Asian, East Asian&#39;,
                                   &#39;South East Asian, East Asian&#39;, 
                                  &#39;South East Asian, South Asian, East Asian&#39;) ~&#39;Asian unspecified&#39;,
      #grouped_ancest = append(grouped_ancest,&#39;Asian unspecified&#39;)
      
      
    study_ancest == &#39;Greater Middle Eastern (Middle Eastern, North African or Persian)&#39; ~&#39;Middle Eastern&#39;,
      #grouped_ancest = append(grouped_ancest,&#39;Middle Eastern&#39;)
   study_ancest %in% c(&#39;Aboriginal Australian&#39;, &#39;Oceanian&#39;) ~&#39;Oceanic&#39;,
      #grouped_ancest = append(grouped_ancest,&#39;Oceanic&#39;)
      
      
  grepl(&quot;, &quot;, study_ancest) ~ &#39;Multiple&#39;,
      #grouped_ancest = append(grouped_ancest,&#39;Multiple&#39;)
      
      
  study_ancest %in% &quot;Hispanic or Latin American&quot; ~&#39;Hispanic/Latin American&#39;,
      #grouped_ancest = append(grouped_ancest,&#39;Hispanic/Latin American&#39;)
      
      
  TRUE~study_ancest
      #grouped_ancest = append(grouped_ancest,study_ancest)
    )
  
    
  
}


grouped_ancest = vector()

for(study_ancest in unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY)){
  
grouped_ancest[study_ancest] = group_ancestry_fn(study_ancest)

}


# ancest_group = data.frame(group_ancest = grouped_ancest,
#                          BROAD_ANCESTRAL_CATEGORY =
#                            unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY))
# 
# }

grouped_ancest_map = data.frame(ancestry_group = grouped_ancest,
                                BROAD_ANCESTRAL_CATEGORY = unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY)
                                )

gwas_ancest_info = dplyr::left_join(
            gwas_ancest_info,
            grouped_ancest_map)</code></pre>
<pre><code>Joining with `by = join_by(BROAD_ANCESTRAL_CATEGORY)`</code></pre>
<pre class="r"><code>gwas_ancest_info %&gt;% 
  group_by(ancestry_group) %&gt;% 
  summarise(n = sum(NUMBER_OF_INDIVDUALS, na.rm = TRUE))</code></pre>
<pre><code># A tibble: 18 Ã— 2
   ancestry_group                              n
   &lt;chr&gt;                                   &lt;dbl&gt;
 1 African                                 60441
 2 African American or Afro-Caribbean    8278986
 3 African unspecified                   3028217
 4 Asian unspecified                     4697042
 5 Central Asian                           42945
 6 East Asian                          124056468
 7 European                           4235129752
 8 Hispanic/Latin American              10840275
 9 Middle Eastern                         348192
10 Multiple                             56292255
11 NR                                  110448036
12 Native American                         91973
13 Oceanic                                116432
14 Other                                  194359
15 Other admixed ancestry                 107563
16 South Asian                           6138158
17 South East Asian                       170602
18 Sub-Saharan African                    827401</code></pre>
<pre class="r"><code>gwas_ancest_info %&gt;% 
  group_by(ancestry_group) %&gt;% 
  mutate(ancest_cumsum = cumsum(NUMBER_OF_INDIVDUALS)) %&gt;% 
  ggplot(aes(x=DATE,y=ancest_cumsum/(10^6), fill = ancestry_group)) + 
  #geom_area() + 
  geom_area() + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  theme_classic() + 
  labs(x = &quot;Year&quot;, y = &quot;Individuals in GWAS catalog (millons)&quot;) + 
  scale_fill_brewer(palette = &quot;Set1&quot;)</code></pre>
<pre><code>Warning: Removed 32641 rows containing non-finite outside the scale range
(`stat_align()`).</code></pre>
<pre><code>Warning in RColorBrewer::brewer.pal(n, pal): n too large, allowed maximum for palette Set1 is 9
Returning the palette you asked for with that many colors</code></pre>
<p><img src="figure/disease_inves_by_ancest.Rmd/grouped_ancest-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-grouped_ancest-1">
Past versions of grouped_ancest-1.png
</button>
</p>
<div id="fig-grouped_ancest-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/2fd5755c1e0ffdca6c453751d9ce6c164e192cf6/docs/figure/disease_inves_by_ancest.Rmd/grouped_ancest-1.png" target="_blank">2fd5755</a>
</td>
<td>
Isobel Beasley
</td>
<td>
2022-02-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>inner_join(
           gwas_study_info %&gt;% select(STUDY_ACCESSION, 
                                      `DISEASE/TRAIT`, 
                                      MAPPED_TRAIT),
           gwas_ancest_info %&gt;% select(STUDY_ACCESSION, 
                                       NUMBER_OF_INDIVDUALS,
                                       BROAD_ANCESTRAL_CATEGORY,
                                       ancestry_group)) %&gt;% 
  group_by(BROAD_ANCESTRAL_CATEGORY, 
           `DISEASE/TRAIT`) %&gt;% 
  summarise(n = sum(NUMBER_OF_INDIVDUALS)) %&gt;% 
  group_by(BROAD_ANCESTRAL_CATEGORY) %&gt;% 
  slice_max(n,n = 5)</code></pre>
<pre><code>Joining with `by = join_by(STUDY_ACCESSION)`
`summarise()` has grouped output by &#39;BROAD_ANCESTRAL_CATEGORY&#39;. You can
override using the `.groups` argument.</code></pre>
<pre><code># A tibble: 803 Ã— 3
# Groups:   BROAD_ANCESTRAL_CATEGORY [192]
   BROAD_ANCESTRAL_CATEGORY                               `DISEASE/TRAIT`      n
   &lt;chr&gt;                                                  &lt;chr&gt;            &lt;int&gt;
 1 Aboriginal Australian                                  Urinary albumiâ€¦    746
 2 Aboriginal Australian                                  Otitis media       391
 3 Aboriginal Australian                                  Type 2 diabetes    391
 4 Aboriginal Australian                                  Body mass index    361
 5 African American or Afro-Caribbean                     Cataracts       262576
 6 African American or Afro-Caribbean                     Body mass index 228346
 7 African American or Afro-Caribbean                     Type 2 diabetes 202731
 8 African American or Afro-Caribbean                     Diastolic blooâ€¦ 200474
 9 African American or Afro-Caribbean                     Systolic bloodâ€¦ 200474
10 African American or Afro-Caribbean, African unspecifiâ€¦ Type 2 diabetes 287510
# â„¹ 793 more rows</code></pre>
</div>
</div>
<div id="disease-statistics-cdc" class="section level2">
<h2>Disease statistics CDC</h2>
<pre class="r"><code>cdc_stats = data.table::fread(&quot;data/cdc/Underlying Cause of Death, 1999-2020.txt&quot;,
                              drop = c(&quot;Notes&quot;, &quot;Race Code&quot;, &quot;Cause of death Code&quot;)) %&gt;% 
             filter(!if_any(everything(), ~.x == &quot;&quot;))

cdc_stats %&gt;% 
  group_by(Race) %&gt;% 
  slice_max(Deaths,n=10)</code></pre>
<pre><code># A tibble: 40 Ã— 6
# Groups:   Race [4]
   Race      `Cause of death` Deaths Population `Crude Rate` `Age Adjusted Rate`
   &lt;chr&gt;     &lt;chr&gt;             &lt;int&gt;    &lt;int64&gt; &lt;chr&gt;        &lt;chr&gt;              
 1 Americanâ€¦ Atheroscleroticâ€¦  15927   88362592 18.0         33.9               
 2 Americanâ€¦ Bronchus or lunâ€¦  15334   88362592 17.4         28.3               
 3 Americanâ€¦ Acute myocardiaâ€¦  13755   88362592 15.6         26.7               
 4 Americanâ€¦ Chronic obstrucâ€¦  10500   88362592 11.9         22.0               
 5 Americanâ€¦ Atheroscleroticâ€¦   9090   88362592 10.3         16.7               
 6 Americanâ€¦ Unspecified diaâ€¦   8307   88362592 9.4          15.0               
 7 Americanâ€¦ Alcoholic cirrhâ€¦   8129   88362592 9.2          10.7               
 8 Americanâ€¦ Stroke, not speâ€¦   6055   88362592 6.9          13.8               
 9 Americanâ€¦ Pneumonia, unspâ€¦   5728   88362592 6.5          12.5               
10 Americanâ€¦ Unspecified demâ€¦   5092   88362592 5.8          13.9               
# â„¹ 30 more rows</code></pre>
<pre class="r"><code># wonder cdc 
# https://wonder.cdc.gov/wonder/help/QuickStart.html#
# https://wonder.cdc.gov/controller/datarequest/D76;jsessionid=2A56E973A3DF13BC5DFDD9C43725

# https://cran.r-project.org/web/packages/whomds/index.html
# whomds: Calculate Results from WHO Model Disability Survey Data</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_Australia.utf8  LC_CTYPE=English_Australia.utf8   
[3] LC_MONETARY=English_Australia.utf8 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.utf8    

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] ggplot2_3.5.2   dplyr_1.1.4     workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] bit_4.0.5          gtable_0.3.6       jsonlite_2.0.0     compiler_4.3.1    
 [5] renv_1.0.3         promises_1.3.3     tidyselect_1.2.1   Rcpp_1.0.14       
 [9] stringr_1.5.1      git2r_0.36.2       callr_3.7.6        later_1.4.2       
[13] jquerylib_0.1.4    scales_1.4.0       yaml_2.3.10        fastmap_1.2.0     
[17] R6_2.6.1           labeling_0.4.3     generics_0.1.4     knitr_1.50        
[21] tibble_3.2.1       rprojroot_2.1.0    RColorBrewer_1.1-3 bslib_0.9.0       
[25] pillar_1.10.2      rlang_1.1.6        utf8_1.2.5         cachem_1.1.0      
[29] stringi_1.8.7      httpuv_1.6.13      xfun_0.52          getPass_0.2-4     
[33] fs_1.6.6           sass_0.4.10        bit64_4.0.5        cli_3.6.5         
[37] withr_3.0.2        magrittr_2.0.3     ps_1.9.1           grid_4.3.1        
[41] digest_0.6.37      processx_3.8.6     rstudioapi_0.15.0  lifecycle_1.0.4   
[45] vctrs_0.6.5        data.table_1.17.4  evaluate_1.0.3     glue_1.8.0        
[49] farver_2.1.2       whisker_0.4.1      rmarkdown_2.29     httr_1.4.7        
[53] tools_4.3.1        pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
