<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-06" />

<title>Non-ontology based trait collapsing</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Non-ontology based trait collapsing</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree1b8a8388c191e8db32270ddeba37a4a9df334da4targetblank1b8a838a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/1b8a8388c191e8db32270ddeba37a4a9df334da4" target="_blank">1b8a838</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree1b8a8388c191e8db32270ddeba37a4a9df334da4targetblank1b8a838a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/1b8a8388c191e8db32270ddeba37a4a9df334da4" target="_blank">1b8a838</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/~$IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/~$IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/who/
    Ignored:    diseases.txt
    Ignored:    not_found_diseases.txt
    Ignored:    orig_phecode_map.csv
    Ignored:    original_phecodes_pheinfo.csv
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l1.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l2.csv
    Ignored:    output/trait_ontology/
    Ignored:    renv/
    Ignored:    sup_table.xlsx
    Ignored:    zooma.tsv
    Ignored:    zooma_res.tsv

Untracked files:
    Untracked:  analysis/garbage_icd_codes.Rmd
    Untracked:  analysis/group_cancer_diseases.Rmd
    Untracked:  analysis/map_trait_to_icd10.Rmd
    Untracked:  disease_mapping.R

Unstaged changes:
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/exclude_infectious_diseases.Rmd
    Modified:   analysis/gbd_data_plots.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_1_disease_group_non_cancer.Rmd
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   analysis/other_disease_filtering.Rmd
    Modified:   data/icd/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/disease_trait_terms_simplify.Rmd</code>) and HTML
(<code>docs/disease_trait_terms_simplify.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/1b8a8388c191e8db32270ddeba37a4a9df334da4/docs/disease_trait_terms_simplify.html" target="_blank">1b8a838</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/c36747c3a18d95d7b25d28c78cb091579df0e533/docs/disease_trait_terms_simplify.html" target="_blank">c36747c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/fd6c194734496e8347f7c396c7d404291bcde18f/analysis/disease_trait_terms_simplify.Rmd" target="_blank">fd6c194</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
More typo + structure fixing …
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/87f9ba35c09b8e6daf8b78345e581be44cad65f7/docs/disease_trait_terms_simplify.html" target="_blank">87f9ba3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/9de036aa01b0bb0fe8ce34449d0239bc15fea99d/analysis/disease_trait_terms_simplify.Rmd" target="_blank">9de036a</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
More typo + structure fixing …
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/ffeabc125c9e48f0fd1089782dd0a4803641ec11/docs/disease_trait_terms_simplify.html" target="_blank">ffeabc1</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/cd768cb85eb5c2b72f28fa55a4f040e7099fbd09/analysis/disease_trait_terms_simplify.Rmd" target="_blank">cd768cb</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
…maybe fixing typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/224e1c5fcc8cec4071cc18ca7e84da20c355201d/docs/disease_trait_terms_simplify.html" target="_blank">224e1c5</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/85452c25ba25fabe1f2b8714b3c6259852d3e144/analysis/disease_trait_terms_simplify.Rmd" target="_blank">85452c2</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Updating regex for disease trait capturing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/a2fae4e0033b89e75e0a1e48f2321c7413df03d6/docs/disease_trait_terms_simplify.html" target="_blank">a2fae4e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/6da2f7d30d747f6ab91a91f5e8dfde1baa499879/analysis/disease_trait_terms_simplify.Rmd" target="_blank">6da2f7d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Improving cancer grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/7412343b33b53b7ddf9201f6d35c9590b263f141/docs/disease_trait_terms_simplify.html" target="_blank">7412343</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/fe7efac7d18e53a9dead70f1ddbb1d5081e2ae04/docs/disease_trait_terms_simplify.html" target="_blank">fe7efac</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/cb08805f61d926acf29b6d2ec31793715df1a438/docs/disease_trait_terms_simplify.html" target="_blank">cb08805</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/096f4348c10fdce90a5046d2b45712a96de38f78/analysis/disease_trait_terms_simplify.Rmd" target="_blank">096f434</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
More disease term grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/6a82fc88a4e50bf832df39b92720ed72ec905f59/docs/disease_trait_terms_simplify.html" target="_blank">6a82fc8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/069a54b58fa0a152f869b3ce3dc89efe4eb25b4f/analysis/disease_trait_terms_simplify.Rmd" target="_blank">069a54b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Dealing with disease progression
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/51c6c29bd80320de148501aafc8b76bfb72cba3b/docs/disease_trait_terms_simplify.html" target="_blank">51c6c29</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/0597bb55e476a8ef979a7cfad4b71ae72081e56c/analysis/disease_trait_terms_simplify.Rmd" target="_blank">0597bb5</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
workflowr::wflow_publish("analysis/disease_trait_terms_simplify.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/a27c404032db9abad7c00f250200347dd5a1d3ba/docs/disease_trait_terms_simplify.html" target="_blank">a27c404</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/49e052c9130d958930bea3d1b885cf9a685df541/analysis/disease_trait_terms_simplify.Rmd" target="_blank">49e052c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/disease_trait_terms_simplify.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/527497c144f030088935ce6c15581e6d75e08626/docs/disease_trait_terms_simplify.html" target="_blank">527497c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/ed265ce98ce3dfa6ffac6ffd49a2022760d681fe/analysis/disease_trait_terms_simplify.Rmd" target="_blank">ed265ce</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/disease_trait_terms_simplify.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/27ea97b4eac3f716010451fc6e27d7fc7e036176/docs/disease_trait_terms_simplify.html" target="_blank">27ea97b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/8b77275db5463a4935c40b805bf8c62176b70d04/analysis/disease_trait_terms_simplify.Rmd" target="_blank">8b77275</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/disease_trait_terms_simplify.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/932ca3bd3012d8129b7e413827d516486aacaba0/docs/disease_trait_terms_simplify.html" target="_blank">932ca3b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/142ada8e9d38f340bad5b91b5988cffd1be28899/analysis/disease_trait_terms_simplify.Rmd" target="_blank">142ada8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/disease_trait_terms_simplify.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/813a3ad62b560350707d0b78a1bf222bf2fd272a/docs/disease_trait_terms_simplify.html" target="_blank">813a3ad</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/ac1586059bacf9effe651ad9603bc49bac06fcc2/analysis/disease_trait_terms_simplify.Rmd" target="_blank">ac15860</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/disease_trait_terms_simplify.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/0cd6e223a28799b8be69bd60d6a871feba88cf22/docs/disease_trait_terms_simplify.html" target="_blank">0cd6e22</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/174ec547abc1b9191d0796129f68e5f4043045b1/analysis/disease_trait_terms_simplify.Rmd" target="_blank">174ec54</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Fixing + improving format of initial disease trait name harmonization
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Set up</h1>
<pre class="r"><code>library(dplyr)
library(data.table)
library(stringr)</code></pre>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_cat/gwas_study_info_trait_cat.csv&quot;))</code></pre>
</div>
<div id="initial-summary" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Initial summary</h1>
<div id="objectives-of-this-analysis" class="section level2"
number="2.1">
<h2><span class="header-section-number">2.1</span> Objectives of this
analysis:</h2>
<ul>
<li>Collapsing and standardizing disease trait terms in GWAS study
metadata without relying on ontology mappings.</li>
<li>Initial harmonization of disease trait terms (fixing spelling,
collapsing studies that study different aspects of the same
disease)</li>
</ul>
</div>
</div>
<div id="number-of-studies-per-trait---before-any-grouping"
class="section level1" number="3">
<h1><span class="header-section-number">3</span> Number of studies per
trait - before any grouping</h1>
<p>Let’s look at the number of studies associated with each unique
disease/trait term before any grouping or collapsing.</p>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))

head(n_studies_trait)</code></pre>
<pre><code># A tibble: 6 × 2
  all_disease_terms         n_studies
  &lt;chr&gt;                         &lt;int&gt;
1 type 2 diabetes mellitus        144
2 major depressive disorder       108
3 alzheimer disease               104
4 asthma                           99
5 schizophrenia                    99
6 breast carcinoma                 98</code></pre>
<pre class="r"><code>dim(n_studies_trait)</code></pre>
<pre><code>[1] 3202    2</code></pre>
</div>
<div id="disease-trait-collapsing-and-cleaning" class="section level1"
number="4">
<h1><span class="header-section-number">4</span> Disease trait
collapsing and cleaning</h1>
<pre class="r"><code># Basic cleaning of disease terms
# Removes trailing commas, 
gwas_study_info$all_disease_terms = sub(&quot;,$&quot;, &quot;&quot;, gwas_study_info$all_disease_terms)

# trims whitespace
gwas_study_info$all_disease_terms = stringr::str_trim(gwas_study_info$all_disease_terms)

# remove apostrophes
gwas_study_info$all_disease_terms = stringr::str_remove_all(gwas_study_info$all_disease_terms, &quot;&#39;|’&quot;)

# initializes a new column (collected_all_disease_terms) to standardized traits
# collapse more traits 
gwas_study_info = gwas_study_info |&gt; 
  mutate(collected_all_disease_terms = all_disease_terms) </code></pre>
<div
id="where-a-study-investigates-a-given-aspect-of-of-disease-reduce-this-to-a-study-of-this-disease"
class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Where a study
investigates a given aspect of of disease, reduce this to a study of
this disease</h2>
<div id="disease-susceptibility" class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Disease
susceptibility</h3>
<div id="susceptibility-to-x-measurement-decreased-susceptibility-to"
class="section level4" number="4.1.1.1">
<h4><span class="header-section-number">4.1.1.1</span> Susceptibility to
X measurement, decreased susceptibility to)</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_replace_all(collected_all_disease_terms, 
                    &quot;susceptibility to (.*?) measurement(?=,|$)&quot;, &quot;\\1&quot;)

         ) |&gt;
    mutate(collected_all_disease_terms = 
        str_replace_all(collected_all_disease_terms, 
                    &quot;susceptibility to (.*?) (.*?) measurement(?=,|$)&quot;, &quot;\\1 \\2&quot;)

         )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_remove(collected_all_disease_terms, 
                   pattern = &quot;(?&lt;=^|, )susceptibility to &quot;
         )
  )</code></pre>
</div>
<div id="decreased-susceptibility-to-x" class="section level4"
number="4.1.1.2">
<h4><span class="header-section-number">4.1.1.2</span> Decreased
susceptibility to X</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_remove_all(collected_all_disease_terms,
                       pattern = &quot;(?&lt;=^|, )decreased susceptibility to&quot;)
  )</code></pre>
</div>
<div id="predisposition" class="section level4" number="4.1.1.3">
<h4><span class="header-section-number">4.1.1.3</span>
Predisposition</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_remove_all(collected_all_disease_terms,
                       pattern = &quot; predisposition measurement(?=,|$)&quot;)
  )</code></pre>
</div>
<div id="specific-examples-changes-to-make---bruisability"
class="section level4" number="4.1.1.4">
<h4><span class="header-section-number">4.1.1.4</span> Specific examples
/ changes to make - bruisability</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
        str_replace_all(collected_all_disease_terms, 
                    &quot;(?&lt;=^|, )bruising susceptibility(?=,|$)&quot;,
                    &quot;bruisability&quot;)
    )</code></pre>
</div>
</div>
<div id="disease-severity" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> Disease
severity</h3>
<div id="severity-symptom-severity-exacerbation" class="section level4"
number="4.1.2.1">
<h4><span class="header-section-number">4.1.2.1</span> Severity /
symptom severity / exacerbation</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
          stringr::str_remove(collected_all_disease_terms, &quot; symptom severity measurement(?=,|$)&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
          stringr::str_remove(collected_all_disease_terms, &quot; severity measurement(?=,|$)&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
          stringr::str_remove(collected_all_disease_terms, &quot; exacerbation measurement(?=,|$)&quot;)
         ) </code></pre>
</div>
<div id="specific-example-psoriasis-area-and-severity-index"
class="section level4" number="4.1.2.2">
<h4><span class="header-section-number">4.1.2.2</span> Specific example:
psoriasis area and severity index</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
          stringr::str_remove(collected_all_disease_terms, &quot;, psoriasis area and severity index(?=,|$)&quot;)
         )</code></pre>
</div>
<div id="specific-example-uveal-melanoma-disease-severity"
class="section level4" number="4.1.2.3">
<h4><span class="header-section-number">4.1.2.3</span> Specific example:
uveal melanoma disease severity</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
          stringr::str_replace_all(collected_all_disease_terms,
                           &quot;(?&lt;=^|, )uveal melanoma disease severity(?=,|$)&quot;,
                           &quot;uveal melanoma&quot;
          )
         )</code></pre>
</div>
</div>
<div id="disease-symptom-measurement" class="section level3"
number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> Disease symptom
measurement</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_remove_all(collected_all_disease_terms,
                         &quot; symptom measurement(?=,|$)&quot;)
         )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_remove_all(collected_all_disease_terms,
                         &quot; symptoms measurement(?=,|$)&quot;)
         )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_remove_all(collected_all_disease_terms,
                         &quot; symptom count(?=,|$)&quot;)
         )</code></pre>
</div>
<div id="general-measurement" class="section level3" number="4.1.4">
<h3><span class="header-section-number">4.1.4</span> General
measurement</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_replace_all(collected_all_disease_terms, 
                    &quot;(.*?) (.*?) measurement(?=,|$) &quot;, &quot;\\1 \\2&quot;)
  )</code></pre>
<div id="dependence-addiction-withdrawal-measurement"
class="section level4" number="4.1.4.1">
<h4><span class="header-section-number">4.1.4.1</span> Dependence /
addiction / withdrawal measurement</h4>
<pre class="r"><code># dependence measurement -&gt; dependence
gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\bdependence\\s+measurement(?=,|$)&quot;,
                           &quot;dependence&quot;)) 

# addiction measurement -&gt; addiction
gwas_study_info = gwas_study_info |&gt; 
  mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\baddiction\\s+measurement(?=,|$)&quot;,
                           &quot;addiction&quot;))

# withdrawal measurement -&gt; withdrawal
gwas_study_info = gwas_study_info |&gt; 
  mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\bwithdrawal\\s+measurement(?=,|$)&quot;,
                           &quot;withdrawal&quot;))</code></pre>
</div>
<div id="syndrome-disorder-measurement" class="section level4"
number="4.1.4.2">
<h4><span class="header-section-number">4.1.4.2</span> Syndrome /
disorder measurement</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\bsyndrome\\s+measurement(?=,|$)&quot;,
                           &quot;syndrome&quot;))

gwas_study_info = gwas_study_info |&gt; 
      mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\bdisorder\\s+measurement(?=,|$)&quot;,
                           &quot;disorder&quot;)) </code></pre>
</div>
<div id="allergy-measurement" class="section level4" number="4.1.4.3">
<h4><span class="header-section-number">4.1.4.3</span> Allergy
measurement</h4>
<pre class="r"><code># allergy measurement -&gt; allergy
gwas_study_info = gwas_study_info |&gt; 
  mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\ballergy\\s+measurement(?=,|$)&quot;,
                           &quot;allergy&quot;)) </code></pre>
</div>
</div>
<div id="symptom-measurement" class="section level3" number="4.1.5">
<h3><span class="header-section-number">4.1.5</span> Symptom
measurement</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_remove_all(collected_all_disease_terms,
                         &quot; symptom measurement(?=,|$)&quot;)
         )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_remove_all(collected_all_disease_terms,
                         &quot; symptoms measurement(?=,|$)&quot;)
         )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_remove_all(collected_all_disease_terms,
                         &quot; symptom count(?=,|$)&quot;)
         )</code></pre>
<div id="specific-example-autism" class="section level4"
number="4.1.5.1">
<h4><span class="header-section-number">4.1.5.1</span> Specific example:
autism</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;(?&lt;=^|, )autism spectrum disorder symptom(?=,|$)&quot;,
                            &quot;autism&quot;)
         ) </code></pre>
</div>
<div id="specific-example-respiratory-symptom-change"
class="section level4" number="4.1.5.2">
<h4><span class="header-section-number">4.1.5.2</span> Specific example:
respiratory symptom change</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_remove_all(collected_all_disease_terms,
                          &quot;, respiratory symptom change measurement(?=,|$)&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;respiratory symptom change(?=,|$)&quot;,
                          &quot;respiratory symptom&quot; 
         ))</code></pre>
</div>
</div>
</div>
<div id="age-of-onset-of" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Age of onset of</h2>
<div id="non-specific-correction" class="section level4"
number="4.2.0.1">
<h4><span class="header-section-number">4.2.0.1</span> Non-specific
correction</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_remove_all(collected_all_disease_terms,
                              pattern = &quot;(?&lt;=^|, )age of onset of &quot;)
         )</code></pre>
</div>
<div id="specific-correction-febrile-seizure" class="section level4"
number="4.2.0.2">
<h4><span class="header-section-number">4.2.0.2</span> Specific
correction: febrile seizure</h4>
<pre class="r"><code># febrile seizure (within the age range of 3 months to 6 years)
gwas_study_info = gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms ,
                                  pattern = &quot;febrile seizure \\(within the age range of 3 months to 6 years\\)&quot;,
                                   &quot;febrile seizure&quot;
                          )  
        )</code></pre>
</div>
</div>
<div id="family-history-of" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Family history
of</h2>
<div id="non-specific-example" class="section level4" number="4.3.0.1">
<h4><span class="header-section-number">4.3.0.1</span> Non-specific
example</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_remove_all(collected_all_disease_terms,
                         pattern = &quot;(?&lt;=^|, )family history of &quot;))</code></pre>
</div>
</div>
<div id="disease-progression" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Disease
progression</h2>
<div id="disease-progression-measurement" class="section level3"
number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Disease progression
measurement</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_remove_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )disease progression measurement(?=,|$)&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_remove_all(collected_all_disease_terms,
                          &quot;disease progression measurement&quot;
         ))</code></pre>
</div>
</div>
<div id="time-to-remission" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Time to
remission</h2>
<div id="covid" class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> COVID</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )time to remission of covid-19 symptoms(?=,|$)&quot;,
                          &quot;covid-19&quot;
         ))</code></pre>
</div>
</div>
<div id="alzheimers-disease---additional-examples"
class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Alzheimer’s disease -
additional examples</h2>
<div id="neuropathological-change" class="section level3"
number="4.6.1">
<h3><span class="header-section-number">4.6.1</span> Neuropathological
change</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )alzheimers disease neuropathologic change(?=,|$)&quot;,
                          &quot;alzheimers disease&quot;
         ))</code></pre>
</div>
<div id="biomarker-measurement" class="section level3" number="4.6.2">
<h3><span class="header-section-number">4.6.2</span> Biomarker
measurement</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )alzheimers disease biomarker measurement(?=,|$)&quot;,
                          &quot;alzheimers disease&quot;
         ))</code></pre>
</div>
</div>
<div id="cancer-aggressiveness-measurement" class="section level2"
number="4.7">
<h2><span class="header-section-number">4.7</span> Cancer aggressiveness
measurement</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_remove_all(collected_all_disease_terms,
                                  pattern = &quot;(?&lt;=^|, )cancer aggressiveness measurement(?=,|$)&quot;
                          )
 ) </code></pre>
</div>
<div id="other-specific-non-defined-measurement" class="section level2"
number="4.8">
<h2><span class="header-section-number">4.8</span> Other specific /
non-defined measurement</h2>
<div id="anxiety" class="section level3" number="4.8.1">
<h3><span class="header-section-number">4.8.1</span> Anxiety</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )anxiety measurement(?=,|$)&quot;,
                          &quot;anxiety&quot;
         ))</code></pre>
</div>
<div id="coronary-atherosclerosis" class="section level3"
number="4.8.2">
<h3><span class="header-section-number">4.8.2</span> Coronary
atherosclerosis</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )coronary atherosclerosis measurement(?=,|$)&quot;,
                          &quot;coronary atherosclerosis&quot;
         ))</code></pre>
</div>
</div>
<div id="cutaneous-psoriasis" class="section level2" number="4.9">
<h2><span class="header-section-number">4.9</span> Cutaneous
psoriasis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )cutaneous psoriasis measurement(?=,|$)&quot;,
                          &quot;cutaneous psoriasis&quot;
         ))</code></pre>
<div id="depressive-episode" class="section level3" number="4.9.1">
<h3><span class="header-section-number">4.9.1</span> Depressive
episode</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )depressive episode measurement(?=,|$)&quot;,
                          &quot;depressive episode&quot;
         ))</code></pre>
</div>
<div id="insomnia" class="section level3" number="4.9.2">
<h3><span class="header-section-number">4.9.2</span> Insomnia</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )insomnia measurement(?=,|$)&quot;,
                          &quot;insomnia&quot;
         ))</code></pre>
</div>
<div id="lewy-body-dementia" class="section level3" number="4.9.3">
<h3><span class="header-section-number">4.9.3</span> Lewy body
dementia</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;lewy body dementia measurement(?=,|$)&quot;,
                          &quot;lewy body dementia&quot;
         ))</code></pre>
</div>
<div id="sleep-apnea" class="section level3" number="4.9.4">
<h3><span class="header-section-number">4.9.4</span> Sleep apnea</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )sleep apnea measurement(?=,|$)&quot;,
                          &quot;sleep apnea&quot;
         ))

# during rem sleep
gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )sleep apnea measurement during rem sleep(?=,|$)&quot;,
                          &quot;sleep apnea&quot;
         ))

# during non-rem sleep
gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )sleep apnea measurement during non-rem sleep(?=,|$)&quot;,
                          &quot;sleep apnea&quot;
         ))</code></pre>
</div>
<div id="substance-abuse-measurement" class="section level3"
number="4.9.5">
<h3><span class="header-section-number">4.9.5</span> Substance abuse
measurement</h3>
</div>
<div id="alcohol" class="section level3" number="4.9.6">
<h3><span class="header-section-number">4.9.6</span> Alcohol</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )alcohol withdrawal delirium(?=,|$)&quot;,
                          &quot;alcohol withdrawal&quot;
         ))


alcohol_use_disorder_terms &lt;- c(&quot;(?&lt;=^|, )alcohol dependence(?=,|$)&quot;,
                             &quot;(?&lt;=^|, )alcohol withdrawal(?=,|$)&quot;,
                             &quot;(?&lt;=^|, )alcohol abuse(?=,|$)&quot;,
                             &quot;(?&lt;=^|, )alcohol use disorder(?=,|$)&quot;,
                             &quot;(?&lt;=^|, )addictive alcohol use(?=,|$)&quot;
                             )

gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            paste0(alcohol_use_disorder_terms, collapse = &quot;|&quot;),
                            &quot;alcohol-related disorders&quot;
           ))</code></pre>
</div>
<div id="cocaine" class="section level3" number="4.9.7">
<h3><span class="header-section-number">4.9.7</span> Cocaine</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;cocaine use disorder(?=,|$)|cocaine dependence(?=,|$)&quot;,
                          &quot;cocaine-related disorders&quot;
         ))</code></pre>
</div>
<div id="nicotine" class="section level3" number="4.9.8">
<h3><span class="header-section-number">4.9.8</span> Nicotine</h3>
<pre class="r"><code>nicotine_use_disorder_terms &lt;- c(&quot;(?&lt;=^|, )nicotine dependence(?=,|$)&quot;,
                             &quot;(?&lt;=^|, )nicotine withdrawal(?=,|$)&quot;,
                             &quot;(?&lt;=^|, )nicotine addiction(?=,|$)&quot;
                             )

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          paste0(nicotine_use_disorder_terms, collapse = &quot;|&quot;),
                          &quot;nicotine-related disorders&quot;
         ))</code></pre>
</div>
</div>
<div id="methamphetamine-use-disorders" class="section level2"
number="4.10">
<h2><span class="header-section-number">4.10</span> Methamphetamine use
disorders</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )methamphetamine dependence(?=,|$)|(?&lt;=^|, )methamphetamine-induced psychosis(?=,|$)&quot;,
                          &quot;methamphetamine use disorders&quot;
         ))</code></pre>
</div>
<div id="deal-with-basic-synonyms" class="section level2" number="4.11">
<h2><span class="header-section-number">4.11</span> Deal with basic
synonyms</h2>
<div id="adhd-attention-deficit-hyperactivity-disorder"
class="section level3" number="4.11.1">
<h3><span class="header-section-number">4.11.1</span> ADHD = attention
deficit hyperactivity disorder</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;(?&lt;=^|, )attention deficit hyperactivity disorder(?=,|$)&quot;,
                            &quot;adhd&quot;
         ))</code></pre>
</div>
<div id="anemia-phenotype---anemia" class="section level3"
number="4.11.2">
<h3><span class="header-section-number">4.11.2</span> Anemia (phenotype)
-&gt; anemia</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms =
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )anemia \\(phenotype\\)(?=,|$)&quot;,
                          &quot;anemia&quot;
         ))</code></pre>
</div>
<div id="autism-autism-spectrum-disorder-asperger-syndrome"
class="section level3" number="4.11.3">
<h3><span class="header-section-number">4.11.3</span> Autism = Autism
Spectrum Disorder = Asperger Syndrome</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;(?&lt;=^|, )autism spectrum disorder(?=,|$)&quot;,
                         &quot;autism&quot;)
        ) |&gt;
    mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;(?&lt;=^|, )asperger syndrome(?=,|$)&quot;,
                         &quot;autism&quot;)
        )</code></pre>
</div>
<div id="carcinoma-cancer" class="section level3" number="4.11.4">
<h3><span class="header-section-number">4.11.4</span> Carcinoma =
Cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)
         )</code></pre>
</div>
<div id="coronary-artery-disease-coronary-atherosclerosis"
class="section level3" number="4.11.5">
<h3><span class="header-section-number">4.11.5</span> Coronary artery
disease = coronary atherosclerosis</h3>
<p><a
href="https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001645"
class="uri">https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001645</a></p>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )coronary atherosclerosis(?=,|$)&quot;,
                          &quot;coronary artery disease&quot;
         ))</code></pre>
</div>
<div id="head-and-neck-cancer-head-and-neck-malignant-neoplasia"
class="section level3" number="4.11.6">
<h3><span class="header-section-number">4.11.6</span> Head and neck
cancer = head and neck malignant neoplasia</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )head and neck malignant neoplasia(?=,|$)&quot;,
                          &quot;head and neck cancer&quot;
         ))</code></pre>
</div>
<div id="hepatitis-virus-infection-hepatitis-infection"
class="section level3" number="4.11.7">
<h3><span class="header-section-number">4.11.7</span> Hepatitis virus
infection = hepatitis infection</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )hepatitis a virus infection(?=,|$)&quot;,
                          &quot;hepatitis a infection&quot;
         )) |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )hepatitis b virus infection(?=,|$)&quot;,
                          &quot;hepatitis b infection&quot;
         )) |&gt;
        mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )hepatitis c virus infection(?=,|$)&quot;,
                          &quot;hepatitis c infection&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )chronic hepatitis c virus infection(?=,|$)&quot;,
                          &quot;chronic hepatitis, hepatitis c infection&quot;
         )) |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )chronic hepatitis b virus infection(?=,|$)&quot;,
                          &quot;chronic hepatitis, hepatitis b infection&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )hepatitis virus-related liver cancer(?=,|$)&quot;,
                          &quot;hepatitis, liver cancer&quot;
         ))</code></pre>
</div>
<div id="narcolepsy-narcolepsy-without-cataplexy" class="section level3"
number="4.11.8">
<h3><span class="header-section-number">4.11.8</span> Narcolepsy =
Narcolepsy without cataplexy</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )narcolepsy without cataplexy(?=,|$)&quot;,
                          &quot;narcolepsy&quot;
         ))</code></pre>
</div>
<div id="mumps-virus-infection-mumps" class="section level3"
number="4.11.9">
<h3><span class="header-section-number">4.11.9</span> Mumps virus
infection = mumps</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )mumps virus infectious disease(?=,|$)&quot;,
                          &quot;mumps&quot;
         ))</code></pre>
</div>
<div id="renal-cancer-kidney-cancer" class="section level3"
number="4.11.10">
<h3><span class="header-section-number">4.11.10</span> Renal cancer =
kidney cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )renal cancer(?=,|$)&quot;,
                          &quot;kidney cancer&quot;
         ))</code></pre>
</div>
<div id="rubella-infect-rubella" class="section level3"
number="4.11.11">
<h3><span class="header-section-number">4.11.11</span> Rubella infect =
rubella</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )rubella infection(?=,|$)&quot;,
                          &quot;rubella&quot;
         ))</code></pre>
</div>
<div id="type-1-diabetes-type-1-diabetes-mellitus"
class="section level3" number="4.11.12">
<h3><span class="header-section-number">4.11.12</span> Type 1 diabetes =
type 1 diabetes mellitus</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;(?&lt;=^|, )type 1 diabetes(?=,|$)&quot;,
                         &quot;type 1 diabetes mellitus&quot;)
        )</code></pre>
</div>
<div
id="type-1-diabetes-mellitus-nephropathy-type-1-diabetes-nephropathy"
class="section level3" number="4.11.13">
<h3><span class="header-section-number">4.11.13</span> Type 1 diabetes
mellitus nephropathy = type 1 diabetes nephropathy</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;(?&lt;=^|, )type 1 diabetes mellitus nephropathy(?=,|$)&quot;,
                         &quot;type 1 diabetes nephropathy&quot;)
        )</code></pre>
</div>
<div
id="von-willebrand-disease-hereditary-or-acquired-von-willebrand-disease"
class="section level3" number="4.11.14">
<h3><span class="header-section-number">4.11.14</span> Von willebrand
disease (hereditary or acquired) = von willebrand disease</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms =
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )von willebrand disease \\(hereditary or acquired\\)(?=,|$)&quot;,
                          &quot;von willebrand disease&quot;
         ))</code></pre>
</div>
</div>
<div id="typos" class="section level2" number="4.12">
<h2><span class="header-section-number">4.12</span> Typos</h2>
<div id="alzheimer---alzheimers" class="section level3" number="4.12.1">
<h3><span class="header-section-number">4.12.1</span> Alzheimer -&gt;
alzheimers</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms =
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )alzheimer disease(?=,|$)&quot;,
                          &quot;alzheimers disease&quot;
         ))</code></pre>
</div>
<div id="parkinson---parkinsons" class="section level3" number="4.12.2">
<h3><span class="header-section-number">4.12.2</span> Parkinson -&gt;
parkinsons</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms =
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;(?&lt;=^|, )parkinson disease(?=,|$)&quot;,
                          &quot;parkinsons disease&quot;
         ))</code></pre>
</div>
</div>
</div>
<div id="putting-it-all-together" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Putting it all
together</h1>
<div id="final-summary---number-of-unique-diseases-studied"
class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Final summary -
number of unique diseases studied</h2>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$collected_all_disease_terms[gwas_study_info$collected_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()

length(unique(diseases))</code></pre>
<pre><code>[1] 2187</code></pre>
</div>
<div id="frequency-of-unique-disease-terms-after-harmonization"
class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Frequency of unique
disease terms after harmonization</h2>
<pre class="r"><code># make frequency table
freq &lt;- table(as.factor(diseases))

# sort in decreasing order
freq_sorted &lt;- sort(freq, decreasing = TRUE)

# show top N, e.g. top 10
print(&quot;Top 5 most frequent disease terms after harmonization:&quot;)</code></pre>
<pre><code>[1] &quot;Top 5 most frequent disease terms after harmonization:&quot;</code></pre>
<pre class="r"><code>head(freq_sorted, 5)</code></pre>
<pre><code>
   chronic kidney disease              hypertension  type 2 diabetes mellitus 
                    10828                      6988                       922 
major depressive disorder   coronary artery disease 
                      471                       456 </code></pre>
<pre class="r"><code>print(&quot;Number of unique disease terms after harmonization:&quot;)</code></pre>
<pre><code>[1] &quot;Number of unique disease terms after harmonization:&quot;</code></pre>
<pre class="r"><code>sum(freq[freq == 1])</code></pre>
<pre><code>[1] 409</code></pre>
<pre class="r"><code>print(&quot;Average number of studies per disease term after harmonization:&quot;)</code></pre>
<pre><code>[1] &quot;Average number of studies per disease term after harmonization:&quot;</code></pre>
<pre class="r"><code>mean(freq)</code></pre>
<pre><code>[1] 20.7316</code></pre>
</div>
<div
id="save-the-updated-gwas_study_info-with-repaired-harmonized-disease-terms"
class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Save the updated
gwas_study_info with repaired / harmonized disease terms</h2>
<pre class="r"><code>data.table::fwrite(
  gwas_study_info,
   here::here(&quot;output/gwas_cat/gwas_study_info_disease_trait_simplified.csv&quot;)
  )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] stringr_1.5.1     data.table_1.17.8 dplyr_1.1.4       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] jsonlite_2.0.0    compiler_4.3.1    renv_1.0.3        promises_1.3.3   
 [5] tidyselect_1.2.1  Rcpp_1.1.0        git2r_0.36.2      callr_3.7.6      
 [9] later_1.4.2       jquerylib_0.1.4   yaml_2.3.10       fastmap_1.2.0    
[13] here_1.0.1        R6_2.6.1          generics_0.1.4    knitr_1.50       
[17] tibble_3.3.0      rprojroot_2.1.0   bslib_0.9.0       pillar_1.11.0    
[21] rlang_1.1.6       utf8_1.2.6        cachem_1.1.0      stringi_1.8.7    
[25] httpuv_1.6.16     xfun_0.52         getPass_0.2-4     fs_1.6.6         
[29] sass_0.4.10       cli_3.6.5         withr_3.0.2       magrittr_2.0.3   
[33] ps_1.9.1          digest_0.6.37     processx_3.8.6    rstudioapi_0.17.1
[37] lifecycle_1.0.4   vctrs_0.6.5       evaluate_1.0.4    glue_1.8.0       
[41] whisker_0.4.1     rmarkdown_2.29    httr_1.4.7        tools_4.3.1      
[45] pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
