<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-10-22" />

<title>Get full article text for GWAS Catalog studies</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Get full article text for GWAS Catalog
studies</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-10-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreeb43e9a99e0280f45d629a78b0ab176516cadc479targetblankb43e9a9a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/b43e9a99e0280f45d629a78b0ab176516cadc479" target="_blank">b43e9a9</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreeb43e9a99e0280f45d629a78b0ab176516cadc479targetblankb43e9a9a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/b43e9a99e0280f45d629a78b0ab176516cadc479" target="_blank">b43e9a9</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    .venv/
    Ignored:    analysis/.DS_Store
    Ignored:    ancestry_dispar_env/
    Ignored:    data/.DS_Store
    Ignored:    data/RCDCFundingSummary_01042026.xlsx
    Ignored:    data/cdc/
    Ignored:    data/cohort/
    Ignored:    data/epmc/
    Ignored:    data/europe_pmc/
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/IHME-GBD_2023_DATA-73cc01fd-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/2025AA/
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/hp_umls_mapping.csv
    Ignored:    data/icd/lancet_conditions_icd10.xlsx
    Ignored:    data/icd/manual_disease_icd10_mappings.xlsx
    Ignored:    data/icd/mondo_umls_mapping.csv
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/phecode_to_icd10_manual_mapping.xlsx
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/semiautomatic_ICD-pheno_UKB_subset.txt
    Ignored:    data/icd/umls-2025AA-mrconso.zip
    Ignored:    figures/
    Ignored:    output/.DS_Store
    Ignored:    output/abstracts/
    Ignored:    output/doccano/
    Ignored:    output/fulltexts/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_cohorts/
    Ignored:    output/icd_map/
    Ignored:    output/trait_ontology/
    Ignored:    pubmedbert-cohort-ner-model/
    Ignored:    pubmedbert-cohort-ner/
    Ignored:    r-spacyr/
    Ignored:    renv/
    Ignored:    venv/

Unstaged changes:
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/gwas_to_gbd.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/map_trait_to_icd10.Rmd
    Modified:   analysis/missing_cohort_info.Rmd
    Modified:   analysis/replication_ancestry_bias.Rmd
    Modified:   analysis/text_for_cohort_labels.Rmd
    Modified:   analysis/trait_ontology_categorization.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/get_full_text.Rmd</code>) and
HTML (<code>docs/get_full_text.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b43e9a99e0280f45d629a78b0ab176516cadc479/analysis/get_full_text.Rmd" target="_blank">b43e9a9</a>
</td>
<td>
IJbeasley
</td>
<td>
2026-01-12
</td>
<td>
Update getting full text
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/ac0d1a7c1691a458765321cbdc43867e43f9a6ee/docs/get_full_text.html" target="_blank">ac0d1a7</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/8642872a5fe5572b64be5cdd6d68045f9766ab36/docs/get_full_text.html" target="_blank">8642872</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/da4d7305c55369025b3f6e89b5faa00987a7e50a/analysis/get_full_text.Rmd" target="_blank">da4d730</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Now run on all texts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/fb5cfd9414e0621d6167b74b3f189911f4be6498/docs/get_full_text.html" target="_blank">fb5cfd9</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/8ed4c37b75a352304f5f4d5bdb5e2f43bdc3183a/analysis/get_full_text.Rmd" target="_blank">8ed4c37</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Now run on all texts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/861028395fe749895e6b71000052500c51d195fd/docs/get_full_text.html" target="_blank">8610283</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/7d504e3db75bc28e18a04e156a746e10a41d3faa/analysis/get_full_text.Rmd" target="_blank">7d504e3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
More fixing of download full text
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/16f4c191dd3ff8d584744a999026381e72bc72b6/docs/get_full_text.html" target="_blank">16f4c19</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/3df4096aa7693c669469c94ef3cb12954f01b243/analysis/get_full_text.Rmd" target="_blank">3df4096</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-27
</td>
<td>
Update + improve full text downloading - test run
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/1439951c3ae14ee8cdfba142d63d9f99b527269c/docs/get_full_text.html" target="_blank">1439951</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/481aebe82123251dcc9708ca5972905d12e966c5/analysis/get_full_text.Rmd" target="_blank">481aebe</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-24
</td>
<td>
Update code for getting full texts
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="required-packages" class="section level2">
<h2>Required packages</h2>
<pre class="r"><code>library(httr)
library(xml2)
library(stringr)
library(here)
library(dplyr)
library(data.table)</code></pre>
</div>
<div id="get-pmcids" class="section level1">
<h1>Get PMCIDs</h1>
<div id="get-pubmed-ids-from-gwas-catalog" class="section level2">
<h2>Get Pubmed ids from GWAS catalog</h2>
<pre class="r"><code># gwas_study_info &lt;- data.table::fread(here::here(&quot;output/gwas_cat/gwas_study_info_trait_group_l2.csv&quot;))

## Step 1: 
# get only relevant disease studies
# gwas_study_info &lt;- data.table::fread(here::here(&quot;output/gwas_cat/gwas_study_info_trait_group_l2.csv&quot;))
gwas_study_info &lt;- data.table::fread(here::here(&quot;output/icd_map/gwas_study_gbd_causes.csv&quot;))

gwas_study_info = gwas_study_info |&gt;
  dplyr::rename_with(~ gsub(&quot; &quot;, &quot;_&quot;, .x))

# gwas_study_info &lt;- gwas_study_info |&gt;
#   dplyr::filter(DISEASE_STUDY == TRUE)

pmids &lt;- unique(gwas_study_info$PUBMED_ID)

#pmids &lt;- sample(pmids, size = 500)

length(pmids)</code></pre>
<pre><code>[1] 993</code></pre>
</div>
<div id="convert-pubmed-ids-to-pmcids" class="section level2">
<h2>Convert Pubmed IDs to PMCIDs</h2>
<pre class="r"><code># get PMID to PMCID mapping using Europe PMC file:
convert_pmid_df &lt;- fread(here::here(&quot;data/europe_pmc/PMID_PMCID_DOI.csv&quot;))</code></pre>
<pre><code>Warning in fread(here::here(&quot;data/europe_pmc/PMID_PMCID_DOI.csv&quot;)): Found and
resolved improper quoting out-of-sample. First healed line 8637909:
&lt;&lt;17234576,PMC2742408,&quot;https://doi.org/10.1102/1470-7330.2007.0001&gt;&gt;. If the
fields are not quoted (e.g. field separator does not appear within any field),
try quote=&quot;&quot; to avoid this warning.</code></pre>
<pre class="r"><code>convert_pmid_df &lt;- convert_pmid_df |&gt;
  dplyr::rename(pmcids = PMCID
                ) |&gt;
  dplyr::mutate(pmcids = ifelse(is.na(pmcids),
                                &quot;&quot;,
                                pmcids
                                )
                )

convert_pmid_df =
  convert_pmid_df |&gt;
  select(-DOI)

convert_pmid_df &lt;-
  convert_pmid_df |&gt;
  dplyr::filter(!is.na(PMID))

converted_ids = 
  convert_pmid_df |&gt;
  filter(PMID %in% pmids)

dim(converted_ids)</code></pre>
<pre><code>[1] 993   2</code></pre>
<pre class="r"><code>length(pmids)</code></pre>
<pre><code>[1] 993</code></pre>
<pre class="r"><code>converted_ids |&gt;
  filter(pmcids == &quot;&quot;) |&gt;
  dim()</code></pre>
<pre><code>[1] 202   2</code></pre>
<pre class="r"><code># convert PMID to PMCID
# convert_pmid_to_pmcid &lt;- function(pmid_vec,
#                                   tool = &quot;myTool&quot;,
#                                   email = &quot;you@example.com&quot;,
#                                   format = &quot;json&quot;,
#                                   batch_size = 50,
#                                   sleep_time = 1) {
# 
#   base_url &lt;- &quot;https://pmc.ncbi.nlm.nih.gov/tools/idconv/api/v1/articles/&quot;
# 
#   batches &lt;- split(pmid_vec,
#                    ceiling(seq_along(pmid_vec) / batch_size))
#   #browser()
# 
#   pmcid_list = purrr::map(batches,
#              function(pmid_vec) {
# 
# 
#                ids_param &lt;- paste(pmid_vec,
#                                   collapse = &quot;,&quot;)
# 
#                query &lt;- list(ids = ids_param,
#                              idtype = &quot;pmid&quot;,
#                              tool = tool,
#                              email = email,
#                              format = format
#                              )
# 
#                resp &lt;- httr::GET(base_url,
#                                  query = query)
# 
#                httr::stop_for_status(resp)
# 
#                content_text &lt;- httr::content(resp,
#                                              as = &quot;text&quot;,
#                                              encoding = &quot;UTF-8&quot;)
# 
#                # Handle cases where records might be empty or missing pmcid
#                parsed &lt;- jsonlite::fromJSON(content_text,
#                                             flatten = TRUE)
# 
#                parsed$records[is.na(parsed$records)] = &quot;&quot;
# 
#                pmcid &lt;- parsed$records |&gt;
#                         pull(pmcid)
# 
#                Sys.sleep(sleep_time)
# 
#                return(pmcid)
#              }
#   )
# 
#  pmcid_list = unlist(pmcid_list)
#  names(pmcid_list) &lt;- pmid_vec
#  return(pmcid_list)
# }
# 
# pmcids &lt;- convert_pmid_to_pmcid(pmids)
# 
# get_pmcid_europepmc &lt;- function(pmid_vec) {
#   base_url &lt;- &quot;https://www.ebi.ac.uk/europepmc/webservices/rest/search&quot;
#   
#   purrr::map_dfr(pmid_vec, function(pmid) {
#     query &lt;- list(
#       query = paste0(&quot;ext_id:&quot;, pmid),
#       format = &quot;json&quot;
#     )
#     resp &lt;- httr::GET(base_url, query = query)
#     if (httr::status_code(resp) != 200) {
#       return(tibble(pmid = pmid, 
#                     pmcid = NA_character_))
#     }
#     
#     dat &lt;- jsonlite::fromJSON(httr::content(resp, 
#                                             as = &quot;text&quot;, 
#                                             encoding = &quot;UTF-8&quot;))
#     
#     if (length(dat$resultList$result) == 0) {
#       return(tibble(pmid = pmid, pmcid = NA_character_))
#     }
#     
#     pmcid &lt;- dat$resultList$result$pmcid
#     tibble(pmid = pmid, pmcid = pmcid)
#   })
# }
# 
# pmids_missing = names(pmcids[pmcids == &quot;&quot;])
# 
# get_pmcid_europepmc(pmids_missing) -&gt; pmcid_europepmc_df
# 
# converted_ids &lt;-
# data.frame(pmids = names(pmcids),
#            pmcids = pmcids
#            ) 
# 
# # checked using pmids to pmcs conversion from europe pmc webservice
# # can map 38367033 -&gt; PMC12560237
# converted_ids &lt;- converted_ids |&gt;
#   mutate(pmcids = ifelse(pmids == &quot;38367033&quot;, 
#                          &quot;PMC12560237&quot;, 
#                          pmcids
#                          )
#          )
# 
# data.table::fwrite(converted_ids,
#                    here::here(&quot;output/gwas_cat/gwas_pubmed_to_pmcid_mapping.csv&quot;)
#                    )
# 
# # How many missing? 
# sum(pmcids == &quot;&quot;)
# 
# pmcids &lt;- pmcids[pmcids != &quot;&quot;]</code></pre>
</div>
</div>
<div id="download-full-texts-from-european-pmc" class="section level1">
<h1>Download full texts from European PMC</h1>
<pre class="r"><code>pmcids &lt;-
converted_ids$pmcids |&gt;
  unique()

pmcids &lt;- pmcids[pmcids != &quot;&quot;]

length(pmcids)</code></pre>
<pre><code>[1] 791</code></pre>
<pre class="r"><code>print(&quot;Percentage of pmids with pmcid:&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of pmids with pmcid:&quot;</code></pre>
<pre class="r"><code>round(100 * length(pmcids) / length(pmids), digits = 2)</code></pre>
<pre><code>[1] 79.66</code></pre>
<pre class="r"><code>download_pmc_text &lt;- function(pmcid, 
                              out_dir = here::here(&quot;output/fulltexts/europe_pmc/&quot;)
                              ) {


  url_xml &lt;- paste0(&quot;https://www.ebi.ac.uk/&quot;,
                    &quot;europepmc/webservices/rest/&quot;,
                    pmcid,
                    &quot;/fullTextXML&quot;
                    )
  
  resp &lt;- GET(url_xml)
  
  # MED/20708005
  
  # ---- Fallback URL ----
  if(status_code(resp) != 200){
    
    #print(paste0(&quot;Trying alternative URL for &quot;, pmcid))
    
    url_xml &lt;- paste0(&quot;https://europepmc.org/&quot;,
                       &quot;oai.cgi?verb=GetRecord&quot;,
                       &quot;&amp;metadataPrefix=pmc&quot;,
                       &quot;&amp;identifier=oai:europepmc.org:&quot;,
                       pmcid)
    
    resp &lt;- GET(url_xml)
  
  }
  
  # ---- Fail if still bad ----
  if(status_code(resp) != 200){
    
  #message(&quot;Failed to fetch XML for &quot;, pmcid)
  
  return(NULL)
    
  }
  
  # ---- Parse XML ----
  xml_content &lt;- read_xml(
    content(resp, 
            as = &quot;text&quot;, 
            encoding = &quot;UTF-8&quot;)
  )
  
  xml_content &lt;- read_xml(content(resp,
                                  as = &quot;text&quot;,
                                  encoding = &quot;UTF-8&quot;)
                          )
  
  article_node = xml_find_first(xml_content, 
                               &quot;//*[local-name() = &#39;article&#39;]&quot;
                               )
  
   if (is.na(article_node)) {
    message(&quot;No &lt;article&gt; node found for &quot;, pmcid)
     
    return(NULL)
   }
  
    # --- Save ---
  write_xml(article_node, 
            paste0(out_dir, pmcid, &quot;.xml&quot;)
            )
  
} 


for(article in pmcids[pmcids != &quot;&quot;]){

download_pmc_text(article)

}</code></pre>
<pre class="r"><code>print(&quot;Number of downloaded full text files&quot;)</code></pre>
<pre><code>[1] &quot;Number of downloaded full text files&quot;</code></pre>
<pre class="r"><code>print(&quot;From European PMC:&quot;)</code></pre>
<pre><code>[1] &quot;From European PMC:&quot;</code></pre>
<pre class="r"><code>n_euro_pmc &lt;- length(list.files(here::here(&quot;output/fulltexts/europe_pmc/&quot;),
                  pattern = &quot;\\.xml$&quot;)
       )

print(&quot;Percentage of pmids with full text from European PMC:&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of pmids with full text from European PMC:&quot;</code></pre>
<pre class="r"><code>round(100 * n_euro_pmc / length(pmids), digits = 2)</code></pre>
<pre><code>[1] 52.77</code></pre>
<pre class="r"><code>converted_ids &lt;-
data.table::fread(here::here(&quot;output/gwas_cat/gwas_pubmed_to_pmcid_mapping.csv&quot;)
)

convert_xml_text &lt;- function(xml_content,
                             text #output text file
                             ){
  
    for(section in 1:xml_length(xml_content)){

    section_node = xml_child(xml_content, section)

    if(length(xml_path(xml_find_all(section_node,
                                    &quot;.//*[.//title and .//p]&quot;
                                    )
                       )
            ) == 0
       )
      {

      # Get section name:
      section_name = xml_text(xml_find_all(section_node, 
                                           &quot;.//title&quot;))
      
      section_name = str_squish(section_name)
      
      if(!rlang::is_empty(section_name)) {
         text = c(text, paste0(&quot;\n\n&quot;, section_name, &quot;\n&quot;))
      }

      # Get paragraphs
      para_nodes = xml_find_all(section_node, &quot;.//p&quot;)
      para_texts = xml_text(para_nodes)
      para_texts = str_squish(para_texts)
      
      
      if(!rlang::is_empty(para_texts)) {
        text = c(text, paste0(&quot;\n&quot;, 
                              para_texts, 
                              &quot;\n&quot;)
                 )
      }
      
      if(rlang::is_empty(section_name) &amp;&amp; 
         rlang::is_empty(para_texts)) {
        
        all_node_text &lt;- xml_text(section_node)
        
        if(!rlang::is_empty(all_node_text)){
        
        text = c(text, paste0(&quot;\n&quot;, 
                              all_node_text, 
                              &quot;\n&quot;)
                 )
        
        }
      }
      
      label &lt;- xml_text(xml_find_all(section_node, 
                                     &quot;.//label&quot;))
      
      href &lt;- xml_attr(xml_find_all(section_node, 
                                    &quot;.//media&quot;), 
                       &quot;href&quot;)
      
      if(!rlang::is_empty(label) &amp;&amp; !rlang::is_empty(href)) {
        
        text = c(text, paste0(&quot;\n&quot;, label, &quot;. &quot;, href, &quot;\n&quot;))
        
      }
      
      

    } else {

    for(subsection in 1:xml_length(section_node)){

    subsection_node = xml_child(section_node,
                                subsection
                                )
    
    if(length(xml_children(subsection_node)) == 0){
      
      if(xml_name(subsection_node) == &quot;title&quot;){
        
        text = c(text, 
               paste0(&quot;\n\n&quot;, 
                      xml_text(subsection_node), 
                      &quot;\n&quot;)
               )
        
      } else {
      
      text = c(text, 
               paste0(&quot;\n&quot;, 
                      xml_text(subsection_node), 
                      &quot;\n&quot;)
               )
      
      }
      
      next
      
    }
                

    # Get section name:
    subsection_name = xml_text(xml_find_all(subsection_node, 
                                            &quot;.//title&quot;)
                               )
    
    subsection_name = str_squish(subsection_name)
    
    if(!rlang::is_empty(subsection_name)) {
    # Add spaces around section titles
    text = c(text, paste0(&quot;\n\n&quot;, 
                          subsection_name, 
                          &quot;\n&quot;)
             )
    }

    # Get paragraphs
    para_nodes = xml_find_all(subsection_node, 
                              &quot;.//p&quot;)
    
    para_texts = xml_text(para_nodes)
    para_texts = str_squish(para_texts)
    
    if(!rlang::is_empty(para_texts)) {
    # Add spaces around paragraphs
    text = c(text, paste0(&quot;\n&quot;, 
                          para_texts, 
                          &quot;\n&quot;)
             )
    }
    
    if(rlang::is_empty(subsection_name) &amp;&amp;
         rlang::is_empty(para_texts)) {    
    
        all_node_text &lt;- xml_text(subsection_node)
        
        if(!rlang::is_empty(all_node_text)){
        
        text = c(text, paste0(&quot;\n&quot;, 
                              all_node_text, 
                              &quot;\n&quot;)
                 )
        
        }
      }
            
    }
}
  }
  
  return(text)
  
}

extract_app_text &lt;- function(xml_back, 
                             text){
  
  # Add separator for appendices section
  text &lt;- c(text, &quot;\n\n=== APPENDICES ===\n&quot;)
  
  #browser()
  
  for(node_id in 1:xml_length(xml_back)){
    
    node = xml_child(xml_back, 
                     node_id)
    
    #print(node)
    
    if(xml_name(node) == &quot;app-group&quot; &amp; 
       length(xml_find_all(node, &quot;.//sec&quot;)) &gt; 0) {
      
      app_node = xml_find_all(node, &quot;.//sec&quot;)
      
      text = convert_xml_text(app_node, text)
      
    } else if(xml_name(node) == &quot;ref-list&quot;) {
      
      next 
      
    } else {
      
      text = convert_xml_text(node, 
                              text)
    }
  }


download_pmc_text &lt;- function(pmcid, 
                              out_dir = here::here(&quot;output/fulltexts/europepmc&quot;)
                              ) {


  url_xml &lt;- paste0(&quot;https://www.ebi.ac.uk/europepmc/webservices/rest/&quot;,
                    pmcid,
                    &quot;/fullTextXML&quot;
                    )

  resp &lt;- GET(url_xml)

  if (status_code(resp) != 200) stop(&quot;Failed to fetch XML for &quot;, pmcid)

  xml_content &lt;- read_xml(content(resp,
                                  as = &quot;text&quot;,
                                  encoding = &quot;UTF-8&quot;)
                          )

  # Get text body xml content
  #browser()
  xml_body = xml_child(xml_content, &quot;body&quot;)
  xml_back = xml_child(xml_content, &quot;back&quot;)
  #browser()
  
  # Build text file
  # By converting xml structure into sections and subsections
  text = c()
  text = convert_xml_text(xml_body,
                          text
                          )
  
  text = c(text, &quot;\n\n&quot;)
  text = extract_app_text(xml_back,
                          text
                          )
  
    # if Nat Genet article
  if(grepl(&quot;Nature genetics&quot;, 
        xml_text(xml_find_all(xml_content, 
                              &quot;.//journal-title&quot;)),
        ignore.case = TRUE
        )
  ){
    
    # Find all figure nodes
    xml_figures &lt;- xml_find_all(xml_content,
                                     &quot;.//fig&quot;)
    
    if (length(xml_figures) != 0){
     
      text = c(text, &quot;\n\nFigures:\n&quot;)
      
    }
    
    for(nodes in 1:length(xml_figures)){
      
      figure_node = xml_figures[nodes]
      
      # Extract label:
      label &lt;- xml_text(xml_find_all(figure_node, 
                                     &quot;.//label&quot;))
      
      # Extract title
      title = xml_text(xml_find_all(figure_node, &quot;.//title&quot;))
      
      if(!rlang::is_empty(label) | !rlang::is_empty(title)){
        
        text = c(text,
                 paste0(&quot;\n&quot;, label, &quot;. &quot;, title, &quot;\n&quot;)
                 )
      }
      
      # Extract caption
      caption = xml_text(xml_find_all(figure_node, 
                                      &quot;.//caption//p&quot;))
      
      if(!rlang::is_empty(caption)){
        
        text = c(text,
                 paste0(&quot;\n&quot;, caption, &quot;\n&quot;)
                 )
      }
      
    }
    
    }
  
  
  # --- Save ---
  text_full &lt;- paste(text, collapse = &quot; &quot;)

  txt_file &lt;- file.path(out_dir,
                        paste0(pmcid, &quot;.txt&quot;)
                        )
  writeLines(text_full,
             txt_file,
             useBytes = TRUE)

  #message(&quot;✅ Cleaned text saved for &quot;, pmcid)
  invisible(text_full)
}


safe_download_pmc_text &lt;- purrr::safely(download_pmc_text)


# Download full texts for all PMCIDs
for(pmcid in pmcids){
  if(pmcid != &quot;&quot;){
    
    result &lt;- safe_download_pmc_text(pmcid)

    # if(!is.null(result$error)){
    #   message(&quot;❌ Failed to download text for &quot;, pmcid,
    #           &quot;: &quot;, result$error)
    # }
  }
}

# How many texts saved? 
length(list.files(here::here(&quot;output/fulltexts/europepmc&quot;),
                  pattern = &quot;\\.txt$&quot;)
       )</code></pre>
<div
id="for-remaining-pmcids-pmids-without-full-text-try-downloading-using-ncbi-cloud-service"
class="section level2">
<h2>For remaining PMCIDs / PMIDs without full text, try downloading
using NCBI Cloud Service</h2>
<pre class="bash"><code>
# get list of pmcids with full text - author_manuscript available
# available in XML and plain text for text mining purposes.
aws s3 cp s3://pmc-oa-opendata/author_manuscript/txt/metadata/txt/author_manuscript.filelist.txt output/fulltexts/aws_locations/  --no-sign-request 

# get list of pmcids with full text - non-commercial use
# oa_noncomm 
aws s3 cp s3://pmc-oa-opendata/oa_noncomm/txt/metadata/txt/oa_noncomm.filelist.txt output/fulltexts/aws_locations/  --no-sign-request 

# get list of pmcids with full text, commercial list
# oa_comm
aws s3 cp s3://pmc-oa-opendata/oa_comm/txt/metadata/txt/oa_comm.filelist.txt output/fulltexts/aws_locations/  --no-sign-request 

# get list of pmcids with full text, commercial list
# oa_other
aws s3 cp s3://pmc-oa-opendata/oa_other/txt/metadata/txt/oa_other.filelist.txt output/fulltexts/aws_locations/  --no-sign-request 
</code></pre>
</div>
<div id="identify-full-texts-downloaded" class="section level2">
<h2>Identify full texts downloaded</h2>
<pre class="r"><code>europeanpmc_full_texts &lt;- 
list.files(here::here(&quot;output/fulltexts/europe_pmc&quot;),
                  pattern = &quot;\\.xml&quot;
           )

# get pmcids of these files
europeanpmc_full_texts &lt;-
  gsub(&quot;\\.xml$&quot;, 
       &quot;&quot;, 
       europeanpmc_full_texts
       ) </code></pre>
</div>
<div id="get-paths-of-full-texts-could-download" class="section level2">
<h2>Get paths of full texts could download:</h2>
<pre class="r"><code>left_over_pmcids = pmcids[!pmcids %in% europeanpmc_full_texts]

print(&quot;Number of remaining pmcids without full text:&quot;)</code></pre>
<pre><code>[1] &quot;Number of remaining pmcids without full text:&quot;</code></pre>
<pre class="r"><code>length(left_over_pmcids)</code></pre>
<pre><code>[1] 267</code></pre>
<pre class="r"><code>author_manu = data.table::fread(here::here(&quot;output/fulltexts/aws_locations/author_manuscript.filelist.txt&quot;))

oa_noncomm = data.table::fread(here::here(&quot;output/fulltexts/aws_locations/oa_noncomm.filelist.txt&quot;))

oa_comm = data.table::fread(here::here(&quot;output/fulltexts/aws_locations/oa_comm.filelist.txt&quot;))

author_manu_to_get &lt;-
author_manu |&gt;
  dplyr::filter(AccessionID %in% left_over_pmcids)

print(&quot;Number of papers to download in Author Manuscripts section:&quot;)</code></pre>
<pre><code>[1] &quot;Number of papers to download in Author Manuscripts section:&quot;</code></pre>
<pre class="r"><code>nrow(author_manu_to_get)</code></pre>
<pre><code>[1] 160</code></pre>
<pre class="r"><code>oa_noncomm_to_get = 
oa_noncomm |&gt;
#  dplyr::filter(PMID %in% names(left_over_pmcids)) 
  dplyr::filter(AccessionID %in% left_over_pmcids)

oa_comm_to_get = 
oa_comm |&gt;
#  dplyr::filter(PMID %in% names(left_over_pmcids)) 
  dplyr::filter(AccessionID %in% left_over_pmcids)

print(&quot;Number of papers to download in Open Access PMC section:&quot;)</code></pre>
<pre><code>[1] &quot;Number of papers to download in Open Access PMC section:&quot;</code></pre>
<pre class="r"><code>nrow(oa_noncomm_to_get) + nrow(oa_comm_to_get)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>oa_noncomm_to_get &lt;-
  oa_noncomm_to_get |&gt;
  dplyr::filter(!c(AccessionID %in% author_manu_to_get$AccessionID))

oa_ncomm_to_get &lt;-
  oa_noncomm_to_get |&gt;
  dplyr::filter(!c(AccessionID %in% author_manu_to_get$AccessionID))

not_available = left_over_pmcids[!c(left_over_pmcids %in% 
                                          c(oa_noncomm_to_get$AccessionID, 
                                            oa_comm_to_get$AccessionID,
                                            author_manu_to_get$AccessionID)
                                          )]

print(&quot;Number of papers without full text available in NCBI Cloud Service:&quot;)</code></pre>
<pre><code>[1] &quot;Number of papers without full text available in NCBI Cloud Service:&quot;</code></pre>
<pre class="r"><code>length(not_available)</code></pre>
<pre><code>[1] 106</code></pre>
<pre class="r"><code>file_paths = 
c(oa_noncomm_to_get$Key,
  oa_comm_to_get$Key,
  author_manu_to_get$Key)

file_paths &lt;- str_replace_all(file_paths,
                              pattern = &quot;txt&quot;,
                              replacement = &quot;xml&quot;)

# percentage not available, from all papers
100 * length(not_available) / length(pmids)</code></pre>
<pre><code>[1] 10.67472</code></pre>
</div>
<div id="download-remaining-full-texts-from-ncbi-cloud-service"
class="section level2">
<h2>Download remaining full texts from NCBI Cloud Service</h2>
<pre class="r"><code>writeLines(
  file_paths,
  here::here(&quot;output/fulltexts/aws_locations/selected_paths.txt&quot;)
)

system(
  paste(
    &quot;xargs -I {} aws s3 cp&quot;,
    &quot;s3://pmc-oa-opendata/{}&quot;,
    shQuote(here::here(&quot;output/fulltexts/ncbi_cloud/&quot;)),
    &quot;--no-sign-request&quot;,
    &quot;&lt;&quot;,
    shQuote(here::here(&quot;output/fulltexts/aws_locations/selected_paths.txt&quot;))
  )
)</code></pre>
</div>
</div>
<div id="download-pdfs-using-open-access-information-from-open-alex"
class="section level1">
<h1>Download PDFs using Open Access information from Open Alex</h1>
<div id="pmids-that-couldnt-be-converted-to-pmcids"
class="section level3">
<h3>PMIDs that couldn’t be converted to PMCIDs</h3>
<pre class="r"><code># old getting dois:
entrez_info &lt;-
entrez_summary(db=&quot;pubmed&quot;, 
               id=not_convertable_pmids)

dois &lt;-
entrez_info |&gt;
  purrr::map(function(x) {
    
    x$articleids |&gt; 
      filter(idtype == &quot;doi&quot;) |&gt; 
      pull(value)
  }
)</code></pre>
<pre class="r"><code>library(openalexR)</code></pre>
<pre><code>openalexR v2.0.0 introduces breaking changes.
See NEWS.md for details.

To suppress this message, add `openalexR.message = suppressed` to your .Renviron file.</code></pre>
<pre class="r"><code>convert_pmid_df &lt;- fread(here::here(&quot;data/europe_pmc/PMID_PMCID_DOI.csv&quot;))</code></pre>
<pre><code>Warning in fread(here::here(&quot;data/europe_pmc/PMID_PMCID_DOI.csv&quot;)): Found and
resolved improper quoting out-of-sample. First healed line 8637909:
&lt;&lt;17234576,PMC2742408,&quot;https://doi.org/10.1102/1470-7330.2007.0001&gt;&gt;. If the
fields are not quoted (e.g. field separator does not appear within any field),
try quote=&quot;&quot; to avoid this warning.</code></pre>
<pre class="r"><code>not_convertable_pmids &lt;- converted_ids |&gt; 
                         filter(pmcids == &quot;&quot;) |&gt; 
                         pull(PMID)

doi_information &lt;-
convert_pmid_df |&gt;
  filter(PMID %in% not_convertable_pmids)

doi_information |&gt;
  filter(DOI == &quot;&quot;)</code></pre>
<pre><code>Empty data.table (0 rows and 3 cols): PMID,PMCID,DOI</code></pre>
<pre class="r"><code>doi_information$PMID |&gt; unique() |&gt; length()</code></pre>
<pre><code>[1] 202</code></pre>
<pre class="r"><code>length(not_convertable_pmids)</code></pre>
<pre><code>[1] 202</code></pre>
<pre class="r"><code># get open alex works for pmids
open_alex_works &lt;- oa_fetch(
  doi = unique(doi_information$DOI),
  entity = &quot;works&quot;,
  options = list(select = c(&quot;doi&quot;, 
                            &quot;open_access&quot;))
)

# no best open access location: 
open_alex_works |&gt; 
  filter(is.na(oa_url)) |&gt;
  nrow()</code></pre>
<pre><code>[1] 110</code></pre>
<pre class="r"><code># pdf link available:
open_alex_works |&gt; 
  filter(grepl(&quot;pdf&quot;, oa_url)) |&gt;
  nrow()</code></pre>
<pre><code>[1] 65</code></pre>
<pre class="r"><code>to_download_pdfs &lt;-
open_alex_works |&gt; 
  filter(grepl(&quot;.pdf&quot;, oa_url)) |&gt;
  pull(oa_url)

  writeLines(
    to_download_pdfs,
    here::here(&quot;output/fulltexts/pdfs/pdf_links_to_download.txt&quot;))</code></pre>
<pre class="bash"><code>
cd output/fulltexts/pdfs

while read -r url; do
  curl -O &quot;$url&quot;
done &lt; pdf_links_to_download.txt
</code></pre>
</div>
<div id="pmcids-not-found-in-author-manuscripts-or-open-access-sections"
class="section level3">
<h3>PMCIDs not found in Author Manuscripts or Open Access sections</h3>
<pre class="r"><code>doi_information &lt;-
convert_pmid_df |&gt;
  filter(PMCID %in% not_available)

doi_information |&gt;
  filter(DOI == &quot;&quot;)

doi_information &lt;-
  doi_information |&gt;
  filter(DOI != &quot;&quot;)

# get open alex works for pmcids
open_alex_works &lt;- oa_fetch(
  doi = doi_information$DOI,
  entity = &quot;works&quot;,
  options = list(select = c(#&quot;title&quot;,
                            &quot;doi&quot;, 
                            &quot;open_access&quot;
                            ))
)

# no best open access location: 
open_alex_works |&gt; 
  filter(is.na(oa_url)) |&gt;
  nrow()

# pdf link available:
open_alex_works |&gt; 
  filter(grepl(&quot;pdf&quot;, oa_url)) |&gt;
  nrow()

to_download_pdfs &lt;-
open_alex_works |&gt; 
  filter(grepl(&quot;.pdf&quot;, oa_url)) |&gt;
  pull(oa_url)

  writeLines(
    to_download_pdfs,
    here::here(&quot;output/fulltexts/pdfs/pdf_links_to_download_pt2.txt&quot;))</code></pre>
<pre class="bash"><code>
cd output/fulltexts/pdfs

while read -r url; do
  curl -O &quot;$url&quot;
done &lt; pdf_links_to_download_pt2.txt
</code></pre>
</div>
</div>
<div id="test-not-used---europe-pmc-author-manuscripts"
class="section level1">
<h1>Test: not used - Europe PMC Author Manuscripts</h1>
<pre class="bash"><code>
curl -s https://europepmc.org/ftp/manuscripts/ \
  | grep -o &#39;author_manuscript_txt[^&quot;]*\.filelist\.txt&#39; \
  | sort -u \
  | while read -r file; do
      curl -O &quot;https://europepmc.org/ftp/manuscripts/$file&quot;
    done
</code></pre>
<pre class="r"><code>all_file_lists &lt;- list.files(here::here(&quot;data/epmc&quot;))

author_manu_epmc &lt;- all_file_lists |&gt;
                    purrr::map(function(file_name) {
                      
                      file_path = here::here(&quot;data/epmc&quot;,
                                             file_name
                                             )
                      
                      df &lt;- fread(file_path)
                      
                      return(df)
                    }
                    ) |&gt;
                    bind_rows()

author_manu_epmc |&gt;
  filter(AccessionID %in% not_available)

author_manu_epmc |&gt;
  filter(PMID %in% not_avaliable_pmids)

author_manu_epmc |&gt;
  filter(PMID %in% pmids)

author_manu_epmc |&gt;
  filter(PMID %in% not_convertable_pmids)</code></pre>
</div>
<div id="test-not-used---download-with-ftp-service-where-avaliable"
class="section level1">
<h1>Test: not used - Download with ftp service, where avaliable</h1>
<pre class="r"><code># not_available
not_convertable_pmids &lt;- converted_ids |&gt; 
                         filter(pmcids == &quot;&quot;) |&gt; 
                         pull(PMID)

all_tgz_links = c()

for(article_id in &quot;PMC2613843&quot;){

url &lt;- paste0(&quot;https://www.ncbi.nlm.nih.gov/pmc/utils/oa/oa.fcgi?id=&quot;,
              article_id)
  
resp &lt;- GET(url)

xml_data &lt;- xml_child(content(resp), &quot;records&quot;)

tgz_link &lt;- xml_find_first(xml_data, 
                           &quot;.//link[@format=&#39;tgz&#39;]/@href&quot;)
tgz_link &lt;- xml_text(tgz_link)

if (is.na(tgz_link)) {
  
  print(&quot;No tar.gz link found.&quot;)
  
} else {
  
  all_tgz_links &lt;- append(all_tgz_links, 
                          tgz_link)
  
}
}</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.7.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] openalexR_2.0.1   data.table_1.17.8 dplyr_1.1.4       here_1.0.1       
[5] stringr_1.6.0     xml2_1.4.0        httr_1.4.7        workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] jsonlite_2.0.0      compiler_4.3.1      BiocManager_1.30.26
 [4] renv_1.0.3          promises_1.3.3      tidyselect_1.2.1   
 [7] Rcpp_1.1.0          git2r_0.36.2        callr_3.7.6        
[10] later_1.4.4         jquerylib_0.1.4     yaml_2.3.10        
[13] fastmap_1.2.0       R6_2.6.1            generics_0.1.4     
[16] curl_7.0.0          knitr_1.50          tibble_3.3.0       
[19] rprojroot_2.1.0     bslib_0.9.0         pillar_1.11.1      
[22] rlang_1.1.6         cachem_1.1.0        stringi_1.8.7      
[25] httpuv_1.6.16       xfun_0.55           getPass_0.2-4      
[28] fs_1.6.6            sass_0.4.10         cli_3.6.5          
[31] withr_3.0.2         magrittr_2.0.4      ps_1.9.1           
[34] digest_0.6.37       processx_3.8.6      rstudioapi_0.17.1  
[37] lifecycle_1.0.4     vctrs_0.6.5         evaluate_1.0.5     
[40] glue_1.8.0          whisker_0.4.1       rmarkdown_2.30     
[43] tools_4.3.1         pkgconfig_2.0.3     htmltools_0.5.8.1  </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
