<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-28" />

<title>group_non_cancer_diseases</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">group_non_cancer_diseases</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreed8f04de6b968807b53a3cf04682c98f09b2c68a7targetblankd8f04dea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/d8f04de6b968807b53a3cf04682c98f09b2c68a7" target="_blank">d8f04de</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreed8f04de6b968807b53a3cf04682c98f09b2c68a7targetblankd8f04dea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/d8f04de6b968807b53a3cf04682c98f09b2c68a7" target="_blank">d8f04de</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/~$IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/~$IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/who/
    Ignored:    diseases.txt
    Ignored:    not_found_diseases.txt
    Ignored:    orig_phecode_map.csv
    Ignored:    original_phecodes_pheinfo.csv
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l1.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l2.csv
    Ignored:    output/trait_ontology/
    Ignored:    renv/
    Ignored:    sup_table.xlsx
    Ignored:    zooma.tsv
    Ignored:    zooma_res.tsv

Untracked files:
    Untracked:  analysis/garbage_icd_codes.Rmd
    Untracked:  analysis/map_trait_to_icd10.Rmd
    Untracked:  disease_mapping.R

Unstaged changes:
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/exclude_infectious_diseases.Rmd
    Modified:   analysis/gbd_data_plots.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_1_disease_group_non_cancer.Rmd
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   analysis/trait_ontology_categorization.Rmd
    Modified:   data/icd/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/group_non_cancer_diseases.Rmd</code>) and HTML
(<code>docs/group_non_cancer_diseases.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/d8f04de6b968807b53a3cf04682c98f09b2c68a7/analysis/group_non_cancer_diseases.Rmd" target="_blank">d8f04de</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-28
</td>
<td>
Using ICD / PheCode mapping
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1">
<h1>Set up</h1>
<pre class="r"><code>library(dplyr)
library(stringr)
library(data.table)</code></pre>
<div id="ontology-help---for-getting-disease-subtypes"
class="section level2">
<h2>Ontology help - for getting disease subtypes</h2>
<pre class="r"><code>source(here::here(&quot;code/get_term_descendants.R&quot;))</code></pre>
<div id="load-data" class="section level3">
<h3>Load Data</h3>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_cat/gwas_study_info_group.csv&quot;))</code></pre>
</div>
<div id="how-many-unique-traits-are-there-now" class="section level3">
<h3>How many unique traits are there now?</h3>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;,
                               gwas_study_info$collected_all_disease_terms[gwas_study_info$collected_all_disease_terms != &quot;&quot;])  |&gt;
  unlist() |&gt;
  stringr::str_trim()

diseases &lt;- unique(diseases)

print(length(diseases))</code></pre>
<pre><code>[1] 1894</code></pre>
</div>
</div>
</div>
<div id="filtering-some-traits" class="section level1">
<h1>Filtering some traits</h1>
<div id="abnormality-of-skin-pigmentation" class="section level2">
<h2>Abnormality of skin pigmentation</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;abnormality of skin pigmentation&quot;, collected_all_disease_terms)) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code> [1] &quot;Skin colour saturation&quot;                                             
 [2] &quot;Perceived skin darkness&quot;                                            
 [3] &quot;Skin pigmentation&quot;                                                  
 [4] &quot;Non-melanoma skin cancer&quot;                                           
 [5] &quot;Skin luminance&quot;                                                     
 [6] &quot;Skin red/green component&quot;                                           
 [7] &quot;Skin yellow/blue component&quot;                                         
 [8] &quot;Skin colour - Brown (UKB data field 1717)&quot;                          
 [9] &quot;Skin colour - Dark olive (UKB data field 1717)&quot;                     
[10] &quot;Skin colour - Fair (UKB data field 1717)&quot;                           
[11] &quot;Skin colour - Light olive (UKB data field 1717)&quot;                    
[12] &quot;Skin colour - Very fair (UKB data field 1717)&quot;                      
[13] &quot;Skin colour - Dark olive (UKB data field 1717) (Gene-based burden)&quot; 
[14] &quot;Skin colour - Brown (UKB data field 1717) (Gene-based burden)&quot;      
[15] &quot;Skin colour - Fair (UKB data field 1717) (Gene-based burden)&quot;       
[16] &quot;Skin colour - Light olive (UKB data field 1717) (Gene-based burden)&quot;
[17] &quot;Skin colour - Very fair (UKB data field 1717) (Gene-based burden)&quot;  
[18] &quot;Skin pigmentation traits&quot;                                           
[19] &quot;Tatto pigmentation&quot;                                                 </code></pre>
<pre class="r"><code># as can see, most of these traits are not about true skin pigmentation abnormalities
# just about skin pigmentation in general

# if `DISEASE/TRAIT` contains &quot;UKB data field 1717&quot; (which is just a skin colour measurment field)
# then remove &quot;abnormality of skin pigmentation&quot; from collected_all_disease_terms
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;UKB data field 1717&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;abnormality of skin pigmentation&quot;
         ),
         collected_all_disease_terms
         )
  )

# also remove if `DISEASE/TRAIT` contains &quot;Tatto pigmentation&quot;
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;tatto pigmentation&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;abnormality of skin pigmentation&quot;
         ),
         collected_all_disease_terms
         )
  )

# also remove if `DISEASE/TRAIT` is any of the following:
# - Perceived skin darkness 
# - Skin red/green component
# - Skin yellow/blue component
# - Skin pigmentation traits
# - Skin pigmentation 
# - Skin luminance

remove_terms &lt;- c(&quot;Perceived skin darkness&quot;,
                  &quot;Skin red/green component&quot;,
                  &quot;Skin yellow/blue component&quot;,
                  &quot;Skin pigmentation traits&quot;,
                  &quot;Skin pigmentation&quot;,
                  &quot;Skin luminance&quot;,
                  &quot;Skin colour saturation&quot;
                  )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(`DISEASE/TRAIT` %in% remove_terms,
                stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;abnormality of skin pigmentation&quot;
         ),
         collected_all_disease_terms
         )
  )

# this leaves PUBMED_ID 23548203, which according to suplementary table 1
# is `Number of non-melanoma skin cancer (5+)` 
# so change to &quot;non-melanoma skin cancer&quot;

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(PUBMED_ID == &quot;23548203&quot;,
                stringr::str_replace_all(collected_all_disease_terms,
                          pattern = &quot;abnormality of skin pigmentation&quot;,
                          replacement = &quot;non-melanoma skin cancer&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="handedness" class="section level2">
<h2>Handedness</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;handedness&quot;, collected_all_disease_terms)) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code> [1] &quot;Handedness (non-right-handed vs right-handed)&quot;                          
 [2] &quot;Handedness (Left-handed vs. non-left-handed)&quot;                           
 [3] &quot;Handedness (left-handed vs. right-handed)&quot;                              
 [4] &quot;Relative hand skill in reading disability&quot;                              
 [5] &quot;Relative hand skill&quot;                                                    
 [6] &quot;Handedness in dyslexia&quot;                                                 
 [7] &quot;Handedness&quot;                                                             
 [8] &quot;Handedness (chirality/laterality): Right-handed (UKB data field 1707_1)&quot;
 [9] &quot;Handedness (Right-handed vs. non-right-handed)&quot;                         
[10] &quot;Ambidextrousness&quot;                                                       
[11] &quot;Ambidextrousness (ambidextrous vs right-handed)&quot;                        
[12] &quot;Left-handedness&quot;                                                        
[13] &quot;Handedness (confirmatory factor analysis Factor 36)&quot;                    </code></pre>
<pre class="r"><code># remove handedness
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;handedness&quot;
         ))

# also remove functional laterality if Handedness chirality laterality is in DISEASE/TRAIT
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;Handedness&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;functional laterality&quot;
         ),
         collected_all_disease_terms
         )
  )

# also remove functional laterality if Usual side of head for mobile phone use is in DISEASE/TRAIT
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;Usual side of head for mobile phone use&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;functional laterality&quot;
         ),
         collected_all_disease_terms
         )
  )

# also remove functional laterality if PUBMED_ID=20585627
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(PUBMED_ID == &quot;20585627&quot;,
                stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;functional laterality&quot;
         ),
         collected_all_disease_terms
         )
  ) </code></pre>
</div>
<div id="attached-earlobe" class="section level2">
<h2>Attached earlobe</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;attached earlobe&quot;, collected_all_disease_terms)) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code>[1] &quot;Common traits (Other)&quot; &quot;Earlobe attachment&quot;   </code></pre>
<pre class="r"><code># remove attached earlobe
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;attached earlobe&quot;
         ))</code></pre>
</div>
<div id="facial-wrinkling" class="section level2">
<h2>Facial wrinkling</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;facial wrinkling&quot;, collected_all_disease_terms)) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code>[1] &quot;Facial wrinkles&quot;                                
[2] &quot;Facial wrinkles (forehead)&quot;                     
[3] &quot;Facial wrinkles (frown lines)&quot;                  
[4] &quot;Facial wrinkles (crow&#39;s feet)&quot;                  
[5] &quot;Facial wrinkles (principal components analysis)&quot;
[6] &quot;Facial wrinkles (under eye)&quot;                    </code></pre>
<pre class="r"><code># remove facial wrinkling
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;facial wrinkling&quot;
         ))</code></pre>
</div>
<div
id="sneezing-in-response-to-bright-light-autosomal-dominant-compelling-helio-ophthalmic-outburst-syndrome"
class="section level2">
<h2>Sneezing in response to bright light (autosomal dominant compelling
helio-ophthalmic outburst syndrome)</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;autosomal dominant compelling helio-ophthalmic outburst syndrome&quot;, collected_all_disease_terms)) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code>[1] &quot;Photic sneeze reflex&quot;</code></pre>
<pre class="r"><code># remove autosomal dominant compelling helio-ophthalmic outburst syndrome
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_remove_all(collected_all_disease_terms,
                          pattern = &quot;autosomal dominant compelling helio-ophthalmic outburst syndrome&quot;
         ))</code></pre>
</div>
</div>
<div id="group-some-non-cancer-diseases-together"
class="section level1">
<h1>Group some non-cancer diseases together</h1>
<div id="abnormal-brain-morphology" class="section level2">
<h2>Abnormal brain morphology</h2>
<pre class="r"><code># if DISEASE/TRAIT contains Unidentified bright object on brain MRI, 
# then replace abnormal brain morphology with Other abnormal findings on diagnostic imaging of central nervous system 
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;Unidentified bright object on brain MRI&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;abnormal brain morphology&quot;),
                          &quot;other abnormal findings on diagnostic imaging of central nervous system&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="abnormality-of-gait" class="section level2">
<h2>Abnormality of gait</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                            &quot;gait imbalance&quot;,
                            &quot;decreased walking ability&quot;)
                            ),
                          &quot;abnormality of gait&quot;
         ))</code></pre>
</div>
<div id="abdominal-abscess" class="section level2">
<h2>Abdominal abscess</h2>
<pre class="r"><code>gwas_study_info |&gt;
  filter(grepl(vec_to_grep_pattern(&quot;abdominal abscess&quot;), 
               collected_all_disease_terms, perl = T)) </code></pre>
<pre><code>   DATE_ADDED_TO_CATALOG PUBMED_ID FIRST_AUTHOR       DATE JOURNAL
                  &lt;IDat&gt;     &lt;int&gt;       &lt;char&gt;     &lt;IDat&gt;  &lt;char&gt;
1:            2025-07-14  35173190    Tangden T 2022-02-16 Sci Rep
                                   LINK
                                 &lt;char&gt;
1: www.ncbi.nlm.nih.gov/pubmed/35173190
                                                                                                                                                          STUDY
                                                                                                                                                         &lt;char&gt;
1: A genome-wide association study in a large community-based cohort identifies multiple loci associated with susceptibility to bacterial and viral infections.
         DISEASE/TRAIT
                &lt;char&gt;
1: Abdominal infection
                                                  INITIAL_SAMPLE_SIZE
                                                               &lt;char&gt;
1: 34,948 European ancestry cases, 302,536 European ancestry controls
   REPLICATION_SAMPLE_SIZE      PLATFORM_[SNPS_PASSING_QC] ASSOCIATION_COUNT
                    &lt;char&gt;                          &lt;char&gt;             &lt;int&gt;
1:                         Affymetrix [16500000] (imputed)                26
                                  MAPPED_TRAIT
                                        &lt;char&gt;
1: digestive system disease, abdominal abscess
                                                             MAPPED_TRAIT_URI
                                                                       &lt;char&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000405, http://www.ebi.ac.uk/efo/EFO_1001753
   STUDY_ACCESSION        GENOTYPING_TECHNOLOGY SUBMISSION_DATE
            &lt;char&gt;                       &lt;char&gt;          &lt;lgcl&gt;
1:    GCST90103953 Genome-wide genotyping array              NA
   STATISTICAL_MODEL BACKGROUND_TRAIT MAPPED_BACKGROUND_TRAIT
              &lt;lgcl&gt;           &lt;lgcl&gt;                  &lt;char&gt;
1:                NA               NA                        
   MAPPED_BACKGROUND_TRAIT_URI COHORT FULL_SUMMARY_STATISTICS
                        &lt;char&gt; &lt;char&gt;                  &lt;char&gt;
1:                               UKBB                      no
   SUMMARY_STATS_LOCATION    GXE                               disease_terms
                   &lt;char&gt; &lt;char&gt;                                      &lt;char&gt;
1:                            no digestive system disease, abdominal abscess
   MAPPED_TRAIT_CATEGORY background_disease_terms BACKGROUND_TRAIT_CATEGORY
                  &lt;char&gt;                   &lt;char&gt;                    &lt;char&gt;
1:      Disease/Disorder                                              Other
   DISEASE_STUDY                           all_disease_terms
          &lt;lgcl&gt;                                      &lt;char&gt;
1:          TRUE digestive system disease, abdominal abscess
                   collected_all_disease_terms
                                        &lt;char&gt;
1: abdominal abscess, digestive system disease</code></pre>
<pre class="r"><code># just one study - 35173190, has abdominal abscess
# they define unusually as: D73.3, K35-37, K57, K61, K63.0, K65, K75.0, K81, K83.0
# &#39;Abdominal infections&#39;

gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(PUBMED_ID == &quot;35173190&quot;,
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;abdominal abscess&quot;),
                          &quot;abdominal infections code&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="abnormal-mammogram" class="section level2">
<h2>Abnormal mammogram</h2>
<pre class="r"><code># if DISEASE/TRAIT contains Abnormal mammogram, then change Abnormality of the breast to abnormal mammogram

gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;Abnormal findings on mammogram or breast exam&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;abnormality of the breast&quot;),
                          &quot;abnormal mammogram&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="achalasia-of-cardia" class="section level2">
<h2>Achalasia of cardia</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;achalasia&quot;),
                          &quot;achalasia of cardia&quot;
         ))</code></pre>
</div>
<div id="acute-pancreatitis" class="section level2">
<h2>Acute pancreatitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;asparaginase-induced acute pancreatitis&quot;),
                          &quot;acute pancreatitis&quot;
         ))</code></pre>
</div>
<div id="altitude-sickness" class="section level2">
<h2>Altitude sickness</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;chronic mountain sickness&quot;),
                          &quot;altitude sickness&quot;
         ))</code></pre>
</div>
<div id="alopecia" class="section level2">
<h2>Alopecia</h2>
<pre class="r"><code>alopecia_terms &lt;- c(&quot;frontal fibrosing alopecia&quot; 
                   )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(alopecia_terms),
                          &quot;alopecia&quot;
         )
  )
         
         
drug_induced_alopecia_terms &lt;- c(
                               &quot;chemotherapy-induced alopecia&quot;
                               )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(drug_induced_alopecia_terms),
                          &quot;drug-induced androgenic alopecia&quot;
         )
  )</code></pre>
</div>
<div id="alcohol-and-nicotine-codependence" class="section level2">
<h2>Alcohol and nicotine codependence</h2>
<pre class="r"><code># alcohol and nicotine codependence -&gt; alcohol dependence, nicotine dependence

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;alcohol and nicotine codependence&quot;),
                          &quot;alcohol dependence, nicotine dependence&quot;
         )) 

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;alcohol dependence&quot;),
                          &quot;alcohol-related disorders&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;nicotine dependence&quot;),
                          &quot;tobacco use disorder&quot;
         ))</code></pre>
</div>
<div id="amyloidosis" class="section level2">
<h2>Amyloidosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;al amyloidosis&quot;),
                          &quot;amyloidosis&quot;
         ))</code></pre>
</div>
<div id="anxiety-disorder" class="section level2">
<h2>Anxiety disorder</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;anxiety&quot;),
                          &quot;anxiety disorder&quot;
         )) </code></pre>
</div>
<div id="aplastic-anemia" class="section level2">
<h2>Aplastic anemia</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;severe aplastic anemia&quot;),
                          &quot;aplastic anemia&quot;
         ))</code></pre>
</div>
<div id="androgenic-alopecia" class="section level2">
<h2>Androgenic alopecia</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;androgenetic alopecia&quot;),
                          &quot;androgenic alopecia&quot;
         ))  </code></pre>
</div>
<div id="angioedema---angioneurotic-oedema" class="section level2">
<h2>angioedema -&gt; Angioneurotic oedema</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;angioedema&quot;),
                          &quot;angioneurotic oedema&quot;
         ))</code></pre>
</div>
<div id="astigmatism" class="section level2">
<h2>Astigmatism</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;corneal astigmatism&quot;),
                          &quot;astigmatism&quot;
         ))</code></pre>
</div>
<div id="asthma" class="section level2">
<h2>Asthma</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms  = 
           stringr::str_replace_all(collected_all_disease_terms,
                            vec_to_grep_pattern(c(&quot;atopic asthma&quot;, &quot;chronic obstructive asthma&quot;)),
                            &quot;asthma&quot;
           )) |&gt;
    mutate(collected_all_disease_terms  = 
           stringr::str_replace_all(collected_all_disease_terms,
                            vec_to_grep_pattern(c(&quot;childhood onset asthma&quot;,
                                                    &quot;adult onset asthma&quot;,
                                                    &quot;aspirin-induced asthma&quot;
                                                  )
                            ),
                            &quot;asthma&quot;
           ))</code></pre>
</div>
<div id="atrial-fibrillation-and-flutter" class="section level2">
<h2>Atrial fibrillation and flutter</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;atrial flutter&quot;),
                          &quot;atrial fibrillation and flutter&quot;
         ))</code></pre>
</div>
<div id="atopic-dermatitis" class="section level2">
<h2>Atopic dermatitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;recalcitrant atopic dermatitis&quot;),
                          &quot;atopic dermatitis&quot;
         ))</code></pre>
</div>
<div id="eczema" class="section level2">
<h2>Eczema</h2>
<pre class="r"><code># if DISEASE/STUDY contains eczema, change eczematoid dermatitis to eczema
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;eczema&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;eczematoid dermatitis&quot;),
                          &quot;eczema&quot;
         ),
         collected_all_disease_terms
         )
  )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;hand eczema&quot;),
                          &quot;eczema&quot;
         )
  )</code></pre>
</div>
<div id="post-operative" class="section level2">
<h2>Post-operative</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms  = 
         stringr::str_remove_all(collected_all_disease_terms,
                         pattern = &quot;^post-operative |^postoperative&quot;
         ))</code></pre>
</div>
<div id="bipolar-disorder" class="section level2">
<h2>Bipolar disorder</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;bipolar ii disorder&quot;,
                              &quot;bipolar i disorder&quot;
                              )
                            ),
                          &quot;bipolar disorder&quot;
         ))</code></pre>
</div>
<div id="bacterial-infection" class="section level2">
<h2>Bacterial infection</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;bacterial infection&quot;
                             )
                             ),
                          &quot;bacterial infection nos&quot;
         ))</code></pre>
</div>
<div id="benign-mammary-dysplasia" class="section level2">
<h2>Benign mammary dysplasia</h2>
<pre class="r"><code># if `DISEASE/TRAIT contains Benign mammary dysplasia`, then change abnormality of the breast to benign mammary dysplasia
gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;Benign mammary dysplasia&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;abnormality of the breast&quot;),
                          &quot;benign mammary dysplasia&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="blindness-and-low-vision" class="section level2">
<h2>Blindness and low vision</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;blindness&quot;,
                             &quot;progressive visual loss&quot;,
                             &quot;visual loss&quot;
                             )
                             ),
                          &quot;blindness and low vision&quot;
         ))

# Disorders of optic nerve and visual pathways
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;visual pathway disorder&quot;,
                             &quot;optic nerve disorder&quot;
                             )
                             ),
                          &quot;disorders of optic nerve and visual pathways&quot;
         ))

# visuospatial impairment
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;visuospatial impairment&quot;
                             )
                             ),
                          &quot;other and unspecified symptoms and signs involving cognitive functions and awareness&quot;
         ))</code></pre>
</div>
<div id="cafe-au-lait-spot" class="section level2">
<h2>Cafe-au-lait spot</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;cafe-au-lait spot&quot;),
                          &quot;café au lait spots&quot;
         ))</code></pre>
</div>
<div id="cancer-of-eye" class="section level2">
<h2>Cancer of eye</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;ocular melanoma&quot;)
                             ),
                          &quot;cancer of eye&quot;
         ))</code></pre>
</div>
<div id="carbuncle-and-furuncle" class="section level2">
<h2>Carbuncle and furuncle</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;carbuncle&quot;, &quot;furuncle&quot;)
                             ),
                          &quot;carbuncle and furuncle&quot;
         ))</code></pre>
</div>
<div id="cardiac-arrhythmia" class="section level2">
<h2>Cardiac arrhythmia</h2>
<pre class="r"><code>arrhythmia_terms &lt;-
c(&quot;ventricular arrhythmia&quot;,
  &quot;torsades de pointes&quot;
  )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(arrhythmia_terms),
                          &quot;cardiac arrhythmia&quot;
         ))</code></pre>
</div>
<div id="cardiomyopathy" class="section level2">
<h2>Cardiomyopathy</h2>
<pre class="r"><code>cardiomyopathy_terms &lt;- c(&quot;nonischemic cardiomyopathy&quot;)

gwas_study_info =
gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms,
                                  pattern = vec_to_grep_pattern(cardiomyopathy_terms),
                                   &quot;cardiomyopathy&quot;
                          )  
        )</code></pre>
</div>
<div id="celiac-disease" class="section level2">
<h2>Celiac disease</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;refractory celiac disease&quot;),
                                   &quot;celiac disease&quot;
                          )  
        )</code></pre>
</div>
<div id="central-and-perpheral-ertigo" class="section level2">
<h2>Central and perpheral ertigo</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;central nervous system origin vertigo&quot;),
                          &quot;central origin vertigo&quot;
         ))


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;peripheral vertigo&quot;),
                          &quot;peripheral or central vertigo&quot;
         ))</code></pre>
</div>
<div id="cerebral-atherosclerosis" class="section level2">
<h2>Cerebral atherosclerosis</h2>
<pre class="r"><code># if Brain vascular atherosclerosis in DISEASE/TRAIT, then change vascular brain injury to  cerebral atherosclerosis 

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;Brain vascular atherosclerosis&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;vascular brain injury&quot;),
                          &quot;cerebral atherosclerosis&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="cerebrovascular-disease" class="section level2">
<h2>Cerebrovascular disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
        ifelse(grepl(&quot;Vascular brain injury&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;vascular brain injury&quot;
                             )
                             ),
                          &quot;cerebrovascular disease&quot;
         ),
         collected_all_disease_terms)
  )</code></pre>
</div>
<div id="occlusion-of-cerebral-arteries" class="section level2">
<h2>Occlusion of cerebral arteries</h2>
<pre class="r"><code># if Brain vascular stenosis in DISEASE/TRAIT, then change vascular brain injury to  occlusion of cerebral arteries

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;Brain vascular stenosis&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;vascular brain injury&quot;),
                          &quot;occlusion of cerebral arteries&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="cerebral-infarction" class="section level2">
<h2>Cerebral infarction</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                             &quot;mri defined brain infarct&quot;,
                             &quot;cardioembolic stroke&quot;)
                             ),
                          &quot;cerebral infarction&quot;
         ))</code></pre>
</div>
<div id="chronic-kidney-disease" class="section level2">
<h2>Chronic kidney disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;stage 5 chronic kidney disease&quot;),
                          &quot;chronic kidney disease&quot;
         ))</code></pre>
</div>
<div id="chronic-rhinosinusitis" class="section level2">
<h2>Chronic rhinosinusitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;chronic rhinosinusitis with nasal polyps&quot;),
                          &quot;chronic rhinosinusitis&quot;
         ))</code></pre>
</div>
<div id="chronic-sinusitis" class="section level2">
<h2>Chronic sinusitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;chronic sinus infection&quot;),
                          &quot;chronic sinusitis&quot;
         ))</code></pre>
</div>
<div id="crohns-disease" class="section level2">
<h2>Crohn’s Disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;perianal crohns disease&quot;),
                          &quot;crohns disease&quot;
         ))</code></pre>
</div>
<div id="cholangitis" class="section level2">
<h2>Cholangitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;sclerosing cholangitis&quot;),
                          &quot;cholangitis&quot;
         ))</code></pre>
</div>
<div id="cluster-headache-syndrome" class="section level2">
<h2>Cluster headache syndrome</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;cluster headache&quot;),
                          &quot;cluster headache syndrome&quot;
         ))</code></pre>
</div>
<div id="circumscribed-brain-atrophy" class="section level2">
<h2>Circumscribed brain atrophy</h2>
<pre class="r"><code>brain_atropy &lt;- c(&quot;frontotemporal dementia&quot;,
                  &quot;grn-related frontotemporal lobar degeneration with tdp43 inclusions&quot;,
                  &quot;pick disease&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(brain_atropy),
                          &quot;circumscribed brain atrophy&quot;
         ))</code></pre>
</div>
<div id="contact-dermatitis" class="section level2">
<h2>Contact dermatitis</h2>
<pre class="r"><code>pattern &lt;- vec_to_grep_pattern(&quot;contact dermatitis due to nickel&quot;)


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = pattern,
                          &quot;contact dermatitis&quot;
         ))</code></pre>
</div>
<div id="congenital-heart-diseases" class="section level2">
<h2>Congenital heart disease/s</h2>
<pre class="r"><code>congenital_heart_disease_terms &lt;- c(
  &quot;heart septal defect&quot;,  
  &quot;atrial heart septal defect&quot;,
  
  &quot;congenital left-sided heart lesions&quot;,
  &quot;congenital right-sided heart lesions&quot;,
  
  &quot;congenital anomaly of the great arteries&quot;, # equiv = &quot;Congenital malformation of great arteries, unspecified&quot;
  
  # malformation of cardiac septum, 
    &quot;abnormal cardiac septum morphology&quot;,
    &quot;atrioventricular canal defect&quot;
)
  

url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0005207/descendants&quot;

congenital_heart_disease_terms &lt;- c(congenital_heart_disease_terms,
                                  get_descendants(url)
                                  ) |&gt;
  unique()</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 102
[1] &quot;\n Some example terms&quot;
[1] &quot;double outlet right ventricle with subaortic or doubly committed ventricular septal defect with pulmonary stenosis&quot;
[2] &quot;gata6-related congenital heart disease with or without pancreatic agenesis or neonatal diabetes&quot;                   
[3] &quot;double outlet right ventricle with non-committed subpulmonary ventricular septal defect&quot;                           
[4] &quot;congenitally uncorrected transposition of the great arteries with cardiac malformation&quot;                            
[5] &quot;congenitally uncorrected transposition of the great arteries with coarctation&quot;                                     </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(congenital_heart_disease_terms),
                          &quot;congenital heart disease&quot;
         )
  )


url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0024239/descendants&quot;

more_congenital_heart_disease_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 229
[1] &quot;\n Some example terms&quot;
[1] &quot;double outlet right ventricle with subaortic or doubly committed ventricular septal defect with pulmonary stenosis&quot;
[2] &quot;double outlet right ventricle with atrioventricular septal defect, pulmonary stenosis, heterotaxy&quot;                 
[3] &quot;gata6-related congenital heart disease with or without pancreatic agenesis or neonatal diabetes&quot;                   
[4] &quot;double outlet right ventricle with subaortic or doubly committed ventricular septal defect&quot;                        
[5] &quot;pulmonary valve agenesis-ventricular septal defect-persistent ductus arteriosus syndrome&quot;                          </code></pre>
<pre class="r"><code>congenital_heart_disease_terms &lt;- c(congenital_heart_disease_terms,
                                  more_congenital_heart_disease_terms
                                  ) |&gt;
  unique() |&gt;
  str_length_sort()


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(congenital_heart_disease_terms),
                          &quot;congenital anomaly of cardiovascular system&quot;
         )
  )</code></pre>
</div>
<div id="cardiac-congenital-anomalies" class="section level2">
<h2>Cardiac congenital anomalies</h2>
<pre class="r"><code>cardiac_congenital_anomalies_terms &lt;- c(
  &quot;bicuspid aortic valve&quot;
)

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(cardiac_congenital_anomalies_terms),
                          &quot;cardiac congenital anomalies&quot;
         )
  )</code></pre>
</div>
<div id="congenital-hypertrophic-pyloric-stenosis"
class="section level2">
<h2>Congenital hypertrophic pyloric stenosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;infantile hypertrophic pyloric stenosis&quot;),
                          &quot;congenital hypertrophic pyloric stenosis&quot;
         ))</code></pre>
</div>
<div id="congenital-hypothyroidism" class="section level2">
<h2>Congenital hypothyroidism</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;congenital hypothyroidism due to developmental anomaly&quot;),
                          &quot;congenital hypothyroidism&quot;
         ))</code></pre>
</div>
<div id="congestive-heart-failure-chf-nos" class="section level2">
<h2>Congestive heart failure (CHF) NOS</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                            &quot;diastolic heart failure&quot;,
                            &quot;systolic heart failure&quot;)
                            ),
                          &quot;congestive heart failure \\(chf\\) nos&quot;
         ))</code></pre>
</div>
<div id="coronary-artery-aneurysm-and-dissection"
class="section level2">
<h2>Coronary artery aneurysm and dissection</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                            &quot;coronary aneurysm&quot;,
                            &quot;coronary artery dissection&quot;)
                            ),
                          &quot;coronary artery aneurysm and dissection&quot;
         ))

# Aortic aneurysm

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                            &quot;thoracic aortic aneurysm&quot;)
                            ),
                          &quot;aortic aneurysm&quot;
         ))</code></pre>
</div>
<div id="creutzfeldt-jacob-disease" class="section level2">
<h2>Creutzfeldt jacob disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                          c(&quot;sporadic creutzfeld jacob disease&quot;,
                            &quot;creutzfeldt jacob disease&quot;,
                            &quot;creutzfeldt-jacob disease&quot;)
                          ),
                          &quot;creutzfeldt-jakob disease&quot;
         ))</code></pre>
</div>
<div id="chronic-pain" class="section level2">
<h2>Chronic pain</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                            &quot;chronic widespread pain&quot;,
                            &quot;multisite chronic pain&quot;,
                            &quot;chronic musculoskeletal pain&quot;,
                            &quot;chronic pain syndrome&quot;)
                            ),
                          &quot;chronic pain&quot;
         ))</code></pre>
</div>
<div id="cystic-fibrosis-with-intestinal-manifestations"
class="section level2">
<h2>Cystic fibrosis with intestinal manifestations</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;cystic fibrosis associated meconium ileus&quot;),
                          &quot;cystic fibrosis with intestinal manifestations&quot;
         ))</code></pre>
</div>
<div id="dental-caries" class="section level2">
<h2>Dental caries</h2>
<pre class="r"><code>dental_caries_terms &lt;- c(&quot;pit and fissure surface dental caries&quot;,
                         &quot;smooth surface dental caries&quot;,
                         &quot;primary dental caries&quot;,
                         &quot;enamel caries&quot;,
                         &quot;permanent dental caries&quot;
                         )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(dental_caries_terms),
                          &quot;dental caries&quot;
         )
)</code></pre>
</div>
<div id="dermatomyositis" class="section level2">
<h2>Dermatomyositis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;clinically amyopathic dermatomyositis&quot;),
                          &quot;dermatomyositis&quot;
         ))</code></pre>
</div>
<div id="dilated-cardiomyopathy" class="section level2">
<h2>Dilated cardiomyopathy</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;idiopathic dilated cardiomyopathy&quot;),
                          &quot;dilated cardiomyopathy&quot;
         ))</code></pre>
</div>
<div id="disorders-of-tooth-development" class="section level2">
<h2>Disorders of tooth development</h2>
<pre class="r"><code>tooth_dev_terms &lt;- c(&quot;dental enamel hypoplasia&quot;,
                     &quot;tooth agenesis&quot;,
                     &quot;molar-incisor hypomineralization&quot;
                     )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(tooth_dev_terms),
                          &quot;disorders of tooth development&quot;
         )
  )</code></pre>
</div>
<div id="depressive-episode" class="section level2">
<h2>Depressive episode</h2>
<pre class="r"><code>depress_epi &lt;- c(&quot;depressive&quot;,
                 &quot;depression&quot;,
                 &quot;depressive disorder&quot;
                 )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(depress_epi),
                          &quot;depressive episode&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(depress_epi),
                          &quot;depressive episode&quot;
         ))</code></pre>
</div>
<div id="disorders-of-purine-and-pyrimidine-metabolism"
class="section level2">
<h2>Disorders of purine and pyrimidine metabolism</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;hyperuricemia&quot;),
                          &quot;disorders of purine and pyrimidine metabolism&quot;
         ))</code></pre>
</div>
<div id="diseases-of-white-blood-cells" class="section level2">
<h2>Diseases of white blood cells</h2>
<pre class="r"><code>wbc_terms &lt;- c(&quot;leukopenia&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(wbc_terms),
                          &quot;diseases of white blood cells&quot;
         )
  )</code></pre>
</div>
<div id="disorders-of-amino-acid-metabolism" class="section level2">
<h2>Disorders of amino-acid metabolism</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;amino acid metabolism disease&quot;),
                          &quot;disorders of amino-acid metabolism&quot;
         ))</code></pre>
</div>
<div id="drug-allergy" class="section level2">
<h2>Drug allergy</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;beta-lactam allergy&quot;),
                          &quot;drug allergy&quot;
         ))</code></pre>
</div>
<div id="degeneration-of-macula-and-posterior-pole"
class="section level2">
<h2>Degeneration of macula and posterior pole</h2>
<pre class="r"><code>macular &lt;- c(&quot;macular degeneration&quot;,
             &quot;atrophic macular degeneration&quot;,
             &quot;retinal drusen&quot;
             )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(macular),
                          &quot;degeneration of macula and posterior pole&quot;
         ))</code></pre>
</div>
<div id="essential-hypertension" class="section level2">
<h2>Essential hypertension</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(
                              &quot;early onset hypertension&quot;,
                              &quot;treatment-resistant hypertension&quot;)),
                          &quot;essential hypertension&quot;
         ))</code></pre>
</div>
<div id="febrile-seizures" class="section level2">
<h2>Febrile seizures</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                                                &quot;mmr-related febrile seizures&quot;,
                                                &quot;febrile seizure&quot;)
                                              ),
                          &quot;febrile convulsions&quot;
         ))</code></pre>
</div>
<div id="food-allergy" class="section level2">
<h2>Food allergy</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                                  vec_to_grep_pattern(
                                    c(&quot;peanut allergy&quot;,
                                      &quot;milk allergy&quot;,
                                      &quot;egg allergy&quot;,
                                      &quot;wheat allergic reaction&quot;
                                     )),
                          &quot;food allergy&quot;
         ))</code></pre>
</div>
<div id="glaucoma" class="section level2">
<h2>Glaucoma</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0005041/descendants&quot;

glaucoma_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 19
[1] &quot;\n Some example terms&quot;
[1] &quot;cyp1b1-related glaucoma with or without anterior segment dysgenesis&quot;
[2] &quot;glaucoma secondary to spherophakia/ectopia lentis and megalocornea&quot; 
[3] &quot;hereditary glaucoma, primary closed-angle&quot;                          
[4] &quot;primary angle closure glaucoma&quot;                                     
[5] &quot;secondary dysgenetic glaucoma&quot;                                      </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(glaucoma_terms),
                          &quot;glaucoma&quot;
         ))</code></pre>
</div>
<div id="hordeolum-and-other-deep-inflammation-of-eyelid"
class="section level2">
<h2>Hordeolum and other deep inflammation of eyelid</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;hordeolum&quot;),
                          &quot;hordeolum and other deep inflammation of eyelid&quot;
         ))</code></pre>
</div>
<div id="graft-vs-host-disease" class="section level2">
<h2>Graft vs host disease</h2>
<pre class="r"><code>graft_vs_host_terms &lt;- c(&quot;chronic graft versus host disease&quot;,
                         &quot;chronic graft vs. host disease&quot;,
                         &quot;acute graft versus host disease&quot;,
                         &quot;acute graft vs. host disease&quot;
                         )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(graft_vs_host_terms),
                          &quot;graft versus host disease&quot;
         ))</code></pre>
</div>
<div id="gingival-and-periodontal-diseases" class="section level2">
<h2>Gingival and periodontal diseases</h2>
<pre class="r"><code>ginival_and_periodontal_terms &lt;- c(&quot;periodontal pocket&quot;,
                                   &quot;periodontal disorder&quot;,
                                   &quot;gingival disease&quot;,
                                   &quot;gingival bleeding&quot;
                               )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(ginival_and_periodontal_terms),
                          &quot;gingival and periodontal diseases&quot;
         )
  )</code></pre>
</div>
<div id="hemiplegia" class="section level2">
<h2>Hemiplegia</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;hemiparesis&quot;)),
                          &quot;hemiplegia&quot;
         ))</code></pre>
</div>
<div id="hearing-loss" class="section level2">
<h2>Hearing loss</h2>
<pre class="r"><code>gwas_study_info  = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;age-related hearing impairment&quot;),
                          &quot;presbycusis&quot;
         ))

gwas_study_info  = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                            &quot;deafness&quot;,
                            &quot;noise-induced hearing loss&quot;)
                            ),
                          &quot;hearing loss&quot;
         ))</code></pre>
</div>
<div id="hereditary-hemochromatosis" class="section level2">
<h2>Hereditary hemochromatosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;hereditary hemochromatosis type 1&quot;),
                          &quot;hereditary hemochromatosis&quot;
         ))</code></pre>
</div>
<div id="hiv-disease-resulting-in-encephalopathy"
class="section level2">
<h2>HIV disease resulting in encephalopathy</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                                               &quot;aids dementia&quot;,
                                               &quot;hiv-associated neurocognitive disorder&quot;)
                                              ),
                          &quot;hiv disease resulting in encephalopathy&quot;
         ))</code></pre>
</div>
<div id="inherited-retinal-dystrophy" class="section level2">
<h2>Inherited retinal dystrophy</h2>
<pre class="r"><code>url &lt;-  &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0019118/descendants&quot;

inherited_retinal_dystrophy_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 341
[1] &quot;\n Some example terms&quot;
[1] &quot;spondyloepiphyseal dysplasia, sensorineural hearing loss, impaired intellectual development, and leber congenital amaurosis&quot;
[2] &quot;x-linked intellectual disability-limb spasticity-retinal dystrophy-diabetes insipidus syndrome&quot;                             
[3] &quot;microcephaly with or without chorioretinopathy, lymphedema, or intellectual disability&quot;                                     
[4] &quot;retinal vasculopathy with cerebral leukoencephalopathy and systemic manifestations&quot;                                         
[5] &quot;retinal dystrophy with inner retinal dysfunction and ganglion cell anomalies&quot;                                               </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(inherited_retinal_dystrophy_terms),
                          &quot;hereditary retinal dystrophy&quot;
         ))</code></pre>
</div>
<div id="intentional-self-harm-by-unspecified-means"
class="section level2">
<h2>Intentional self-harm by unspecified means</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;self-injurious behavior&quot;,
                                                &quot;self-injurious ideation&quot;)),
                          &quot;intentional self-harm by unspecified means&quot;
         ))</code></pre>
</div>
<div id="infertility-male" class="section level2">
<h2>Infertility (male)</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
                  stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;azoospermia&quot;,
                              &quot;sertoli cell-only syndrome&quot;)
                            ),
                          &quot;male infertility&quot;
                          )  )</code></pre>
</div>
<div id="induratio-penis-plastica" class="section level2">
<h2>Induratio penis plastica</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
                  stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;peyronie disease&quot;)
                            ),
                          &quot;induratio penis plastica&quot;
                          )  )</code></pre>
</div>
<div id="intracerebral-hemorrhage" class="section level2">
<h2>Intracerebral hemorrhage</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
                  stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;non-lobar intracerebral hemorrhage&quot;,
                              &quot;lobar intracerebral hemorrhage&quot;)
                            ),
                          &quot;intracerebral hemorrhage&quot;
                          )  )</code></pre>
</div>
<div id="idiopathic-generalized-epilepsy" class="section level2">
<h2>Idiopathic generalized epilepsy</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F36803009/descendants&quot;

idiopathic_generalized_epilepsy_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 4
[1] &quot;\n Some example terms&quot;
[1] &quot;epilepsy with generalized tonic-clonic seizures alone (disorder)&quot;
[2] &quot;juvenile myoclonic epilepsy&quot;                                     
[3] &quot;childhood absence epilepsy&quot;                                      
[4] &quot;juvenile absence epilepsy&quot;                                       
[5] NA                                                                </code></pre>
<pre class="r"><code>idiopathic_generalized_epilepsy_terms = stringr::str_remove_all(
                                        idiopathic_generalized_epilepsy_terms, 
                                        &quot; \\(disorder\\)$&quot;
                                        )

idiopathic_generalized_epilepsy_terms = c(&quot;epilepsy with generalized tonic-clonic seizures&quot;,
                                          idiopathic_generalized_epilepsy_terms
                                         )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(idiopathic_generalized_epilepsy_terms),
                          &quot;generalized idiopathic epilepsy and epileptic syndromes&quot;
         ))</code></pre>
</div>
<div id="juvenile-idiopathic-arthritis" class="section level2">
<h2>Juvenile idiopathic arthritis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;rheumatoid factor-negative juvenile idiopathic arthritis&quot;),
                          &quot;juvenile idiopathic arthritis&quot;
         ))</code></pre>
</div>
<div id="keratoconjunctivitis" class="section level2">
<h2>Keratoconjunctivitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;keratoconjunctivitis sicca&quot;),
                          &quot;keratoconjunctivitis&quot;
         ))</code></pre>
</div>
<div id="learning-disorder" class="section level2">
<h2>Learning disorder</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;dyslexia&quot;,
                                                &quot;mathematics disorder&quot;,
                                                &quot;disorder of written expression&quot;
                                                )
                                              ),
                          &quot;learning disorder&quot;
         ))</code></pre>
</div>
<div id="lewy-body-dementia" class="section level2">
<h2>Lewy body dementia</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;lewy body attribute&quot;),
                          &quot;lewy body dementia&quot;
         ))</code></pre>
</div>
<div id="rheumatoid-arthritis" class="section level2">
<h2>Rheumatoid arthritis</h2>
<pre class="r"><code>ra_terms &lt;- c(&quot;acpa-positive rheumatoid arthritis&quot;,
               &quot;acpa-negative rheumatoid arthritis&quot;,
              &quot;adult-onset stills disease&quot;)

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms,
                                  pattern = vec_to_grep_pattern(ra_terms),
                                   &quot;rheumatoid arthritis&quot;
                          )  
        )</code></pre>
</div>
<div id="migraine" class="section level2">
<h2>Migraine</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;migraine with aura&quot;,
                              &quot;migraine without aura&quot;,
                              &quot;migraine disorder&quot;
                              )
                            ),
                          &quot;migraine&quot;
         ))</code></pre>
</div>
<div id="multiple-sclerosis" class="section level2">
<h2>Multiple sclerosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;relapsing-remitting multiple sclerosis&quot;),
                          &quot;multiple sclerosis&quot;
         ))</code></pre>
</div>
<div id="myasthenia-gravis" class="section level2">
<h2>Myasthenia gravis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;late-onset myasthenia gravis&quot;),
                          &quot;myasthenia gravis&quot;
         ))</code></pre>
</div>
<div id="nephrolithiasis" class="section level2">
<h2>Nephrolithiasis</h2>
<pre class="r"><code>nephro_terms &lt;- c(&quot;uric acid nephrolithiasis&quot;,
                   &quot;calcium phosphate nephrolithiasis&quot;,
                    &quot;calcium oxalate nephrolithiasis&quot;,
                   &quot;struvite nephrolithiasis&quot;)


gwas_study_info =
gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms,
                                  pattern = vec_to_grep_pattern(nephro_terms),
                                   &quot;nephrolithiasis&quot;
                          )  
        )</code></pre>
</div>
<div id="nephritis-and-nephropathy-with-pathological-lesion"
class="section level2">
<h2>Nephritis and nephropathy with pathological lesion</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;acute tubulointerstitial nephritis&quot;,
                                              &quot;iga glomerulonephritis&quot;,
                                              &quot;membranous glomerulonephritis&quot;,
                                              &quot;lupus nephritis&quot;)
                                              ),
                          &quot;nephritis and nephropathy with pathological lesion&quot;
         ))</code></pre>
</div>
<div id="neurofibromatosis" class="section level2">
<h2>Neurofibromatosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;neurofibromatosis type 1&quot;,
                              &quot;neurofibromatosis type 2&quot;)
                          ),
                          &quot;neurofibromatosis&quot;
         ))</code></pre>
</div>
<div id="neuromyelitis-optica" class="section level2">
<h2>Neuromyelitis optica</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;aquaporin-4 antibody positive neuromyelitis optica&quot;,
                              &quot;aquaporin-4 antibody negative neuromyelitis optica&quot;,
                              &quot;aqp4-igg-positive neuromyelitis optica&quot;,
                              &quot;aqp4-igg-negative neuromyelitis optica&quot;
                            )
                          ),
                          &quot;neuromyelitis optica&quot;
         ))</code></pre>
</div>
<div id="noninflammatory-disorders-of-vagina" class="section level2">
<h2>Noninflammatory disorders of vagina</h2>
<pre class="r"><code>non_inflam_terms &lt;- c(&quot;abnormal vaginal discharge itching&quot;,
                      &quot;abnormal vaginal discharge smell&quot;,
                      &quot;vaginal discharge&quot;)

gwas_study_info = 
  gwas_study_info |&gt;
   mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms,
                                  pattern = 
                                    vec_to_grep_pattern(
                                      non_inflam_terms
                                      ),
                                   &quot;noninflammatory disorders of vagina&quot;
                          )  
        )</code></pre>
</div>
<div id="obesity" class="section level2">
<h2>Obesity</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms,
                                  pattern = 
                                    vec_to_grep_pattern(
                                      c(&quot;morbid obesity&quot;,
                                         &quot;metabolically healthy obesity&quot;
                                        )
                                      ),
                                   &quot;obesity&quot;
                          )  
        )</code></pre>
</div>
<div id="obsessive-compulsive-disorder" class="section level2">
<h2>Obsessive-compulsive disorder</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c
                                              (&quot;obsessive-compulsive trait&quot;,
                                                &quot;obsessive-compulsive&quot;)
                                              ),
                          &quot;obsessive-compulsive disorder&quot;
         ))</code></pre>
</div>
<div id="osteonecrosis" class="section level2">
<h2>Osteonecrosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;idiopathic osteonecrosis of the femoral head&quot;),
                          &quot;osteonecrosis&quot;
         ))</code></pre>
</div>
<div id="other-epilepsy" class="section level2">
<h2>Other epilepsy</h2>
<pre class="r"><code>gwas_study_info =
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;mesial temporal lobe epilepsy with hippocampal sclerosis&quot;,
                              &quot;rolandic epilepsy&quot;)
                            ),
                          &quot;epilepsy&quot;
         )) </code></pre>
</div>
<div id="chromosomal-anomalies" class="section level2">
<h2>Chromosomal anomalies</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(
                            &quot;22q11.2 deletion syndrome&quot;,
                            &quot;fragile x syndrome&quot;)
                            ),
                          &quot;chromosomal anomalies&quot;
         ))</code></pre>
</div>
<div id="opiod-dependence" class="section level2">
<h2>Opiod dependence</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;heroin dependence&quot;,
                                              &quot;opioid use disorder&quot;)),
                          &quot;opioid dependence&quot;
         ))</code></pre>
</div>
<div id="other-and-unspecified-cirrhosis-of-liver"
class="section level2">
<h2>Other and unspecified cirrhosis of liver</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;hepatitis c induced liver cirrhosis&quot;,
                                                &quot;biliary liver cirrhosis&quot;)
                                              ),
                          &quot;other and unspecified cirrhosis of liver&quot;
         ))</code></pre>
</div>
<div id="other-cardiac-conduction-disorders" class="section level2">
<h2>Other cardiac conduction disorders</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;familial long qt syndrome&quot;)
                                              ),
                          &quot;other cardiac conduction disorders&quot;
         ))</code></pre>
</div>
<div id="other-cerebral-degenerations" class="section level2">
<h2>Other cerebral degenerations</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;brain atrophy&quot;
                                                )
                                              ),
                          &quot;other cerebral degenerations&quot;
         ))</code></pre>
</div>
<div id="other-specified-degenerative-diseases-of-nervous-system"
class="section level2">
<h2>Other specified degenerative diseases of nervous system</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;corticobasal degeneration disorder&quot;
                                                )
                                              ),
                          &quot;other specified degenerative diseases of nervous system&quot;
         ))

# progressive supranuclear palsy -&gt; Dementia with cerebral degenerations

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;progressive supranuclear palsy&quot;),
                          &quot;dementia with cerebral degenerations&quot;
         ))</code></pre>
</div>
<div id="other-chronic-nonalcoholic-liver-disease"
class="section level2">
<h2>Other chronic nonalcoholic liver disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;non-alcoholic fatty liver disease&quot;
                                                )),
                          &quot;other chronic nonalcoholic liver disease&quot;
         ))</code></pre>
</div>
<div id="other-disorders-of-bone-and-cartilage" class="section level2">
<h2>Other disorders of bone and cartilage</h2>
<pre class="r"><code>other_bone_cartilage_terms &lt;- c(&quot;tietze syndrome&quot;
                               )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(other_bone_cartilage_terms),
                          &quot;other disorders of bone and cartilage&quot;
         )
  )</code></pre>
</div>
<div id="other-eating-disorders" class="section level2">
<h2>Other eating disorders</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;binge eating&quot;)
                                              ),
                          &quot;other eating disorders&quot;
         ))</code></pre>
</div>
<div id="other-specified-inflammatory-liver-diseases"
class="section level2">
<h2>Other specified inflammatory liver diseases</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;primary biliary cholangitis&quot;,
                                                &quot;primary sclerosing cholangitis&quot;
                                                )
                                              ),
                          &quot;non-alcoholic steatohepatitis&quot;
         ))</code></pre>
</div>
<div id="other-haemoglobinopathies" class="section level2">
<h2>Other haemoglobinopathies</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;hemoglobin e disease&quot;),
                          &quot;other haemoglobinopathies&quot;
         ))</code></pre>
</div>
<div id="other-paralytic-syndromes" class="section level2">
<h2>Other paralytic syndromes</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;paraplegia&quot;,
                              &quot;quadriplegia&quot;)
                            ),
                          &quot;other paralytic syndromes&quot;
         ))</code></pre>
</div>
<div id="parkinsons-disease" class="section level2">
<h2>Parkinsons disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;young adult-onset parkinsonism&quot;),
                          &quot;parkinsons disease&quot;
         ))</code></pre>
</div>
<div id="perinatal-jaundice" class="section level2">
<h2>Perinatal jaundice</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;neonatal jaundice&quot;),
                          &quot;perinatal jaundice&quot;
         ))</code></pre>
</div>
<div id="periodontitis" class="section level2">
<h2>Periodontitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;aggressive periodontitis&quot;),
                          &quot;periodontitis&quot;
         ))</code></pre>
</div>
<div id="phlebitis-and-thrombophlebitis" class="section level2">
<h2>Phlebitis and thrombophlebitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;thrombophlebitis&quot;)),
                          &quot;phlebitis and thrombophlebitis&quot;
         ))</code></pre>
</div>
<div id="phobias" class="section level2">
<h2>Phobias</h2>
<pre class="r"><code># social anxiety disorder -&gt; social phobias
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;social anxiety disorder&quot;),
                          &quot;social phobias&quot;
         ))

# specific phobia -&gt; Specific \\(isolated\\) phobias

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;specific phobia&quot;),
                          &quot;specific \\(isolated\\) phobias&quot;
         ))</code></pre>
</div>
<div id="primary-ovarian-failure" class="section level2">
<h2>Primary ovarian failure</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;primary ovarian insufficiency&quot;),
                          &quot;primary ovarian failure&quot;
         ))</code></pre>
</div>
<div id="proteinuria" class="section level2">
<h2>Proteinuria</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;albuminuria&quot;,
                                                &quot;moderate albuminuria&quot;)
                                              ),
                          &quot;proteinuria&quot;
         ))</code></pre>
</div>
<div id="prurigo" class="section level2">
<h2>Prurigo</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;prurigo&quot;),
                          &quot;other prurigo&quot;
         ))</code></pre>
</div>
<div id="precocious-puberty" class="section level2">
<h2>Precocious puberty</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;central precocious puberty&quot;),
                          &quot;precocious puberty&quot;
         ))</code></pre>
</div>
<div id="psoriasis" class="section level2">
<h2>Psoriasis</h2>
<pre class="r"><code>psoriasis_terms &lt;- c(&quot;cutaneous psoriasis&quot;)


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(psoriasis_terms),
                          &quot;psoriasis&quot;)
  )</code></pre>
</div>
<div id="psychosis-pyschotic-pyschotic-symptoms" class="section level2">
<h2>Psychosis (pyschotic, pyschotic symptoms)</h2>
<pre class="r"><code>psychosis_terms &lt;- c(&quot;psychotic&quot;,
                     &quot;psychotic symptoms&quot;
                     )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(psychosis_terms),
                          &quot;psychosis&quot;)
  )</code></pre>
</div>
<div id="pulmonary-fibrosis" class="section level2">
<h2>Pulmonary fibrosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;idiopathic pulmonary fibrosis&quot;),
                          &quot;pulmonary fibrosis&quot;
         ))</code></pre>
</div>
<div id="retinoschisis-and-retinoschisis-and-retinal-cysts"
class="section level2">
<h2>Retinoschisis and retinoschisis and retinal cysts</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;retinoschisis&quot;),
                          &quot;retinoschisis and retinal cysts&quot;
         ))</code></pre>
</div>
<div id="separation-of-retinal-layers" class="section level2">
<h2>Separation of retinal layers</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;central serous retinopathy&quot;,
                                              &quot;chronic central serous retinopathy&quot;)
                                              ),
                          &quot;separation of retinal layers&quot;
         ))</code></pre>
</div>
<div id="rhinitis" class="section level2">
<h2>Rhinitis</h2>
<pre class="r"><code>rhinitis_terms &lt;- c(&quot;non-allergic rhinitis&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(rhinitis_terms),
                          &quot;rhinitis&quot;
         )
  )</code></pre>
</div>
<div id="sciatica" class="section level2">
<h2>Sciatica</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;ldh-related sciatica&quot;)
                                              ),
                          &quot;sciatica&quot;
         )) </code></pre>
</div>
<div id="schizoaffective-disorder" class="section level2">
<h2>Schizoaffective disorder</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;schizoaffective disorder-bipolar type&quot;),
                          &quot;schizoaffective disorder&quot;)
         )</code></pre>
</div>
<div id="schizophrenia" class="section level2">
<h2>Schizophrenia</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;treatment refractory schizophrenia&quot;),
                          &quot;schizophrenia&quot;)
         )</code></pre>
</div>
<div id="scoliosis" class="section level2">
<h2>Scoliosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;adolescent idiopathic scoliosis&quot;),
                          &quot;scoliosis&quot;)
         )</code></pre>
</div>
<div id="sickle-cell-disease-and-related-diseases"
class="section level2">
<h2>Sickle cell disease and related diseases</h2>
<pre class="r"><code># if DISEASE/TRAIT contains sickle cell disease, change sickle cell disease and related diseases to sickle cell anemia
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(grepl(&quot;sickle cell disease&quot;, 
                     `DISEASE/TRAIT`,
                     ignore.case = TRUE),
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;sickle cell disease and related diseases&quot;),
                          &quot;sickle cell anemia&quot;
         ),
         collected_all_disease_terms
         )
  )</code></pre>
</div>
<div id="sleep-apnea" class="section level2">
<h2>Sleep apnea</h2>
<pre class="r"><code>sleep_apnea_terms &lt;- c(&quot;sleep apnea during non-rem sleep&quot;,
                      &quot;sleep apnea during rem sleep&quot;,
                      &quot;obstructive sleep apnea&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(sleep_apnea_terms),
                          &quot;sleep apnea&quot;)
  )</code></pre>
</div>
<div id="sleep-disorders" class="section level2">
<h2>Sleep disorders</h2>
<pre class="r"><code>sleep_disorder_terms &lt;- c(&quot;sleepiness&quot;,
                          &quot;somnambulism&quot;,
                          &quot;rem sleep behavior disorder&quot;
                         )


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(sleep_disorder_terms),
                          &quot;sleep disorders&quot;
         )
  )</code></pre>
</div>
<div id="staphylococcus-infections" class="section level2">
<h2>Staphylococcus infections</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(
                            c(&quot;staphylococcus aureus infection&quot;,
                              &quot;skin and soft tissue staphylococcus aureus infection&quot;,
                              &quot;methicillin-resistant staphylococcus aureus infection&quot;)),
                          &quot;staphylococcus infections&quot;
         ))</code></pre>
</div>
<div id="strabismus" class="section level2">
<h2>Strabismus</h2>
<pre class="r"><code>url &lt;-  &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_540/descendants&quot;

strabismus_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 26
[1] &quot;\n Some example terms&quot;
[1] &quot;abnormal retinal correspondence&quot; &quot;brown&#39;s tendon sheath syndrome&quot; 
[3] &quot;internuclear ophthalmoplegia&quot;    &quot;duane retraction syndrome 3&quot;    
[5] &quot;duane retraction syndrome 2&quot;    </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(strabismus_terms),
                          &quot;strabismus&quot;
         ))</code></pre>
</div>
<div id="stroke" class="section level2">
<h2>Stroke</h2>
<pre class="r"><code>other_nonspec_stroke &lt;- c(&quot;large artery stroke&quot;,
                         &quot;small vessel stroke&quot;,
                         &quot;stroke outcome&quot;,
                         &quot;stroke disorder&quot;
                         )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(other_nonspec_stroke),
                          &quot;stroke&quot;
         ))</code></pre>
</div>
<div id="type-1-diabetes" class="section level2">
<h2>Type 1 diabetes</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;latent autoimmune diabetes in adults&quot;),
                          &quot;type 1 diabetes mellitus&quot;
         ))</code></pre>
</div>
<div id="type-2-diabetes-with-a-ophthalmic-manifestations"
class="section level2">
<h2>Type 2 diabetes with a ophthalmic manifestations</h2>
<pre class="r"><code>type_2_eye_terms &lt;- c(&quot;diabetes mellitus type 2 associated cataract&quot;,
                      &quot;diabetic maculopathy, type 2 diabetes mellitus&quot;,
                      &quot;diabetic macular edema, type 2 diabetes mellitus&quot;,
                      &quot;proliferative diabetic retinopathy, type 2 diabetes mellitus&quot;
                     )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(type_2_eye_terms),
                          &quot;type 2 diabetes with ophthalmic manifestations&quot;
         ))

# for 30487263, all discovery samples are type 2 diabetes - so 
# replace proliferative diabetic retinopathy with type 2 diabetes with ophthalmic manifestations
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(PUBMED_ID == 30487263,
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;proliferative diabetic retinopathy&quot;),
                          &quot;type 2 diabetes with ophthalmic manifestations&quot;
         ),
         collected_all_disease_terms
         )
  )

# for pubmed id: 31482010
# all samples are type 2 diabetes
# so replace diabetic retinopathy with type 2 diabetes with ophthalmic manifestations
gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         ifelse(PUBMED_ID == 31482010,
                stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(&quot;proliferative diabetic retinopathy&quot;),
                          &quot;type 2 diabetes with ophthalmic manifestations&quot;
         ),
         collected_all_disease_terms
         )
)</code></pre>
</div>
<div id="treatment-resistant" class="section level2">
<h2>Treatment resistant</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_remove_all(collected_all_disease_terms,
                       &quot;^treatment-resistant |^treatment resistant |^treatment-resistant &quot;
         ))</code></pre>
</div>
<div id="uveitis" class="section level2">
<h2>Uveitis</h2>
<pre class="r"><code>uveitis_terms &lt;- c(&quot;anterior uveitis&quot;,
                   &quot;iritis&quot;,
                   &quot;vogt-koyanagi-harada disease&quot;,
                   &quot;birdshot chorioretinopathy&quot;
                   )

gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          pattern = vec_to_grep_pattern(uveitis_terms),
                          &quot;uveitis&quot;
         ))</code></pre>
</div>
<div id="other-specified-retinal-disorders" class="section level2">
<h2>Other specified retinal disorders</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms  = 
         stringr::str_replace_all(collected_all_disease_terms,
                          vec_to_grep_pattern(c(&quot;retinal edema&quot;
                                              )),
                          &quot;other specified retinal disorders&quot;
         ))</code></pre>
</div>
</div>
<div id="save" class="section level1">
<h1>Save</h1>
<div id="how-many-unique-traits-are-there-now-1" class="section level2">
<h2>How many unique traits are there now?</h2>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;,
                               gwas_study_info$collected_all_disease_terms[gwas_study_info$collected_all_disease_terms != &quot;&quot;])  |&gt;
  unlist() |&gt;
  stringr::str_trim()

diseases &lt;- unique(diseases)

print(length(diseases))</code></pre>
<pre><code>[1] 1718</code></pre>
<pre class="r"><code>fwrite(
  gwas_study_info,
  here::here(&quot;output/gwas_cat/gwas_study_info_group.csv&quot;)
  )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] jsonlite_2.0.0    httr_1.4.7        data.table_1.17.8 stringr_1.5.1    
[5] dplyr_1.1.4       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] compiler_4.3.1    renv_1.0.3        promises_1.3.3    tidyselect_1.2.1 
 [5] Rcpp_1.1.0        git2r_0.36.2      callr_3.7.6       later_1.4.2      
 [9] jquerylib_0.1.4   yaml_2.3.10       fastmap_1.2.0     here_1.0.1       
[13] R6_2.6.1          generics_0.1.4    curl_6.4.0        knitr_1.50       
[17] tibble_3.3.0      rprojroot_2.1.0   bslib_0.9.0       pillar_1.11.0    
[21] rlang_1.1.6       cachem_1.1.0      stringi_1.8.7     httpuv_1.6.16    
[25] xfun_0.52         getPass_0.2-4     fs_1.6.6          sass_0.4.10      
[29] cli_3.6.5         withr_3.0.2       magrittr_2.0.3    ps_1.9.1         
[33] digest_0.6.37     processx_3.8.6    rstudioapi_0.17.1 lifecycle_1.0.4  
[37] vctrs_0.6.5       evaluate_1.0.4    glue_1.8.0        whisker_0.4.1    
[41] rmarkdown_2.29    tools_4.3.1       pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
