<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-08" />

<title>Level 1 Disease Trait Grouping - Non-cancer</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Level 1 Disease Trait Grouping -
Non-cancer</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-12-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree2b73f21a720d219970e9ba82b18654917f9e3002targetblank2b73f21a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/2b73f21a720d219970e9ba82b18654917f9e3002" target="_blank">2b73f21</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree2b73f21a720d219970e9ba82b18654917f9e3002targetblank2b73f21a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/2b73f21a720d219970e9ba82b18654917f9e3002" target="_blank">2b73f21</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    .venv/
    Ignored:    analysis/.DS_Store
    Ignored:    ancestry_dispar_env/
    Ignored:    data/.DS_Store
    Ignored:    data/cdc/
    Ignored:    data/cohort/
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/IHME-GBD_2023_DATA-73cc01fd-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/2025AA/
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/hp_umls_mapping.csv
    Ignored:    data/icd/lancet_conditions_icd10.xlsx
    Ignored:    data/icd/manual_disease_icd10_mappings.xlsx
    Ignored:    data/icd/mondo_umls_mapping.csv
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/phecode_to_icd10_manual_mapping.xlsx
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/semiautomatic_ICD-pheno_UKB_subset.txt
    Ignored:    data/icd/umls-2025AA-mrconso.zip
    Ignored:    figures/
    Ignored:    human_dictionary/
    Ignored:    igsr_populations.tsv
    Ignored:    output/.DS_Store
    Ignored:    output/abstracts/
    Ignored:    output/doccano/
    Ignored:    output/fulltexts/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_cohorts/
    Ignored:    output/icd_map/
    Ignored:    output/trait_ontology/
    Ignored:    pubmedbert-cohort-ner-model/
    Ignored:    pubmedbert-cohort-ner/
    Ignored:    r-spacyr/
    Ignored:    renv/
    Ignored:    venv/
    Ignored:    visualization.Rdata

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/get_full_text.Rmd
    Modified:   analysis/gwas_to_gbd.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   analysis/missing_cohort_info.Rmd
    Modified:   analysis/replication_ancestry_bias.Rmd
    Modified:   analysis/text_for_cohort_labels.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/level_1_disease_group_non_cancer.Rmd</code>) and HTML
(<code>docs/level_1_disease_group_non_cancer.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/2b73f21a720d219970e9ba82b18654917f9e3002/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">2b73f21</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-12-29
</td>
<td>
Archiving old GWAS trait conversion
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/abd5cd85e6dd47bb16c0c5ba10ced4070c6b971a/docs/level_1_disease_group_non_cancer.html" target="_blank">abd5cd8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/43ca9588576e6e32f4e86ce0c19ab2c0e44c2090/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">43ca958</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-24
</td>
<td>
More fixing diseases
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/9fa0815504e7fb7ba72c2b0027a88d3f5d2ddb44/docs/level_1_disease_group_non_cancer.html" target="_blank">9fa0815</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b1256b2c71eaf949f00fb4debbae7adbefb44389/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">b1256b2</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
More icd codes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/3828be82b55a4b007351ea1cd2a116f31007da42/docs/level_1_disease_group_non_cancer.html" target="_blank">3828be8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/935db3659eee21ad70111ff56a3b7d9667ed4e56/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">935db36</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
Using icd codes to help grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/4bdd10062db3c90674b89506fc192ea854810f52/docs/level_1_disease_group_non_cancer.html" target="_blank">4bdd100</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/94d5bd5b6558897566b998c75fe218de0c6401f1/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">94d5bd5</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
More typo + structure fixing …
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/8b518cc076b70f808d677ab51ea11cd6840086c7/docs/level_1_disease_group_non_cancer.html" target="_blank">8b518cc</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/61c94d317868c0692dfaba3b9beaf533d10d63c1/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">61c94d3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
…maybe fixing typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/23665d1f812f8e1d79b6bb84cd98dfe3fb49c794/docs/level_1_disease_group_non_cancer.html" target="_blank">23665d1</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/914b2e4d39de838c0c33b2d36580f2b0ab4337f8/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">914b2e4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Excluding infectious disease
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/f1d115d2f54931805068a77b53078844145b1133/docs/level_1_disease_group_non_cancer.html" target="_blank">f1d115d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/44b6e865b3621fa4a58794b230328c1fcfd0e1d9/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">44b6e86</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
More fixing up of disease grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/b2440217593e5e07979ca2ab02bbb04c4fd44a31/docs/level_1_disease_group_non_cancer.html" target="_blank">b244021</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/61ecfb67d6959fa7ba11769f4a6cf2f1d53c01d1/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">61ecfb6</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
More correction to cardiovascular disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/778adb7367fbf77c9d5aa4a153ca3bced6ab476d/docs/level_1_disease_group_non_cancer.html" target="_blank">778adb7</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/4f4993540498d9c67253709e5959983ff5e4de8c/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">4f49935</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Better grouping of cardiovascular disease
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/cb730ee81dd723769ee5d9dee2d6acf6c8cf7030/docs/level_1_disease_group_non_cancer.html" target="_blank">cb730ee</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b01d9aa3b57a3fe2534f8361fa36983822fb5e80/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">b01d9aa</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Improving cancer grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/f15743e7c5892f850f8b3ae639edf2b38ee1f337/docs/level_1_disease_group_non_cancer.html" target="_blank">f15743e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/18a4e855746d7954b8a17dab785a74085f4bab84/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">18a4e85</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
More disease grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/922c9c389504e1e52215aac9bfff966cd3f06865/docs/level_1_disease_group_non_cancer.html" target="_blank">922c9c3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/c60171395c0b967700270f2b4dc07c7421e40db4/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">c601713</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Even more disease term grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/add5ecca974a046bab671d86ff42578ab6c92cca/docs/level_1_disease_group_non_cancer.html" target="_blank">add5ecc</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/465a6899c96d9121f9a8788bd2efd01aedb751b2/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">465a689</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
More disease term grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/7504c04311912b5fcd2713027d672da108f270d0/docs/level_1_disease_group_non_cancer.html" target="_blank">7504c04</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/f01005f56729861ea3bc93a299ceb9e59ae4871f/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">f01005f</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
workflowr::wflow_publish("analysis/level_1_disease_group_non_cancer.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/9aa118eefa3f99476b45d22c18426e75686533a2/docs/level_1_disease_group_non_cancer.html" target="_blank">9aa118e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/ffbf74a538da80378b9854e07111393880703caf/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">ffbf74a</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Further grouping of disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/b19b361b89211b1e2194727f13a659bfd6936dd2/docs/level_1_disease_group_non_cancer.html" target="_blank">b19b361</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/9cc22baacefb040edb910e3155cb32017f853215/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">9cc22ba</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Dealing with duplicate disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/1679f9db8422bc91b7b826e88275a1bdfb2f3339/docs/level_1_disease_group_non_cancer.html" target="_blank">1679f9d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/2250e22191cdd8c16e7118e0fe8ee2bac6c0cce6/docs/level_1_disease_group_non_cancer.html" target="_blank">2250e22</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/e3de56c9875922b0116effb3452d27d40b533d92/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">e3de56c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Update cardiac disease grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/e713c34a2d0a7fcc48decf6e0209016edcfb38b3/docs/level_1_disease_group_non_cancer.html" target="_blank">e713c34</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/934b11f4eff2a42e418f61341f57546fd090cd80/analysis/level_1_disease_group_non_cancer.Rmd" target="_blank">934b11f</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/level_1_disease_group_non_cancer.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Set up</h1>
<pre class="r"><code>library(dplyr)
library(data.table)
library(stringr)</code></pre>
<div id="ontology-help---for-getting-disease-subtypes"
class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Ontology help - for
getting disease subtypes</h2>
<pre class="r"><code>source(here::here(&quot;code/get_term_descendants.R&quot;))</code></pre>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_cat/gwas_study_info_disease_trait_filtered_v2.csv&quot;))</code></pre>
</div>
</div>
<div id="initial-summary" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Initial summary</h1>
<div id="objectives-of-this-analysis" class="section level2"
number="2.1">
<h2><span class="header-section-number">2.1</span> Objectives of this
analysis:</h2>
<p>Level 1 disease grouping: - collapse disease causes, e.g. contact
dermatitis due to nickel to contact dermatitis - collapse disease
subtypes, e.g. bipolar I and bipolar II to bipolar disorder - collapse
disease onset times, e.g. early-onset alzheimers disease and late-onset
alzheimers disease to alzheimers disease - collapse disease stages -
collapse disease complications, e.g. device complication, trauma
complication to complication</p>
</div>
<div id="grouping---level-1-set-up" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Grouping - level 1
set up</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = collected_all_disease_terms)</code></pre>
</div>
<div id="initial-summary---number-of-unique-study-terms"
class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Initial summary -
number of unique study terms</h2>
<div id="when-separate-studies-with-multiple-terms"
class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> When separate
studies with multiple terms</h3>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
 gwas_study_info$collected_all_disease_terms[gwas_study_info$collected_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()

length(unique(diseases))</code></pre>
</div>
</div>
</div>
<div id="disease-complications" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Disease
complications</h1>
<div id="nervous-system-complications-caused-by-herpes-zoster"
class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Nervous system
complications caused by herpes zoster</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;herpes zoster, nervous system disease&quot;),
                          &quot;nervous system disease&quot;
                          )
         )</code></pre>
</div>
<div id="herpes-zoster-psoriasis" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> herpes zoster,
psoriasis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;herpes zoster, psoriasis&quot;),
                          &quot;psoriasis&quot;
         )
         )</code></pre>
</div>
<div id="herpes-zoster-rheumatoid-arthritis" class="section level2"
number="3.3">
<h2><span class="header-section-number">3.3</span> herpes zoster,
rheumatoid arthritis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;herpes zoster, rheumatoid arthritis&quot;),
                          &quot;rheumatoid arthritis&quot;
                          
         )
         )</code></pre>
</div>
</div>
<div id="disturbances-to-senses" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Disturbances to
senses</h1>
<pre class="r"><code>disturb_senses_terms &lt;- c(&quot;disturbances of sensation of smell and taste&quot;,
                          &quot;abnormality of the sense of smell&quot;,
                          &quot;ageusia&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(disturb_senses_terms),
                          &quot;disturbances to senses&quot;
         )
         )</code></pre>
</div>
<div id="disease-stage-grouping" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Disease stage
grouping</h1>
<div id="chronic-kidney-disease-ckd" class="section level2"
number="5.1">
<h2><span class="header-section-number">5.1</span> Chronic kidney
disease (CKD)</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;chronic kidney disease stage 5&quot;),
                          &quot;chronic kidney disease&quot;
         ))


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;stage 5 chronic kidney disease&quot;),
                          &quot;chronic kidney disease&quot;
         ))</code></pre>
</div>
</div>
<div id="disease-cause-grouping" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Disease cause
grouping</h1>
</div>
<div id="disease-complication" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Disease
complication</h1>
<pre class="r"><code>disease_complication_terms = c(&quot;device complication&quot;,
                             &quot;trauma complication&quot;,
                             &quot;adverse effect&quot;,
                             &quot;aseptic loosening&quot;
                             )

pattern &lt;- vec_to_grep_pattern(disease_complication_terms)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = pattern,
                                   &quot;complication&quot;
                          )  
        )</code></pre>
</div>
<div id="cardiovascular-diseases" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Cardiovascular
diseases</h1>
<div id="hypertension" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Hypertension</h2>
<div id="pulmonary-arterial-hypertension" class="section level3"
number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Pulmonary arterial
hypertension</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001361/descendants&quot;

pulmonary_arterial_hypertension_terms &lt;- get_descendants(url)

pattern &lt;- vec_to_grep_pattern(pulmonary_arterial_hypertension_terms)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = pattern,
                          &quot;pulmonary arterial hypertension&quot;
         ))</code></pre>
</div>
<div id="atrial-fibrillation-flutter" class="section level3"
number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> Atrial fibrillation
&amp; flutter</h3>
</div>
<div id="atrial-fibrillation" class="section level3" number="8.1.3">
<h3><span class="header-section-number">8.1.3</span> Atrial
fibrillation</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;post-operative atrial fibrillation&quot;),
                          &quot;atrial fibrillation, post-operative&quot;
         ))



url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F49436004/descendants&quot;

atrial_fibrillation_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(atrial_fibrillation_terms),
                          &quot;atrial fibrillation&quot;
         ))</code></pre>
<div id="atrial-flutter" class="section level4" number="8.1.3.1">
<h4><span class="header-section-number">8.1.3.1</span> Atrial
flutter</h4>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F5370000/descendants&quot;

atrial_flutter_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(atrial_flutter_terms),
                          &quot;atrial flutter&quot;
         ))</code></pre>
</div>
</div>
</div>
<div id="heart-disease" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Heart disease</h2>
<div id="hypertensive-heart-disease" class="section level3"
number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Hypertensive heart
disease</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F64715009/descendants&quot;

hypertensive_heart_disease_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(hypertensive_heart_disease_terms),
                          &quot;hypertensive heart disease&quot;
         ))</code></pre>
</div>
<div id="ischemic-heart-disease" class="section level3" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Ischemic heart
disease</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/cvdo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_3394/descendants&quot;

ischemic_heart_disease_terms &lt;- get_descendants(url)

url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F414545008/descendants&quot;

ischemic_heart_disease_terms_snomed &lt;- get_descendants(url)

ischemic_heart_disease_terms = c(ischemic_heart_disease_terms,
                                 ischemic_heart_disease_terms_snomed
                                 ) |&gt;
                                str_length_sort()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(ischemic_heart_disease_terms),
                          &quot;ischemic heart disease&quot;
         ))</code></pre>
<div id="coronary-artery-disease" class="section level4"
number="8.2.2.1">
<h4><span class="header-section-number">8.2.2.1</span> Coronary artery
disease</h4>
<p>Likely also add, as ICD10 I25.1, seems to be coronary artery disease
And gbd includes this in ischemic heart disease</p>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001645/descendants&quot;

coronary_artery_disease_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(coronary_artery_disease_terms),
                          #pattern = paste0(coronary_artery_disease_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;coronary artery disease&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;coronary artery disease&quot;),
                          # &quot;coronary artery disease&quot;,
                          &quot;ischemic heart disease&quot;
         ))</code></pre>
</div>
<div id="myocardial-infarction" class="section level4" number="8.2.2.2">
<h4><span class="header-section-number">8.2.2.2</span> Myocardial
infarction</h4>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000612/descendants&quot;

myocardial_infarction_terms &lt;- get_descendants(url)

myocardial_infarction_terms &lt;- c(&quot;subsequent st elevation \\(stemi\\) and non-st elevation \\(nstemi\\) myocardial infarction&quot;,
                              myocardial_infarction_terms
                              )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(myocardial_infarction_terms), 
                          #pattern = paste0(myocardial_infarction_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                                  &quot;myocardial infarction&quot;
                          )
         )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;post-operative myocardial infarction&quot;),
                          &quot;myocardial infarction, post-operative&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(&quot;myocardial infarction&quot;),
                          &quot;ischemic heart disease&quot;
         ))</code></pre>
</div>
<div id="fixing" class="section level4" number="8.2.2.3">
<h4><span class="header-section-number">8.2.2.3</span> Fixing</h4>
<pre class="r"><code>to_fix &lt;- c(&quot;non-st elevation ischemic heart disease&quot;,
            &quot;post-operative ischemic heart disease&quot;,
            &quot;st elevation ischemic heart disease&quot;,
            &quot;subsequent st elevation \\(stemi\\) and non-st elevation \\(nstemi\\) ischemic heart disease&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(to_fix),
                          #paste0(to_fix, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;ischemic heart disease&quot;
         ))</code></pre>
</div>
</div>
<div id="non-rheumatic-heart-valvular-disease" class="section level3"
number="8.2.3">
<h3><span class="header-section-number">8.2.3</span> (Non-rheumatic)
Heart valvular disease</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0009551/descendants&quot;

heart_valve_disease_terms &lt;- get_descendants(url)

# make non-rheumatic 
rheumatic_terms &lt;- c(&quot;rheumatic pulmonary valve disease&quot;,
                     &quot;rheumatic disease of mitral valve&quot;)

heart_valve_disease_terms &lt;- 
  heart_valve_disease_terms[!heart_valve_disease_terms %in% rheumatic_terms]

gwas_study_info = gwas_study_info |&gt; 
          mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                   pattern = vec_to_grep_pattern(heart_valve_disease_terms),
                                  # pattern = paste0(heart_valve_disease_terms, 
                                  #                  collapse = &quot;(?=,|$) |\\b&quot;),
                                   &quot;heart valve disease&quot;
                                  )  
          )</code></pre>
</div>
</div>
<div id="other-cardiovascular-diseases-gbd" class="section level2"
number="8.3">
<h2><span class="header-section-number">8.3</span> Other cardiovascular
diseases (GBD)</h2>
<div id="aortic-aneurysm" class="section level3" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Aortic
aneurysm</h3>
<pre class="r"><code>aortic_aneurysm_terms &lt;- c(&quot;thoracic aortic aneurysm&quot;,
                          &quot;abdominal aortic aneurysm&quot;
                          )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(aortic_aneurysm_terms),
                          # pattern = paste0(aortic_aneurysm_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;aortic aneurysm&quot;
         ))</code></pre>
</div>
<div id="endocarditis" class="section level3" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Endocarditis</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;bacterial endocarditis&quot;),
                          &quot;endocarditis&quot;
         ))</code></pre>
</div>
<div id="cardiomyopathy" class="section level3" number="8.3.3">
<h3><span class="header-section-number">8.3.3</span> Cardiomyopathy</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/hp/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0001638/descendants&quot;

cardiomyopathy_terms &lt;- get_descendants(url)

cardiomyopathy_terms &lt;- c(&quot;chagas cardiomyopathy&quot;,
                          &quot;ischemic cardiomyopathy&quot;,
                          &quot;idiopathic cardiomyopathy&quot;,
                          &quot;nonischemic cardiomyopathy&quot;,
                          &quot;peripartum cardiomyopathy&quot;, #? maybe include in pregnancy 
                          cardiomyopathy_terms
                          )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(cardiomyopathy_terms),
                          # pattern = paste0(cardiomyopathy_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;cardiomyopathy&quot;
         ))

gwas_study_info =
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;\\bidiopathic cardiomyopathy\\b&quot;),
                                   &quot;cardiomyopathy&quot;
                          )  
        )</code></pre>
</div>
<div id="stroke" class="section level3" number="8.3.4">
<h3><span class="header-section-number">8.3.4</span> Stroke</h3>
<p><a
href="https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000712?lang=en"
class="uri">https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000712?lang=en</a></p>
<pre class="r"><code>hemorrhage_terms &lt;- c(&quot;intracerebral hemorrhage&quot;,
                      &quot;non-lobar intracerebral hemorrhage&quot;,
                      &quot;lobar intracerebral hemorrhage&quot;,
                      &quot;parenchymal hematoma&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(hemorrhage_terms),
                          # pattern = paste0(hemorrhage_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;hemorrhagic stroke&quot;)
  )

stroke_terms &lt;- c(&quot;stroke outcome&quot;,
                  &quot;large artery stroke&quot;,
                  &quot;small vessel stroke&quot;,
                  &quot;ischemic stroke&quot;,
                  &quot;stroke disorder&quot;,
                  &quot;cardioembolic stroke&quot;,
                  &quot;hemorrhagic stroke&quot;,
                  &quot;intracranial hemorrhage&quot;,
                  &quot;subdural hemorrhage&quot;
                  )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(stroke_terms),
                          # paste0(stroke_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;stroke&quot;)
  )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;post-operative stroke&quot;),
                          &quot;post-operative, stroke&quot;)
  )</code></pre>
</div>
<div id="lower-extremity-peripheral-artery-disease"
class="section level3" number="8.3.5">
<h3><span class="header-section-number">8.3.5</span> Lower extremity
peripheral artery disease</h3>
<pre class="r"><code># studies in gwas catalog don&#39;t specify lower or upper extremity
# but most peripheral artery disease is in lower extremities

gwas_study_info =
 gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;\\bperipheral artery disease|\\bperipheral arterial disease&quot;),
                          &quot;lower extremity peripheral arterial disease&quot;
         ))</code></pre>
</div>
</div>
<div id="other-non-gbd" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Other (non-GBD)</h2>
<div id="carotid-artery-disease" class="section level3" number="8.4.1">
<h3><span class="header-section-number">8.4.1</span> Carotid artery
disease</h3>
<pre class="r"><code>carotid_artery_disease_terms &lt;- c(&quot;carotid artery thrombosis&quot;,
                             &quot;carotid atherosclerosis&quot;
                             )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(carotid_artery_disease_terms),
                          # pattern = paste0(carotid_artery_disease_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;carotid artery disease&quot;
         ))</code></pre>
</div>
<div id="other-hypertension" class="section level3" number="8.4.2">
<h3><span class="header-section-number">8.4.2</span> Other
hypertension</h3>
<pre class="r"><code>url &lt;-  &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000537/descendants&quot;

hypertension_terms &lt;- get_descendants(url)

hypertension_terms = c(&quot;primary hypertension&quot;,
                       &quot;chronic venous hypertension&quot;,
                       &quot;portal hypertension&quot;,
                      hypertension_terms
                      )

hypertension_terms = hypertension_terms[!hypertension_terms %in% 
                                        c(pulmonary_arterial_hypertension_terms,
                                          hypertensive_heart_disease_terms,
                                          &quot;pulmonary arterial hypertension&quot;,
                                          &quot;hypertensive heart disease&quot;
                                          )
                                        ]

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(hypertension_terms),
                          # pattern = paste0(hypertension_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;hypertension&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(hypertension_terms),
                          # pattern = paste0(hypertension_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;hypertension&quot;
         ))</code></pre>
</div>
<div id="heart-block" class="section level3" number="8.4.3">
<h3><span class="header-section-number">8.4.3</span> Heart block</h3>
</div>
<div id="bundle-branch-block" class="section level3" number="8.4.4">
<h3><span class="header-section-number">8.4.4</span> Bundle branch
block</h3>
<pre class="r"><code>bundle_branch_block_terms &lt;- c(&quot;left bundle branch block&quot;,
                             &quot;right bundle branch block&quot;,
                             &quot;complete right bundle branch block&quot;
                             )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(bundle_branch_block_terms),
                          &quot;bundle branch block&quot;
         )
  )</code></pre>
<pre class="r"><code>heart_block_terms &lt;- c(&quot;first degree atrioventricular block&quot;,
                      &quot;second degree atrioventricular block&quot;,
                      &quot;third-degree atrioventricular block&quot;,
                      &quot;atrioventricular block&quot;,
                      &quot;bundle branch block&quot;,
                      &quot;complete right heart block&quot;,
                      &quot;left heart block&quot;
                      )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(heart_block_terms),
                          # pattern = paste0(heart_block_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;heart block&quot;
         ))</code></pre>
</div>
<div id="tachycardia" class="section level3" number="8.4.5">
<h3><span class="header-section-number">8.4.5</span> Tachycardia</h3>
<pre class="r"><code>tacycardia_terms &lt;- c(&quot;paroxysmal tachycardia&quot;,
                     &quot;ventricular tachycardia&quot;,
                     &quot;supraventricular tachycardia&quot;,
                     &quot;paroxysmal supraventricular tachycardia&quot;,
                     &quot;paroxysmal ventricular tachycardia&quot;
                     )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(tacycardia_terms),
                          # pattern = paste0(tacycardia_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;tachycardia&quot;
         )
  )</code></pre>
</div>
<div id="cardiac-arrest" class="section level3" number="8.4.6">
<h3><span class="header-section-number">8.4.6</span> Cardiac arrest</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                         vec_to_grep_pattern(&quot;sudden cardiac arrest&quot;),
                          &quot;cardiac arrest&quot;
         )
  )</code></pre>
</div>
<div id="other-cardiac-arrhythmias" class="section level3"
number="8.4.7">
<h3><span class="header-section-number">8.4.7</span> Other cardiac
arrhythmias</h3>
<pre class="r"><code>other_cardiac_arrhythmia_terms &lt;- c(&quot;ventricular fibrillation&quot;,
                                   &quot;ventricular arrhythmia&quot;,
                                   &quot;cardiac arrest&quot;,
                                   &quot;bradycardia&quot;,
                                   &quot;cardiac arrhythmia&quot;
                                   )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(other_cardiac_arrhythmia_terms),
                          # pattern = paste0(other_cardiac_arrhythmia_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;other cardiac arrhythmias&quot;
         )
  )</code></pre>
</div>
<div id="abnormal-cardiovascular-system-morphology"
class="section level3" number="8.4.8">
<h3><span class="header-section-number">8.4.8</span> Abnormal
cardiovascular system morphology</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mp/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMP_0002127/descendants&quot;

mp_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(mp_terms),
                          &quot;abnormal cardiovascular system morphology&quot;
         )
  )</code></pre>
</div>
<div id="thrombosis" class="section level3" number="8.4.9">
<h3><span class="header-section-number">8.4.9</span> Thrombosis</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0000831/descendants&quot;

thrombosis_terms &lt;- get_descendants(url)

thrombosis_terms &lt;- c(&quot;abnormal thrombosis&quot;,
                      thrombosis_terms)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(thrombosis_terms),
                          # pattern = paste0(thrombosis_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;thrombotic disease&quot;
         )
  )</code></pre>
</div>
<div id="arterial-occlusion" class="section level3" number="8.4.10">
<h3><span class="header-section-number">8.4.10</span> Arterial
occlusion</h3>
<pre class="r"><code>arterial_occlusion_terms &lt;- c(&quot;cerebral artery occlusion&quot;,
                              &quot;occlusion precerebral artery&quot;,
                              &quot;small artery occlusion&quot;
                              )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(arterial_occlusion_terms),
                          # pattern = paste0(arterial_occlusion_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;arterial occlusion&quot;
         )
  )</code></pre>
</div>
<div id="other-vascular-disorders-diseases" class="section level3"
number="8.4.11">
<h3><span class="header-section-number">8.4.11</span> Other vascular
disorders / diseases</h3>
<pre class="r"><code>other_vascular_disorder_terms &lt;- c(&quot;skin vascular disease&quot;,
                                   &quot;peripheral vascular disease&quot;,
                                   &quot;vascular disease&quot;,
                                   &quot;vasculitis&quot;,
                                   &quot;arteritis&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(other_vascular_disorder_terms),
                          &quot;other vascular disorders&quot;
         )
  )</code></pre>
</div>
</div>
<div id="congestive-heart-failure" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Congestive heart
failure</h2>
<pre class="r"><code>congestive_heart_failure_terms &lt;- c(&quot;systolic heart failure&quot;,
                                 &quot;diastolic heart failure&quot;,
                                 &quot;cor pulmonale&quot;,
                                 &quot;congenital heart disease&quot;,
                                 &quot;chronic pulmonary heart disease&quot;
                                 )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(congestive_heart_failure_terms),
                          # pattern = paste0(congestive_heart_failure_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;congestive heart failure&quot;
         ))</code></pre>
</div>
<div id="brain-infarction" class="section level2" number="8.6">
<h2><span class="header-section-number">8.6</span> Brain infarction</h2>
<pre class="r"><code>brain_infarction_terms &lt;- c(&quot;cerebral infarction&quot;,
                          &quot;mri defined brain infarct&quot;,
                          &quot;small cerebral infarction&quot;
                          )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(brain_infarction_terms),
                          # pattern = paste0(brain_infarction_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;brain infarction&quot;
         ))</code></pre>
</div>
<div id="vascular-insufficiency" class="section level2" number="8.7">
<h2><span class="header-section-number">8.7</span> Vascular
insufficiency</h2>
<pre class="r"><code>vascular_insufficiency_terms &lt;- c(&quot;vascular insufficiency disorder&quot;,
                                 &quot;chronic venous insufficiency&quot;,
                                  &quot;chronic intestinal vascular insufficiency&quot;
                                 )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(vascular_insufficiency_terms),
                          # pattern = paste0(vascular_insufficiency_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;vascular insufficiency&quot;
         )
  )</code></pre>
</div>
</div>
<div id="disease-subtype-grouping-non-cancer" class="section level1"
number="9">
<h1><span class="header-section-number">9</span> Disease subtype
grouping (non-cancer)</h1>
<div id="abornmal-total-eosinophil-count" class="section level2"
number="9.1">
<h2><span class="header-section-number">9.1</span> Abornmal total
eosinophil count</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/hp/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0020064/descendants&quot;

abnormal_total_eosinophil_count_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(abnormal_total_eosinophil_count_terms),
                          # pattern = paste0(abnormal_total_eosinophil_count_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;abnormal total eosinophil count&quot;
         ))</code></pre>
</div>
<div id="abortion" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Abortion</h2>
<pre class="r"><code>pregnancy_loss_terms &lt;- c(&quot;habitual abortion&quot;,
                          &quot;incomplete abortion&quot;,
                          &quot;spontaneous abortion&quot;,
                          &quot;abortion&quot;,
                          &quot;spontaneous loss of pregnancy&quot;,
                          &quot;incomplete loss of pregnancy&quot;
                          )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(pregnancy_loss_terms),
                          # paste0(pregnancy_loss_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;loss of pregnancy&quot;
         ))

# exact synonyms (to avoid partial matches
# https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_1001491?lang=en</code></pre>
</div>
<div id="acute-pancreatitis" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Acute
pancreatitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;asparaginase-induced acute pancreatitis&quot;),
                          &quot;acute pancreatitis&quot;
         ))</code></pre>
</div>
<div id="age-related-macular-degeneration" class="section level2"
number="9.4">
<h2><span class="header-section-number">9.4</span> Age-related macular
degeneration</h2>
<pre class="r"><code># from: http://purl.obolibrary.org/obo/HP_0011141
age_related_macular_degeneration_terms &lt;- c(&quot;age-related cortical cataract&quot;,
                                             &quot;age-related nuclear cataract&quot;,
                                             &quot;age-related posterior subcapsular cataract&quot;
                                         )

url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001365/descendants&quot;

age_related_macular_degeneration_terms &lt;- c(age_related_macular_degeneration_terms,
                                             get_descendants(url)
                                             )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(age_related_macular_degeneration_terms),
                          # pattern = paste0(age_related_macular_degeneration_terms, collapse = &quot;(?=,|$) |\\b&quot;),
                          &quot;age-related macular degeneration&quot;
         ))</code></pre>
</div>
<div id="alcohol-use-disorders" class="section level2" number="9.5">
<h2><span class="header-section-number">9.5</span> Alcohol-use
disorders</h2>
</div>
<div id="alcoholic-liver-disease" class="section level2" number="9.6">
<h2><span class="header-section-number">9.6</span> Alcoholic liver
disease</h2>
<pre class="r"><code>alcoholic_liver_disease_terms &lt;- c(&quot;alcoholic liver cirrhosis&quot;,
                                   &quot;alcoholic fatty liver disease&quot;,
                                   &quot;alcoholic hepatitis&quot;
                                   )


gwas_study_info = gwas_study_info |&gt;
      mutate(l1_all_disease_terms = 
           stringr::str_replace_all(l1_all_disease_terms,
                            pattern = vec_to_grep_pattern(alcoholic_liver_disease_terms),
                            &quot;alcoholic liver disease&quot;
           ))</code></pre>
</div>
<div id="alzheimers-disease" class="section level2" number="9.7">
<h2><span class="header-section-number">9.7</span> Alzheimer’s
disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(
                                    c(&quot;early-onset alzheimers disease&quot;,
                                  &quot;late-onset alzheimers disease&quot;)),
                          &quot;alzheimers disease&quot;
         ))</code></pre>
</div>
<div id="allergic-contact-dermatitis" class="section level2"
number="9.8">
<h2><span class="header-section-number">9.8</span> Allergic contact
dermatitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;allergic contact dermatitis of eyelid&quot;),
                          &quot;allergic contact dermatitis&quot;
         ))</code></pre>
</div>
<div id="allergic-rhinitis" class="section level2" number="9.9">
<h2><span class="header-section-number">9.9</span> Allergic
rhinitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;seasonal allergic rhinitis&quot;),
                          &quot;allergic rhinitis&quot;
         ))</code></pre>
</div>
<div id="aplastic-anemia" class="section level2" number="9.10">
<h2><span class="header-section-number">9.10</span> Aplastic anemia</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0015909/descendants&quot;

aplastic_anemia_terms &lt;- get_descendants(url)

aplastic_anemia_terms &lt;- c(&quot;severe aplastic anemia&quot;,
                           aplastic_anemia_terms
                          )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(aplastic_anemia_terms),
                          &quot;aplastic anemia&quot;
         ))</code></pre>
</div>
<div id="autoimmune-pancreatitis" class="section level2" number="9.11">
<h2><span class="header-section-number">9.11</span> Autoimmune
pancreatitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(l1_all_disease_terms  = 
           stringr::str_replace_all(l1_all_disease_terms,
                            vec_to_grep_pattern(&quot;autoimmune pancreatitis type 1&quot;),
                            &quot;autoimmune pancreatitis&quot;
           )) </code></pre>
</div>
<div id="cataract" class="section level2" number="9.12">
<h2><span class="header-section-number">9.12</span> Cataract</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0005129/descendants&quot;

cataract_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(cataract_terms),
                          &quot;cataract&quot;
         ))</code></pre>
</div>
<div id="charcot-marie-tooth-disease-type-1a" class="section level2"
number="9.13">
<h2><span class="header-section-number">9.13</span> Charcot-marie-tooth
disease type 1a</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;charcot-marie-tooth disease type 1a, decreased fine motor function&quot;),
                          &quot;charcot-marie-tooth disease type 1a&quot;
         ))


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;charcot-marie-tooth disease type 1a, decreased walking ability&quot;),
                          &quot;charcot-marie-tooth disease type 1a&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;charcot-marie-tooth disease type 1a, gait imbalance&quot;),
                          &quot;charcot-marie-tooth disease type 1a, gait disturbance&quot;
         ))</code></pre>
</div>
<div id="chronic-pain" class="section level2" number="9.14">
<h2><span class="header-section-number">9.14</span> Chronic pain</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0012532/descendants&quot;

chronic_pain_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(chronic_pain_terms),
                          &quot;chronic pain&quot;
         ))</code></pre>
</div>
<div id="cholecystitis" class="section level2" number="9.15">
<h2><span class="header-section-number">9.15</span> Cholecystitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;acute cholecystitis&quot;),
                          &quot;cholecystitis&quot;
         ))</code></pre>
</div>
<div id="communication-disorder" class="section level2" number="9.16">
<h2><span class="header-section-number">9.16</span> Communication
disorder</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002182/descendants&quot;

communication_disorder_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(communication_disorder_terms),
                          &quot;communication disorder&quot;
         ))</code></pre>
</div>
<div id="delirium" class="section level2" number="9.17">
<h2><span class="header-section-number">9.17</span> Delirium</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;post-operative delirium&quot;),
                          &quot;delirium, post-operative&quot;
         ))</code></pre>
</div>
<div id="dermatitis" class="section level2" number="9.18">
<h2><span class="header-section-number">9.18</span> Dermatitis</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_2723/descendants&quot;

dermatitis_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(dermatitis_terms),
                          &quot;dermatitis&quot;
         ))

# see: http://www.ebi.ac.uk/efo/EFO_0000274
eczema_terms &lt;- c(&quot;atopic eczema&quot;,
                 &quot;hand eczema&quot;,
                 &quot;eczematoid dermatitis&quot;, # see: http://purl.obolibrary.org/obo/HP_0000964
                 &quot;recalcitrant dermatitis&quot; # assuming same as recalcitrant atopic dermatitis http://www.ebi.ac.uk/efo/EFO_1000651
                 )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(eczema_terms),
                          &quot;dermatitis&quot;
         ))</code></pre>
</div>
<div id="diabetic-eye-disease" class="section level2" number="9.19">
<h2><span class="header-section-number">9.19</span> Diabetic eye
disease</h2>
<pre class="r"><code>diabetic_eye_terms &lt;- c(&quot;diabetic maculopathy&quot;,
                        &quot;diabetic macular edema&quot;,
                        &quot;diabetic retinopathy&quot;,
                        &quot;proliferative diabetic retinopathy&quot;,
                        &quot;non-proliferative diabetic retinopathy&quot;,
                        &quot;diabetes mellitus type 2 associated cataract&quot;)


gwas_study_info = 
  gwas_study_info |&gt; 
  mutate(l1_all_disease_terms  = 
           stringr::str_replace_all(l1_all_disease_terms,
                                    pattern = vec_to_grep_pattern(diabetic_eye_terms),
                                    &quot;diabetic eye disease&quot;
           )  
  )</code></pre>
</div>
<div id="diabetic-neuropathy" class="section level2" number="9.20">
<h2><span class="header-section-number">9.20</span> Diabetic
neuropathy</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;diabetic polyneuropathy&quot;),
                          &quot;diabetic neuropathy&quot;
         ))</code></pre>
</div>
<div id="deficiency-anemia" class="section level2" number="9.21">
<h2><span class="header-section-number">9.21</span> Deficiency
anemia</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;iron deficiency anemia&quot;),
                          &quot;deficiency anemia&quot;
         ))</code></pre>
</div>
<div id="encephalopathy" class="section level2" number="9.22">
<h2><span class="header-section-number">9.22</span> Encephalopathy</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;delayed encephalopathy after acute poisoning&quot;),
                          &quot;encephalopathy, poisoning&quot;
         ))


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;encephalopathy acute infection-induced&quot;),
                          &quot;encephalopathy&quot;
         ))</code></pre>
</div>
<div id="gout" class="section level2" number="9.23">
<h2><span class="header-section-number">9.23</span> Gout</h2>
<pre class="r"><code># from: http://www.ebi.ac.uk/efo/EFO_0004274
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;renal overload-type gout&quot;),
                          &quot;gout&quot;
         ))</code></pre>
</div>
<div id="hemolytic-anemia" class="section level2" number="9.24">
<h2><span class="header-section-number">9.24</span> Hemolytic
anemia</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0003664/descendants&quot;


hemolytic_anemia_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(hemolytic_anemia_terms),
                          &quot;hemolytic anemia&quot;
         ))</code></pre>
</div>
<div id="hereditary-progressive-muscular-distrophy"
class="section level2" number="9.25">
<h2><span class="header-section-number">9.25</span> Hereditary
progressive muscular distrophy</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F193225000/descendants&quot;

hereditary_progressive_muscular_dystrophy_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(hereditary_progressive_muscular_dystrophy_terms),
                          &quot;hereditary progressive muscular dystrophy&quot;
         ))</code></pre>
</div>
<div id="hernia-of-the-abdominal-wall" class="section level2"
number="9.26">
<h2><span class="header-section-number">9.26</span> Hernia of the
abdominal wall</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/hp/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0004299/children&quot;

hernia_abdominal_wall_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(hernia_abdominal_wall_terms),
                          &quot;hernia of the abdominal wall&quot;
         ))</code></pre>
</div>
<div id="hyperaldosteronism" class="section level2" number="9.27">
<h2><span class="header-section-number">9.27</span>
Hyperaldosteronism</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;primary hyperaldosteronism&quot;),
                          &quot;hyperaldosteronism&quot;
         ))</code></pre>
</div>
<div id="hyperlipidemia" class="section level2" number="9.28">
<h2><span class="header-section-number">9.28</span> Hyperlipidemia</h2>
<pre class="r"><code>hyperlipidemia_terms &lt;- c(&quot;hypercholesterolemia&quot;,
                         &quot;familial hypercholesterolemia&quot;,
                         &quot;familial hyperlipidemia&quot;
                         )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(hyperlipidemia_terms),
                          &quot;hyperlipidemia&quot;
         ))</code></pre>
</div>
<div id="inborn-errors-of-metabolism" class="section level2"
number="9.29">
<h2><span class="header-section-number">9.29</span> Inborn errors of
metabolism</h2>
<pre class="r"><code>inborn_error_metab &lt;- c(&quot;inborn disorder of amino acid metabolism&quot;, # ICD9:270, ICD9:270.9
                        &quot;inborn disorder of amino acid transport&quot;, # ICD9:270.0
                        &quot;inborn disorder of porphyrin metabolism&quot;, # ? ebi suggests ... ICD10CM:E80.4 - doesn&#39;t seem right match, but close - E80 range likely captures it
                        &quot;inborn carbohydrate metabolic disorder&quot;, # ICD9:271.8
                        &quot;familial lipoprotein lipase deficiency&quot;, # ICD9:272.3
                        &quot;lactose intolerance&quot;, # ICD10: E73
                        &quot;hereditary hemochromatosis&quot;, # ICD10CM:E83.110, ICD9: 275.01
                        &quot;alpha 1-antitrypsin deficiency&quot;, # ICD9:273.4
                        &quot;urea cycle disorder&quot;, # ICD9CM:270.6
                        &quot;gaucher disease&quot;, # ICD10CM:E75.22, ICD9:272.7
                        &quot;plasma protein metabolism disease&quot;, # ? ICD9:273.8
                        &quot;disorder of metabolite absorption and transport&quot;
                        )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(inborn_error_metab),
                          &quot;inborn errors of metabolism&quot;
         ))</code></pre>
</div>
<div id="intestinal-obstruction" class="section level2" number="9.30">
<h2><span class="header-section-number">9.30</span> Intestinal
obstruction</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0004565/descendants&quot;

intestinal_obstruction_terms &lt;- get_descendants(url)

# also add: http://purl.obolibrary.org/obo/DOID_8437

intestinal_obstruction_terms = c(intestinal_obstruction_terms,
                                 &quot;intestinal impaction&quot;
                                 )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(intestinal_obstruction_terms),
                          &quot;intestinal obstruction&quot;
         ))</code></pre>
</div>
<div id="chronic-kidney-disease" class="section level2" number="9.31">
<h2><span class="header-section-number">9.31</span> Chronic kidney
disease</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0003884/descendants&quot;


kidney_disease_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(kidney_disease_terms),
                          &quot;chronic kidney disease&quot;
         ))</code></pre>
</div>
<div id="penile-disorder" class="section level2" number="9.32">
<h2><span class="header-section-number">9.32</span> Penile disorder</h2>
<pre class="r"><code>penile_disorder &lt;- c(&quot;balanitis&quot;,
                     &quot;balanoposthitis&quot;,
                     &quot;priapism&quot;,
                     &quot;phimosis&quot;,
                     &quot;erectile dysfunction&quot;,
                     &quot;leukoplakia of penis&quot;)


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(penile_disorder),
                          &quot;penile disorder&quot;
         ))</code></pre>
</div>
<div id="photosensitivity-disease" class="section level2" number="9.33">
<h2><span class="header-section-number">9.33</span> Photosensitivity
disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
        stringr::str_replace_all(l1_all_disease_terms,
                         vec_to_grep_pattern(
                           c(&quot;phototoxic dermatitis&quot;,
                           &quot;skin sensitivity to sun&quot;
                           )),
                         &quot;photosensitivity disease&quot;)
        )</code></pre>
</div>
<div id="phobic-disorder" class="section level2" number="9.34">
<h2><span class="header-section-number">9.34</span> Phobic disorder</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_1001908/descendants&quot;

phobic_disorder_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(phobic_disorder_terms),
                          &quot;phobic disorder&quot;
         ))</code></pre>
</div>
<div id="mineral-metabolism-disease" class="section level2"
number="9.35">
<h2><span class="header-section-number">9.35</span> Mineral metabolism
disease</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0009556/descendants&quot;

mineral_metabolism_disorder_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(mineral_metabolism_disorder_terms),
                          &quot;mineral metabolism disease&quot;
         ))</code></pre>
</div>
<div id="non-alcoholic-fatty-liver-disease-nafld" class="section level2"
number="9.36">
<h2><span class="header-section-number">9.36</span> Non-alcoholic fatty
liver disease (NAFLD)</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                                  vec_to_grep_pattern(
                                    c(&quot;non-alcoholic steatohepatitis&quot;,
                                      &quot;non-alcoholic liver disease&quot;
                                     )
                                   ),
                          &quot;non-alcoholic fatty liver disease&quot;
         ))</code></pre>
</div>
<div id="orofacial-cleft" class="section level2" number="9.37">
<h2><span class="header-section-number">9.37</span> Orofacial cleft</h2>
<pre class="r"><code>orofacial_cleft_terms &lt;- c(&quot;cleft lip&quot;,
                            &quot;cleft palate&quot;
                            )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(orofacial_cleft_terms),
                          &quot;orofacial cleft&quot;
         ))</code></pre>
</div>
<div id="osteoarthritis" class="section level2" number="9.38">
<h2><span class="header-section-number">9.38</span> Osteoarthritis</h2>
<pre class="r"><code>osteoarthritis_terms &lt;- c(&quot;knee osteoarthritis&quot;,
                         &quot;hip osteoarthritis&quot;,
                         &quot;hand osteoarthritis&quot;,
                         &quot;spine osteoarthritis&quot;,
                         &quot;toe osteoarthritis&quot;
                         )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(osteoarthritis_terms),
                          &quot;osteoarthritis&quot;
         )
  )</code></pre>
</div>
<div id="ovarian-dysfunction" class="section level2" number="9.39">
<h2><span class="header-section-number">9.39</span> Ovarian
dysfunction</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0009003/descendants&quot;

ovarian_dysfunction_terms &lt;- get_descendants(url)

ovarian_dysfunction_terms &lt;- c(ovarian_dysfunction_terms,
                              &quot;premature ovarian failure&quot;,
                              &quot;premature ovarian insufficiency&quot;
                              )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(ovarian_dysfunction_terms),
                          &quot;ovarian dysfunction&quot;
         ))</code></pre>
</div>
<div id="pancreatitis" class="section level2" number="9.40">
<h2><span class="header-section-number">9.40</span> Pancreatitis</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000278/descendants&quot;

pancreatitis_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(pancreatitis_terms),
                          &quot;pancreatitis&quot;
         ))</code></pre>
</div>
<div id="peridontal-disease" class="section level2" number="9.41">
<h2><span class="header-section-number">9.41</span> Peridontal
disease</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_3388/descendants&quot;

periodontal_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(periodontal_terms),
                          &quot;periodontal disease&quot;
         ))</code></pre>
</div>
<div id="pituitary-gland-disease" class="section level2" number="9.42">
<h2><span class="header-section-number">9.42</span> Pituitary gland
disease</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0009607/descendants&quot;

pituitary_gland_disease_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(pituitary_gland_disease_terms),
                          &quot;pituitary gland disease&quot;
         ))</code></pre>
</div>
<div id="polycythemia" class="section level2" number="9.43">
<h2><span class="header-section-number">9.43</span> Polycythemia</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0005804/descendants&quot;

polycythemia_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(polycythemia_terms),
                          &quot;polycythemia&quot;
         ))</code></pre>
</div>
<div id="pulmonary-embolism" class="section level2" number="9.44">
<h2><span class="header-section-number">9.44</span> Pulmonary
embolism</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;pulmonary embolism, pulmonary infarction&quot;),
                          &quot;pulmonary embolism&quot;
         ))</code></pre>
</div>
<div id="psychosis" class="section level2" number="9.45">
<h2><span class="header-section-number">9.45</span> Psychosis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;psychosis predisposition&quot;),
                          &quot;psychosis&quot;)
         )</code></pre>
</div>
<div id="retinopathy" class="section level2" number="9.46">
<h2><span class="header-section-number">9.46</span> Retinopathy</h2>
<pre class="r"><code>retinopathy_terms &lt;- c(&quot;chronic central serous retinopathy&quot;,
                       &quot;central serous retinopathy&quot;
                       )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
           stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(retinopathy_terms),
                          &quot;retinopathy&quot;)
  )</code></pre>
</div>
<div id="sciatica" class="section level2" number="9.47">
<h2><span class="header-section-number">9.47</span> Sciatica</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_remove_all(l1_all_disease_terms,
                          &quot;^ldh-related sciatica, |, ldh-related sciatica$&quot;
         )
  )</code></pre>
</div>
<div id="sickle-cell-disease" class="section level2" number="9.48">
<h2><span class="header-section-number">9.48</span> Sickle cell
disease</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;sickle cell anemia&quot;),
                          &quot;sickle cell disease and related diseases&quot;)
         )</code></pre>
</div>
<div id="staphylococcus-aureus-infection" class="section level2"
number="9.49">
<h2><span class="header-section-number">9.49</span> Staphylococcus
aureus infection</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;methicillin-resistant staphylococcus aureus infection&quot;),
                          &quot;staphylococcus aureus infection&quot;
         ))</code></pre>
</div>
<div id="systemic-scleroderma" class="section level2" number="9.50">
<h2><span class="header-section-number">9.50</span> Systemic
scleroderma</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000717/descendants&quot;

scleroderma_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(scleroderma_terms),
                          &quot;systemic scleroderma&quot;
         ))</code></pre>
</div>
<div id="tenosynovitis" class="section level2" number="9.51">
<h2><span class="header-section-number">9.51</span> Tenosynovitis</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
        stringr::str_replace_all(l1_all_disease_terms,
                         vec_to_grep_pattern(&quot;stenosing tenosynovitis&quot;),
                         &quot;tenosynovitis&quot;)
        )</code></pre>
</div>
<div id="thrombocytopenia" class="section level2" number="9.52">
<h2><span class="header-section-number">9.52</span>
Thrombocytopenia</h2>
<pre class="r"><code>thrombocytopenia_terms &lt;- c(&quot;thrombocytopenia 4&quot;,
                           &quot;acquired thrombocytopenia&quot;,
                           &quot;primary thrombocytopenia&quot;
                           )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(thrombocytopenia_terms),
                          &quot;thrombocytopenia&quot;
         ))</code></pre>
</div>
<div id="tinea" class="section level2" number="9.53">
<h2><span class="header-section-number">9.53</span> Tinea</h2>
<pre class="r"><code>tinea_terms &lt;- c(&quot;tinea pedis&quot;,
                 &quot;tinea unguium&quot;,
                 &quot;dermatophytosis&quot;
                 )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(tinea_terms),
                          &quot;tinea&quot;
         ))</code></pre>
</div>
<div id="vitamin-b-deficiency" class="section level2" number="9.54">
<h2><span class="header-section-number">9.54</span> Vitamin b
deficiency</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;vitamin b12 deficiency&quot;),
                          &quot;vitamin b deficiency&quot;
         ))</code></pre>
</div>
<div id="visual-impairment" class="section level2" number="9.55">
<h2><span class="header-section-number">9.55</span> Visual
impairment</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0000505/descendants&quot;

visual_impairment_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(visual_impairment_terms),
                          &quot;visual impairment&quot;
         ))</code></pre>
</div>
</div>
<div id="further-grouping-of-terms" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Further grouping of
terms</h1>
<div id="abnormal-refraction" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Abnormal
refraction</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/upheno/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0000539/descendants&quot;

abnormal_refraction_terms &lt;- get_descendants(url)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          pattern = vec_to_grep_pattern(abnormal_refraction_terms),
                          &quot;abnormality of refraction&quot;
         ))</code></pre>
</div>
</div>
<div id="final-summary---number-of-unique-study-terms"
class="section level1" number="11">
<h1><span class="header-section-number">11</span> Final summary - number
of unique study terms</h1>
<div id="deal-with-duplicate-terms-created-during-grouping"
class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Deal with duplicate
terms created during grouping</h2>
<pre class="r"><code>gwas_study_info = 
 gwas_study_info |&gt;
  rowwise() |&gt;
  mutate(l1_all_disease_terms = paste0(sort(unique(unlist(strsplit(l1_all_disease_terms, &quot;, &quot;)))),
                                      collapse = &quot;, &quot;)
         ) |&gt;
  ungroup()</code></pre>
</div>
<div id="deal-with-hanging-commas-and-spaces" class="section level2"
number="11.2">
<h2><span class="header-section-number">11.2</span> Deal with hanging
commas and spaces</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = stringr::str_remove_all(l1_all_disease_terms, &quot;^,|,$&quot;)
         ) |&gt;
  mutate(l1_all_disease_terms = stringr::str_trim(l1_all_disease_terms)
         ) </code></pre>
</div>
<div id="final-summary---number-of-unique-study-terms-pairs"
class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> Final summary -
number of unique study terms pairs</h2>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(l1_all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(l1_all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))


head(n_studies_trait)

dim(n_studies_trait)</code></pre>
<div id="when-separate-studies-with-multiple-terms-1"
class="section level3" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> When separate
studies with multiple terms</h3>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l1_all_disease_terms[gwas_study_info$l1_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()


test &lt;- data.frame(trait = unique(diseases))

length(unique(diseases))

# make frequency table
freq &lt;- table(as.factor(diseases))

# sort in decreasing order
freq_sorted &lt;- sort(freq, decreasing = TRUE)

# show top N, e.g. top 10
head(freq_sorted, 10)</code></pre>
</div>
<div id="save-the-updated-gwas_study_info-with-harmonized-disease-terms"
class="section level3" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> Save the updated
gwas_study_info with harmonized disease terms</h3>
<pre class="r"><code>fwrite(gwas_study_info,
        here::here(&quot;output/gwas_cat/gwas_study_info_group_l1.csv&quot;)
         )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
