<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-08" />

<title>Level 1 Disease Trait Grouping - Cancer traits</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Level 1 Disease Trait Grouping - Cancer
traits</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-12-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree66f7b4e5597e95d18a588ea4fd3597bb1e8ad8c5targetblank66f7b4ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/66f7b4e5597e95d18a588ea4fd3597bb1e8ad8c5" target="_blank">66f7b4e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree66f7b4e5597e95d18a588ea4fd3597bb1e8ad8c5targetblank66f7b4ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/66f7b4e5597e95d18a588ea4fd3597bb1e8ad8c5" target="_blank">66f7b4e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    .venv/
    Ignored:    analysis/.DS_Store
    Ignored:    ancestry_dispar_env/
    Ignored:    data/.DS_Store
    Ignored:    data/cdc/
    Ignored:    data/cohort/
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/IHME-GBD_2023_DATA-73cc01fd-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/2025AA/
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/hp_umls_mapping.csv
    Ignored:    data/icd/lancet_conditions_icd10.xlsx
    Ignored:    data/icd/manual_disease_icd10_mappings.xlsx
    Ignored:    data/icd/mondo_umls_mapping.csv
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/phecode_to_icd10_manual_mapping.xlsx
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/semiautomatic_ICD-pheno_UKB_subset.txt
    Ignored:    data/icd/umls-2025AA-mrconso.zip
    Ignored:    figures/
    Ignored:    human_dictionary/
    Ignored:    igsr_populations.tsv
    Ignored:    output/.DS_Store
    Ignored:    output/abstracts/
    Ignored:    output/doccano/
    Ignored:    output/fulltexts/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_cohorts/
    Ignored:    output/icd_map/
    Ignored:    output/trait_ontology/
    Ignored:    pubmedbert-cohort-ner-model/
    Ignored:    pubmedbert-cohort-ner/
    Ignored:    r-spacyr/
    Ignored:    renv/
    Ignored:    venv/
    Ignored:    visualization.Rdata

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/get_full_text.Rmd
    Modified:   analysis/gwas_to_gbd.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_1_disease_group_non_cancer.Rmd
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   analysis/missing_cohort_info.Rmd
    Modified:   analysis/replication_ancestry_bias.Rmd
    Modified:   analysis/text_for_cohort_labels.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/level_1_disease_group_cancer.Rmd</code>) and HTML
(<code>docs/level_1_disease_group_cancer.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/66f7b4e5597e95d18a588ea4fd3597bb1e8ad8c5/analysis/level_1_disease_group_cancer.Rmd" target="_blank">66f7b4e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-12-29
</td>
<td>
Archiving old GWAS trait conversion
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/8072a8ef8d8bde576a5acd28cd20b84f25a89307/docs/level_1_disease_group_cancer.html" target="_blank">8072a8e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/bcd91ff3d72af16f74d002438be525df29a403a0/analysis/level_1_disease_group_cancer.Rmd" target="_blank">bcd91ff</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-24
</td>
<td>
More fixing diseases
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/137d32695df5ad678f77d266f51861d43af1a871/docs/level_1_disease_group_cancer.html" target="_blank">137d326</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/03f1751af38db5d81971eed307ca080b6a12f1f2/analysis/level_1_disease_group_cancer.Rmd" target="_blank">03f1751</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
More icd codes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/0fd42876497b695d09079ca9e5afe85ba1deb475/docs/level_1_disease_group_cancer.html" target="_blank">0fd4287</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/3b88f25b54a14852ad4db1288a44156bf14f93cf/docs/level_1_disease_group_cancer.html" target="_blank">3b88f25</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/df0281b52970a108b87635e5267ffc0b67feddb9/analysis/level_1_disease_group_cancer.Rmd" target="_blank">df0281b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
More typo + structure fixing …
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/02263dddf4de67a29550d226691f710054f9f657/docs/level_1_disease_group_cancer.html" target="_blank">02263dd</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/72709e95d1cccb677d2e50a6a299d3c3bd0f3098/analysis/level_1_disease_group_cancer.Rmd" target="_blank">72709e9</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
…maybe fixing typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/f7ea2576e3f677bae1fbf0cbd0311ee2f6a212d5/docs/level_1_disease_group_cancer.html" target="_blank">f7ea257</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/a13c2724078f2c117ff031a140e4ab93cce3e243/docs/level_1_disease_group_cancer.html" target="_blank">a13c272</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/003c2267e67d5b1cfa67d6e210e30f2ea8beabfb/analysis/level_1_disease_group_cancer.Rmd" target="_blank">003c226</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
More fixing up of disease grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/3d701f261b30e32f7b1f3cc4f8cac29f894cfaa8/docs/level_1_disease_group_cancer.html" target="_blank">3d701f2</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/4f70a33619a609c142e08f72d19994b0a25b1366/docs/level_1_disease_group_cancer.html" target="_blank">4f70a33</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/41b1b7c78a61852772721a44bcc0bac0e743a978/analysis/level_1_disease_group_cancer.Rmd" target="_blank">41b1b7c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Better grouping of cardiovascular disease
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/fa95c620639ee64ada0dee01f018baaf56f97475/docs/level_1_disease_group_cancer.html" target="_blank">fa95c62</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/57e46da16276ebf0aa3771874d41cd7e30f41f70/analysis/level_1_disease_group_cancer.Rmd" target="_blank">57e46da</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
More typo fixing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/cfd2ef84a043e6a8700f9f2d8e86bcfccf45795e/docs/level_1_disease_group_cancer.html" target="_blank">cfd2ef8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/7df4726b725a4389442b08815bd220ace0dd1fdf/analysis/level_1_disease_group_cancer.Rmd" target="_blank">7df4726</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Dealing with non-specific cancer labels
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/2aa6027f63ec41956972459f48065f13d84934de/docs/level_1_disease_group_cancer.html" target="_blank">2aa6027</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b6f20c42ffa346161f08ae0e5cea9c7981f132fc/analysis/level_1_disease_group_cancer.Rmd" target="_blank">b6f20c4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Adding more benign neoplasm
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/83152bd539f14eb80ee5122f0e7de5f782cf5952/docs/level_1_disease_group_cancer.html" target="_blank">83152bd</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/d7db734d7e894e250565e5dc44cb21eaf5deecc9/docs/level_1_disease_group_cancer.html" target="_blank">d7db734</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/53bf24e83f1dd219c090bff37a261f5102e9ba4e/analysis/level_1_disease_group_cancer.Rmd" target="_blank">53bf24e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
More cancer typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/b0f0ff577a65ea2c078a3fd3dc73a33ac992c4a0/docs/level_1_disease_group_cancer.html" target="_blank">b0f0ff5</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/e8fb82ca554069ce66615507f54accca7b06ee5b/analysis/level_1_disease_group_cancer.Rmd" target="_blank">e8fb82c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Correcting some cancer grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/de1a7406ef0f6929bb4063795747fff3462f6584/docs/level_1_disease_group_cancer.html" target="_blank">de1a740</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/69d6255631190f9d436151d112d7e3e3e11585af/analysis/level_1_disease_group_cancer.Rmd" target="_blank">69d6255</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Improving cancer grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/da4e2ccead3e621595163d49733303dbc1d9a0d0/docs/level_1_disease_group_cancer.html" target="_blank">da4e2cc</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/019691496b8d11c1fab9ba01fa8a438168db2d6b/analysis/level_1_disease_group_cancer.Rmd" target="_blank">0196914</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
More disease grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/937b4601f8db81ada93b281503b1a063f1007c3c/docs/level_1_disease_group_cancer.html" target="_blank">937b460</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/3ac50bd1f7bbfa492dc1a1f96c10f7c4abe6fdf7/analysis/level_1_disease_group_cancer.Rmd" target="_blank">3ac50bd</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Even more disease term grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/7c6dee88a9b6589c24035adbae91373db669e70d/docs/level_1_disease_group_cancer.html" target="_blank">7c6dee8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/445142184b40e7256e593707d6e985995d7e1121/analysis/level_1_disease_group_cancer.Rmd" target="_blank">4451421</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Grouping more neoplasms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/2e145f8cf31a5e73f86cff6e7753ae9c1440512f/docs/level_1_disease_group_cancer.html" target="_blank">2e145f8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/2702dc1df5ef879b9d14a5901a3e6b8f2c77700b/analysis/level_1_disease_group_cancer.Rmd" target="_blank">2702dc1</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
workflowr::wflow_publish("analysis/level_1_disease_group_cancer.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/7fe9a06522f25ada1d957fb35f0c3976d748a83a/docs/level_1_disease_group_cancer.html" target="_blank">7fe9a06</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/81a1d22e7d3574eecccba52bd17b35b0648453b6/analysis/level_1_disease_group_cancer.Rmd" target="_blank">81a1d22</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
workflowr::wflow_publish("analysis/level_1_disease_group_cancer.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/1f89b203ba3320e8525a738c232e704ae373c92b/docs/level_1_disease_group_cancer.html" target="_blank">1f89b20</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/fdd60ed13f36d58017a4f241006b32abb53d343c/analysis/level_1_disease_group_cancer.Rmd" target="_blank">fdd60ed</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
More disease term grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/bf45a695d011e5adf921d602aa238bbb56eafff0/docs/level_1_disease_group_cancer.html" target="_blank">bf45a69</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/1414cad810662ddf218b7d850f602c7d8447456b/analysis/level_1_disease_group_cancer.Rmd" target="_blank">1414cad</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
workflowr::wflow_publish("analysis/level_1_disease_group_cancer.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/3c8309cba2a866c31c2a4fa36c7f666dbcfe6ac2/docs/level_1_disease_group_cancer.html" target="_blank">3c8309c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/17a16b0568cc32537a563f99d252bfa8e5878ac2/analysis/level_1_disease_group_cancer.Rmd" target="_blank">17a16b0</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Further grouping of disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/778ac1e5e852390a65e1e9ba417c60037a4cc9d3/docs/level_1_disease_group_cancer.html" target="_blank">778ac1e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/bb5431c248ee3c0817ecc0010deaa8b2a8151eeb/analysis/level_1_disease_group_cancer.Rmd" target="_blank">bb5431c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Dealing with duplicate disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/9f6997944f597beb2db9edbe85a0cc9aa2990cb5/docs/level_1_disease_group_cancer.html" target="_blank">9f69979</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/9ca183a43197ca703180df4ab805ce841735c03e/docs/level_1_disease_group_cancer.html" target="_blank">9ca183a</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/50ef69d9fb5863dcdfcb1dd86b10766078897b33/analysis/level_1_disease_group_cancer.Rmd" target="_blank">50ef69d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Update cancer grouping
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(dplyr)
library(data.table)
library(stringr)</code></pre>
<div id="ontology-help---for-getting-disease-subtypes"
class="section level2" number="0.1">
<h2><span class="header-section-number">0.1</span> Ontology help - for
getting disease subtypes</h2>
<pre class="r"><code>source(here::here(&quot;code/get_term_descendants.R&quot;))</code></pre>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_cat/gwas_study_info_group_l1.csv&quot;))</code></pre>
</div>
<div id="initial-summary---number-of-unique-study-terns"
class="section level1" number="1">
<h1><span class="header-section-number">1</span> Initial summary -
number of unique study terns</h1>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(l1_all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(l1_all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))


head(n_studies_trait)

dim(n_studies_trait)</code></pre>
<div id="when-separate-studies-with-multiple-terms"
class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> When separate studies
with multiple terms</h2>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l1_all_disease_terms[gwas_study_info$l1_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()


length(unique(diseases))


test &lt;- data.frame(trait = unique(diseases))</code></pre>
</div>
</div>
<div id="disease-subtype-grouping-cancer" class="section level1"
number="2">
<h1><span class="header-section-number">2</span> Disease subtype
grouping (cancer)</h1>
<div id="astrocytoma" class="section level3" number="2.0.1">
<h3><span class="header-section-number">2.0.1</span> Astrocytoma</h3>
<pre class="r"><code>gwas_study_info |&gt;
    filter(grepl(&quot;astrocytoma&quot;, l1_all_disease_terms)) |&gt; 
    pull(STUDY) |&gt; 
    unique()

# all comes from one cancer study - so a central nervous system cancer

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  =
          ifelse(PUBMED_ID == &quot;36810956&quot;,
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;astrocytoma&quot;),
                                   &quot;central nervous system cancer&quot;
                          ),
          l1_all_disease_terms
        )
 )</code></pre>
</div>
<div id="bone-cancer" class="section level3" number="2.0.2">
<h3><span class="header-section-number">2.0.2</span> Bone cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002129/descendants&quot;

bone_cancer_terms &lt;- get_descendants(url)

bone_cancer_terms = stringr::str_replace_all(bone_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

bone_cancer_terms = c(&quot;malignant bone neoplasm&quot;,
                      bone_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(bone_cancer_terms),
                                   &quot;bone cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="bladder-cancer" class="section level3" number="2.0.3">
<h3><span class="header-section-number">2.0.3</span> Bladder cancer</h3>
<pre class="r"><code>url &lt;-  &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_4007/descendants&quot;

# maybe do uninary bladder cancer instead

bladder_cancer_terms &lt;- get_descendants(url)

bladder_cancer_terms = stringr::str_replace_all(bladder_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(bladder_cancer_terms),
                                   &quot;bladder cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="breast-cancer" class="section level3" number="2.0.4">
<h3><span class="header-section-number">2.0.4</span> Breast cancer</h3>
<pre class="r"><code>breast_cancer_terms &lt;- grep(&quot;breast cancer&quot;, unique(diseases), value = T)

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(breast_cancer_terms),
                                   &quot;breast cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;breast cancer in situ&quot;),
                                   &quot;breast cancer&quot;
                          )  
        ) |&gt;
     mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;invasive lobular cancer&quot;),
                                   &quot;breast cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="benign-neoplasms" class="section level3" number="2.0.5">
<h3><span class="header-section-number">2.0.5</span> Benign
neoplasms</h3>
<div id="benign-neoplasm-of-blood-vessel" class="section level4"
number="2.0.5.1">
<h4><span class="header-section-number">2.0.5.1</span> Benign neoplasm
of blood vessel</h4>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F92017000/descendants&quot;

benign_blood_vessel_terms &lt;- get_descendants(url)

benign_blood_vessel_terms &lt;- stringr::str_replace_all(benign_blood_vessel_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)


gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms = 
        stringr::str_replace_all(l1_all_disease_terms, 
                    vec_to_grep_pattern(benign_blood_vessel_terms),
                    &quot;benign neoplasm&quot;)
         )</code></pre>
</div>
</div>
<div id="labelled-benign-neoplasms" class="section level3"
number="2.0.6">
<h3><span class="header-section-number">2.0.6</span> Labelled benign
neoplasms</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;(?&lt;=^|, )benign neoplasm of (.*?)(?=,|$)|(?&lt;=^|, )benign neoplasm of (.*?) (.*?)(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;(?&lt;=^|, )benign (.*?) neoplasm(?=,|$)|(?&lt;=^|, )benign (.*?) (.*?) neoplasm(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;(?&lt;=^|, )(.*?) benign neoplasm(?=,|$)|(.*?) (.*?) neoplasm(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;(?&lt;=^|, )polyp of (.*?)(?=,|$)|(?&lt;=^|, )polyp of (.*?) (.*?)(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;(?&lt;=^|, ) (.*?) polyp(?=,|$)|(?&lt;=^|, )(.*?) (.*?) polyp (?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) </code></pre>
</div>
</div>
<div id="other-benign-neoplasms" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Other benign
neoplasms</h1>
<pre class="r"><code># https://my.clevelandclinic.org/health/diseases/21477-adenomas - benign
other_benign_neoplasms = c(&quot;adenomatous colon polyp&quot;,
                           &quot;colorectal adenoma&quot;,
                           &quot;pituitary gland adenoma&quot;,
                           &quot;aldosterone-producing adenoma&quot;,
                           &quot;metachronous colorectal adenoma&quot;,
                           &quot;adenomatous colon polyp&quot;,
                           &quot;female genital tract polyp&quot;,
                           &quot;\\bpolyp\\b&quot;,
                           &quot;uterine leiomyoma&quot;,
                           &quot;hepatic hemangioma&quot;,
                           &quot;lobular capilliary hemangioma&quot;, 
                           &quot;hemangioma of subcutaneous tissue&quot;,
                           &quot;benign prostatic hyperplasia&quot;,
                           &quot;melanocytic nevus&quot;,
                           &quot;hemangioma&quot;,
                           &quot;lymphangioma&quot;,
                           &quot;vestibular schwannoma&quot;,
                           &quot;schwannoma&quot;,
                           &quot;skin lipoma&quot;, # likely benign ... 
                           &quot;lipoma&quot;,
                           &quot;hamartoma&quot;,
                           &quot;seborrheic keratosis&quot;,
                           &quot;actinic keratosis&quot;,
                           &quot;keratosis&quot;,
                           &quot;meningioma&quot;, # most are benign (80%)
                           &quot;common wart&quot;,
                           &quot;plantar wart&quot;,
                           &quot;penile Fibromatosis&quot;
                           )

other_benign_neoplasms = str_length_sort(other_benign_neoplasms)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    vec_to_grep_pattern(other_benign_neoplasms),
                    &quot;benign neoplasm&quot;)
         )</code></pre>
<div id="central-nervous-system-cancer" class="section level3"
number="3.0.1">
<h3><span class="header-section-number">3.0.1</span> Central nervous
system cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000326/descendants&quot;

cns_cancer_terms &lt;- get_descendants(url)

cns_cancer_terms = stringr::str_replace_all(cns_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>pattern = vec_to_grep_pattern(cns_cancer_terms)

gwas_study_info = gwas_study_info |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = pattern,
                                   &quot;central nervous system cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="cervical-cancer" class="section level3" number="3.0.2">
<h3><span class="header-section-number">3.0.2</span> Cervical
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_4362/descendants&quot;

cervical_cancer_terms &lt;- get_descendants(url)

cervical_cancer_terms = stringr::str_replace_all(cervical_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

cervical_cancer_terms = c(&quot;cervical intraepithelial neoplasia grade 2/3&quot;,
                          &quot;uterine cervical cancer in situ&quot;,
                          cervical_cancer_terms)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;uterine cervical cancer in situ&quot;),
                                   &quot;cervical cancer&quot;
                          )  
        )


gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(cervical_cancer_terms),
                                   &quot;cervical cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="colorectal-cancer" class="section level3" number="3.0.3">
<h3><span class="header-section-number">3.0.3</span> Colorectal
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0005575/descendants&quot;

colorectal_cancer_terms &lt;- get_descendants(url)

colorectal_cancer_terms = stringr::str_replace_all(colorectal_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

colorectal_cancer_terms= c(&quot;metastatic colorectal cancer&quot;,
                             &quot;rectum cancer&quot;,
                          colorectal_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(colorectal_cancer_terms),
                                   &quot;colorectal cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;colorectal mucinous adenocarcinoma&quot;),
                                   &quot;colorectal cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;metachronous colorectal adenoma&quot;),
                                   &quot;colorectal cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="endometrial-cancer" class="section level3" number="3.0.4">
<h3><span class="header-section-number">3.0.4</span> Endometrial
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0011962/descendants&quot;

endometrial_cancer_terms &lt;- get_descendants(url)

# also: http://www.ebi.ac.uk/efo/EFO_1001514: endometrial endometrioid carcinoma
endometrial_cancer_terms = c(&quot;endometrial endometrioid carcinoma&quot;,
                             endometrial_cancer_terms)

endometrial_cancer_terms = stringr::str_replace_all(endometrial_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(endometrial_cancer_terms),
                                   &quot;endometrial cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="esophageal-cancer" class="section level3" number="3.0.5">
<h3><span class="header-section-number">3.0.5</span> Esophageal
cancer</h3>
<pre class="r"><code>esophageal_cancer_terms &lt;- c(&quot;esophageal adenocarcinoma&quot;,
                             &quot;esophageal squamous cell cancer&quot;)


gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(esophageal_cancer_terms),
                                   &quot;esophageal cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="eye-cancer-to-add" class="section level3" number="3.0.6">
<h3><span class="header-section-number">3.0.6</span> Eye cancer (to
add)</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002236/descendants&quot;

ocular_cancer_terms &lt;- get_descendants(url)

ocular_cancer_terms = stringr::str_replace_all(ocular_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(ocular_cancer_terms),
                                   &quot;ocular cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="gallbladder-and-biliary-tract-cancer" class="section level3"
number="3.0.7">
<h3><span class="header-section-number">3.0.7</span> Gallbladder and
biliary tract cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                         vec_to_grep_pattern(&quot;cancer of gallbladder and extrahepatic biliary tract&quot;),
                         &quot;gallbladder and biliary tract cancer&quot;
         )
         )</code></pre>
</div>
<div id="hodgkin-lymphoma" class="section level3" number="3.0.8">
<h3><span class="header-section-number">3.0.8</span> Hodgkin
lymphoma</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms ,
                          vec_to_grep_pattern(&quot;nodular sclerosis hodgkin lymphoma&quot;),
                          &quot;hodgkins lymphoma&quot;
         ))</code></pre>
</div>
<div id="head-and-neck-cancer" class="section level3" number="3.0.9">
<h3><span class="header-section-number">3.0.9</span> Head and neck
cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms ,
                          vec_to_grep_pattern(&quot;head and neck squamous cell cancer&quot;),
                          &quot;head and neck cancer, squamous cell cancer&quot;
         ))</code></pre>
</div>
<div id="intestinal-cancer-non--colorectal" class="section level3"
number="3.0.10">
<h3><span class="header-section-number">3.0.10</span> Intestinal cancer
(non- colorectal)</h3>
<pre class="r"><code>intestinal_cancer_terms &lt;- c(&quot;small intestine cancer&quot;,
                           &quot;small bowel cancer&quot;,
                           &quot;small intestine cancer&quot;)</code></pre>
</div>
<div id="kidney-cancer" class="section level3" number="3.0.11">
<h3><span class="header-section-number">3.0.11</span> Kidney cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_263/descendants&quot;


kidney_cancer_terms &lt;- get_descendants(url)

kidney_cancer_terms = c(&quot;renal cell carcinoma&quot;,
                       &quot;clear cell renal carcinoma&quot;,
                       &quot;clear cell renal cell carcinoma&quot;,
                       kidney_cancer_terms)

kidney_cancer_terms = stringr::str_replace_all(kidney_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

kidney_cancer_terms =  stringr::str_replace_all(kidney_cancer_terms,
                          vec_to_grep_pattern(&quot;renal cancer&quot;),
                          &quot;kidney cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(kidney_cancer_terms),
                                   &quot;kidney cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="laryngeal-cancer" class="section level3" number="3.0.12">
<h3><span class="header-section-number">3.0.12</span> Laryngeal
cancer</h3>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = 
                                    vec_to_grep_pattern(
                                      c(&quot;laryngeal squamous cell cancer&quot;,
                                      &quot;laryngeal cancer&quot;)
                                      ),
                                   &quot;larynx cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="leukemia" class="section level3" number="3.0.13">
<h3><span class="header-section-number">3.0.13</span> Leukemia</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000565/descendants&quot;

leukemia_terms &lt;- get_descendants(url)

leukemia_terms &lt;- c(&quot;b-cell acute lymphoblastic leukemia with t\\(1;19\\)\\(q23;p13.3\\); e2a-pbx1 \\(tcf3-pbx1\\)&quot;,
                    leukemia_terms)

leukemia_terms = stringr::str_replace_all(leukemia_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(leukemia_terms),
                                   &quot;leukemia&quot;
                          )  
        )</code></pre>
</div>
<div id="lip-and-oral-cavity-cancer" class="section level3"
number="3.0.14">
<h3><span class="header-section-number">3.0.14</span> Lip and oral
cavity cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0005570/descendants&quot;

lip_oral_cavity_cancer_terms &lt;- get_descendants(url)

lip_oral_cavity_cancer_terms = stringr::str_replace_all(lip_oral_cavity_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(lip_oral_cavity_cancer_terms),
                                   &quot;lip and oral cavity cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="liver-cancer" class="section level3" number="3.0.15">
<h3><span class="header-section-number">3.0.15</span> Liver cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002691/descendants&quot;

liver_cancer_terms &lt;- get_descendants(url)

liver_cancer_terms = stringr::str_replace_all(liver_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)


liver_cancer_terms = c(&quot;hepatitis virus-related liver cancer&quot;,
                       liver_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(liver_cancer_terms),
                                   &quot;liver cancer&quot;
                          )  
        )

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;hepatitis virus-related liver cancer&quot;),
                                   &quot;liver cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="lung-cancer" class="section level3" number="3.0.16">
<h3><span class="header-section-number">3.0.16</span> Lung cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0008903/descendants&quot;

lung_cancer_terms &lt;- get_descendants(url)

lung_cancer_terms = stringr::str_replace_all(lung_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(lung_cancer_terms),
                                   &quot;lung cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="ovarian-cancer" class="section level3" number="3.0.17">
<h3><span class="header-section-number">3.0.17</span> Ovarian
cancer</h3>
<pre class="r"><code>url &lt;-  &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0008170/descendants&quot;

ovarian_cancer_terms &lt;- get_descendants(url)

ovarian_cancer_terms = stringr::str_replace_all(ovarian_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

ovarian_cancer_terms = c(&quot;high grade serous ovarian cancer&quot;,
                         &quot;high grade ovarian serous adenocarcinoma&quot;,
                         &quot;high grade ovarian cancer&quot;,
                         &quot;high grade ovarian cancers&quot;,
                         &quot;ovarian endometrioid cancer&quot;, # http://www.ebi.ac.uk/efo/EFO_1001515 - ovarian edometrioid carcinoma
                         
                         &quot;ovarian serous cancer&quot;, # http://www.ebi.ac.uk/efo/EFO_1001516 - ovarian serous carcinoma
                       ovarian_cancer_terms
                       )</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(ovarian_cancer_terms),
                                   &quot;ovarian cancer&quot;
                  
        )
 )
        
gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;high grade ovarian cancer&quot;),
                                   &quot;ovarian cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="pancreatic-cancer" class="section level3" number="3.0.18">
<h3><span class="header-section-number">3.0.18</span> Pancreatic
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0009831/descendants&quot;

pancreatic_cancer_terms &lt;- get_descendants(url)

pancreatic_cancer_terms = stringr::str_replace_all(pancreatic_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(pancreatic_cancer_terms),
                                   &quot;pancreatic cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="peripheral-nervous-system-cancer" class="section level2"
number="3.1">
<h2><span class="header-section-number">3.1</span> Peripheral nervous
system cancer</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0021089/descendants&quot;

peripheral_nervous_system_cancer_terms &lt;- get_descendants(url)

peripheral_nervous_system_cancer_terms = stringr::str_replace_all(peripheral_nervous_system_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(peripheral_nervous_system_cancer_terms),
                                   &quot;peripheral nervous system cancer&quot;
                          )  
        )</code></pre>
<div id="prostate-cancer" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Prostate
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_10283/descendants&quot;

prostate_cancer_terms &lt;- get_descendants(url)

prostate_cancer_terms = stringr::str_replace_all(prostate_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

prostate_cancer_terms = c(&quot;grade iii prostatic intraepithelial neoplasia&quot;,
                          &quot;metastatic prostate cancer&quot;,
                          prostate_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = vec_to_grep_pattern(prostate_cancer_terms),
                                   &quot;prostate cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="mesothelioma" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Mesothelioma</h3>
<pre class="r"><code>mesothelioma_terms = c(&quot;pleural mesothelioma&quot;,
                       &quot;malignant pleural mesothelioma&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(mesothelioma_terms),
                                   &quot;mesothelioma&quot;
                          )  
        )</code></pre>
</div>
<div id="neuroendocrine-tumor" class="section level3" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> Neuroendocrine
tumor</h3>
<pre class="r"><code>neuroendo_terms &lt;- c(&quot;pulmonary neuroendocrine tumor&quot;,
                     &quot;small intestine neuroendocrine tumor&quot;,
                     &quot;pancreatic neuroendocrine tumor&quot;,
                     &quot;carcinoid tumor&quot; #http://www.ebi.ac.uk/efo/EFO_0004243
)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                    pattern = vec_to_grep_pattern(neuroendo_terms),
                                   &quot;neuroendocrine tumor&quot;
                          )  
        )</code></pre>
</div>
<div id="non-hodgkins-lymphoma" class="section level3" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> Non-Hodgkins
lymphoma</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0005952/descendants&quot;

nhl_terms &lt;- get_descendants(url)

nhl_terms = stringr::str_replace_all(nhl_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)



nhl_terms = c(&quot;central nervous system non-hodgkin lymphoma&quot;,
              &quot;lymphoblastic lymphoma&quot;,
              &quot;extranodal nasal nk/t cell lymphoma&quot;, # https://www.ebi.ac.uk/ols4/ontologies/ordo/classes/http%253A%252F%252Fwww.orpha.net%252FORDO%252FOrphanet_86879
              &quot;follicular lymphoma&quot;, # http://purl.obolibrary.org/obo/DOID_0050873
              &quot;marginal zone b-cell lymphoma&quot;,
              &quot;diffuse large b-cell lymphoma&quot;,
              nhl_terms)

# also likely that reticulum cell sarcoma is NHL
# see; http://www.ebi.ac.uk/efo/EFO_0005287
# https://pubmed.ncbi.nlm.nih.gov/6328875/
nhl_terms = c(&quot;reticulum cell sarcoma&quot;,
              nhl_terms)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(nhl_terms),
                                   &quot;non-hodgkins lymphoma&quot;
                          )  
        )</code></pre>
</div>
<div id="non-melanoma-skin-cancer" class="section level3"
number="3.1.5">
<h3><span class="header-section-number">3.1.5</span> Non-melanoma skin
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0009260/descendants&quot;

non_melanoma_skin_cancer_terms &lt;- get_descendants(url)

non_melanoma_skin_cancer_terms = stringr::str_replace_all(non_melanoma_skin_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(non_melanoma_skin_cancer_terms),
                                   &quot;non-melanoma skin cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="other-pharygnx-cancer" class="section level3" number="3.1.6">
<h3><span class="header-section-number">3.1.6</span> Other pharygnx
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_0060119/descendants&quot;

other_pharynx_cancer_terms &lt;- get_descendants(url)

other_pharynx_cancer_terms = stringr::str_replace_all(other_pharynx_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

other_pharynx_cancer_terms = c(&quot;hypopharyngeal cancer&quot;,
                               other_pharynx_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(other_pharynx_cancer_terms),
                                   &quot;other pharynx cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="soft-tissue-sarcoma" class="section level3" number="3.1.7">
<h3><span class="header-section-number">3.1.7</span> Soft tissue
sarcoma</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_1001968/descendants&quot;

soft_tissue_sarcoma_terms &lt;- get_descendants(url)

soft_tissue_sarcoma_terms = stringr::str_replace_all(soft_tissue_sarcoma_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

soft_tissue_sarcoma_terms = c(&quot;kaposis sarcoma&quot;,
                              &quot;iatrogenic kaposis sarcoma&quot;,
                              soft_tissue_sarcoma_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(soft_tissue_sarcoma_terms),
                                   &quot;soft tissue sarcoma&quot;
                          )  
        )

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;sarcoma, soft tissue sarcoma&quot;),
                                   &quot;soft tissue sarcoma&quot;
                          )
 )</code></pre>
<div id="ewing-sarcoma" class="section level4" number="3.1.7.1">
<h4><span class="header-section-number">3.1.7.1</span> Ewing
sarcoma</h4>
<pre class="r"><code># can be either bone or soft tissue sarcoma
# hard to tell from these studies: 
gwas_study_info |&gt; 
  filter(grepl(&quot;ewing&quot;, l1_all_disease_terms))  |&gt; 
  select(PUBMED_ID, `DISEASE/TRAIT`, COHORT, STUDY)</code></pre>
</div>
</div>
<div id="stomach-cancer" class="section level3" number="3.1.8">
<h3><span class="header-section-number">3.1.8</span> Stomach cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_10534/descendants&quot;

stomach_cancer_terms &lt;- get_descendants(url)

stomach_cancer_terms = stringr::str_replace_all(stomach_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

stomach_cancer_terms = c(
                          &quot;diffuse stomach cancer&quot;,
                          &quot;gastric cancer&quot;,
                          &quot;gastric intestinal type adenocarcinoma&quot;,
                          &quot;gastric cardia cancer&quot;,
                          &quot;cardia cancer&quot;,
                          stomach_cancer_terms
                          )</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(stomach_cancer_terms),
                                   &quot;stomach cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;diffuse stomach cancer&quot;),
                                   &quot;stomach cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="squamous-cell-carcinoma" class="section level3" number="3.1.9">
<h3><span class="header-section-number">3.1.9</span> Squamous cell
carcinoma</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;cutaneous squamous cell cancer&quot;),
                                   &quot;non-melanoma skin cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="testicular-cancer" class="section level3" number="3.1.10">
<h3><span class="header-section-number">3.1.10</span> Testicular
cancer</h3>
</div>
<div id="thyroid-cancer" class="section level3" number="3.1.11">
<h3><span class="header-section-number">3.1.11</span> Thyroid
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_1781/descendants&quot;

thyroid_cancer_terms &lt;- get_descendants(url)

thyroid_cancer_terms = stringr::str_replace_all(thyroid_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

thyroid_cancer_terms = c(&quot;differentiated thyroid cancer&quot;,
                         thyroid_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(thyroid_cancer_terms),
                                   &quot;thyroid cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="uterine-cancer" class="section level3" number="3.1.12">
<h3><span class="header-section-number">3.1.12</span> Uterine
cancer</h3>
<pre class="r"><code>uterine_cancer_terms &lt;- c(&quot;uterine corpus cancer&quot;,
                          &quot;uterine adnexa cancer&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(uterine_cancer_terms),
                                   &quot;uterine cancer&quot;
                          )  
        )</code></pre>
</div>
</div>
</div>
<div id="reducing-non-specific-cancer-terms" class="section level1"
number="4">
<h1><span class="header-section-number">4</span> Reducing non-specific
cancer terms</h1>
<div id="brain-neoplasm" class="section level3" number="4.0.1">
<h3><span class="header-section-number">4.0.1</span> Brain neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;brain neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique() 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;brain neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;central nervous system cancer&quot;,
                 l1_all_disease_terms
         )
         )

# still leaves one study (with Brain Tumor)

gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;brain neoplasm&quot;) |&gt; 
  pull(PUBMED_ID) |&gt;
  unique() 

# from paper sup tables, ICD-10 code of brain tumor term is C71 - malignant neoplasm of brain
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;brain neoplasm&quot; &amp; 
                 PUBMED_ID == 34594039,
                 &quot;central nervous system cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="breast-neoplasm" class="section level3" number="4.0.2">
<h3><span class="header-section-number">4.0.2</span> Breast
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;breast neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;breast neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;breast cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="bone-neoplasm" class="section level3" number="4.0.3">
<h3><span class="header-section-number">4.0.3</span> Bone neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;bone neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;bone neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;bone cancer&quot;,
                 l1_all_disease_terms
         )
         )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;bone neoplasm&quot; &amp; 
                 grepl(&quot;benign&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="cecal-neoplasm" class="section level3" number="4.0.4">
<h3><span class="header-section-number">4.0.4</span> Cecal neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;cecal neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;cecal neoplasm&quot; &amp; 
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;colorectal cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="colonic-neoplasm" class="section level3" number="4.0.5">
<h3><span class="header-section-number">4.0.5</span> Colonic
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;colonic neoplasm&quot;) |&gt; 
  select(MAPPED_TRAIT, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct() |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;malignant|cancer&quot;, `DISEASE/TRAIT`, ignore.case = T) &amp;
                l1_all_disease_terms == &quot;colonic neoplasm&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;colonic neoplasm&quot;),
                          &quot;colorectal cancer&quot;),
                l1_all_disease_terms
         )
  )

# also specific example where measuring rectal cancer vs colon cancer

gwas_study_info |&gt; 
  filter(grepl(&quot;colonic neoplasm&quot;, l1_all_disease_terms)) |&gt; 
  select(MAPPED_TRAIT, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct() |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(STUDY_ACCESSION == &quot;GCST90179122&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;colonic neoplasm&quot;),
                          &quot;colorectal cancer&quot;),
                l1_all_disease_terms
         )
  )</code></pre>
</div>
<div id="endometrial-neoplasm" class="section level3" number="4.0.6">
<h3><span class="header-section-number">4.0.6</span> Endometrial
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;endometrial neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;endometrial neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;endometrial cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="esophageal-neoplasm" class="section level3" number="4.0.7">
<h3><span class="header-section-number">4.0.7</span> Esophageal
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;neoplasm of esophagus&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;neoplasm of esophagus&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;esophageal cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="eye-neoplasm" class="section level3" number="4.0.8">
<h3><span class="header-section-number">4.0.8</span> Eye neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;eye neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;eye neoplasm&quot; &amp; 
                 grepl(&quot;cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;ocular cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="gallbladder-neoplasm" class="section level3" number="4.0.9">
<h3><span class="header-section-number">4.0.9</span> Gallbladder
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;gallbladder neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;gallbladder neoplasm&quot; &amp; 
                 grepl(&quot;cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;gallbladder and biliary tract cancer&quot;,
                 l1_all_disease_terms
         )
         )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;gallbladder neoplasm&quot; &amp; 
                `DISEASE/TRAIT` == &quot;Gallbladder adenomyomatosis&quot;,
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )

# also specific example where measuring sclerosing cholangitis &amp; gallbladder cancer
gwas_study_info |&gt;
  filter(grepl(&quot;gallbladder neoplasm&quot;, l1_all_disease_terms)) |&gt;
  select(STUDY_ACCESSION, `DISEASE/TRAIT`)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(STUDY_ACCESSION == &quot;GCST005857&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;gallbladder neoplasm&quot;),
                          &quot;gallbladder and biliary tract cancer&quot;),
                l1_all_disease_terms
         ))</code></pre>
</div>
<div id="glioma-can-be-benign-or-malignant" class="section level3"
number="4.0.10">
<h3><span class="header-section-number">4.0.10</span> Glioma (can be
benign or malignant)</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;central nervous system cancer, glioma&quot;,
                          &quot;central nervous system cancer&quot;
         ))


gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;glioma&quot;) |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_URI, `DISEASE/TRAIT`) |&gt; 
  distinct()

# assume where measure survival and glioma, it is cancer
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;survival&quot;, MAPPED_TRAIT, ignore.case = T),
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;glioma&quot;,
                          &quot;central nervous system cancer&quot;),
         l1_all_disease_terms
         )
  )

# assme where measure is grade, it is cancer
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;grade&quot;, `DISEASE/TRAIT`, ignore.case = T),
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;glioma&quot;,
                          &quot;central nervous system cancer&quot;),
         l1_all_disease_terms
         )
  )

# Adult diffuse glioma - assume maglignant
# https://pmc.ncbi.nlm.nih.gov/articles/PMC9245936/
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;adult diffuse&quot;, `DISEASE/TRAIT`, ignore.case = T),
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;glioma&quot;,
                          &quot;central nervous system cancer&quot;),
         l1_all_disease_terms
         )
  )

gwas_study_info |&gt;
  filter(`DISEASE/TRAIT` == &quot;Glioma (pediatric/youth onset)&quot;) |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_URI, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct()

# from paper; seems malignant
# https://pubmed.ncbi.nlm.nih.gov/31040135/

gwas_study_info = 
gwas_study_info |&gt;
  mutate(l1_all_disease_terms  =
         ifelse(STUDY_ACCESSION == &quot;GCST008912&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;glioma&quot;),
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms
         ))


gwas_study_info = 
gwas_study_info |&gt;
  mutate(l1_all_disease_terms  =
         ifelse(`DISEASE/TRAIT` == &quot;Glioblastoma&quot; &amp; l1_all_disease_terms == &quot;glioma&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;glioma&quot;),
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms
         ))


# for pubmed id: 22886559 
# majority (~90%) graded glioma, gliobastoma and Oligodendroglioma
# so assume malignant
gwas_study_info =
gwas_study_info |&gt;
  mutate(l1_all_disease_terms  =
         ifelse(PUBMED_ID == 22886559 &amp; l1_all_disease_terms == &quot;glioma&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;glioma&quot;),
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms)
  )


# for pubmed id: 29743610
# majority (~60%) are glioblastoma
# so assume malignant

gwas_study_info =
gwas_study_info |&gt;
  mutate(l1_all_disease_terms  =
         ifelse(PUBMED_ID == 29743610 &amp; l1_all_disease_terms == &quot;glioma&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;glioma&quot;),
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms)
  )

# for pubmed id: 36810956
# seems to primarily include high grade glioma
# so assume malignant
gwas_study_info = 
gwas_study_info |&gt;
  mutate(l1_all_disease_terms  =
         ifelse(PUBMED_ID == 36810956 &amp; l1_all_disease_terms == &quot;glioma&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;glioma&quot;),
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms)
  )

# pubmed id: 30714141 
# considers glioma cancer
gwas_study_info =
gwas_study_info |&gt;
  mutate(l1_all_disease_terms  =
         ifelse(PUBMED_ID == 30714141 &amp; l1_all_disease_terms == &quot;glioma&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;glioma&quot;),
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms)
  )

# pubmed id: 34319593
# considers glioma a maglignant tumor
gwas_study_info =
gwas_study_info |&gt;
  mutate(l1_all_disease_terms  =
         ifelse(PUBMED_ID == 34319593 &amp; l1_all_disease_terms == &quot;glioma&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;glioma&quot;),
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms)
  )</code></pre>
</div>
<div id="glottis-neoplasm" class="section level3" number="4.0.11">
<h3><span class="header-section-number">4.0.11</span> Glottis
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;glottis neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;glottis neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;larynx cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="kidney-neoplasm" class="section level3" number="4.0.12">
<h3><span class="header-section-number">4.0.12</span> Kidney
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;kidney neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;kidney neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;kidney cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="laryngeal-neoplasm" class="section level3" number="4.0.13">
<h3><span class="header-section-number">4.0.13</span> Laryngeal
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;laryngeal neoplasm&quot;) |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_URI, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct()

# one study - 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(STUDY_ACCESSION == &quot;GCST90041889&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;laryngeal neoplasm&quot;),
                          &quot;larynx cancer&quot;),
                l1_all_disease_terms
         ))</code></pre>
</div>
<div id="liver-neoplasm" class="section level3" number="4.0.14">
<h3><span class="header-section-number">4.0.14</span> Liver
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;liver neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;liver neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;liver cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="lung-neoplasm" class="section level3" number="4.0.15">
<h3><span class="header-section-number">4.0.15</span> Lung neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;lung neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;lung neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;lung cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="lymphoid-neoplasm" class="section level3" number="4.0.16">
<h3><span class="header-section-number">4.0.16</span> Lymphoid
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;lymphoid neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;lymphoid neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;malignant lymphoid tumor&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="meningeal-neoplasm" class="section level3" number="4.0.17">
<h3><span class="header-section-number">4.0.17</span> Meningeal
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;meningeal neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;meningeal neoplasm&quot; &amp; 
                 grepl(&quot;benign&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="mature-b-cell-neoplasm" class="section level3" number="4.0.18">
<h3><span class="header-section-number">4.0.18</span> Mature b-cell
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;neoplasm of mature b-cells&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;neoplasm of mature b-cells&quot; &amp; 
                 `DISEASE/TRAIT` == &quot;Follicular lymphoma&quot;,
                 &quot;non-hodgkins lymphoma&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="mouth-neoplasm" class="section level3" number="4.0.19">
<h3><span class="header-section-number">4.0.19</span> Mouth
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;mouth neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;mouth neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;lip and oral cavity cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="myeloid-neoplasm" class="section level3" number="4.0.20">
<h3><span class="header-section-number">4.0.20</span> Myeloid
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;myeloid neoplasm&quot;, l1_all_disease_terms)) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;myeloid neoplasm&quot; &amp; 
                 grepl(&quot;Myeloid leukemia&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;leukemia&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="neuroendocrine-neoplasm" class="section level3"
number="4.0.21">
<h3><span class="header-section-number">4.0.21</span> Neuroendocrine
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;neuroendocrine neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;neuroendocrine neoplasm&quot; &amp; 
                 grepl(&quot;PheCode 209&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;neuroendocrine tumor&quot;,
                 l1_all_disease_terms
         )
         )

# ? to double check: neuroendocrine tumor is malignant
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;neuroendocrine neoplasm&quot; &amp; 
                 grepl(&quot;neuroendocrine tumor&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;neuroendocrine tumor&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="ovarian-neoplasm" class="section level3" number="4.0.22">
<h3><span class="header-section-number">4.0.22</span> Ovarian
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;ovarian neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;ovarian neoplasm&quot; &amp;
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;ovarian cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="nasopharyngeal-neoplasm" class="section level3"
number="4.0.23">
<h3><span class="header-section-number">4.0.23</span> Nasopharyngeal
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;nasopharyngeal neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;nasopharyngeal neoplasm&quot; &amp; 
                 grepl(&quot;carcinoma|cancer|malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;nasopharyngeal cancer&quot;,
                 l1_all_disease_terms
         )
         )

gwas_study_info |&gt; 
  filter(grepl(&quot;nasopharyngeal neoplasm&quot;, l1_all_disease_terms)) |&gt; 
  pull(`DISEASE/TRAIT`)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;nasopharyngeal neoplasm&quot; &amp; 
                 grepl(&quot;nasopharyngeal carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;nasopharyngeal cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="pancreatic-neoplasm" class="section level3" number="4.0.24">
<h3><span class="header-section-number">4.0.24</span> Pancreatic
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;pancreatic neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)

# Intraductal papillary mucinous neoplasm of the pancreas is a benign precursor lesion

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
         ifelse(l1_all_disease_terms == &quot;pancreatic neoplasm&quot; &amp; 
                STUDY_ACCESSION == &quot;GCST90104145&quot;,
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="sigmoid-neoplasm" class="section level3" number="4.0.25">
<h3><span class="header-section-number">4.0.25</span> Sigmoid
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;sigmoid neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;sigmoid neoplasm&quot; &amp; 
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;colorectal cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="skin-neoplasm" class="section level3" number="4.0.26">
<h3><span class="header-section-number">4.0.26</span> Skin neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;skin neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;skin neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;non-melanoma skin cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="stomach-neoplasm" class="section level3" number="4.0.27">
<h3><span class="header-section-number">4.0.27</span> Stomach
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;stomach neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;stomach neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;stomach cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="testicular-neoplasm" class="section level3" number="4.0.28">
<h3><span class="header-section-number">4.0.28</span> Testicular
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;testicular neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
        ifelse(l1_all_disease_terms == &quot;testicular neoplasm&quot; &amp;
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;testicular cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="tongue-neoplasm" class="section level3" number="4.0.29">
<h3><span class="header-section-number">4.0.29</span> Tongue
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;tongue neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;tongue neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;lip and oral cavity cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="uterine-neoplasm" class="section level3" number="4.0.30">
<h3><span class="header-section-number">4.0.30</span> Uterine
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;uterine neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;uterine neoplasm&quot; &amp; 
                 grepl(&quot;benign&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="urogenital-neoplasm" class="section level3" number="4.0.31">
<h3><span class="header-section-number">4.0.31</span> Urogenital
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;urogenital neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;urogenital neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;urogenital cancer&quot;,
                 l1_all_disease_terms
         )
         )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;urogenital neoplasm&quot; &amp; 
                 grepl(&quot;benign&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
<div id="vulvar-neoplasm" class="section level4" number="4.0.31.1">
<h4><span class="header-section-number">4.0.31.1</span> Vulvar
neoplasm</h4>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;vulvar neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;vulvar neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;vulvar cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
</div>
<div id="ocular-melanoma" class="section level3" number="4.0.32">
<h3><span class="header-section-number">4.0.32</span> Ocular
Melanoma</h3>
<pre class="r"><code>ocular_melanoma_terms &lt;- c(&quot;uveal melanoma&quot;,
                           &quot;uveal melanoma disease severity&quot;,
                           &quot;epithelioid cell uveal melanoma&quot;,
                           &quot;choroidal melanoma&quot;,
                           &quot;ocular melanoma disease severity&quot;
                           )

ocular_melanoma_terms = str_length_sort(ocular_melanoma_terms)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(ocular_melanoma_terms),
                          &quot;ocular melanoma&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;ocular melanoma disease severity&quot;),
                          &quot;ocular melanoma&quot;
         ))</code></pre>
</div>
</div>
<div id="other" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Other …</h1>
<div id="benign-neoplasm-colorectal-cancer" class="section level3"
number="5.0.1">
<h3><span class="header-section-number">5.0.1</span> Benign neoplasm,
colorectal cancer</h3>
<pre class="r"><code>gwas_study_info |&gt;
  filter(l1_all_disease_terms == &quot;benign neoplasm, colorectal cancer&quot;) |&gt; 
  select(MAPPED_TRAIT, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(STUDY_ACCESSION == &quot;GCST90093303&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;benign neoplasm, colorectal cancer&quot;),
                          &quot;colorectal cancer&quot;),
                l1_all_disease_terms
         ))</code></pre>
</div>
<div id="more-other-pharynx-cancer" class="section level3"
number="5.0.2">
<h3><span class="header-section-number">5.0.2</span> More other pharynx
cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;larynx cancer, pharynx cancer&quot;),
                          &quot;larynx cancer, other pharynx cancer&quot;
         ))</code></pre>
</div>
<div id="malignant-melanoma-of-skin" class="section level3"
number="5.0.3">
<h3><span class="header-section-number">5.0.3</span> Malignant melanoma
of skin</h3>
<div id="cutaneous-melanoma-to-malignant-melanoma-of-skin"
class="section level4" number="5.0.3.1">
<h4><span class="header-section-number">5.0.3.1</span> Cutaneous
melanoma to malignant melanoma of skin</h4>
<pre class="r"><code>gwas_study_info =
  gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;cutaneous melanoma&quot;),
                          &quot;malignant melanoma of skin&quot;
         ))</code></pre>
</div>
<div id="dealing-with-studies-just-labelled-as-melanoma"
class="section level4" number="5.0.3.2">
<h4><span class="header-section-number">5.0.3.2</span> Dealing with
studies just labelled as “melanoma”</h4>
<pre class="r"><code>gwas_study_info |&gt;
 filter(l1_all_disease_terms == &quot;melanoma&quot;) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

# checked UKB data field 40006 (ICD10 codes) 
# https://biobank.ctsu.ox.ac.uk/ukb/field.cgi?id=40006
# malignant melanoma of skin includes: 
# Malignant melanoma of trunk 
# Malignant melanoma of upper limb, including shoulder
# Malignant melanoma of lower limb, including hip

# checked UKB data field 20001 
# https://biobank.ctsu.ox.ac.uk/ukb/field.cgi?id=20001
# malignant melanoma is a subcategory of skin cancer


malignant_skin_melanoma &lt;- c(&quot;ICD10 C43&quot;,
                             &quot;survival in skin melanoma&quot;,
                             &quot;Skin melanoma specific survival&quot;,
                             &quot;malignant melanoma of skin&quot;,
                             &quot;malignant melanoma of trunk&quot;,
                             &quot;Malignant melanoma \\(UKB data field 20001\\)&quot;,
                             &quot;malignant melanoma \\(UKB data field 20001_1059\\)&quot;,
                             &quot;malignant melanoma of upper limb, including shoulder&quot;,
                             &quot;malignant melanoma of lower limb, including hip&quot;,
                             &quot;ICD10 D03&quot;, # skin melanoma in situ 
                             &quot;Melanoma in situ \\(UKB data field 40006\\)&quot;
)


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;melanoma&quot; &amp;
                grepl(paste0(malignant_skin_melanoma, collapse = &quot;|\\b&quot;),
                      `DISEASE/TRAIT`, 
                      ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;melanoma&quot;),
                          &quot;malignant melanoma of skin&quot;),
                l1_all_disease_terms
         ))

# UKBB malignant melanoma of skin
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;melanoma&quot; &amp;
                grepl(&quot;UKBB&quot;, COHORT, ignore.case = T) &amp;
                grepl(&quot;malignant melanoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;melanoma&quot;),
                          &quot;malignant melanoma of skin&quot;),
                l1_all_disease_terms
         ))

# cutaneous melanoma in title
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;melanoma&quot; &amp; 
                grepl(&quot;\\bcutaneous melanoma&quot;, STUDY, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;melanoma&quot;), 
                          &quot;malignant melanoma of skin&quot;),
                l1_all_disease_terms
         ))
                  

gwas_study_info |&gt;
 filter(l1_all_disease_terms == &quot;\\bmelanoma&quot;) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

gwas_study_info |&gt;
 filter(l1_all_disease_terms == &quot;\\bmelanoma&quot;) |&gt;
  pull(PUBMED_ID) |&gt;
  unique()

# Checking the clinical trials that make up pubmed id 27023328
# https://clinicaltrials.gov/study/NCT01153763 - cutaneous melanoma
# https://clinicaltrials.gov/study/NCT01266967 - not specified, likely skin melanoma by MeSH terms
# https://clinicaltrials.gov/study/NCT01227889 - not specified, likely skin melanoma by MeSH terms
# https://clinicaltrials.gov/study/NCT01584648 - cutaneous melanoma
# https://clinicaltrials.gov/study/NCT01597908 - cutaneous melanoma
# thus, likely malignant melanoma of skin
                             
# for pubmed ID 21983785, 
# seems likely malignant melanoma of skin
# as test in situ vs invasive, and use non-skin cancer controls
# https://pmc.ncbi.nlm.nih.gov/articles/PMC3227560/#SM

# for pubmed id: 23455637 - uses one of the same cohorts as 21983785 (genoMEL)
# thus likely malignant melanoma of skin
# https://pubmed.ncbi.nlm.nih.gov/23455637/

# for pubmed id: 21706340
# Cutaneous malignant melanoma
# therefore, malignant melanoma of skin

# pubmed id: 19578364 also uses GenoMEL consortium
# therefore, malignant melanoma of skin

# pubmed id: 18488026
# cutaneous malignant melanoma
# therefore, malignant melanoma of skin

# pubmed id: 21983787
# also uses GenoMEL consortium
# therefore, malignant melanoma of skin

# pubmed id: 28212542
# cutaneous melanoma
# therefore, malignant melanoma of skin

# pubmed id: 24980573
# skin cancer melanoma discussion 
# therefore, malignant melanoma of skin

# pubmed id: 35626014
# not entirely clear, but likely malignant melanoma of skin

# pubmed id: 34724200
# &quot;current study focused on melanomas of the skin&quot;
# hence, malignant melanoma of skin

# pubmed id: 34290314
# lists ICD10 codes as C43 (malignant melanoma of skin) 
# for melanoma - thus malignant melanoma of skin

# pubmed id: 36064556
# lists cutaneous melanoma

malignant_skin_melanoma_studies &lt;- c(27023328,
                                     21983785,
                                     23455637,
                                     21706340,
                                     19578364,
                                     18488026,
                                     21983787,
                                     28212542,
                                     24980573,
                                     35626014,
                                     34724200,
                                     34290314,
                                     36064556)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(PUBMED_ID %in% malignant_skin_melanoma_studies &amp;
                grepl(&quot;\\bmelanoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;melanoma&quot;),
                          &quot;malignant melanoma of skin&quot;),
                l1_all_disease_terms
         ))


# honestly not sure of pubmed id: 32887889
# ? guess but likely malignant melanoma of skin
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(PUBMED_ID == 32887889 &amp;
                grepl(&quot;\\bmelanoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;melanoma&quot;),
                          &quot;malignant melanoma of skin&quot;),
                l1_all_disease_terms
         ))


# also not of pubmed id: 33409738
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(PUBMED_ID == 33409738 &amp;
                grepl(&quot;\\bmelanoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;melanoma&quot;),
                          &quot;malignant melanoma of skin&quot;),
                l1_all_disease_terms
         ))</code></pre>
</div>
</div>
<div id="in-situ-cancer" class="section level3" number="5.0.4">
<h3><span class="header-section-number">5.0.4</span> In situ cancer</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;in situ&quot;, l1_all_disease_terms)) |&gt; 
  pull(l1_all_disease_terms) |&gt;
  unique()

# cervical cancer
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;uterine cervix cancer in situ&quot;),
                          &quot;cervical cancer&quot;
         ))


gwas_study_info |&gt; 
  filter(grepl(&quot;in situ&quot;, l1_all_disease_terms))  |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique()

# strictly speaking is unspecified skin cancer
# likely ICD10 DO4 is non-melanoma skin cancer
# PheCode 172.3 - maps to D04.9
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;in situ&quot;, l1_all_disease_terms) &amp; 
                grepl(&quot;ICD10 D04|PheCode 172.3&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;skin cancer in situ&quot;),
                          &quot;non-melanoma skin cancer&quot;),
                l1_all_disease_terms
         )
  )

# in situ cancer -&gt; to cancer

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;in situ cancer&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;in situ cancer&quot;),
                          &quot;cancer&quot;),
                l1_all_disease_terms
         )
  )</code></pre>
</div>
<div id="non-specific-cancer-terms" class="section level3"
number="5.0.5">
<h3><span class="header-section-number">5.0.5</span> Non-specific cancer
terms</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms==&quot;cancer&quot;)  |&gt; 
  select(`DISEASE/TRAIT`, l1_all_disease_terms) |&gt;
  head()

# ICD10 Z85.4: Personal history of malignant neoplasm of genital organs
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;cancer&quot; &amp; 
                grepl(&quot;ICD10 Z85.4&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;cancer&quot;),
                          &quot;urogenital cancer&quot;),
                l1_all_disease_terms
         )
  )

# ICD10 Z85.1: Personal history of malignant neoplasm of trachea, bronchus and lung 
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;cancer&quot; &amp; 
                grepl(&quot;ICD10 Z85.1&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;cancer&quot;),
                           &quot;tracheal bronchus and lung cancer&quot;),
                l1_all_disease_terms
         )
  )

# ICD10 Z85.0: Personal history of malignant neoplasm of digestive organs
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;cancer&quot; &amp; 
                grepl(&quot;ICD10 Z85.0&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;cancer&quot;),
                          &quot;digestive system cancer&quot;),
                l1_all_disease_terms
         )
  )

# Cancer of intrathoracic organs (PheCode 164)

# Malignant neoplasm of retroperitoneum and peritoneum (PheCode 159.4)&quot;
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;cancer&quot; &amp; 
                grepl(&quot;PheCode 164|PheCode 159.4&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;cancer&quot;),
                          &quot;peritoneum cancer, retroperitoneal cancer&quot;),
                l1_all_disease_terms
         )
  )
# Malignant neoplasm of other and ill-defined sites within the digestive organs and peritoneum (PheCode 159)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(l1_all_disease_terms == &quot;cancer&quot; &amp; 
                grepl(&quot;PheCode 159$&quot;, `DISEASE/TRAIT`, ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;cancer&quot;),
                          &quot;digestive system cancer, peritoneum cancer&quot;),
                l1_all_disease_terms
         )
  )</code></pre>
</div>
<div id="peritoneum-cancer---peritoneal-cancer" class="section level3"
number="5.0.6">
<h3><span class="header-section-number">5.0.6</span> peritoneum cancer
-&gt; peritoneal cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;peritoneum cancer&quot;),
                          &quot;peritoneal cancer&quot;
         ))</code></pre>
</div>
<div id="bladder-tumor" class="section level3" number="5.0.7">
<h3><span class="header-section-number">5.0.7</span> Bladder tumor</h3>
<pre class="r"><code>gwas_study_info |&gt;
  filter(l1_all_disease_terms == &quot;bladder tumor&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
</div>
<div id="squamous-cell-cancer" class="section level3" number="5.0.8">
<h3><span class="header-section-number">5.0.8</span> Squamous cell
cancer</h3>
<pre class="r"><code>gwas_study_info |&gt;
    filter(grepl(&quot;squamous cell cancer&quot;, l1_all_disease_terms)) |&gt;
    select(`DISEASE/TRAIT`, l1_all_disease_terms) |&gt;
    distinct()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;lung cancer, squamous cell cancer&quot;),
                          &quot;lung cancer&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;esophageal cancer, squamous cell cancer&quot;),
                          &quot;esophageal cancer&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;head and neck cancer, squamous cell cancer&quot;),
                          &quot;head and neck cancer&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;head and neck cancer, pain, squamous cell cancer&quot;),
                          &quot;head and neck cancer, cancer pain&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;non-melanoma skin cancer, squamous cell cancer&quot;),
                          &quot;non-melanoma skin cancer&quot;
         ))</code></pre>
</div>
<div id="female-reproductive-organ" class="section level3"
number="5.0.9">
<h3><span class="header-section-number">5.0.9</span> Female reproductive
organ</h3>
<pre class="r"><code>gwas_study_info |&gt;
    filter(grepl(&quot;female reproductive organ cancer&quot;, l1_all_disease_terms)) |&gt;
    select(`DISEASE/TRAIT`, l1_all_disease_terms) |&gt;
    distinct()


gwas_study_info = gwas_study_info |&gt;
    mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;female reproductive organ cancer&quot; &amp; 
                 grepl(&quot;benign&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )


gwas_study_info |&gt;
    filter(grepl(&quot;\\breproductive system cancer&quot;, l1_all_disease_terms)) |&gt;
    select(`DISEASE/TRAIT`, l1_all_disease_terms) |&gt;
    distinct()

gwas_study_info = gwas_study_info |&gt;
    mutate(l1_all_disease_terms =
         ifelse(l1_all_disease_terms == &quot;reproductive system cancer&quot; &amp; 
                 grepl(&quot;female reproductive cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;female reproductive organ cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="male-reproductive-organ" class="section level3"
number="5.0.10">
<h3><span class="header-section-number">5.0.10</span> Male reproductive
organ</h3>
<pre class="r"><code>gwas_study_info |&gt;
    filter(grepl(&quot;\\bmale reproductive organ cancer&quot;, l1_all_disease_terms)) |&gt;
    select(`DISEASE/TRAIT`, l1_all_disease_terms) |&gt;
    distinct()</code></pre>
</div>
<div id="small-cell-cancer" class="section level3" number="5.0.11">
<h3><span class="header-section-number">5.0.11</span> Small cell
cancer</h3>
<pre class="r"><code>gwas_study_info |&gt;
    filter(grepl(&quot;\\bsmall cell cancer&quot;, l1_all_disease_terms)) |&gt;
    select(`DISEASE/TRAIT`, l1_all_disease_terms) |&gt;
    distinct()


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(`DISEASE/TRAIT` == &quot;Small-cell lung cancer&quot;,
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;small cell cancer&quot;),
                          &quot;lung cancer&quot;),
         l1_all_disease_terms
         )
)</code></pre>
</div>
<div id="central-nervous-system-cancer-nervous-system-cancer"
class="section level3" number="5.0.12">
<h3><span class="header-section-number">5.0.12</span> Central nervous
system cancer, nervous system cancer</h3>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         case_when(l1_all_disease_terms == &quot;central nervous system cancer, nervous system cancer&quot; ~ &quot;central nervous system cancer&quot;,
                   TRUE ~ l1_all_disease_terms)
         ) </code></pre>
</div>
<div id="fix-non-melanoma-skin-cancer-woopsy" class="section level3"
number="5.0.13">
<h3><span class="header-section-number">5.0.13</span> Fix non-melanoma
skin cancer, woopsy</h3>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         case_when(l1_all_disease_terms == &quot;non-malignant skin melanoma skin cancer&quot; ~ &quot;non-melanoma skin cancer&quot;,
                   l1_all_disease_terms == &quot;non-malignant melanoma of skin skin cancer&quot; ~ &quot;non-melanoma skin cancer&quot;,
                   TRUE ~ l1_all_disease_terms)
         ) 

# 
# gwas_study_info =
# gwas_study_info |&gt;
#   mutate(l1_all_disease_terms =
#          stringr::str_replace_all(l1_all_disease_terms,
#                            &quot;non-malignant skin melanoma skin cancer&quot;,
#                           &quot;non-melanoma skin cancer&quot;
#          )
#          ) </code></pre>
</div>
<div id="unspecified-skin-cancer" class="section level3"
number="5.0.14">
<h3><span class="header-section-number">5.0.14</span> Unspecified skin
cancer</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;skin cancer&quot;, l1_all_disease_terms) &amp; 
         !grepl(&quot;non-melanoma&quot;, l1_all_disease_terms)
         )  |&gt; 
  select(STUDY, 
         `DISEASE/TRAIT`, 
         all_disease_terms, 
         l1_all_disease_terms) |&gt; 
  distinct()

# make them listed under both malignant melanoma of skin and non-melanoma skin cancer

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(grepl(&quot;skin cancer&quot;, l1_all_disease_terms) &amp; 
                !grepl(&quot;melanoma&quot;, l1_all_disease_terms),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;skin cancer&quot;),
                          &quot;malignant melanoma of skin, non-melanoma skin cancer&quot;),
                l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="unspecified-lymphoma" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Unspecified
lymphoma</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;lymphoma&quot;, l1_all_disease_terms) &amp; 
          !grepl(&quot;hodgkin&quot;, l1_all_disease_terms)) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt; 
  unique()

# PheCode 202.23 maps to ICD-9  200.1   Lymphosarcoma
# which as from: http://snomed.info/id/188498009, is a form of non-Hodgkin&#39;s lymphoma

# PheCode 202.24 code maps to ICD-9 200.6,  Anaplastic large cell lymphoma a form of non-Hodgkin&#39;s lymphoma

# all ICD10 C83 codes are non-Hodgkin&#39;s lymphoma

# ICD10 C85.1 maps to PheCode 202.2 Non-Hodgkins lymphoma

nhl_terms &lt;- c(&quot;B cell non-Hodgkin lymphoma&quot;,
               &quot;PheCode 202.23&quot;,
               &quot;PheCode 202.24&quot;,
               &quot;ICD10 C83&quot;,
               &quot;ICD10 C85.1&quot;
               )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(grepl(&quot;lymphoma&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                !grepl(&quot;hodgkin&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                grepl(paste0(nhl_terms, collapse = &quot;|&quot;), 
                      `DISEASE/TRAIT`, 
                      ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;lymphoma&quot;),
                          &quot;non-hodgkin lymphoma&quot;),
                l1_all_disease_terms
         )
         )


# Non-follicular lymphoma (UKB data field 40006) likely non-hodgkin lymphoma
# as ICD10 C83: Non-follicular lymphoma is non-hodgkin lymphoma

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(grepl(&quot;lymphoma&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                !grepl(&quot;hodgkin&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                grepl(&quot;Non-follicular lymphoma \\(UKB data field 40006\\)&quot;, 
                      `DISEASE/TRAIT`, 
                      ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;lymphoma&quot;),
                          &quot;non-hodgkin lymphoma&quot;),
                l1_all_disease_terms
         )
         )


# Cancer code, self-reported: lymphoma (UKB data field 20001_1047)
# includes both hodgkin and non-hodgkin lymphoma
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(grepl(&quot;lymphoma&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                !grepl(&quot;hodgkin&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                grepl(&quot;Cancer code, self-reported: lymphoma \\(UKB data field 20001_1047\\)&quot;, 
                      `DISEASE/TRAIT`, 
                      ignore.case = T),
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;lymphoma&quot;),
                          &quot;non-hodgkin lymphoma, hodgkin lymphoma&quot;),
                l1_all_disease_terms
         )
         )

# for pubmed id: 34594039
# from sup table 1; 
# Malignant lymphoma    Malignant_Lymphoma  is defined PheCodes 201/202 CD2_NONFOLLICULAR_LYMPHOMA
# thus includes both hodgkin and non-hodgkin lymphoma
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(grepl(&quot;lymphoma&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                !grepl(&quot;hodgkin&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                PUBMED_ID == 34594039,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;lymphoma&quot;),
                          &quot;non-hodgkin lymphoma, hodgkin lymphoma&quot;),
                l1_all_disease_terms
         )
         )

# pubmed id: 23349640
# includes both hodgkin and non-hodgkin lymphoma
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(grepl(&quot;lymphoma&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                !grepl(&quot;hodgkin&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                PUBMED_ID == 23349640,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;lymphoma&quot;),
                          &quot;non-hodgkin lymphoma, hodgkin lymphoma&quot;),
                l1_all_disease_terms
         )
         )

# not entirely sure for pubmed id: 36344522
# perhaps need to read further in, but seems like it is both hodgkin and non-hodgkin lymphoma
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         ifelse(grepl(&quot;lymphoma&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                !grepl(&quot;hodgkin&quot;, l1_all_disease_terms, ignore.case = T) &amp; 
                PUBMED_ID == 36344522,
                stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(&quot;lymphoma&quot;),
                          &quot;non-hodgkin lymphoma, hodgkin lymphoma&quot;),
                l1_all_disease_terms
         )
         )</code></pre>
<div
id="breast-cancer-cancer-colon-and-rectum-cancer-tracheal-bronchus-and-lung-cancer-ovarian-cancer-prostate-cancer"
class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> breast cancer,
cancer, colon and rectum cancer, tracheal bronchus and lung cancer,
ovarian cancer, prostate cancer</h3>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt;
  mutate(l1_all_disease_terms =
         case_when(l1_all_disease_terms == &quot;breast cancer, cancer, colorectal cancer, lung cancer, ovarian cancer, prostate cancer&quot; ~ 
                     &quot;breast cancer, colorectal cancer, lung cancer, ovarian cancer, prostate cancer&quot;,
                   TRUE ~ l1_all_disease_terms)
         )</code></pre>
</div>
</div>
</div>
<div id="final-summary---number-of-unique-study-terns"
class="section level1" number="6">
<h1><span class="header-section-number">6</span> Final summary - number
of unique study terns</h1>
<div id="deal-with-duplicate-terms-created-during-grouping"
class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Deal with duplicate
terms created during grouping</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
  rowwise() |&gt;
  mutate(l1_all_disease_terms = paste0(sort(unique(unlist(strsplit(l1_all_disease_terms, &quot;, &quot;)))),
                                      collapse = &quot;, &quot;)
         ) |&gt;
  ungroup()</code></pre>
</div>
<div id="deal-with-hanging-commas-and-spaces" class="section level2"
number="6.2">
<h2><span class="header-section-number">6.2</span> Deal with hanging
commas and spaces</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = stringr::str_remove_all(l1_all_disease_terms, &quot;^,|,$&quot;)
         ) |&gt;
  mutate(l1_all_disease_terms = stringr::str_trim(l1_all_disease_terms)
         ) </code></pre>
</div>
<div id="final-summary---number-of-unique-study-terms-pairs"
class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Final summary -
number of unique study terms pairs</h2>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(l1_all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(l1_all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))


head(n_studies_trait)

dim(n_studies_trait)</code></pre>
<div id="when-separate-studies-with-multiple-terms-1"
class="section level3" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> When separate
studies with multiple terms</h3>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l1_all_disease_terms[gwas_study_info$l1_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()


test &lt;- data.frame(trait = unique(diseases))

length(unique(diseases))

# make frequency table
freq &lt;- table(as.factor(diseases))

# sort in decreasing order
freq_sorted &lt;- sort(freq, decreasing = TRUE)

# show top N, e.g. top 10
head(freq_sorted, 10)</code></pre>
</div>
<div id="save-the-updated-gwas_study_info-with-harmonized-disease-terms"
class="section level3" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Save the updated
gwas_study_info with harmonized disease terms</h3>
<pre class="r"><code>fwrite(gwas_study_info,
        here::here(&quot;output/gwas_cat/gwas_study_info_group_l1_v2.csv&quot;)
         )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
