<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-08" />

<title>Level 1 Disease Trait Grouping - Cancer traits</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Level 1 Disease Trait Grouping - Cancer
traits</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreefdd60ed13f36d58017a4f241006b32abb53d343ctargetblankfdd60eda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/fdd60ed13f36d58017a4f241006b32abb53d343c" target="_blank">fdd60ed</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreefdd60ed13f36d58017a4f241006b32abb53d343ctargetblankfdd60eda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/fdd60ed13f36d58017a4f241006b32abb53d343c" target="_blank">fdd60ed</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/who/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l1.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l2.csv
    Ignored:    output/trait_ontology/
    Ignored:    renv/

Unstaged changes:
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   code/get_term_descendants.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/level_1_disease_group_cancer.Rmd</code>) and HTML
(<code>docs/level_1_disease_group_cancer.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/fdd60ed13f36d58017a4f241006b32abb53d343c/analysis/level_1_disease_group_cancer.Rmd" target="_blank">fdd60ed</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
More disease term grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/bf45a695d011e5adf921d602aa238bbb56eafff0/docs/level_1_disease_group_cancer.html" target="_blank">bf45a69</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/1414cad810662ddf218b7d850f602c7d8447456b/analysis/level_1_disease_group_cancer.Rmd" target="_blank">1414cad</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
workflowr::wflow_publish("analysis/level_1_disease_group_cancer.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/3c8309cba2a866c31c2a4fa36c7f666dbcfe6ac2/docs/level_1_disease_group_cancer.html" target="_blank">3c8309c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/17a16b0568cc32537a563f99d252bfa8e5878ac2/analysis/level_1_disease_group_cancer.Rmd" target="_blank">17a16b0</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Further grouping of disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/778ac1e5e852390a65e1e9ba417c60037a4cc9d3/docs/level_1_disease_group_cancer.html" target="_blank">778ac1e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/bb5431c248ee3c0817ecc0010deaa8b2a8151eeb/analysis/level_1_disease_group_cancer.Rmd" target="_blank">bb5431c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Dealing with duplicate disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/9f6997944f597beb2db9edbe85a0cc9aa2990cb5/docs/level_1_disease_group_cancer.html" target="_blank">9f69979</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/9ca183a43197ca703180df4ab805ce841735c03e/docs/level_1_disease_group_cancer.html" target="_blank">9ca183a</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/50ef69d9fb5863dcdfcb1dd86b10766078897b33/analysis/level_1_disease_group_cancer.Rmd" target="_blank">50ef69d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Update cancer grouping
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(dplyr)
library(data.table)
library(stringr)</code></pre>
<div id="ontology-help---for-getting-disease-subtypes"
class="section level2" number="0.1">
<h2><span class="header-section-number">0.1</span> Ontology help - for
getting disease subtypes</h2>
<pre class="r"><code>source(here::here(&quot;code/get_term_descendants.R&quot;))</code></pre>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_cat/gwas_study_info_group_l1.csv&quot;))</code></pre>
</div>
<div id="initial-summary---number-of-unique-study-terns"
class="section level1" number="1">
<h1><span class="header-section-number">1</span> Initial summary -
number of unique study terns</h1>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(l1_all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(l1_all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))


head(n_studies_trait)</code></pre>
<pre><code># A tibble: 6 × 2
  l1_all_disease_terms      n_studies
  &lt;chr&gt;                         &lt;int&gt;
1 type 2 diabetes mellitus        145
2 asthma                          134
3 alzheimers disease              124
4 breast cancer                   112
5 major depressive disorder       108
6 schizophrenia                   108</code></pre>
<pre class="r"><code>dim(n_studies_trait)</code></pre>
<pre><code>[1] 2771    2</code></pre>
<div id="when-separate-studies-with-multiple-terms"
class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> When separate studies
with multiple terms</h2>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l1_all_disease_terms[gwas_study_info$l1_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()


length(unique(diseases))</code></pre>
<pre><code>[1] 1952</code></pre>
<pre class="r"><code>test &lt;- data.frame(trait = unique(diseases))</code></pre>
</div>
</div>
<div id="cancer-aggressiveness-measurement" class="section level1"
number="2">
<h1><span class="header-section-number">2</span> Cancer aggressiveness
measurement</h1>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_remove_all(l1_all_disease_terms,
                                  pattern = &quot;^cancer aggressiveness&quot;
                          )
 ) </code></pre>
</div>
<div id="disease-subtype-grouping-cancer" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> Disease subtype
grouping (cancer)</h1>
<div id="bone-cancer" class="section level3" number="3.0.1">
<h3><span class="header-section-number">3.0.1</span> Bone cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002129/descendants&quot;

bone_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 232
[1] &quot;\n Some example terms&quot;
[1] &quot;acute myeloid leukemia with abnormal bone marrow eosinophils inv(16)(p13q22) or t(16;16)(p13;q22)&quot;
[2] &quot;acute myeloid leukemia and myelodysplastic syndromes related to topoisomerase type 2 inhibitor&quot;   
[3] &quot;acute myeloid leukemia and myelodysplastic syndromes related to alkylating agent&quot;                 
[4] &quot;acute myeloid leukemia and myelodysplastic syndromes related to radiation&quot;                        
[5] &quot;therapy related acute myeloid leukemia and myelodysplastic syndrome&quot;                              </code></pre>
<pre class="r"><code>bone_cancer_terms = stringr::str_replace_all(bone_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

bone_cancer_terms = c(&quot;malignant bone neoplasm&quot;,
                      bone_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = paste0(bone_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;bone cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="bladder-cancer" class="section level3" number="3.0.2">
<h3><span class="header-section-number">3.0.2</span> Bladder cancer</h3>
<pre class="r"><code>url &lt;-  &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_4007/descendants&quot;

# maybe do uninary bladder cancer instead

bladder_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 33
[1] &quot;\n Some example terms&quot;
[1] &quot;micropapillary variant infiltrating bladder urothelial carcinoma&quot;
[2] &quot;lymphoma-like variant infiltrating bladder urothelial carcinoma&quot; 
[3] &quot;plasmacytoid variant infiltrating bladder urothelial carcinoma&quot;  
[4] &quot;microcystic variant infiltrating bladder urothelial carcinoma&quot;   
[5] &quot;infiltrating bladder urothelial carcinoma sarcomatoid variant&quot;   </code></pre>
<pre class="r"><code>bladder_cancer_terms = stringr::str_replace_all(bladder_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(bladder_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;bladder cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="breast-cancer" class="section level3" number="3.0.3">
<h3><span class="header-section-number">3.0.3</span> Breast cancer</h3>
<pre class="r"><code>breast_cancer_terms &lt;- grep(&quot;breast cancer&quot;, unique(diseases), value = T)

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(breast_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;breast cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = &quot;breast cancer in situ&quot;,
                                   &quot;breast cancer&quot;
                          )  
        ) |&gt;
     mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = &quot;invasive lobular cancer&quot;,
                                   &quot;breast cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="benign-neoplasms" class="section level3" number="3.0.4">
<h3><span class="header-section-number">3.0.4</span> Benign
neoplasms</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;benign neoplasm of (.*?)(?=,|$)|benign neoplasm of (.*?) (.*?)(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;benign (.*?) neoplasm(?=,|$)|benign (.*?) (.*?) neoplasm(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;(.*?) benign neoplasm(?=,|$)|(.*?) (.*?) neoplasm(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;polyp of (.*?)(?=,|$)|polyp of (.*?) (.*?)(?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    &quot;(?=,|^)(.*?) polyp(?=,|$)|(?=,|^)(.*?) (.*?) polyp (?=,|$)&quot;, 
                    &quot;benign neoplasm&quot;)
         ) 

# https://my.clevelandclinic.org/health/diseases/21477-adenomas - benign
other_benign_neoplasms = c(&quot;adenomatous colon polyp&quot;,
                           &quot;colorectal adenoma&quot;,
                           &quot;pituitary gland adenoma&quot;,
                           &quot;aldosterone-producing adenoma&quot;,
                           &quot;female genital tract polyp&quot;,
                           &quot;polyp&quot;,
                           &quot;uterine leiomyoma&quot;,
                           &quot;hepatic hemangioma&quot;,
                           &quot;lobular capilliary hemangioma&quot;, 
                           &quot;hemangioma of subcutaneous tissue&quot;,
                           &quot;hemangioma&quot;,
                           &quot;lymphangioma&quot;
                           )

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms = 
        str_replace_all(l1_all_disease_terms, 
                    paste0(other_benign_neoplasms, collapse = &quot;(?=,|$)|\\b&quot;), 
                    &quot;benign neoplasm&quot;)
         )</code></pre>
</div>
<div id="central-nervous-system-cancer" class="section level3"
number="3.0.5">
<h3><span class="header-section-number">3.0.5</span> Central nervous
system cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000326/descendants&quot;

cns_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 173
[1] &quot;\n Some example terms&quot;
[1] &quot;central nervous system ewing sarcoma/peripheral primitive neuroectodermal tumor&quot;
[2] &quot;malignant central nervous system mesenchymal, non-meningothelial neoplasm&quot;      
[3] &quot;malignant peripheral nerve sheath tumor with mesenchymal differentiation&quot;       
[4] &quot;diffuse pediatric-type high-grade glioma, h3-wildtype and idh-wildtype&quot;         
[5] &quot;childhood central nervous system primitive neuroectodermal neoplasm&quot;            </code></pre>
<pre class="r"><code>cns_cancer_terms = stringr::str_replace_all(cns_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>pattern = paste0(cns_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;)

gwas_study_info = gwas_study_info |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = pattern,
                                   &quot;central nervous system cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="cervical-cancer" class="section level3" number="3.0.6">
<h3><span class="header-section-number">3.0.6</span> Cervical
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_4362/descendants&quot;

cervical_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 37
[1] &quot;\n Some example terms&quot;
[1] &quot;signet ring cell variant cervical mucinous adenocarcinoma&quot;
[2] &quot;villoglandular variant cervical mucinous adenocarcinoma&quot;  
[3] &quot;glassy cell variant cervical adenosquamous carcinoma&quot;     
[4] &quot;intestinal variant cervical mucinous adenocarcinoma&quot;      
[5] &quot;endocervical type cervical mucinous adenocarcinoma&quot;       </code></pre>
<pre class="r"><code>cervical_cancer_terms = stringr::str_replace_all(cervical_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

cervical_cancer_terms = c(&quot;cervical intraepithelial neoplasia grade 2/3&quot;,
                          &quot;uterine cervical cancer in situ&quot;,
                          cervical_cancer_terms)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = &quot;uterine cervical cancer in situ&quot;,
                                   &quot;cervical cancer&quot;
                          )  
        )</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = paste0(cervical_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;cervical cancer&quot;
                          )  
        )

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = &quot;uterine cervical cancer in situ&quot;,
                                   &quot;cervical cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="colorectal-cancer" class="section level3" number="3.0.7">
<h3><span class="header-section-number">3.0.7</span> Colorectal
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0005575/descendants&quot;

colorectal_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 106
[1] &quot;\n Some example terms&quot;
[1] &quot;pold1-related polyposis and colorectal cancer syndrome&quot;
[2] &quot;pole-related polyposis and colorectal cancer syndrome&quot; 
[3] &quot;colorectal cancer, hereditary nonpolyposis, type 7&quot;    
[4] &quot;colorectal cancer, hereditary nonpolyposis, type 6&quot;    
[5] &quot;colon mucosa-associated lymphoid tissue lymphoma&quot;      </code></pre>
<pre class="r"><code>colorectal_cancer_terms = stringr::str_replace_all(colorectal_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

colorectal_cancer_terms= c(&quot;metastatic colorectal cancer&quot;,
                             &quot;rectum cancer&quot;,
                          colorectal_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = paste0(colorectal_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;colorectal cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = &quot;colorectal mucinous adenocarcinoma&quot;,
                                   &quot;colorectal cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = &quot;metachronous colorectal adenoma&quot;,
                                   &quot;colorectal cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="endometrial-cancer" class="section level3" number="3.0.8">
<h3><span class="header-section-number">3.0.8</span> Endometrial
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0011962/descendants&quot;

endometrial_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 23
[1] &quot;\n Some example terms&quot;
[1] &quot;endometrial endometrioid adenocarcinoma with spindled epithelial cells&quot;
[2] &quot;ovarian endometrioid adenocarcinoma with squamous differentiation&quot;     
[3] &quot;villoglandular endometrial endometrioid adenocarcinoma&quot;                
[4] &quot;secretory uterine corpus endometrioid adenocarcinoma&quot;                  
[5] &quot;mucin-rich endometrial endometrioid adenocarcinoma&quot;                    </code></pre>
<pre class="r"><code># also: http://www.ebi.ac.uk/efo/EFO_1001514: endometrial endometrioid carcinoma
endometrial_cancer_terms = c(&quot;endometrial endometrioid carcinoma&quot;,
                             endometrial_cancer_terms)

endometrial_cancer_terms = stringr::str_replace_all(endometrial_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(endometrial_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;endometrial cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="esophageal-cancer" class="section level3" number="3.0.9">
<h3><span class="header-section-number">3.0.9</span> Esophageal
cancer</h3>
<pre class="r"><code>esophageal_cancer_terms &lt;- c(&quot;esophageal adenocarcinoma&quot;,
                             &quot;esophageal squamous cell cancer&quot;)


gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(esophageal_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;esophageal cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="eye-cancer-to-add" class="section level3" number="3.0.10">
<h3><span class="header-section-number">3.0.10</span> Eye cancer (to
add)</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002236/descendants&quot;

ocular_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 75
[1] &quot;\n Some example terms&quot;
[1] &quot;lacrimal gland carcinoma ex pleomorphic adenoma&quot;
[2] &quot;intermediate cell type ciliary body melanoma&quot;   
[3] &quot;ocular melanoma with extraocular extension&quot;     
[4] &quot;medium/large size posterior uveal melanoma&quot;     
[5] &quot;metastatic malignant neoplasm in the eye&quot;       </code></pre>
<pre class="r"><code>ocular_cancer_terms = stringr::str_replace_all(ocular_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(ocular_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;ocular cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="gallbladder-and-bilary-tract-cancer" class="section level3"
number="3.0.11">
<h3><span class="header-section-number">3.0.11</span> Gallbladder and
bilary tract cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                         &quot;cancer of gallbladder and extrahepatic biliary tract&quot;,
                         &quot;gallbladder and bilary tract cancer&quot;
         )
         )</code></pre>
</div>
<div id="hodgkin-lymphoma" class="section level3" number="3.0.12">
<h3><span class="header-section-number">3.0.12</span> Hodgkin
lymphoma</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms ,
                          &quot;nodular sclerosis hodgkin lymphoma&quot;,
                          &quot;hodgkins lymphoma&quot;
         ))</code></pre>
</div>
<div id="head-and-neck-cancer" class="section level3" number="3.0.13">
<h3><span class="header-section-number">3.0.13</span> Head and neck
cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms ,
                          &quot;head and neck squamous cell cancer&quot;,
                          &quot;head and neck cancer, squamous cell cancer&quot;
         ))</code></pre>
</div>
<div id="intestinal-cancer-non--colorectal" class="section level3"
number="3.0.14">
<h3><span class="header-section-number">3.0.14</span> Intestinal cancer
(non- colorectal)</h3>
<pre class="r"><code>intestinal_cancer_terms &lt;- c(&quot;small intestine cancer&quot;,
                           &quot;small bowel cancer&quot;,
                           &quot;small intestine cancer&quot;)</code></pre>
</div>
<div id="kidney-cancer" class="section level3" number="3.0.15">
<h3><span class="header-section-number">3.0.15</span> Kidney cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_263/descendants&quot;


kidney_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 51
[1] &quot;\n Some example terms&quot;
[1] &quot;eosinophilic variant of chromophobe renal cell carcinoma&quot;
[2] &quot;childhood renal cell carcinoma with mit translocations&quot;  
[3] &quot;kidney pelvis sarcomatoid transitional cell carcinoma&quot;   
[4] &quot;infiltrating renal pelvis transitional cell carcinoma&quot;   
[5] &quot;classic variant of chromophobe renal cell carcinoma&quot;     </code></pre>
<pre class="r"><code>kidney_cancer_terms = c(&quot;renal cell carcinoma&quot;,
                       &quot;clear cell renal carcinoma&quot;,
                       &quot;clear cell renal cell carcinoma&quot;,
                       kidney_cancer_terms)

kidney_cancer_terms = stringr::str_replace_all(kidney_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

kidney_cancer_terms =  stringr::str_replace_all(kidney_cancer_terms,
                          &quot;renal cancer&quot;,
                          &quot;kidney cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(kidney_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;kidney cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="laryngeal-cancer" class="section level3" number="3.0.16">
<h3><span class="header-section-number">3.0.16</span> Laryngeal
cancer</h3>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = &quot;laryngeal squamous cell cancer|laryngeal cancer&quot;,
                                   &quot;larynx cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="leukemia" class="section level3" number="3.0.17">
<h3><span class="header-section-number">3.0.17</span> Leukemia</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000565/descendants&quot;

leukemia_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 133
[1] &quot;\n Some example terms&quot;
[1] &quot;b-cell acute lymphoblastic leukemia with t(1;19)(q23;p13.3); e2a-pbx1 (tcf3-pbx1)&quot;
[2] &quot;therapy related acute myeloid leukemia and myelodysplastic syndrome&quot;              
[3] &quot;acute myeloid leukemia, flt3 tyrosine kinase domain point mutation&quot;               
[4] &quot;acute myeloid leukemia, non-kmt2a mllt10 rearrangement positive&quot;                  
[5] &quot;blast phase chronic myelogenous leukemia, bcr-abl1 positive&quot;                      </code></pre>
<pre class="r"><code>leukemia_terms = stringr::str_replace_all(leukemia_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(leukemia_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;leukemia&quot;
                          )  
        )</code></pre>
</div>
<div id="lip-and-oral-cavity-cancer" class="section level3"
number="3.0.18">
<h3><span class="header-section-number">3.0.18</span> Lip and oral
cavity cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0005570/descendants&quot;

lip_oral_cavity_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 51
[1] &quot;\n Some example terms&quot;
[1] &quot;major salivary gland carcinoma ex pleomorphic adenoma&quot;
[2] &quot;salivary gland carcinoma ex pleomorphic adenoma&quot;      
[3] &quot;mucoepidermoid carcinoma of submandibular gland&quot;      
[4] &quot;parotid gland carcinoma ex pleomorphic adenoma&quot;       
[5] &quot;major salivary gland mucoepidermoid carcinoma&quot;        </code></pre>
<pre class="r"><code>lip_oral_cavity_cancer_terms = stringr::str_replace_all(lip_oral_cavity_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(lip_oral_cavity_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;lip and oral cavity cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="liver-cancer" class="section level3" number="3.0.19">
<h3><span class="header-section-number">3.0.19</span> Liver cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002691/descendants&quot;

liver_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 76
[1] &quot;\n Some example terms&quot;
[1] &quot;squamous cell carcinoma of gallbladder and extrahepatic biliary tract&quot;
[2] &quot;undifferentiated carcinoma of liver and intrahepatic biliary tract&quot;   
[3] &quot;squamous cell carcinoma of liver and intrahepatic biliary tract&quot;      
[4] &quot;adenocarcinoma of gallbladder and extrahepatic biliary tract&quot;         
[5] &quot;combined hepatocellular carcinoma and cholangiocarcinoma&quot;             </code></pre>
<pre class="r"><code>liver_cancer_terms = stringr::str_replace_all(liver_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)


liver_cancer_terms = c(&quot;hepatitis virus-related liver cancer&quot;,
                       liver_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(liver_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;liver cancer&quot;
                          )  
        )

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = &quot;hepatitis virus-related liver cancer&quot;,
                                   &quot;liver cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="lung-cancer" class="section level3" number="3.0.20">
<h3><span class="header-section-number">3.0.20</span> Lung cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0008903/descendants&quot;

lung_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 63
[1] &quot;\n Some example terms&quot;
[1] &quot;mixed mucinous and nonmucinous bronchioloalveolar adenocarcinoma&quot;
[2] &quot;well-differentiated fetal adenocarcinoma of the lung&quot;            
[3] &quot;lung combined large cell neuroendocrine carcinoma&quot;               
[4] &quot;primary pulmonary diffuse large b-cell lymphoma&quot;                 
[5] &quot;large cell lung carcinoma, clear cell variant&quot;                   </code></pre>
<pre class="r"><code>lung_cancer_terms = stringr::str_replace_all(lung_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(lung_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;lung cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="ovarian-cancer" class="section level3" number="3.0.21">
<h3><span class="header-section-number">3.0.21</span> Ovarian
cancer</h3>
<pre class="r"><code>url &lt;-  &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0008170/descendants&quot;

ovarian_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 77
[1] &quot;\n Some example terms&quot;
[1] &quot;theca steroid-producing cell malignant tumor of ovary, not further specified&quot;
[2] &quot;ovarian endometrioid adenocarcinoma with squamous differentiation&quot;           
[3] &quot;malignant non-dysgerminomatous germ cell tumor of ovary&quot;                     
[4] &quot;ovarian yolk sac tumor, polyvesicular vitelline pattern&quot;                     
[5] &quot;malignant dysgerminomatous germ cell tumor of ovary&quot;                         </code></pre>
<pre class="r"><code>ovarian_cancer_terms = stringr::str_replace_all(ovarian_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

ovarian_cancer_terms = c(&quot;high grade serous ovarian cancer&quot;,
                         &quot;high grade ovarian cancer&quot;,
                         &quot;high grade ovarian cancers&quot;,
                         &quot;ovarian endometrioid cancer&quot;, # http://www.ebi.ac.uk/efo/EFO_1001515 - ovarian edometrioid carcinoma
                         
                         &quot;ovarian serous cancer&quot;, # http://www.ebi.ac.uk/efo/EFO_1001516 - ovarian serous carcinoma
                       ovarian_cancer_terms
                       )</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = paste0(ovarian_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;ovarian cancer&quot;
                  
        )
 )
        
gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = &quot;high grade ovarian cancer&quot;,
                                   &quot;ovarian cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="pancreatic-cancer" class="section level3" number="3.0.22">
<h3><span class="header-section-number">3.0.22</span> Pancreatic
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0009831/descendants&quot;

pancreatic_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 37
[1] &quot;\n Some example terms&quot;
[1] &quot;pancreatic intraductal papillary-mucinous neoplasm with an associated invasive carcinoma&quot;
[2] &quot;pancreatic intraductal papillary-mucinous neoplasm with high grade dysplasia&quot;            
[3] &quot;pancreatic intraductal papillary-mucinous neoplasm with low grade dysplasia&quot;             
[4] &quot;pancreatic mucinous-cystic neoplasm with an associated invasive carcinoma&quot;               
[5] &quot;undifferentiated pancreatic carcinoma with osteoclast-like giant cells&quot;                  </code></pre>
<pre class="r"><code>pancreatic_cancer_terms = stringr::str_replace_all(pancreatic_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(pancreatic_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;pancreatic cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="prostate-cancer" class="section level3" number="3.0.23">
<h3><span class="header-section-number">3.0.23</span> Prostate
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_10283/descendants&quot;

prostate_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 21
[1] &quot;\n Some example terms&quot;
[1] &quot;lymphoepithelioma-like acinar prostate adenocarcinoma&quot;
[2] &quot;prostate signet ring cell adenocarcinoma&quot;             
[3] &quot;castration-resistant prostate carcinoma&quot;              
[4] &quot;prostate transitional cell carcinoma&quot;                 
[5] &quot;prostate embryonal rhabdomyosarcoma&quot;                  </code></pre>
<pre class="r"><code>prostate_cancer_terms = stringr::str_replace_all(prostate_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

prostate_cancer_terms = c(&quot;grade iii prostatic intraepithelial neoplasia&quot;,
                          &quot;metastatic prostate cancer&quot;,
                          prostate_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms,
                                  pattern = paste0(prostate_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;prostate cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="mesothelioma" class="section level3" number="3.0.24">
<h3><span class="header-section-number">3.0.24</span> Mesothelioma</h3>
<pre class="r"><code>mesothelioma_terms = c(&quot;pleural mesothelioma&quot;,
                       &quot;malignant pleural mesothelioma&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(mesothelioma_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;mesothelioma&quot;
                          )  
        )</code></pre>
</div>
<div id="non-hodgkins-lymphoma" class="section level3" number="3.0.25">
<h3><span class="header-section-number">3.0.25</span> Non-Hodgkins
lymphoma</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0005952/descendants&quot;

nhl_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 93
[1] &quot;\n Some example terms&quot;
[1] &quot;b-cell acute lymphoblastic leukemia with t(1;19)(q23;p13.3); e2a-pbx1 (tcf3-pbx1)&quot;
[2] &quot;epstein-barr virus-positive diffuse large b-cell lymphoma of the elderly&quot;         
[3] &quot;diffuse large b-cell lymphoma of the central nervous system&quot;                      
[4] &quot;small intestinal mucosa-associated lymphoid tissue lymphoma&quot;                      
[5] &quot;primary cutaneous diffuse large b-cell lymphoma, leg type&quot;                        </code></pre>
<pre class="r"><code>nhl_terms = stringr::str_replace_all(leukemia_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

nhl_terms = c(&quot;central nervous system non-hodgkin lymphoma&quot;,
              &quot;lymphoblastic lymphoma&quot;,
              &quot;extranodal nasal nk/t cell lymphoma&quot;, # https://www.ebi.ac.uk/ols4/ontologies/ordo/classes/http%253A%252F%252Fwww.orpha.net%252FORDO%252FOrphanet_86879
              &quot;follicular lymphoma&quot;, # http://purl.obolibrary.org/obo/DOID_0050873
              &quot;marginal zone b-cell lymphoma&quot;,
              &quot;diffuse large b-cell lymphoma&quot;,
              nhl_terms)</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(nhl_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;non-hodgkins lymphoma&quot;
                          )  
        )</code></pre>
</div>
<div id="non-melanoma-skin-cancer" class="section level3"
number="3.0.26">
<h3><span class="header-section-number">3.0.26</span> Non-melanoma skin
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0009260/descendants&quot;

non_melanoma_skin_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 11
[1] &quot;\n Some example terms&quot;
[1] &quot;salivary gland basal cell adenocarcinoma&quot;
[2] &quot;skin adenoid basal cell carcinoma&quot;       
[3] &quot;external ear basal cell carcinoma&quot;       
[4] &quot;skin basosquamous cell carcinoma&quot;        
[5] &quot;cervical adenoid basal carcinoma&quot;        </code></pre>
<pre class="r"><code>non_melanoma_skin_cancer_terms = stringr::str_replace_all(non_melanoma_skin_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(non_melanoma_skin_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;non-melanoma skin cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="other-pharygnx-cancer" class="section level3" number="3.0.27">
<h3><span class="header-section-number">3.0.27</span> Other pharygnx
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_0060119/descendants&quot;

other_pharynx_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 16
[1] &quot;\n Some example terms&quot;
[1] &quot;pharynx squamous cell carcinoma&quot; &quot;adenoid squamous cell carcinoma&quot;
[3] &quot;tonsil squamous cell carcinoma&quot;  &quot;adenoid basal cell carcinoma&quot;   
[5] &quot;aryepiglottic fold cancer&quot;      </code></pre>
<pre class="r"><code>other_pharynx_cancer_terms = stringr::str_replace_all(other_pharynx_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

other_pharynx_cancer_terms = c(&quot;hypopharyngeal cancer&quot;,
                               other_pharynx_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(other_pharynx_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;other pharynx cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="soft-tissue-sarcoma" class="section level3" number="3.0.28">
<h3><span class="header-section-number">3.0.28</span> Soft tissue
sarcoma</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_1001968/descendants&quot;

soft_tissue_sarcoma_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 191
[1] &quot;\n Some example terms&quot;
[1] &quot;malignant peripheral nerve sheath tumor with mesenchymal differentiation&quot;
[2] &quot;rhabdomyosarcoma with mixed embryonal and alveolar features&quot;             
[3] &quot;undifferentiated pleomorphic sarcoma, inflammatory variant&quot;              
[4] &quot;low grade fibromyxoid sarcoma with giant collagen rosettes&quot;              
[5] &quot;epithelioid malignant peripheral nerve sheath tumor&quot;                     </code></pre>
<pre class="r"><code>soft_tissue_sarcoma_terms = stringr::str_replace_all(soft_tissue_sarcoma_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(soft_tissue_sarcoma_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;soft tissue sarcoma&quot;
                          )  
        )</code></pre>
</div>
<div id="stomach-cancer" class="section level3" number="3.0.29">
<h3><span class="header-section-number">3.0.29</span> Stomach
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_10534/descendants&quot;

stomach_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 32
[1] &quot;\n Some example terms&quot;
[1] &quot;gastric signet ring cell adenocarcinoma&quot;
[2] &quot;malignant gastric granular cell tumor&quot;  
[3] &quot;malignant gastric germ cell tumor&quot;      
[4] &quot;hereditary diffuse gastric cancer&quot;      
[5] &quot;gastric papillary adenocarcinoma&quot;       </code></pre>
<pre class="r"><code>stomach_cancer_terms = stringr::str_replace_all(stomach_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

stomach_cancer_terms = c(
                          &quot;diffuse stomach cancer&quot;,
                          &quot;gastric cancer&quot;,
                          stomach_cancer_terms
                          )</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(stomach_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;stomach cancer&quot;
                          )  
        ) |&gt;
   mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = &quot;diffuse stomach cancer&quot;,
                                   &quot;stomach cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="squamous-cell-carcinoma" class="section level3"
number="3.0.30">
<h3><span class="header-section-number">3.0.30</span> Squamous cell
carcinoma</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = &quot;cutaneous squamous cell cancer&quot;,
                                   &quot;squamous cell cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="testicular-cancer" class="section level3" number="3.0.31">
<h3><span class="header-section-number">3.0.31</span> Testicular
cancer</h3>
</div>
<div id="thyroid-cancer" class="section level3" number="3.0.32">
<h3><span class="header-section-number">3.0.32</span> Thyroid
cancer</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_1781/descendants&quot;

thyroid_cancer_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 24
[1] &quot;\n Some example terms&quot;
[1] &quot;thyroid gland mixed medullary and follicular cell-derived carcinoma&quot;
[2] &quot;thyroid gland spindle epithelial tumor with thymus-like elements&quot;   
[3] &quot;spindle epithelial tumor with thymus-like differentiation tumor&quot;    
[4] &quot;diffuse sclerosing papillary thyroid carcinoma&quot;                     
[5] &quot;differentiated high-grade thyroid carcinoma&quot;                        </code></pre>
<pre class="r"><code>thyroid_cancer_terms = stringr::str_replace_all(thyroid_cancer_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)

thyroid_cancer_terms = c(&quot;differentiated thyroid cancer&quot;,
                         thyroid_cancer_terms)</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(thyroid_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;thyroid cancer&quot;
                          )  
        )</code></pre>
</div>
<div id="uterine-cancer" class="section level3" number="3.0.33">
<h3><span class="header-section-number">3.0.33</span> Uterine
cancer</h3>
<pre class="r"><code>uterine_cancer_terms &lt;- c(&quot;uterine corpus cancer&quot;,
                          &quot;uterine adnexa cancer&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l1_all_disease_terms  = 
          stringr::str_replace_all(l1_all_disease_terms ,
                                  pattern = paste0(uterine_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;uterine cancer&quot;
                          )  
        )</code></pre>
</div>
</div>
<div id="reducing-non-specific-cancer-terms" class="section level1"
number="4">
<h1><span class="header-section-number">4</span> Reducing non-specific
cancer terms</h1>
<div id="colonic-neoplasm" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Colonic neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;colonic neoplasm&quot;) |&gt; 
  select(MAPPED_TRAIT, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct() |&gt;
  unique()</code></pre>
<pre><code>                                                 MAPPED_TRAIT
                                                       &lt;char&gt;
 1:                                          colonic neoplasm
 2: disease free survival, colonic neoplasm, overall survival
 3:                                          colonic neoplasm
 4:                                          colonic neoplasm
 5:                                          colonic neoplasm
 6:                                          colonic neoplasm
 7:                                          colonic neoplasm
 8:                                          colonic neoplasm
 9:                                          colonic neoplasm
10:                                          colonic neoplasm
11:                                          colonic neoplasm
                                                                DISEASE/TRAIT
                                                                       &lt;char&gt;
 1:                                                              Colon cancer
 2:                                                  Survival in colon cancer
 3:                        ICD10 C18.2: Malignant neoplasm of ascending colon
 4:                     ICD10 C18.9: Malignant neoplasm of colon, unspecified
 5:                                                        Right colon cancer
 6:                                                         Left colon cancer
 7:                                   Left colon cancer vs right colon cancer
 8:                     ICD10 C18.9: Malignant neoplasm of colon, unspecified
 9:                        ICD10 C18.2: Malignant neoplasm of ascending colon
10: ICD10 C18.9: Malignant neoplasm of colon, unspecified (Gene-based burden)
11:    ICD10 C18.2: Malignant neoplasm of ascending colon (Gene-based burden)
    STUDY_ACCESSION
             &lt;char&gt;
 1:      GCST004167
 2:      GCST002822
 3:    GCST90043849
 4:    GCST90043855
 5:    GCST90162553
 6:    GCST90162554
 7:    GCST90179121
 8:    GCST90079580
 9:    GCST90079578
10:    GCST90083566
11:    GCST90083564</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;malignant|canceer&quot;, `DISEASE/TRAIT`, ignore.case = T) &amp;
                l1_all_disease_terms == &quot;colonic neoplasm&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          &quot;colonic neoplasm&quot;,
                          &quot;colorectal cancer&quot;),
                l1_all_disease_terms
         )
  )</code></pre>
<div id="glioma-can-be-benign-or-malignant" class="section level3"
number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Glioma (can be
benign or malignant)</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;central nervous system cancer, glioma&quot;,
                          &quot;central nervous system cancer&quot;
         ))


gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;glioma&quot;) |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_URI, `DISEASE/TRAIT`) |&gt; 
  distinct()</code></pre>
<pre><code>                         MAPPED_TRAIT
                               &lt;char&gt;
 1:                            glioma
 2:                            glioma
 3:                            glioma
 4:                            glioma
 5:                            glioma
 6:          overall survival, glioma
 7: progression free survival, glioma
 8:                            glioma
 9:                            glioma
10:                            glioma
11:                            glioma
12:                            glioma
13:                            glioma
14:                            glioma
15:                            glioma
16:                            glioma
17:                            glioma
                                                              MAPPED_TRAIT_URI
                                                                        &lt;char&gt;
 1:                                       http://www.ebi.ac.uk/efo/EFO_0005543
 2:                                       http://www.ebi.ac.uk/efo/EFO_0005543
 3:                                       http://www.ebi.ac.uk/efo/EFO_0005543
 4:                                       http://www.ebi.ac.uk/efo/EFO_0005543
 5:                                       http://www.ebi.ac.uk/efo/EFO_0005543
 6: http://www.ebi.ac.uk/efo/EFO_0000638, http://www.ebi.ac.uk/efo/EFO_0005543
 7: http://www.ebi.ac.uk/efo/EFO_0004920, http://www.ebi.ac.uk/efo/EFO_0005543
 8:                                       http://www.ebi.ac.uk/efo/EFO_0005543
 9:                                       http://www.ebi.ac.uk/efo/EFO_0005543
10:                                       http://www.ebi.ac.uk/efo/EFO_0005543
11:                                       http://www.ebi.ac.uk/efo/EFO_0005543
12:                                       http://www.ebi.ac.uk/efo/EFO_0005543
13:                                       http://www.ebi.ac.uk/efo/EFO_0005543
14:                                       http://www.ebi.ac.uk/efo/EFO_0005543
15:                                       http://www.ebi.ac.uk/efo/EFO_0005543
16:                                       http://www.ebi.ac.uk/efo/EFO_0005543
17:                                       http://www.ebi.ac.uk/efo/EFO_0005543
                                                                         DISEASE/TRAIT
                                                                                &lt;char&gt;
 1:                                                                             Glioma
 2:                                                                       Glioblastoma
 3:                                                            Non-glioblastoma glioma
 4:                                                                Glioma (high-grade)
 5:                                                                 Glioma (low-grade)
 6:                                                         Overall survival in glioma
 7:                                                Progression-free survival in glioma
 8:                                                Adult diffuse glioma (IDH mutation)
 9:                                                Adult diffuse glioma (IDH wildtype)
10:                             Adult diffuse glioma (IDH mutation, 1p/19q codeletion)
11:                          Adult diffuse glioma (IDH mutation, 1p/19q non-codeleted)
12:    Adult diffuse triple positive glioma (IDH and TERT mutations, 1p19q codeletion)
13:                                      Adult diffuse glioma (IDH and TERT mutations)
14:                                           Adult diffuse glioma (IDH mutation only)
15:                                          Adult diffuse glioma (TERT mutation only)
16: Adult diffuse triple negative glioma (IDH and TERT wildtype, 1p/19q non-codeleted)
17:                                                     Glioma (pediatric/youth onset)</code></pre>
<pre class="r"><code># assume where measure survival and glioma, it is cancer
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;survival&quot;, MAPPED_TRAIT, ignore.case = T),
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;glioma&quot;,
                          &quot;central nervous system cancer&quot;),
         l1_all_disease_terms
         )
  )

# assme where measure is grade, it is cancer
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;grade&quot;, `DISEASE/TRAIT`, ignore.case = T),
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;glioma&quot;,
                          &quot;central nervous system cancer&quot;),
         l1_all_disease_terms
         )
  )

# Adult diffuse glioma - assume maglignant
# https://pmc.ncbi.nlm.nih.gov/articles/PMC9245936/
gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(grepl(&quot;adult diffuse&quot;, `DISEASE/TRAIT`, ignore.case = T),
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;glioma&quot;,
                          &quot;central nervous system cancer&quot;),
         l1_all_disease_terms
         )
  )

gwas_study_info |&gt;
  filter(`DISEASE/TRAIT` == &quot;Glioma (pediatric/youth onset)&quot;) |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_URI, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct()</code></pre>
<pre><code>   MAPPED_TRAIT                     MAPPED_TRAIT_URI
         &lt;char&gt;                               &lt;char&gt;
1:       glioma http://www.ebi.ac.uk/efo/EFO_0005543
                    DISEASE/TRAIT STUDY_ACCESSION
                           &lt;char&gt;          &lt;char&gt;
1: Glioma (pediatric/youth onset)      GCST008912</code></pre>
<pre class="r"><code># from paper; seems malignant
# https://pubmed.ncbi.nlm.nih.gov/31040135/

gwas_study_info = 
gwas_study_info |&gt;
  mutate(ifelse(STUDY_ACCESSION == &quot;GCST008912&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          &quot;glioma&quot;,
                          &quot;central nervous system cancer&quot;),
                l1_all_disease_terms
         ))</code></pre>
</div>
<div id="laryngeal-neoplasm" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> Laryngeal
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;laryngeal neoplasm&quot;) |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_URI, `DISEASE/TRAIT`, STUDY_ACCESSION) |&gt; 
  distinct()</code></pre>
<pre><code>         MAPPED_TRAIT                     MAPPED_TRAIT_URI
               &lt;char&gt;                               &lt;char&gt;
1: laryngeal neoplasm http://www.ebi.ac.uk/efo/EFO_0003817
                                                                  DISEASE/TRAIT
                                                                         &lt;char&gt;
1: Cancer code, self-reported: larynx/throat cancer (UKB data field 20001_1006)
   STUDY_ACCESSION
            &lt;char&gt;
1:    GCST90041889</code></pre>
<pre class="r"><code># one study - 

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         ifelse(STUDY_ACCESSION == &quot;GCST90041889&quot;,
                stringr::str_replace_all(l1_all_disease_terms,
                          &quot;laryngeal neoplasm&quot;,
                          &quot;larynx cancer&quot;),
                l1_all_disease_terms
         ))</code></pre>
</div>
<div id="sigmoid-neoplasm" class="section level3" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> Sigmoid
neoplasm</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;sigmoid neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;ICD10 C18.7: Malignant neoplasm of sigmoid colon&quot;                              
[2] &quot;ICD10 C18.7: Malignant neoplasm of sigmoid colon&quot;                              
[3] &quot;Malignant neoplasm of sigmoid colon (UKB data field 40006) (Gene-based burden)&quot;
[4] &quot;Malignant neoplasm of sigmoid colon (UKB data field 40006)&quot;                    
[5] &quot;ICD10 C18.7: Malignant neoplasm of sigmoid colon (Gene-based burden)&quot;          </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;sigmoid neoplasm&quot; &amp; 
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;colorectal cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
</div>
<div id="cecal-neoplasm" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Cecal neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;cecal neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;Malignant neoplasm of cecum (UKB data field 40006) (Gene-based burden)&quot;
[2] &quot;Malignant neoplasm of cecum (UKB data field 40006)&quot;                    </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;cecal neoplasm&quot; &amp; 
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;colorectal cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="testicular-neoplasm" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Testicular
neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;testicular neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;ICD10 C62.1: Malignant neoplasm of descended testis&quot;   
[2] &quot;ICD10 C62.9: Malignant neoplasm of testis, unspecified&quot;</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;testicular neoplasm&quot; &amp;
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;testicular cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="ovarian-neoplasm" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Ovarian neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;ovarian neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;Malignant neoplasm of ovary (PheCode 184.11)&quot;                          
[2] &quot;ICD10 C56: Malignant neoplasm of ovary&quot;                                
[3] &quot;ICD10 C56: Malignant neoplasm of ovary&quot;                                
[4] &quot;Malignant neoplasm of ovary (UKB data field 40006) (Gene-based burden)&quot;
[5] &quot;ICD10 C56: Malignant neoplasm of ovary&quot;                                
[6] &quot;Malignant neoplasm of ovary (UKB data field 40006)&quot;                    
[7] &quot;ICD10 C56: Malignant neoplasm of ovary (Gene-based burden)&quot;            </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;ovarian neoplasm&quot; &amp;
                 grepl(&quot;malignant&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;ovarian cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="pancreatic-neoplasm" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Pancreatic
neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;pancreatic neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;Intraductal papillary mucinous neoplasm of the pancreas&quot;</code></pre>
<pre class="r"><code># Intraductal papillary mucinous neoplasm of the pancreas is a benign precursor lesion

gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;pancreatic neoplasm&quot; &amp; 
                `DISEASE/TRAIT` == &quot;Intraductal papillary mucinous neoplasm of the pancreas&quot;,
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="kidney-neoplasm" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Kidney neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;kidney neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;Cancer code, self-reported: kidney/renal cell cancer (UKB data field 20001_1034)&quot;            
[2] &quot;ICD10 C64: Malignant neoplasm of kidney, except renal pelvis&quot;                                
[3] &quot;Malignant neoplasm of kidney, except pelvis (PheCode 189.11)&quot;                                
[4] &quot;Kidney / renal cell cancer (UKB data field 20001) (Gene-based burden)&quot;                       
[5] &quot;Kidney / renal cell cancer (UKB data field 20001)&quot;                                           
[6] &quot;Malignant neoplasm of kidney, except renal pelvis (UKB data field 40006) (Gene-based burden)&quot;
[7] &quot;Malignant neoplasm of kidney, except renal pelvis (UKB data field 40006)&quot;                    
[8] &quot;ICD10 C64: Malignant neoplasm of kidney, except renal pelvis&quot;                                
[9] &quot;ICD10 C64: Malignant neoplasm of kidney, except renal pelvis (Gene-based burden)&quot;            </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;kidney neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;kidney cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="nasopharyngeal-neoplasm" class="section level2" number="4.7">
<h2><span class="header-section-number">4.7</span> Nasopharyngeal
neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;nasopharyngeal neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code> [1] &quot;Nasopharyngeal carcinoma&quot;                                                            
 [2] &quot;Nasopharyngeal carcinoma&quot;                                                            
 [3] &quot;Nasopharyngeal carcinoma&quot;                                                            
 [4] &quot;Nasopharyngeal carcinoma&quot;                                                            
 [5] &quot;Nasopharyngeal carcinoma&quot;                                                            
 [6] &quot;Nasopharyngeal carcinoma&quot;                                                            
 [7] &quot;Nasopharyngeal carcinoma (SNP x SNP interaction)&quot;                                    
 [8] &quot;Nasopharyngeal carcinoma&quot;                                                            
 [9] &quot;Malignant neoplasm of nasopharynx (Union C11)&quot;                                       
[10] &quot;Nasopharyngeal carcinoma&quot;                                                            
[11] &quot;Response to radiotherapy in nasopharyngeal carcinoma (primary lesion efficacy)&quot;      
[12] &quot;Response to radiotherapy in nasopharyngeal carcinoma (positive lymph nodes efficacy)&quot;
[13] &quot;Cancer of nasopharynx (PheCode 149.2)&quot;                                               
[14] &quot;Dysphagia in nasopharyngeal carcinoma treated with radiotherapy&quot;                     
[15] &quot;Myelosuppression in nasopharyngeal carcinoma treated with radiotherapy&quot;              </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;nasopharyngeal neoplasm&quot; &amp; 
                 grepl(&quot;carcinoma|cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;nasopharyngeal cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="eye-neoplasm" class="section level2" number="4.8">
<h2><span class="header-section-number">4.8</span> Eye neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;eye neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;Cancer code, self-reported: eye and/or adnexal cancer (UKB data field 20001_1030)&quot;
[2] &quot;Cancer of eye (PheCode 190)&quot;                                                      </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;eye neoplasm&quot; &amp; 
                 grepl(&quot;cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;ocular cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="gallbladder-neoplasm" class="section level2" number="4.9">
<h2><span class="header-section-number">4.9</span> Gallbladder
neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;gallbladder neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`)</code></pre>
<pre><code>[1] &quot;Gallbladder cancer&quot;            &quot;Gallbladder cancer&quot;           
[3] &quot;Gallbladder cancer&quot;            &quot;ICD10 C23: Gallbladder cancer&quot;
[5] &quot;Gallbladder adenomyomatosis&quot;  </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;gallbladder neoplasm&quot; &amp; 
                 grepl(&quot;cancer&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;gallbladder and bilary tract cancer&quot;,
                 l1_all_disease_terms
         )
         )

gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;gallbladder neoplasm&quot; &amp; 
                `DISEASE/TRAIT` == &quot;Gallbladder adenomyomatosis&quot;,
                 &quot;benign neoplasm&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="skin-neoplasm" class="section level2" number="4.10">
<h2><span class="header-section-number">4.10</span> Skin neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;skin neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code> [1] &quot;Other non-epithelial cancer of skin (PheCode 172.2)&quot;                                                                     
 [2] &quot;Neoplasm of uncertain behavior of skin (PheCode 173)&quot;                                                                    
 [3] &quot;ICD10 C44.8: Overlapping lesion of skin&quot;                                                                                 
 [4] &quot;ICD10 C44.9:  Malignant neoplasm of skin, unspecified&quot;                                                                   
 [5] &quot;ICD10 C44.0:  Other and unspecified malignant neoplasm of skin of lip&quot;                                                   
 [6] &quot;ICD10 C44.1: Other and unspecified malignant neoplasm of skin of eyelid, including canthus&quot;                              
 [7] &quot;ICD10 C44.2: Other and unspecified malignant neoplasm of skin of ear and external auricular canal&quot;                       
 [8] &quot;ICD10 C44.3: Other and unspecified malignant neoplasm of skin of other and unspecified parts of face&quot;                    
 [9] &quot;ICD10 C44.4: Other and unspecified malignant neoplasm of skin of scalp and neck&quot;                                         
[10] &quot;ICD10 C44.5: Other and unspecified malignant neoplasm of skin of trunk&quot;                                                  
[11] &quot;ICD10 C44.6: Other and unspecified malignant neoplasm of skin of upper limb, including shoulder&quot;                         
[12] &quot;ICD10 C44.7: Other and unspecified malignant neoplasm of skin of lower limb, including hip&quot;                              
[13] &quot;Carcinoma in situ of skin (PheCode 172.3)&quot;                                                                               
[14] &quot;ICD10 D04: Carcinoma in situ of skin&quot;                                                                                    
[15] &quot;Skin cancer (UKB data field 20001)&quot;                                                                                      
[16] &quot;Carcinoma in situ of skin (UKB data field 40006) (Gene-based burden)&quot;                                                    
[17] &quot;Malignant neoplasm of skin of eyelid, including canthus (UKB data field 40006) (Gene-based burden)&quot;                      
[18] &quot;Malignant neoplasm of skin of lower limb, including hip (UKB data field 40006) (Gene-based burden)&quot;                      
[19] &quot;Malignant neoplasm of skin of parts of face (UKB data field 40006) (Gene-based burden)&quot;                                  
[20] &quot;Malignant neoplasm of skin of scalp and neck (UKB data field 40006) (Gene-based burden)&quot;                                 
[21] &quot;Malignant neoplasm of skin of trunk (UKB data field 40006) (Gene-based burden)&quot;                                          
[22] &quot;Malignant neoplasm of skin of upper limb, including shoulder (UKB data field 40006) (Gene-based burden)&quot;                 
[23] &quot;Malignant neoplasm of skin, unspecified (UKB data field 40006) (Gene-based burden)&quot;                                      
[24] &quot;Malignant neoplasm of skin (UKB data field 40006) (Gene-based burden)&quot;                                                   
[25] &quot;Malignant neoplasm of skin of ear and external auricular canal (UKB data field 40006) (Gene-based burden)&quot;               
[26] &quot;Skin cancer (UKB data field 20001) (Gene-based burden)&quot;                                                                  
[27] &quot;Malignant neoplasm of skin of eyelid, including canthus (UKB data field 40006)&quot;                                          
[28] &quot;Malignant neoplasm of skin of lower limb, including hip (UKB data field 40006)&quot;                                          
[29] &quot;Malignant neoplasm of skin of parts of face (UKB data field 40006)&quot;                                                      
[30] &quot;Malignant neoplasm of skin of scalp and neck (UKB data field 40006)&quot;                                                     
[31] &quot;Malignant neoplasm of skin of trunk (UKB data field 40006)&quot;                                                              
[32] &quot;Malignant neoplasm of skin of upper limb, including shoulder (UKB data field 40006)&quot;                                     
[33] &quot;Malignant neoplasm of skin, unspecified (UKB data field 40006)&quot;                                                          
[34] &quot;Malignant neoplasm of skin (UKB data field 40006)&quot;                                                                       
[35] &quot;Malignant neoplasm of skin of ear and external auricular canal (UKB data field 40006)&quot;                                   
[36] &quot;Carcinoma in situ of skin (UKB data field 40006)&quot;                                                                        
[37] &quot;Skin cancer&quot;                                                                                                             
[38] &quot;ICD10 C44: Other and unspecified malignant neoplasm of skin&quot;                                                             
[39] &quot;ICD10 D04: Carcinoma in situ of skin (Gene-based burden)&quot;                                                                
[40] &quot;ICD10 C44: Other and unspecified malignant neoplasm of skin (Gene-based burden)&quot;                                         
[41] &quot;ICD10 C44.1: Other and unspecified malignant neoplasm of skin of eyelid, including canthus (Gene-based burden)&quot;          
[42] &quot;ICD10 C44.2: Other and unspecified malignant neoplasm of skin of ear and external auricular canal (Gene-based burden)&quot;   
[43] &quot;ICD10 C44.3: Other and unspecified malignant neoplasm of skin of other and unspecified parts of face (Gene-based burden)&quot;
[44] &quot;ICD10 C44.4: Other and unspecified malignant neoplasm of skin of scalp and neck (Gene-based burden)&quot;                     
[45] &quot;ICD10 C44.5: Other and unspecified malignant neoplasm of skin of trunk (Gene-based burden)&quot;                              
[46] &quot;ICD10 C44.6: Other and unspecified malignant neoplasm of skin of upper limb, including shoulder (Gene-based burden)&quot;     
[47] &quot;ICD10 C44.7: Other and unspecified malignant neoplasm of skin of lower limb, including hip (Gene-based burden)&quot;          </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;skin neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;non-melanoma skin cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="liver-neoplasm" class="section level2" number="4.11">
<h2><span class="header-section-number">4.11</span> Liver neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;liver neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code>[1] &quot;ICD10 C22: Malignant neoplasm of liver and intrahepatic bile ducts&quot;                    
[2] &quot;Cancer of liver and intrahepatic bile duct (PheCode 155)&quot;                              
[3] &quot;ICD10 C22: Malignant neoplasm of liver and intrahepatic bile ducts (Gene-based burden)&quot;
[4] &quot;Hepatic cancer&quot;                                                                        </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;liver neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;liver cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="lymphoid-neoplasm" class="section level2" number="4.12">
<h2><span class="header-section-number">4.12</span> lymphoid
neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;lymphoid neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique()</code></pre>
<pre><code>[1] &quot;Cancer code, self-reported: malignant lymph node, unspecified (UKB data field 20001_1070)&quot;</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;lymphoid neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;maglinant lymphoid tumor&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="brain-neoplasm" class="section level2" number="4.13">
<h2><span class="header-section-number">4.13</span> Brain neoplasm</h2>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(l1_all_disease_terms == &quot;brain neoplasm&quot;) |&gt; 
  pull(`DISEASE/TRAIT`) |&gt;
  unique() </code></pre>
<pre><code> [1] &quot;ICD10 C71: Malignant neoplasm of brain&quot;                                                               
 [2] &quot;Cancer code, self-reported: brain cancer / primary malignant brain tumour (UKB data field 20001_1032)&quot;
 [3] &quot;ICD10 C71.1: Malignant neoplasm of brain, frontal lobe&quot;                                               
 [4] &quot;ICD10 C71.2: Malignant neoplasm of brain, temporal lobe&quot;                                              
 [5] &quot;ICD10 C71.9: Malignant neoplasm of brain, unspecified&quot;                                                
 [6] &quot;Cancer of brain (PheCode 191.11)&quot;                                                                     
 [7] &quot;Overall survival in brain cancer&quot;                                                                     
 [8] &quot;Brain cancer specific survival&quot;                                                                       
 [9] &quot;Malignant neoplasm of brain (UKB data field 40006) (Gene-based burden)&quot;                               
[10] &quot;Malignant neoplasm of brain (UKB data field 40006)&quot;                                                   
[11] &quot;ICD10 C71: Malignant neoplasm of brain (Gene-based burden)&quot;                                           
[12] &quot;ICD10 C71.9: Malignant neoplasm of brain, unspecified (Gene-based burden)&quot;                            
[13] &quot;Brain tumor&quot;                                                                                          </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(ifelse(l1_all_disease_terms == &quot;brain neoplasm&quot; &amp; 
                 grepl(&quot;malignant|cancer|carcinoma&quot;, `DISEASE/TRAIT`, ignore.case = T),
                 &quot;central nervous system cancer&quot;,
                 l1_all_disease_terms
         )
         )</code></pre>
</div>
<div id="ocular-melanoma" class="section level2" number="4.14">
<h2><span class="header-section-number">4.14</span> Ocular Melanoma</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;uveal melanoma|uveal melanoma disease severity|epithelioid cell uveal melanoma|choroidal melanoma&quot;,
                          &quot;ocular melanoma&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;ocular melanoma disease severity&quot;,
                          &quot;ocular melanoma&quot;
         ))</code></pre>
</div>
</div>
<div id="final-summary---number-of-unique-study-terns"
class="section level1" number="5">
<h1><span class="header-section-number">5</span> Final summary - number
of unique study terns</h1>
<div id="deal-with-duplicate-terms-created-during-grouping"
class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Deal with duplicate
terms created during grouping</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
  rowwise() |&gt;
  mutate(l1_all_disease_terms = paste0(sort(unique(unlist(strsplit(l1_all_disease_terms, &quot;, &quot;)))),
                                      collapse = &quot;, &quot;)
         )</code></pre>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(l1_all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(l1_all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))


head(n_studies_trait)</code></pre>
<pre><code># A tibble: 6 × 2
  l1_all_disease_terms      n_studies
  &lt;chr&gt;                         &lt;int&gt;
1 type 2 diabetes mellitus        145
2 asthma                          134
3 breast cancer                   125
4 alzheimers disease              124
5 major depressive disorder       108
6 schizophrenia                   108</code></pre>
<pre class="r"><code>dim(n_studies_trait)</code></pre>
<pre><code>[1] 2565    2</code></pre>
<div id="when-separate-studies-with-multiple-terms-1"
class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> When separate
studies with multiple terms</h3>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l1_all_disease_terms[gwas_study_info$l1_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()


test &lt;- data.frame(trait = unique(diseases))

length(unique(diseases))</code></pre>
<pre><code>[1] 1788</code></pre>
<pre class="r"><code># make frequency table
freq &lt;- table(as.factor(diseases))

# sort in decreasing order
freq_sorted &lt;- sort(freq, decreasing = TRUE)

# show top N, e.g. top 10
head(freq_sorted, 10)</code></pre>
<pre><code>
   chronic kidney disease              hypertension  type 2 diabetes mellitus 
                    10835                      7089                       922 
  coronary artery disease           benign neoplasm major depressive disorder 
                      501                       472                       471 
       alzheimers disease             schizophrenia             breast cancer 
                      422                       368                       355 
                   asthma 
                      348 </code></pre>
</div>
<div id="save-the-updated-gwas_study_info-with-harmonized-disease-terms"
class="section level3" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Save the updated
gwas_study_info with harmonized disease terms</h3>
<pre class="r"><code>fwrite(gwas_study_info,
        here::here(&quot;output/gwas_cat/gwas_study_info_group_l1_v2.csv&quot;)
         )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] jsonlite_2.0.0    httr_1.4.7        stringr_1.5.1     data.table_1.17.8
[5] dplyr_1.1.4       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] compiler_4.3.1    renv_1.0.3        promises_1.3.3    tidyselect_1.2.1 
 [5] Rcpp_1.1.0        git2r_0.36.2      callr_3.7.6       later_1.4.2      
 [9] jquerylib_0.1.4   yaml_2.3.10       fastmap_1.2.0     here_1.0.1       
[13] R6_2.6.1          generics_0.1.4    curl_6.4.0        knitr_1.50       
[17] tibble_3.3.0      rprojroot_2.1.0   bslib_0.9.0       pillar_1.11.0    
[21] rlang_1.1.6       utf8_1.2.6        cachem_1.1.0      stringi_1.8.7    
[25] httpuv_1.6.16     xfun_0.52         getPass_0.2-4     fs_1.6.6         
[29] sass_0.4.10       cli_3.6.5         withr_3.0.2       magrittr_2.0.3   
[33] ps_1.9.1          digest_0.6.37     processx_3.8.6    rstudioapi_0.17.1
[37] lifecycle_1.0.4   vctrs_0.6.5       evaluate_1.0.4    glue_1.8.0       
[41] whisker_0.4.1     rmarkdown_2.29    tools_4.3.1       pkgconfig_2.0.3  
[45] htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
