<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-08" />

<title>More disease grouping to match GBD</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">More disease grouping to match GBD</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreedd87c61c27fd30ccff7d94e46083bd5eb67e09adtargetblankdd87c61a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/dd87c61c27fd30ccff7d94e46083bd5eb67e09ad" target="_blank">dd87c61</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreedd87c61c27fd30ccff7d94e46083bd5eb67e09adtargetblankdd87c61a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/dd87c61c27fd30ccff7d94e46083bd5eb67e09ad" target="_blank">dd87c61</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/
    Ignored:    data/who/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l1.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l2.csv
    Ignored:    output/trait_ontology/
    Ignored:    renv/

Unstaged changes:
    Modified:   code/get_term_descendants.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/level_2_disease_group.Rmd</code>)
and HTML (<code>docs/level_2_disease_group.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/dd87c61c27fd30ccff7d94e46083bd5eb67e09ad/analysis/level_2_disease_group.Rmd" target="_blank">dd87c61</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-23
</td>
<td>
Using icd codes to help grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/904bb1d0d078d006d30f157acbafc6acd9d9af3c/docs/level_2_disease_group.html" target="_blank">904bb1d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/bbcc1676c9a33b4a2199b649926510ba7fca22af/analysis/level_2_disease_group.Rmd" target="_blank">bbcc167</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Even more typo etc.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/75debe4c12411621d570ab227fadebd1368d6129/docs/level_2_disease_group.html" target="_blank">75debe4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b3e3287da4a335cd80e4186341176b52ffc9a628/analysis/level_2_disease_group.Rmd" target="_blank">b3e3287</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
…maybe fixing typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/b8ee7f07d92c6f898e85cf959d874e098cc41688/docs/level_2_disease_group.html" target="_blank">b8ee7f0</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/3305f6a8b86b0653f5ec4f7012a73e0e18ecee7b/analysis/level_2_disease_group.Rmd" target="_blank">3305f6a</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-22
</td>
<td>
…maybe fixing typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/200442f03481b96230a841801e60acf9b7c4cb00/docs/level_2_disease_group.html" target="_blank">200442f</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/614204e032f071ee4fc5135bc853b3a91534370c/analysis/level_2_disease_group.Rmd" target="_blank">614204e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
More fixing up of disease grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/7b87d93c8595e65f23ef1ab81577527729d1258b/docs/level_2_disease_group.html" target="_blank">7b87d93</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/da5c4b49d681c5d53ea42436af286a4658dad68e/analysis/level_2_disease_group.Rmd" target="_blank">da5c4b4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
More correction to cardiovascular disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/08b0db3ddeb111f49c0b2cacd59aa4d04b9bc557/docs/level_2_disease_group.html" target="_blank">08b0db3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/bb8ae95f2583268b67410838196df8d6bd996712/analysis/level_2_disease_group.Rmd" target="_blank">bb8ae95</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Better grouping of cardiovascular disease
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/7cf28036662d5013560dbcb788431c05c048292f/docs/level_2_disease_group.html" target="_blank">7cf2803</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/39262a4a846a05c1b81e8ea439cab0200fd752a6/analysis/level_2_disease_group.Rmd" target="_blank">39262a4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-17
</td>
<td>
More typo fixing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/c0cf9bdc52529d159fea52be7533b96398ff1652/docs/level_2_disease_group.html" target="_blank">c0cf9bd</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/3519a0bfd3e1c2631a965a7582a0dff1ff6979ff/analysis/level_2_disease_group.Rmd" target="_blank">3519a0b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Collapsing traits to gbd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/f1b18b0bf08e63d99da7e6f085d27c81451cdeeb/docs/level_2_disease_group.html" target="_blank">f1b18b0</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/afe44b4f5e2ff16b4dfc7c0fcbd9f7295f33cb6b/analysis/level_2_disease_group.Rmd" target="_blank">afe44b4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Collapsing traits to gbd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/c204ac4a66909ea9bf50885a198c574f72d739e3/docs/level_2_disease_group.html" target="_blank">c204ac4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/7fa03f54ef7ad0840180b470b08344a3c3a619a8/analysis/level_2_disease_group.Rmd" target="_blank">7fa03f5</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
More cancer typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/8f1639b4a91cd9c6ee219aea57a6c7018327b9ff/docs/level_2_disease_group.html" target="_blank">8f1639b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/345ad9bb454c1275f4cfa876353dd7cea2e7e6df/analysis/level_2_disease_group.Rmd" target="_blank">345ad9b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
More cancer typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/a15dd406f4679a9b446a67607c69a7e78d2ab7f4/docs/level_2_disease_group.html" target="_blank">a15dd40</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/16ead66d25b831ad97e4d2028b5bf5982597fbd2/analysis/level_2_disease_group.Rmd" target="_blank">16ead66</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Correcting some cancer grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/6018e42ac1a4b3073373f866fc26f6c4257521ec/docs/level_2_disease_group.html" target="_blank">6018e42</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/02a0b9d6f649e6ec5215d8d27fe1600c25468a38/analysis/level_2_disease_group.Rmd" target="_blank">02a0b9d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Improving cancer grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/6f6669630b17d0b849553c2970acbbfcb58f0367/docs/level_2_disease_group.html" target="_blank">6f66696</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/66cff1cdf7340f4895407ca678b9cb50eefd1195/analysis/level_2_disease_group.Rmd" target="_blank">66cff1c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-16
</td>
<td>
Even more disease term grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/21b6c022bf20fd6f601b018635550be1f603ed4a/docs/level_2_disease_group.html" target="_blank">21b6c02</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/5ec31117cbb45e8a4a939b7f54d293e4a2244b06/docs/level_2_disease_group.html" target="_blank">5ec3111</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/30d773ee3645affdf8c26d5b62834ea433e6621c/docs/level_2_disease_group.html" target="_blank">30d773e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/8d64a38fe4e1747fd2af2251d88c34813da05b1d/docs/level_2_disease_group.html" target="_blank">8d64a38</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b3088d8152dfca1738961b18074117123ad55f9c/analysis/level_2_disease_group.Rmd" target="_blank">b3088d8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-15
</td>
<td>
workflowr::wflow_publish("analysis/level_2_disease_group.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/b89d6615743221c6e229d6e7a0b43523d7df1feb/docs/level_2_disease_group.html" target="_blank">b89d661</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/c0fcab7e592e668ccfce22929defe33801bb84ad/analysis/level_2_disease_group.Rmd" target="_blank">c0fcab7</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/level_2_disease_group.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/ead4d8e06b04d72a98fb1091b0b2f3719fd3f39b/docs/level_2_disease_group.html" target="_blank">ead4d8e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/3964f77a8be7995f892061bfe465a2d9fa084a0f/analysis/level_2_disease_group.Rmd" target="_blank">3964f77</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/level_2_disease_group.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/8fb639dcf3bd391ab7610366e2d408bf9ef36736/docs/level_2_disease_group.html" target="_blank">8fb639d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/edeb6f522e44312c61808e1dedce38544e02857d/analysis/level_2_disease_group.Rmd" target="_blank">edeb6f5</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
workflowr::wflow_publish("analysis/level_2_disease_group.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/fe9170498876f7a9dd8d6d40501c9a5e33006972/docs/level_2_disease_group.html" target="_blank">fe91704</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/9c64867ea57e32dc6c8220deb37459d87a1baae2/analysis/level_2_disease_group.Rmd" target="_blank">9c64867</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-09
</td>
<td>
Minor fixing of disease trait categorisation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/fa509c01976de9a3f410ad605721320d95c43bf6/docs/level_2_disease_group.html" target="_blank">fa509c0</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/c9602c7b808057c59abf2834efbb5ab340a7b967/analysis/level_2_disease_group.Rmd" target="_blank">c9602c7</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
More grouping to match GBD
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Set up</h1>
<pre class="r"><code>library(dplyr)
library(data.table)
library(ggplot2)
library(stringr)</code></pre>
<div id="ontology-help---for-getting-disease-subtypes"
class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Ontology help - for
getting disease subtypes</h2>
<pre class="r"><code>source(here::here(&quot;code/get_term_descendants.R&quot;))</code></pre>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_cat/gwas_study_info_group_l1_v2.csv&quot;))</code></pre>
</div>
</div>
<div id="objectives" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Objectives:</h1>
<ul>
<li>Further group disease terms (level 2 categories) to match GBD
(globalqburden of disease) categories more closely.</li>
</ul>
<div id="grouping---level-2-set-up" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Grouping - level 2
set up</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l2_all_disease_terms = l1_all_disease_terms)</code></pre>
</div>
</div>
<div id="maternal-and-neonatal-disorders" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> Maternal and neonatal
disorders</h1>
<div id="maternal-disorders" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Maternal
disorders</h2>
<div id="section" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> </h3>
</div>
</div>
</div>
<div id="neoplasms" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Neoplasms</h1>
<div id="lip-and-oral-cavity-cancer" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> Lip and oral cavity
cancer</h2>
<pre class="r"><code>gwas_study_info |&gt; 
 filter(grepl(vec_to_grep_pattern(&quot;lip and oral cavity cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                     all_disease_terms
                                                &lt;char&gt;
1:                                  oral cavity cancer
2:                                      mouth neoplasm
3:                                       tongue cancer
4:                         major salivary gland cancer
5: human papilloma virus infection, oral cavity cancer
6:                                     tongue neoplasm
                                          l2_all_disease_terms
                                                        &lt;char&gt;
1:                                  lip and oral cavity cancer
2:                                  lip and oral cavity cancer
3:                                  lip and oral cavity cancer
4:                                  lip and oral cavity cancer
5: human papilloma virus infection, lip and oral cavity cancer
6:                                  lip and oral cavity cancer</code></pre>
</div>
<div id="nasopharynx-cancer" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Nasopharynx
cancer</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
  mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;nasopharyngeal cancer&quot;),
                                   &quot;nasopharynx cancer&quot;
                          )  
        )
  
gwas_study_info |&gt; 
 filter(grepl(vec_to_grep_pattern(&quot;nasopharynx cancer&quot;),
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct()</code></pre>
<pre><code>         all_disease_terms l2_all_disease_terms
                    &lt;char&gt;               &lt;char&gt;
1: nasopharyngeal neoplasm   nasopharynx cancer</code></pre>
</div>
<div id="other-pharynx-cancer" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Other pharynx
cancer</h2>
<pre class="r"><code>gwas_study_info |&gt; 
 filter(grepl(vec_to_grep_pattern(&quot;other pharynx cancer&quot;),
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                       all_disease_terms
                                                  &lt;char&gt;
1:                                     oropharynx cancer
2: laryngeal squamous cell carcinoma, hypopharynx cancer
3:    human papilloma virus infection, oropharynx cancer
4:                                         tonsil cancer
5:                              hypopharyngeal carcinoma
6:                   pharynx cancer, laryngeal carcinoma
                                    l2_all_disease_terms
                                                  &lt;char&gt;
1:                                  other pharynx cancer
2:                   larynx cancer, other pharynx cancer
3: human papilloma virus infection, other pharynx cancer
4:                                  other pharynx cancer
5:                                  other pharynx cancer
6:                   larynx cancer, other pharynx cancer</code></pre>
</div>
<div id="esophageal-cancer" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Esophageal
cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;esophageal cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                            all_disease_terms
                                                       &lt;char&gt;
1:              esophageal adenocarcinoma, barretts esophagus
2:                                  esophageal adenocarcinoma
3: esophageal adenocarcinoma, gastroesophageal reflux disease
4:                         esophageal squamous cell carcinoma
5:                    esophageal carcinoma, gastric carcinoma
6:              squamous cell carcinoma, esophageal carcinoma
                                 l2_all_disease_terms
                                               &lt;char&gt;
1:              barretts esophagus, esophageal cancer
2:                                  esophageal cancer
3: esophageal cancer, gastroesophageal reflux disease
4:                                  esophageal cancer
5:                  esophageal cancer, stomach cancer
6:                                  esophageal cancer</code></pre>
</div>
<div id="stomach-cancer" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Stomach cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;stomach cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                            all_disease_terms
                                                       &lt;char&gt;
1:                                          gastric carcinoma
2:                    esophageal carcinoma, gastric carcinoma
3:                                   gastric cardia carcinoma
4:                                     gastric adenocarcinoma
5: lung carcinoma, squamous cell carcinoma, gastric carcinoma
6:                                             gastric cancer
                l2_all_disease_terms
                              &lt;char&gt;
1:                    stomach cancer
2: esophageal cancer, stomach cancer
3:                    stomach cancer
4:                    stomach cancer
5:       lung cancer, stomach cancer
6:                    stomach cancer</code></pre>
</div>
<div id="colon-and-rectum-cancer" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Colon and rectum
cancer</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;colorectal cancer&quot;),
                                   &quot;colon and rectum cancer&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;colon and rectum cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt; 
  head()</code></pre>
<pre><code>                                                                                                                                                                                                                                           all_disease_terms
                                                                                                                                                                                                                                                      &lt;char&gt;
1:                                                                                                                                                                                                                                         colorectal cancer
2:                                                                                                                                                                                                                 sclerosing cholangitis, colorectal cancer
3:                                                                                                                                                                                                                     colorectal cancer, colorectal adenoma
4:                                                                                                                                                                                                                              metastatic colorectal cancer
5: lung carcinoma, estrogen-receptor negative breast cancer, ovarian endometrioid carcinoma, colorectal cancer, prostate carcinoma, ovarian serous carcinoma, breast carcinoma, ovarian carcinoma, lung adenocarcinoma, squamous cell lung carcinoma, cancer
6:                                                                                                                                                                                                                                             rectum cancer
                                                                           l2_all_disease_terms
                                                                                         &lt;char&gt;
1:                                                                      colon and rectum cancer
2:                                              colon and rectum cancer, sclerosing cholangitis
3:                                                     benign neoplasm, colon and rectum cancer
4:                                                                      colon and rectum cancer
5: breast cancer, cancer, colon and rectum cancer, lung cancer, ovarian cancer, prostate cancer
6:                                                                      colon and rectum cancer</code></pre>
</div>
<div id="liver-cancer" class="section level2" number="4.7">
<h2><span class="header-section-number">4.7</span> Liver cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;liver cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                         all_disease_terms
                                                    &lt;char&gt;
1:   hepatitis b virus infection, hepatocellular carcinoma
2:              sclerosing cholangitis, cholangiocarcinoma
3:              cholangiocarcinoma, sclerosing cholangitis
4:        sclerosing cholangitis, hepatocellular carcinoma
5:   hepatitis c virus infection, hepatocellular carcinoma
6: hepatocellular carcinoma, non-alcoholic steatohepatitis
                              l2_all_disease_terms
                                            &lt;char&gt;
1:             hepatitis b infection, liver cancer
2:            liver cancer, sclerosing cholangitis
3:            liver cancer, sclerosing cholangitis
4:            liver cancer, sclerosing cholangitis
5:             hepatitis c infection, liver cancer
6: liver cancer, non-alcoholic fatty liver disease</code></pre>
</div>
<div id="gallbladder-and-biliary-tract-cancer" class="section level2"
number="4.8">
<h2><span class="header-section-number">4.8</span> Gallbladder and
biliary tract cancer</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
        case_when(
          l2_all_disease_terms == &quot;cancer of gallbladder and extrahepatic biliary tract&quot; ~ &quot;gallbladder and biliary tract cancer&quot;,
          TRUE ~ l2_all_disease_terms
                 )
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;gallbladder and biliary tract cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct()</code></pre>
<pre><code>                                         all_disease_terms
                                                    &lt;char&gt;
1:            sclerosing cholangitis, gallbladder neoplasm
2:                                    gallbladder neoplasm
3: carcinoma of gallbladder and extrahepatic biliary tract
                                           l2_all_disease_terms
                                                         &lt;char&gt;
1: gallbladder and biliary tract cancer, sclerosing cholangitis
2:                         gallbladder and biliary tract cancer
3:                         gallbladder and biliary tract cancer</code></pre>
</div>
<div id="pancreatic-cancer" class="section level2" number="4.9">
<h2><span class="header-section-number">4.9</span> Pancreatic
cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;pancreatic cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                                                         all_disease_terms
                                                                                                                                    &lt;char&gt;
1:                                                                                                                    pancreatic carcinoma
2:                                                                                                        pancreatic ductal adenocarcinoma
3:                                                                                                       pancreatic carcinoma, neutropenia
4:              breast carcinoma, estrogen-receptor positive breast cancer, metastatic prostate cancer, pancreatic carcinoma, hypertension
5:               breast carcinoma, estrogen-receptor positive breast cancer, metastatic prostate cancer, pancreatic carcinoma, proteinuria
6: breast carcinoma, estrogen-receptor positive breast cancer, metastatic prostate cancer, pancreatic carcinoma, hypertension, proteinuria
                                                           l2_all_disease_terms
                                                                         &lt;char&gt;
1:                                                            pancreatic cancer
2:                                                            pancreatic cancer
3:                                               neutropenia, pancreatic cancer
4:              breast cancer, hypertension, pancreatic cancer, prostate cancer
5:               breast cancer, pancreatic cancer, prostate cancer, proteinuria
6: breast cancer, hypertension, pancreatic cancer, prostate cancer, proteinuria</code></pre>
</div>
<div id="larynx-cancer" class="section level2" number="4.10">
<h2><span class="header-section-number">4.10</span> Larynx cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;larynx cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                       all_disease_terms
                                                  &lt;char&gt;
1:                     laryngeal squamous cell carcinoma
2: laryngeal squamous cell carcinoma, hypopharynx cancer
3:                                   laryngeal carcinoma
4:                                    laryngeal neoplasm
5:                                      glottis neoplasm
6:                   pharynx cancer, laryngeal carcinoma
                  l2_all_disease_terms
                                &lt;char&gt;
1:                       larynx cancer
2: larynx cancer, other pharynx cancer
3:                       larynx cancer
4:                       larynx cancer
5:                       larynx cancer
6: larynx cancer, other pharynx cancer</code></pre>
</div>
<div id="tracheal-bronchus-and-lung-cancer" class="section level2"
number="4.11">
<h2><span class="header-section-number">4.11</span> Tracheal, bronchus,
and lung cancer</h2>
<pre class="r"><code>resp_cancer_terms = c(&quot;lung cancer&quot;,
                      &quot;bronchus cancer&quot;,
                      &quot;tracheal cancer&quot;,
                      &quot;respiratory system cancer&quot;
                        )

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  =
          ifelse(l2_all_disease_terms != &quot;tracheal bronchus and lung cancer&quot;,
          stringr::str_replace_all(l2_all_disease_terms,
                                   pattern = vec_to_grep_pattern(resp_cancer_terms),
                                  #pattern = paste0(resp_cancer_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                   &quot;tracheal bronchus and lung cancer&quot;
                          ),
          l2_all_disease_terms
          )
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;tracheal bronchus and lung cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                             all_disease_terms
                                                        &lt;char&gt;
1:                               non-small cell lung carcinoma
2:                                         lung adenocarcinoma
3:                                squamous cell lung carcinoma
4:               lung carcinoma, family history of lung cancer
5:          lung adenocarcinoma, family history of lung cancer
6: squamous cell lung carcinoma, family history of lung cancer
                l2_all_disease_terms
                              &lt;char&gt;
1: tracheal bronchus and lung cancer
2: tracheal bronchus and lung cancer
3: tracheal bronchus and lung cancer
4: tracheal bronchus and lung cancer
5: tracheal bronchus and lung cancer
6: tracheal bronchus and lung cancer</code></pre>
</div>
<div id="malignant-skin-melanoma" class="section level2" number="4.12">
<h2><span class="header-section-number">4.12</span> Malignant skin
melanoma</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;malignant melanoma of skin&quot;),
                                   &quot;malignant skin melanoma&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;malignant skin melanoma&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                   all_disease_terms
                              &lt;char&gt;
1:                cutaneous melanoma
2:                          melanoma
3: neuroblastoma, cutaneous melanoma
4:                       skin cancer
5:                    skin carcinoma
6:  melanoma, immune system toxicity
                                l2_all_disease_terms
                                              &lt;char&gt;
1:                           malignant skin melanoma
2:                           malignant skin melanoma
3:            malignant skin melanoma, neuroblastoma
4: malignant skin melanoma, non-melanoma skin cancer
5: malignant skin melanoma, non-melanoma skin cancer
6:   immune system toxicity, malignant skin melanoma</code></pre>
</div>
<div id="non-melanoma-skin-cancer" class="section level2" number="4.13">
<h2><span class="header-section-number">4.13</span> Non-melanoma skin
cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;non-melanoma skin cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct()</code></pre>
<pre><code>                               all_disease_terms
                                          &lt;char&gt;
1: squamous cell carcinoma, basal cell carcinoma
2:                        keratinocyte carcinoma
3:                          basal cell carcinoma
4:             cutaneous squamous cell carcinoma
5:                   non-melanoma skin carcinoma
6:                                   skin cancer
7:                                 skin neoplasm
8:                        skin carcinoma in situ
9:                                skin carcinoma
                                l2_all_disease_terms
                                              &lt;char&gt;
1:                          non-melanoma skin cancer
2:                          non-melanoma skin cancer
3:                          non-melanoma skin cancer
4:                          non-melanoma skin cancer
5:                          non-melanoma skin cancer
6: malignant skin melanoma, non-melanoma skin cancer
7:                          non-melanoma skin cancer
8:                          non-melanoma skin cancer
9: malignant skin melanoma, non-melanoma skin cancer</code></pre>
</div>
<div id="soft-tissue-and-other-extraosseous-sarcomas"
class="section level2" number="4.14">
<h2><span class="header-section-number">4.14</span> Soft tissue and
other extraosseous sarcomas</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;soft tissue sarcoma&quot;),
                                   &quot;soft tissue and other extraosseous sarcomas&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;soft tissue and other extraosseous sarcomas&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>       all_disease_terms                                 l2_all_disease_terms
                  &lt;char&gt;                                               &lt;char&gt;
1: sarcoma, fibrosarcoma sarcoma, soft tissue and other extraosseous sarcomas
2:       kaposis sarcoma          soft tissue and other extraosseous sarcomas</code></pre>
</div>
<div id="malignant-neoplasm-of-bone-and-articular-cartilage"
class="section level2" number="4.15">
<h2><span class="header-section-number">4.15</span> Malignant neoplasm
of bone and articular cartilage</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(c(&quot;bone cancer&quot;,&quot;osteosarcoma&quot;)),
                                   &quot;malignant neoplasm of bone and articular cartilage&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;malignant neoplasm of bone and articular cartilage&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                  all_disease_terms
                                             &lt;char&gt;
1:                                     osteosarcoma
2:                           acute myeloid leukemia
3:                                 myeloid leukemia
4:                          malignant bone neoplasm
5: acute myeloid leukemia, myelodysplastic syndrome
6:                                    bone neoplasm
                                                           l2_all_disease_terms
                                                                         &lt;char&gt;
1:                           malignant neoplasm of bone and articular cartilage
2:                           malignant neoplasm of bone and articular cartilage
3:                           malignant neoplasm of bone and articular cartilage
4:                           malignant neoplasm of bone and articular cartilage
5: malignant neoplasm of bone and articular cartilage, myelodysplastic syndrome
6:                           malignant neoplasm of bone and articular cartilage</code></pre>
</div>
<div id="breast-cancer" class="section level2" number="4.16">
<h2><span class="header-section-number">4.16</span> Breast cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;breast cancer&quot;), 
              l2_all_disease_terms,
              perl = T
              )) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                            all_disease_terms
                                                       &lt;char&gt;
1:                   estrogen-receptor negative breast cancer
2:                                           breast carcinoma
3:                   estrogen-receptor positive breast cancer
4:                                          breast carcinoma,
5: estrogen-receptor positive breast cancer, breast carcinoma
6: estrogen-receptor negative breast cancer, breast carcinoma
   l2_all_disease_terms
                 &lt;char&gt;
1:        breast cancer
2:        breast cancer
3:        breast cancer
4:        breast cancer
5:        breast cancer
6:        breast cancer</code></pre>
</div>
<div id="cervical-cancer" class="section level2" number="4.17">
<h2><span class="header-section-number">4.17</span> Cervical cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;cervical cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                     all_disease_terms
                                                &lt;char&gt;
1:                                  cervical carcinoma
2:                                     cervical cancer
3:                dysplasia of cervix, cervical cancer
4:                          dysplasia, cervical cancer
5:                    uterine cervix carcinoma in situ
6: cervical carcinoma, human papilloma virus infection
                               l2_all_disease_terms
                                             &lt;char&gt;
1:                                  cervical cancer
2:                                  cervical cancer
3:             cervical cancer, dysplasia of cervix
4:                       cervical cancer, dysplasia
5:                                  cervical cancer
6: cervical cancer, human papilloma virus infection</code></pre>
</div>
<div id="uterine-cancer" class="section level2" number="4.18">
<h2><span class="header-section-number">4.18</span> Uterine cancer</h2>
<pre class="r"><code># ? is endometrial cancer a subset of uterine cancer for GBD?
# is for ontology: http://purl.obolibrary.org/obo/MONDO_0002715
gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;endometrial cancer&quot;),
                                   &quot;uterine cancer&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;uterine cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                    all_disease_terms l2_all_disease_terms
                               &lt;char&gt;               &lt;char&gt;
1: endometrial endometrioid carcinoma       uterine cancer
2:              endometrial carcinoma       uterine cancer
3:               endometrial neoplasm       uterine cancer
4:                  uterine carcinoma       uterine cancer
5:       endometrial cancer, covid-19       uterine cancer
6:              uterine corpus cancer       uterine cancer</code></pre>
</div>
<div id="ovarian-cancer" class="section level2" number="4.19">
<h2><span class="header-section-number">4.19</span> Ovarian cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;ovarian cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                                                                                                                                                                           all_disease_terms
                                                                                                                                                                                                                                                      &lt;char&gt;
1:                                                                                                                                                                                                                                         ovarian carcinoma
2:                                                                                                                                                                                                                       malignant epithelial tumor of ovary
3:                                                                                                                                                                                                   prostate carcinoma, breast carcinoma, ovarian carcinoma
4: lung carcinoma, estrogen-receptor negative breast cancer, ovarian endometrioid carcinoma, colorectal cancer, prostate carcinoma, ovarian serous carcinoma, breast carcinoma, ovarian carcinoma, lung adenocarcinoma, squamous cell lung carcinoma, cancer
5:                                                                                                                                                                                                                           ovarian mucinous adenocarcinoma
6:                                                                                                                                                                                                                                  ovarian serous carcinoma
                                                                                                 l2_all_disease_terms
                                                                                                               &lt;char&gt;
1:                                                                                                     ovarian cancer
2:                                                                                                     ovarian cancer
3:                                                                     breast cancer, ovarian cancer, prostate cancer
4: breast cancer, cancer, colon and rectum cancer, tracheal bronchus and lung cancer, ovarian cancer, prostate cancer
5:                                                                                                     ovarian cancer
6:                                                                                                     ovarian cancer</code></pre>
</div>
<div id="prostate-cancer" class="section level2" number="4.20">
<h2><span class="header-section-number">4.20</span> Prostate cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;prostate cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                         all_disease_terms
                                                    &lt;char&gt;
1:                                      prostate carcinoma
2:   cancer aggressiveness measurement, prostate carcinoma
3: prostate carcinoma, breast carcinoma, ovarian carcinoma
4:       metastatic prostate cancer, peripheral neuropathy
5:                              metastatic prostate cancer
6:                prostate carcinoma, erectile dysfunction
                             l2_all_disease_terms
                                           &lt;char&gt;
1:                                prostate cancer
2:                                prostate cancer
3: breast cancer, ovarian cancer, prostate cancer
4:         peripheral neuropathy, prostate cancer
5:                                prostate cancer
6:          erectile dysfunction, prostate cancer</code></pre>
</div>
<div id="testicular-cancer" class="section level2" number="4.21">
<h2><span class="header-section-number">4.21</span> Testicular
cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;testicular cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct()</code></pre>
<pre><code>                              all_disease_terms
                                         &lt;char&gt;
1:                         testicular carcinoma
2: testicular carcinoma, cardiovascular disease
3:                          testicular neoplasm
                        l2_all_disease_terms
                                      &lt;char&gt;
1:                         testicular cancer
2: cardiovascular disease, testicular cancer
3:                         testicular cancer</code></pre>
</div>
<div id="kidney-cancer" class="section level2" number="4.22">
<h2><span class="header-section-number">4.22</span> Kidney cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;kidney cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head() </code></pre>
<pre><code>                all_disease_terms l2_all_disease_terms
                           &lt;char&gt;               &lt;char&gt;
1:           renal cell carcinoma        kidney cancer
2:                 nephroblastoma        kidney cancer
3:                  kidney cancer        kidney cancer
4:     clear cell renal carcinoma        kidney cancer
5:                renal carcinoma        kidney cancer
6: papillary renal cell carcinoma        kidney cancer</code></pre>
</div>
<div id="bladder-cancer" class="section level2" number="4.23">
<h2><span class="header-section-number">4.23</span> Bladder cancer</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;urinary bladder cancer&quot;),
                                   &quot;bladder cancer&quot;
                          )  
        )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;bladder cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() </code></pre>
<pre><code>                                            all_disease_terms
                                                       &lt;char&gt;
1:                                  urinary bladder carcinoma
2: urinary bladder carcinoma, disease progression measurement
3:                                    urinary bladder cancer,
4:                                     urinary bladder cancer
   l2_all_disease_terms
                 &lt;char&gt;
1:       bladder cancer
2:       bladder cancer
3:       bladder cancer
4:       bladder cancer</code></pre>
</div>
<div id="brain-and-central-nervous-system-cancer" class="section level2"
number="4.24">
<h2><span class="header-section-number">4.24</span> Brain and central
nervous system cancer</h2>
<pre class="r"><code>gwas_study_info = 
  gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;central nervous system cancer&quot;),
                                   &quot;brain and central nervous system cancer&quot;
                          )  
        )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;brain and central nervous system cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                        all_disease_terms
                                                   &lt;char&gt;
1:                                glioblastoma multiforme
2:                          central nervous system cancer
3:                                                 glioma
4:                  central nervous system cancer, glioma
5: central nervous system cancer, glioblastoma multiforme
6:                                         brain neoplasm
                      l2_all_disease_terms
                                    &lt;char&gt;
1: brain and central nervous system cancer
2: brain and central nervous system cancer
3: brain and central nervous system cancer
4: brain and central nervous system cancer
5: brain and central nervous system cancer
6: brain and central nervous system cancer</code></pre>
</div>
<div id="eye-cancer" class="section level2" number="4.25">
<h2><span class="header-section-number">4.25</span> Eye cancer</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = 
                                    vec_to_grep_pattern(c(&quot;ocular melanoma&quot;,
                                                       &quot;ocular cancer&quot;)
                                                       ),
                                   &quot;eye cancer&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;eye cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt; 
  head()</code></pre>
<pre><code>                                 all_disease_terms l2_all_disease_terms
                                            &lt;char&gt;               &lt;char&gt;
1:                                  uveal melanoma           eye cancer
2:                              choroidal melanoma           eye cancer
3:                 epithelioid cell uveal melanoma           eye cancer
4: uveal melanoma, epithelioid cell uveal melanoma           eye cancer
5:                 uveal melanoma disease severity           eye cancer
6:                                   ocular cancer           eye cancer</code></pre>
</div>
<div id="neuroblastoma-and-other-peripheral-nervous-cell-tumors"
class="section level2" number="4.26">
<h2><span class="header-section-number">4.26</span> Neuroblastoma and
other peripheral nervous cell tumors</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                   pattern = vec_to_grep_pattern(
                                     c(&quot;neuroblastoma&quot;,
                                       &quot;peripheral nervous system cancer&quot;)
                                   ),
                                   &quot;neuroblastoma and other peripheral nervous cell tumors&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;neuroblastoma and other peripheral nervous cell tumors&quot;),
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                   all_disease_terms
                              &lt;char&gt;
1:                     neuroblastoma
2: neuroblastoma, cutaneous melanoma
                                                              l2_all_disease_terms
                                                                            &lt;char&gt;
1:                          neuroblastoma and other peripheral nervous cell tumors
2: malignant skin melanoma, neuroblastoma and other peripheral nervous cell tumors</code></pre>
</div>
<div id="thyroid-cancer" class="section level2" number="4.27">
<h2><span class="header-section-number">4.27</span> Thyroid cancer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;thyroid cancer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                  all_disease_terms l2_all_disease_terms
                             &lt;char&gt;               &lt;char&gt;
1: differentiated thyroid carcinoma       thyroid cancer
2:      papillary thyroid carcinoma       thyroid cancer
3:     follicular thyroid carcinoma       thyroid cancer
4:                thyroid carcinoma       thyroid cancer
5:                   thyroid cancer       thyroid cancer</code></pre>
</div>
<div id="mesothelioma" class="section level2" number="4.28">
<h2><span class="header-section-number">4.28</span> Mesothelioma</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;mesothelioma&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                all_disease_terms l2_all_disease_terms
                           &lt;char&gt;               &lt;char&gt;
1: malignant pleural mesothelioma         mesothelioma
2:                   mesothelioma         mesothelioma
3:           pleural mesothelioma         mesothelioma</code></pre>
</div>
<div id="hodgkins-lymphoma" class="section level2" number="4.29">
<h2><span class="header-section-number">4.29</span> Hodgkins
lymphoma</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;hodgkins lymphoma&quot;),
                                   &quot;hodgkin lymphoma&quot;
                          )  
        )


gwas_study_info |&gt; 
 filter(grepl(vec_to_grep_pattern(&quot;hodgkin lymphoma&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                   all_disease_terms
                                                              &lt;char&gt;
1:                                nodular sclerosis hodgkin lymphoma
2:                                                 hodgkins lymphoma
3:                                                          lymphoma
4: hodgkins lymphoma, multiple myeloma, chronic lymphocytic leukemia
5:        hodgkins lymphoma, multiple myeloma, non-hodgkins lymphoma
6:                                                  lymphoma, asthma
                                        l2_all_disease_terms
                                                      &lt;char&gt;
1:                                          hodgkin lymphoma
2:                                          hodgkin lymphoma
3:                    hodgkin lymphoma, non-hodgkin lymphoma
4:              hodgkin lymphoma, leukemia, multiple myeloma
5: hodgkin lymphoma, multiple myeloma, non-hodgkins lymphoma
6:            asthma, hodgkin lymphoma, non-hodgkin lymphoma</code></pre>
</div>
<div id="non-hodgkin-lymphoma" class="section level2" number="4.30">
<h2><span class="header-section-number">4.30</span> Non-hodgkin
lymphoma</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;non-hodgkins lymphoma&quot;),
                                   &quot;non-hodgkin lymphoma&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;non-hodgkin lymphoma&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                     all_disease_terms
                                                &lt;char&gt;
1:   diffuse large b-cell lymphoma, multiple sclerosis
2:             follicular lymphoma, multiple sclerosis
3:   marginal zone b-cell lymphoma, multiple sclerosis
4: diffuse large b-cell lymphoma, rheumatoid arthritis
5:           rheumatoid arthritis, follicular lymphoma
6: rheumatoid arthritis, marginal zone b-cell lymphoma
                         l2_all_disease_terms
                                       &lt;char&gt;
1:   multiple sclerosis, non-hodgkin lymphoma
2:   multiple sclerosis, non-hodgkin lymphoma
3:   multiple sclerosis, non-hodgkin lymphoma
4: non-hodgkin lymphoma, rheumatoid arthritis
5: non-hodgkin lymphoma, rheumatoid arthritis
6: non-hodgkin lymphoma, rheumatoid arthritis</code></pre>
</div>
<div id="multiple-myeloma" class="section level2" number="4.31">
<h2><span class="header-section-number">4.31</span> Multiple
myeloma</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;multiple myeloma&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                   all_disease_terms
                                                              &lt;char&gt;
1:                                                  multiple myeloma
2:                           multiple myeloma, peripheral neuropathy
3:             multiple myeloma, chemotherapy-induced oral mucositis
4:                           multiple myeloma, monoclonal gammopathy
5: hodgkins lymphoma, multiple myeloma, chronic lymphocytic leukemia
6:        hodgkins lymphoma, multiple myeloma, non-hodgkins lymphoma
                                       l2_all_disease_terms
                                                     &lt;char&gt;
1:                                         multiple myeloma
2:                  multiple myeloma, peripheral neuropathy
3:    chemotherapy-induced oral mucositis, multiple myeloma
4:                  monoclonal gammopathy, multiple myeloma
5:             hodgkin lymphoma, leukemia, multiple myeloma
6: hodgkin lymphoma, multiple myeloma, non-hodgkin lymphoma</code></pre>
</div>
<div id="leukemia" class="section level2" number="4.32">
<h2><span class="header-section-number">4.32</span> Leukemia</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;leukemia&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                       all_disease_terms
                                                                  &lt;char&gt;
1:                                          acute lymphoblastic leukemia
2:                      multiple sclerosis, chronic lymphocytic leukemia
3:                    rheumatoid arthritis, chronic lymphocytic leukemia
4:            systemic lupus erythematosus, chronic lymphocytic leukemia
5: acute lymphoblastic leukemia, asparaginase-induced acute pancreatitis
6:                                          chronic lymphocytic leukemia
                     l2_all_disease_terms
                                   &lt;char&gt;
1:                               leukemia
2:           leukemia, multiple sclerosis
3:         leukemia, rheumatoid arthritis
4: leukemia, systemic lupus erythematosus
5:                 leukemia, pancreatitis
6:                               leukemia</code></pre>
</div>
<div id="other-malignant-neoplasms" class="section level2"
number="4.33">
<h2><span class="header-section-number">4.33</span> Other malignant
neoplasms</h2>
<pre class="r"><code>gwas_study_info =
  gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
         case_when(
         l2_all_disease_terms == &quot;cancer&quot; ~ &quot;other malignant neoplasms&quot;,
         TRUE ~ l2_all_disease_terms
                 )
         )

gwas_study_info =
  gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
         ifelse(PUBMED_ID == 27790247,
                stringr::str_replace_all(l2_all_disease_terms,
                                        pattern = &quot;, cancer,&quot;,
                                        &quot;, other malignant neoplasms,&quot;
                                        ),
                l2_all_disease_terms
                
         )
  )


### dealing with measuring cancer caused factor terms
gwas_study_info |&gt; 
  filter(grepl(&quot;^cancer,&quot;, l2_all_disease_terms)) |&gt; 
  pull(l2_all_disease_terms) |&gt; 
  unique()</code></pre>
<pre><code>[1] &quot;cancer, chronic obstructive pulmonary disease&quot;
[2] &quot;cancer, cardiotoxicity&quot;                       
[3] &quot;cancer, hand-foot syndrome&quot;                   
[4] &quot;cancer, peripheral neuropathy&quot;                
[5] &quot;cancer, immune system toxicity&quot;               
[6] &quot;cancer, hypothyroidism&quot;                       
[7] &quot;cancer, radiation-induced disorder&quot;           
[8] &quot;cancer, osteonecrosis&quot;                        </code></pre>
<pre class="r"><code>gwas_study_info =
  gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
         ifelse(grepl(&quot;^cancer,&quot;, l2_all_disease_terms),
                stringr::str_replace_all(l2_all_disease_terms,
                                        pattern = &quot;^cancer,&quot;,
                                        &quot;other malignant neoplasms,&quot;
                                        ),
                l2_all_disease_terms
                
         )
  )

other_malignant_terms &lt;- c(
                           &quot;retroperitoneal cancer&quot;,
                           &quot;peritoneal cancer&quot;,
                           &quot;ewing sarcoma&quot;,
                           
                           &quot;digestive system cancer&quot;,
                           &quot;intestinal cancer&quot;,
                           &quot;small intestine cancer&quot;,
                           
                           &quot;female reproductive organ cancer&quot;,
                           &quot;male reproductive organ cancer&quot;,
                           &quot;vulvar cancer&quot;,                           
                           &quot;testicular germ cell tumor&quot;,
                           &quot;urogenital cancer&quot;,
                           
                           &quot;squamous cell cancer&quot;,

                           &quot;head and neck cancer&quot;,
                           &quot;malignant tumor of floor of mouth&quot;, 
                           &quot;nasal cavity cancer&quot;, #? not sure if should be somewhere else .. 
                           
                           &quot;malignant lymphoid tumor&quot;,
                           &quot;neuroendocrine tumor&quot;,
                           &quot;lymphatic system cancer&quot;,
                           
                           &quot;childhood cancer&quot; #? maybe sort furtrher

                           )

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                   pattern = vec_to_grep_pattern(other_malignant_terms),
                                  # pattern = paste0(other_malignant_terms, collapse = &quot;(?=,|$)|\\b&quot;),
                                  &quot;other malignant neoplasms&quot;
                          )  
        )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other malignant neoplasms&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt; 
  head()</code></pre>
<pre><code>                      all_disease_terms
                                 &lt;char&gt;
1:              squamous cell carcinoma
2:                               cancer
3:     childhood cancer, cardiomyopathy
4:              neuroendocrine neoplasm
5: small intestine neuroendocrine tumor
6:      pancreatic neuroendocrine tumor
                        l2_all_disease_terms
                                      &lt;char&gt;
1:                 other malignant neoplasms
2:                 other malignant neoplasms
3: cardiomyopathy, other malignant neoplasms
4:                 other malignant neoplasms
5:                 other malignant neoplasms
6:                 other malignant neoplasms</code></pre>
</div>
<div id="other-neoplasms" class="section level2" number="4.34">
<h2><span class="header-section-number">4.34</span> Other neoplasms</h2>
<pre class="r"><code>gwas_study_info =
  gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
         case_when(
         l2_all_disease_terms == &quot;benign neoplasm&quot; ~ &quot;other neoplasms&quot;,
         TRUE ~ l2_all_disease_terms
                 )
         )

unknown_sig_terms &lt;- c(&quot;intracranial germ cell tumor&quot;,
                       &quot;bladder tumor&quot;)

gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(unknown_sig_terms),
                                   &quot;other neoplasms&quot;
                          )
 )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other neoplasms&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                         all_disease_terms l2_all_disease_terms
                                    &lt;char&gt;               &lt;char&gt;
1:            benign prostatic hyperplasia      other neoplasms
2:                      colorectal adenoma      other neoplasms
3: colorectal cancer, endometrial neoplasm      other neoplasms
4:      upper aerodigestive tract neoplasm      other neoplasms
5:                              meningioma      other neoplasms
6:   nasopharyngeal neoplasm, hearing loss      other neoplasms</code></pre>
</div>
</div>
<div id="cardiovascular-diseases" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Cardiovascular
diseases</h1>
<div id="rheumatic-heart-disease" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Rheumatic heart
disease</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;rheumatic heart disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>         all_disease_terms    l2_all_disease_terms
                    &lt;char&gt;                  &lt;char&gt;
1: rheumatic heart disease rheumatic heart disease</code></pre>
</div>
<div id="ischemic-heart-disease" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Ischemic heart
disease</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;ischemic heart disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                      all_disease_terms
                                                                 &lt;char&gt;
1:                              non-obstructive coronary artery disease
2:                 mucocutaneous lymph node syndrome, coronary aneurysm
3:                           migraine disorder, coronary artery disease
4:                                              coronary artery disease
5:           type 2 diabetes mellitus, obesity, coronary artery disease
6: alzheimer disease, type 2 diabetes mellitus, coronary artery disease
                                                   l2_all_disease_terms
                                                                 &lt;char&gt;
1:                                               ischemic heart disease
2:            ischemic heart disease, mucocutaneous lymph node syndrome
3:                                     ischemic heart disease, migraine
4:                                               ischemic heart disease
5:            ischemic heart disease, obesity, type 2 diabetes mellitus
6: alzheimers disease, ischemic heart disease, type 2 diabetes mellitus</code></pre>
<p>== coronary artery disease (<a
href="https://www.ncbi.nlm.nih.gov/books/NBK209964/"
class="uri">https://www.ncbi.nlm.nih.gov/books/NBK209964/</a>)</p>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;coronary artery disease&quot;),
                                   &quot;ischemic heart disease&quot;
                          )  
        )</code></pre>
</div>
<div id="stroke" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Stroke</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;stroke&quot;),
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                        all_disease_terms       l2_all_disease_terms
                                   &lt;char&gt;                     &lt;char&gt;
1:               intracerebral hemorrhage                     stroke
2:     non-lobar intracerebral hemorrhage                     stroke
3:         lobar intracerebral hemorrhage                     stroke
4:                    small vessel stroke                     stroke
5: alzheimer disease, small vessel stroke alzheimers disease, stroke
6:                        ischemic stroke                     stroke</code></pre>
</div>
<div id="hypertensive-heart-disease" class="section level2"
number="5.4">
<h2><span class="header-section-number">5.4</span> Hypertensive heart
disease</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;hypertensive heart disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                            all_disease_terms
                                       &lt;char&gt;
1: hypertensive heart disease, kidney disease
2:                 hypertensive heart disease
                         l2_all_disease_terms
                                       &lt;char&gt;
1: hypertensive heart disease, kidney disease
2:                 hypertensive heart disease</code></pre>
</div>
<div id="non-rheumatic-valvular-heart-disease" class="section level2"
number="5.5">
<h2><span class="header-section-number">5.5</span> Non-rheumatic
valvular heart disease</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;heart valve disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                             all_disease_terms l2_all_disease_terms
                                        &lt;char&gt;               &lt;char&gt;
1: aortic stenosis, aortic valve calcification  heart valve disease
2:                             aortic stenosis  heart valve disease
3:                  aortic valve calcification  heart valve disease
4:                       mitral valve prolapse  heart valve disease
5:                        aortic valve disease  heart valve disease
6:                         heart valve disease  heart valve disease</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;heart valve disease&quot;),
                                   &quot;non-rheumatic valvular heart disease&quot;
                          )  
        )



gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;non-rheumatic valvular heart disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                             all_disease_terms
                                        &lt;char&gt;
1: aortic stenosis, aortic valve calcification
2:                             aortic stenosis
3:                  aortic valve calcification
4:                       mitral valve prolapse
5:                        aortic valve disease
6:                         heart valve disease
                   l2_all_disease_terms
                                 &lt;char&gt;
1: non-rheumatic valvular heart disease
2: non-rheumatic valvular heart disease
3: non-rheumatic valvular heart disease
4: non-rheumatic valvular heart disease
5: non-rheumatic valvular heart disease
6: non-rheumatic valvular heart disease</code></pre>
</div>
<div id="cardiomyopathy-and-myocarditis" class="section level2"
number="5.6">
<h2><span class="header-section-number">5.6</span> Cardiomyopathy and
myocarditis</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(c(&quot;cardiomyopathy&quot;,
                                                                  &quot;myocarditis&quot;)),
                                   &quot;cardiomyopathy and myocarditis&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;cardiomyopathy and myocarditis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                  all_disease_terms
                             &lt;char&gt;
1: childhood cancer, cardiomyopathy
2:      hypertrophic cardiomyopathy
3:           dilated cardiomyopathy
4:            chagas cardiomyopathy
5:        peripartum cardiomyopathy
6:         takotsubo cardiomyopathy
                                        l2_all_disease_terms
                                                      &lt;char&gt;
1: cardiomyopathy and myocarditis, other malignant neoplasms
2:                            cardiomyopathy and myocarditis
3:                            cardiomyopathy and myocarditis
4:                            cardiomyopathy and myocarditis
5:                            cardiomyopathy and myocarditis
6:                            cardiomyopathy and myocarditis</code></pre>
</div>
<div id="pulmonary-arterial-hypertension" class="section level2"
number="5.7">
<h2><span class="header-section-number">5.7</span> Pulmonary arterial
hypertension</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;pulmonary arterial hypertension&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                 all_disease_terms            l2_all_disease_terms
                            &lt;char&gt;                          &lt;char&gt;
1: pulmonary arterial hypertension pulmonary arterial hypertension</code></pre>
</div>
<div id="atrial-fibrillation-flutter" class="section level2"
number="5.8">
<h2><span class="header-section-number">5.8</span> Atrial fibrillation
&amp; flutter</h2>
<pre class="r"><code>afib_terms &lt;- c(&quot;atrial fibrillation&quot;,
                &quot;atrial flutter&quot;,
                &quot;post-operative atrial fibrillation&quot;)

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(afib_terms),
                                   &quot;atrial fibrillation and flutter&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;atrial fibrillation and flutter&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                all_disease_terms
                                                                                           &lt;char&gt;
1:                                                                            atrial fibrillation
2: heart failure, diabetes mellitus, stroke, atrial fibrillation, coronary artery disease, cancer
3: stroke, atrial fibrillation, coronary artery disease, heart failure, diabetes mellitus, cancer
4:                                                                                 atrial flutter
5:                                                             post-operative atrial fibrillation
                                                                                                           l2_all_disease_terms
                                                                                                                         &lt;char&gt;
1:                                                                                              atrial fibrillation and flutter
2: atrial fibrillation and flutter, other malignant neoplasms, diabetes mellitus, heart failure, ischemic heart disease, stroke
3: atrial fibrillation and flutter, other malignant neoplasms, diabetes mellitus, heart failure, ischemic heart disease, stroke
4:                                                                                              atrial fibrillation and flutter
5:                                                                              atrial fibrillation and flutter, post-operative</code></pre>
</div>
<div id="aortic-aneurysm" class="section level2" number="5.9">
<h2><span class="header-section-number">5.9</span> Aortic aneurysm</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/cvdo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_3627/descendants&quot;

aortic_aneurysm_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 6
[1] &quot;\n Some example terms&quot;
[1] &quot;ruptured thoracoabdominal aortic aneurysm&quot;
[2] &quot;ruptured abdominal aortic aneurysm&quot;       
[3] &quot;ruptured thoracic aortic aneurysm&quot;        
[4] &quot;abdominal aortic aneurysm&quot;                
[5] &quot;ruptured aortic aneurysm&quot;                 </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(aortic_aneurysm_terms),
                                   &quot;aortic aneurysm&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;aortic aneurysm&quot;), 
              l2_all_disease_terms,
              perl = T
              )) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
</div>
<div id="lower-extremity-peripheral-arterial-disease"
class="section level2" number="5.10">
<h2><span class="header-section-number">5.10</span> Lower extremity
peripheral arterial disease</h2>
<pre class="r"><code>lower_extremity_peripheral_arterial_disease_terms &lt;- c(&quot;raynaud disease&quot;
                                                        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;lower extremity peripheral arterial disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                       all_disease_terms
                                                  &lt;char&gt;
1:                           peripheral arterial disease
2: type 2 diabetes mellitus, peripheral arterial disease
3:        diabetes mellitus, peripheral arterial disease
                                                    l2_all_disease_terms
                                                                  &lt;char&gt;
1:                           lower extremity peripheral arterial disease
2: lower extremity peripheral arterial disease, type 2 diabetes mellitus
3:        diabetes mellitus, lower extremity peripheral arterial disease</code></pre>
</div>
<div id="endocarditis" class="section level2" number="5.11">
<h2><span class="header-section-number">5.11</span> Endocarditis</h2>
<pre class="r"><code>gwas_study_info |&gt; 
 filter(grepl(vec_to_grep_pattern(&quot;endocarditis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                         all_disease_terms
                                                    &lt;char&gt;
1: staphylococcus aureus infection, bacterial endocarditis
2:                                            endocarditis
                            l2_all_disease_terms
                                          &lt;char&gt;
1: endocarditis, staphylococcus aureus infection
2:                                  endocarditis</code></pre>
</div>
<div id="other-cardiovascular-and-circulatory-diseases"
class="section level2" number="5.12">
<h2><span class="header-section-number">5.12</span> Other cardiovascular
and circulatory diseases</h2>
<pre class="r"><code>other_cardiovascular_terms &lt;- c(&quot;tachycardia&quot;,
                                &quot;other cardiac arrhythmias&quot;,
                                &quot;heart block&quot;,
                                &quot;carotid artery disease&quot;,
                                &quot;hypertension&quot;,
                                &quot;pericarditis&quot;,
                                &quot;coronary artery calcification&quot;,
                                &quot;arterial occlusion&quot;,
                                &quot;other vascular disorders&quot;,
                                &quot;congestive heart failure&quot;,
                                &quot;heart failure&quot;,
                                &quot;thrombotic diseas&quot;,
                                &quot;arterial embolism&quot;,
                                &quot;cardiac embolism&quot;,
                                &quot;venus embolism&quot;,
                                &quot;venus thrombosis&quot;,
                                &quot;pulmonary embolism&quot;,                                
                                &quot;arterial thrombosis&quot;,
                                &quot;thromboembolism&quot;,
                                &quot;vascular insufficiency&quot;,
                                &quot;brain infarction&quot;,
                                &quot;heart murmur&quot;
                                 )

gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(other_cardiovascular_terms),
                                   &quot;other cardiovascular and circulatory diseases&quot;
                          )
 )

gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;(?&lt;=^|, ) other vascular disorders(?=,|$)&quot;,
                                   &quot;other cardiovascular and circulatory diseases&quot;
                          )
 )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other cardiovascular and circulatory diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                                all_disease_terms
                                                                                                           &lt;char&gt;
1:                                                                                                   hypertension
2:                                                                                hypertension, alzheimer disease
3:                                                                              ischemic stroke, cardiac embolism
4:                                                                        ischemic stroke, small artery occlusion
5: ischemic stroke, venous thromboembolism, stroke, deep vein thrombosis, pulmonary embolism, abnormal thrombosis
6:                                                        type 2 diabetes mellitus, coronary artery calcification
                                                                                                      l2_all_disease_terms
                                                                                                                    &lt;char&gt;
1:                                                                           other cardiovascular and circulatory diseases
2:                                                       alzheimers disease, other cardiovascular and circulatory diseases
3:                                                                   other cardiovascular and circulatory diseases, stroke
4:                                                                   other cardiovascular and circulatory diseases, stroke
5: deep vein thrombosis, other cardiovascular and circulatory diseases, stroke, thrombotic disease, venous thromboembolism
6:                                                 other cardiovascular and circulatory diseases, type 2 diabetes mellitus</code></pre>
</div>
</div>
<div id="chronic-respiratory-diseases" class="section level1"
number="6">
<h1><span class="header-section-number">6</span> Chronic respiratory
diseases</h1>
<div id="chronic-obstructive-pulmonary-disease" class="section level2"
number="6.1">
<h2><span class="header-section-number">6.1</span> Chronic obstructive
pulmonary disease</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;chronic obstructive pulmonary disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                   all_disease_terms
                                                              &lt;char&gt;
1:                             chronic obstructive pulmonary disease
2:         chronic obstructive pulmonary disease, chronic bronchitis
3: digestive system carcinoma, chronic obstructive pulmonary disease
4:                     cancer, chronic obstructive pulmonary disease
5:             lung carcinoma, chronic obstructive pulmonary disease
6:                     asthma, chronic obstructive pulmonary disease
                                                       l2_all_disease_terms
                                                                     &lt;char&gt;
1:                                    chronic obstructive pulmonary disease
2:                chronic bronchitis, chronic obstructive pulmonary disease
3:         chronic obstructive pulmonary disease, other malignant neoplasms
4:         other malignant neoplasms, chronic obstructive pulmonary disease
5: chronic obstructive pulmonary disease, tracheal bronchus and lung cancer
6:                            asthma, chronic obstructive pulmonary disease</code></pre>
</div>
<div id="pneumoconiosis" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Pneumoconiosis</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;pneumoconiosis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>   all_disease_terms l2_all_disease_terms
              &lt;char&gt;               &lt;char&gt;
1:    pneumoconiosis       pneumoconiosis</code></pre>
</div>
<div id="asthma" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Asthma</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;asthma&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                all_disease_terms
                                                           &lt;char&gt;
1:                                                         asthma
2:                                       asthma, allergic disease
3:                                         childhood onset asthma
4: childhood onset asthma, respiratory symptom change measurement
5:                                         age of onset of asthma
6:                                         aspirin-induced asthma
       l2_all_disease_terms
                     &lt;char&gt;
1:                   asthma
2: allergic disease, asthma
3:                   asthma
4:                   asthma
5:                   asthma
6:                   asthma</code></pre>
</div>
<div id="interstitial-lung-disease-and-pulmonary-sarcoidosis"
class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Interstitial lung
disease and pulmonary sarcoidosis</h2>
<pre class="r"><code>interstitial_lung_disease_terms &lt;- c(&quot;pulmonary sarcoidosis&quot;,
                                   &quot;interstitial lung disease&quot;
                                   )

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(interstitial_lung_disease_terms),
                                   &quot;interstitial lung disease and pulmonary sarcoidosis&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;interstitial lung disease and pulmonary sarcoidosis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                 all_disease_terms
                                            &lt;char&gt;
1: rheumatoid arthritis, interstitial lung disease
2:                       interstitial lung disease
3: systemic scleroderma, interstitial lung disease
                                                        l2_all_disease_terms
                                                                      &lt;char&gt;
1: interstitial lung disease and pulmonary sarcoidosis, rheumatoid arthritis
2:                       interstitial lung disease and pulmonary sarcoidosis
3: interstitial lung disease and pulmonary sarcoidosis, systemic scleroderma</code></pre>
</div>
<div id="other-chronic-respiratory-diseases" class="section level2"
number="6.5">
<h2><span class="header-section-number">6.5</span> Other chronic
respiratory diseases</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other chronic respiratory diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
</div>
</div>
<div id="digestive-diseases" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Digestive diseases</h1>
<div id="cirrhosis-other-chronic-liver-diseases" class="section level2"
number="7.1">
<h2><span class="header-section-number">7.1</span> Cirrhosis &amp; other
chronic liver diseases</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F328383001/descendants&quot;

chronic_liver_disease_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 114
[1] &quot;\n Some example terms&quot;
[1] &quot;hepatic ascites co-occurrent with chronic active hepatitis due to toxic liver disease&quot;
[2] &quot;cirrhosis of liver co-occurrent and due to primary sclerosing cholangitis (disorder)&quot; 
[3] &quot;chronic hepatitis c co-occurrent with human immunodeficiency virus infection&quot;         
[4] &quot;primary biliary cirrhosis co-occurrent with systemic scleroderma (disorder)&quot;          
[5] &quot;pulmonary fibrosis, hepatic hyperplasia, bone marrow hypoplasia syndrome&quot;             </code></pre>
<pre class="r"><code>chronic_liver_disease_terms &lt;- c(&quot;primary biliary cirrhosis&quot;,
                                 &quot;alcoholic liver cirrhosis&quot;,
                                 &quot;chronic hepatitis B virus infection&quot;, 
                                 &quot;acute-on-chronic liver failure&quot;,
                                 &quot;non-alcoholic fatty liver disease&quot;,
                                 &quot;cirrhosis of liver&quot;,
                                 &quot;primary biliary cirrhosis&quot;,
                                 &quot;chronic hepatitis&quot;,
                                 &quot;liver disease&quot;,
                                 chronic_liver_disease_terms)

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(chronic_liver_disease_terms),
                                   &quot;cirrhosis and other chronic liver diseases&quot;
                          )  
        )

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = &quot;(?&lt;=^|, ) liver disease(?=,|$)&quot;,
                                   &quot;cirrhosis and other chronic liver diseases&quot;
                          )  
        )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;cirrhosis and other chronic liver diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                        all_disease_terms
                                                   &lt;char&gt;
1:                              primary biliary cirrhosis
2:                       alagille syndrome, liver disease
3:              inflammatory bowel disease, liver disease
4: non-alcoholic fatty liver disease severity measurement
5:    non-alcoholic fatty liver disease, hepatic fibrosis
6:                          non-alcoholic steatohepatitis
                                                     l2_all_disease_terms
                                                                   &lt;char&gt;
1:                             cirrhosis and other chronic liver diseases
2:          alagille syndrome, cirrhosis and other chronic liver diseases
3: inflammatory bowel disease, cirrhosis and other chronic liver diseases
4:                             cirrhosis and other chronic liver diseases
5:           hepatic fibrosis, cirrhosis and other chronic liver diseases
6:                             cirrhosis and other chronic liver diseases</code></pre>
</div>
<div id="upper-digestive-system-diseases" class="section level2"
number="7.2">
<h2><span class="header-section-number">7.2</span> Upper digestive
system diseases</h2>
<pre class="r"><code>upper_dig_terms &lt;- c(&quot;peptic ulcer diseases&quot;,
                     &quot;duodenitis&quot;,
                     &quot;gastritis&quot;,
                     &quot;gastroesophageal reflux disease&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(upper_dig_terms),
                                   &quot;upper digestive system diseases&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;upper digestive system diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                 all_disease_terms
                                                            &lt;char&gt;
1:                                 gastroesophageal reflux disease
2:      esophageal adenocarcinoma, gastroesophageal reflux disease
3:             barretts esophagus, gastroesophageal reflux disease
4:                                                       gastritis
5:      gastroesophageal reflux disease, major depressive disorder
6: post-traumatic stress disorder, gastroesophageal reflux disease
                                              l2_all_disease_terms
                                                            &lt;char&gt;
1:                                 upper digestive system diseases
2:              esophageal cancer, upper digestive system diseases
3:             barretts esophagus, upper digestive system diseases
4:                                 upper digestive system diseases
5:      upper digestive system diseases, major depressive disorder
6: upper digestive system diseases, post-traumatic stress disorder</code></pre>
</div>
<div id="appendicitis" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Appendicitis</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;appendicitis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>   all_disease_terms l2_all_disease_terms
              &lt;char&gt;               &lt;char&gt;
1:      appendicitis         appendicitis</code></pre>
</div>
<div id="paralytic-ileus-and-intestinal-obstruction"
class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Paralytic ileus and
intestinal obstruction</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(c(&quot;paralytic ileus&quot;,
                                                             &quot;intestinal obstruction&quot;)
                                                             ),
                                   &quot;paralytic ileus and intestinal obstruction&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;paralytic ileus and intestinal obstruction&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                            all_disease_terms
                                                       &lt;char&gt;
1:                  cystic fibrosis associated meconium ileus
2:                                     intestinal obstruction
3:                                            paralytic ileus
4:                                       intestinal impaction
5: cystic fibrosis, cystic fibrosis associated meconium ileus
6:                                                      ileus
                                          l2_all_disease_terms
                                                        &lt;char&gt;
1:                  paralytic ileus and intestinal obstruction
2:                  paralytic ileus and intestinal obstruction
3:                  paralytic ileus and intestinal obstruction
4:                  paralytic ileus and intestinal obstruction
5: cystic fibrosis, paralytic ileus and intestinal obstruction
6:                  paralytic ileus and intestinal obstruction</code></pre>
</div>
<div id="inguinal-femoral-and-abdominal-hernia" class="section level2"
number="7.5">
<h2><span class="header-section-number">7.5</span> Inguinal femoral and
abdominal hernia</h2>
<pre class="r"><code>hernia_terms &lt;- c(&quot;inguinal hernia&quot;,
                  &quot;femoral hernia&quot;,
                  &quot;abdominal hernia&quot;,
                  &quot;hernia of the abdominal wall&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(hernia_terms),
                                   &quot;inguinal femoral and abdominal hernia&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;inguinal femoral and abdominal hernia&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>              all_disease_terms                  l2_all_disease_terms
                         &lt;char&gt;                                &lt;char&gt;
1:              inguinal hernia inguinal femoral and abdominal hernia
2: hernia of the abdominal wall inguinal femoral and abdominal hernia
3:               femoral hernia inguinal femoral and abdominal hernia
4:             umbilical hernia inguinal femoral and abdominal hernia
5:               ventral hernia inguinal femoral and abdominal hernia
6:            incisional hernia inguinal femoral and abdominal hernia</code></pre>
</div>
<div id="inflammatory-bowel-disease" class="section level2"
number="7.6">
<h2><span class="header-section-number">7.6</span> Inflammatory bowel
disease</h2>
<pre class="r"><code>ibd_terms &lt;- c(&quot;crohns disease&quot;,
               &quot;ulcerative colitis&quot;,
               &quot;inflammatory bowel disease&quot;,
               &quot;enteritis&quot;,
               &quot;gastroenteritis&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(ibd_terms),
                                   &quot;inflammatory bowel disease&quot;
                          )  
        )

gwas_study_info |&gt;
  filter(grepl(vec_to_grep_pattern(&quot;inflammatory bowel disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                          all_disease_terms
                                     &lt;char&gt;
1:                           crohns disease
2:               inflammatory bowel disease
3:                       ulcerative colitis
4:   inflammatory bowel disease, leukopenia
5:     inflammatory bowel disease, alopecia
6: inflammatory bowel disease, pancreatitis
                       l2_all_disease_terms
                                     &lt;char&gt;
1:               inflammatory bowel disease
2:               inflammatory bowel disease
3:               inflammatory bowel disease
4:   inflammatory bowel disease, leukopenia
5:     alopecia, inflammatory bowel disease
6: inflammatory bowel disease, pancreatitis</code></pre>
</div>
<div id="vascular-intestinal-disorders" class="section level2"
number="7.7">
<h2><span class="header-section-number">7.7</span> Vascular intestinal
disorders</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;vascular intestinal disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
</div>
<div id="gallbladder-and-biliary-diseases" class="section level2"
number="7.8">
<h2><span class="header-section-number">7.8</span> Gallbladder and
biliary diseases</h2>
<pre class="r"><code>gal_bile_terms = c(&quot;gallbladder disease&quot;,
                   &quot;bile duct disorder&quot;,
                   &quot;biliary tract disease&quot;,
                   &quot;cholelithiasis&quot;,
                   &quot;cholecystitis&quot;)


gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(gal_bile_terms),
                                   &quot;gallbladder and biliary diseases&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(&quot;gallbladder and biliary diseases&quot;, 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                    all_disease_terms
                               &lt;char&gt;
1: sickle cell anemia, cholelithiasis
2:                     cholelithiasis
3:                 bile duct disorder
4:                gallbladder disease
5:              biliary tract disease
6:      cholelithiasis, cholecystitis
                                                         l2_all_disease_terms
                                                                       &lt;char&gt;
1: gallbladder and biliary diseases, sickle cell disease and related diseases
2:                                           gallbladder and biliary diseases
3:                                           gallbladder and biliary diseases
4:                                           gallbladder and biliary diseases
5:                                           gallbladder and biliary diseases
6:         gallbladder and biliary diseases, gallbladder and biliary diseases</code></pre>
</div>
<div id="pancreatitis" class="section level2" number="7.9">
<h2><span class="header-section-number">7.9</span> Pancreatitis</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(&quot;pancreatitis&quot;, 
              l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                               all_disease_terms
                                                                          &lt;char&gt;
1:         alcoholic liver cirrhosis, alcoholic pancreatitis, alcohol dependence
2:                             alcoholic liver cirrhosis, alcoholic pancreatitis
3:                                      inflammatory bowel disease, pancreatitis
4:         acute lymphoblastic leukemia, asparaginase-induced acute pancreatitis
5: autoimmune pancreatitis type 1, salivary gland lesion, lachrymal gland lesion
6:                                                        alcoholic pancreatitis
                                               l2_all_disease_terms
                                                             &lt;char&gt;
1: alcohol-related disorders, alcoholic liver disease, pancreatitis
2:                            alcoholic liver disease, pancreatitis
3:                         inflammatory bowel disease, pancreatitis
4:                                           leukemia, pancreatitis
5:      lachrymal gland lesion, pancreatitis, salivary gland lesion
6:                                                     pancreatitis</code></pre>
</div>
<div id="other-digestive-diseases" class="section level2" number="7.10">
<h2><span class="header-section-number">7.10</span> Other digestive
diseases</h2>
<pre class="r"><code># I84-I84.9, 
# K20-K20.9, 
# K22-K22.6, 
# K22.8-K24, 
# K31-K31.8, 
# K38-K38.2, 
# K57-K62, 
# K62.2-K62.6, 
# K62.8-K62.9, 
# K64-K64.9, 
# K66.8, K67, 
# K68, 
# K77, 
# K90-K90.9, 
# K92.8, 
# K93.8

# 579 - celiac disease

other_digestive_terms &lt;- c(&quot;esophagitis&quot;,
                           &quot;eosinophilic esophagitis&quot;,
                           &quot;esophageal ulcer&quot;,
                           # &quot;barretts esophagus&quot;,
                           &quot;diverticulitis&quot;,
                           &quot;celiac disease&quot;,
                           &quot;irritable bowel syndrome&quot;,
                           &quot;anal fissure&quot;,
                           &quot;anal fistula&quot;,
                           #? &quot;anal polyp&quot;
                           &quot;rectal prolapse&quot;,
                           &quot;rectal abscess&quot;
                           )

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(other_digestive_terms),
                                   &quot;other digestive diseases&quot;
                          )
 )


gwas_study_info |&gt;
 filter(grepl(&quot;other digestive diseases&quot;, 
              l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                                                                                                                                                                                all_disease_terms
                                                                                                                                                                                                                                                           &lt;char&gt;
1: autoimmune thyroid disease, systemic lupus erythematosus, type 1 diabetes mellitus, ankylosing spondylitis, psoriasis, common variable immunodeficiency, celiac disease, ulcerative colitis, crohns disease, autoimmune disease, juvenile idiopathic arthritis
2:                                                                                                                                                                                                                                                 celiac disease
3:                                                                                                                                                   rheumatoid arthritis, frontotemporal dementia, psoriasis, celiac disease, ulcerative colitis, crohns disease
4:                                                                                                                  rheumatoid arthritis, type 1 diabetes mellitus, psoriasis, celiac disease, ulcerative colitis, crohns disease, progressive supranuclear palsy
5:                                                                                                                   rheumatoid arthritis, amyotrophic lateral sclerosis, type 1 diabetes mellitus, psoriasis, celiac disease, ulcerative colitis, crohns disease
6:                                                                                                              rheumatoid arthritis, type 1 diabetes mellitus, corticobasal degeneration disorder, psoriasis, celiac disease, ulcerative colitis, crohns disease
                                                                                                                                                                                                                                                                           l2_all_disease_terms
                                                                                                                                                                                                                                                                                         &lt;char&gt;
1: ankylosing spondylitis, autoimmune disease, autoimmune thyroid disease, other digestive diseases, common variable immunodeficiency, inflammatory bowel disease, juvenile idiopathic arthritis, psoriasis, systemic lupus erythematosus, type 1 diabetes mellitus, inflammatory bowel disease
2:                                                                                                                                                                                                                                                                     other digestive diseases
3:                                                                                                                                                   other digestive diseases, inflammatory bowel disease, frontotemporal dementia, psoriasis, rheumatoid arthritis, inflammatory bowel disease
4:                                                                                                                  other digestive diseases, inflammatory bowel disease, progressive supranuclear palsy, psoriasis, rheumatoid arthritis, type 1 diabetes mellitus, inflammatory bowel disease
5:                                                                                                                   amyotrophic lateral sclerosis, other digestive diseases, inflammatory bowel disease, psoriasis, rheumatoid arthritis, type 1 diabetes mellitus, inflammatory bowel disease
6:                                                                                                              other digestive diseases, corticobasal degeneration disorder, inflammatory bowel disease, psoriasis, rheumatoid arthritis, type 1 diabetes mellitus, inflammatory bowel disease</code></pre>
</div>
</div>
<div id="neurological-disorders" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Neurological
disorders</h1>
<div id="alzheimers-disease-and-other-dementias" class="section level2"
number="8.1">
<h2><span class="header-section-number">8.1</span> Alzheimer’s disease
and other dementias</h2>
<pre class="r"><code>dementia &lt;- c(&quot;alzheimers disease biomarker measurement&quot;,
              &quot;alzheimers disease neuropathologic change&quot;,
              &quot;aids dementia&quot;,
              &quot;dementia&quot;,
              &quot;frontotemporal dementia&quot;,
              &quot;lewy body dementia&quot;,
              &quot;vascular dementia&quot;,
              &quot;alzheimers disease&quot;,
              &quot;neurodegenerative disease&quot;
)

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(dementia),
                                   &quot;alzheimer&#39;s disease and other dementias&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;alzheimer&#39;s disease and other dementias&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                         all_disease_terms
                                                    &lt;char&gt;
1:                                       alzheimer disease
2:                       age of onset of alzheimer disease
3:                         hypertension, alzheimer disease
4:                    family history of alzheimers disease
5: alzheimer disease, family history of alzheimers disease
6:                                      lewy body dementia
                                                                     l2_all_disease_terms
                                                                                   &lt;char&gt;
1:                                                alzheimer&#39;s disease and other dementias
2:                                                alzheimer&#39;s disease and other dementias
3: alzheimer&#39;s disease and other dementias, other cardiovascular and circulatory diseases
4:                                                alzheimer&#39;s disease and other dementias
5:                                                alzheimer&#39;s disease and other dementias
6:                                                alzheimer&#39;s disease and other dementias</code></pre>
</div>
<div id="parkinsons-disease" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Parkinsons
disease</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;parkinsons disease&quot;,
                                   &quot;parkinson&#39;s disease&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(&quot;parkinson&#39;s disease&quot;, l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                              all_disease_terms
                                                         &lt;char&gt;
1:                                            parkinson disease
2:                            age of onset of parkinson disease
3:                   frontotemporal dementia, parkinson disease
4:                       lewy body attribute, parkinson disease
5:                             schizophrenia, parkinson disease
6: dementia, parkinson disease, disease progression measurement
                                           l2_all_disease_terms
                                                         &lt;char&gt;
1:                                          parkinson&#39;s disease
2:                                          parkinson&#39;s disease
3: alzheimer&#39;s disease and other dementias, parkinson&#39;s disease
4: alzheimer&#39;s disease and other dementias, parkinson&#39;s disease
5:                           parkinson&#39;s disease, schizophrenia
6: alzheimer&#39;s disease and other dementias, parkinson&#39;s disease</code></pre>
</div>
<div id="idiopathic-epilepsy" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Idiopathic
epilepsy</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;idiopathic generalized epilepsy&quot;,
                                   &quot;idiopathic epilepsy&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(&quot;idiopathic epilepsy&quot;, l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>             all_disease_terms l2_all_disease_terms
                        &lt;char&gt;               &lt;char&gt;
1:  childhood absence epilepsy  idiopathic epilepsy
2:   juvenile absence epilepsy  idiopathic epilepsy
3: juvenile myoclonic epilepsy  idiopathic epilepsy</code></pre>
</div>
<div id="multiple-sclerosis" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Multiple
sclerosis</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(&quot;multiple sclerosis&quot;, l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                   all_disease_terms
                                              &lt;char&gt;
1:                                multiple sclerosis
2:      type 2 diabetes mellitus, multiple sclerosis
3:  multiple sclerosis, chronic lymphocytic leukemia
4: diffuse large b-cell lymphoma, multiple sclerosis
5:           follicular lymphoma, multiple sclerosis
6: marginal zone b-cell lymphoma, multiple sclerosis
                           l2_all_disease_terms
                                         &lt;char&gt;
1:                           multiple sclerosis
2: multiple sclerosis, type 2 diabetes mellitus
3:                 leukemia, multiple sclerosis
4:     multiple sclerosis, non-hodgkin lymphoma
5:     multiple sclerosis, non-hodgkin lymphoma
6:     multiple sclerosis, non-hodgkin lymphoma</code></pre>
</div>
<div id="motor-neuron-disease" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Motor neuron
disease</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(&quot;motor neuron disease&quot;, l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>      all_disease_terms l2_all_disease_terms
                 &lt;char&gt;               &lt;char&gt;
1: motor neuron disease motor neuron disease</code></pre>
</div>
<div id="headache-disorders" class="section level2" number="8.6">
<h2><span class="header-section-number">8.6</span> Headache
disorders</h2>
<pre class="r"><code>headache_terms &lt;- c(&quot;headache disorder&quot;,
                    &quot;cluster headache&quot;,
                    &quot;migraine&quot;
                    )



gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  #pattern = &quot;\\bheadache disorder\\b|cluster headache\\b|migraine\\b&quot;,
                                  pattern = vec_to_grep_pattern(headache_terms),
                                   &quot;headache disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(&quot;(?&lt;=^|, )headache disorders&quot;, 
              l2_all_disease_terms, 
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                            all_disease_terms
                                       &lt;char&gt;
1: migraine disorder, coronary artery disease
2:                      migraine without aura
3:                          migraine disorder
4:                         migraine with aura
5:                           cluster headache
6:        bipolar disorder, migraine disorder
                         l2_all_disease_terms
                                       &lt;char&gt;
1: ischemic heart disease, headache disorders
2:                         headache disorders
3:                         headache disorders
4:                         headache disorders
5:                         headache disorders
6:       bipolar disorder, headache disorders</code></pre>
</div>
<div id="other-neurological-disorders" class="section level2"
number="8.7">
<h2><span class="header-section-number">8.7</span> Other neurological
disorders</h2>
<pre class="r"><code>other_neuro_terms &lt;- c(&quot;huntington disease&quot;,
                       &quot;hereditary ataxia&quot;,
                       &quot;torsion dystonia&quot;,
                       &quot;x-linked dystonia-parkinsonism&quot;,
                       &quot;isolated dystonia&quot;,
                       &quot;limb dystonia&quot;,
                       &quot;myoclonus&quot;,
                       &quot;restless legs syndrome&quot;,
                        &quot;chronic inflammatory demyelinating polyneuropathy&quot;,
                       &quot;demyelinating disease of central nervous system&quot;,
                       &quot;myasthenia gravis&quot;,
                       &quot;complex regional pain syndrome&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(other_neuro_terms),
                                   &quot;other neurological disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other neurological disorders&quot;), 
              l2_all_disease_terms, 
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                     all_disease_terms
                                                &lt;char&gt;
1:                              restless legs syndrome
2: huntington disease, disease progression measurement
3:                        late-onset myasthenia gravis
4:                      complex regional pain syndrome
5:                  age of onset of huntington disease
6:                                   myasthenia gravis
           l2_all_disease_terms
                         &lt;char&gt;
1: other neurological disorders
2: other neurological disorders
3: other neurological disorders
4: other neurological disorders
5: other neurological disorders
6: other neurological disorders</code></pre>
</div>
</div>
<div id="mental-disorders" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Mental disorders</h1>
<div id="schizophrenia" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Schizophrenia</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;schizophrenia&quot;), 
              l2_all_disease_terms, 
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                  all_disease_terms
                                                             &lt;char&gt;
1:                                                    schizophrenia
2:       major depressive disorder, schizophrenia, bipolar disorder
3:                                  schizophrenia, bipolar disorder
4:                         schizophrenia, major depressive disorder
5:                          schizophrenia, type 2 diabetes mellitus
6: treatment refractory schizophrenia, drug-induced agranulocytosis
                                         l2_all_disease_terms
                                                       &lt;char&gt;
1:                                              schizophrenia
2: bipolar disorder, major depressive disorder, schizophrenia
3:                            bipolar disorder, schizophrenia
4:                   major depressive disorder, schizophrenia
5:                    schizophrenia, type 2 diabetes mellitus
6:                drug-induced agranulocytosis, schizophrenia</code></pre>
</div>
<div id="depressive-disorders" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Depressive
disorders</h2>
<pre class="r"><code>depressive_terms &lt;- c(&quot;depressive disorder&quot;,
                      &quot;depressive symptom&quot;,
                      &quot;depressive episode&quot;,
                      &quot;major depressive disorders&quot;,
                      &quot;major depressive disorder&quot;,
                      &quot;major depressive episode&quot;,
                      &quot;depressive&quot;
)


gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(depressive_terms),
                                   &quot;depressive disorders&quot;
                          )  
        ) |&gt;
   mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;depressive disorder&quot;),
                                   &quot;depressive disorders&quot;
                          )  
        ) 
  
  
gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;depressive disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                            all_disease_terms
                                                       &lt;char&gt;
1:                                  major depressive disorder
2: major depressive disorder, schizophrenia, bipolar disorder
3:                major depressive disorder, bipolar disorder
4:                bipolar disorder, major depressive disorder
5:                   schizophrenia, major depressive disorder
6:  depressive symptom measurement, major depressive disorder
                                    l2_all_disease_terms
                                                  &lt;char&gt;
1:                                  depressive disorders
2: bipolar disorder, depressive disorders, schizophrenia
3:                bipolar disorder, depressive disorders
4:                bipolar disorder, depressive disorders
5:                   depressive disorders, schizophrenia
6:            depressive disorders, depressive disorders</code></pre>
</div>
<div id="anxiety-disorders" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Anxiety
disorders</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mesh/terms/http%253A%252F%252Fid.nlm.nih.gov%252Fmesh%252FD001008/descendants&quot;

anxiety_terms &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 15
[1] &quot;\n Some example terms&quot;
[1] &quot;obsessive-compulsive disorder&quot; &quot;generalized anxiety disorder&quot; 
[3] &quot;neurocirculatory asthenia&quot;     &quot;excoriation disorder&quot;         
[5] &quot;anxiety, separation&quot;          </code></pre>
<pre class="r"><code>anxiety_terms &lt;- c(anxiety_terms, 
                   &quot;obsessive-compulsive symptom measurement&quot;,
                   &quot;obsessive-compulsive disorder&quot;,
                   &quot;obsessive-compulsive&quot;
                   )


gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(anxiety_terms),
                                   &quot;anxiety disorders&quot;
                          )  
        ) |&gt;
   mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;(?&lt;=^|, ) anxiety disorder(?=,|$)|(?&lt;=^|, ) anxiety measurement(?=,|$)&quot;,
                                   &quot;anxiety disorders&quot;
                          )  
        ) |&gt;
     mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;(?&lt;=^|, ) anxiety(?=,|$)&quot;,
                                   &quot;anxiety disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;anxiety disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                         all_disease_terms
                                                                    &lt;char&gt;
1:                                            generalized anxiety disorder
2: obsessive-compulsive disorder, attention deficit hyperactivity disorder
3:                                           obsessive-compulsive disorder
4:                                obsessive-compulsive symptom measurement
5:                 obsessive-compulsive disorder, autism spectrum disorder
6:                                                          panic disorder
        l2_all_disease_terms
                      &lt;char&gt;
1:         anxiety disorders
2:   adhd, anxiety disorders
3:         anxiety disorders
4:         anxiety disorders
5: autism, anxiety disorders
6:         anxiety disorders</code></pre>
</div>
<div id="eating-disorders" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> Eating disorders</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = 
                                    vec_to_grep_pattern(
                                      c(&quot;bulimia nervosa&quot;,
                                        &quot;anorexia nervosa&quot;,
                                        &quot;binge eating&quot;,
                                        &quot;eating disorder&quot;
                                      )
                                    ),
                                  &quot;eating disorders&quot;
                          )  
        ) |&gt;
   mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;anorexia&quot;,
                                  &quot;eating disorders&quot;
                          )  
        )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;eating disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                   all_disease_terms               l2_all_disease_terms
                              &lt;char&gt;                             &lt;char&gt;
1:    bipolar disorder, binge eating eating disorders, bipolar disorder
2:    binge eating, bipolar disorder eating disorders, bipolar disorder
3:                  anorexia nervosa                   eating disorders
4:                   eating disorder                   eating disorders
5:                   bulimia nervosa                   eating disorders
6: bipolar disorder, eating disorder bipolar disorder, eating disorders</code></pre>
</div>
<div id="autism-spectrum-disorders" class="section level2" number="9.5">
<h2><span class="header-section-number">9.5</span> Autism spectrum
disorders</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(&quot;autism&quot;),
                                   &quot;autism spectrum disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;autism spectrum disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                         all_disease_terms
                                                    &lt;char&gt;
1:                        autism spectrum disorder symptom
2:                                       asperger syndrome
3:                 autism spectrum disorder, schizophrenia
4:                                autism spectrum disorder
5: obsessive-compulsive disorder, autism spectrum disorder
6:                                                  autism
                           l2_all_disease_terms
                                         &lt;char&gt;
1:                    autism spectrum disorders
2:                    autism spectrum disorders
3:     autism spectrum disorders, schizophrenia
4:                    autism spectrum disorders
5: autism spectrum disorders, anxiety disorders
6:                    autism spectrum disorders</code></pre>
</div>
<div id="adhd" class="section level2" number="9.6">
<h2><span class="header-section-number">9.6</span> ADHD</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;adhd&quot;),
                                   &quot;attention-deficit/hyperactivity disorder&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;attention-deficit/hyperactivity disorder&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                                                all_disease_terms
                                                                                                                           &lt;char&gt;
1:                                                        obsessive-compulsive disorder, attention deficit hyperactivity disorder
2:                                                                                       attention deficit hyperactivity disorder
3:                                            attention deficit hyperactivity disorder, oppositional defiant disorder measurement
4:                                                                     attention deficit hyperactivity disorder, conduct disorder
5: attention deficit hyperactivity disorder, bipolar disorder, autism spectrum disorder, schizophrenia, major depressive disorder
6:                                                                     attention deficit hyperactivity disorder, bipolar disorder
                                                                                                         l2_all_disease_terms
                                                                                                                       &lt;char&gt;
1:                                                                attention-deficit/hyperactivity disorder, anxiety disorders
2:                                                                                   attention-deficit/hyperactivity disorder
3:                                                    attention-deficit/hyperactivity disorder, oppositional defiant disorder
4:                                                                 attention-deficit/hyperactivity disorder, conduct disorder
5: attention-deficit/hyperactivity disorder, autism spectrum disorders, bipolar disorder, depressive disorders, schizophrenia
6:                                                                 attention-deficit/hyperactivity disorder, bipolar disorder</code></pre>
</div>
<div id="conduct-disorder" class="section level2" number="9.7">
<h2><span class="header-section-number">9.7</span> Conduct disorder</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;conduct disorder&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                           all_disease_terms
                                                                                      &lt;char&gt;
1:                                                                          conduct disorder
2:                                attention deficit hyperactivity disorder, conduct disorder
3: attention deficit hyperactivity disorder, conduct disorder, oppositional defiant disorder
                                                                        l2_all_disease_terms
                                                                                      &lt;char&gt;
1:                                                                          conduct disorder
2:                                attention-deficit/hyperactivity disorder, conduct disorder
3: attention-deficit/hyperactivity disorder, conduct disorder, oppositional defiant disorder</code></pre>
</div>
<div id="idiopathic-developmental-intellectual-disability"
class="section level2" number="9.8">
<h2><span class="header-section-number">9.8</span> Idiopathic
developmental intellectual disability</h2>
<pre class="r"><code>terms &lt;- c(&quot;developmental disability&quot;)

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(terms),
                                   &quot;idiopathic developmental intellectual disability&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;idiopathic developmental intellectual disability&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>          all_disease_terms                             l2_all_disease_terms
                     &lt;char&gt;                                           &lt;char&gt;
1: developmental disability idiopathic developmental intellectual disability</code></pre>
</div>
<div id="other-mental-disorders" class="section level2" number="9.9">
<h2><span class="header-section-number">9.9</span> Other mental
disorders</h2>
<div id="personality-disorders" class="section level3" number="9.9.1">
<h3><span class="header-section-number">9.9.1</span> Personality
disorders</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0002028/descendants&quot;

personality_disorders &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 10
[1] &quot;\n Some example terms&quot;
[1] &quot;obsessive-compulsive personality disorder&quot;
[2] &quot;narcissistic personality disorder&quot;        
[3] &quot;schizotypal personality disorder&quot;         
[4] &quot;histrionic personality disorder&quot;          
[5] &quot;antisocial personality disorder&quot;          </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(personality_disorders),
                                   &quot;personality disorders&quot;
                          )  
        )</code></pre>
</div>
<div id="mood-disorders" class="section level3" number="9.9.2">
<h3><span class="header-section-number">9.9.2</span> Mood disorders</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0004247/descendants&quot;

mood_disorders &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 10
[1] &quot;\n Some example terms&quot;
[1] &quot;mixed anxiety and depressive disorder&quot;
[2] &quot;treatment resistant depression&quot;       
[3] &quot;major depressive disorder&quot;            
[4] &quot;postpartum depression&quot;                
[5] &quot;depressive disorder&quot;                  </code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(mood_disorders),
                                   &quot;mood disorder&quot;
                          )  
        )</code></pre>
</div>
<div id="sleep-disorders" class="section level3" number="9.9.3">
<h3><span class="header-section-number">9.9.3</span> Sleep
disorders</h3>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/doid/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FDOID_535/descendants&quot;

sleep_disorders &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 16
[1] &quot;\n Some example terms&quot;
[1] &quot;periodic limb movement disorder&quot; &quot;advanced sleep phase syndrome 3&quot;
[3] &quot;advanced sleep phase syndrome 2&quot; &quot;advanced sleep phase syndrome 1&quot;
[5] &quot;advanced sleep phase syndrome 4&quot;</code></pre>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0008568/descendants&quot;

other_sleep_disorders &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 26
[1] &quot;\n Some example terms&quot;
[1] &quot;autosomal dominant cerebellar ataxia, deafness and narcolepsy&quot;
[2] &quot;hereditary sensory neuropathy-deafness-dementia syndrome&quot;     
[3] &quot;rapid eye movement sleep disorder&quot;                            
[4] &quot;substance-induced sleep disorder&quot;                             
[5] &quot;drug induced central sleep apnea&quot;                             </code></pre>
<pre class="r"><code>sleep_disorders &lt;- c(sleep_disorders,
                    other_sleep_disorders)

sleep_disorders &lt;- str_length_sort(sleep_disorders)

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(sleep_disorders),
                                   &quot;sleep disorders&quot;
                          )  
        )</code></pre>
<pre class="r"><code>other_mental_disorders &lt;- c(&quot;manic or hypomanic episode&quot;,
                            &quot;mental or behavioural disorder&quot;,
                            &quot;mental disorder&quot;,
                            &quot;post-traumatic stress disorder&quot;,
                            &quot;stress-related disorder&quot;,
                            &quot;acute stress reaction&quot;,
                            &quot;occupation-related stress disorder&quot;,
                            &quot;psychotic symptom&quot;,
                            &quot;psychosis&quot;,
                            &quot;psychiatric disorder&quot;,
                             &quot;personality disorders&quot;,
                            &quot;personality disorder&quot;,
                            &quot;mood disorder&quot;,
                            &quot;sleep disorders&quot;,
                            &quot;sleep disorder&quot;,
                            &quot;mixed anxiety disorders and depressive disorders&quot;,
                            &quot;emotional symptom&quot;,
                            &quot;dissociative disorder&quot;,
                            &quot;hallucinations&quot;,
                            &quot;somatoform disorder&quot;,
                            &quot;schizoaffective disorder&quot;,
                            &quot;phobic disorder&quot;,
                            &quot;psychotic&quot;
                            
)


gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(other_mental_disorders),
                                   &quot;other mental disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other mental disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                            all_disease_terms
                                                       &lt;char&gt;
1:                                           bipolar disorder
2: major depressive disorder, schizophrenia, bipolar disorder
3:                major depressive disorder, bipolar disorder
4:                            schizophrenia, bipolar disorder
5:                bipolar disorder, major depressive disorder
6:                             bipolar disorder, binge eating
                                          l2_all_disease_terms
                                                        &lt;char&gt;
1:                                      other mental disorders
2: other mental disorders, depressive disorders, schizophrenia
3:                other mental disorders, depressive disorders
4:                       other mental disorders, schizophrenia
5:                other mental disorders, depressive disorders
6:                    eating disorders, other mental disorders</code></pre>
</div>
</div>
</div>
<div id="substance-use-disorders" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Substance use
disorders</h1>
<div id="alcohol-use-disorders" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Alcohol use
disorders</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = 
                                    vec_to_grep_pattern(
                                      c(&quot;alcohol-related disorders&quot;,
                                        &quot;alcohol and nicotine codependence&quot;,
                                        &quot;alcohol use disorder&quot;
                                      )),
                                   &quot;alcohol use disorders&quot;
                          )  
        )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;alcohol use disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                       all_disease_terms
                                                                  &lt;char&gt;
1:                                                    alcohol dependence
2: alcoholic liver cirrhosis, alcoholic pancreatitis, alcohol dependence
3:                                        alcohol dependence measurement
4:                                     alcohol and nicotine codependence
5:                                                    alcohol withdrawal
6:                                    age of onset of alcohol dependence
                                           l2_all_disease_terms
                                                         &lt;char&gt;
1:                                        alcohol use disorders
2: alcohol use disorders, alcoholic liver disease, pancreatitis
3:                                        alcohol use disorders
4:                                        alcohol use disorders
5:                                        alcohol use disorders
6:                                        alcohol use disorders</code></pre>
</div>
<div id="opioid-use-disorders" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Opioid use
disorders</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(
                                    c(&quot;opioid-related disorders&quot;,
                                      &quot;opioid dependence&quot;,
                                      &quot;opioid use disorder&quot;
                                      )
                                    ),
                                   &quot;opioid use disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;opioid use disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>      all_disease_terms l2_all_disease_terms
                 &lt;char&gt;               &lt;char&gt;
1:    opioid dependence opioid use disorders
2:  opioid use disorder opioid use disorders
3: opioid use disorder, opioid use disorders</code></pre>
</div>
<div id="cocaine-use-disorders" class="section level2" number="10.3">
<h2><span class="header-section-number">10.3</span> Cocaine use
disorders</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;cocaine-related disorders&quot;),
                                   &quot;cocaine use disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;cocaine use disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>      all_disease_terms  l2_all_disease_terms
                 &lt;char&gt;                &lt;char&gt;
1:   cocaine dependence cocaine use disorders
2: cocaine use disorder cocaine use disorders</code></pre>
</div>
<div id="amphetamine-use-disorders" class="section level2"
number="10.4">
<h2><span class="header-section-number">10.4</span> Amphetamine use
disorders</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;methamphetamine&quot;,
                                   &quot;amphetamine&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(&quot;amphetamine&quot;, 
              l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                   all_disease_terms
                                                              &lt;char&gt;
1:                                        methamphetamine dependence
2:                                 methamphetamine-induced psychosis
3: alcohol dependence, heroin dependence, methamphetamine dependence
                                                  l2_all_disease_terms
                                                                &lt;char&gt;
1:                                           amphetamine use disorders
2:                                           amphetamine use disorders
3: alcohol use disorders, heroin dependence, amphetamine use disorders</code></pre>
</div>
<div id="cannabis-use-disorders" class="section level2" number="10.5">
<h2><span class="header-section-number">10.5</span> Cannabis use
disorders</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;cannabis dependence&quot;),
                                   &quot;cannabis use disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;cannabis use disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                     all_disease_terms
                                                &lt;char&gt;
1:                                 cannabis dependence
2:                     cannabis dependence measurement
3: cannabis dependence, schizophrenia, substance abuse
4:                cannabis dependence, substance abuse
5:      hiv infection, cannabis dependence measurement
                                     l2_all_disease_terms
                                                   &lt;char&gt;
1:                                 cannabis use disorders
2:                                 cannabis use disorders
3: cannabis use disorders, schizophrenia, substance abuse
4:                cannabis use disorders, substance abuse
5:                                 cannabis use disorders</code></pre>
</div>
<div id="other-drug-use-disorders" class="section level2" number="10.6">
<h2><span class="header-section-number">10.6</span> Other drug use
disorders</h2>
<pre class="r"><code>other_drug_use_terms &lt;- c(&quot;heroin dependence&quot;,
                          &quot;drug dependence&quot;,
                          &quot;nictone dependence&quot;,
                          &quot;substance abuse&quot;,
                          &quot;drug misuse&quot;,
                          &quot;nicotine-related disorders&quot;
                          )

gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(other_drug_use_terms),
                                   &quot;other drug use disorders&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other drug use disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                   all_disease_terms     l2_all_disease_terms
                              &lt;char&gt;                   &lt;char&gt;
1: nicotine dependence symptom count other drug use disorders
2:               nicotine dependence other drug use disorders
3: nicotine withdrawal symptom count other drug use disorders
4:   nicotine withdrawal measurement other drug use disorders
5:                 heroin dependence other drug use disorders
6:                   drug dependence other drug use disorders</code></pre>
</div>
</div>
<div id="diabetes-and-kidney-diseases" class="section level1"
number="11">
<h1><span class="header-section-number">11</span> Diabetes and kidney
diseases</h1>
<div id="diabetes-mellitus-type-1" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Diabetes mellitus
type 1</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;type 1 diabetes mellitus&quot;),
                                   &quot;diabetes mellitus type 1&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;diabetes mellitus type 1&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                                                                                                                                                                                all_disease_terms
                                                                                                                                                                                                                                                           &lt;char&gt;
1:                                                                                                                                                                                                                                       type 1 diabetes mellitus
2: autoimmune thyroid disease, systemic lupus erythematosus, type 1 diabetes mellitus, ankylosing spondylitis, psoriasis, common variable immunodeficiency, celiac disease, ulcerative colitis, crohns disease, autoimmune disease, juvenile idiopathic arthritis
3:                                                                                                                  rheumatoid arthritis, type 1 diabetes mellitus, psoriasis, celiac disease, ulcerative colitis, crohns disease, progressive supranuclear palsy
4:                                                                                                                   rheumatoid arthritis, amyotrophic lateral sclerosis, type 1 diabetes mellitus, psoriasis, celiac disease, ulcerative colitis, crohns disease
5:                                                                                                              rheumatoid arthritis, type 1 diabetes mellitus, corticobasal degeneration disorder, psoriasis, celiac disease, ulcerative colitis, crohns disease
6:                                                                                                                                                                                  type 1 diabetes mellitus, type 2 diabetes mellitus, neuropathy, diabetic foot
                                                                                                                                                                                                                                                                           l2_all_disease_terms
                                                                                                                                                                                                                                                                                         &lt;char&gt;
1:                                                                                                                                                                                                                                                                     diabetes mellitus type 1
2: ankylosing spondylitis, autoimmune disease, autoimmune thyroid disease, other digestive diseases, common variable immunodeficiency, inflammatory bowel disease, juvenile idiopathic arthritis, psoriasis, systemic lupus erythematosus, diabetes mellitus type 1, inflammatory bowel disease
3:                                                                                                                  other digestive diseases, inflammatory bowel disease, progressive supranuclear palsy, psoriasis, rheumatoid arthritis, diabetes mellitus type 1, inflammatory bowel disease
4:                                                                                                                   amyotrophic lateral sclerosis, other digestive diseases, inflammatory bowel disease, psoriasis, rheumatoid arthritis, diabetes mellitus type 1, inflammatory bowel disease
5:                                                                                                              other digestive diseases, corticobasal degeneration disorder, inflammatory bowel disease, psoriasis, rheumatoid arthritis, diabetes mellitus type 1, inflammatory bowel disease
6:                                                                                                                                                                                                                diabetic foot, neuropathy, diabetes mellitus type 1, type 2 diabetes mellitus</code></pre>
</div>
<div id="diabetes-mellitus-type-2" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Diabetes mellitus
type 2</h2>
<pre class="r"><code>gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;type 2 diabetes mellitus&quot;),
                                   &quot;diabetes mellitus type 2&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(&quot;diabetes mellitus type 2&quot;, 
              l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                      all_disease_terms
                                                                 &lt;char&gt;
1:                                             type 2 diabetes mellitus
2:                         type 2 diabetes mellitus, multiple sclerosis
3:                       type 2 diabetes mellitus, diabetic maculopathy
4:           type 2 diabetes mellitus, obesity, coronary artery disease
5:                              schizophrenia, type 2 diabetes mellitus
6: alzheimer disease, type 2 diabetes mellitus, coronary artery disease
                                                                        l2_all_disease_terms
                                                                                      &lt;char&gt;
1:                                                                  diabetes mellitus type 2
2:                                              multiple sclerosis, diabetes mellitus type 2
3:                                            diabetic eye disease, diabetes mellitus type 2
4:                                 ischemic heart disease, obesity, diabetes mellitus type 2
5:                                                   schizophrenia, diabetes mellitus type 2
6: alzheimer&#39;s disease and other dementias, ischemic heart disease, diabetes mellitus type 2</code></pre>
</div>
<div id="chronic-kidney-disease" class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> Chronic kidney
disease</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;chronic kidney disease&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                        all_disease_terms
                                                                   &lt;char&gt;
1:                                                 chronic kidney disease
2:                                    chronic kidney disease, proteinuria
3:                         diabetic nephropathy, type 2 diabetes mellitus
4:                                                   diabetic nephropathy
5: type 1 diabetes mellitus, chronic kidney disease, diabetic nephropathy
6:                         type 1 diabetes mellitus, diabetic nephropathy
                               l2_all_disease_terms
                                             &lt;char&gt;
1:                           chronic kidney disease
2:              chronic kidney disease, proteinuria
3: chronic kidney disease, diabetes mellitus type 2
4:                           chronic kidney disease
5: chronic kidney disease, diabetes mellitus type 1
6: chronic kidney disease, diabetes mellitus type 1</code></pre>
</div>
<div id="acute-glomerulonephritis" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> Acute
glomerulonephritis</h2>
<pre class="r"><code>glomerulonephritis_terms &lt;- c(&quot;chronic glomerulonephritis&quot;,
                              &quot;membranous glomerulonephritis&quot;,
                              &quot;proliferative glomerulonephritis&quot;)

gwas_study_info |&gt;
  mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(glomerulonephritis_terms),
                                   &quot;glomerulonephritis&quot;
                          )  
        )</code></pre>
<pre><code>        DATE_ADDED_TO_CATALOG PUBMED_ID FIRST_AUTHOR       DATE
                       &lt;IDat&gt;     &lt;int&gt;       &lt;char&gt;     &lt;IDat&gt;
     1:            2018-01-17  29093273    Almgren P 2017-11-02
     2:            2018-01-17  29093273    Almgren P 2017-11-02
     3:            2018-01-17  29093273    Almgren P 2017-11-02
     4:            2018-01-17  29093273    Almgren P 2017-11-02
     5:            2018-01-17  29093273    Almgren P 2017-11-02
    ---                                                        
142851:            2024-09-12  37984853   Eissman JM 2023-11-20
142852:            2024-09-12  37984853   Eissman JM 2023-11-20
142853:            2024-09-12  37984853   Eissman JM 2023-11-20
142854:            2024-09-12  37984853   Eissman JM 2023-11-20
142855:            2024-09-12  37984853   Eissman JM 2023-11-20
                  JOURNAL                                 LINK
                   &lt;char&gt;                               &lt;char&gt;
     1:       JCI Insight www.ncbi.nlm.nih.gov/pubmed/29093273
     2:       JCI Insight www.ncbi.nlm.nih.gov/pubmed/29093273
     3:       JCI Insight www.ncbi.nlm.nih.gov/pubmed/29093273
     4:       JCI Insight www.ncbi.nlm.nih.gov/pubmed/29093273
     5:       JCI Insight www.ncbi.nlm.nih.gov/pubmed/29093273
    ---                                                       
142851: Alzheimers Dement www.ncbi.nlm.nih.gov/pubmed/37984853
142852: Alzheimers Dement www.ncbi.nlm.nih.gov/pubmed/37984853
142853: Alzheimers Dement www.ncbi.nlm.nih.gov/pubmed/37984853
142854: Alzheimers Dement www.ncbi.nlm.nih.gov/pubmed/37984853
142855: Alzheimers Dement www.ncbi.nlm.nih.gov/pubmed/37984853
                                                                     STUDY
                                                                    &lt;char&gt;
     1:  Genetic determinants of circulating GIP and GLP-1 concentrations.
     2:  Genetic determinants of circulating GIP and GLP-1 concentrations.
     3:  Genetic determinants of circulating GIP and GLP-1 concentrations.
     4:  Genetic determinants of circulating GIP and GLP-1 concentrations.
     5:  Genetic determinants of circulating GIP and GLP-1 concentrations.
    ---                                                                   
142851: Sex-specific genetic architecture of late-life memory performance.
142852: Sex-specific genetic architecture of late-life memory performance.
142853: Sex-specific genetic architecture of late-life memory performance.
142854: Sex-specific genetic architecture of late-life memory performance.
142855: Sex-specific genetic architecture of late-life memory performance.
                                                                 DISEASE/TRAIT
                                                                        &lt;char&gt;
     1:    Insulin levels in response to oral glucose tolerance test (fasting)
     2:   Glucagon levels in response to oral glucose tolerance test (fasting)
     3:        GIP levels in response to oral glucose tolerance test (fasting)
     4:      GLP-1 levels in response to oral glucose tolerance test (fasting)
     5: Insulin levels in response to oral glucose tolerance test (30 minutes)
    ---                                                                       
142851:                  Baseline memory in normal cognition x sex interaction
142852:                                      Baseline memory x sex interaction
142853:                Baseline memory in impaired cognition x sex interaction
142854:                  Baseline memory in normal cognition x sex interaction
142855:                                      Baseline memory x sex interaction
                                                             INITIAL_SAMPLE_SIZE
                                                                          &lt;char&gt;
     1:                                       3,344 Swedish ancestry individuals
     2:                                       3,344 Swedish ancestry individuals
     3:                                       3,344 Swedish ancestry individuals
     4:                                       3,344 Swedish ancestry individuals
     5:                                       3,344 Swedish ancestry individuals
    ---                                                                         
142851: 12,789 European ancestry individuals, 1,775 African ancestry individuals
142852:                                       3,367 African ancestry individuals
142853:                                       1,242 African ancestry individuals
142854:                                       1,775 African ancestry individuals
142855:                                     24,216 European ancestry individuals
                   REPLICATION_SAMPLE_SIZE          PLATFORM_[SNPS_PASSING_QC]
                                    &lt;char&gt;                              &lt;char&gt;
     1: 4,905 Finnish ancestry individuals           Illumina [at least 81396]
     2: 4,905 Finnish ancestry individuals           Illumina [at least 81396]
     3: 4,905 Finnish ancestry individuals           Illumina [at least 81396]
     4: 4,905 Finnish ancestry individuals           Illumina [at least 81396]
     5: 4,905 Finnish ancestry individuals           Illumina [at least 81396]
    ---                                                                       
142851:                                    Affymetrix, Illumina [NR] (imputed)
142852:                                    Affymetrix, Illumina [NR] (imputed)
142853:                                    Affymetrix, Illumina [NR] (imputed)
142854:                                    Affymetrix, Illumina [NR] (imputed)
142855:                                    Affymetrix, Illumina [NR] (imputed)
        ASSOCIATION_COUNT
                    &lt;int&gt;
     1:                 1
     2:                 4
     3:                 2
     4:                 4
     5:                 3
    ---                  
142851:                56
142852:                55
142853:                37
142854:                48
142855:               106
                                                                        MAPPED_TRAIT
                                                                              &lt;char&gt;
     1:                                  insulin measurement, glucose tolerance test
     2:                                 glucagon measurement, glucose tolerance test
     3: glucose-dependent insulinotropic peptide measurement, glucose tolerance test
     4:                  glucagon-like peptide-1 measurement, glucose tolerance test
     5:                                  insulin measurement, glucose tolerance test
    ---                                                                             
142851:                              memory performance, sex interaction measurement
142852:                              memory performance, sex interaction measurement
142853:                              memory performance, sex interaction measurement
142854:                              memory performance, sex interaction measurement
142855:                              memory performance, sex interaction measurement
                                                                  MAPPED_TRAIT_URI
                                                                            &lt;char&gt;
     1: http://www.ebi.ac.uk/efo/EFO_0004467, http://www.ebi.ac.uk/efo/EFO_0004307
     2: http://www.ebi.ac.uk/efo/EFO_0008463, http://www.ebi.ac.uk/efo/EFO_0004307
     3: http://www.ebi.ac.uk/efo/EFO_0008464, http://www.ebi.ac.uk/efo/EFO_0004307
     4: http://www.ebi.ac.uk/efo/EFO_0008465, http://www.ebi.ac.uk/efo/EFO_0004307
     5: http://www.ebi.ac.uk/efo/EFO_0004467, http://www.ebi.ac.uk/efo/EFO_0004307
    ---                                                                           
142851: http://www.ebi.ac.uk/efo/EFO_0004874, http://www.ebi.ac.uk/efo/EFO_0008343
142852: http://www.ebi.ac.uk/efo/EFO_0004874, http://www.ebi.ac.uk/efo/EFO_0008343
142853: http://www.ebi.ac.uk/efo/EFO_0004874, http://www.ebi.ac.uk/efo/EFO_0008343
142854: http://www.ebi.ac.uk/efo/EFO_0004874, http://www.ebi.ac.uk/efo/EFO_0008343
142855: http://www.ebi.ac.uk/efo/EFO_0004874, http://www.ebi.ac.uk/efo/EFO_0008343
        STUDY_ACCESSION        GENOTYPING_TECHNOLOGY SUBMISSION_DATE
                 &lt;char&gt;                       &lt;char&gt;          &lt;lgcl&gt;
     1:      GCST005159 Genome-wide genotyping array              NA
     2:      GCST005162 Genome-wide genotyping array              NA
     3:      GCST005167 Genome-wide genotyping array              NA
     4:      GCST005164 Genome-wide genotyping array              NA
     5:      GCST005160 Genome-wide genotyping array              NA
    ---                                                             
142851:    GCST90448438 Genome-wide genotyping array              NA
142852:    GCST90448439 Genome-wide genotyping array              NA
142853:    GCST90448440 Genome-wide genotyping array              NA
142854:    GCST90448441 Genome-wide genotyping array              NA
142855:    GCST90448442 Genome-wide genotyping array              NA
        STATISTICAL_MODEL BACKGROUND_TRAIT          MAPPED_BACKGROUND_TRAIT
                   &lt;lgcl&gt;           &lt;lgcl&gt;                           &lt;char&gt;
     1:                NA               NA                                 
     2:                NA               NA                                 
     3:                NA               NA                                 
     4:                NA               NA                                 
     5:                NA               NA                                 
    ---                                                                    
142851:                NA               NA                                 
142852:                NA               NA                                 
142853:                NA               NA cognitive impairment measurement
142854:                NA               NA                                 
142855:                NA               NA                                 
                 MAPPED_BACKGROUND_TRAIT_URI                COHORT
                                      &lt;char&gt;                &lt;char&gt;
     1:                                                           
     2:                                                           
     3:                                                           
     4:                                                           
     5:                                                           
    ---                                                           
142851:                                      ADNI|NACC|ROSMAP|MARS
142852:                                      ADNI|NACC|ROSMAP|MARS
142853: http://www.ebi.ac.uk/efo/EFO_0007998 ADNI|NACC|ROSMAP|MARS
142854:                                      ADNI|NACC|ROSMAP|MARS
142855:                                      ADNI|NACC|ROSMAP|MARS
        FULL_SUMMARY_STATISTICS SUMMARY_STATS_LOCATION    GXE disease_terms
                         &lt;char&gt;                 &lt;char&gt; &lt;char&gt;        &lt;char&gt;
     1:                      no                            no              
     2:                      no                            no              
     3:                      no                            no              
     4:                      no                            no              
     5:                      no                            no              
    ---                                                                    
142851:                      no                           yes              
142852:                      no                           yes              
142853:                      no                           yes              
142854:                      no                           yes              
142855:                      no                           yes              
        MAPPED_TRAIT_CATEGORY background_disease_terms
                       &lt;char&gt;                   &lt;char&gt;
     1:           Measurement                         
     2:           Measurement                         
     3:           Measurement                         
     4:           Measurement                         
     5:           Measurement                         
    ---                                               
142851:           Measurement                         
142852:           Measurement                         
142853:           Measurement                         
142854:           Measurement                         
142855:           Measurement                         
        BACKGROUND_TRAIT_CATEGORY DISEASE_STUDY all_disease_terms
                           &lt;char&gt;        &lt;lgcl&gt;            &lt;char&gt;
     1:                     Other         FALSE                  
     2:                     Other         FALSE                  
     3:                     Other         FALSE                  
     4:                     Other         FALSE                  
     5:                     Other         FALSE                  
    ---                                                          
142851:                     Other         FALSE                  
142852:                     Other         FALSE                  
142853:               Measurement         FALSE                  
142854:                     Other         FALSE                  
142855:                     Other         FALSE                  
        collected_all_disease_terms l1_all_disease_terms l2_all_disease_terms
                             &lt;char&gt;               &lt;char&gt;               &lt;char&gt;
     1:                                                                      
     2:                                                                      
     3:                                                                      
     4:                                                                      
     5:                                                                      
    ---                                                                      
142851:                                                                      
142852:                                                                      
142853:                                                                      
142854:                                                                      
142855:                                                                      </code></pre>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;acute glomerulonephritis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
</div>
</div>
<div id="skin-and-subcutaneous-diseases" class="section level1"
number="12">
<h1><span class="header-section-number">12</span> Skin and subcutaneous
diseases</h1>
<div id="dermatitis" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Dermatitis</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;dermatitis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                  all_disease_terms l2_all_disease_terms
                             &lt;char&gt;               &lt;char&gt;
1:              pemphigus foliaceus           dermatitis
2:                        pemphigus           dermatitis
3:               pemphigus vulgaris           dermatitis
4:            seborrheic dermatitis           dermatitis
5:                    atopic eczema           dermatitis
6: contact dermatitis due to nickel           dermatitis</code></pre>
</div>
<div id="psoriasis" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Psoriasis</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;psoriasis&quot;),
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                                                                                                                                                                                                all_disease_terms
                                                                                                                                                                                                                                                           &lt;char&gt;
1:                                                                                                                                                                                                                                                      psoriasis
2: autoimmune thyroid disease, systemic lupus erythematosus, type 1 diabetes mellitus, ankylosing spondylitis, psoriasis, common variable immunodeficiency, celiac disease, ulcerative colitis, crohns disease, autoimmune disease, juvenile idiopathic arthritis
3:                                                                                                                                                                                                                     cutaneous psoriasis measurement, psoriasis
4:                                                                                                                                                                                                                                             psoriasis vulgaris
5:                                                                                                                                                                                                                                            psoriatic arthritis
6:                                                                                                                                                                                                cutaneous psoriasis measurement, psoriatic arthritis, psoriasis
                                                                                                                                                                                                                                                                           l2_all_disease_terms
                                                                                                                                                                                                                                                                                         &lt;char&gt;
1:                                                                                                                                                                                                                                                                                    psoriasis
2: ankylosing spondylitis, autoimmune disease, autoimmune thyroid disease, other digestive diseases, common variable immunodeficiency, inflammatory bowel disease, juvenile idiopathic arthritis, psoriasis, systemic lupus erythematosus, diabetes mellitus type 1, inflammatory bowel disease
3:                                                                                                                                                                                                                                                                                    psoriasis
4:                                                                                                                                                                                                                                                                                    psoriasis
5:                                                                                                                                                                                                                                                                                    psoriasis
6:                                                                                                                                                                                                                                                                                    psoriasis</code></pre>
</div>
<div id="bacterial-skin-diseases" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> Bacterial skin
diseases</h2>
<pre class="r"><code>bacterial_skin_disease_terms &lt;- c(&quot;staphylococcal skin infections&quot;,
                                  &quot;skin and soft tissue staphylococcus aureus infection&quot;,
                                  &quot;cellulitis&quot;
                                  # &quot;skin infection&quot;
                                  )

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(bacterial_skin_disease_terms),
                                   &quot;bacterial skin diseases&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;bacterial skin diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                      all_disease_terms
                                                 &lt;char&gt;
1: skin and soft tissue staphylococcus aureus infection
2:                       staphylococcal skin infections
3:                                  cellulitis, abscess
4:                                           cellulitis
5:                             cellulitis, lymphangitis
                    l2_all_disease_terms
                                  &lt;char&gt;
1:               bacterial skin diseases
2:               bacterial skin diseases
3:      abscess, bacterial skin diseases
4:               bacterial skin diseases
5: bacterial skin diseases, lymphangitis</code></pre>
</div>
<div id="scabies" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> Scabies</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;scabies&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
</div>
<div id="fungal-skin-diseases" class="section level2" number="12.5">
<h2><span class="header-section-number">12.5</span> Fungal skin
diseases</h2>
<pre class="r"><code>fungal_skin_disease_terms &lt;- c(&quot;tinea&quot;,
                               &quot;dermatomycosis&quot;)

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(fungal_skin_disease_terms),
                                   &quot;fungal skin diseases&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(&quot;fungal skin diseases&quot;, l2_all_disease_terms)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                 all_disease_terms                       l2_all_disease_terms
                            &lt;char&gt;                                     &lt;char&gt;
1: dermatomycosis, dermatophytosis fungal skin diseases, fungal skin diseases
2:                 dermatophytosis                       fungal skin diseases
3:                   tinea unguium                       fungal skin diseases
4:                     tinea pedis                       fungal skin diseases
5:                  dermatomycosis                       fungal skin diseases
6:                           tinea                       fungal skin diseases</code></pre>
</div>
<div id="viral-skin-diseases" class="section level2" number="12.6">
<h2><span class="header-section-number">12.6</span> Viral skin
diseases</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;viral skin diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
</div>
<div id="acne-vulgaris" class="section level2" number="12.7">
<h2><span class="header-section-number">12.7</span> Acne vulgaris</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(&quot;acne&quot;),
                                   &quot;acne vulgaris&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;acne vulgaris&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>   all_disease_terms l2_all_disease_terms
              &lt;char&gt;               &lt;char&gt;
1:              acne        acne vulgaris</code></pre>
</div>
<div id="pruritus" class="section level2" number="12.8">
<h2><span class="header-section-number">12.8</span> Pruritus</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;pruritus&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                         all_disease_terms
                                    &lt;char&gt;
1: non-alcoholic steatohepatitis, pruritus
2:                                pruritus
                                   l2_all_disease_terms
                                                 &lt;char&gt;
1: cirrhosis and other chronic liver diseases, pruritus
2:                                             pruritus</code></pre>
</div>
<div id="urticaria" class="section level2" number="12.9">
<h2><span class="header-section-number">12.9</span> Urticaria</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;urticaria&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>       all_disease_terms  l2_all_disease_terms
                  &lt;char&gt;                &lt;char&gt;
1: urticaria, angioedema angioedema, urticaria
2:             urticaria             urticaria</code></pre>
</div>
<div id="decubitus-ulcer" class="section level2" number="12.10">
<h2><span class="header-section-number">12.10</span> Decubitus
ulcer</h2>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;decubitus ulcer&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>   all_disease_terms l2_all_disease_terms
              &lt;char&gt;               &lt;char&gt;
1:   decubitus ulcer      decubitus ulcer</code></pre>
</div>
<div id="other-skin-and-subcutaneous-diseases" class="section level2"
number="12.11">
<h2><span class="header-section-number">12.11</span> Other skin and
subcutaneous diseases</h2>
<pre class="r"><code>other_skin_disease_terms &lt;- c(&quot;sebaceous gland disease&quot;,
                              &quot;rosacea&quot;,
                              &quot;erythematosquamous dermatosis&quot;,
                              &quot;dry skin&quot;,
                              &quot;skin tags&quot;,
                              &quot;dermatochalasis&quot;,
                              &quot;epidermal thickening&quot;,
                              &quot;epidermal inclusion cyst&quot;,
                              &quot;cutaneous lupus erythematosus&quot;,
                              &quot;androgenetic alopecia&quot;,
                              &quot;chemotherapy-induced alopecia&quot;,
                              &quot;cutaneous leishmaniasis&quot;)

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(other_skin_disease_terms),
                                   &quot;other skin and subcutaneous diseases&quot;
                          )
 )
                                   

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other skin and subcutaneous diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                 all_disease_terms
                                            &lt;char&gt;
1:                           androgenetic alopecia
2:                   chemotherapy-induced alopecia
3:                    rosacea severity measurement
4:                   cutaneous lupus erythematosus
5:                                         rosacea
6: breast carcinoma, chemotherapy-induced alopecia
                                  l2_all_disease_terms
                                                &lt;char&gt;
1:                other skin and subcutaneous diseases
2:                other skin and subcutaneous diseases
3:                other skin and subcutaneous diseases
4:                other skin and subcutaneous diseases
5:                other skin and subcutaneous diseases
6: breast cancer, other skin and subcutaneous diseases</code></pre>
</div>
</div>
<div id="sense-organ-diseases" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Sense organ
diseases</h1>
<div id="blindness-and-vision-loss" class="section level2"
number="13.1">
<h2><span class="header-section-number">13.1</span> Blindness and vision
loss</h2>
<pre class="r"><code>vision_loss_terms &lt;- c(&quot;blindness&quot;,
                       &quot;color vision disorder&quot;,
                       &quot;vision disorder&quot;,
                       &quot;visuospatial impairment&quot;,
                       &quot;pathological blindness and vision loss&quot;,
                       &quot;visual impairment&quot;,
                       &quot;myopia&quot;,
                       &quot;refractive error&quot;,
                       &quot;hyperopia&quot;,
                       &quot;astigmatism&quot;,
                       &quot;corneal astigmatism&quot;,
                       &quot;presbyopia&quot;,
                       &quot;anisometropia&quot;,
                       &quot;esotropia&quot;,
                       &quot;non-accomodative esotropia&quot;,
                       &quot;accommodative esotropia&quot;,
                       &quot;abnormality of refraction&quot;,
                       &quot;abnormality of vision&quot;,
                       &quot;age-related macular degeneration&quot;,
                       &quot;degeneration of macula and posterior pole&quot;,
                       &quot;age-related cataract&quot;,
                       &quot;cataract&quot;)

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(vision_loss_terms),
                                   &quot;blindness and vision loss&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;blindness and vision loss&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                                           all_disease_terms
                                                                                      &lt;char&gt;
1:                                                                       corneal astigmatism
2:                                                                  wet macular degeneration
3:                                                                          refractive error
4:                                                          age-related macular degeneration
5: atrophic macular degeneration, age-related macular degeneration, wet macular degeneration
6:                                                             atrophic macular degeneration
        l2_all_disease_terms
                      &lt;char&gt;
1: blindness and vision loss
2: blindness and vision loss
3: blindness and vision loss
4: blindness and vision loss
5: blindness and vision loss
6: blindness and vision loss</code></pre>
</div>
<div id="age-related-and-other-hearing-loss" class="section level2"
number="13.2">
<h2><span class="header-section-number">13.2</span> Age-related and
other hearing loss</h2>
<pre class="r"><code>hearing_loss_terms &lt;- c(&quot;sensorineural hearing loss&quot;,
                        &quot;sensorineural hearing impairment&quot;,
                        &quot;age-related hearing impairment&quot;,
                        &quot;conductive hearing impairment&quot;,
                        &quot;noise-induced hearing loss&quot;,
                        &quot;hearing loss&quot;,
                        &quot;meniere disease&quot;,
                        &quot;deafness&quot;)


gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(hearing_loss_terms),
                                   &quot;age-related and other hearing loss&quot;
                          )  
        )


gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;(?&lt;=^|, ) hearing loss(?=,|$)&quot;,
                                   &quot;age-related and other hearing loss&quot;
                          )  
        )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;age-related and other hearing loss&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                   all_disease_terms
                                              &lt;char&gt;
1: charcot-marie-tooth disease type 1a, hearing loss
2:                    age-related hearing impairment
3:                                      hearing loss
4:                        noise-induced hearing loss
5:                         hearing loss, ototoxicity
6:                                   meniere disease
                                                      l2_all_disease_terms
                                                                    &lt;char&gt;
1: charcot-marie-tooth disease type 1a, age-related and other hearing loss
2:                                      age-related and other hearing loss
3:                                      age-related and other hearing loss
4:                                      age-related and other hearing loss
5:                         age-related and other hearing loss, ototoxicity
6:                                      age-related and other hearing loss</code></pre>
</div>
<div id="other-sense-organ-diseases" class="section level2"
number="13.3">
<h2><span class="header-section-number">13.3</span> Other sense organ
diseases</h2>
<pre class="r"><code>other_sense_terms &lt;- c(&quot;abnormality of the sense of smell&quot;,
                       &quot;disturbances of sensation of smell and taste&quot;)

gwas_study_info =
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = vec_to_grep_pattern(other_sense_terms),
                                   &quot;other sense organ diseases&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;other sense organ diseases&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
</div>
</div>
<div id="other-neurodegenerative-disorder-to-do" class="section level1"
number="14">
<h1><span class="header-section-number">14</span> Other
neurodegenerative disorder (to do)</h1>
<p>take url: “<a
href="http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F52522001/descendants"
class="uri">http://www.ebi.ac.uk/ols4/api/ontologies/snomed/terms/http%253A%252F%252Fsnomed.info%252Fid%252F52522001/descendants</a>”</p>
</div>
<div id="other-neurological-disorders-1" class="section level1"
number="15">
<h1><span class="header-section-number">15</span> Other neurological
disorders</h1>
<pre class="r"><code>other_neuro &lt;- c(&quot;mild neurocognitive disorder&quot;,
                 &quot;hiv-associated neurocognitive disorder&quot;)</code></pre>
</div>
<div id="other-non-communicable-diseases" class="section level1"
number="16">
<h1><span class="header-section-number">16</span> Other non-communicable
diseases</h1>
<div id="congenital-birth-defects" class="section level2" number="16.1">
<h2><span class="header-section-number">16.1</span> Congenital birth
defects</h2>
<div id="neural-tube-defects" class="section level3" number="16.1.1">
<h3><span class="header-section-number">16.1.1</span> Neural tube
defects</h3>
</div>
<div id="congenital-heart-defects" class="section level3"
number="16.1.2">
<h3><span class="header-section-number">16.1.2</span> Congenital heart
defects</h3>
</div>
<div id="orofacial-clefts" class="section level3" number="16.1.3">
<h3><span class="header-section-number">16.1.3</span> Orofacial
clefts</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
           stringr::str_replace_all(
             l2_all_disease_terms,
             vec_to_grep_pattern(&quot;orofacial cleft&quot;),
             &quot;orofacial clefts&quot;
           )
  )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;orofacial clefts&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>            all_disease_terms l2_all_disease_terms
                       &lt;char&gt;               &lt;char&gt;
1:                  cleft lip     orofacial clefts
2:               cleft palate     orofacial clefts
3:            orofacial cleft     orofacial clefts
4: orofacial cleft, cleft lip     orofacial clefts
5:    cleft palate, cleft lip     orofacial clefts</code></pre>
</div>
<div id="down-syndrome" class="section level3" number="16.1.4">
<h3><span class="header-section-number">16.1.4</span> Down syndrome</h3>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;down syndrome&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                              all_disease_terms
                                         &lt;char&gt;
1: down syndrome, atrioventricular canal defect
2:                                down syndrome
                                         l2_all_disease_terms
                                                       &lt;char&gt;
1: congenital anomaly of cardiovascular system, down syndrome
2:                                              down syndrome</code></pre>
</div>
<div id="klinefelter-syndrome" class="section level3" number="16.1.5">
<h3><span class="header-section-number">16.1.5</span> Klinefelter
syndrome</h3>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;klinefelter syndrome&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>Empty data.table (0 rows and 2 cols): all_disease_terms,l2_all_disease_terms</code></pre>
<pre class="r"><code>icd10 &lt;- readxl::read_xlsx(here::here(&quot;data/icd/cdc_valid_icd10_Sep_23_2025.xlsx&quot;))

icd10 &lt;- icd10 |&gt;
  rename_with(~ tolower(gsub(&quot; &quot;, &quot;_&quot;, .x))) </code></pre>
</div>
<div id="other-chromosomal-abnormalities" class="section level3"
number="16.1.6">
<h3><span class="header-section-number">16.1.6</span> Other chromosomal
abnormalities</h3>
<pre class="r"><code># Q87-Q87.8, Q91-Q93.9, Q95-Q95.9, Q97-Q97.9, Q99-Q99.8

other_chrom_abn_icd10 &lt;-
  c(paste0(&quot;Q&quot;, seq(87, 87.8, by = 0.1) * 10),
    paste0(&quot;Q&quot;, seq(91, 93.9, by = 0.1) * 10),
    paste0(&quot;Q&quot;, seq(95, 95.9, by = 0.1) * 10),
    paste0(&quot;Q&quot;, seq(97, 97.9, by = 0.1) * 10),
    paste0(&quot;Q&quot;, seq(99, 99.8, by = 0.1) * 10)
    )

other_chrom_abn_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(other_chrom_abn_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()

other_chrom_abn &lt;- c(&quot;marfran syndrome&quot;,
                     &quot;fragile x syndrome&quot;,
                     &quot;22q11.2 deletion syndrome&quot;,
                     &quot;chromosomal disorder&quot;)

url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0019040/descendants&quot;

chromosomal_disorders &lt;- get_descendants(url)</code></pre>
<pre><code>[1] &quot;Number of terms collected:&quot;
[1] 587
[1] &quot;\n Some example terms&quot;
[1] &quot;neurodevelopmental disorder-craniofacial dysmorphism-cardiac defect-hip dysplasia syndrome due to 9q21 microdeletion&quot;
[2] &quot;wilms tumor, aniridia, genitourinary anomalies, intellectual disability, and obesity syndrome&quot;                       
[3] &quot;severe neonatal hypotonia-seizures-encephalopathy syndrome due to 5q31.3 microdeletion&quot;                              
[4] &quot;severe motor and intellectual disabilities-sensorineural deafness-dystonia syndrome&quot;                                 
[5] &quot;syndrome caused by partial chromosomal duplication of the short arm of chromosome 9&quot;                                 </code></pre>
<pre class="r"><code>other_chrom_abn &lt;- c(other_chrom_abn, chromosomal_disorders)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(c(other_chrom_abn_terms, other_chrom_abn)),
                          &quot;other chromosomal abnormalities&quot;
         )
  )</code></pre>
</div>
<div id="congenital-heart-abnormalities" class="section level3"
number="16.1.7">
<h3><span class="header-section-number">16.1.7</span> Congenital heart
abnormalities</h3>
<pre class="r"><code># Q20-Q28.9
congen_heart_icd10 &lt;-
  paste0(&quot;Q&quot;, seq(20, 28.9, by = 0.1) * 10) 


congen_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(congen_heart_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          &quot;congenital anomaly of cardiovascular system&quot;,
                          &quot;congenital heart abnormalities&quot;
         )
  )</code></pre>
</div>
<div id="congenital-musculoskeletal-and-limb-anomalies"
class="section level3" number="16.1.8">
<h3><span class="header-section-number">16.1.8</span> Congenital
musculoskeletal and limb anomalies</h3>
<pre class="r"><code># Q65-Q79, Q79.6-Q79.9
congen_musculo_icd10 &lt;-
  c(paste0(&quot;Q&quot;, seq(65, 79, by = 0.1) * 10),
    paste0(&quot;Q&quot;, seq(79.6, 79.9, by = 0.1) * 10)
    )

congen_musculo_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(congen_musculo_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()


congen_muscle &lt;- c(&quot;osteochondrodysplasia&quot;,
                   &quot;congenital deformities of limbs&quot;,
                   &quot;lower limb asymmetry&quot;,
                   &quot;familial clubfoot with or without associated lower limb anomalies&quot;,
                   &quot;abnormality of limbs&quot;,
                   &quot;abnormal foot morphology&quot;,
                   congen_musculo_terms
                   )


gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(c(congen_musculo_terms, congen_muscle)),
                          &quot;congenital musculoskeletal and limb anomalies&quot;
         )
  )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;congenital musculoskeletal and limb anomalies&quot;), 
              l1_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l1_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                                   all_disease_terms
                                                              &lt;char&gt;
1:     charcot-marie-tooth disease type 1a, abnormal foot morphology
2: familial clubfoot with or without associated lower limb anomalies
3:                                   congenital deformities of limbs
4:                                          abnormal foot morphology
5:                                              abnormality of limbs
6:                                              lower limb asymmetry
                                                             l1_all_disease_terms
                                                                           &lt;char&gt;
1: congenital musculoskeletal and limb anomalies, other chromosomal abnormalities
2:                                  congenital musculoskeletal and limb anomalies
3:                                  congenital musculoskeletal and limb anomalies
4:                                  congenital musculoskeletal and limb anomalies
5:                                  congenital musculoskeletal and limb anomalies
6:                                  congenital musculoskeletal and limb anomalies</code></pre>
</div>
<div id="urogenital-congenital-anomalies" class="section level3"
number="16.1.9">
<h3><span class="header-section-number">16.1.9</span> Urogenital
congenital anomalies</h3>
<pre class="r"><code># P96.0, Q50-Q60.6, Q63-Q64.9

urogenital_icd10 &lt;-
  c(&quot;P960&quot;,
    paste0(&quot;Q&quot;, seq(50, 60.6, by = 0.1) * 10),
    paste0(&quot;Q&quot;, seq(63, 64.9, by = 0.1) * 10)
    )

urogenital_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(urogenital_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()


urogenital_terms &lt;- 
  c(&quot;abnormal morphology of female internal genitalia&quot;,
    &quot;abnormality of the genital system&quot;,
    &quot;functional abnormality of the bladder&quot;,
    &quot;congenital anomaly of kidney and urinary tract&quot;,
    &quot;bladder exstrophy&quot;, # ICD9:753.5
    tolower(urogenital_terms))

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(urogenital_terms),
                          &quot;urogenital congenital anomalies&quot;
         )
  )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;urogenital congenital anomalies&quot;), 
              l1_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l1_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                     all_disease_terms
                                                &lt;char&gt;
1:                                   bladder exstrophy
2: behcets syndrome, abnormality of the genital system
3:                   abnormality of the genital system
4:      congenital anomaly of kidney and urinary tract
5:               functional abnormality of the bladder
6:    abnormal morphology of female internal genitalia
                                l1_all_disease_terms
                                              &lt;char&gt;
1:                   urogenital congenital anomalies
2: urogenital congenital anomalies, behcets syndrome
3:                   urogenital congenital anomalies
4:                   urogenital congenital anomalies
5:                   urogenital congenital anomalies
6:                   urogenital congenital anomalies</code></pre>
</div>
<div id="digestive-congenital-anomalies" class="section level3"
number="16.1.10">
<h3><span class="header-section-number">16.1.10</span> Digestive
congenital anomalies</h3>
<pre class="r"><code># Q38-Q45.9, Q79.0-Q79.5

digestive_icd10 &lt;-
  c(paste0(&quot;Q&quot;, seq(38, 45.9, by = 0.1) * 10),
    paste0(&quot;Q&quot;, seq(79, 79.5, by = 0.1) * 10)
    )

digestive_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(digestive_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()

digestive_terms &lt;- tolower(digestive_terms)
digestive_terms &lt;- stringr::str_remove_all(digestive_terms, &quot;, unspecified$&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(l1_all_disease_terms  = 
         stringr::str_replace_all(l1_all_disease_terms,
                          vec_to_grep_pattern(tolower(digestive_terms)),
                          &quot;digestive congenital anomalies&quot;
         )
  )


gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;digestive congenital anomalies&quot;), 
              l1_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l1_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                  all_disease_terms
                             &lt;char&gt;
1: alagille syndrome, liver disease
                            l1_all_disease_terms
                                          &lt;char&gt;
1: digestive congenital anomalies, liver disease</code></pre>
</div>
</div>
<div id="urinary-diseases-and-male-infertility" class="section level2"
number="16.2">
<h2><span class="header-section-number">16.2</span> Urinary diseases and
male infertility</h2>
<div id="urinary-tract-infections-and-interstitial-nephritis"
class="section level3" number="16.2.1">
<h3><span class="header-section-number">16.2.1</span> Urinary tract
infections and interstitial nephritis</h3>
<pre class="r"><code># N10-N12.9, N13.6, N15, N15.1-N16.8, N30-N30.3, N30.8-N30.9, N34-N34.3, N39.0-N39.2

uti_icd10 &lt;-
  c(paste0(&quot;N&quot;, seq(10, 12.9, by = 0.1) * 10),
    &quot;N136&quot;,
    paste0(&quot;N&quot;, seq(151, 168, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(30, 30.3, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(308, 309, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(34, 34.3, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(390, 392, by = 0.1) * 10)
    )


uti_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(uti_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()

uti_terms &lt;-
  c(&quot;acute cystitis&quot;,
    &quot;chronic cystitis&quot;,
    &quot;chronic interstitial cystitis&quot;,
    &quot;urinary tract infection&quot;,
    &quot;interstitial nephritis&quot;,
    &quot;pyelonephritis&quot;,
    &quot;urethritis&quot;,
    uti_terms
    )

gwas_study_info = gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
           stringr::str_replace_all(
             l2_all_disease_terms,
             vec_to_grep_pattern(uti_terms),
             &quot;urinary tract infections and interstitial nephritis&quot;
           )
  )</code></pre>
</div>
<div id="urolithiasis" class="section level3" number="16.2.2">
<h3><span class="header-section-number">16.2.2</span> Urolithiasis</h3>
<pre class="r"><code># N20-N23.0

urolithiasis_icd10 &lt;-
  paste0(&quot;N&quot;, seq(20, 23.0, by = 0.1) * 10)

urolithiasis_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(urolithiasis_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()

urolithiasis_terms &lt;-
  c(&quot;lower urinary tract calculus&quot;,
    &quot;bladder calculus&quot;,
    &quot;calcium oxalate nephrolithiasis&quot;,
    &quot;nephrolithiasis&quot;,
    &quot;renal colic&quot;, #icd-9 788.0
    urolithiasis_terms
    )</code></pre>
</div>
<div id="benign-prostatic-hyperplasia" class="section level3"
number="16.2.3">
<h3><span class="header-section-number">16.2.3</span> Benign prostatic
hyperplasia</h3>
</div>
<div id="male-infertility" class="section level3" number="16.2.4">
<h3><span class="header-section-number">16.2.4</span> Male
infertility</h3>
<pre class="r"><code>gwas_study_info |&gt;
  filter(grepl(vec_to_grep_pattern(&quot;male infertility&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct()</code></pre>
<pre><code>               all_disease_terms l2_all_disease_terms
                          &lt;char&gt;               &lt;char&gt;
1:              male infertility     male infertility
2:                   azoospermia     male infertility
3: male infertility, azoospermia     male infertility</code></pre>
</div>
<div id="other-urinary-diseases" class="section level3" number="16.2.5">
<h3><span class="header-section-number">16.2.5</span> Other urinary
diseases</h3>
<pre class="r"><code># N25-N28.1, N29-N29.8, N31-N32.0, N32.3-N32.4, N36-N36.9, N39, N41-N41.9, N44-N44.0, N45-N45.9, N49-N49.9

urine_icd10 &lt;- 
  c(paste0(&quot;N&quot;, seq(25, 28.1, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(29, 29.8, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(31, 32.0, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(32.3, 32.4, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(36, 36.9, by = 0.1) * 10),
    &quot;N39&quot;,
    paste0(&quot;N&quot;, seq(41, 41.9, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(44, 44.0, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(45, 45.9, by = 0.1) * 10),
    paste0(&quot;N&quot;, seq(49, 49.9, by = 0.1) * 10)
    )

urine_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(urine_icd10, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique() |&gt;
  tolower()

urine_terms &lt;- stringr::str_remove_all(urine_terms, &quot;, unspecified$&quot;)
urine_terms &lt;- stringr::str_remove_all(urine_terms, &quot;, site not specified$&quot;)


urinary_diseases_terms &lt;- c(&quot;urinary incontinence&quot;,
                            &quot;stress urinary incontinence&quot;,
                            &quot;urgency urinary incontinence&quot;,
                            &quot;urinary system disease&quot;,
                            &quot;bladder neck obstruction&quot;,
                            &quot;urethral disease&quot;,
                            &quot;urethral syndrome&quot;,
                            &quot;uterine inflammatory disease&quot;,
                            &quot;enuresis&quot;,
                            &quot;neurogenic bladder&quot;, # icd9 596.54
                            &quot;bladder diverticulum&quot;, # icd9 596.3
                            urine_terms
                            )

urinary_diseases_terms &lt;- str_length_sort(urinary_diseases_terms)

gwas_study_info = gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
           stringr::str_replace_all(
             l2_all_disease_terms,
             vec_to_grep_pattern(urinary_diseases_terms),
             &quot;other urinary diseases&quot;
           )
  )</code></pre>
</div>
</div>
<div id="gynocological-diseases" class="section level2" number="16.3">
<h2><span class="header-section-number">16.3</span> Gynocological
diseases</h2>
<div id="uterine-fibroids" class="section level3" number="16.3.1">
<h3><span class="header-section-number">16.3.1</span> Uterine
fibroids</h3>
<pre class="r"><code># D25-D26, D28.2
icd10_uterine_fibroids &lt;-
  c(paste0(&quot;D&quot;, seq(25, 26, by = 0.1) * 10),
    &quot;D282&quot;
    )

icd10_uterine_fibroids_terms &lt;-
  icd10 |&gt;
  filter(grepl(paste0(icd10_uterine_fibroids, collapse = &quot;|&quot;), code)) |&gt;
  pull(`short_description_(valid_icd-10_fy2025)`) |&gt;
  unique()

uterine_fibroid_terms &lt;- 
  c(&quot;uterine leiomyoma&quot;,
    &quot;uterine fibroid&quot;)



gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(uterine_fibroid_terms),
                                   &quot;uterine fibroids&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;uterine fibroids&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>              all_disease_terms          l2_all_disease_terms
                         &lt;char&gt;                        &lt;char&gt;
1:              uterine fibroid              uterine fibroids
2: uterine fibroid, menorrhagia menorrhagia, uterine fibroids</code></pre>
</div>
<div id="polycystic-ovary-syndrome" class="section level3"
number="16.3.2">
<h3><span class="header-section-number">16.3.2</span> Polycystic ovary
syndrome</h3>
<pre class="r"><code>gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;polycystic ovary syndrome&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                                     all_disease_terms
                                                &lt;char&gt;
1:                           polycystic ovary syndrome
2: polycystic ovary syndrome, type 2 diabetes mellitus
                                  l2_all_disease_terms
                                                &lt;char&gt;
1:                           polycystic ovary syndrome
2: polycystic ovary syndrome, diabetes mellitus type 2</code></pre>
</div>
<div id="female-infertility" class="section level3" number="16.3.3">
<h3><span class="header-section-number">16.3.3</span> Female
infertility</h3>
<pre class="r"><code>gwas_study_info |&gt;
  filter(grepl(
         vec_to_grep_pattern(&quot;female infertility&quot;),
         l2_all_disease_terms,
         perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() </code></pre>
<pre><code>                                                                all_disease_terms
                                                                           &lt;char&gt;
1:                                                             female infertility
2:                                              female infertility, endometriosis
3: female infertility, tubal factor infertility, chlamydophila infectious disease
4:                                                       tubal factor infertility
                l2_all_disease_terms
                              &lt;char&gt;
1:                female infertility
2: endometriosis, female infertility
3:                female infertility
4:                female infertility</code></pre>
</div>
<div id="endometriosis" class="section level3" number="16.3.4">
<h3><span class="header-section-number">16.3.4</span> Endometriosis</h3>
<pre class="r"><code>endo_terms &lt;- c(&quot;ovarian endometriosis&quot;,
                &quot;endometriosis of pelvic peritoneum&quot;
                )

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(endo_terms),
                                   &quot;endometriosis&quot;
                          )  
        )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;endometriosis&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                          all_disease_terms                l2_all_disease_terms
                                     &lt;char&gt;                              &lt;char&gt;
1:                            endometriosis                       endometriosis
2:        female infertility, endometriosis   endometriosis, female infertility
3:         migraine disorder, endometriosis   endometriosis, headache disorders
4: endometriosis, major depressive disorder endometriosis, depressive disorders
5:                    ovarian endometriosis                       endometriosis
6:       endometriosis of pelvic peritoneum                       endometriosis</code></pre>
</div>
<div id="genital-prolapse" class="section level3" number="16.3.5">
<h3><span class="header-section-number">16.3.5</span> Genital
prolapse</h3>
<pre class="r"><code>fem_genital_prolapse_terms &lt;- c(&quot;uterine prolapse&quot;,
                              &quot;cystocele&quot;,
                              &quot;rectocele&quot;
)

gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(fem_genital_prolapse_terms),
                                   &quot;genital prolapse&quot;
                          )
 )
        
        
gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;genital prolapse&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>   all_disease_terms l2_all_disease_terms
              &lt;char&gt;               &lt;char&gt;
1:  uterine prolapse     genital prolapse
2:         rectocele     genital prolapse</code></pre>
</div>
<div id="premenstrual-syndrome" class="section level3" number="16.3.6">
<h3><span class="header-section-number">16.3.6</span> Premenstrual
syndrome</h3>
</div>
<div id="other-gynocological-diseases" class="section level3"
number="16.3.7">
<h3><span class="header-section-number">16.3.7</span> Other
gynocological diseases</h3>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l2_all_disease_terms)  |&gt; 
            unlist() |&gt;
            stringr::str_trim()

# N72-N72.0, N75-N77.8, N83-N83.9

other_gyno_terms &lt;- c(&quot;cervicitis&quot;, #N72
                      &quot;bartholin gland disease&quot;, # N75
                      &quot;bartholin duct cyst&quot;,
                      &quot;vaginitis&quot;, # N76
                      &quot;vaginal inflammation&quot;,
                      &quot;postmenopausal atrophic vaginitis&quot;,                      
                      &quot;atrophic vaginitis&quot;,
                      &quot;vulvovaginitis&quot;,
                      &quot;ulceration of vulva&quot;, # N77
                      &quot;ovarian cyst&quot;, # N83
                      &quot;follicular cyst&quot;
                      )

# pregnancy_terms &lt;- grep(&quot;pregnancy&quot;, diseases, value = T)

# gyno_terms &lt;- c(
#                 &quot;female reproductive system disease&quot;,
#                 &quot;female genital tract fistula&quot;,
#                 &quot;placenta disease&quot;, 
#                 &quot;ovarian gynecological diseases&quot;,
#                 &quot;vaginal disorder&quot;,
# 
# 
#                 &quot;abnormal delivery&quot;,
#                 pregnancy_terms)


gwas_study_info = gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(other_gyno_terms),
                                   &quot;other gynecological diseases&quot;
                          )  
        )</code></pre>
</div>
</div>
<div id="hemoglobinopathies-and-hemolytic-anemias"
class="section level2" number="16.4">
<h2><span class="header-section-number">16.4</span> Hemoglobinopathies
and hemolytic anemias</h2>
<pre class="r"><code>hemoglobinopathies_terms &lt;- c(&quot;sickle cell disease and related diseases&quot;,
                              &quot;thalassemia&quot;,
                              &quot;inherited hemoglobinopathy&quot;,
                              &quot;hemoglobin e disease&quot;
)

hemopath_hemo_anemias &lt;- c(hemoglobinopathies_terms, &quot;hemolytic anemia&quot;)

gwas_study_info = gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
           stringr::str_replace_all(
             l2_all_disease_terms,
             vec_to_grep_pattern(hemopath_hemo_anemias),
             &quot;hemoglobinopathies and hemolytic anemias&quot;
           )
  )

gwas_study_info |&gt;
 filter(grepl(vec_to_grep_pattern(&quot;hemoglobinopathies and hemolytic anemias&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                          all_disease_terms
                                     &lt;char&gt;
1: sickle cell disease and related diseases
2:                       sickle cell anemia
3:       sickle cell anemia, cholelithiasis
4:                     hemoglobin e disease
5:      sickle cell anemia, thromboembolism
6:                familial hemolytic anemia
                                                                      l2_all_disease_terms
                                                                                    &lt;char&gt;
1:                                                hemoglobinopathies and hemolytic anemias
2:                                                hemoglobinopathies and hemolytic anemias
3:              gallbladder and biliary diseases, hemoglobinopathies and hemolytic anemias
4:                                                hemoglobinopathies and hemolytic anemias
5: hemoglobinopathies and hemolytic anemias, other cardiovascular and circulatory diseases
6:                                                hemoglobinopathies and hemolytic anemias</code></pre>
<div id="endocrine-metabolic-blood-and-immune-disorders"
class="section level3" number="16.4.1">
<h3><span class="header-section-number">16.4.1</span> Endocrine,
metabolic, blood, and immune disorders</h3>
<pre class="r"><code>endo_terms &lt;- c(&quot;ovarian dysfunction&quot;,
                &quot;ovarian disease&quot;,
                &quot;menstrual disorder&quot;,
            #    &quot;premenstral tension&quot; #icd9 625.4, icd10 N94.3,
                &quot;adrenocortical insufficiency&quot;, # ICD9:255.4, ICD10:E27.1 / ICD10:E27.4
                &quot;cushing syndrome&quot;, # ICD9:255.0, ICD10:E24
                &quot;hyperaldosteronism&quot;, # ICD10CM:E26, ICD9:255.1
                &quot;delayed puberty&quot;, #ICD10:E30.1, ICD9:259.1
                &quot;central precocious puberty&quot;,  #?ICD10: E30.1
            &quot;endocrine system disease&quot; # ????
                )

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(endo_terms),
                                   &quot;other endocrine disorders&quot;
                          )  
        )

thyroid_terms &lt;- c(&quot;congenital hypothyroidism due to developmental anomaly&quot;, #ICD-10:E03.1
                   &quot;hypothyroidism&quot;,
                   &quot;nontoxic goiter&quot;,#ICD-10: E04
                   &quot;thyrotoxicosis&quot;, #ICD-10: E05
                   &quot;toxic nodular goiter&quot;, # ICD9:242.3)
                   &quot;thyroiditis&quot;, #ICD-10: E06, IDC9:245
                   &quot;hashimotos thyroiditis&quot;, #ICD9:245.2
                   &quot;graves disease&quot;, #ICD9:242.0,
                   &quot;autoimmune thyroid disease&quot; #ICD10:E06.3,
                   )

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(thyroid_terms),
                                   &quot;thyroid disorders&quot;
                          )  
        )


other_blood_disorders &lt;- c(&quot;neutropenia&quot;,
                           &quot;cryoglobulinemia&quot;)

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(other_blood_disorders),
                                   &quot;other blood disorders&quot;
                          )  
        )

metabolism_disorders &lt;- c(&quot;hypoglycemia&quot;,
                          &quot;hypoparathyroidism&quot;,
                          &quot;parathyroid disease&quot;, # ICD9: 252
                          &quot;primary hyperparathyroidism&quot;, # ICD10 E21.0
                          &quot;hyperparathyroidism&quot;, #ID20 E20
                          # &quot;secondary hyperparathyroidism of renal origin&quot; not incl, as otherwise specified in ICD10CM:N25.81
                          &quot;obesity&quot;,
                          &quot;metabolic syndrome&quot;,
                          &quot;metabolic syndrome x&quot;,
                          &quot;inborn errors of metabolism&quot;,
                          &quot;metabolic disease&quot;,
                          &quot;mineral metabolism disease&quot;, # ICD9:275.8, ICD9:275.9, ICD10:E83
                          &quot;acidosis&quot;, # ICD9:276.2
                          &quot;disorder of acid-base balance&quot;, # ? ICD10-cm E87.8
                          &quot;bilirubin metabolism disease&quot;
                          )

gwas_study_info = gwas_study_info |&gt;
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms,
                                  pattern = vec_to_grep_pattern(metabolism_disorders),
                                   &quot;other metabolic disorders&quot;
                          )  
        )</code></pre>
</div>
</div>
<div id="oral-disorders" class="section level2" number="16.5">
<h2><span class="header-section-number">16.5</span> Oral disorders</h2>
<pre class="r"><code>oral_disorders_terms &lt;- c(&quot;dental caries&quot;,
                           &quot;tooth disease&quot;,
                          &quot;toothache&quot;,
                           &quot;periodontal disease&quot;
)


gwas_study_info = gwas_study_info |&gt;
  mutate(l2_all_disease_terms = 
           stringr::str_replace_all(
             l2_all_disease_terms,
             vec_to_grep_pattern(oral_disorders_terms),
             &quot;oral disorders&quot;
           )
  )


gwas_study_info |&gt; 
 filter(grepl(vec_to_grep_pattern(&quot;oral disorders&quot;), 
              l2_all_disease_terms,
              perl = T)) |&gt;
  select(all_disease_terms, l2_all_disease_terms) |&gt;
  distinct() |&gt;
  head()</code></pre>
<pre><code>                       all_disease_terms l2_all_disease_terms
                                  &lt;char&gt;               &lt;char&gt;
1:                         dental caries       oral disorders
2:              aggressive periodontitis       oral disorders
3: pit and fissure surface dental caries       oral disorders
4:          smooth surface dental caries       oral disorders
5:                         periodontitis       oral disorders
6:                             toothache       oral disorders</code></pre>
</div>
</div>
<div id="check-compatibility-with-gbd-data" class="section level1"
number="17">
<h1><span class="header-section-number">17</span> Check compatibility
with gbd data</h1>
<div id="weird-fix" class="section level2" number="17.1">
<h2><span class="header-section-number">17.1</span> Weird fix</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;disorderss&quot;,
                                   &quot;disorders&quot;
                          )  
        )


gwas_study_info = 
gwas_study_info |&gt; 
 mutate(l2_all_disease_terms  = 
          stringr::str_replace_all(l2_all_disease_terms ,
                                  pattern = &quot;anxiety disorders disorderss&quot;,
                                   &quot;anxiety disorders&quot;
                          )  
        )</code></pre>
<pre class="r"><code>gbd_data &lt;- data.table::fread(here::here(&quot;data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv&quot;))

gbd_data$cause &lt;- stringr::str_remove_all(gbd_data$cause, &quot;,&quot;)

diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l2_all_disease_terms[gwas_study_info$l2_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()

gbd_data$cause[!tolower(gbd_data$cause) %in% unique(diseases)] |&gt; sort() |&gt; unique()</code></pre>
<pre><code> [1] &quot;Acute glomerulonephritis&quot;                                         
 [2] &quot;Aortic aneurysm&quot;                                                  
 [3] &quot;Bipolar disorder&quot;                                                 
 [4] &quot;Congenital birth defects&quot;                                         
 [5] &quot;Drug use disorders&quot;                                               
 [6] &quot;Endocrine metabolic blood and immune disorders&quot;                   
 [7] &quot;Gynecological diseases&quot;                                           
 [8] &quot;Other chronic respiratory diseases&quot;                               
 [9] &quot;Other musculoskeletal disorders&quot;                                  
[10] &quot;Other sense organ diseases&quot;                                       
[11] &quot;Scabies&quot;                                                          
[12] &quot;Sudden infant death syndrome&quot;                                     
[13] &quot;Total burden related to Non-alcoholic fatty liver disease (NAFLD)&quot;
[14] &quot;Urinary diseases and male infertility&quot;                            
[15] &quot;Vascular intestinal disorders&quot;                                    
[16] &quot;Viral skin diseases&quot;                                              </code></pre>
<pre class="r"><code>gbd_data =
  gbd_data |&gt;
  mutate(cause = tolower(cause))

gwas_disease_traits = data.frame(cause = diseases)
  # gwas_study_info |&gt;
  # filter(DISEASE_STUDY == T) |&gt;
  # select(all_disease_terms, l1_all_disease_terms, cause = l2_all_disease_terms) |&gt;
  # distinct()

left_join(gwas_disease_traits, 
          gbd_data) |&gt;
  head()</code></pre>
<pre><code>Joining with `by = join_by(cause)`</code></pre>
<pre><code>Warning in left_join(gwas_disease_traits, gbd_data): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 3 of `x` matches multiple rows in `y`.
ℹ Row 19 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.</code></pre>
<pre><code>                              cause                                measure
1 idiopathic dilated cardiomyopathy                                   &lt;NA&gt;
2                  orofacial clefts                                   &lt;NA&gt;
3 tracheal bronchus and lung cancer DALYs (Disability-Adjusted Life Years)
4 tracheal bronchus and lung cancer                             Prevalence
5 tracheal bronchus and lung cancer                              Incidence
6 tracheal bronchus and lung cancer DALYs (Disability-Adjusted Life Years)
  location  sex      age metric year       val     upper     lower
1     &lt;NA&gt; &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;   NA        NA        NA        NA
2     &lt;NA&gt; &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;   NA        NA        NA        NA
3   Global Both All ages   Rate 2019 580.36100 627.79984 532.74652
4   Global Both All ages   Rate 2019  40.27440  43.51721  37.12978
5   Global Both All ages   Rate 2019  28.16826  30.49575  25.77712
6   Global Both All ages   Rate 2019 580.36100 627.79984 532.74652</code></pre>
<pre class="r"><code>gwas_study_info |&gt; select(cause = l2_all_disease_terms) |&gt;
  distinct() |&gt;
  left_join(gbd_data) |&gt;
  head()</code></pre>
<pre><code>Joining with `by = join_by(cause)`</code></pre>
<pre><code>                               cause                                measure
                              &lt;char&gt;                                 &lt;char&gt;
1:                                                                     &lt;NA&gt;
2: idiopathic dilated cardiomyopathy                                   &lt;NA&gt;
3:                  orofacial clefts                                   &lt;NA&gt;
4: tracheal bronchus and lung cancer DALYs (Disability-Adjusted Life Years)
5: tracheal bronchus and lung cancer                             Prevalence
6: tracheal bronchus and lung cancer                              Incidence
   location    sex      age metric  year       val     upper     lower
     &lt;char&gt; &lt;char&gt;   &lt;char&gt; &lt;char&gt; &lt;int&gt;     &lt;num&gt;     &lt;num&gt;     &lt;num&gt;
1:     &lt;NA&gt;   &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;    NA        NA        NA        NA
2:     &lt;NA&gt;   &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;    NA        NA        NA        NA
3:     &lt;NA&gt;   &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt;    NA        NA        NA        NA
4:   Global   Both All ages   Rate  2019 580.36100 627.79984 532.74652
5:   Global   Both All ages   Rate  2019  40.27440  43.51721  37.12978
6:   Global   Both All ages   Rate  2019  28.16826  30.49575  25.77712</code></pre>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$l2_all_disease_terms[gwas_study_info$l2_all_disease_terms != &quot;&quot;])  |&gt; 
            unlist() |&gt;
            stringr::str_trim()

length(unique(diseases))</code></pre>
<pre><code>[1] 1338</code></pre>
<pre class="r"><code># make frequency table
freq &lt;- table(as.factor(diseases))

# sort in decreasing order
freq_sorted &lt;- sort(freq, decreasing = TRUE)

# show top N, e.g. top 10
head(freq_sorted, 10)</code></pre>
<pre><code>
                       chronic kidney disease 
                                        10889 
other cardiovascular and circulatory diseases 
                                         7585 
                     diabetes mellitus type 2 
                                          922 
                       other mental disorders 
                                          828 
                         depressive disorders 
                                          726 
                       ischemic heart disease 
                                          579 
      alzheimer&#39;s disease and other dementias 
                                          514 
                              other neoplasms 
                                          509 
                    blindness and vision loss 
                                          415 
                    other metabolic disorders 
                                          381 </code></pre>
<div id="save-the-updated-gwas_study_info-with-harmonized-disease-terms"
class="section level3" number="17.1.1">
<h3><span class="header-section-number">17.1.1</span> Save the updated
gwas_study_info with harmonized disease terms</h3>
<pre class="r"><code>gwas_study_info &lt;- fwrite(gwas_study_info,
                          here::here(&quot;output/gwas_cat/gwas_study_info_trait_group_l2.csv&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] jsonlite_2.0.0    httr_1.4.7        stringr_1.5.1     ggplot2_3.5.2    
[5] data.table_1.17.8 dplyr_1.1.4       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       compiler_4.3.1     renv_1.0.3         promises_1.3.3    
 [5] tidyselect_1.2.1   Rcpp_1.1.0         git2r_0.36.2       callr_3.7.6       
 [9] later_1.4.2        jquerylib_0.1.4    scales_1.4.0       readxl_1.4.5      
[13] yaml_2.3.10        fastmap_1.2.0      here_1.0.1         R6_2.6.1          
[17] generics_0.1.4     curl_6.4.0         knitr_1.50         tibble_3.3.0      
[21] rprojroot_2.1.0    RColorBrewer_1.1-3 bslib_0.9.0        pillar_1.11.0     
[25] rlang_1.1.6        cachem_1.1.0       stringi_1.8.7      httpuv_1.6.16     
[29] xfun_0.52          getPass_0.2-4      fs_1.6.6           sass_0.4.10       
[33] cli_3.6.5          withr_3.0.2        magrittr_2.0.3     ps_1.9.1          
[37] grid_4.3.1         digest_0.6.37      processx_3.8.6     rstudioapi_0.17.1 
[41] lifecycle_1.0.4    vctrs_0.6.5        evaluate_1.0.4     glue_1.8.0        
[45] cellranger_1.1.0   farver_2.1.2       whisker_0.4.1      rmarkdown_2.29    
[49] tools_4.3.1        pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
