<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-26" />

<title>Mapping GWAS traits to ICD 10</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mapping GWAS traits to ICD 10</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-12-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreef5976201a207a0d66bf93e4d3d03c78041ab2d8atargetblankf597620a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/f5976201a207a0d66bf93e4d3d03c78041ab2d8a" target="_blank">f597620</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreef5976201a207a0d66bf93e4d3d03c78041ab2d8atargetblankf597620a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/f5976201a207a0d66bf93e4d3d03c78041ab2d8a" target="_blank">f597620</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    .venv/
    Ignored:    analysis/.DS_Store
    Ignored:    ancestry_dispar_env/
    Ignored:    data/.DS_Store
    Ignored:    data/cohort/
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/IHME-GBD_2023_DATA-73cc01fd-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/hp_umls_mapping.csv
    Ignored:    data/icd/lancet_conditions_icd10.xlsx
    Ignored:    data/icd/manual_disease_icd10_mappings.xlsx
    Ignored:    data/icd/mondo_umls_mapping.csv
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/phecode_to_icd10_manual_mapping.xlsx
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/semiautomatic_ICD-pheno_UKB_subset.txt
    Ignored:    human_dictionary/
    Ignored:    igsr_populations.tsv
    Ignored:    output/.DS_Store
    Ignored:    output/abstracts/
    Ignored:    output/doccano/
    Ignored:    output/fulltexts/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_cohorts/
    Ignored:    output/icd_map/
    Ignored:    output/trait_ontology/
    Ignored:    pubmedbert-cohort-ner-model/
    Ignored:    pubmedbert-cohort-ner/
    Ignored:    r-spacyr/
    Ignored:    renv/
    Ignored:    venv/

Untracked files:
    Untracked:  analysis/get_abstracts.Rmd
    Untracked:  code/extract_cdc_meta.R
    Untracked:  code/figure_4a.R
    Untracked:  code/poster_figures.R
    Untracked:  code/umls_ontology.R
    Untracked:  data/cdc/
    Untracked:  data/icd/2025AA/
    Untracked:  data/icd/umls-2025AA-mrconso.zip
    Untracked:  figures/
    Untracked:  visualization.Rdata

Unstaged changes:
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/get_full_text.Rmd
    Modified:   analysis/gwas_to_gbd.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_1_disease_group_non_cancer.Rmd
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   analysis/missing_cohort_info.Rmd
    Modified:   analysis/replication_ancestry_bias.Rmd
    Modified:   analysis/text_for_cohort_labels.Rmd
    Modified:   code/custom_plotting.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/map_trait_to_icd10.Rmd</code>)
and HTML (<code>docs/map_trait_to_icd10.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/f5976201a207a0d66bf93e4d3d03c78041ab2d8a/analysis/map_trait_to_icd10.Rmd" target="_blank">f597620</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-12-29
</td>
<td>
Update mapping to ICD-10 codes (to keep year)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/1f555b6dd010b4f1d714a28e64114765790ce532/docs/map_trait_to_icd10.html" target="_blank">1f555b6</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-12-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b4527b83be01781943ce91e1225a9d35fc4ce0ac/analysis/map_trait_to_icd10.Rmd" target="_blank">b4527b8</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-12-29
</td>
<td>
Update mapping to ICD-10 codes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/757b4b4a6e2ce1f399322aaf0fa2e42ec3dd892d/docs/map_trait_to_icd10.html" target="_blank">757b4b4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/6019c960c8c3e579050afa394a5c304c78e223b2/analysis/map_trait_to_icd10.Rmd" target="_blank">6019c96</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-09
</td>
<td>
Even more correcting of icd 10 codes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/0feea16517cba21cda901985969eef16a2332679/docs/map_trait_to_icd10.html" target="_blank">0feea16</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/a8f16288fbccd427444d6e8c2ec740c9e844b92c/analysis/map_trait_to_icd10.Rmd" target="_blank">a8f1628</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-08
</td>
<td>
Include study accession in icd 10 map
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/50ebebc4e4ad2798495b995ea929015c9745d66a/docs/map_trait_to_icd10.html" target="_blank">50ebebc</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/9bbe0ddf1641e9ebf392e866c41b11946258a4fb/analysis/map_trait_to_icd10.Rmd" target="_blank">9bbe0dd</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-08
</td>
<td>
Updating icd 10 mapping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/ec027a3c722388c1c87eba69103f7e736b49e378/docs/map_trait_to_icd10.html" target="_blank">ec027a3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/cb8a570c803f4d528369b7b99fca0dcae3fa735c/analysis/map_trait_to_icd10.Rmd" target="_blank">cb8a570</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-08
</td>
<td>
Updating disease icd code mapping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/41d6fe5b65f1628ed48b63351422535aad788c41/docs/map_trait_to_icd10.html" target="_blank">41d6fe5</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/97d340dae80dcce5fec2e5fe2db9d93851769ae1/analysis/map_trait_to_icd10.Rmd" target="_blank">97d340d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-28
</td>
<td>
workflowr::wflow_publish("analysis/map_trait_to_icd10.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1">
<h1>Set up</h1>
<pre class="r"><code>library(dplyr)
library(stringr)
library(data.table)</code></pre>
<div id="ontology-help---for-getting-disease-subtypes"
class="section level2">
<h2>Ontology help - for getting disease subtypes</h2>
<pre class="r"><code>source(here::here(&quot;code/get_term_descendants.R&quot;))</code></pre>
<div id="load-data" class="section level3">
<h3>Load Data</h3>
<pre class="r"><code># gwas_study_info &lt;- fread(here::here(&quot;output/gwas_cat/gwas_study_info_group_v2.csv&quot;))

gwas_study_info &lt;- data.table::fread(here::here(&quot;output/gwas_cat/gwas_study_all_group.csv&quot;))

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = stringi::stri_trans_general(collected_all_disease_terms, &quot;Latin-ASCII&quot;)
           )

# study accession = GCST003209
# replace benign neoplasm with colorectal cancer, endometrial cancer

gwas_study_info = 
  gwas_study_info |&gt;
  mutate(collected_all_disease_terms = ifelse(STUDY_ACCESSION == &quot;GCST003209&quot; |
                                              STUDY_ACCESSION == &quot;GCST003208&quot; ,
                                              &quot;colorectal cancer, endometrial cancer&quot;,
                                              collected_all_disease_terms))

# for study accession = GCST90133383
# replace benign neoplasm with testicular cancer, hearing loss</code></pre>
</div>
</div>
</div>
<div id="automatic-mapping-of-gwas-traits-to-icd-10"
class="section level1">
<h1>Automatic mapping of GWAS traits to ICD 10</h1>
<div id="create-a-mapping-table" class="section level2">
<h2>Create a mapping table</h2>
<pre class="r"><code>disease_mapping &lt;- gwas_study_info |&gt;
  filter(DISEASE_STUDY == T) |&gt;
  tidyr::separate_longer_delim(cols = collected_all_disease_terms, 
                               delim = &quot;, &quot;) |&gt;
  select(`DISEASE/TRAIT`, 
         collected_all_disease_terms, 
         PUBMED_ID,
         YEAR,
         STUDY_ACCESSION) |&gt;
  distinct()

disease_mapping =
  disease_mapping |&gt;
  filter(collected_all_disease_terms != &quot;&quot;)

print(&quot;Number of unique disease trait &amp; study pairs&quot;)</code></pre>
<pre><code>[1] &quot;Number of unique disease trait &amp; study pairs&quot;</code></pre>
<pre class="r"><code>nrow(disease_mapping)</code></pre>
<pre><code>[1] 46316</code></pre>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;,
                               gwas_study_info$collected_all_disease_terms[gwas_study_info$collected_all_disease_terms != &quot;&quot;])  |&gt;
  unlist() |&gt;
  stringr::str_trim()

diseases &lt;- unique(diseases)

print(&quot;Number of unique disease terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of unique disease terms&quot;</code></pre>
<pre class="r"><code>print(length(diseases))</code></pre>
<pre><code>[1] 1993</code></pre>
</div>
<div id="get-icd10-codes-from-author-provided-diseasetrait-column"
class="section level2">
<h2>Get ICD10 codes from author provided DISEASE/TRAIT column</h2>
<pre class="r"><code>disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(grepl(&quot;ICD10&quot;, `DISEASE/TRAIT`),
                             str_extract(`DISEASE/TRAIT`, 
                                         &quot;(?&lt;=ICD10 )[^:]+(?=:)|(?&lt;=ICD10 )[^ ]+&quot;),
                             NA),
         icd10_code_origin = ifelse(grepl(&quot;ICD10&quot;, `DISEASE/TRAIT`),
                                    &quot;Study Provided&quot;,
                                    NA)
)

# fix weird ICD10 codes: R11.103.11 -&gt; R11
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = stringr::str_replace(icd10_code, 
                                         pattern = &quot;R11.103.11&quot;,
                                         replacement = &quot;R11&quot;)
         )

# fix weird ICD10 codes: D63.165.8 -&gt; D63, N18.3-N18.9
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = stringr::str_replace(icd10_code, 
                                         pattern = &quot;D63.165.8&quot;,
                                         replacement = &quot;D63, N18.3-N18.9&quot;)
         ) 

# Additionally, some studies provide ICD10 codes in the DISEASE/TRAIT column without the &quot;ICD10&quot; prefix
# e.g. Source of report of K76 (other diseases of liver) (UKB data field 131671)
# let&#39;s try to capture these too
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(is.na(icd10_code) &amp;
                             grepl(&quot;Source of report of&quot;, `DISEASE/TRAIT`),
                             str_extract(`DISEASE/TRAIT`,
                                         &quot;(?&lt;=Source of report of\\s)[A-Z][0-9]{1,2}(?:\\.[0-9A-Z]+)?\\b&quot;),
                             icd10_code),
         icd10_code_origin = ifelse(grepl(&quot;Source of report of&quot;, 
                                          `DISEASE/TRAIT`),
                                    &quot;Study Provided&quot;,
                                    icd10_code_origin)
) |&gt;
  mutate(icd10_code = str_remove_all(icd10_code, 
                                     pattern = &quot;Source of report of &quot;)
         )

# Another example where  some studies provide ICD10 codes in the DISEASE/TRAIT column without the &quot;ICD10&quot; prefix
# e.g. Insulin-dependent diabetes mellitus (Union E10)
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(is.na(icd10_code) &amp;
                             grepl(&quot;Union&quot;, `DISEASE/TRAIT`),
                             str_extract(`DISEASE/TRAIT`, 
                                         &quot;Union [A-Z][0-9]{1,2}(?:\\.[0-9A-Z]+)?\\b&quot;),
                             icd10_code),
         icd10_code_origin = ifelse(grepl(&quot;Union&quot;, 
                                          `DISEASE/TRAIT`),
                                    &quot;Study Provided&quot;,
                                    icd10_code_origin)
) |&gt;
  mutate(icd10_code = str_remove_all(icd10_code, 
                                     pattern = &quot;Union &quot;)
  )

disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(is.na(icd10_code) &amp;
                             grepl(&quot;ICD [A-Z][0-9]|ICD-10 [A-Z][0-9]&quot;, 
                                   `DISEASE/TRAIT`),
                        str_match(`DISEASE/TRAIT`,
                                  &quot;ICD(?:-10|10)?\\s*([A-Z][0-9]{1,2}(?:\\.[0-9A-Z]+)?)&quot;)[,2],
                             icd10_code),
         icd10_code_origin = ifelse(grepl(&quot;ICD [A-Z][0-9]|ICD-10 [A-Z][0-9]&quot;,
                                          `DISEASE/TRAIT`),
                                    &quot;Study Provided&quot;,
                                    icd10_code_origin)
)

n_study_provided = 
disease_mapping |&gt;
  filter(icd10_code_origin == &quot;Study Provided&quot;) |&gt;
  nrow()

n_studies =
  disease_mapping |&gt;
  nrow()

print(&quot;Percentage of studies that provide ICD10 codes:&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of studies that provide ICD10 codes:&quot;</code></pre>
<pre class="r"><code>print(round(n_study_provided / n_studies * 100, 2))</code></pre>
<pre><code>[1] 8.55</code></pre>
</div>
<div id="get-icd10-codes-from-author-provided-phecodes"
class="section level2">
<h2>Get ICD10 codes from author provided PheCodes</h2>
<div id="get-phecodes-for-diseases" class="section level3">
<h3>Get Phecodes for diseases</h3>
<pre class="r"><code>disease_mapping &lt;- disease_mapping |&gt;
  mutate(
    phecode = str_extract(`DISEASE/TRAIT`, &quot;(?&lt;=PheCode )[^)]+&quot;)
  ) |&gt;
  mutate(phecode = as.numeric(phecode))</code></pre>
</div>
<div id="convert-phecodes-to-icd10" class="section level3">
<h3>Convert Phecodes to ICD10</h3>
<pre class="r"><code># phecode to ICD10 mapping from https://wei-lab.app.vumc.org/phecode-data/phecode_international_version

phecodes &lt;- fread(here::here(&quot;data/icd/phecode_international_version_unrolled.csv&quot;))

phecode_icd_map =
  phecodes |&gt;
  select(icd10_code = ICD10, 
         phecode = PheCode
         ) |&gt;
  filter(!is.na(phecode))

phecode_icd_map =
  phecode_icd_map |&gt;
  filter(phecode %in% unique(disease_mapping$phecode))

# if more than one ICD10 code per phecode, collapse into a single row
phecode_icd_map =
  phecode_icd_map |&gt;
group_by(phecode) |&gt;
  summarise(icd10_code = 
            str_flatten(unique(icd10_code), 
                        collapse = &quot;, &quot;, 
                        na.rm = T), 
            .groups = &quot;drop&quot;)

# label the origin of these ICD10 codes as &quot;Study PheCode Mapping&quot;
phecode_icd_map =
  phecode_icd_map |&gt;
  mutate(icd10_code_origin = &quot;Study PheCode Mapping&quot;)

disease_mapping =
  rows_patch(disease_mapping,
              phecode_icd_map,
              by = c(&quot;phecode&quot;),
              unmatched = &quot;ignore&quot;,
              )
# left_join(disease_mapping,
#           phecode_icd_map,
#           by = c(&quot;phecode&quot;,&quot;icd10_code_origin&quot;, &quot;icd10_code&quot;),
#           relationship = &quot;many-to-one&quot;,
#           na_matches = &quot;never&quot;)

# disease_mapping =
#   disease_mapping |&gt;
#   mutate(icd10_code_origin = &quot;Study PheCode&quot;)

print(&quot;Number of ICD10 code obtained&quot;)</code></pre>
<pre><code>[1] &quot;Number of ICD10 code obtained&quot;</code></pre>
<pre class="r"><code>disease_mapping |&gt;
    filter(!is.na(icd10_code)) |&gt; 
  group_by(icd10_code_origin) |&gt; 
  summarise(n = n(), 
            percent = n()/n_studies *100)</code></pre>
<pre><code># A tibble: 2 × 3
  icd10_code_origin         n percent
  &lt;chr&gt;                 &lt;int&gt;   &lt;dbl&gt;
1 Study PheCode Mapping  6780   14.6 
2 Study Provided         3961    8.55</code></pre>
<pre class="r"><code>disease_mapping |&gt;
  filter(!is.na(icd10_code_origin)) |&gt;
  nrow()</code></pre>
<pre><code>[1] 10741</code></pre>
<pre class="r"><code>disease_mapping |&gt;
  filter(!is.na(icd10_code_origin)) |&gt;
  head()</code></pre>
<pre><code>                                                                          DISEASE/TRAIT
1              Source of report of M45 (ankylosing spondylitis) (UKB data field 131913)
2         Source of report of I21 (acute myocardial infarction) (UKB data field 131299)
3                         Source of report of B07 (viral warts) (UKB data field 130189)
4                         Source of report of D86 (sarcoidosis) (UKB data field 130687)
5                Source of report of E03 (other hypothyroidism) (UKB data field 130697)
6 Source of report of E10 (insulin-dependent diabetes mellitus) (UKB data field 130707)
  collected_all_disease_terms PUBMED_ID YEAR STUDY_ACCESSION icd10_code
1      ankylosing spondylitis  36779085 2022    GCST90103423        M45
2 acute myocardial infarction  36779085 2022    GCST90103349        I21
3             benign neoplasm  36779085 2022    GCST90103384        B07
4                 sarcoidosis  36779085 2022    GCST90103387        D86
5              hypothyroidism  36779085 2022    GCST90103388        E03
6    type 1 diabetes mellitus  36779085 2022    GCST90103390        E10
  icd10_code_origin phecode
1    Study Provided      NA
2    Study Provided      NA
3    Study Provided      NA
4    Study Provided      NA
5    Study Provided      NA
6    Study Provided      NA</code></pre>
<pre class="r"><code># per pubmed id, are there multiple icd10 code origins?
disease_mapping |&gt;
  filter(!is.na(icd10_code_origin)) |&gt;
  group_by(PUBMED_ID) |&gt;
  summarise(n_icd10_code_origins = n_distinct(icd10_code_origin)) |&gt;
  pull(n_icd10_code_origins) |&gt;
  summary()</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.000   1.000   1.000   1.059   1.000   2.000 </code></pre>
<pre class="r"><code># which pubmed ids have multiple icd10 code origins?
disease_mapping |&gt;
  filter(!is.na(icd10_code_origin)) |&gt;
  group_by(PUBMED_ID) |&gt;
  summarise(n_icd10_code_origins = n_distinct(icd10_code_origin)) |&gt;
  filter(n_icd10_code_origins &gt; 1)</code></pre>
<pre><code># A tibble: 1 × 2
  PUBMED_ID n_icd10_code_origins
      &lt;int&gt;                &lt;int&gt;
1  34737426                    2</code></pre>
<pre class="r"><code># per study accession, are there multiple icd10 code origins?
disease_mapping |&gt;
  filter(!is.na(icd10_code_origin)) |&gt;
  group_by(STUDY_ACCESSION) |&gt;
  summarise(n_icd10_code_origins = n_distinct(icd10_code_origin)) |&gt;
  pull(n_icd10_code_origins) |&gt;
  summary()</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      1       1       1       1       1       1 </code></pre>
</div>
<div id="complete-phecode-to-icd10-mappings" class="section level3">
<h3>Complete PheCode to ICD10 mappings</h3>
<p>Some Phecodes do not have ICD10 mappings in the above Phecode to
ICD10 mapping file. Here we’ve created a file with manual mappings for
these missing Phecodes.</p>
<pre class="r"><code># how many missing ICD10 codes for PheCodes?
disease_mapping |&gt;
    filter(is.na(icd10_code) &amp; !is.na(phecode)) |&gt; 
    nrow()</code></pre>
<pre><code>[1] 913</code></pre>
<pre class="r"><code># load manual mapping file
phecode_icd_manual = 
  readxl::read_xlsx(here::here(&quot;data/icd/phecode_to_icd10_manual_mapping.xlsx&quot;))

to_add = phecode_icd_manual |&gt;
  mutate(phecode = as.numeric(phecode),
         icd10_code_origin = &quot;Study PheCode (Manual Mapping)&quot;) |&gt;
  select(-phenotype) |&gt;
  distinct()

disease_mapping =
  rows_patch(disease_mapping,
             to_add,
             unmatched = &quot;ignore&quot;)</code></pre>
<pre><code>Matching, by = &quot;phecode&quot;</code></pre>
<pre class="r"><code># 
# nrow(phecode_icd_manual)
# 
# phecode = c(&quot;38.30&quot;,
#             &quot;79.90&quot;,
#             &quot;110.00&quot;,
#             &quot;530.13&quot;,
#             &quot;562.20&quot;,
#             &quot;580.10&quot;,
#             &quot;174.00&quot;,
#             &quot;174.20&quot;,
#             &quot;189.10&quot;,
#             &quot;218.00&quot;,
#             &quot;228.10&quot;,
#             &quot;224.00&quot;,
#             &quot;250.15&quot;,
#             &quot;250.25&quot;,
#             &quot;250.40&quot;,
#             &quot;285.21&quot;,
#             &quot;292.11&quot;,
#             &quot;362.26&quot;,
#             &quot;362.23&quot;,
#             &quot;362.27&quot;,
#             &quot;362.50&quot;,
#             &quot;452.20&quot;,
#             &quot;528.10&quot;,
#             &quot;535.90&quot;,
#             &quot;724.10&quot;,
#             &quot;724.22&quot;,
#             &quot;740.00&quot;,
#             &quot;743.10&quot;,
#             &quot;743.12&quot;,
#             &quot;743.13&quot;,
#             &quot;172.21&quot;,
#             &quot;274.00&quot;,
#             &quot;286.00&quot;,
#             &quot;282.00&quot;,
#             &quot;280.00&quot;,
#             &quot;272.00&quot;,
#             &quot;276.00&quot;,
#             &quot;338.00&quot;,
#             &quot;350.00&quot;,
#             &quot;401.00&quot;,
#             &quot;411.00&quot;,
#             &quot;414.20&quot;,
#             &quot;415.1&quot;,
#             &quot;427.21&quot;,
#             &quot;429.00&quot;,
#             &quot;585.00&quot;,
#             &quot;735.22&quot;,
#             &quot;724.00&quot;,
#             &quot;722.00&quot;,
#             &quot;716.10&quot;,
#             &quot;709.00&quot;,
#             &quot;706.00&quot;,
#             &quot;592.00&quot;,
#             &quot;571.00&quot;,
#             &quot;580.00&quot;,
#             &quot;170.00&quot;,
#             &quot;172.00&quot;,
#             &quot;264.00&quot;,
#             &quot;291.00&quot;,
#             &quot;555.00&quot;,
#             &quot;562.00&quot;,
#             &quot;578.00&quot;,
#             &quot;783.1&quot;,
#             &quot;536.80&quot;,
#             &quot;427.40&quot;,
#             &quot;443.00&quot;,
#             &quot;41.80&quot;,
#             &quot;41.90&quot;,
#             &quot;244.00&quot;,
#             &quot;250.00&quot;,
#             &quot;427.40&quot;,
#             &quot;526.40&quot;,
#             &quot;977.00&quot;,
#             &quot;840.20&quot;,
#             &quot;823.00&quot;,
#             &quot;751.00&quot;,
#             &quot;743.00&quot;,
#             &quot;270.38&quot;,
#             &quot;504.10&quot;,
#             &quot;253.40&quot;,
#             &quot;279.20&quot;,
#             &quot;426.22&quot;,
#             &quot;537.1&quot;,
#             &quot;707.20&quot;,
#             &quot;736.10&quot;,
#             &quot;789.10&quot;,
#             &quot;290.13&quot;,
#             &quot;327.70&quot;,
#             &quot;433.60&quot;,
#             &quot;695.00&quot;,
#             &quot;602.30&quot;,
#             &quot;375.10&quot;,
#             &quot;560.00&quot;,
#             &quot;586.10&quot;,
#             &quot;593.20&quot;,
#             &quot;620.10&quot;,
#             &quot;475.90&quot;,
#             &quot;799.00&quot;,
#             &quot;662.10&quot;,
#             &quot;855.00&quot;,
#             &quot;536.70&quot;,
#             &quot;856.00&quot;,
#             &quot;519.10&quot;,
#             &quot;771.20&quot;,
#             &quot;792.10&quot;,
#             &quot;525.10&quot;,
#             &quot;724.20&quot;,
#             &quot;611.11&quot;,
#             &quot;556.11&quot;,
#             &quot;767.00&quot;,
#             &quot;736.50&quot;,
#             &quot;526.42&quot;)
# 
# icd10_code = c(&quot;A49.9&quot;,
#                &quot;B34.9&quot;,
#                &quot;B35, B36&quot;,
#                &quot;K22.7&quot;,
#                &quot;K57&quot;,
#                &quot;N00, N01, N02, N03, N04, N05, N06, N07&quot;,
#                &quot;Z85.3&quot;,
#                &quot;C50&quot;,
#                &quot;C64, C65&quot;,
#                &quot;D25, D26&quot;,
#                &quot;D18.0&quot;,
#                &quot;E00, E00.0, E00.1, E00.2, E00.9, E01.8, E02, E03.0, E03.1, E03.2, E03.3, E03.8, E03.9, E89.0&quot;,
#                &quot;E10.5&quot;,
#                &quot;E11.5&quot;,
#                &quot;R73.0&quot;,
#                &quot;D63&quot;,
#                &quot;R47.0&quot;,
#                &quot;H35.3&quot;,
#                &quot;H35.3&quot;,
#                &quot;H35.3&quot;,
#                &quot;H35.5&quot;,
#                &quot;I80&quot;,
#                &quot;K12.30&quot;,
#                &quot;K29.7, K29.8, K29.9&quot;,
#                &quot;M43.2&quot;,
#                &quot;M21.5&quot;,
#                &quot;M13.9, M15.0, M15.1, M15.2, M15.3, M15.4, M16, M16.0, M16.1, M16.3, M16.6, M16.7, M16.9, M17.1, M17.4, M17.5, M18.0, M18.1, M18.5, M18.9, M19.0, M19.2&quot;,
#                &quot;M81&quot;,
#                &quot;M81.8&quot;,
#                &quot;M81.8&quot;,
#                &quot;C44&quot;,
#                &quot;M10, M10.0, M10.1, M10.2, M10.4, M10.9, M11.0, M11.1, M11.2, M11.8, M11.9, M67.9&quot;,
#                &quot;D65, D66, D67, D68, D68, D68.0, D68.1, D68.2, D68.3, D68.4, D68.8, D68.9, O72.3, O99.1&quot;,
#                &quot;D55, D55.0, D55.1, D55.2, D55.3, D55.8, D55.9, D56, D56.0, D56.1, D56.2, D56.3, D56.4, D56.8, D56.9, D57, D57.0, D57.1, D57.2, D57.3, D57.8, D58, D58.0, D58.1, D58.2, D58.8, D58.9, M90.4&quot;,
#                &quot;D50, D50.0, D50.1, D50.8, D50.9&quot;,
#                &quot;E78.0, E78.1, E78.2, E78.3, E78.4, E78.5, E78.9&quot;,
#                &quot;E86, E87.0, E87.1, E87.2, E87.3, E87.4, E87.5, E87.6, E87.7, E87.8, R63.1&quot;,
#                &quot;R52.0, R52.2, R52.9&quot;,
#                &quot;R25, R25.0, R25.1, R25.2, R25.3, R25.8, R26, R26.0, R26.1, R26.8, R27, R27.0, R27.8, R29.0, R29.2, R43, R43.0, R43.1, R43.2&quot;,
#                &quot;I10, I11, I11.0, I11.9, I12, I12.0, I12.9, I13, I13.0, I13.1, I13.2, I13.9, I15, I15.0, I15.1, I15.2, I15.8, I15.9, I67.4&quot;,
#                &quot;I20, I20.0, I20.1, I20.8, I20.9, I21, I21.0, I21.1, I21.2, I21.3, I21.4, I21.9, I22, I22.0, I22.1, I22.8, I22.9, I23, I23.0, I23.1, I23.2, I23.3, I23.6, I23.8, I24, I24.0, I24.1, I24.8, I24.9, I25, I25.1, I25.2, I25.3, I25.4, I25.5, I25.6, I25.8, I25.9, I34.1, I51.0, I51.3, Z95.1, Z95.5&quot;,
#                &quot;I25.10&quot;,
#                &quot;I26, I26.0&quot;,
#                &quot;I48&quot;,
#                &quot;I51.8&quot;,
#                &quot;N17, N17.0, N17.1, N17.2, N17.8, N17.9, N18, N18.0, N18.9, N19, Y60.2, Y61.2, Y62.0, Y84.1, Z49.1, Z49.2, Z99.2&quot;,
#                &quot;M21.5&quot;,
#                &quot;M40.2, M43.2, M43.8, M48.8, M49.8, M50.0, M99.6&quot;,
#                &quot;G55.1, M46.4, M50, M50.0, M50.0, M50.1, M50.2, M50.3, M50.8, M50.9, M51.3, M51.4, M96.1&quot;,
#                &quot;M13.0&quot;,
#                &quot;L94.3, M33, M33.0, M33.1, M33.2, M33.9, M34, M34.0, M34.1, M34.2, M34.8, M34.9, M35.0, M35.1, M35.5, M35.8, M35.9, M36.0, M36.8, M65.3, N16.4&quot;,
#                &quot;K09.8, L70, L70.0, L70.1, L70.2, L70.3, L70.4, L70.5, L70.8, L70.9, L72, L72.0, L72.1, L72.2, L72.8, L72.9, L73.0, L85.3&quot;,
#                &quot;N30, N30.0, N30.1, N30.2, N30.3, N30.8, N30.9, N34, N34.0, N34.2, N34.3, N35.1, N37&quot;,
#                 &quot;K70.4, K72, K72.1, K72.9, K74.0, K74.1, K74.2, K74.3, K74.4, K74.5, K74.6, K75.0, K75.1, K76.0, K76.6, K76.7&quot;,
#                &quot;B52.0, N00.0, N00.1, N00.2, N00.3, N00.4, N00.5, N00.6, N00.7, N01, N01.0, N01.1, N01.2, N01.3, N01.4, N01.5, N01.6, N01.7, N01.9, N02.0, N02.1, N02.2, N02.3, N02.4, N02.5, N02.6, N02.7, N03, N03.0, N03.1, N03.2, N03.3, N03.4, N03.5, N03.6, N03.7, N03.9, N04, N04.0, N04.1, N04.2, N04.3, N04.4, N04.5, N04.6, N04.7, N04.8, N04.9, N05, N05.0, N05.1, N05.2, N05.3, N05.4, N05.5, N05.6, N05.7, N05.9, N06.0, N06.1, N06.2, N06.3, N06.4, N06.5, N06.6, N06.7, N07.0, N07.1, N07.2, N07.3, N07.4, N07.5, N07.6, N07.7, N08, N08.1, N08.2, N08.3, N08.4, N08.5, N08.8, N14, N14.0, N14.1, N14.2, N14.3, N14.4, N15.0, N15.8, N16.1, N16.2, N16.3, N16.4, N16.5&quot;,
#                &quot;C40, C40.0, C40.1, C40.2, C40.3, C40.8, C40.9, C41, C41.0, C41.1, C41.2, C41.3, C41.4, C41.9, C47, C47.0, C47.1, C47.2, C47.3, C47.4, C47.5, C47.6, C47.8, C47.9, C49, C49.0, C49.1, C49.2, C49.3, C49.4, C49.5, C49.6, C49.8, C49.9&quot;,
# &quot;C43, C43.0, C43.1, C43.2, C43.3, C43.4, C43.5, C43.6, C43.7, C43.8, C43.9, C44.0, C44.1, C44.2, C44.3, C44.4, C44.5, C44.6, C44.7, C44.8, C44.9, D03, D03.0, D03.1, D03.2, D03.3, D03.4, D03.5, D03.6, D03.7, D03.8, D03.9, D04, D04.0, D04.1, D04.2, D04.3, D04.4, D04.5, D04.6, D04.7, D04.8, D04.9&quot;,
# &quot;R62.0, R62.8, R62.9&quot;,
# &quot;F06, F06.1, F07.0, F07.1, F07.2, F07.8, F07.9, F23, F23.0, F23.1, F23.8, F23.9, G47.1, R40.0, R40.1&quot;,
# &quot;K50, K50.0, K50.1, K50.8, K50.9, K51, K51.0, K51.1, K51.2, K51.3, K51.4, K51.5, K51.8, K51.9&quot;,
# &quot;K57, K57.0, K57.1, K57.2, K57.3, K57.4, K57.5, K57.8, K57.9&quot;,
# &quot;K62.5, K92.0, K92.1, K92.2&quot;,
# &quot;R50.8&quot;,
# &quot;K30&quot;,
# &quot;I46, I46.0, I46.9, I49.0&quot;,
# &quot;E10.5, E11.5, E14.5, I73, I73.0, I73.8, I73.9, I79.1, I79.2, I79.8&quot;,
# &quot;B96.8&quot;,
# &quot;U82, U83, U84&quot;,
# &quot;E00, E00.0, E00.1, E00.2, E00.9, E01.8, E02, E03.0, E03.1, E03.2, E03.3, E03.8, E03.9, E89.0&quot;,
# &quot;E10, E10.0, E10.1, E10.2, E10.3, E10.3, E10.3, E10.4, E10.4, E10.6, E10.7, E10.8, E10.9, E11, E11.0, E11.1, E11.2, E11.3, E11.4, E11.6, E11.7, E11.8, E11.9, E12.3, E13, E13.1, E13.3, E13.4, E13.5, E13.6, E13.7, E13.8, E13.9, E14.9, G59.0, G63.2, H36.0, R73.0, R73.9, R81, R82.4, Z96.4&quot;,
# &quot;I46, I46.0, I46.9, I49.0&quot;,
# &quot;K07.6&quot;,
# &quot;Z88.9&quot;,
# &quot;S43.4&quot;,
# &quot;S82.1, S82.3, S82.8&quot;,
#  &quot;Q50.0, Q50.1, Q50.2, Q50.3, Q50.4, Q50.5, Q50.6, Q51, Q51.0, Q51.1, Q51.2, Q51.3, Q51.4, Q51.5, Q51.6, Q51.7, Q51.8, Q51.9, Q52.0, Q52.1, Q52.2, Q52.3, Q52.4, Q52.5, Q52.6, Q52.7, Q52.8, Q52.9, Q53, Q53.0, Q53.1, Q53.2, Q53.9, Q54, Q54.0, Q54.1, Q54.2, Q54.3, Q54.4, Q54.8, Q54.9, Q55, Q55.0, Q55.1, Q55.2, Q55.3, Q55.4, Q55.5, Q55.6, Q55.8, Q55.9, Q56, Q56.0, Q56.1, Q56.2, Q56.3, Q56.4, Q60, Q60.0, Q60.1, Q60.2, Q60.3, Q60.4, Q60.5, Q60.6, Q61, Q61.0, Q61.1, Q61.2, Q61.3, Q61.4, Q61.5, Q61.8, Q61.9, Q62, Q62.0, Q62.1, Q62.3, Q62.4, Q62.5, Q62.6, Q62.7, Q62.8, Q63, Q63.0, Q63.1, Q63.2, Q63.3, Q63.8, Q63.9, Q64.0, Q64.1, Q64.2, Q64.3, Q64.4, Q64.5, Q64.6, Q64.7, Q64.8, Q64.9&quot;,
# &quot;M48.4, M48.5, M80.5, M80.8, M81.6, M81.9, M84.4, M85.9, M89.9&quot;,
# &quot;E88.0&quot;,
# &quot;J84.1, J84.2&quot;,
# &quot;E23.6&quot;,
# &quot;D89.8&quot;,
# &quot;I44.1&quot;,
# &quot;K31.8, K31.9&quot;,
# &quot;L97&quot;,
# &quot;M21.0, M21.1, M21.9&quot;,
# &quot;R11.10&quot;,
# &quot;F03&quot;,
# &quot;G47.6, G25.8&quot;,
# &quot;G43.6&quot;,
# &quot;L49&quot;,
# &quot;N42.3&quot;,
# &quot;H04.1&quot;,
# &quot;K56.6&quot;,
# &quot;N28.8&quot;,
# &quot;R31.2&quot;,
# &quot;N87&quot;,
# &quot;R09.8&quot;,
# &quot;R53&quot;,
# &quot;T38.0, T50.0&quot;,
# &quot;T85.1, T85.8&quot;,
# &quot;K91.4, K91.8, Y83.3&quot;,
# &quot;I97.8&quot;,
# &quot;J95.0&quot;,
# &quot;R25.2&quot;,
# &quot;R87.6&quot;,
# &quot;K08.1&quot;,
# &quot;M53.2, M53.3&quot;,
# &quot;R92&quot;,
# &quot;K55.2&quot;,
# &quot;M53.0, M53.1&quot;,
# &quot;M21.1, M21.8&quot;,
# &quot;K07.6&quot;)
# 
# writexl::write_xlsx(data.frame(phecode, icd10_code) |&gt; arrange(phecode),
#                     here::here(&quot;data/icd/phecode_to_icd10_manual_mapping.xlsx&quot;)
#                     )
# 
# icd10_code_origin = rep(&quot;Study PheCode (Manual Mapping)&quot;, 
#                         length(phecode))
# 
# to_add = data.frame(phecode, 
#                     icd10_code, 
#                     icd10_code_origin)
# 
# to_add = to_add |&gt; distinct()
# 
# to_add = 
#   to_add |&gt;
#   mutate(phecode = as.numeric(phecode))
# 
# disease_mapping =
#   rows_patch(disease_mapping,
#              to_add,
#              unmatched = &quot;ignore&quot;)</code></pre>
</div>
<div id="how-many-diseases-are-not-mapped-yet" class="section level3">
<h3>How many diseases are not mapped yet?</h3>
<pre class="r"><code>print(&quot;Number of ICD10 codes obtained&quot;)</code></pre>
<pre><code>[1] &quot;Number of ICD10 codes obtained&quot;</code></pre>
<pre class="r"><code>disease_mapping |&gt;
    filter(!is.na(icd10_code)) |&gt; 
  group_by(icd10_code_origin) |&gt; 
  summarise(n = n(), 
            percent = n()/n_studies *100)</code></pre>
<pre><code># A tibble: 3 × 3
  icd10_code_origin                  n percent
  &lt;chr&gt;                          &lt;int&gt;   &lt;dbl&gt;
1 Study PheCode (Manual Mapping)   875    1.89
2 Study PheCode Mapping           6780   14.6 
3 Study Provided                  3961    8.55</code></pre>
<pre class="r"><code># we were able to get PheCodes or ICD10 codes directly for roughly 25% of studies 

disease_mapping_matched =
  disease_mapping |&gt;
  filter(icd10_code != &quot;&quot; | !is.na(icd10_code))

not_found_diseases &lt;- diseases[!diseases %in% disease_mapping_matched$collected_all_disease_terms] 
not_found_diseases &lt;- not_found_diseases[not_found_diseases != &quot;&quot;]

print(&quot;Number of diseases not mapped to a single ICD10 code yet:&quot;)</code></pre>
<pre><code>[1] &quot;Number of diseases not mapped to a single ICD10 code yet:&quot;</code></pre>
<pre class="r"><code>print(length(not_found_diseases))</code></pre>
<pre><code>[1] 653</code></pre>
</div>
</div>
<div id="get-icd10-codes-by-matching-diseasetrait-terms"
class="section level2">
<h2>Get ICD10 codes by matching <code>DISEASE/TRAIT</code> terms</h2>
<div id="tidy-diseasetrait-column-to-better-match-terms"
class="section level3">
<h3>Tidy <code>DISEASE/TRAIT</code> Column to better match terms</h3>
<div id="removing-model-related-terms" class="section level4">
<h4>Removing model related terms</h4>
<pre class="r"><code># removing &#39;(Gene-based burden)&#39; in `DISEASE/TRAIT` column
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(Gene-based burden\\)&quot;))

# removing &#39;(1df test)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(1df test\\)|\\s*\\(1df\\)&quot;))

# removing `(2df test)`
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(2df test\\)||\\s*\\(2df\\)&quot;))

# removing &#39;(3df test)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(3df test\\)|\\s*\\(3df\\)&quot;))

# removing &#39;(multivariate analysis)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(multivariate analysis\\)&quot;))

# removing &#39;(ordinary least squares (OLS))&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(ordinary least squares \\(OLS\\)\\)&quot;))

# remove &#39;(CNV mirror model)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(CNV mirror model\\)&quot;))

# remove &#39;(CNV U-shape model)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(CNV U-shape model\\)&quot;))

# removing &#39;(MTAG)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(MTAG\\)&quot;))

# removing &#39;(pleiotropy)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(pleiotropy\\)&quot;))

# removing &#39;(multi-trait analysis)&#39;


# remove &#39;SNP x SNP interaction&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(SNP x SNP interaction\\)&quot;))</code></pre>
</div>
<div id="removing-study-related-phenotype-ids" class="section level4">
<h4>Removing study related phenotype IDs</h4>
<pre class="r"><code># removing references to UKB data fields in `DISEASE/TRAIT` column
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(UKB data field \\d+\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(UKB data field \\d+_\\d+\\)&quot;))


# removing Phecode references (e.g. (Phecode 250.2))
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(Phe[Cc]ode [0-9]+\\.[0-9]+\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(Phe[Cc]ode [0-9]+\\)&quot;))

# remove any substring of the form (Number_Number) — e.g. (13013_41)
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = 
           str_remove_all(`DISEASE/TRAIT`, 
                          &quot;\\s*\\(\\d+_\\d+\\)&quot;)
         )</code></pre>
</div>
<div id="remove-patient-genotype-effects" class="section level4">
<h4>Remove patient genotype effects</h4>
<pre class="r"><code># removing genotype effect (e.g. (fetal genotype effect))
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = 
           str_remove_all(`DISEASE/TRAIT`, 
                          &quot;\\s*\\([^)]*genotype effect\\)&quot;)
         ) |&gt;
    mutate(`DISEASE/TRAIT` = 
           str_remove_all(`DISEASE/TRAIT`, 
                          &quot;\\s*\\([^)]* effect\\)&quot;)
         )

# remove &#39;(adjusted for APOE e4 dosage)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\(adjusted for APOE e4 dosage\\)&quot;))

# remove &#39;(maternal):&#39; &amp; &#39;(paternal):
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(maternal\\):&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(paternal\\):&quot;))


# remove &#39;Biological Grandparent &#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;Biological Grandparent &quot;))

# remove &#39;Biological Father: &#39;, &#39;Biological Sibling: &#39;, &#39;Biological Mother: &#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;Biological Father: &quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;Biological Sibling: &quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, &quot;Biological Mother: &quot;))</code></pre>
</div>
<div id="removing-trait-specific-terms" class="section level4">
<h4>Removing trait specific terms</h4>
<pre class="r"><code># remove  (apnea hypopnea index), 
# (average respiratory event duration), 
#  (micro-arousal index)
#  (percentage of N3 sleep time during total sleep time)
# (percentage of N3 sleep time during sleep period time)
# (average oxyhemoglobin desaturation per event)
#  (average oxyhemoglobin saturation across sleep episode)
# (percentage sleep with oxyhemoglobin saturation less than 90%)
#  (wake time during sleep period time)
# (minimum oxyhemoglobin saturation across sleep episode)
#  (oxygen desaturation index)
# (average oxygen saturation during sleep)
# (apnea hypopnea index, change over time)

disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(apnea hypopnea index\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(average respiratory event duration\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(micro-arousal index\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(percentage of N3 sleep time during total sleep time\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(percentage of N3 sleep time during sleep period time\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(average oxyhemoglobin desaturation per event\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(average oxyhemoglobin saturation across sleep episode\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(percentage sleep with oxyhemoglobin saturation less than 90%\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(wake time during sleep period time\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(minimum oxyhemoglobin saturation across sleep episode\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(oxygen desaturation index\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(average oxygen saturation during sleep\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`,
                                          &quot;\\s*\\(apnea hypopnea index, change over time\\)&quot;))

disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot; during REM sleep$| during non-REM sleep$&quot;))


# ends in  levels in coronary artery disease, make it just coronary artery disease
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = 
           ifelse(str_detect(`DISEASE/TRAIT`,
                             &quot;(?i)\\s*levels in coronary artery disease$&quot;),
                  &quot;coronary artery disease&quot;,
                  `DISEASE/TRAIT`))

# ends in levels in chronic kidney disease, make it just chronic kidney disease
# ends in  levels in coronary artery disease, make it just coronary artery disease
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = 
           ifelse(str_detect(`DISEASE/TRAIT`,
                             &quot;(?i)\\s*levels in chronic kidney disease$&quot;),
                  &quot;chronic kidney disease&quot;,
                  `DISEASE/TRAIT`))

# ends in levels in type 2 diabetes, make it just type 2 diabetes
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = 
           ifelse(str_detect(`DISEASE/TRAIT`,
                             &quot;(?i)\\s*levels in type 2 diabetes$&quot;),
                  &quot;type 2 diabetes&quot;,
                  `DISEASE/TRAIT`))

# ends in levels in prediabetes, make it just prediabetes
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = 
           ifelse(str_detect(`DISEASE/TRAIT`,
                             &quot;(?i)\\s*levels in prediabetes$&quot;),
                  &quot;prediabetes&quot;,
                  `DISEASE/TRAIT`))


# remove str_extract(x, &quot;(?&lt;=Takes medication for )\\w+&quot;)
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` =
           ifelse(str_detect(`DISEASE/TRAIT`,
                             &quot;(?i)Takes medication for \\w+&quot;),
                  str_extract(`DISEASE/TRAIT`,
                              &quot;(?&lt;=Takes medication for )\\w+&quot;),
                  `DISEASE/TRAIT`))
                                          

# remove BMI adjustments 
# &#39;(BMI adjusted)&#39;, or &#39;(adjusted for BMI)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(BMI adjusted\\)|\\s*\\(adjusted for BMI\\)&quot;))

# remove &#39;trait&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*trait$&quot;))


# remove  (slight), (severe), (generalised), (localised)
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(slight\\)|\\s*\\(severe\\)|\\s*\\(generalised\\)|\\s*\\(localised\\)&quot;))</code></pre>
</div>
<div id="removing-patient-condition-related-terms"
class="section level4">
<h4>Removing patient condition related terms</h4>
<pre class="r"><code># removing &#39;(time to event)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(time to event\\)&quot;))

# removing age of onset
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;\\s*\\(age of onset\\)|\\s*\\(age at onset\\)&quot;)) |&gt;
  mutate(`DISEASE/TRAIT` = str_remove_all(`DISEASE/TRAIT`, 
                                          &quot;^Age of onset of &quot;))


# remove visit times
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` =
  str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(\\d+ month visit\\)&quot;)
  )

# remove &#39;(baseline)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` =
  str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(baseline\\)&quot;)
  )

# remove &#39;(hospitalized vs population)&#39;, &#39;(hospitalized vs not hospitalized)&#39;, &#39;(severe vs population)&#39;
disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` =
  str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(hospitalized vs population\\)|\\s*\\(hospitalized vs not hospitalized\\)|\\s*\\(severe vs population\\)&quot;)
  )

# remove (RELEASE [Number])
disease_mapping = 
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` =
  str_remove_all(`DISEASE/TRAIT`, &quot;\\s*\\(RELEASE \\d+\\)&quot;)
  )</code></pre>
</div>
<div id="remove-white-space" class="section level4">
<h4>Remove white space</h4>
<pre class="r"><code>disease_mapping =
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = str_squish(`DISEASE/TRAIT`))</code></pre>
</div>
</div>
<div id="match-diseasetrait-to-icd-10-descriptions"
class="section level3">
<h3>Match disease/trait to ICD 10 descriptions</h3>
<pre class="r"><code>matched_icd10_desc  = 
phecodes |&gt;
    mutate(collected_all_disease_terms = tolower(iconv(ICD_DESCRIPTION, 
                                                       to = &quot;UTF-8&quot;))
           ) |&gt;
    select(collected_all_disease_terms,    
           icd10_code = ICD10) 

# filter missing strings
matched_icd10_desc = 
  matched_icd10_desc |&gt;
  filter(!is.na(collected_all_disease_terms) &amp; 
         collected_all_disease_terms != &quot;&quot;) |&gt;
  filter(!is.na(icd10_code) &amp; 
         icd10_code != &quot;&quot;)


matched_icd10_desc =
  matched_icd10_desc |&gt;
  group_by(collected_all_disease_terms) |&gt;
  summarise(icd10_code = str_flatten(unique(icd10_code), 
                                     collapse = &quot;, &quot;, 
                                     na.rm = T), 
            .groups = &quot;drop&quot;) 
  

matched_icd10_desc = 
  matched_icd10_desc |&gt;
    mutate(icd10_code_origin = &quot;ICD Description Match (DISEASE/TRAIT)&quot;)

# match by DISEASE/TRAIT
matched_icd10_desc =
  matched_icd10_desc |&gt;
  rename(`DISEASE/TRAIT` = collected_all_disease_terms)

disease_mapping = 
disease_mapping |&gt;
rows_patch(matched_icd10_desc,
           unmatched = &quot;ignore&quot;)

print(&quot;Number of ICD10 codes obtained&quot;)</code></pre>
<pre><code>[1] &quot;Number of ICD10 codes obtained&quot;</code></pre>
<pre class="r"><code>disease_mapping |&gt;
    filter(!is.na(icd10_code)) |&gt; 
  group_by(icd10_code_origin) |&gt; 
  summarise(n = n(), 
            percent = n()/n_studies *100)</code></pre>
<pre><code># A tibble: 4 × 3
  icd10_code_origin                         n percent
  &lt;chr&gt;                                 &lt;int&gt;   &lt;dbl&gt;
1 ICD Description Match (DISEASE/TRAIT)  3945    8.52
2 Study PheCode (Manual Mapping)          875    1.89
3 Study PheCode Mapping                  6780   14.6 
4 Study Provided                         3961    8.55</code></pre>
</div>
<div id="match-diseasetrait-to-phenotypes-corresponding-to-phecodes"
class="section level3">
<h3>Match disease/trait to phenotypes (corresponding to PheCodes)</h3>
<pre class="r"><code>phenotype_icd_map =
  phecodes |&gt;
group_by(Phenotype) |&gt;
  summarise(icd10_code = 
            str_flatten(ICD10, 
                        collapse = &quot;, &quot;, 
                        na.rm = T), 
            .groups = &quot;drop&quot;)

matched_phenotypes =
phenotype_icd_map #|&gt;
#filter(tolower(Phenotype) %in% not_found_diseases)

matched_phenotypes =
matched_phenotypes |&gt;
  mutate(collected_all_disease_terms = tolower(Phenotype)) |&gt;
  select(collected_all_disease_terms, icd10_code) |&gt;
  mutate(icd10_code_origin = &quot;Phecode Phenotype Match (DISEASE/TRAIT)&quot;)

# match by DISEASE/TRAIT
matched_phenotypes =
  matched_phenotypes |&gt;
  rename(`DISEASE/TRAIT` = collected_all_disease_terms)

disease_mapping =
disease_mapping |&gt;
rows_patch(matched_phenotypes,
           unmatched = &quot;ignore&quot;)

disease_mapping |&gt;
  filter(icd10_code_origin == &quot;Phecode Phenotype Match (DISEASE/TRAIT)&quot;) |&gt;
  nrow()</code></pre>
<pre><code>[1] 580</code></pre>
<pre class="r"><code>disease_mapping |&gt;
    filter(!is.na(icd10_code)) |&gt; 
  group_by(icd10_code_origin) |&gt; 
  summarise(n = n(), 
            percent = n()/n_studies *100)</code></pre>
<pre><code># A tibble: 5 × 3
  icd10_code_origin                           n percent
  &lt;chr&gt;                                   &lt;int&gt;   &lt;dbl&gt;
1 ICD Description Match (DISEASE/TRAIT)    3945    8.52
2 Phecode Phenotype Match (DISEASE/TRAIT)   580    1.25
3 Study PheCode (Manual Mapping)            875    1.89
4 Study PheCode Mapping                    6780   14.6 
5 Study Provided                           3961    8.55</code></pre>
</div>
<div id="how-many-diseases-are-not-mapped-yet-1" class="section level3">
<h3>How many diseases are not mapped yet?</h3>
<pre class="r"><code>matched =
  disease_mapping |&gt;
  filter(icd10_code != &quot;&quot;) |&gt;
  pull(collected_all_disease_terms)

not_found_diseases &lt;- diseases[!diseases %in% 
                               matched
                               ] 

not_found_diseases &lt;- not_found_diseases[not_found_diseases != &quot;&quot;]

print(length(not_found_diseases))</code></pre>
<pre><code>[1] 652</code></pre>
</div>
<div id="match-diseasetrait-to-umls" class="section level3">
<h3>Match disease/trait to UMLS</h3>
<pre class="r"><code>unmapped_disease_terms &lt;-
  disease_mapping |&gt;
  filter(is.na(icd10_code)) |&gt;
  pull(`DISEASE/TRAIT`) |&gt;
  unique() |&gt;
  tolower()

# get UMLS CUIs with ICD10 mappings
umls_data &lt;-
  data.table::fread(here::here(&quot;data/icd/2025AA/META/MRCONSO.RRF&quot;),
                    sep = &quot;|&quot;,
                    header = FALSE,
                    quote = &quot;&quot;,
                    fill = TRUE,
                    na.strings = c(&quot;&quot;, &quot;NA&quot;)
  )

colnames(umls_data)[1:18] &lt;- c(
  &quot;CUI&quot;,&quot;LAT&quot;,&quot;TS&quot;,&quot;LUI&quot;,&quot;STT&quot;,&quot;SUI&quot;,&quot;ISPREF&quot;,
  &quot;AUI&quot;,&quot;SAUI&quot;,&quot;SCUI&quot;,&quot;SDUI&quot;,&quot;SAB&quot;,&quot;TTY&quot;,&quot;CODE&quot;,
  &quot;STR&quot;,&quot;SRL&quot;,&quot;SUPPRESS&quot;,&quot;CVF&quot;
)

umls_cuis_icd10 &lt;-
  umls_data |&gt;
  filter(SAB %in% c(&quot;ICD10&quot;, &quot;ICD10CM&quot;)) |&gt;
  select(CODE, CUI) |&gt;
  group_by(CUI) |&gt;
  summarise(CODE = str_flatten(unique(CODE),
                               collapse = &quot;, &quot;,
                               na.rm = T),
            .groups = &quot;drop&quot;
            ) |&gt;
  rename(icd10_code = CODE) 



umls_icd10 &lt;-
  umls_data |&gt;
  filter(CUI %in% umls_cuis_icd10$CUI)

umls_icd10 =
  umls_icd10 |&gt;
  left_join(umls_cuis_icd10,
            by = &quot;CUI&quot;)

# overlap with umls terms
disease_trait_umls  &lt;-
  umls_icd10 |&gt;
  filter(tolower(STR) %in% unmapped_disease_terms) |&gt;
  mutate(`DISEASE/TRAIT` = tolower(STR)) |&gt;
  # mutate(CODE = ifelse(SAB %in% c(&quot;ICD10&quot;, &quot;ICD10CM&quot;),
  #                      CODE,
  #                      NA)) |&gt;
  select(`DISEASE/TRAIT`,
         icd10_code) |&gt;
  group_by(`DISEASE/TRAIT`) |&gt;
  summarise(icd10_code = str_flatten(unique(icd10_code),
                               collapse = &quot;, &quot;,
                               na.rm = T),
            .groups = &quot;drop&quot;
            ) |&gt;
  distinct() |&gt;
  mutate(icd10_code_origin = &quot;UMLS term match&quot;)

 disease_mapping &lt;-
  disease_mapping |&gt;
  mutate(`DISEASE/TRAIT` = tolower(`DISEASE/TRAIT`)) |&gt;
  rows_patch(disease_trait_umls,
             by = &quot;DISEASE/TRAIT&quot;)</code></pre>
</div>
</div>
</div>
<div id="get-icd10-codes-by-matching-collected_all_disease_terms-terms"
class="section level1">
<h1>Get ICD10 codes by matching <code>collected_all_disease_terms</code>
terms</h1>
<div id="match-collected_all_disease_terms-to-icd10-descriptions"
class="section level3">
<h3>Match collected_all_disease_terms to ICD10 descriptions</h3>
<pre class="r"><code>matched_icd10_desc = 
  matched_icd10_desc |&gt;
  rename(collected_all_disease_terms = `DISEASE/TRAIT`)

matched_icd10_desc = 
  matched_icd10_desc |&gt;
  mutate(icd10_code_origin = &quot;ICD Description Match (collected_all_disease_terms)&quot;)

disease_mapping =
  rows_patch(disease_mapping,
             matched_icd10_desc,
             unmatched = &quot;ignore&quot;)</code></pre>
<pre><code>Matching, by = &quot;collected_all_disease_terms&quot;</code></pre>
</div>
<div id="how-many-diseases-are-not-mapped-yet-2" class="section level3">
<h3>How many diseases are not mapped yet?</h3>
<pre class="r"><code>print(&quot;Number of ICD10 codes obtained&quot;)</code></pre>
<pre><code>[1] &quot;Number of ICD10 codes obtained&quot;</code></pre>
<pre class="r"><code>disease_mapping |&gt;
    filter(!is.na(icd10_code)) |&gt; 
    group_by(icd10_code_origin) |&gt; 
    summarise(n = n(), 
              percent = n()/n_studies *100)</code></pre>
<pre><code># A tibble: 7 × 3
  icd10_code_origin                                       n percent
  &lt;chr&gt;                                               &lt;int&gt;   &lt;dbl&gt;
1 ICD Description Match (DISEASE/TRAIT)                3945    8.52
2 ICD Description Match (collected_all_disease_terms)  9575   20.7 
3 Phecode Phenotype Match (DISEASE/TRAIT)               580    1.25
4 Study PheCode (Manual Mapping)                        875    1.89
5 Study PheCode Mapping                                6780   14.6 
6 Study Provided                                       3961    8.55
7 UMLS term match                                      4820   10.4 </code></pre>
<pre class="r"><code>matched =
  disease_mapping |&gt;
  filter(icd10_code != &quot;&quot;) |&gt;
  pull(collected_all_disease_terms)

not_found_diseases &lt;- diseases[!diseases %in% 
                               matched
                               ] 

not_found_diseases &lt;- not_found_diseases[not_found_diseases != &quot;&quot;]

print(length(not_found_diseases))</code></pre>
<pre><code>[1] 455</code></pre>
</div>
<div id="match-collected_all_disease_terms-to-icd10-codes-by-phenotypes"
class="section level3">
<h3>Match collected_all_disease_terms to ICD10 codes (by
phenotypes)</h3>
<pre class="r"><code>matched_phenotypes = 
  matched_phenotypes |&gt;
  rename(collected_all_disease_terms = `DISEASE/TRAIT`)

matched_phenotypes = 
  matched_phenotypes |&gt;
  mutate(icd10_code_origin = &quot;Phecode Phenotype Match (collected_all_disease_terms)&quot;)

# match by collected_all_disease_terms
disease_mapping = 
disease_mapping |&gt;
rows_patch(matched_phenotypes,
           unmatched = &quot;ignore&quot;)</code></pre>
<pre><code>Matching, by = &quot;collected_all_disease_terms&quot;</code></pre>
</div>
<div id="how-many-diseases-are-not-mapped-yet-3" class="section level3">
<h3>How many diseases are not mapped yet?</h3>
<pre class="r"><code>print(&quot;Number of ICD10 codes obtained&quot;)</code></pre>
<pre><code>[1] &quot;Number of ICD10 codes obtained&quot;</code></pre>
<pre class="r"><code>disease_mapping |&gt;
    filter(!is.na(icd10_code)) |&gt; 
    group_by(icd10_code_origin) |&gt; 
    summarise(n = n(), 
              percent = n()/n_studies *100)</code></pre>
<pre><code># A tibble: 8 × 3
  icd10_code_origin                                         n percent
  &lt;chr&gt;                                                 &lt;int&gt;   &lt;dbl&gt;
1 ICD Description Match (DISEASE/TRAIT)                  3945    8.52
2 ICD Description Match (collected_all_disease_terms)    9575   20.7 
3 Phecode Phenotype Match (DISEASE/TRAIT)                 580    1.25
4 Phecode Phenotype Match (collected_all_disease_terms)  2194    4.74
5 Study PheCode (Manual Mapping)                          875    1.89
6 Study PheCode Mapping                                  6780   14.6 
7 Study Provided                                         3961    8.55
8 UMLS term match                                        4820   10.4 </code></pre>
<pre class="r"><code>matched =
  disease_mapping |&gt;
  filter(icd10_code != &quot;&quot;) |&gt;
  pull(collected_all_disease_terms)

not_found_diseases &lt;- diseases[!diseases %in% 
                               matched
                               ] 

not_found_diseases &lt;- not_found_diseases[not_found_diseases != &quot;&quot;]

print(length(not_found_diseases))</code></pre>
<pre><code>[1] 442</code></pre>
</div>
<div id="umls-match-collected_all_disease_terms" class="section level3">
<h3>UMLS match collected_all_disease_terms</h3>
<pre class="r"><code>unmapped_terms &lt;-
  disease_mapping |&gt;
  filter(is.na(icd10_code)) |&gt;
  pull(collected_all_disease_terms) |&gt;
  unique()

# overlap with umls terms
collected_trait_umls  &lt;-
  umls_icd10 |&gt;
  filter(tolower(STR) %in% unmapped_terms) |&gt;
  mutate(collected_all_disease_terms = tolower(STR)) |&gt;
  # mutate(CODE = ifelse(SAB %in% c(&quot;ICD10&quot;, &quot;ICD10CM&quot;),
  #                      CODE,
  #                      NA)) |&gt;
  select(collected_all_disease_terms,
         icd10_code) |&gt;
  group_by(collected_all_disease_terms) |&gt;
  summarise(icd10_code = str_flatten(unique(icd10_code),
                               collapse = &quot;, &quot;,
                               na.rm = T),
            .groups = &quot;drop&quot;
            ) |&gt;
  distinct() |&gt;
  mutate(icd10_code_origin = &quot;UMLS term match (collected_all_disease_terms)&quot;)

disease_mapping &lt;-
  disease_mapping |&gt;
  rows_patch(collected_trait_umls,
             by = &quot;collected_all_disease_terms&quot;)</code></pre>
</div>
</div>
<div id="manual-mapping-of-gwas-traits-to-icd-10"
class="section level1">
<h1>Manual mapping of GWAS traits to ICD 10</h1>
<pre class="r"><code>manual_icd10_map &lt;-
  readxl::read_xlsx(here::here(&quot;data/icd/manual_disease_icd10_mappings.xlsx&quot;))

manual_icd10_map =
  manual_icd10_map |&gt;
  select(collected_all_disease_terms = mapped_trait, 
         icd10_code) |&gt;
  mutate(collected_all_disease_terms = stringr::str_squish(tolower(collected_all_disease_terms))) |&gt;
  mutate(icd10_code_origin = &quot;Manual Mapping (collected_all_disease_terms)&quot;)

# disease_mapping =
#   bind_rows(disease_mapping, to_add) |&gt;
#   distinct()
disease_mapping =
  rows_patch(disease_mapping,
             manual_icd10_map,
             unmatched = &quot;ignore&quot;)</code></pre>
<pre><code>Matching, by = &quot;collected_all_disease_terms&quot;</code></pre>
<pre class="r"><code>disease_mapping |&gt;
  filter(icd10_code_origin == &quot;Manual Mapping (collected_all_disease_terms)&quot;) |&gt;
  nrow()</code></pre>
<pre><code>[1] 1172</code></pre>
<pre class="r"><code># repeat for `DISEASE/TRAIT`
manual_icd10_map =
  manual_icd10_map |&gt;
  select(`DISEASE/TRAIT` = collected_all_disease_terms,
         icd10_code) |&gt;
  mutate(icd10_code_origin = &quot;Manual Mapping (DISEASE/TRAIT)&quot;)

disease_mapping =
  rows_patch(disease_mapping |&gt; mutate(`DISEASE/TRAIT` = tolower(`DISEASE/TRAIT`)),
             manual_icd10_map,
             unmatched = &quot;ignore&quot;)</code></pre>
<pre><code>Matching, by = &quot;DISEASE/TRAIT&quot;</code></pre>
<div id="additional-manual-mapping" class="section level2">
<h2>Additional manual mapping</h2>
<pre class="r"><code>disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                            grepl(&quot;Anthracycline-induced cardiotoxicity|Trastuzumab-induced cardiotoxicity&quot;, `DISEASE/TRAIT`, ignore.case = T),
                            &quot;I42.7, T45.1&quot;,
                             icd10_code)) |&gt;
mutate(icd10_code_origin = 
         ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                  grepl(&quot;Anthracycline-induced cardiotoxicity|Trastuzumab-induced cardiotoxicity&quot;, `DISEASE/TRAIT`, ignore.case = T),
                &quot;Manual Mapping (from DISEASE/TRAIT)&quot;,
                icd10_code_origin)
         )

disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                            grepl(&quot;Induration at the injection site after COVID-19&quot;, `DISEASE/TRAIT`, ignore.case = T),
                            &quot;R23.4&quot;,
                             icd10_code)) |&gt;
  mutate(collected_all_disease_terms = 
           ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                    grepl(&quot;Induration at the injection site after COVID-19&quot;, `DISEASE/TRAIT`, ignore.case = T),
                  &quot;induration of skin&quot;,
                  collected_all_disease_terms)) |&gt;
  mutate(icd10_code_origin =
           ifelse(collected_all_disease_terms == &quot;induration of skin&quot;,
                  &quot;Manual Mapping (from DISEASE/TRAIT)&quot;,
                  icd10_code_origin)
         )


# N64.5
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                            grepl(&quot;Induration \\(&gt;= grade 2\\) in breast cancer treated with radiotherapy&quot;, 
                                  `DISEASE/TRAIT`, ignore.case = T),
                            &quot;N64.5&quot;,
                             icd10_code)) |&gt;
  mutate(collected_all_disease_terms = 
           ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                    grepl(&quot;Induration \\(&gt;= grade 2\\) in breast cancer treated with radiotherapy&quot;, 
                          `DISEASE/TRAIT`, ignore.case = T),
                  &quot;induration of breast&quot;,
                  collected_all_disease_terms)) |&gt;
  mutate(icd10_code_origin =
           ifelse(collected_all_disease_terms == &quot;induration of breast&quot;,
                  &quot;Manual Mapping (from DISEASE/TRAIT)&quot;,
                  icd10_code_origin)
         )


# T45.1
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                            grepl(&quot;Nivolumab-induced immune-related adverse events in cancer|Response to immune checkpoint inhibitors in melanoma&quot;, 
                                  `DISEASE/TRAIT`, ignore.case = T),
                            &quot;T45.1&quot;,
                             icd10_code)) |&gt;
  mutate(icd10_code_origin =
           ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                    grepl(&quot;Nivolumab-induced immune-related adverse events in cancer|Response to immune checkpoint inhibitors in melanoma&quot;, 
                          `DISEASE/TRAIT`, ignore.case = T),
                  &quot;Manual Mapping (from DISEASE/TRAIT)&quot;,
                  icd10_code_origin)
         )


disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                            grepl(&quot;Methotrexate-related central neurotoxicity in children treated for acute lymphoblastic leukemia&quot;, 
                                  `DISEASE/TRAIT`, ignore.case = T),
                            &quot;G92, T45.1&quot;,
                             icd10_code)) |&gt;
  mutate(icd10_code_origin =
           ifelse(collected_all_disease_terms == &quot;toxicity&quot; &amp; 
                    grepl(&quot;Methotrexate-related central neurotoxicity in children treated for acute lymphoblastic leukemia&quot;, 
                          `DISEASE/TRAIT`, ignore.case = T),
                  &quot;Manual Mapping (from DISEASE/TRAIT)&quot;,
                  icd10_code_origin)
         )

## if DISEASE/TRAIT ==  Abnormalities of forces of labour
## and collected_all_disease_terms == abnormal delivery
## if icd10_code is missing, map to
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;abnormal delivery&quot; &amp; 
                            grepl(&quot;^Abnormalities of forces of labour$&quot;, 
                                  `DISEASE/TRAIT`, 
                                  ignore.case = T) &amp;
                            icd10_code == &quot;&quot;,
                            &quot;O62&quot;,
                             icd10_code)) |&gt;
  mutate(icd10_code_origin = 
         ifelse(collected_all_disease_terms == &quot;abnormal delivery&quot; &amp; 
                  grepl(&quot;^Abnormalities of forces of labour$&quot;, 
                        `DISEASE/TRAIT`, 
                        ignore.case = T),
                &quot;Manual Mapping (from DISEASE/TRAIT)&quot;,
                icd10_code_origin)
         )

## if DISEASE/TRAIT == Ischemic heart disease
## and collected_all_disease_terms == heart disease
## if icd10_code is missing, map to &quot;I20-I21.6, I21.9-I25.9, Z82.4-Z82.49&quot;
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;heart disease&quot; &amp; 
                            grepl(&quot;^Ischemic heart disease$&quot;, 
                                  `DISEASE/TRAIT`, 
                                  ignore.case = T) &amp;
                            icd10_code == &quot;&quot;,
                            &quot;I20-I21.6, I21.9-I25.9, Z82.4-Z82.49&quot;,
                             icd10_code)) |&gt;
  mutate(icd10_code_origin = 
         ifelse(collected_all_disease_terms == &quot;heart disease&quot; &amp; 
                  grepl(&quot;^Ischemic heart disease$&quot;, 
                        `DISEASE/TRAIT`, 
                        ignore.case = T),
                &quot;Manual Mapping (from DISEASE/TRAIT)&quot;,
                icd10_code_origin)
         )


# if collected_all_disease_terms == &quot;heart disease&quot;
# and icd10_code is missing
disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;heart disease&quot; &amp;
                            icd10_code == &quot;&quot;,
                            &quot;C34&quot;,
                             icd10_code)) |&gt;
  mutate(icd10_code_origin = 
         ifelse(collected_all_disease_terms == &quot;heart disease&quot; &amp;
                  icd10_code == &quot;C34&quot;,
                &quot;Manual Mapping (collected_all_disease_terms)&quot;,
                icd10_code_origin)
         )</code></pre>
<div id="how-many-diseases-are-not-mapped-yet-4" class="section level3">
<h3>How many diseases are not mapped yet?</h3>
<pre class="r"><code># disease_mapping =
#   disease_mapping |&gt;
#   filter(icd10_code != &quot;&quot;)

matched &lt;- c(disease_mapping_matched$collected_all_disease_terms,
             matched_phenotypes$collected_all_disease_terms,
             to_add$collected_all_disease_terms,
             manual_icd10_map$collected_all_disease_terms)</code></pre>
<pre><code>Warning: Unknown or uninitialised column: `collected_all_disease_terms`.
Unknown or uninitialised column: `collected_all_disease_terms`.</code></pre>
<pre class="r"><code>not_found_diseases &lt;- diseases[!diseases %in% matched] 
not_found_diseases &lt;- not_found_diseases[not_found_diseases != &quot;&quot;]

print(length(not_found_diseases))</code></pre>
<pre><code>[1] 617</code></pre>
</div>
</div>
</div>
<div id="inferring-icd10-codes-from-similar-study-icd10-codes"
class="section level1">
<h1>Inferring ICD10 codes from similar study ICD10 codes</h1>
<pre class="r"><code># similar studies:
study_icd_map =
disease_mapping |&gt;
filter(!is.na(icd10_code)) |&gt;
filter(icd10_code_origin == &quot;Study Provided&quot; | icd10_code_origin == &quot;Study PheCode Mapping&quot;) 

study_icd_map =
  study_icd_map |&gt;
  select(collected_all_disease_terms, icd10_code) |&gt;
  distinct()

# remove Z95.1 and Z95.5 from coronary artery disease
study_icd_map =
  study_icd_map |&gt;
  mutate(icd10_code = ifelse(collected_all_disease_terms == &quot;coronary artery disease&quot;,
                             str_replace_all(icd10_code, &quot;Z95.1|Z95.5&quot;, &quot;&quot;),
                             icd10_code))

# assume ovarian cancer terms more specific than 
# Malignant neoplasm of ovary and other uterine adnexa (PheCode 184.1)
# study_icd_map =
#   study_icd_map |&gt;
#   filter(!( collected_all_disease_terms == &quot;ovarian cancer&quot; &amp;&amp;phecode == &quot;184.10&quot;)
#   )

study_icd_map = 
  study_icd_map |&gt;
  mutate(collected_all_disease_terms = str_trim(collected_all_disease_terms)) |&gt;
  mutate(collected_all_disease_terms = str_remove_all(collected_all_disease_terms, &quot;^, &quot;)) 
  

study_icd_map =
  study_icd_map |&gt;
  filter(icd10_code != &quot;&quot; &amp; !is.na(icd10_code))


study_icd_map =
  study_icd_map |&gt;
  tidyr::separate_longer_delim(icd10_code, delim = &quot;, &quot;) |&gt;
  tidyr::separate_longer_delim(icd10_code, delim = &quot;,&quot;)

study_icd_map =
  study_icd_map |&gt;
  group_by(collected_all_disease_terms) |&gt;
  summarise(icd10_code = str_flatten(unique(sort(icd10_code)),
                                     collapse = &quot;, &quot;, 
                                     na.rm = T), 
            .groups = &quot;drop&quot;)

study_icd_map = 
  study_icd_map |&gt;
  mutate(icd10_code_origin = &quot;Inferred from similar studies&quot;)

disease_mapping = 
  rows_patch(disease_mapping,
             study_icd_map,
             unmatched = &quot;ignore&quot;)</code></pre>
<pre><code>Matching, by = &quot;collected_all_disease_terms&quot;</code></pre>
<pre class="r"><code>disease_mapping |&gt;
  filter(is.na(icd10_code)) |&gt;
  nrow()</code></pre>
<pre><code>[1] 495</code></pre>
</div>
<div id="how-was-icd-10-code-inferred" class="section level1">
<h1>How was ICD-10 code inferred?</h1>
<pre class="r"><code>disease_mapping  |&gt; 
group_by(icd10_code_origin) |&gt; 
summarise(n = n()) |&gt; 
arrange(desc(n))</code></pre>
<pre><code># A tibble: 13 × 2
   icd10_code_origin                                         n
   &lt;chr&gt;                                                 &lt;int&gt;
 1 UMLS term match (collected_all_disease_terms)         10995
 2 ICD Description Match (collected_all_disease_terms)    9575
 3 Study PheCode Mapping                                  6780
 4 UMLS term match                                        4820
 5 Study Provided                                         3961
 6 ICD Description Match (DISEASE/TRAIT)                  3945
 7 Phecode Phenotype Match (collected_all_disease_terms)  2194
 8 Manual Mapping (collected_all_disease_terms)           1172
 9 Inferred from similar studies                           899
10 Study PheCode (Manual Mapping)                          875
11 Phecode Phenotype Match (DISEASE/TRAIT)                 580
12 &lt;NA&gt;                                                    495
13 Manual Mapping (DISEASE/TRAIT)                           25</code></pre>
</div>
<div id="saving-disease-mapping" class="section level1">
<h1>Saving disease mapping</h1>
<div id="add-description-of-icd10-codes-to-mapping-table"
class="section level2">
<h2>Add description of ICD10 codes to mapping table</h2>
<div id="prepare-fix-disease-mappings" class="section level3">
<h3>Prepare / fix disease mappings</h3>
<div id="fixing-multiple-icd10-codes-missing-commas"
class="section level4">
<h4>Fixing multiple ICD10 codes missing commas</h4>
<pre class="r"><code>disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;F40 F41 F42&quot;,
    replacement = &quot;F40, F41, F42&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;G47.8. G47.9&quot;,
    replacement = &quot;G47.8, G47.9&quot;
    )
  )</code></pre>
</div>
<div id="converting-icd-10-cm-codes-to-who-icd-10"
class="section level4">
<h4>Converting ICD-10 cm codes to WHO ICD-10</h4>
<pre class="r"><code>disease_mapping =
  disease_mapping |&gt;
  mutate(icd10_code = str_split(icd10_code, &quot;,\\s*&quot;)) |&gt;
  tidyr::unnest(icd10_code) 


disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;B37.49&quot;,
    replacement = &quot;B37.4&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;E05.90&quot;,
    replacement = &quot;E05.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;E11.319|E11.31|E11.329.9|E11.329.|E11.32|E11.3.&quot;,
    replacement = &quot;E11.3&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;E11.3.&quot;,
    replacement = &quot;E11.3&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;E13.621.|E13.62&quot;,
    replacement = &quot;E11.3&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;E78.00&quot;,
    replacement = &quot;E78.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;F10.10&quot;,
    replacement = &quot;F10.1&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;F17.201.|F17.20&quot;,
    replacement = &quot;F17.2&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;F430&quot;,
    replacement = &quot;F43.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;G47.00&quot;,
    replacement = &quot;G47.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H01.00|H01.09.&quot;,
    replacement = &quot;H01.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H01.09.&quot;,
    replacement = &quot;H01.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H029&quot;,
    replacement = &quot;H02.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H60.90&quot;,
    replacement = &quot;H60.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H60.90&quot;,
    replacement = &quot;H60.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H91.8X9.&quot;,
    replacement = &quot;H91.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H91.90&quot;,
    replacement = &quot;H91.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;H93.299.|H93.29&quot;,
    replacement = &quot;H01&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;I70.20&quot;,
    replacement = &quot;I70.2&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;I82.409.|I82.40|I82.4&quot;,
    replacement = &quot;I82&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;J30.9&quot;,
    replacement = &quot;J30.2&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;J45.998.|J45.909.|J45.901.|J45.99|J45.90&quot;,
    replacement = &quot;J45.9&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;J98.457.6&quot;,
    replacement = &quot;J98.4&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;K05.30-31&quot;,
    replacement = &quot;K05.3&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;K29.70&quot;,
    replacement = &quot;K29.7&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;K59.00&quot;,
    replacement = &quot;K59.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;L08.89&quot;,
    replacement = &quot;L08.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M06.99|M06.90&quot;,
    replacement = &quot;M06.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M10.99&quot;,
    replacement = &quot;M10.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M10.99&quot;,
    replacement = &quot;M10.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M13.90|M13.94|M13.96|M13.97|M13.99&quot;,
    replacement = &quot;M13.9&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M19.07&quot;,
    replacement = &quot;M19.0&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M19.90|M19.91|M19.94|M19.97|M19.99&quot;,
    replacement = &quot;M19.9&quot;
    )
  )



disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M25.50|M25.51|M25.55|M25.569.|M25.56|M25.571.|M25.57&quot;,
    replacement = &quot;M25.5&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M25.76|M25.77&quot;,
    replacement = &quot;M25.7&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M43.16&quot;,
    replacement = &quot;M43.1&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M47.80|M47.82|M47.86&quot;,
    replacement = &quot;M47.8&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M47.92|M47.96&quot;,
    replacement = &quot;M47.9&quot;
    )
  )

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M48.02|M48.06&quot;,
    replacement = &quot;M48.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M54.22&quot;,
    replacement = &quot;M54.2&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M54.30|M54.39&quot;,
    replacement = &quot;M54.3&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M54.56|M54.57|M54.59&quot;,
    replacement = &quot;M54.5&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M54.99&quot;,
    replacement = &quot;M54.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M65.34&quot;,
    replacement = &quot;M65.3&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M65.96&quot;,
    replacement = &quot;M65.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M72.04&quot;,
    replacement = &quot;M72.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M79.09&quot;,
    replacement = &quot;M79.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M79.65|M79.66|M79.67&quot;,
    replacement = &quot;M79.6&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M79.79&quot;,
    replacement = &quot;M79.7&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M79.86&quot;,
    replacement = &quot;M79.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;M81.99&quot;,
    replacement = &quot;M81.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;N390&quot;,
    replacement = &quot;N39.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;N50.89&quot;,
    replacement = &quot;N50.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;N52&quot;,
    replacement = &quot;F52&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;N52.9&quot;,
    replacement = &quot;F52.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;N814&quot;,
    replacement = &quot;N81.4&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;P29.12&quot;,
    replacement = &quot;P29.1&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R06.00|R06.09&quot;,
    replacement = &quot;R06.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R06.83&quot;,
    replacement = &quot;R06.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R07.89&quot;,
    replacement = &quot;R07.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R07.8|R07.9&quot;,
    replacement = &quot;R07&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R09.89&quot;,
    replacement = &quot;R09.8&quot;
    )
    ) 


disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R10.30&quot;,
    replacement = &quot;R10.3&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R10.8|R10.9&quot;,
    replacement = &quot;R10&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R11.0&quot;,
    replacement = &quot;R11&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R13.10&quot;,
    replacement = &quot;R13&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R13.1&quot;,
    replacement = &quot;R13&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R14.0&quot;,
    replacement = &quot;R14&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R19.7&quot;,
    replacement = &quot;R19&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R29.898.|R29.89&quot;,
    replacement = &quot;R29.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R31.29&quot;,
    replacement = &quot;R31.2&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R31.2|R31.9&quot;,
    replacement = &quot;R31&quot;
    )
    ) 


disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R53.82|R53.83|R5382&quot;,
    replacement = &quot;R53.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R53.8&quot;,
    replacement = &quot;R53&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R56.9&quot;,
    replacement = &quot;R56&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
  mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R73.09|R73.02&quot;,
    replacement = &quot;R73.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R80.9&quot;,
    replacement = &quot;R80&quot;
    )
    ) 
 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R82.998.|R82.99&quot;,
    replacement = &quot;R82.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R871&quot;,
    replacement = &quot;R87.1&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R87.6&quot;,
    replacement = &quot;R87&quot;
    )
    ) 


disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R87.61&quot;,
    replacement = &quot;R87.6&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;T50.905.&quot;,
    replacement = &quot;T50.9&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;T78.40&quot;,
    replacement = &quot;T78.4&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;T780&quot;,
    replacement = &quot;T78.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;T81.149.88&quot;,
    replacement = &quot;T81.1&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;T81.815.013.&quot;,
    replacement = &quot;T81.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;T84.84&quot;,
    replacement = &quot;T84.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;T887&quot;,
    replacement = &quot;T88.7&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;Z86.79&quot;,
    replacement = &quot;Z86.7&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;Z87.09&quot;,
    replacement = &quot;Z87.0&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;Z87.39&quot;,
    replacement = &quot;Z87.3&quot;
    )
    ) 


disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;Z87.42&quot;,
    replacement = &quot;Z87.4&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;Z87.891.&quot;,
    replacement = &quot;Z87.8&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;W44.9&quot;,
    replacement = &quot;W44&quot;
    )
    ) 

disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;U80&quot;,
    replacement = &quot;U82&quot;
    )
    ) 

disease_mapping =
  disease_mapping |&gt;
  distinct()</code></pre>
</div>
</div>
<div id="add-icd10-descriptions" class="section level3">
<h3>Add ICD10 descriptions</h3>
<div id="other-icd10-descriptions-to-add" class="section level4">
<h4>Other ICD10 descriptions to add</h4>
<pre class="r"><code>other_icd10_desc = 
  data.frame(
    icd10_code = c(&quot;A09.9&quot;, 
                  &quot;A41&quot;,
                  &quot;B95&quot;,
                  &quot;B96&quot;,
                  &quot;B97&quot;,
                  &quot;B98&quot;,
                  &quot;C44&quot;,
                  &quot;C79&quot;,
                  &quot;C80.0&quot;,
                  &quot;C80.9&quot;,
                  &quot;C90&quot;,
                  &quot;D07&quot;,
                  &quot;D35&quot;,
                  &quot;D37&quot;,
                  &quot;D47&quot;,
                  &quot;D48&quot;,
                  &quot;E03&quot;,
                  &quot;E04&quot;,
                  &quot;E21&quot;,
                  &quot;E23&quot;,
                  &quot;E27&quot;,
                  &quot;E61&quot;,
                  &quot;E80&quot;,
                  &quot;E87&quot;,
                  &quot;E89&quot;,
                  &quot;F05&quot;,
                  &quot;F17&quot;,
                  &quot;F41&quot;,
                  &quot;F43.0&quot;,
                  &quot;F53&quot;,
                  &quot;G31&quot;,
                  &quot;G45&quot;,
                  &quot;G46&quot;,
                  &quot;G62&quot;,
                  &quot;G93&quot;,
                  &quot;G99&quot;,
                  &quot;H47&quot;,
                  &quot;H50&quot;,
                  &quot;H57&quot;,
                  &quot;H83&quot;,
                  &quot;H91&quot;,
                  &quot;H92&quot;,
                  &quot;H93&quot;,
                  &quot;I08&quot;,
                  &quot;I27&quot;,
                  &quot;I27.2&quot;,
                  &quot;I44&quot;,
                  &quot;I45&quot;,
                  &quot;I48.0&quot;,
                  &quot;I49&quot;,
                  &quot;I51&quot;,
                  &quot;I62&quot;,
                  &quot;I71&quot;,
                  &quot;J15&quot;,
                  &quot;J16&quot;,
                  &quot;J34&quot;,
                  &quot;J38&quot;,
                  &quot;J69&quot;,
                  &quot;J84&quot;,
                  &quot;J95&quot;,
                  &quot;J96&quot;,
                  &quot;K05&quot;,
                  &quot;K07&quot;,
                  &quot;K08&quot;,
                  &quot;K09&quot;,
                  &quot;K10&quot;,
                  &quot;K12&quot;,
                  &quot;K13&quot;,
                  &quot;K22&quot;,
                  &quot;K22.7&quot;,
                  &quot;K52&quot;,
                  &quot;K56&quot;,
                  &quot;K64&quot;,
                  &quot;K64.0&quot;,
                  &quot;K64.9&quot;,
                  &quot;K74&quot;,
                  &quot;K75&quot;,
                  &quot;K85.9&quot;,
                  &quot;K86&quot;,
                  &quot;K91&quot;,
                  &quot;K91.8&quot;,
                  &quot;K92&quot;,
                  &quot;L02&quot;,
                  &quot;L13&quot;,
                  &quot;L30&quot;,
                  &quot;L65&quot;,
                  &quot;L73&quot;,
                  &quot;L85&quot;,
                  &quot;L89.1&quot;,
                  &quot;L98&quot;,
                  &quot;M05&quot;,
                  &quot;M06&quot;,
                  &quot;M07&quot;,
                  &quot;M11&quot;,
                  &quot;M13&quot;,
                  &quot;M18&quot;,
                  &quot;M19&quot;,
                  &quot;M24&quot;,
                  &quot;M25&quot;,
                  &quot;M31&quot;,
                  &quot;M35&quot;,
                  &quot;M43&quot;,
                  &quot;M48&quot;,
                  &quot;M53&quot;,
                  &quot;M62&quot;,
                  &quot;M66&quot;,
                  &quot;M67&quot;,
                  &quot;M71&quot;,
                  &quot;M77&quot;,
                  &quot;M79&quot;,
                  &quot;M79.7&quot;,
                  &quot;M80&quot;,
                  &quot;M81&quot;,
                  &quot;M85&quot;,
                  &quot;M89&quot;,
                  &quot;M96&quot;,
                  &quot;N02&quot;,
                  &quot;N18.3&quot;,
                  &quot;N18.4&quot;,
                  &quot;N28&quot;,
                  &quot;N39&quot;,
                  &quot;N48&quot;,
                  &quot;N73&quot;,
                  &quot;N76&quot;,
                  &quot;N88&quot;,
                  &quot;N89&quot;,
                  &quot;N91&quot;,
                  &quot;N92&quot;,
                  &quot;N93&quot;,
                  &quot;N94&quot;,
                  &quot;N99&quot;,
                  &quot;O14&quot;,
                  &quot;O04&quot;,
                  &quot;O26&quot;,
                  &quot;O32&quot;,
                  &quot;O34&quot;,
                  &quot;O36&quot;,
                  &quot;O68&quot;,
                  &quot;O75&quot;,
                  &quot;O99&quot;,
                  &quot;R03&quot;,
                  &quot;R07&quot;,
                  &quot;R09&quot;,
                  &quot;R10&quot;,
                  &quot;R19&quot;,
                  &quot;R29&quot;,
                  &quot;R29.6&quot;,
                  &quot;R39&quot;,
                  &quot;R40&quot;,
                  &quot;R47&quot;,
                  &quot;R57&quot;,
                  &quot;T95.8&quot;,
                  &quot;W44&quot;,
                  &quot;Y95&quot;,
                  &quot;Z86.3&quot;,
                  &quot;Z87.3&quot;,
                  &quot;Z87.4&quot;,
                  &quot;Z87.7&quot;,
                  &quot;Z87.8&quot;,
                  &quot;T85.8&quot;,
                  &quot;Z91.0&quot;,
                  &quot;Z88.9&quot;,
                  &quot;Z88.8&quot;,
                  &quot;Z88&quot;,
                  &quot;Z92.6&quot;,
                  &quot;N90&quot;,
                  &quot;U82&quot;
                  ),
    icd10_description = c(&quot;Gastroenteritis and colitis of unspecified origin&quot;,
                         &quot;Other sepsis&quot;,
                         &quot;Streptococcus and staphylococcus as the cause of diseases classified to other chapters&quot;,
                         &quot;Other specified bacterial agents as the cause of diseases classified to other chapters&quot;,
                         &quot;Viral agents as the cause of diseases classified to other chapters&quot;,
                         &quot;Other specified infectious agents as the cause of diseases classified to other chapters&quot;,
                         &quot;Other malignant neoplasms of skin&quot;,
                         &quot;Secondary malignant neoplasm of other and unspecified sites&quot;,
                         &quot;Malignant neoplasm, primary site unknown, so stated&quot;,
                         &quot;Malignant neoplasm, primary site unspecified&quot;,
                         &quot;Multiple myeloma and malignant plasma cell neoplasms&quot;,
                         &quot;Carcinoma in situ of other and unspecified genital organs&quot;,
                         &quot;Benign neoplasm of other and unspecified endocrine glands&quot;,
                         &quot;Neoplasm of uncertain or unknown behaviour of oral cavity and digestive organs&quot;,
                         &quot;Other neoplasms of uncertain or unknown behaviour of lymphoid, haematopoietic and related tissue&quot;,
                         &quot;Neoplasm of uncertain or unknown behaviour of other and unspecified sites&quot;,
                         &quot;Other hypothyroidism&quot;,
                         &quot;Other nontoxic goitre&quot;,
                         &quot;Hyperparathyroidism and other disorders of parathyroid gland&quot;,
                         &quot;Hypofunction and other disorders of pituitary gland&quot;,
                         &quot;Other disorders of adrenal gland&quot;,
                         &quot;Deficiency of other nutrient elements&quot;,
                         &quot;Disorders of porphyrin and bilirubin metabolism&quot;,
                         &quot;Other disorders of fluid, electrolyte and acid-base balance&quot;,
                         &quot;Postprocedural endocrine and metabolic disorders, not elsewhere classified&quot;,
                         &quot;Delirium, not induced by alcohol and other psychoactive substances&quot;,
                         &quot;Mental and behavioural disorders due to use of tobacco&quot;,
                         &quot;Other anxiety disorders&quot;,
                         &quot;Acute stress reaction&quot;,
                         &quot;Mental and behavioural disorders associated with the puerperium, not elsewhere classified&quot;,
                         &quot;Other degenerative diseases of nervous system, not elsewhere classified&quot;,
                         &quot;Transient cerebral ischaemic attacks and related syndromes&quot;,
                         &quot;Vascular syndromes of brain in cerebrovascular diseases&quot;,
                         &quot;Other polyneuropathies&quot;,
                         &quot;Other disorders of brain&quot;,
                         &quot;Other disorders of nervous system in diseases classified elsewhere&quot;,
                         &quot;Other disorders of optic [2nd] nerve and visual pathways&quot;,
                         &quot;Other strabismus&quot;,
                         &quot;Other disorders of eye and adnexa&quot;,
                         &quot;Other diseases of inner ear&quot;,
                         &quot;Other hearing loss&quot;,
                         &quot;Otalgia and effusion of ear&quot;,
                         &quot;Other disorders of ear, not elsewhere classified&quot;,
                         &quot;Multiple valve diseases&quot;,
                         &quot;Other pulmonary heart diseases&quot;,
                         &quot;Other secondary pulmonary hypertension&quot;,
                         &quot;Atrioventricular and left bundle-branch block&quot;,
                         &quot;Other conduction disorders&quot;,
                         &quot;Paroxysmal atrial fibrillation&quot;,
                         &quot;Ventricular fibrillation and flutter&quot;,
                         &quot;Complications and ill-defined descriptions of heart disease&quot;,
                         &quot;Other nontraumatic intracranial haemorrhage&quot;,
                         &quot;Aortic aneurysm and dissection&quot;,
                         &quot;Bacterial pneumonia, not elsewhere classified&quot;,
                         &quot;Pneumonia due to other infectious organisms, not elsewhere classified&quot;,
                         &quot;Other disorders of nose and nasal sinuses&quot;,
                         &quot;Diseases of vocal cords and larynx, not elsewhere classified&quot;,
                         &quot;Pneumonitis due to solids and liquids&quot;,
                         &quot;Other interstitial pulmonary diseases&quot;,
                         &quot;Postprocedural respiratory disorders, not elsewhere classified&quot;,
                         &quot;Respiratory failure, not elsewhere classified&quot;,
                         &quot;Gingivitis and periodontal diseases&quot;,
                         &quot;Dentofacial anomalies [including malocclusion]&quot;,
                         &quot;Other disorders of teeth and supporting structures&quot;,
                         &quot;Cysts of oral region, not elsewhere classified&quot;,
                         &quot;Other diseases of jaws&quot;,
                         &quot;Stomatitis and related lesions&quot;,
                         &quot;Other diseases of lip and oral mucosa&quot;,
                         &quot;Other diseases of oesophagus&quot;,
                         &quot;Barrett oesophagus&quot;,
                         &quot;Other noninfective gastroenteritis and colitis&quot;,
                         &quot;Paralytic ileus and intestinal obstruction without hernia&quot;,
                         &quot;Haemorrhoids and perianal venous thrombosis&quot;,
                         &quot;First degree haemorrhoids&quot;,
                         &quot;Haemorrhoids, unspecified&quot;,
                         &quot;Fibrosis and cirrhosis of liver&quot;,
                         &quot;Other inflammatory liver diseases&quot;,
                         &quot;Acute pancreatitis, unspecified&quot;,
                         &quot;Other diseases of pancreas&quot;,
                         &quot;Postprocedural disorders of digestive system, not elsewhere classified&quot;,
                         &quot;Other postprocedural disorders of digestive system, not elsewhere classified&quot;,
                         &quot;Other diseases of digestive system&quot;,
                         &quot;Cutaneous abscess, furuncle and carbuncle&quot;,
                         &quot;Other bullous disorders&quot;,
                         &quot;Other dermatitis&quot;,
                         &quot;Other nonscarring hair loss&quot;,
                         &quot;Other follicular disorders&quot;,
                         &quot;Other epidermal thickening&quot;,
                         &quot;Stage II decubitus ulcer&quot;,
                         &quot;Other disorders of skin and subcutaneous tissue, not elsewhere classified&quot;,
                         &quot;Seropositive rheumatoid arthritis&quot;,
                         &quot;Other rheumatoid arthritis&quot;,
                         &quot;Psoriatic and enteropathic arthropathies&quot;,
                         &quot;Other crystal arthropathies&quot;,
                         &quot;Other arthritis&quot;,
                         &quot;Arthrosis of first carpometacarpal joint&quot;,
                         &quot;Other arthrosis&quot;,
                         &quot;Other specific joint derangements&quot;,
                         &quot;Other joint disorders, not elsewhere classified&quot;,
                         &quot;Other necrotizing vasculopathies&quot;,
                         &quot;Other systemic involvement of connective tissue&quot;,
                         &quot;Other deforming dorsopathies&quot;,
                         &quot;Other spondylopathies&quot;,
                         &quot;Other dorsopathies, not elsewhere classified&quot;,
                         &quot;Other disorders of muscle&quot;,
                         &quot;Spontaneous rupture of synovium and tendon&quot;,
                         &quot;Other disorders of synovium and tendon&quot;,
                         &quot;Other bursopathies&quot;,
                         &quot;Other enthesopathies&quot;,
                         &quot;Other soft tissue disorders, not elsewhere classified&quot;,
                         &quot;Fibromyalgia&quot;,
                         &quot;Osteoporosis with pathological fracture&quot;,
                         &quot;Osteoporosis without pathological fracture&quot;,
                         &quot;Other disorders of bone density and structure&quot;,
                         &quot;Other disorders of bone&quot;,
                         &quot;Postprocedural musculoskeletal disorders, not elsewhere classified&quot;,
                         &quot;Recurrent and persistent haematuria&quot;,
                         &quot;Chronic kidney disease, stage 3&quot;,
                         &quot;Chronic kidney disease, stage 4&quot;,
                         &quot;Other disorders of kidney and ureter, not elsewhere classified&quot;,
                         &quot;Other disorders of urinary system&quot;,
                         &quot;Other disorders of penis&quot;,
                         &quot;Other female pelvic inflammatory diseases&quot;,
                         &quot;Other inflammation of vagina and vulva&quot;,
                         &quot;Other noninflammatory disorders of cervix uteri&quot;,
                         &quot;Other noninflammatory disorders of vagina&quot;,
                         &quot;Other noninflammatory disorders of vulva and perineum&quot;,
                         &quot;Excessive, frequent and irregular menstruation&quot;,
                         &quot;Other abnormal uterine and vaginal bleeding&quot;,
                         &quot;Pain and other conditions associated with female genital organs and menstrual cycle&quot;,
                         &quot;Postprocedural disorders of genitourinary system, not elsewhere classified&quot;,
                         &quot;Pre-eclampsia&quot;,
                         &quot;Medical abortion&quot;,
                         &quot;Maternal care for other conditions predominantly related to pregnancy&quot;,
                         &quot;Maternal care for known or suspected malpresentation of fetus&quot;,
                         &quot;Maternal care for known or suspected abnormality of pelvic organs&quot;,
                         &quot;Maternal care for other known or suspected fetal problems&quot;,
                         &quot;Labour and delivery complicated by fetal stress [distress]&quot;,
                         &quot;Other complications of labour and delivery, not elsewhere classified&quot;,
                         &quot;Other maternal diseases classifiable elsewhere but complicating pregnancy, childbirth and the puerperium&quot;,
                         &quot;Abnormal blood-pressure reading, without diagnosis&quot;,
                         &quot;Pain in throat and chest&quot;,
                         &quot;Other symptoms and signs involving the circulatory and respiratory systems&quot;,
                         &quot;Abdominal and pelvic pain&quot;,
                         &quot;Other symptoms and signs involving the digestive system and abdomen&quot;,
                         &quot;Other symptoms and signs involving the nervous and musculoskeletal systems&quot;,
                         &quot;Tendency to fall, not elsewhere classified&quot;,
                         &quot;Other symptoms and signs involving the urinary system&quot;,
                         &quot;Somnolence, stupor and coma&quot;,
                         &quot;Speech disturbances, not elsewhere classified&quot;,
                         &quot;Shock, not elsewhere classified&quot;,
                         &quot;Other complications of internal prosthetic devices, implants and grafts, not elsewhere classified&quot;,
                         &quot;Foreign body entering into or through eye or natural orifice&quot;,
                         &quot;Nosocomial condition&quot;,
                         &quot;Personal history of endocrine, nutritional and metabolic diseases&quot;,
                         &quot;Personal history of diseases of the musculoskeletal system and connective tissue&quot;,
                         &quot;Personal history of diseases of the genitourinary system&quot;,
                         &quot;Personal history of congenital malformations, deformations and chromosomal abnormalities&quot;,
                         &quot;Personal history of other specified conditions&quot;,
                         &quot;Other complications of internal prosthetic devices, implants and grafts, not elsewhere classified&quot;,
                         &quot;Personal history of allergy, other than to drugs and biological substances&quot;,
                         &quot;Personal history of allergy to unspecified drugs, medicaments and biological substances&quot;,
                         &quot;Personal history of allergy to other drugs, medicaments and biological substances
&quot;,
&quot;Personal history of allergy to drugs, medicaments and biological substances&quot;,
&quot;Personal history of chemotherapy for neoplastic disease&quot;,
&quot;Other noninflammatory disorders of vulva and perineum&quot;,
&quot;Resistance to betalactam antibiotics&quot;)
  )</code></pre>
<pre class="r"><code>manual_icd10_map &lt;-
  readxl::read_xlsx(here::here(&quot;data/icd/manual_disease_icd10_mappings.xlsx&quot;))

icd10_descriptions =
  phecodes |&gt;
  select(icd10_code = ICD10, 
         icd10_description = ICD_DESCRIPTION
         ) |&gt;
  distinct()

# Expand multiple ICD codes into rows
to_add_expanded &lt;- manual_icd10_map |&gt;
  mutate(icd10_code = str_split(icd10_code, &quot;,\\s*&quot;)) |&gt;
  tidyr::unnest(icd10_code)

icd10_descriptions = 
  bind_rows(
    icd10_descriptions,
    to_add_expanded |&gt;
      select(icd10_code, 
             icd10_description = icd10_desc),
    other_icd10_desc
  ) 

icd10_descriptions = icd10_descriptions |&gt; distinct()

icd10_descriptions = 
  icd10_descriptions |&gt;
  group_by(icd10_code) |&gt;
  summarise(icd10_description = 
            str_flatten(unique(icd10_description), collapse = &quot;; &quot;, na.rm = T), 
            .groups = &quot;drop&quot;
            )

disease_mapping =
  left_join(disease_mapping,
            icd10_descriptions,
            by = &quot;icd10_code&quot;,
            relationship = &quot;many-to-one&quot;,
            na_matches = &quot;never&quot;
            ) 


disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_code = str_replace_all(
    icd10_code,
    pattern = &quot;R871&quot;,
    replacement = &quot;R87&quot;
    )
    ) 


disease_mapping = 
  disease_mapping |&gt; 
    mutate(icd10_description = 
    ifelse(icd10_code == &quot;R87&quot;,
           &quot;Abnormal findings in specimens from female genital organs&quot;,
           icd10_description)
    ) </code></pre>
</div>
</div>
<div id="missing-descriptions-for-icd10-codes" class="section level3">
<h3>Missing descriptions for ICD10 codes</h3>
<pre class="r"><code># check if any ICD10 codes are missing descriptions
disease_mapping |&gt; 
  filter(is.na(icd10_description)) |&gt;
  head()</code></pre>
<pre><code># A tibble: 6 × 9
  `DISEASE/TRAIT`         collected_all_diseas…¹ PUBMED_ID  YEAR STUDY_ACCESSION
  &lt;chr&gt;                   &lt;chr&gt;                      &lt;int&gt; &lt;int&gt; &lt;chr&gt;          
1 smoking interaction in… lung cancer             29059373  2017 GCST005910     
2 smoking interaction in… lung cancer             29059373  2017 GCST005909     
3 smoking interaction in… lung cancer             29059373  2017 GCST005911     
4 breast cancer (estroge… breast cancer           29058716  2017 GCST005076     
5 breast cancer (estroge… breast cancer           29058716  2017 GCST005076     
6 breast cancer           breast cancer           29058716  2017 GCST005077     
# ℹ abbreviated name: ¹​collected_all_disease_terms
# ℹ 4 more variables: icd10_code &lt;chr&gt;, icd10_code_origin &lt;chr&gt;, phecode &lt;dbl&gt;,
#   icd10_description &lt;chr&gt;</code></pre>
</div>
<div id="noticing-some-mistakes-in-mapped-trait-by-icd-10-codes"
class="section level3">
<h3>Noticing some mistakes in MAPPED trait by icd 10 codes</h3>
<pre class="r"><code>gwas_study_info |&gt; 
  filter(grepl(&quot;ICD10 F05&quot;, 
               `DISEASE/TRAIT`)) |&gt; 
  select(`DISEASE/TRAIT`, MAPPED_TRAIT, collected_all_disease_terms, PUBMED_ID)</code></pre>
<pre><code>                                                                  DISEASE/TRAIT
                                                                         &lt;char&gt;
1: ICD10 F05: Delirium due to known physiological condition (Gene-based burden)
2:                       ICD10 F05.9: Delirium, unspecified (Gene-based burden)
3:                                           ICD10 F05.9: Delirium, unspecified
4:                     ICD10 F05: Delirium due to known physiological condition
                  MAPPED_TRAIT collected_all_disease_terms PUBMED_ID
                        &lt;char&gt;                      &lt;char&gt;     &lt;int&gt;
1: alcohol withdrawal delirium   alcohol-related disorders  34662886
2:                    delirium                    delirium  34662886
3:                    delirium                    delirium  34662886
4: alcohol withdrawal delirium   alcohol-related disorders  34662886</code></pre>
<pre class="r"><code>disease_mapping |&gt; 
  filter(icd10_code == &quot;F05&quot;)</code></pre>
<pre><code># A tibble: 3 × 9
  `DISEASE/TRAIT`         collected_all_diseas…¹ PUBMED_ID  YEAR STUDY_ACCESSION
  &lt;chr&gt;                   &lt;chr&gt;                      &lt;int&gt; &lt;int&gt; &lt;chr&gt;          
1 behavioral disturbance… atypical behavior       25897833  2015 GCST002863     
2 icd10 f05: delirium du… alcohol-related disor…  34662886  2021 GCST90083772   
3 icd10 f05: delirium du… alcohol-related disor…  34662886  2021 GCST90079786   
# ℹ abbreviated name: ¹​collected_all_disease_terms
# ℹ 4 more variables: icd10_code &lt;chr&gt;, icd10_code_origin &lt;chr&gt;, phecode &lt;dbl&gt;,
#   icd10_description &lt;chr&gt;</code></pre>
<pre class="r"><code># F05 refers to delirium not induced by alcohol and other psychoactive substances
# yet, the mapped trait is alcohol withdrawal delirium
# replace with delirium
disease_mapping = 
disease_mapping |&gt;
  mutate(collected_all_disease_terms = 
         ifelse(icd10_code == &quot;F05&quot;,
                &quot;delirium&quot;,
                collected_all_disease_terms)
         )

gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = 
         ifelse(grepl(&quot;ICD10 F05&quot;, `DISEASE/TRAIT`),
                str_replace_all(collected_all_disease_terms,
                                pattern = &quot;alcohol-related disorders&quot;,
                                replacement = &quot;delirium&quot;),
                collected_all_disease_terms)
         )</code></pre>
</div>
<div id="save-disease-map" class="section level3">
<h3>Save disease map</h3>
<pre class="r"><code>disease_mapping = 
  disease_mapping |&gt; 
  arrange(YEAR,
          PUBMED_ID,
          STUDY_ACCESSION,
          collected_all_disease_terms, 
          icd10_code)

fwrite(disease_mapping,
       here::here(&quot;output/icd_map/gwas_disease_to_icd10_mapping.csv&quot;)
       )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] jsonlite_2.0.0    httr_1.4.7        data.table_1.17.8 stringr_1.5.2    
[5] dplyr_1.1.4       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] bit_4.6.0         compiler_4.3.1    renv_1.0.3        promises_1.3.3   
 [5] tidyselect_1.2.1  Rcpp_1.1.0        git2r_0.36.2      tidyr_1.3.1      
 [9] callr_3.7.6       later_1.4.4       jquerylib_0.1.4   readxl_1.4.5     
[13] yaml_2.3.10       fastmap_1.2.0     here_1.0.1        R6_2.6.1         
[17] generics_0.1.4    knitr_1.50        tibble_3.3.0      rprojroot_2.1.0  
[21] bslib_0.9.0       pillar_1.11.1     rlang_1.1.6       utf8_1.2.6       
[25] cachem_1.1.0      stringi_1.8.7     httpuv_1.6.16     xfun_0.53        
[29] getPass_0.2-4     fs_1.6.6          sass_0.4.10       bit64_4.6.0-1    
[33] cli_3.6.5         withr_3.0.2       magrittr_2.0.4    ps_1.9.1         
[37] digest_0.6.37     processx_3.8.6    rstudioapi_0.17.1 lifecycle_1.0.4  
[41] vctrs_0.6.5       evaluate_1.0.5    glue_1.8.0        cellranger_1.1.0 
[45] whisker_0.4.1     purrr_1.1.0       rmarkdown_2.30    tools_4.3.1      
[49] pkgconfig_2.0.3   htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
