<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-09-06" />

<title>Non-ontology based trait collapsing</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Non-ontology based trait collapsing</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-09-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreecc7742884dea0d9737324869302e0aa203976701targetblankcc77428a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/cc7742884dea0d9737324869302e0aa203976701" target="_blank">cc77428</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreecc7742884dea0d9737324869302e0aa203976701targetblankcc77428a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/cc7742884dea0d9737324869302e0aa203976701" target="_blank">cc77428</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/gwas_catalog/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l1.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l2.csv
    Ignored:    output/trait_ontology/
    Ignored:    renv/

Untracked files:
    Untracked:  analysis/disease_burden.qmd
    Untracked:  data/gbd/
    Untracked:  data/who/

Unstaged changes:
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_1_disease_group.Rmd
    Modified:   analysis/level_2_disease_group.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/non_ontology_trait_collapse.Rmd</code>) and HTML
(<code>docs/non_ontology_trait_collapse.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/e6995cd7dd1589aa2a4294df2d40d9a85489d8ae/docs/non_ontology_trait_collapse.html" target="_blank">e6995cd</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/7a2b2037dfbe5b1abe580d3356cd7b6e712e902e/analysis/non_ontology_trait_collapse.Rmd" target="_blank">7a2b203</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
workflowr::wflow_publish("analysis/non_ontology_trait_collapse.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/0a2d4efa93cba8863a056248a200adfa23672e46/docs/non_ontology_trait_collapse.html" target="_blank">0a2d4ef</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/6925922cd440fc8ee9acbcee88d45355f172a9fd/analysis/non_ontology_trait_collapse.Rmd" target="_blank">6925922</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
workflowr::wflow_publish("analysis/non_ontology_trait_collapse.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/e3badc38d9d520869484b10e9354ff7444dfa4bc/docs/non_ontology_trait_collapse.html" target="_blank">e3badc3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/d4dc82c98f40555ac6871ee2a11c6dc723f96298/analysis/non_ontology_trait_collapse.Rmd" target="_blank">d4dc82c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-08
</td>
<td>
Small updates to disease trait grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/5a1b60004317b11490bd80892b63dfc45a09d966/docs/non_ontology_trait_collapse.html" target="_blank">5a1b600</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/39d92e6491386c1da0626870f7f7047cf618c986/analysis/non_ontology_trait_collapse.Rmd" target="_blank">39d92e6</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-07
</td>
<td>
Minor trait harmonization update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/26e7c84c927837ef38e51b120ab4f1e61b671dcc/docs/non_ontology_trait_collapse.html" target="_blank">26e7c84</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/5510b11483bf5e0dc21756b0a18d48d999f1f3dd/analysis/non_ontology_trait_collapse.Rmd" target="_blank">5510b11</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-07
</td>
<td>
Fixing data writing typo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/d252706dda2ca7608bffbd90d1538f8b774bada0/docs/non_ontology_trait_collapse.html" target="_blank">d252706</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/cad3bf3bf2863acb9aada8cb4b4e88b03caeb243/analysis/non_ontology_trait_collapse.Rmd" target="_blank">cad3bf3</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-07
</td>
<td>
Separating disease grouping steps
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/dcfc8e1b58fba0dc1e1838dbd3d2dd95b864bbb2/docs/non_ontology_trait_collapse.html" target="_blank">dcfc8e1</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/dc84680a34d58fb82780a1bc9861ed1168d1ea9b/analysis/non_ontology_trait_collapse.Rmd" target="_blank">dc84680</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-06
</td>
<td>
Updating trait collapsing once again
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/358498872fd9a55d61282035d34d2db1c2333b29/docs/non_ontology_trait_collapse.html" target="_blank">3584988</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/82b6e8a5360462a192e567b50bedeb6d0fc19c65/analysis/non_ontology_trait_collapse.Rmd" target="_blank">82b6e8a</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-06
</td>
<td>
Update trait collapsing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/13c5442ffca2e64d1ac03dbb246fd2839bda1420/docs/non_ontology_trait_collapse.html" target="_blank">13c5442</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b209d1fa195046192e6da58eaaa7badc29f07bc8/analysis/non_ontology_trait_collapse.Rmd" target="_blank">b209d1f</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-05
</td>
<td>
Even more trait collapsing
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Set up</h1>
<pre class="r"><code>library(dplyr)
library(data.table)
library(ggplot2)
library(stringr)</code></pre>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_study_info_trait_ontology_info.csv&quot;))</code></pre>
</div>
<div id="initial-summary" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Initial summary</h1>
<div id="objectives-of-this-analysis" class="section level2"
number="2.1">
<h2><span class="header-section-number">2.1</span> Objectives of this
analysis:</h2>
<ul>
<li>Collapsing and standardizing disease trait terms in GWAS study
metadata without relying on ontology mappings.</li>
<li>Initial harmonization of disease trait terms (fixing spelling,
collapsing studies that study different aspects of the same
disease)</li>
</ul>
</div>
</div>
<div id="number-of-studies-per-trait---before-any-grouping"
class="section level1" number="3">
<h1><span class="header-section-number">3</span> Number of studies per
trait - before any grouping</h1>
<p>Let’s look at the number of studies associated with each unique
disease/trait term before any grouping or collapsing.</p>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))

head(n_studies_trait)</code></pre>
<pre><code># A tibble: 6 × 2
  all_disease_terms           n_studies
  &lt;chr&gt;                           &lt;int&gt;
1 , type 2 diabetes mellitus        125
2 , alzheimer disease                95
3 , major depressive disorder        85
4 , breast carcinoma                 84
5 , colorectal cancer                84
6 , schizophrenia                    83</code></pre>
<pre class="r"><code>dim(n_studies_trait)</code></pre>
<pre><code>[1] 3394    2</code></pre>
</div>
<div id="disease-trait-collapsing-and-cleaning" class="section level1"
number="4">
<h1><span class="header-section-number">4</span> Disease trait
collapsing and cleaning</h1>
<pre class="r"><code># Basic cleaning of disease terms
# Removes trailing commas, 
gwas_study_info$all_disease_terms = sub(&quot;,$&quot;, &quot;&quot;, gwas_study_info$all_disease_terms)

# trims whitespace
gwas_study_info$all_disease_terms = stringr::str_trim(gwas_study_info$all_disease_terms)

# remove apostrophes
gwas_study_info$all_disease_terms = stringr::str_remove_all(gwas_study_info$all_disease_terms, &quot;&#39;|’&quot;)

# initializes a new column (collected_all_disease_terms) to standardized traits
# collapse more traits 
gwas_study_info = gwas_study_info |&gt; 
  mutate(collected_all_disease_terms = all_disease_terms) </code></pre>
<div
id="where-a-study-investigates-a-given-aspect-of-of-disease-reduce-this-to-a-study-of-this-disease"
class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Where a study
investigates a given aspect of of disease, reduce this to a study of
this disease</h2>
<div
id="e.g-susceptibility-to-x-measurement-x-disorder-measurement-etc."
class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> e.g Susceptibility
to X measurement, X disorder measurement etc.</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_replace(collected_all_disease_terms, 
                    &quot;^susceptibility to (.*?) measurement$&quot;, &quot;\\1&quot;)

         ) |&gt;
    mutate(collected_all_disease_terms = 
        str_replace(collected_all_disease_terms, 
                    &quot;^susceptibility to (.*?) (.*?) measurement$&quot;, &quot;\\1&quot;)

         ) |&gt;
      mutate(collected_all_disease_terms = 
        str_replace(collected_all_disease_terms, 
                    &quot;^susceptibility to (.*?) (.*?) infection$&quot;, &quot;\\1&quot;)

         ) 


gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_replace(collected_all_disease_terms, 
                    &quot;susceptibility to viral and mycobacterial infections&quot;,
                    &quot;viral and mycobacterial infections&quot;)

         ) |&gt;
    mutate(collected_all_disease_terms = 
        str_replace(collected_all_disease_terms, 
                    &quot;susceptibility to strep throat&quot;,
                    &quot;strep throat&quot;)
         )




gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
          stringr::str_remove(collected_all_disease_terms, &quot; symptom severity measurement&quot;)
         ) |&gt;
  mutate(collected_all_disease_terms = 
          stringr::str_remove(collected_all_disease_terms, &quot; severity measurement| exacerbation measurement&quot;)
         ) 



gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\bdependence\\s+measurement\\b&quot;,
                           &quot;dependence&quot;)) |&gt; 
  mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\ballergy\\s+measurement\\b&quot;,
                           &quot;allergy&quot;)) |&gt; 
  mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\baddiction\\s+measurement\\b&quot;,
                           &quot;addiction&quot;)) |&gt; 
      mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\bsyndrome\\s+measurement\\b&quot;,
                           &quot;syndrome&quot;)) |&gt; 
      mutate(collected_all_disease_terms =
           str_replace_all(collected_all_disease_terms,
                           &quot;\\bdisorder\\s+measurement\\b&quot;,
                           &quot;disorder&quot;)) </code></pre>
</div>
<div id="e.g-age-of-onset-of" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> e.g age of onset
of</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_remove(collected_all_disease_terms,
                         pattern = &quot;age of onset of &quot;))


gwas_study_info = 
gwas_study_info |&gt; 
 mutate(collected_all_disease_terms  = 
          stringr::str_replace_all(collected_all_disease_terms ,
                                  pattern = &quot;age of onset of type 2 diabetes mellitus&quot;,
                                   &quot;type 2 diabetes mellitus&quot;
                          )  
        )</code></pre>
</div>
<div id="e.g.-family-history-of" class="section level3" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> e.g. Family history
of</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_remove(collected_all_disease_terms,
                         pattern = &quot;family history of &quot;))


gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;family history of alzheimer’s disease&quot;,
                          &quot;alzheimers disease&quot;
         ))</code></pre>
</div>
<div id="time-to-remission" class="section level3" number="4.1.4">
<h3><span class="header-section-number">4.1.4</span> Time to
remission</h3>
<div id="covid" class="section level4" number="4.1.4.1">
<h4><span class="header-section-number">4.1.4.1</span> COVID</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;time to remission of covid-19 symptoms&quot;,
                          &quot;covid-19&quot;
         ))</code></pre>
</div>
</div>
<div id="symptom-measurement" class="section level3" number="4.1.5">
<h3><span class="header-section-number">4.1.5</span> Symptom
measurement</h3>
<div id="adhd" class="section level4" number="4.1.5.1">
<h4><span class="header-section-number">4.1.5.1</span> ADHD</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
  mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;adhd symptom measurement&quot;,
                            &quot;adhd&quot;)
         )</code></pre>
</div>
<div id="agoraphobia" class="section level4" number="4.1.5.2">
<h4><span class="header-section-number">4.1.5.2</span> Agoraphobia</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;agoraphobia symptom measurement&quot;,
                          &quot;agoraphobia&quot;
         ))</code></pre>
</div>
<div id="alzheimers-disease---neuropathological-change"
class="section level4" number="4.1.5.3">
<h4><span class="header-section-number">4.1.5.3</span> Alzheimer’s
disease - neuropathological change</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;alzheimer disease neuropathological change&quot;,
                          &quot;alzheimers disease&quot;
         ))</code></pre>
</div>
<div id="asthma" class="section level4" number="4.1.5.4">
<h4><span class="header-section-number">4.1.5.4</span> Asthma</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;asthma exacerbation measurement&quot;,
                          &quot;asthma&quot;
         )) |&gt;
  mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;asthma symptoms measurement&quot;,
                            &quot;asthma&quot;
           )) </code></pre>
</div>
<div id="covid-symptoms" class="section level4" number="4.1.5.5">
<h4><span class="header-section-number">4.1.5.5</span> COVID
symptoms</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;covid-19 symptoms measurement&quot;,
                          &quot;covid-19&quot;)
  )</code></pre>
</div>
<div id="irritable-bowel-syndrome" class="section level4"
number="4.1.5.6">
<h4><span class="header-section-number">4.1.5.6</span> Irritable bowel
syndrome</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;irritable bowel syndrome symptom measurement&quot;,
                          &quot;irritable bowel syndrome&quot;
         ))</code></pre>
</div>
</div>
<div id="obsessive-compulsive-disorder" class="section level3"
number="4.1.6">
<h3><span class="header-section-number">4.1.6</span> Obsessive
compulsive disorder</h3>
<div id="parkinsons-disease" class="section level4" number="4.1.6.1">
<h4><span class="header-section-number">4.1.6.1</span> Parkinson’s
disease</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;parkinsons disease symptom measurement&quot;,
                          &quot;parkinsons disease&quot;
         ))</code></pre>
</div>
<div id="ptsd" class="section level4" number="4.1.6.2">
<h4><span class="header-section-number">4.1.6.2</span> PTSD</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;post-traumatic stress disorder symptom measurement&quot;,
                          &quot;post-traumatic stress disorder&quot;
         ))</code></pre>
</div>
<div id="respiratory-symptom-change" class="section level4"
number="4.1.6.3">
<h4><span class="header-section-number">4.1.6.3</span> Respiratory
symptom change</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;respiratory symptom change measurement&quot; ,
                          &quot;respiratory symptom measurement&quot; 
         ))</code></pre>
</div>
</div>
<div id="measurement" class="section level3" number="4.1.7">
<h3><span class="header-section-number">4.1.7</span> Measurement</h3>
<div id="anxiety-disorder" class="section level4" number="4.1.7.1">
<h4><span class="header-section-number">4.1.7.1</span> Anxiety
disorder</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;anxiety disorder measurement&quot;,
                          &quot;anxiety disorder&quot;
         ))</code></pre>
</div>
<div id="coronary-atherosclerosis" class="section level4"
number="4.1.7.2">
<h4><span class="header-section-number">4.1.7.2</span> Coronary
atherosclerosis</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;coronary atherosclerosis measurement&quot;,
                          &quot;coronary atherosclerosis&quot;)
  ) </code></pre>
</div>
</div>
<div id="insommnia" class="section level3" number="4.1.8">
<h3><span class="header-section-number">4.1.8</span> Insommnia</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;insomnia measurement&quot;,
                          &quot;insomnia&quot;
         ))</code></pre>
<div id="lewy-body-dementia" class="section level4" number="4.1.8.1">
<h4><span class="header-section-number">4.1.8.1</span> Lewy body
dementia</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;lewy body dementia measurement&quot;,
                          &quot;lewy body dementia&quot;
         ))</code></pre>
</div>
<div id="psychosis-predisposition" class="section level4"
number="4.1.8.2">
<h4><span class="header-section-number">4.1.8.2</span> Psychosis
predisposition</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;psychosis predisposition measurement&quot;,
                         &quot;psychosis predisposition&quot;)
        )</code></pre>
</div>
<div id="sleep-apnea-measurement" class="section level4"
number="4.1.8.3">
<h4><span class="header-section-number">4.1.8.3</span> Sleep apnea
measurement</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
        mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;sleep apnea measurement&quot;,
                          &quot;sleep apnea&quot;
         ))</code></pre>
</div>
<div id="susceptibility-to-infectious-disease" class="section level4"
number="4.1.8.4">
<h4><span class="header-section-number">4.1.8.4</span> Susceptibility to
infectious disease</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        str_replace(collected_all_disease_terms, 
                    &quot;susceptibility to infectious disease measurement&quot;,
                    &quot;susceptibility to infectious disease&quot;)
         )</code></pre>
</div>
</div>
<div id="substance-abuse-measurement" class="section level3"
number="4.1.9">
<h3><span class="header-section-number">4.1.9</span> Substance abuse
measurement</h3>
<div id="alcohol" class="section level4" number="4.1.9.1">
<h4><span class="header-section-number">4.1.9.1</span> Alcohol</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;alcohol dependence measurement&quot;,
                          &quot;alcohol dependence&quot;
         ))</code></pre>
</div>
<div id="cannibis" class="section level4" number="4.1.9.2">
<h4><span class="header-section-number">4.1.9.2</span> Cannibis</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;cannabis dependence measurement&quot;,
                          &quot;cannabis dependence&quot;
         ))</code></pre>
</div>
<div id="cocaine" class="section level4" number="4.1.9.3">
<h4><span class="header-section-number">4.1.9.3</span> Cocaine</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;cocaine use disorder|cocaine dependence&quot;,
                          &quot;cocaine-related disorders&quot;
         ))</code></pre>
</div>
<div id="nicotine" class="section level4" number="4.1.9.4">
<h4><span class="header-section-number">4.1.9.4</span> Nicotine</h4>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;nicotine dependence symptom count&quot;,
                          &quot;nicotine dependence&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;nicotine withdrawal symptom count|nicotine withdrawal measurement&quot;,
                          &quot;nicotine withdrawal&quot;
         ))</code></pre>
</div>
</div>
</div>
<div id="deal-with-basic-synonyms" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Deal with basic
synonyms</h2>
<div id="carcinoma-cancer" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Carcinoma =
Cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;\\bcarcinoma&quot;,
                            &quot;cancer&quot;)
         )

gwas_study_info = gwas_study_info |&gt;
      mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;triple-negative breast cancer&quot;,
                            &quot;breast cancer&quot;
           ))</code></pre>
</div>
<div id="coronary-artery-disease-coronary-atherosclerosis"
class="section level3" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Coronary artery
disease = coronary atherosclerosis</h3>
<p><a
href="https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001645"
class="uri">https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001645</a></p>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;coronary atherosclerosis&quot;,
                          &quot;coronary artery disease&quot;
         ))</code></pre>
</div>
<div id="head-and-neck-cancer-head-and-neck-malignant-neoplasia"
class="section level3" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Head and neck
cancer = head and neck malignant neoplasia</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;head and neck malignant neoplasia&quot;,
                          &quot;head and neck cancer&quot;
         ))</code></pre>
</div>
<div id="hepatitis-virus-infection-hepatitis-infection"
class="section level3" number="4.2.4">
<h3><span class="header-section-number">4.2.4</span> Hepatitis virus
infection = hepatitis infection</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;hepatitis a virus infection&quot;,
                          &quot;hepatitis a infection&quot;
         )) |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;hepatitis b virus infection&quot;,
                          &quot;hepatitis b infection&quot;
         )) |&gt;
        mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;hepatitis c virus infection&quot;,
                          &quot;hepatitis c infection&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;chronic hepatitis c virus infection&quot;,
                          &quot;chronic hepatitis, hepatitis c infection&quot;
         )) |&gt;
      mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;chronic hepatitis b virus infection&quot;,
                          &quot;chronic hepatitis, hepatitis b infection&quot;
         ))

gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;hepatitis virus-related liver cancer&quot;,
                          &quot;hepatitis, liver cancer&quot;
         ))</code></pre>
</div>
<div id="hiv" class="section level3" number="4.2.5">
<h3><span class="header-section-number">4.2.5</span> HIV</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;hiv-1 infection&quot;,
                          &quot;hiv infection&quot;
         ))</code></pre>
</div>
<div id="mumps-virus-infection-mumps" class="section level3"
number="4.2.6">
<h3><span class="header-section-number">4.2.6</span> Mumps virus
infection = mumps</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;mumps virus infectious disease&quot;,
                          &quot;mumps&quot;
         ))</code></pre>
</div>
<div id="renal-cancer-kidney-cancer" class="section level3"
number="4.2.7">
<h3><span class="header-section-number">4.2.7</span> Renal cancer =
kidney cancer</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;renal cancer&quot;,
                          &quot;kidney cancer&quot;
         ))</code></pre>
</div>
<div id="rubella-infect-rubella" class="section level3" number="4.2.8">
<h3><span class="header-section-number">4.2.8</span> Rubella infect =
rubella</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;rubella infection&quot;,
                          &quot;rubella&quot;
         ))</code></pre>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;multiple sclerosis symptom measurement&quot;,
                          &quot;multiple sclerosis&quot;
         ))  </code></pre>
</div>
<div id="autism-autism-spectrum-disorder-asperger-syndrome"
class="section level3" number="4.2.9">
<h3><span class="header-section-number">4.2.9</span> Autism = Autism
Spectrum Disorder = Asperger Syndrome</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;autism spectrum disorder&quot;,
                         &quot;autism&quot;)
        ) |&gt;
    mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;asperger syndrome&quot;,
                         &quot;autism&quot;)
        )</code></pre>
</div>
</div>
<div id="type-1-diabetes-type-1-diabetes-mellitus"
class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Type 1 diabetes =
type 1 diabetes mellitus</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;type 1 diabetes mellitus&quot;,
                         &quot;type 1 diabetes&quot;)
        ) |&gt;
    mutate(collected_all_disease_terms = 
        stringr::str_replace_all(collected_all_disease_terms,
                         &quot;type 1 diabetes&quot;,
                         &quot;type 1 diabetes mellitus&quot;)
        )</code></pre>
<div id="adhd-attention-deficit-hyperactivity-disorder"
class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> ADHD = attention
deficit hyperactivity disorder</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt; 
    mutate(collected_all_disease_terms = 
           stringr::str_replace_all(collected_all_disease_terms,
                            &quot;attention deficit hyperactivity disorder&quot;,
                            &quot;adhd&quot;
         ))</code></pre>
</div>
</div>
<div id="narcolepsy-narcolepsy-without-cataplexy" class="section level2"
number="4.4">
<h2><span class="header-section-number">4.4</span> Narcolepsy =
Narcolepsy without cataplexy</h2>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms = 
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;narcolepsy without cataplexy&quot;,
                          &quot;narcolepsy&quot;
         ))</code></pre>
</div>
<div id="typos" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Typos</h2>
<div id="alzheimer---alzheimers" class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Alzheimer -&gt;
alzheimers</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms =
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;alzheimer disease&quot;,
                          &quot;alzheimers disease&quot;
         ))</code></pre>
</div>
<div id="parkinson---parkinsons" class="section level3" number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> Parkinson -&gt;
parkinsons</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  mutate(collected_all_disease_terms =
         stringr::str_replace_all(collected_all_disease_terms,
                          &quot;parkinson disease&quot;,
                          &quot;parkinsons disease&quot;
         ))</code></pre>
</div>
</div>
</div>
<div id="putting-it-all-together" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Putting it all
together</h1>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  rowwise() |&gt;
  mutate(
    collected_all_disease_terms = paste0(
      unique(unlist(stringr::str_split(collected_all_disease_terms, &quot;, &quot;))),
      collapse = &quot;, &quot;
    )
  ) |&gt;
  ungroup()

           
gwas_study_info$collected_all_disease_terms = stringr::str_trim(gwas_study_info$collected_all_disease_terms)
gwas_study_info$collected_all_disease_terms = sub(&quot;,$&quot;, &quot;&quot;, gwas_study_info$collected_all_disease_terms)</code></pre>
<div
id="final-summary---number-of-studies-per-trait---after-harmonization"
class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Final summary -
number of studies per trait - after harmonization</h2>
<pre class="r"><code>n_studies_trait = gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(collected_all_disease_terms, PUBMED_ID) |&gt;
  dplyr::distinct() |&gt;
  dplyr::group_by(collected_all_disease_terms) |&gt;
  dplyr::summarise(n_studies = dplyr::n()) |&gt;
  dplyr::arrange(desc(n_studies))

head(n_studies_trait)</code></pre>
<pre><code># A tibble: 6 × 2
  collected_all_disease_terms n_studies
  &lt;chr&gt;                           &lt;int&gt;
1 , type 2 diabetes mellitus        126
2 , alzheimers disease              104
3 , breast cancer                    97
4 , colorectal cancer                90
5 , major depressive disorder        85
6 , schizophrenia                    85</code></pre>
<pre class="r"><code>dim(n_studies_trait)</code></pre>
<pre><code>[1] 3286    2</code></pre>
<pre class="r"><code>diseases &lt;- stringr::str_split(pattern = &quot;, &quot;, 
                               gwas_study_info$collected_all_disease_terms)  |&gt; 
            unlist() |&gt;
            stringr::str_trim()

length(unique(diseases))</code></pre>
<pre><code>[1] 2204</code></pre>
<div id="save-the-updated-gwas_study_info-with-harmonized-disease-terms"
class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Save the updated
gwas_study_info with harmonized disease terms</h3>
<pre class="r"><code>gwas_study_info &lt;- fwrite(gwas_study_info,
                          here::here(&quot;output/gwas_study_info_trait_ontology_info.csv&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] stringr_1.5.1     ggplot2_3.5.2     data.table_1.17.8 dplyr_1.1.4      
[5] workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       jsonlite_2.0.0     compiler_4.3.1     renv_1.0.3        
 [5] promises_1.3.3     tidyselect_1.2.1   Rcpp_1.1.0         git2r_0.36.2      
 [9] callr_3.7.6        later_1.4.2        jquerylib_0.1.4    scales_1.4.0      
[13] yaml_2.3.10        fastmap_1.2.0      here_1.0.1         R6_2.6.1          
[17] generics_0.1.4     knitr_1.50         tibble_3.3.0       rprojroot_2.1.0   
[21] RColorBrewer_1.1-3 bslib_0.9.0        pillar_1.11.0      rlang_1.1.6       
[25] utf8_1.2.6         cachem_1.1.0       stringi_1.8.7      httpuv_1.6.16     
[29] xfun_0.52          getPass_0.2-4      fs_1.6.6           sass_0.4.10       
[33] cli_3.6.5          withr_3.0.2        magrittr_2.0.3     ps_1.9.1          
[37] grid_4.3.1         digest_0.6.37      processx_3.8.6     rstudioapi_0.17.1 
[41] lifecycle_1.0.4    vctrs_0.6.5        evaluate_1.0.4     glue_1.8.0        
[45] farver_2.1.2       whisker_0.4.1      rmarkdown_2.29     httr_1.4.7        
[49] tools_4.3.1        pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
