<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Replicate observed ancestry biases figures in GWAS dataset</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Replicate observed ancestry biases figures
in GWAS dataset</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree10a35cc5b555c4f9759b44ea044d867e7c9ade61targetblank10a35cca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/10a35cc5b555c4f9759b44ea044d867e7c9ade61" target="_blank">10a35cc</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree10a35cc5b555c4f9759b44ea044d867e7c9ade61targetblank10a35cca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/10a35cc5b555c4f9759b44ea044d867e7c9ade61" target="_blank">10a35cc</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/cdc/
    Ignored:    data/gwas_catalog/

Untracked files:
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   README.md
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/replication_ancestry_bias.Rmd</code>) and HTML
(<code>docs/replication_ancestry_bias.html</code>) files. If youâ€™ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/10a35cc5b555c4f9759b44ea044d867e7c9ade61/analysis/replication_ancestry_bias.Rmd" target="_blank">10a35cc</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
<td>
Updating replication of ancestry biases in gwas
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-gwas-catalog-data" class="section level2">
<h2>Load GWAS catalog data</h2>
<pre class="r"><code>library(dplyr)
library(ggplot2)

gwas_study_info = data.table::fread(&quot;data/gwas_catalog/gwas-catalog-v1.0.3-studies-r2022-02-02.tsv&quot;, 
                                    sep = &quot;\t&quot;, 
                                    quote = &quot;&quot;)

gwas_ancest_info = data.table::fread(&quot;./data/gwas_catalog/gwas_catalog-ancestry_r2022-02-02.tsv&quot;, 
                                     sep = &quot;\t&quot;, 
                                     quote = &quot;&quot;)

gwas_study_info = gwas_study_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_ancest_info = gwas_ancest_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

# Set up custom theme for ggplots
custom_theme &lt;-
  list(
    theme_bw() +
      theme(
        panel.border = element_blank(),
        axis.line = element_line(),
        text = element_text(size = 16),
        legend.position = &quot;bottom&quot;,
        strip.background = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5)
      )
  )</code></pre>
</div>
<div id="plot-figure-martin-et-al.-2019-like" class="section level2">
<h2>Plot figure Martin et al.Â 2019 like</h2>
<div id="for-all-ancestries" class="section level3">
<h3>For all ancestries</h3>
<pre class="r"><code># code adapted from https://github.com/armartin/prs_disparities/blob/master/gwas_disparities_time.R

# https://github.com/armartin/prs_disparities/blob/master/gwas_sfs_pop.R
# gwas cat
# http://bioconductor.org/packages/release/bioc/html/gwascat.html


# following steps from https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-019-0379-x/MediaObjects/41588_2019_379_MOESM1_ESM.pdf


# Some number of individuals are missing
gwas_ancest_info |&gt; dplyr::filter(is.na(NUMBER_OF_INDIVDUALS) | NUMBER_OF_INDIVDUALS == 0) |&gt; nrow()</code></pre>
<pre><code>[1] 35</code></pre>
<pre class="r"><code>gwas_ancest_info |&gt; dplyr::filter(is.na(NUMBER_OF_INDIVDUALS) | NUMBER_OF_INDIVDUALS == 0) |&gt; head()</code></pre>
<pre><code>   STUDY_ACCESSION PUBMEDID     FIRST_AUTHOR       DATE
            &lt;char&gt;    &lt;int&gt;           &lt;char&gt;     &lt;IDat&gt;
1:      GCST002988 26089329         Carty CL 2015-06-18
2:      GCST004339 28298293             Hu Y 2017-03-15
3:      GCST004338 28298293             Hu Y 2017-03-15
4:      GCST004585 28630421             Li D 2017-06-19
5:      GCST002330 24406073      Galanter JM 2014-01-07
6:      GCST001788 23263445 Lawrance-Owen AJ 2012-12-22
                                                              INITIAL_SAMPLE_DESCRIPTION
                                                                                  &lt;char&gt;
1:            up to 1,365 African American cases, up to 13,154 African American controls
2:               3,521 Chinese ancestry individuals, European individuals (see Wu 2013).
3: 3,521 Chinese ancestry individuals, European ancestry individuals (see Wu et al 2013)
4:        692 European ancestry female cases and 3,570 European ancestry female controls
5:                             1,893 Latin American cases, 1,881 Latin American controls
6:                                                     979 European ancestry individuals
                      REPLICATION_SAMPLE_DESCRIPTION       STAGE
                                              &lt;char&gt;      &lt;char&gt;
1: European ancestry individuals (see Traylor, 2012) replication
2:                                              &lt;NA&gt;     initial
3:                                              &lt;NA&gt;     initial
4:      Cases and controls (see Boraska et al, 2014) replication
5: EVE Asthma Consortium, see Torgerson et al., 2011 replication
6:                               (see Medland, 2010) replication
   NUMBER_OF_INDIVDUALS       BROAD_ANCESTRAL_CATEGORY COUNTRY_OF_ORIGIN
                  &lt;int&gt;                         &lt;char&gt;            &lt;char&gt;
1:                   NA                       European                NR
2:                   NA                       European                NR
3:                   NA                       European                NR
4:                   NA                             NR                NR
5:                   NA NR, Hispanic or Latin American                NR
6:                   NA                             NR                NR
   COUNTRY_OF_RECRUITMENT            ADDITONAL_ANCESTRY_DESCRIPTION
                   &lt;char&gt;                                    &lt;char&gt;
1:          Australia, NR                                          
2:                     NR See Wu J. H., et al. 2013. PMID: 23362303
3:                     NR                     See Wu et al 23362303
4:                     NR                                          
5:                     NR                                          
6:                     NR                                          </code></pre>
<pre class="r"><code># Filter missing number of individuals
gwas_ancest_info = gwas_ancest_info |&gt;
  dplyr::filter(!is.na(NUMBER_OF_INDIVDUALS)) |&gt;
  dplyr::filter(NUMBER_OF_INDIVDUALS != 0)

# calculate cumulative number of individuals
gwas_ancest_info = gwas_ancest_info %&gt;%  
                   dplyr::arrange(DATE) |&gt;
                   mutate(cum_num = cumsum(as.numeric(NUMBER_OF_INDIVDUALS))
                          )


# plot cumulative numbers
gwas_ancest_info %&gt;%  
  # group_by(DATE) %&gt;% 
  # slice_max(NUMBER_OF_INDIVDUALS) %&gt;% 
  ggplot(aes(x=DATE,y=cum_num/1e6)) + 
 # geom_line() +
  geom_area() + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  custom_theme + 
  labs(x = &quot;Year&quot;, 
       y = &quot;Individuals in GWAS catalog (millons)&quot;)</code></pre>
<p><img src="figure/replication_ancestry_bias.Rmd/plot_like_martin_et_al_2019_not_grouped-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="group-ancestry-into-broader-categories" class="section level2">
<h2>Group ancestry into broader categories</h2>
</div>
<div id="dealing-with-multiple" class="section level2">
<h2>Dealing with multiple</h2>
<ol style="list-style-type: decimal">
<li>Where multiple, and can get ancestry info from initial sample
description - do that</li>
</ol>
<pre class="r"><code># there&#39;s some kinda gross things in here (e.g. &#39;whites&#39;, &#39;Caucasian non-Hispanic&#39;)
unique(gwas_ancest_info$ADDITONAL_ANCESTRY_DESCRIPTION)[1:20]</code></pre>
<pre><code> [1] &quot;&quot;                                  &quot;Korean&quot;                           
 [3] &quot;Latinos or Africans&quot;               &quot;Asian&quot;                            
 [5] &quot;European&quot;                          &quot;Malawian, Ugandan&quot;                
 [7] &quot;Black African, Ugandan&quot;            &quot;Italian&quot;                          
 [9] &quot;Japanese&quot;                          &quot;Han Chinese&quot;                      
[11] &quot;Polish&quot;                            &quot;Northern European&quot;                
[13] &quot;British, Irish&quot;                    &quot;Erasmus Rucphen&quot;                  
[15] &quot;Han Chinese, Indian, Malay&quot;        &quot;Orcadian, Erasmus Rucphen&quot;        
[17] &quot;Dutch, European American&quot;          &quot;European and European American&quot;   
[19] &quot;mixed European and Asian ancestry&quot; &quot;Chinese, Japanese&quot;                </code></pre>
<pre class="r"><code># find studies where people fall into different ancestry categories
gwas_ancest_info_multiple = gwas_ancest_info |&gt; 
  filter(grepl(&quot;,&quot;, BROAD_ANCESTRAL_CATEGORY)) |&gt;
  dplyr::filter(!BROAD_ANCESTRAL_CATEGORY %in% 
                  c(&quot;African unspecified, Sub-Saharan African&quot;,
                    &quot;Sub-Saharan African, African unspecified&quot;,
                    &#39;African American or Afro-Caribbean, African unspecified&#39;,
                    &#39;Greater Middle Eastern (Middle Eastern, North African or Persian)&#39;,
                    &#39;Sub-Saharan African, African American or Afro-Caribbean&#39;)
  ) 

# let&#39;s see if initial sample description match number of individuals
gwas_ancest_info_multiple = gwas_ancest_info_multiple |&gt; 
  dplyr::select(PUBMEDID, DATE, STAGE,
                BROAD_ANCESTRAL_CATEGORY, 
                NUMBER_OF_INDIVDUALS,
                INITIAL_SAMPLE_DESCRIPTION) |&gt; 
  dplyr::distinct() |&gt;
    mutate(
    extracted_numbers = stringr::str_extract_all(INITIAL_SAMPLE_DESCRIPTION, &quot;\\d{1,3}(?:,\\d{3})*|\\d+&quot;), 
    extracted_numbers = lapply(extracted_numbers, function(x) as.integer(gsub(&quot;,&quot;, &quot;&quot;, x))),      # remove commas, convert to integers
    extracted_sum = sapply(extracted_numbers, sum),
    matches_reported_total = extracted_sum == NUMBER_OF_INDIVDUALS,
    diff_inital_sample_n_individ = extracted_sum - NUMBER_OF_INDIVDUALS
  )

# total number of individuals in studies with multiple ancestries
total_multiple = sum(gwas_ancest_info_multiple$NUMBER_OF_INDIVDUALS)

# proportion individuals in these studies of total number multiple ancestry
gwas_ancest_info_multiple = gwas_ancest_info_multiple |&gt;
  dplyr::mutate(prop_total_multiple = NUMBER_OF_INDIVDUALS / total_multiple)

gwas_extracted_pairs = gwas_ancest_info_multiple |&gt; 
  dplyr::filter(matches_reported_total == T) |&gt; 
  dplyr::rowwise() |&gt;
  dplyr::filter(length(extracted_numbers) &gt; 1) |&gt;
  dplyr::ungroup()

gwas_extracted_pairs =   
  gwas_extracted_pairs |&gt;
  dplyr::mutate(
    # Extract numberâ€“label patterns: captures digits followed by 1â€“10 words (tunable)
    number_label_matches = stringr::str_extract_all(
      INITIAL_SAMPLE_DESCRIPTION,
      &quot;\\b\\d{1,3}(?:,\\d{3})*\\b(?:\\s+\\w+){1,10}&quot;
    )
  ) |&gt;
  dplyr::select(PUBMEDID,
                number_label_matches, 
                BROAD_ANCESTRAL_CATEGORY, 
                INITIAL_SAMPLE_DESCRIPTION) |&gt;
  # Unnest to one pair per row
  tidyr::unnest_longer(number_label_matches) |&gt;
  # Clean and split number from label
  dplyr::mutate(
    number = stringr::str_extract(number_label_matches, &quot;^\\d{1,3}(?:,\\d{3})*&quot;),
    number = as.integer(gsub(&quot;,&quot;, &quot;&quot;, number)),
    label = stringr::str_trim(stringr::str_remove(number_label_matches, &quot;^\\d{1,3}(?:,\\d{3})*&quot;))
  ) 


gwas_extracted_pairs |&gt;
  dplyr::filter(BROAD_ANCESTRAL_CATEGORY %in% c(&quot;NR, European&quot;,
                                                &quot;European, NR&quot;,
                                                &quot;European, Other&quot;))</code></pre>
<pre><code># A tibble: 154 Ã— 6
   PUBMEDID number_label_matches   BROAD_ANCESTRAL_CATEâ€¦Â¹ INITIAL_SAMPLE_DESCRâ€¦Â²
      &lt;int&gt; &lt;chr&gt;                  &lt;chr&gt;                  &lt;chr&gt;                 
 1 18385738 1,926 European ance oâ€¦ European, Other        1,926 European ance oâ€¦
 2 18385738 2,522 European and otâ€¦ European, Other        1,926 European ance oâ€¦
 3 19451621 1,821 cases            European, Other        1,821 cases, 2,258 coâ€¦
 4 19451621 2,258 controls         European, Other        1,821 cases, 2,258 coâ€¦
 5 20351715 4,387 European and unâ€¦ European, NR           4,387 European and unâ€¦
 6 20351715 6,209 European and unâ€¦ European, NR           4,387 European and unâ€¦
 7 20860503 10,365 cases           European, Other        10,365 cases, 16,110 â€¦
 8 20860503 16,110 controls        European, Other        10,365 cases, 16,110 â€¦
 9 21738484 1,190 European and unâ€¦ NR, European           1,190 European and unâ€¦
10 21738484 401 European and unknâ€¦ NR, European           1,190 European and unâ€¦
# â„¹ 144 more rows
# â„¹ abbreviated names: Â¹â€‹BROAD_ANCESTRAL_CATEGORY, Â²â€‹INITIAL_SAMPLE_DESCRIPTION
# â„¹ 2 more variables: number &lt;int&gt;, label &lt;chr&gt;</code></pre>
<pre class="r"><code>gwas_extracted_pairs |&gt;
  dplyr::filter(!BROAD_ANCESTRAL_CATEGORY %in% c(&quot;NR, European&quot;,
                                                &quot;European, NR&quot;,
                                                &quot;European, Other&quot;))</code></pre>
<pre><code># A tibble: 142 Ã— 6
   PUBMEDID number_label_matches   BROAD_ANCESTRAL_CATEâ€¦Â¹ INITIAL_SAMPLE_DESCRâ€¦Â²
      &lt;int&gt; &lt;chr&gt;                  &lt;chr&gt;                  &lt;chr&gt;                 
 1 19724244 90 Hispanic and Europâ€¦ European, Hispanic orâ€¦ 90 Hispanic and Europâ€¦
 2 19724244 90 Hispanic and Europâ€¦ European, Hispanic orâ€¦ 90 Hispanic and Europâ€¦
 3 20460622 973 incident AD cases  European, Hispanic orâ€¦ 973 incident AD casesâ€¦
 4 20460622 2,033 prevalent AD caâ€¦ European, Hispanic orâ€¦ 973 incident AD casesâ€¦
 5 20460622 22,604  controls of Eâ€¦ European, Hispanic orâ€¦ 973 incident AD casesâ€¦
 6 23021708 132 African American   African American or Aâ€¦ 132 African American,â€¦
 7 23021708 103 African American   African American or Aâ€¦ 132 African American,â€¦
 8 23719583 79 European            European, Hispanic orâ€¦ 79 European, African,â€¦
 9 23719583 241 European           European, Hispanic orâ€¦ 79 European, African,â€¦
10 23838604 175 European           African American or Aâ€¦ 175 European, Africanâ€¦
# â„¹ 132 more rows
# â„¹ abbreviated names: Â¹â€‹BROAD_ANCESTRAL_CATEGORY, Â²â€‹INITIAL_SAMPLE_DESCRIPTION
# â„¹ 2 more variables: number &lt;int&gt;, label &lt;chr&gt;</code></pre>
<pre class="r"><code>gwas_extracted_pairs |&gt;
  dplyr::filter(!BROAD_ANCESTRAL_CATEGORY %in% c(&quot;NR, European&quot;,
                                                &quot;European, NR&quot;,
                                                &quot;European, Other&quot;)) |&gt; dplyr::pull(label) |&gt; unique()</code></pre>
<pre><code> [1] &quot;Hispanic and European ancestry cases&quot;                                           
 [2] &quot;Hispanic and European ancestry controls&quot;                                        
 [3] &quot;incident AD cases&quot;                                                              
 [4] &quot;prevalent AD cases&quot;                                                             
 [5] &quot;controls of European and Hispanic ancestry&quot;                                     
 [6] &quot;African American&quot;                                                               
 [7] &quot;European&quot;                                                                       
 [8] &quot;bipolar disorder cases&quot;                                                         
 [9] &quot;pre&quot;                                                                            
[10] &quot;term birth neonates&quot;                                                            
[11] &quot;term delivery mothers&quot;                                                          
[12] &quot;obese individuals&quot;                                                              
[13] &quot;non&quot;                                                                            
[14] &quot;osteonecrosis cases&quot;                                                            
[15] &quot;treated controls&quot;                                                               
[16] &quot;Native American ancestry&quot;                                                       
[17] &quot;cases with pancreatitis&quot;                                                        
[18] &quot;cases without pancreatitis&quot;                                                     
[19] &quot;cases&quot;                                                                          
[20] &quot;controls&quot;                                                                       
[21] &quot;children&quot;                                                                       
[22] &quot;mothers and 237 fathers&quot;                                                        
[23] &quot;African American or European ancestry female individuals with single fibroid&quot;   
[24] &quot;African American or European ancestry female individuals with multiple fibroids&quot;
[25] &quot;Han Chinese and unknown ancestry alcohol drinkers&quot;                              
[26] &quot;Han Chinese and unknown ancestry non&quot;                                           
[27] &quot;preterm birth infant cases&quot;                                                     
[28] &quot;preterm birth infant controls&quot;                                                  
[29] &quot;Asian&quot;                                                                          
[30] &quot;mothers&quot;                                                                        
[31] &quot;fathers&quot;                                                                        
[32] &quot;Asian ancestry cases&quot;                                                           
[33] &quot;Asian ancestry controls&quot;                                                        
[34] &quot;Chinese or European ancestry cases&quot;                                             
[35] &quot;Chinese or European ancestry controls&quot;                                          
[36] &quot;Hispanic&quot;                                                                       
[37] &quot;African American individuals&quot;                                                   
[38] &quot;East Asian ancestry individuals&quot;                                                
[39] &quot;European ancestry individuals&quot;                                                  
[40] &quot;Hispanic individuals&quot;                                                           
[41] &quot;South Asian ancestry individuals&quot;                                               
[42] &quot;Sub&quot;                                                                            
[43] &quot;European ancestry&quot;                                                              
[44] &quot;East Asian ancestry female cases&quot;                                               
[45] &quot;East Asian ancestry female controls&quot;                                            
[46] &quot;European ancestry European&quot;                                                     
[47] &quot;Japanese&quot;                                                                       </code></pre>
<pre class="r"><code>gwas_extracted_pairs |&gt;
  dplyr::filter(!BROAD_ANCESTRAL_CATEGORY %in% c(&quot;NR, European&quot;,
                                                &quot;European, NR&quot;,
                                                &quot;European, Other&quot;)) |&gt;
 dplyr::select(PUBMEDID, 
               BROAD_ANCESTRAL_CATEGORY, 
               label, 
               number) |&gt;
 dplyr::mutate(label_v2 = case_when(
   label %in% c(&quot;European ancestry&quot;, 
                &quot;European ancestry individuals&quot;,
                &quot;European&quot;) ~ &quot;European&quot;,
   
   label %in% c(&quot;Hispanic&quot;,
                &quot;Hispanic ancestry&quot;,
                &quot;Hispanic individuals&quot;) ~ &quot;Hispanic/Latino&quot;,
   
   label %in% c(&#39;East Asian ancestry female cases&#39;,
                &#39;East Asian ancestry female controls&#39;,
                &#39;East Asian ancestry individuals&#39;, 
                &#39;Japanese&#39;) ~ &quot;East Asian&quot;,
   
   label %in% c(&#39;South Asian ancestry individuals&#39;) ~ &quot;South Asian&quot;,
  
   label %in% c(&#39;African American&#39;,
                &quot;African American individuals&quot;) ~ &#39;African&#39;, 
   
   label %in% c(&#39;Asian&#39;, 
                &#39;Asian ancestry cases&#39;, 
                &#39;Asian ancestry controls&#39;) ~ &#39;Asian unspecified&#39;,
   
   label %in% c(&#39;Native American ancestry&#39;) ~ &quot;Other&quot;,
   TRUE~BROAD_ANCESTRAL_CATEGORY
 )
 )</code></pre>
<pre><code># A tibble: 142 Ã— 5
   PUBMEDID BROAD_ANCESTRAL_CATEGORY                       label number label_v2
      &lt;int&gt; &lt;chr&gt;                                          &lt;chr&gt;  &lt;int&gt; &lt;chr&gt;   
 1 19724244 European, Hispanic or Latin American           Hispâ€¦     90 Europeaâ€¦
 2 19724244 European, Hispanic or Latin American           Hispâ€¦     90 Europeaâ€¦
 3 20460622 European, Hispanic or Latin American           inciâ€¦    973 Europeaâ€¦
 4 20460622 European, Hispanic or Latin American           prevâ€¦   2033 Europeaâ€¦
 5 20460622 European, Hispanic or Latin American           contâ€¦  22604 Europeaâ€¦
 6 23021708 African American or Afro-Caribbean, South Asiâ€¦ Afriâ€¦    132 African 
 7 23021708 African American or Afro-Caribbean, South Asiâ€¦ Afriâ€¦    103 African 
 8 23719583 European, Hispanic or Latin American, Africanâ€¦ Euroâ€¦     79 European
 9 23719583 European, Hispanic or Latin American, Africanâ€¦ Euroâ€¦    241 European
10 23838604 African American or Afro-Caribbean, European,â€¦ Euroâ€¦    175 European
# â„¹ 132 more rows</code></pre>
<pre class="r"><code>gwas_ancest_info_multiple |&gt; 
  dplyr::group_by(matches_reported_total, STAGE) |&gt; 
  dplyr::summarise(n = n())  </code></pre>
<pre><code>`summarise()` has grouped output by &#39;matches_reported_total&#39;. You can override
using the `.groups` argument.</code></pre>
<pre><code># A tibble: 4 Ã— 3
# Groups:   matches_reported_total [2]
  matches_reported_total STAGE           n
  &lt;lgl&gt;                  &lt;chr&gt;       &lt;int&gt;
1 FALSE                  initial       249
2 FALSE                  replication   169
3 TRUE                   initial       267
4 TRUE                   replication     1</code></pre>
<pre class="r"><code>gwas_ancest_info_multiple |&gt;
  dplyr::group_by(matches_reported_total, STAGE) |&gt; 
  dplyr::summarise(median_diff = median(abs(diff_inital_sample_n_individ)),
                   mean_diff = mean(abs(diff_inital_sample_n_individ)),
                   min_diff = min(diff_inital_sample_n_individ),
                   max_diff = max(diff_inital_sample_n_individ)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;matches_reported_total&#39;. You can override
using the `.groups` argument.</code></pre>
<pre><code># A tibble: 4 Ã— 6
# Groups:   matches_reported_total [2]
  matches_reported_total STAGE       median_diff mean_diff min_diff max_diff
  &lt;lgl&gt;                  &lt;chr&gt;             &lt;int&gt;     &lt;dbl&gt;    &lt;int&gt;    &lt;int&gt;
1 FALSE                  initial           17803    94252.  -187639  1190995
2 FALSE                  replication        6862    44807.  -741609   337502
3 TRUE                   initial               0        0         0        0
4 TRUE                   replication           0        0         0        0</code></pre>
<pre class="r"><code>gwas_ancest_info_multiple |&gt;
  dplyr::filter(matches_reported_total == F) |&gt;
  ggplot(aes(x = diff_inital_sample_n_individ)) + 
  geom_histogram() + 
  theme_bw() + 
  facet_wrap(~STAGE)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/replication_ancestry_bias.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gwas_ancest_info_multiple |&gt;
  dplyr::filter(matches_reported_total == F) |&gt;
  ggplot(aes(y = diff_inital_sample_n_individ)) + 
  geom_boxplot() + 
  theme_bw() + 
  facet_wrap(~STAGE)</code></pre>
<p><img src="figure/replication_ancestry_bias.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gwas_ancest_info_multiple |&gt;
  arrange(desc(NUMBER_OF_INDIVDUALS)) |&gt;
  head(
  )</code></pre>
<pre><code>   PUBMEDID       DATE       STAGE
      &lt;int&gt;     &lt;IDat&gt;      &lt;char&gt;
1: 34548389 2021-09-21     initial
2: 33243845 2020-11-26 replication
3: 34450027 2021-08-24     initial
4: 34450027 2021-08-24     initial
5: 32888493 2020-09-01     initial
6: 32888493 2020-09-01     initial
                                                                                                 BROAD_ANCESTRAL_CATEGORY
                                                                                                                   &lt;char&gt;
1:                      European, East Asian, South Asian, African American or Afro-Caribbean, Hispanic or Latin American
2:                                                                                                           European, NR
3:                                                                                                   East Asian, European
4:                                                                                                   East Asian, European
5: African American or Afro-Caribbean, African unspecified, European, East Asian, Hispanic or Latin American, South Asian
6: African American or Afro-Caribbean, African unspecified, European, East Asian, Hispanic or Latin American, South Asian
   NUMBER_OF_INDIVDUALS
                  &lt;int&gt;
1:              1159871
2:               959564
3:               890949
4:               890949
5:               746667
6:               746431
                                                                                                                                                                 INITIAL_SAMPLE_DESCRIPTION
                                                                                                                                                                                     &lt;char&gt;
1: 949,116 European ancestry individuals, 166,997 East Asian ancestry individuals, 21,338 South Asian ancestry individuals, 17,459 African American individuals, 4,961 Hispanic individuals
2:                                                                                                                         16,761 Finnish ancestry cases, 201,194 Finnish ancestry controls
3:                                                                        up to 400,604 European, Japanese ancestry female cases, up to 490,345 European, Japanese ancestry female controls
4:                                                                            up to 400,604 Japanese, European ancestry male cases, up to 490,345 Japanese, European ancestry male controls
5:                    746,667 African American or Afro-Caribbean, African ancestry, European ancestry, East Asian ancestry, Hispanic or Latin American and South Asian ancestry individuals
6:                    746,431 African American or Afro-Caribbean, African ancestry, European ancestry, East Asian ancestry, Hispanic or Latin American and South Asian ancestry individuals
                    extracted_numbers extracted_sum matches_reported_total
                               &lt;list&gt;         &lt;int&gt;                 &lt;lgcl&gt;
1: 949116,166997, 21338, 17459,  4961       1159871                   TRUE
2:                       16761,201194        217955                  FALSE
3:                      400604,490345        890949                   TRUE
4:                      400604,490345        890949                   TRUE
5:                             746667        746667                   TRUE
6:                             746431        746431                   TRUE
   diff_inital_sample_n_individ prop_total_multiple
                          &lt;int&gt;               &lt;num&gt;
1:                            0          0.02556391
2:                      -741609          0.02114908
3:                            0          0.01963679
4:                            0          0.01963679
5:                            0          0.01645677
6:                            0          0.01645157</code></pre>
<pre class="r"><code>gwas_ancest_info_multiple |&gt;
  dplyr::slice_max(NUMBER_OF_INDIVDUALS, n = 100) |&gt;
  dplyr::mutate(PUBMEDID = as.factor(PUBMEDID)) |&gt;
  dplyr::mutate(PUBMEDID = reorder(PUBMEDID, NUMBER_OF_INDIVDUALS)) |&gt; 
  ggplot(aes(x = NUMBER_OF_INDIVDUALS, y= PUBMEDID)) +
  geom_col() + 
  theme_bw()</code></pre>
<p><img src="figure/replication_ancestry_bias.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gwas_ancest_info_multiple |&gt;
  dplyr::mutate(prop_number_of_individ = NUMBER_OF_INDIVDUALS / total_multiple) |&gt;
  dplyr::arrange(desc(NUMBER_OF_INDIVDUALS)) |&gt;  
  dplyr::mutate(cum_prop_n_of_individ = cumsum(prop_number_of_individ)) |&gt;
  dplyr::select(PUBMEDID, cum_prop_n_of_individ, NUMBER_OF_INDIVDUALS)</code></pre>
<pre><code>     PUBMEDID cum_prop_n_of_individ NUMBER_OF_INDIVDUALS
        &lt;int&gt;                 &lt;num&gt;                &lt;int&gt;
  1: 34548389            0.02556391              1159871
  2: 33243845            0.04671299               959564
  3: 34450027            0.06634978               890949
  4: 34450027            0.08598657               890949
  5: 32888493            0.10244333               746667
 ---                                                    
682: 29315502            0.99999960                   10
683: 33589590            0.99999974                    6
684: 33589590            0.99999987                    6
685: 34680877            0.99999998                    5
686: 31299468            1.00000000                    1</code></pre>
<pre class="r"><code># in total - studies where initial number of individuals = number of individual
# ~ 70% 
gwas_ancest_info_multiple |&gt;
  dplyr::filter(matches_reported_total == T) |&gt; 
  dplyr::arrange(desc(NUMBER_OF_INDIVDUALS)) |&gt;  
  dplyr::mutate(cum_prop_n_of_individ = cumsum(prop_total_multiple)) |&gt;
  dplyr::select(PUBMEDID, cum_prop_n_of_individ, NUMBER_OF_INDIVDUALS) |&gt; 
  tail()</code></pre>
<pre><code>   PUBMEDID cum_prop_n_of_individ NUMBER_OF_INDIVDUALS
      &lt;int&gt;                 &lt;num&gt;                &lt;int&gt;
1: 22391508             0.7030777                  100
2: 32066683             0.7030798                   95
3: 33669428             0.7030818                   90
4: 33831438             0.7030838                   90
5: 34025683             0.7030857                   87
6: 34025683             0.7030876                   86</code></pre>
<div id="what-are-the-groups" class="section level3">
<h3>What are the groups?</h3>
<pre class="r"><code># code adapted from https://github.com/armartin/prs_disparities/blob/master/gwas_disparities_time.R

group_ancestry_fn = function(study_ancest){
  
   case_when(
     
  # European
  study_ancest %in% c(&#39;European&#39;) ~ &#39;European&#39;,
  
  # African
  study_ancest %in% c(&#39;Sub-Saharan African, African American or Afro-Caribbean&#39;,
                      &#39;African American or Afro-Caribbean, African unspecified&#39;,
                      &#39;Sub-Saharan African, African unspecified&#39;,
                      &#39;African-American or Afro-Caribbean&#39;,
                      &#39;Sub-Saharan African&#39;,
                      &#39;African American or Afro-Caribbean&#39;,
                      &#39;African unspecified&#39;) ~ &#39;African&#39;,
      
  # Asian  
  study_ancest %in% c(&#39;East Asian, Asian unspecified&#39;, 
                      &#39;South Asian, East Asian &#39;,
                      &#39;South Asian, South East Asian&#39;, 
                      &#39;South Asian, South East Asian, East Asian&#39;,
                      &#39;South East Asian, East Asian&#39;, 
                      &#39;South East Asian, South Asian, East Asian&#39;,
                      &#39;South Asian&#39;,
                      &#39;South East Asian&#39;,
                      &#39;Central Asian&#39;,
                      &#39;East Asian&#39;,
                      &#39;Asian unspecified&#39;) ~&#39;Asian&#39;,
  
  # Middle eastern    
  study_ancest == &#39;Greater Middle Eastern (Middle Eastern, North African or Persian)&#39; ~&#39;Middle Eastern&#39;,

  # Oceanic
  study_ancest %in% c(&#39;Aboriginal Australian&#39;, 
                      &#39;Oceanian&#39;) ~&#39;Oceanic&#39;,
  
  # Hispanic/Latin American
  study_ancest %in% &quot;Hispanic or Latin American&quot; ~&#39;Hispanic/Latin American&#39;,
  
  # Other 
  study_ancest %in% c(&#39;Other&#39;,
                      &#39;Other, NR&#39;, 
                      &#39;NR, Other&#39;,
                      &#39;Other admixed ancestry&#39;, 
                      &#39;Native American&#39;) ~ &quot;Other&quot;,
  
  # Not reported
  study_ancest %in% &quot;NR&quot; ~ &quot;Not reported&quot;,
  
  # Multiple    
  grepl(&quot;, &quot;, study_ancest) ~ &#39;Multiple&#39;,
      
  TRUE~study_ancest
    )
  
    
  
}


grouped_ancest = vector()

for(study_ancest in unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY)){
  
grouped_ancest[study_ancest] = group_ancestry_fn(study_ancest)

}

grouped_ancest_map = data.frame(ancestry_group = grouped_ancest,
                                BROAD_ANCESTRAL_CATEGORY = unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY)
                                )


print(head(grouped_ancest_map))</code></pre>
<pre><code>                                   ancestry_group
European                                 European
Other                                       Other
Asian unspecified                           Asian
African American or Afro-Caribbean        African
NR                                   Not reported
South East Asian                            Asian
                                             BROAD_ANCESTRAL_CATEGORY
European                                                     European
Other                                                           Other
Asian unspecified                                   Asian unspecified
African American or Afro-Caribbean African American or Afro-Caribbean
NR                                                                 NR
South East Asian                                     South East Asian</code></pre>
</div>
<div id="apply-this-ancestry-mapping" class="section level3">
<h3>Apply this ancestry mapping</h3>
<pre class="r"><code>gwas_ancest_info = dplyr::left_join(
            gwas_ancest_info,
            grouped_ancest_map)</code></pre>
<pre><code>Joining with `by = join_by(BROAD_ANCESTRAL_CATEGORY)`</code></pre>
<pre class="r"><code>gwas_ancest_info %&gt;% 
  group_by(ancestry_group) %&gt;% 
  summarise(n = sum(NUMBER_OF_INDIVDUALS, na.rm = TRUE))</code></pre>
<pre><code># A tibble: 9 Ã— 2
  ancestry_group                   n
  &lt;chr&gt;                        &lt;dbl&gt;
1 African                   13322669
2 Asian                    135105215
3 European                4235129752
4 Hispanic/Latin American   10840275
5 Middle Eastern              348192
6 Multiple                  55154361
7 Not reported             110448036
8 Oceanic                     116432
9 Other                       404165</code></pre>
<pre class="r"><code># colours from RColorBrewer::brewer.pal(n = 9, &quot;Set1&quot;)
ancestry_colors &lt;- c(
  &quot;African&quot; = &quot;#984EA3&quot;,
  &quot;European&quot; = &quot;#E41A1C&quot;,
  &quot;Asian&quot; = &quot;#377EB8&quot;, #east asian
  # &quot;South Asian&quot; = &quot;#4DAF4A&quot;,
  &quot;Hispanic/Latin American&quot; = &quot;#FF7F00&quot;,
  &quot;Middle Eastern&quot; = &quot;#FFFF33&quot;,
  &quot;Oceanic&quot; = &quot;#A65628&quot;,
  &quot;Other&quot; = &quot;#F781BF&quot;,
  &quot;Multiple&quot; = &quot;#999999&quot;,
  &quot;Not reported&quot; = &quot;black&quot;
  
)

# Define the desired stacking order
ancestry_levels &lt;- c(
  &quot;European&quot;,
  # &quot;East Asian&quot;,
  # &quot;South Asian&quot;,
  &quot;Asian&quot;,
  &quot;African&quot;,
  &quot;Hispanic/Latin American&quot;,
  &quot;Middle Eastern&quot;,
  &quot;Oceanic&quot;,
  &quot;Other&quot;,
  &quot;Multiple&quot;,
  &quot;Not reported&quot;
)

gwas_ancest_info %&gt;% 
  group_by(ancestry_group) %&gt;% 
  mutate(ancestry_group = factor(ancestry_group, levels = ancestry_levels)) %&gt;%
  mutate(ancest_cumsum = cumsum(as.numeric(NUMBER_OF_INDIVDUALS))) %&gt;% 
  ggplot(aes(x=DATE, y=ancest_cumsum/(10^6), fill = ancestry_group)) + 
  #geom_area() + 
  geom_area(position = &#39;stack&#39;) + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  theme_classic() + 
  labs(x = &quot;Year&quot;, y = &quot;Individuals in GWAS catalog (millons)&quot;) + 
  scale_fill_manual(values = ancestry_colors, name=&#39;Ancestry group&#39;) </code></pre>
<p><img src="figure/replication_ancestry_bias.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # scale_fill_brewer(palette = &quot;Set1&quot;)</code></pre>
<pre class="r"><code>inner_join(
           gwas_study_info %&gt;% select(STUDY_ACCESSION, 
                                      `DISEASE/TRAIT`, 
                                      MAPPED_TRAIT),
           gwas_ancest_info %&gt;% select(STUDY_ACCESSION, 
                                       NUMBER_OF_INDIVDUALS,
                                       BROAD_ANCESTRAL_CATEGORY,
                                       ancestry_group)) %&gt;% 
  group_by(BROAD_ANCESTRAL_CATEGORY, 
           `DISEASE/TRAIT`) %&gt;% 
  summarise(n = sum(NUMBER_OF_INDIVDUALS)) %&gt;% 
  group_by(BROAD_ANCESTRAL_CATEGORY) %&gt;% 
  slice_max(n,n = 5)</code></pre>
<pre><code>Joining with `by = join_by(STUDY_ACCESSION)`
`summarise()` has grouped output by &#39;BROAD_ANCESTRAL_CATEGORY&#39;. You can
override using the `.groups` argument.</code></pre>
<pre><code># A tibble: 802 Ã— 3
# Groups:   BROAD_ANCESTRAL_CATEGORY [192]
   BROAD_ANCESTRAL_CATEGORY                               `DISEASE/TRAIT`      n
   &lt;chr&gt;                                                  &lt;chr&gt;            &lt;int&gt;
 1 Aboriginal Australian                                  Urinary albumiâ€¦    746
 2 Aboriginal Australian                                  Otitis media       391
 3 Aboriginal Australian                                  Type 2 diabetes    391
 4 Aboriginal Australian                                  Body mass index    361
 5 African American or Afro-Caribbean                     Cataracts       262576
 6 African American or Afro-Caribbean                     Body mass index 228346
 7 African American or Afro-Caribbean                     Type 2 diabetes 202731
 8 African American or Afro-Caribbean                     Diastolic blooâ€¦ 200474
 9 African American or Afro-Caribbean                     Systolic bloodâ€¦ 200474
10 African American or Afro-Caribbean, African unspecifiâ€¦ Type 2 diabetes 287510
# â„¹ 792 more rows</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_Australia.utf8  LC_CTYPE=English_Australia.utf8   
[3] LC_MONETARY=English_Australia.utf8 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.utf8    

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] ggplot2_3.5.2   dplyr_1.1.4     workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       jsonlite_2.0.0     compiler_4.3.1     renv_1.0.3        
 [5] promises_1.3.3     tidyselect_1.2.1   Rcpp_1.0.14        stringr_1.5.1     
 [9] git2r_0.36.2       tidyr_1.3.1        callr_3.7.6        later_1.4.2       
[13] jquerylib_0.1.4    scales_1.4.0       yaml_2.3.10        fastmap_1.2.0     
[17] R6_2.6.1           labeling_0.4.3     generics_0.1.4     knitr_1.50        
[21] tibble_3.2.1       rprojroot_2.1.0    RColorBrewer_1.1-3 bslib_0.9.0       
[25] pillar_1.10.2      rlang_1.1.6        utf8_1.2.5         cachem_1.1.0      
[29] stringi_1.8.7      httpuv_1.6.13      xfun_0.52          getPass_0.2-4     
[33] fs_1.6.6           sass_0.4.10        cli_3.6.5          withr_3.0.2       
[37] magrittr_2.0.3     ps_1.9.1           grid_4.3.1         digest_0.6.37     
[41] processx_3.8.6     rstudioapi_0.15.0  lifecycle_1.0.4    vctrs_0.6.5       
[45] data.table_1.17.4  evaluate_1.0.3     glue_1.8.0         farver_2.1.2      
[49] whisker_0.4.1      purrr_1.0.4        rmarkdown_2.29     httr_1.4.7        
[53] tools_4.3.1        pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
