<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Replicate observed ancestry biases figures in GWAS dataset</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Replicate observed ancestry biases figures
in GWAS dataset</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree02fe018bafe9143ed172d637ceb22acaeb0a2655targetblank02fe018a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/02fe018bafe9143ed172d637ceb22acaeb0a2655" target="_blank">02fe018</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree02fe018bafe9143ed172d637ceb22acaeb0a2655targetblank02fe018a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/02fe018bafe9143ed172d637ceb22acaeb0a2655" target="_blank">02fe018</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/gwas_catalog/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/trait_ontology/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  data/.DS_Store
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   analysis/collapse_traits.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/replication_ancestry_bias.Rmd</code>) and HTML
(<code>docs/replication_ancestry_bias.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/02fe018bafe9143ed172d637ceb22acaeb0a2655/analysis/replication_ancestry_bias.Rmd" target="_blank">02fe018</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-25
</td>
<td>
Update replicatio Martin et al. 2019
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/ec761592c38ee66054be983d1cb588339aaf05e9/docs/replication_ancestry_bias.html" target="_blank">ec76159</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/10a35cc5b555c4f9759b44ea044d867e7c9ade61/analysis/replication_ancestry_bias.Rmd" target="_blank">10a35cc</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
<td>
Updating replication of ancestry biases in gwas
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(dplyr)
library(ggplot2)
library(data.table)
source(here::here(&quot;code/custom_plotting.R&quot;))</code></pre>
<div id="load-gwas-catalog-data" class="section level2">
<h2>Load GWAS catalog data</h2>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_study_info_trait_ontology_info.csv&quot;))

gwas_ancest_info &lt;-  fread(here::here(&quot;data/gwas_catalog/gwas-catalog-v1.0.3.1-ancestries-r2025-07-21.tsv&quot;),
                         sep = &quot;\t&quot;, 
                         quote = &quot;&quot;)

gwas_study_info = gwas_study_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_ancest_info = gwas_ancest_info |&gt;
  dplyr::rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_ancest_info = gwas_ancest_info |&gt;
                    arrange(DATE)

gwas_study_info = gwas_study_info |&gt;
                  arrange(DATE)</code></pre>
</div>
<div id="plot-figure-martin-et-al.-2019-like" class="section level2">
<h2>Plot figure Martin et al. 2019 like</h2>
<div id="for-all-ancestries" class="section level3">
<h3>For all ancestries</h3>
<pre class="r"><code># code adapted from https://github.com/armartin/prs_disparities/blob/master/gwas_disparities_time.R

# https://github.com/armartin/prs_disparities/blob/master/gwas_sfs_pop.R
# gwas cat
# http://bioconductor.org/packages/release/bioc/html/gwascat.html

# following steps from https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-019-0379-x/MediaObjects/41588_2019_379_MOESM1_ESM.pdf


# Some number of individuals are missing
gwas_ancest_info |&gt; dplyr::filter(is.na(NUMBER_OF_INDIVIDUALS) | NUMBER_OF_INDIVIDUALS == 0) |&gt; nrow()</code></pre>
<pre><code>[1] 46</code></pre>
<pre class="r"><code>gwas_ancest_info |&gt; dplyr::filter(is.na(NUMBER_OF_INDIVIDUALS) | NUMBER_OF_INDIVIDUALS == 0) |&gt; head()</code></pre>
<pre><code>   STUDY_ACCESSION PUBMED_ID     FIRST_AUTHOR       DATE
            &lt;char&gt;     &lt;int&gt;           &lt;char&gt;     &lt;IDat&gt;
1:      GCST000308  19115949          Limou S 2009-01-01
2:      GCST001070  21573907         Parra EJ 2011-05-15
3:      GCST001564  22694930             Li X 2012-06-11
4:      GCST001788  23263445 Lawrance-Owen AJ 2012-12-22
5:      GCST002197  24057671       Chimusa ER 2013-09-18
6:      GCST002330  24406073      Galanter JM 2014-01-07
                                                                                                                               INITIAL_SAMPLE_DESCRIPTION
                                                                                                                                                   &lt;char&gt;
1: 275 European ancestry seropositive non-progressors, 86 European ancestry seropositive rapid progressors, 1,352 European ancestry seronegative controls
2:                                                                                                            1,804 Hispanic cases, 780 Hispanic controls
3:                                                                                          813 European ancestry cases, 1,011 European ancestry controls
4:                                                                                                                      979 European ancestry individuals
5:                                                                  642 South African coloured cases, 91 South African coloured controls, (see Thye 2010)
6:                                                                                              1,893 Latin American cases, 1,881 Latin American controls
                      REPLICATION_SAMPLE_DESCRIPTION       STAGE
                                              &lt;char&gt;      &lt;char&gt;
1:                                 (see Fellay 2007) replication
2:                     European ancestry individuals replication
3:                         TENOR and GABRIEL studies replication
4:                               (see Medland, 2010) replication
5:                                              &lt;NA&gt;     initial
6: EVE Asthma Consortium, see Torgerson et al., 2011 replication
   NUMBER_OF_INDIVIDUALS       BROAD_ANCESTRAL_CATEGORY COUNTRY_OF_ORIGIN
                   &lt;int&gt;                         &lt;char&gt;            &lt;char&gt;
1:                    NA                             NR                NR
2:                    NA                       European                NR
3:                    NA                             NR                NR
4:                    NA                             NR                NR
5:                    NA            Sub-Saharan African                NR
6:                    NA NR, Hispanic or Latin American                NR
   COUNTRY_OF_RECRUITMENT ADDITIONAL_ANCESTRY_DESCRIPTION ANCESTRY_DESCRIPTOR
                   &lt;char&gt;                          &lt;char&gt;              &lt;lgcl&gt;
1:                     NR                                                  NA
2:                     NR                                                  NA
3:                     NR                                                  NA
4:                     NR                                                  NA
5:  Malawi, Ghana, Gambia                                                  NA
6:                     NR                                                  NA
   FOUNDER/GENETICALLY_ISOLATED_POPULATION NUMBER_OF_CASES NUMBER_OF_CONTROLS
                                    &lt;lgcl&gt;          &lt;lgcl&gt;             &lt;lgcl&gt;
1:                                      NA              NA                 NA
2:                                      NA              NA                 NA
3:                                      NA              NA                 NA
4:                                      NA              NA                 NA
5:                                      NA              NA                 NA
6:                                      NA              NA                 NA
   SAMPLE_DESCRIPTION
               &lt;lgcl&gt;
1:                 NA
2:                 NA
3:                 NA
4:                 NA
5:                 NA
6:                 NA</code></pre>
<pre class="r"><code># Filter missing number of individuals
gwas_ancest_info = gwas_ancest_info |&gt;
  dplyr::filter(!is.na(NUMBER_OF_INDIVIDUALS)) |&gt;
  dplyr::filter(NUMBER_OF_INDIVIDUALS != 0)

# calculate cumulative number of individuals
gwas_ancest_info = gwas_ancest_info %&gt;%  
                   dplyr::arrange(DATE) |&gt;
                   mutate(cum_num = cumsum(as.numeric(NUMBER_OF_INDIVIDUALS))
                          )


# plot cumulative numbers
gwas_ancest_info %&gt;%  
  # group_by(DATE) %&gt;% 
  # slice_max(NUMBER_OF_INDIVDUALS) %&gt;% 
  ggplot(aes(x=DATE,y=cum_num/1e6)) + 
 # geom_line() +
  geom_area() + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  custom_theme + 
  labs(x = &quot;Year&quot;, 
       y = &quot;Individuals in GWAS catalog (millons)&quot;)</code></pre>
<p><img src="figure/replication_ancestry_bias.Rmd/plot_like_martin_et_al_2019_not_grouped-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_like_martin_et_al_2019_not_grouped-1">
Past versions of plot_like_martin_et_al_2019_not_grouped-1.png
</button>
</p>
<div id="fig-plot_like_martin_et_al_2019_not_grouped-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/ec761592c38ee66054be983d1cb588339aaf05e9/docs/figure/replication_ancestry_bias.Rmd/plot_like_martin_et_al_2019_not_grouped-1.png" target="_blank">ec76159</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 44 studies / 44 rows
gwas_ancest_info |&gt;
dplyr::filter(is.na(NUMBER_OF_INDIVIDUALS)) |&gt;
nrow()</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># from only 24 gwas papers
gwas_ancest_info |&gt;
    dplyr::filter(is.na(NUMBER_OF_INDIVIDUALS)) |&gt; 
    select(PUBMED_ID) |&gt; 
    distinct() |&gt;
    nrow()</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>gwas_ancest_info |&gt;
  dplyr::filter(PUBMED_ID == 28679651) |&gt;
  select(INITIAL_SAMPLE_DESCRIPTION, 
         REPLICATION_SAMPLE_DESCRIPTION, 
         BROAD_ANCESTRAL_CATEGORY) |&gt;
  distinct()</code></pre>
<pre><code>   INITIAL_SAMPLE_DESCRIPTION REPLICATION_SAMPLE_DESCRIPTION
                       &lt;char&gt;                         &lt;char&gt;
1:        404 cases, controls                           &lt;NA&gt;
2:        194 cases, controls                           &lt;NA&gt;
3:        426 cases, controls                           &lt;NA&gt;
4:         85 cases, controls                           &lt;NA&gt;
5:        535 cases, controls                           &lt;NA&gt;
6:        345 cases, controls                           &lt;NA&gt;
7:        835 cases, controls                           &lt;NA&gt;
8:        844 cases, controls                           &lt;NA&gt;
9:        447 cases, controls                           &lt;NA&gt;
   BROAD_ANCESTRAL_CATEGORY
                     &lt;char&gt;
1:                       NR
2:                       NR
3:                       NR
4:                       NR
5:                       NR
6:                       NR
7:                       NR
8:                       NR
9:                       NR</code></pre>
<pre class="r"><code># 28679651 - problem seems to be that number of controls per disease not specifically listed
# see https://pubmed.ncbi.nlm.nih.gov/28679651/


# although paper they cite as where data comes from (https://www.nature.com/articles/leu2016387#Tab1)
# discloses: 1229 AL amyloidosis patients from Germany, UK and Italy, and 7526 healthy local controls</code></pre>
</div>
<div id="filter-out-na-number-of-individuals" class="section level3">
<h3>Filter out NA number of individuals</h3>
<pre class="r"><code>gwas_ancest_info  =
gwas_ancest_info |&gt;
dplyr::filter(!is.na(NUMBER_OF_INDIVIDUALS))</code></pre>
</div>
</div>
<div id="make-ancestry-groups" class="section level1">
<h1>Make ancestry groups</h1>
<p>Here we make the column ‘ancestry_group’ in the gwas_study_info
datasets, ‘ancestry_group’ defines the broad ancestry group (like in
Martin et al. 2019, European, Greater Middle Eastern etc.) that each
group of individuals belongs to.</p>
<pre class="r"><code>grouped_ancest = vector()
broad_ancest_cat = unique(gwas_ancest_info$BROAD_ANCESTRAL_CATEGORY)

for(study_ancest in broad_ancest_cat){
  
grouped_ancest[study_ancest] = group_ancestry_fn(study_ancest)

}

grouped_ancest_map = data.frame(ancestry_group = grouped_ancest,
                                BROAD_ANCESTRAL_CATEGORY = broad_ancest_cat
                                )


print(&quot;Mapping examples - broad ancestral categories to ancestry groups&quot;)</code></pre>
<pre><code>[1] &quot;Mapping examples - broad ancestral categories to ancestry groups&quot;</code></pre>
<pre class="r"><code>head(grouped_ancest_map)</code></pre>
<pre><code>                                   ancestry_group
European                                 European
Asian unspecified                           Asian
Other                                       Other
African American or Afro-Caribbean        African
NR                                   Not reported
South East Asian                            Asian
                                             BROAD_ANCESTRAL_CATEGORY
European                                                     European
Asian unspecified                                   Asian unspecified
Other                                                           Other
African American or Afro-Caribbean African American or Afro-Caribbean
NR                                                                 NR
South East Asian                                     South East Asian</code></pre>
<pre class="r"><code>gwas_ancest_info = dplyr::left_join(
            gwas_ancest_info,
            grouped_ancest_map,
            by = &quot;BROAD_ANCESTRAL_CATEGORY&quot;)

gwas_ancest_info = gwas_ancest_info |&gt;
                    dplyr::mutate(ancestry_group = factor(ancestry_group, levels = ancestry_levels))</code></pre>
<div id="number-of-individuals-per-ancestry-group-in-millions"
class="section level2">
<h2>Number of individuals per ancestry group (in millions)</h2>
<pre class="r"><code>gwas_ancest_info %&gt;% 
  dplyr::group_by(ancestry_group) %&gt;% 
  dplyr::summarise(n = sum(NUMBER_OF_INDIVIDUALS, na.rm = TRUE)/ 10^6)</code></pre>
<pre><code># A tibble: 9 × 2
  ancestry_group                 n
  &lt;fct&gt;                      &lt;dbl&gt;
1 European                9122.   
2 Asian                    313.   
3 African                  408.   
4 Hispanic/Latin American  179.   
5 Middle Eastern             4.47 
6 Oceanic                    0.134
7 Other                      2.01 
8 Multiple                 186.   
9 Not reported             149.   </code></pre>
<div
id="make-plot-like-martin-et-al.-2019-by-applying-this-ancestry-mapping"
class="section level3">
<h3>Make plot like Martin et al. 2019 by applying this ancestry
mapping</h3>
<pre class="r"><code># Define the desired stacking order
gwas_ancest_info = 
gwas_ancest_info %&gt;% 
  mutate(ancestry_group = factor(ancestry_group, levels = ancestry_levels)) %&gt;%
  group_by(ancestry_group) %&gt;% 
  mutate(ancest_cumsum = cumsum(as.numeric(NUMBER_OF_INDIVIDUALS))) %&gt;% 
  add_final_totals() 


gwas_ancest_info |&gt; 
  ggplot(aes(x=DATE, y=ancest_cumsum/(10^6), fill = ancestry_group)) + 
  geom_area(position = &#39;stack&#39;) + 
  scale_x_date(date_labels = &#39;%Y&#39;, date_breaks = &quot;2 years&quot;) + 
  theme_classic() + 
  labs(x = &quot;Year&quot;, y = &quot;Individuals in GWAS catalog (millons)&quot;) + 
  scale_fill_manual(values = ancestry_colors, name=&#39;Ancestry group&#39;) </code></pre>
<p><img src="figure/replication_ancestry_bias.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/ec761592c38ee66054be983d1cb588339aaf05e9/docs/figure/replication_ancestry_bias.Rmd/unnamed-chunk-2-1.png" target="_blank">ec76159</a>
</td>
<td>
IJBeasley
</td>
<td>
2025-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] data.table_1.17.8 ggplot2_3.5.2     dplyr_1.1.4       workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       jsonlite_2.0.0     compiler_4.3.1     renv_1.0.3        
 [5] promises_1.3.3     tidyselect_1.2.1   Rcpp_1.1.0         stringr_1.5.1     
 [9] git2r_0.36.2       callr_3.7.6        later_1.4.2        jquerylib_0.1.4   
[13] scales_1.4.0       yaml_2.3.10        fastmap_1.2.0      here_1.0.1        
[17] R6_2.6.1           labeling_0.4.3     generics_0.1.4     knitr_1.50        
[21] tibble_3.3.0       rprojroot_2.1.0    RColorBrewer_1.1-3 bslib_0.9.0       
[25] pillar_1.11.0      rlang_1.1.6        utf8_1.2.6         cachem_1.1.0      
[29] stringi_1.8.7      httpuv_1.6.16      xfun_0.52          getPass_0.2-4     
[33] fs_1.6.6           sass_0.4.10        cli_3.6.5          withr_3.0.2       
[37] magrittr_2.0.3     ps_1.9.1           grid_4.3.1         digest_0.6.37     
[41] processx_3.8.6     rstudioapi_0.17.1  lifecycle_1.0.4    vctrs_0.6.5       
[45] evaluate_1.0.4     glue_1.8.0         farver_2.1.2       whisker_0.4.1     
[49] rmarkdown_2.29     httr_1.4.7         tools_4.3.1        pkgconfig_2.0.3   
[53] htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
