<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-10-17" />

<title>Get abstract text for GWAS Catalog studies</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Get abstract text for GWAS Catalog
studies</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-10-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree0d8b8727e9ef63901141f9e5e118e6db4f0e7393targetblank0d8b872a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/0d8b8727e9ef63901141f9e5e118e6db4f0e7393" target="_blank">0d8b872</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree0d8b8727e9ef63901141f9e5e118e6db4f0e7393targetblank0d8b872a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/0d8b8727e9ef63901141f9e5e118e6db4f0e7393" target="_blank">0d8b872</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    IHME_GBD_2019_Level_2.csv
    Ignored:    analysis/.DS_Store
    Ignored:    ancestry_dispar_env/
    Ignored:    author_manuscript.filelist.txt
    Ignored:    author_manuscript_files.txt
    Ignored:    data/.DS_Store
    Ignored:    data/cohort/
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/manual_disease_icd10_mappings.xlsx
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/~$IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/who/
    Ignored:    fulltexts_pmc/
    Ignored:    output/.DS_Store
    Ignored:    output/abstracts/
    Ignored:    output/doccano/
    Ignored:    output/fulltexts/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_cohorts/
    Ignored:    output/icd_map/
    Ignored:    output/trait_ontology/
    Ignored:    pubmedbert-cohort-ner-model/
    Ignored:    pubmedbert-cohort-ner/
    Ignored:    r-spacyr/
    Ignored:    renv/
    Ignored:    venv/

Untracked files:
    Untracked:  analysis/get_full_text.Rmd
    Untracked:  analysis/specific_aims_stats.Rmd
    Untracked:  code/full_text_download.R
    Untracked:  code/get_dbgap_ids.py
    Untracked:  code/get_pmids_from_dbgap.py
    Untracked:  test_cohort_info_overlap.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/correcting_cohort_names.Rmd
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/gbd_data_plots.Rmd
    Modified:   analysis/group_cancer_diseases.Rmd
    Modified:   analysis/gwas_to_gbd.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_1_disease_group_non_cancer.Rmd
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   analysis/map_trait_to_icd10.Rmd
    Modified:   analysis/trait_ontology_categorization.Rmd
    Deleted:    code/abstracts.R
    Modified:   code/pubmedbert_train_test.py

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/text_for_cohort_labels.Rmd</code>) and HTML
(<code>docs/text_for_cohort_labels.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/0d8b8727e9ef63901141f9e5e118e6db4f0e7393/analysis/text_for_cohort_labels.Rmd" target="_blank">0d8b872</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-24
</td>
<td>
Cleaning up abstract collecting code again
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/2afc108f64bdf3160790bb105a7cba49ad647a01/docs/text_for_cohort_labels.html" target="_blank">2afc108</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/748dac2c44b726dbede003c6d4810481e18466a7/analysis/text_for_cohort_labels.Rmd" target="_blank">748dac2</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-10-24
</td>
<td>
Cleaning up abstract collecting
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1">
<h1>Set up</h1>
<div id="required-packages" class="section level2">
<h2>Required packages</h2>
<pre class="r"><code>library(stringr)
library(readxl)
library(dplyr)
library(stringi)
library(rentrez)
library(xml2)
library(jsonlite)

# Improve sentence recognition
# library(reticulate)

# Path to the Python inside your venv
# python_path &lt;- file.path(here::here(), &quot;venv&quot;, &quot;bin&quot;, &quot;python&quot;)  # Mac/Linux
# Sys.setenv(SPACY_PYTHON = python_path)
# 
# use_python(Sys.getenv(&quot;SPACY_PYTHON&quot;, 
#                                       unset = &quot;r-spacyr&quot;), required = TRUE)
# 
# # Check configuration
# py_config()
# 
# py_module_available(&quot;spacy&quot;)
# 
# library(spacyr)
# spacy_initialize(model = &quot;en_core_web_sm&quot;)
# library(spacyr)
# reticulate::virtualenv_create(&quot;r-spacyr&quot;, python = python_exe)
# spacy_install(version = &quot;apple&quot;)
# spacy_download_langmodel(&quot;en_core_web_sm&quot;)
library(tokenizers)</code></pre>
</div>
<div id="getting-required-information-from-gwas-catalog"
class="section level2">
<h2>Getting required information from GWAS catalog</h2>
<div id="step-1-get-only-disease-studies-and-select-relevant-columns"
class="section level3">
<h3>Step 1: Get only disease studies, and select relevant columns</h3>
<pre class="r"><code>## Step 1: 
# get only disease studies
gwas_study_info &lt;- data.table::fread(here::here(&quot;output/gwas_cat/gwas_study_info_trait_group_l2.csv&quot;))

gwas_study_info = gwas_study_info |&gt;
  dplyr::rename_with(~ gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_study_info =
  gwas_study_info |&gt;
  dplyr::filter(DISEASE_STUDY == T) |&gt;
  dplyr::select(-COHORT)</code></pre>
</div>
<div id="step-2-get-cohort-information" class="section level3">
<h3>Step 2: Get cohort information</h3>
<pre class="r"><code>gwas_study_info_cohort =
  data.table::fread(here::here(&quot;output/gwas_cohorts/gwas_cohort_name_corrected.csv&quot;))

gwas_study_info_cohort =
  gwas_study_info_cohort |&gt;
  dplyr::rename_with(~ gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_study_info_cohort =
  gwas_study_info_cohort |&gt;
  select(STUDY_ACCESSION,
         COHORT) |&gt;
  distinct()

gwas_study_info =
  left_join(gwas_study_info,
            gwas_study_info_cohort,
            by = &quot;STUDY_ACCESSION&quot;
  )</code></pre>
</div>
<div id="step-3-add-ancestrypopulation-info" class="section level3">
<h3>Step 3: Add ancestry/population info</h3>
<pre class="r"><code>gwas_ancest_info &lt;-  data.table::fread(here::here(&quot;data/gwas_catalog/gwas-catalog-v1.0.3.1-ancestries-r2025-07-21.tsv&quot;),
                           sep = &quot;\t&quot;,
                           quote = &quot;&quot;)

gwas_ancest_info = gwas_ancest_info |&gt;
  dplyr::rename_with(~ gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_ancest_info =
  gwas_ancest_info |&gt;
  select(STUDY_ACCESSION,
         BROAD_ANCESTRAL_CATEGORY,
         COUNTRY_OF_RECRUITMENT) |&gt;
  distinct() |&gt;
  group_by(STUDY_ACCESSION) |&gt;
  summarise(
    BROAD_ANCESTRAL_CATEGORY = paste(
      unique(
        unlist(strsplit(BROAD_ANCESTRAL_CATEGORY, split = &quot;\\|&quot;))
      ),
      collapse = &quot;|&quot;
    ),
    COUNTRY_OF_RECRUITMENT = paste(
      unique(
        unlist(strsplit(COUNTRY_OF_RECRUITMENT, split = &quot;\\|&quot;))
      ),
      collapse = &quot;|&quot;
    )
  )

gwas_study_info =
  left_join(gwas_study_info,
            gwas_ancest_info,
            by = &quot;STUDY_ACCESSION&quot;
  )</code></pre>
</div>
<div
id="step-4-extract-information-required-to-get-paper-text-abstracts"
class="section level3">
<h3>Step 4: Extract information required to get paper text &amp;
abstracts</h3>
<pre class="r"><code>gwas_study_info &lt;-
  gwas_study_info |&gt;
  #filter(COHORT != &quot;&quot;) |&gt;
  select(PUBMED_ID,
         COHORT,
         DATE,
         BROAD_ANCESTRAL_CATEGORY,
         COUNTRY_OF_RECRUITMENT) |&gt;
  distinct() |&gt;
  group_by(PUBMED_ID,
           DATE,
           BROAD_ANCESTRAL_CATEGORY,
           COUNTRY_OF_RECRUITMENT) |&gt;
  summarise(
    COHORT = paste(
      unique(
        unlist(strsplit(COHORT, split = &quot;\\|&quot;))
      ),
      collapse = &quot;|&quot;
    )
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;PUBMED_ID&#39;, &#39;DATE&#39;,
&#39;BROAD_ANCESTRAL_CATEGORY&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>pmids = gwas_study_info$PUBMED_ID
cohort = gwas_study_info$COHORT
date = gwas_study_info$DATE
country = gwas_study_info$COUNTRY_OF_RECRUITMENT
ancestry = gwas_study_info$BROAD_ANCESTRAL_CATEGORY

# how many papers without cohort information
gwas_study_info |&gt; 
  filter(COHORT == &quot;&quot;) |&gt;
  nrow()</code></pre>
<pre><code>[1] 4006</code></pre>
<pre class="r"><code># papers for which there is cohort information
gwas_study_info |&gt;
  filter(COHORT != &quot;&quot;) |&gt;
  nrow()</code></pre>
<pre><code>[1] 1626</code></pre>
</div>
<div id="just-take-a-sample" class="section level3">
<h3>Just take a sample:</h3>
<pre class="r"><code># Example PMIDs
rows_with_cohort = which(gwas_study_info$COHORT != &quot;&quot;)
rows_without_cohort = which(gwas_study_info$COHORT == &quot;&quot;)

rows = 
  c(rows_with_cohort, 
    sample(rows_without_cohort, length(rows_with_cohort))
    )

pmids = pmids[rows]

cohort = cohort[rows]
names(cohort) = pmids

date = date[rows]
names(date) = pmids

country = country[rows]
names(country) = pmids

ancestry = ancestry[rows]
names(ancestry) = pmids</code></pre>
</div>
</div>
<div id="getting-cohort-names" class="section level2">
<h2>Getting cohort names</h2>
<pre class="r"><code># this xlsx was built from looking at acrynyms / cohort names in the gwas catalog
# and finding the corresponding full names / details of cohorts

cohort_names &lt;- readxl::read_xlsx(here::here(&quot;data/cohort/cohort_desc.xlsx&quot;),
                                 sheet = 1)</code></pre>
<pre><code>New names:
• `` -&gt; `...13`</code></pre>
<pre class="r"><code>cohort_full_names = cohort_names$full_name[!is.na(cohort_names$full_name)]
cohort_full_names &lt;- str_trim(cohort_full_names)
cohort_full_names &lt;- iconv(cohort_full_names, to = &quot;UTF-8&quot;)
cohort_full_names &lt;- gsub(&quot;[\u00A0\r\n]&quot;, &quot; &quot;, cohort_full_names)  # replace non-breaking spaces, CR, LF with space
cohort_full_names &lt;- str_squish(cohort_full_names)  # trims and removes extra spaces
# sort by length of name (longest first) to match longest names first
cohort_full_names &lt;- cohort_full_names[order(-nchar(cohort_full_names))]
cohort_full_names &lt;- cohort_full_names[cohort_full_names != &quot;Not Reported&quot;]


cohort_abbr_names = cohort_names$cohort[!is.na(cohort_names$cohort)]
cohort_abbr_names &lt;- str_trim(cohort_abbr_names)
cohort_abbr_names &lt;- iconv(cohort_abbr_names, to = &quot;UTF-8&quot;)
cohort_abbr_names &lt;- gsub(&quot;[\u00A0\r\n]&quot;, &quot; &quot;, cohort_abbr_names)  # replace non-breaking spaces, CR, LF with space
cohort_abbr_names &lt;- str_squish(cohort_abbr_names)  # trims and removes extra spaces
# remove abbreviations that are too short
cohort_abbr_names &lt;- cohort_abbr_names[nchar(cohort_abbr_names) &gt;= 4]
small_abbr_to_keep &lt;- c(&quot;C4D&quot;, 
                        &quot;BBJ&quot;, 
                        &quot;UKB&quot;, 
                        &quot;MVP&quot;, 
                        &quot;TWB&quot;, 
                        &quot;QBB&quot;,
                        &quot;MEC&quot;
                        )
cohort_abbr_names &lt;- unique(c(cohort_abbr_names, 
                              small_abbr_to_keep
                              ))
cohort_abbr_names &lt;- cohort_abbr_names[!grepl(&quot;?&quot;, cohort_abbr_names)]
# sort by length of name (longest first) to match longest names first
cohort_abbr_names &lt;- cohort_abbr_names[order(-nchar(cohort_abbr_names))]</code></pre>
</div>
</div>
<div id="relevant-abstract-text" class="section level1">
<h1>Relevant abstract text</h1>
<div id="get-abstracts-from-entrez" class="section level2">
<h2>Get abstracts from Entrez</h2>
<pre class="r"><code>set_entrez_key(Sys.getenv(&#39;NCBI_API_KEY&#39;))

get_pubmed_abstracts &lt;- function(pmids, 
                                 batch_size = 200, 
                                 verbose = TRUE) {
  
  n &lt;- length(pmids)
  
  abstracts &lt;- setNames(rep(&quot;MISSING&quot;, 
                            n), 
                        pmids
                        )  # initialize result
  
  # Split PMIDs into batches
  batches &lt;- split(pmids, 
                   ceiling(seq_along(pmids)/batch_size)
                   )
  
  for(i in seq_along(batches)) {
    
    batch_pmids &lt;- batches[[i]]
    
    if(verbose) message(sprintf(&quot;Fetching batch %d of %d (%d PMIDs)...&quot;, 
                                i, 
                                length(batches), 
                                length(batch_pmids)
                                )
                        )
    
    # Fetch XML
    xml_data &lt;- entrez_fetch(db = &quot;pubmed&quot;, 
                             id = paste(batch_pmids, 
                                        collapse = &quot;,&quot;), 
                             rettype = &quot;xml&quot;, 
                             parsed = FALSE
                             )
    
    doc &lt;- read_xml(xml_data)
    articles &lt;- xml_find_all(doc, &quot;.//PubmedArticle&quot;)
    
    for(article in articles) {
      
      pmid_node &lt;- xml_find_first(article, 
                                  &quot;.//PMID&quot;)
      
      pmid &lt;- xml_text(pmid_node)
      
      abstract_nodes &lt;- xml_find_all(article, 
                                     &quot;.//AbstractText&quot;)
      
      if(length(abstract_nodes) &gt; 0) {
        abstracts[pmid] &lt;- paste(xml_text(abstract_nodes), 
                                 collapse = &quot; &quot;)
      }
    }
  }
  
  return(abstracts)
}

pmids = unique(pmids)
abstracts &lt;- get_pubmed_abstracts(pmids)

pmids &lt;- pmids[abstracts != &quot;MISSING&quot;]
date &lt;- date[abstracts != &quot;MISSING&quot;]
cohort &lt;- cohort[abstracts != &quot;MISSING&quot;]
abstracts &lt;- abstracts[abstracts != &quot;MISSING&quot;]

# Loop through abstracts and write each to a file
for (i in seq_along(abstracts)) {
  file_name &lt;- paste0(here::here(&quot;output/abstracts/&quot;), pmids[i], &quot;.txt&quot;)
  writeLines(abstracts[i], file_name)
}</code></pre>
<div id="check-what-abstracts-are-missing" class="section level3">
<h3>Check: What abstracts are missing?</h3>
<pre class="r"><code>abstract_files &lt;- list.files(here::here(&quot;output/abstracts/&quot;), 
                             pattern = &quot;*.txt&quot;, 
                             full.names = FALSE
                             )

pmids_with_abstracts = gsub(&quot;\\.txt$&quot;, &quot;&quot;, abstract_files)

all_pmids = gwas_study_info$PUBMED_ID |&gt; unique()

missing_abstracts = setdiff(all_pmids,
                             pmids_with_abstracts)

print(&quot;Number of missing abstracts:&quot;)</code></pre>
<pre><code>[1] &quot;Number of missing abstracts:&quot;</code></pre>
<pre class="r"><code>length(missing_abstracts)</code></pre>
<pre><code>[1] 42</code></pre>
<pre class="r"><code>library(openalexR)

# get information on these papers from openAlex
oa_example &lt;- 
oa_fetch(entity = &quot;works&quot;,
         pmid = missing_abstracts,
         abstract = TRUE)

oa_example |&gt;
  select(doi, abstract) |&gt;
  distinct()</code></pre>
<pre><code># A tibble: 41 × 2
   doi                                             abstract                     
   &lt;chr&gt;                                           &lt;chr&gt;                        
 1 https://doi.org/10.1038/s41431-020-0636-6        &lt;NA&gt;                        
 2 https://doi.org/10.1038/s41586-022-04826-7       &lt;NA&gt;                        
 3 https://doi.org/10.1038/mp.2009.107              &lt;NA&gt;                        
 4 https://doi.org/10.1016/j.jaci.2015.01.047       &lt;NA&gt;                        
 5 https://doi.org/10.1002/art.23603                &lt;NA&gt;                        
 6 https://doi.org/10.1016/j.jaci.2010.06.051       &lt;NA&gt;                        
 7 https://doi.org/10.1176/appi.ajp.2013.12091228   &lt;NA&gt;                        
 8 https://doi.org/10.1164/rccm.202206-1139le      &quot;\&quot;Identification of a Genet…
 9 https://doi.org/10.1161/circgenetics.111.960989  &lt;NA&gt;                        
10 https://doi.org/10.4088/jcp.15l10127            &quot;Article Abstract Because th…
# ℹ 31 more rows</code></pre>
</div>
</div>
<div id="read-in-abstracts-from-files" class="section level2">
<h2>Read in abstracts from files</h2>
<pre class="r"><code>abstracts &lt;- sapply(abstract_files,
                    function(file) {
                      readLines(here::here(paste0(&quot;output/abstracts/&quot;,file)), 
                                warn = FALSE) |&gt; 
                        paste(collapse = &quot; &quot;)
                    }
                    )

pmids = pmids_with_abstracts 
cohort = cohort[pmids]
date = date[pmids]
country = country[pmids]</code></pre>
</div>
<div id="extract-sentences-from-abstracts-with-cohort-names"
class="section level2">
<h2>Extract sentences from abstracts with cohort names</h2>
<pre class="r"><code>extract_cohort_sentences &lt;- function(text_vector, 
                                     cohort_names, 
                                     ignore_case) {
  
  
  results &lt;- lapply(seq_along(text_vector), function(i) {
    
    abstract &lt;- text_vector[i]
    # Split abstract into sentences
    sentences &lt;- unlist(tokenize_sentences(abstract))

    # For each sentence, find all matching cohort names
    lapply(seq_along(sentences), function(s) {
      
      sentence &lt;- sentences[s]
      
      # Identify cohort names present in this sentence
      matched_cohorts &lt;- cohort_names[str_detect(sentence, 
                                                 regex(cohort_names, 
                                                       ignore_case = ignore_case))]

      data.frame(
        abstract_id = i,
        sentence_id = s,
        sentence = sentence,
        has_cohort = length(matched_cohorts) &gt; 0,
        COHORT = if (length(matched_cohorts) &gt; 0) str_flatten(unique(matched_cohorts), collapse = &quot;|&quot;, na.rm = T) else &quot;&quot;,
        stringsAsFactors = FALSE
      )
    }) |&gt; bind_rows()
  })

  bind_rows(results)
}

cohort_sentences_df &lt;- extract_cohort_sentences(abstracts,
                                                cohort_full_names,
                                                ignore_case = TRUE
                                                )

cohort_sentences_df_p2 = extract_cohort_sentences(abstracts,
                                                  cohort_abbr_names,
                                                  ignore_case = FALSE
                                                  )

cohort_sentences_df =
  bind_rows(cohort_sentences_df,
            cohort_sentences_df_p2
            )

cohort_sentences_df = 
  cohort_sentences_df |&gt;
  distinct()</code></pre>
</div>
<div
id="correct-for-the-over-representation-of-sentences-without-cohort-names"
class="section level2">
<h2>Correct for the over-representation of sentences without cohort
names</h2>
<pre class="r"><code># set the ratio of sentences with/without cohort names
# ratio = 1
# 
# # number of sentences with cohort names
# n_tp_sentences = nrow(cohort_sentences_df |&gt;
#                       filter(COHORT != &quot;&quot;)
#                       )
# cohort_sentences_df  =
#   cohort_sentences_df |&gt;
#   mutate(has_cohort = ifelse(COHORT == &quot;&quot;, FALSE, TRUE)) |&gt;
#   group_by(has_cohort) |&gt;
#   slice_sample(n = ratio*n_tp_sentences,
#                replace = FALSE
#                ) 

filtered_cohort_sentences_df =
  cohort_sentences_df |&gt;
  filter(COHORT != &quot;&quot;)

control_cohort_sentences_df &lt;-
cohort_sentences_df |&gt;
  filter(grepl(&quot;cohort|consortium|study|population|registry|biobank|corsortia&quot;, 
               sentence, 
               ignore.case = TRUE)
         ) |&gt;
  slice_sample(n = 1000)

filtered_cohort_sentences_df =
  bind_rows(filtered_cohort_sentences_df,
            control_cohort_sentences_df
            ) |&gt;
  distinct()

filtered_cohort_sentences_df =
  filtered_cohort_sentences_df |&gt;
  group_by(abstract_id, sentence_id, sentence) |&gt;
  summarise(
    COHORT = str_flatten(unique(COHORT), collapse = &quot;|&quot;, na.rm = T)
  ) |&gt;
  ungroup()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;abstract_id&#39;, &#39;sentence_id&#39;. You can
override using the `.groups` argument.</code></pre>
<pre class="r"><code>abstract_ids = filtered_cohort_sentences_df$abstract_id |&gt; unique() |&gt; sort()

pmids = pmids[abstract_ids]
abstracts = abstracts[abstract_ids]
date = date[abstract_ids]
cohort = cohort[abstract_ids]</code></pre>
</div>
<div id="prepare-data-in-doccano-json-format" class="section level2">
<h2>Prepare data in Doccano JSON format</h2>
<pre class="r"><code># file path for intermediate json file output
json_file = here::here(&quot;output/doccano/abstracts_with_cohort_info.json&quot;)

# file path for final jsonl file output
jsonl_file = here::here(&quot;output/doccano/abstracts_with_cohort_info.jsonl&quot;)

convert_to_doccano_json_sentence_level &lt;- function(pmids,
                                                   date,
                                                   cohort,
                                                   country,
                                                   cohort_sentences_df) {
  
  # set up json list
  doccano_list &lt;- list()
  example_id &lt;- 1

  for(current_sentence in cohort_sentences_df$sentence) {

      # Filter cohort sentences that match this sentence (safe matching)
      df &lt;- cohort_sentences_df |&gt;
             dplyr::filter(sentence == current_sentence) 
      
      # abstract_id &lt;- df$abstract_id
      # matched_cohort &lt;- df$COHORT
      
      for (i in seq_len(nrow(df))) {
        
        matched_cohort &lt;- df$COHORT[i]
        abstract_id &lt;- df$abstract_id[i]
        
        #browser()
      
        if(matched_cohort == &quot;&quot;){
        
          doccano_list[[example_id]] &lt;- list(
          text = current_sentence,
          pubmed_id = pmids[abstract_id],
          date = date[abstract_id],
          country = country[abstract_id],
          gwas_cat_cohort_label = cohort[abstract_id],
          label = list()
          )
        
        } else {
        
        # Find the location of all matches of the cohort name in the sentence
        if(grepl(&quot;\\|&quot;, matched_cohort)) {
           
           # If multiple cohort names, separate
           matched_cohort &lt;- unlist(
                                    strsplit(matched_cohort, 
                                             split = &quot;\\|&quot;)
                                    )
        
           match_locations &lt;- list()
        
           for(current_matched_cohort in matched_cohort){
          
               matches &lt;- str_locate_all(current_sentence,
                                         fixed(current_matched_cohort, 
                                               ignore_case = TRUE)
                                         )[[1]]
          
               match_locations &lt;- append(match_locations, 
                                         list(matches)
                                         )
          
           }
        
           # Combine all match locations into a single matrix
           matches &lt;- do.call(rbind, match_locations)
        
        }  else {
        
           matches &lt;- str_locate_all(current_sentence,
                                     fixed(matched_cohort, 
                                           ignore_case = TRUE)
                                     )[[1]]
      
      }
      
      # Convert matches to 0-based indexing (for doccano)
      matches[, &quot;start&quot;] &lt;- matches[, &quot;start&quot;] - 1  
      
      # Turn match locations into entity list
      entities &lt;- list()
      
      for(k in seq_len(nrow(matches))) {
              entities &lt;- append(entities, list(list(
                start_offset = matches[k, &quot;start&quot;],
                end_offset = matches[k, &quot;end&quot;],
                label = &quot;COHORT&quot;
              )))
      }
      
      # Create Doccano JSON entry
      doccano_list[[example_id]] &lt;- list(
        # id = example_id,
        text = current_sentence,
        pubmed_id = pmids[abstract_id],
        date = date[abstract_id],
        country = country[abstract_id],
        gwas_cat_cohort_label = cohort[abstract_id],
        label = entities
      )
      
      }
      }

      example_id &lt;- example_id + 1
  }
  
  
# Suppose each element of json_list has &#39;labels&#39; as a list of named lists
  doccano_list &lt;- lapply(doccano_list, function(x) {
  x$label &lt;- lapply(x$label, function(l) {
    # convert named list to vector/list format [start, end, label]
    c(l$start_offset, l$end_offset, l$label)
  })
  x
})

  return(doccano_list)
}




# Create JSON as a list
json_list &lt;- convert_to_doccano_json_sentence_level(pmids = pmids,
                                                    date = date,
                                                    cohort = cohort,
                                                    country = country,
                                                    filtered_cohort_sentences_df)



# Write to JSON file
writeLines(toJSON(json_list,
                  auto_unbox = TRUE,
                  pretty = TRUE),
           json_file)


json_data &lt;- fromJSON(json_file,
                      simplifyVector = FALSE)

# Open connection to JSONL file
con &lt;- file(jsonl_file, &quot;w&quot;)

# Loop over each element (object) and write as one line
for (i in seq_along(json_data)) {
  writeLines(toJSON(json_data[[i]], auto_unbox = TRUE), con)
}

# Close connection
close(con)

cat(&quot;JSONL saved to:&quot;, jsonl_file, &quot;\n&quot;)</code></pre>
<pre><code>JSONL saved to: /Users/ibeasley/code/genomics_ancest_disease_dispar/output/doccano/abstracts_with_cohort_info.jsonl </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] openalexR_2.0.1  tokenizers_0.3.0 jsonlite_2.0.0   xml2_1.4.0      
 [5] rentrez_1.2.4    stringi_1.8.7    dplyr_1.1.4      readxl_1.4.5    
 [9] stringr_1.5.2    workflowr_1.7.1 

loaded via a namespace (and not attached):
 [1] compiler_4.3.1    renv_1.0.3        promises_1.3.3    tidyselect_1.2.1 
 [5] Rcpp_1.1.0        git2r_0.36.2      callr_3.7.6       later_1.4.4      
 [9] jquerylib_0.1.4   yaml_2.3.10       fastmap_1.2.0     here_1.0.1       
[13] R6_2.6.1          SnowballC_0.7.1   generics_0.1.4    curl_7.0.0       
[17] knitr_1.50        XML_3.99-0.19     tibble_3.3.0      rprojroot_2.1.0  
[21] bslib_0.9.0       pillar_1.11.1     rlang_1.1.6       utf8_1.2.6       
[25] cachem_1.1.0      httpuv_1.6.16     xfun_0.53         getPass_0.2-4    
[29] fs_1.6.6          sass_0.4.10       cli_3.6.5         withr_3.0.2      
[33] magrittr_2.0.4    ps_1.9.1          digest_0.6.37     processx_3.8.6   
[37] rstudioapi_0.17.1 lifecycle_1.0.4   vctrs_0.6.5       data.table_1.17.8
[41] evaluate_1.0.5    glue_1.8.0        whisker_0.4.1     cellranger_1.1.0 
[45] rmarkdown_2.30    httr_1.4.7        tools_4.3.1       pkgconfig_2.0.3  
[49] htmltools_0.5.8.1</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
