<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-08-24" />

<title>GWAS Trait Categorisation</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GWAS Trait Categorisation</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-08-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree8a523250a094a3c7ff324e21f877acbf20603042targetblank8a52325a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/8a523250a094a3c7ff324e21f877acbf20603042" target="_blank">8a52325</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartree8a523250a094a3c7ff324e21f877acbf20603042targetblank8a52325a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/8a523250a094a3c7ff324e21f877acbf20603042" target="_blank">8a52325</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/gwas_catalog/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l1.csv
    Ignored:    output/gwas_study_info_trait_ontology_info_l2.csv
    Ignored:    output/trait_ontology/
    Ignored:    renv/

Untracked files:
    Untracked:  analysis/disease_burden.qmd
    Untracked:  data/gbd/
    Untracked:  data/who/

Unstaged changes:
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/level_2_disease_group.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/trait_ontology_collapse.Rmd</code>) and HTML
(<code>docs/trait_ontology_collapse.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/8a523250a094a3c7ff324e21f877acbf20603042/analysis/trait_ontology_collapse.Rmd" target="_blank">8a52325</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-09
</td>
<td>
Woops fix typo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/ec6dfc29b9c1b35caa8bd411eb5d40eaae58a25b/docs/trait_ontology_collapse.html" target="_blank">ec6dfc2</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/c096bdd318cf50032fa6ddb8186d960f72ff32dc/analysis/trait_ontology_collapse.Rmd" target="_blank">c096bdd</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-09
</td>
<td>
Fixing disease terms with commas in them
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/cc7742884dea0d9737324869302e0aa203976701/docs/trait_ontology_collapse.html" target="_blank">cc77428</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/0c1ee67f2860d4fd5f491ae36602df76361b51fc/analysis/trait_ontology_collapse.Rmd" target="_blank">0c1ee67</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-09
</td>
<td>
Include phenotypic abornomality in disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/6867f0cbcdcce338a9c941edb1471bc132dec07f/docs/trait_ontology_collapse.html" target="_blank">6867f0c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/7c4f4af631daa17f51fa3626ee964b36a694fa33/analysis/trait_ontology_collapse.Rmd" target="_blank">7c4f4af</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-05
</td>
<td>
Fixing / updating trait ontology collapse
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/5eee83c71ad5ed6034d13b1a19be44322ee01989/docs/trait_ontology_collapse.html" target="_blank">5eee83c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b710a4db85001c23f15255b79a1fade1dccab8d2/analysis/trait_ontology_collapse.Rmd" target="_blank">b710a4d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-25
</td>
<td>
Update trait ontology investigation to group disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/243218e2a8094d602e56712012c41902986dbe84/docs/trait_ontology_collapse.html" target="_blank">243218e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/6ceba65f7523468737192025be09f5703416d3b5/analysis/trait_ontology_collapse.Rmd" target="_blank">6ceba65</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-24
</td>
<td>
workflowr::wflow_publish("analysis/trait_ontology_collapse.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Set up</h1>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, 
                      message = FALSE, 
                      warning = FALSE
                      )

library(data.table)
library(dplyr)
library(ggplot2)
library(stringr)</code></pre>
</div>
<div id="overlap-ontology-terms-and-gwas-traits" class="section level1"
number="2">
<h1><span class="header-section-number">2</span> Overlap ontology terms
and GWAS traits</h1>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_study_info_cohort_corrected.csv&quot;))

# fixing weird terms- where comma is in the term
# gwas_study_info |&gt; 
#  mutate(n_commas_trait = stringr::str_count(MAPPED_TRAIT, &quot;, &quot;),
#         n_commas_uri = stringr::str_count(MAPPED_TRAIT_URI, &quot;,&quot;)) |&gt;
#   filter(n_commas_trait != n_commas_uri)


gwas_study_info =
gwas_study_info |&gt; 
  rowwise() |&gt;
  mutate(MAPPED_TRAIT = case_when(
                         # osteoarthritis, hip ... http://www.ebi.ac.uk/efo/EFO_1000786
                         grepl(&quot;EFO_1000786&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;osteoarthritis, hip&quot;, &quot;hip osteoarthritis&quot;),
                         #  osteoarthritis, hand ... http://www.ebi.ac.uk/efo/EFO_1000789
                         grepl(&quot;EFO_1000789&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;osteoarthritis, hand&quot;, &quot;hand osteoarthritis&quot;),
                         #  osteoarthritis, spine ... http://www.ebi.ac.uk/efo/EFO_1000787
                         grepl(&quot;EFO_1000787&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;osteoarthritis, spine&quot;, &quot;spine osteoarthritis&quot;),
                         #  Hepatitis, Alcoholic, http://www.ebi.ac.uk/efo/EFO_1001345
                         grepl(&quot;EFO_1001345&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;Hepatitis, Alcoholic&quot;, &quot;Alcoholic Hepatitis&quot;),
                         # psoriasis 14, pustular http://purl.obolibrary.org/obo/MONDO_0013626
                         grepl(&quot;MONDO_0013626&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;psoriasis 14, pustula&quot;, &quot;pustular psoriasis 14&quot;),
                         # hypertension, pregnancy-induced http://purl.obolibrary.org/obo/MONDO_0024664
                         grepl(&quot;MONDO_0024664&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;hypertension, pregnancy-induced&quot;, &quot;pregnancy-induced hypertension&quot;),
                         # renal agenesis, unilateral http://purl.obolibrary.org/obo/MONDO_0019636
                         grepl(&quot;MONDO_0019636&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;renal agenesis, unilateral&quot;, &quot;unilateral renal agenesis&quot;),
                         #  Cholecystitis, Acute http://www.ebi.ac.uk/efo/EFO_1001289
                         grepl(&quot;EFO_1001289&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;Cholecystitis, Acute&quot;, &quot;Acute Cholecystitis&quot;),
                         #  Genital neoplasm, female http://www.ebi.ac.uk/efo/EFO_1001331
                         grepl(&quot;EFO_1001331&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;Genital neoplasm, female&quot;, &quot;female reproductive organ cancer&quot;),
                         #  Anemia, Hemolytic, Autoimmune http://www.ebi.ac.uk/efo/EFO_1001264
                          grepl(&quot;EFO_1001264&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;Anemia, Hemolytic, Autoimmune&quot;, &quot;autoimmune haemolytic anemia&quot;),
                         TRUE ~ MAPPED_TRAIT
                         )
  ) |&gt;
  ungroup()

# osteoarthritis, knee ... http://www.ebi.ac.uk/efo/EFO_0004616
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(
                         grepl(&quot;EFO_0004616&quot;, MAPPED_TRAIT_URI),
                         stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;osteoarthritis, knee&quot;, &quot;knee osteoarthritis&quot;),
                         MAPPED_TRAIT
  )
  ) 

# rheumatoid factor-negative juvenile idiopathic arthritis
# polyarticular juvenile idiopathic arthritis, rheumatoid factor negative
# http://www.ebi.ac.uk/efo/EFO_1002020
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;EFO_1002020&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;polyarticular juvenile idiopathic arthritis, rheumatoid factor negative&quot;, 
                                                        &quot;rheumatoid factor-negative juvenile idiopathic arthritis&quot;),
                               MAPPED_TRAIT)
         )

# http://www.ebi.ac.uk/efo/EFO_0007294, hand, foot and mouth disease,
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;EFO_0007294&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;hand, foot and mouth disease&quot;, 
                                                        &quot;hand foot and mouth disease&quot;),
                               MAPPED_TRAIT)
         )

# susceptibility to migriane without aura http://purl.obolibrary.org/obo/MONDO_0011847 

gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;MONDO_0011847&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;migraine without aura, susceptibility to, 4&quot;, &quot;migraine without aura&quot;),
                               MAPPED_TRAIT)
         )

#   neural tube defects, susceptibility to, http://purl.obolibrary.org/obo/MONDO_0020705
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;MONDO_0020705&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;neural tube defects, susceptibility to&quot;, 
                                                        &quot;neural tube defects&quot;),
                               MAPPED_TRAIT)
         )

# infantile diarrhea
# Diarrhea, Infantile http://www.ebi.ac.uk/efo/EFO_1001306
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;EFO_1001306&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;Diarrhea, Infantile&quot;, 
                                                        &quot;infantile diarrhea&quot;),
                               MAPPED_TRAIT)
         )

# self reported traits 
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;EFO_0009803|EFO_0009822|EFO_0009803|EFO_0009817|EFO_0009822|EFO_0009819|EFO_0009823|EFO_0009824&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;, self-reported$&quot;, 
                                                        &quot; self-reported&quot;),
                               MAPPED_TRAIT)
         )


# Hodgkins lymphoma, mixed cellularity http://www.ebi.ac.uk/efo/EFO_1002031
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;EFO_1002031&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, pattern = &quot;Hodgkins lymphoma, mixed cellularity&quot;, &quot;Hodgkins lymphoma mixed cellularity&quot;),
                               MAPPED_TRAIT)
         )

# encephalopathy, acute, infection-induced, http://purl.obolibrary.org/obo/MONDO_0000166
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;MONDO_0000166&quot;, MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;encephalopathy, acute, infection-induced&quot;, 
                                                        &quot;encephalopathy acute infection-induced&quot;),
                               MAPPED_TRAIT)
         )


gwas_study_info = 
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = stringr::str_replace_all(MAPPED_TRAIT, 
                                                 &quot;^level of .+, mitochondrial in blood$&quot;, &quot;blood protein amount&quot;)
         )</code></pre>
<pre class="r"><code>all_gwas_terms = unique(gwas_study_info$MAPPED_TRAIT)

all_gwas_terms = stringr::str_trim(tolower(all_gwas_terms))</code></pre>
<div
id="disease-overlap-how-many-gwas-traits-fall-within-disease-or-disorder-terms"
class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Disease Overlap (How
many GWAS traits fall within disease or disorder terms?)</h2>
<div id="combine-disease-terms" class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Combine disease
terms</h3>
<pre class="r"><code>efo_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000408_descendants.txt&quot;))

mondo_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/mondo_0700096_descendants.txt&quot;))

ncit_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/ncit_C2991_descendants.txt&quot;))

orphanet_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/orphanet_557493_descendants.txt&quot;))

age_of_onset_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/oba_2020000_descendants.txt&quot;))

disease_measurement_terms &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0001444_disease_measurement_terms.txt&quot;))

disease_typos = c(&quot;Alzheimer disease&quot;,
                  &quot;late-onset Alzheimers disease&quot;,
                  &quot;age of onset of Alzheimer disease&quot;,
                  &quot;Chagas cardiomyopathy&quot;,
                  &quot;Parkinson disease&quot;,
                  &quot;Iron deficiency anemia&quot;,
                  &quot;Churg-Strauss syndrome&quot;,
                  &quot;Iridocyclitis&quot;,
                  &quot;Phlebitis&quot; 
                  )

other &lt;- c(&quot;Allergic disease&quot;, 
  &quot;Lewy body dementia&quot;,
  &quot;Lewy body attribute&quot;,
 &quot;non-Hodgkins lymphoma&quot;,
           &quot;Ischemic Stroke&quot;,
           &quot;Lung disease&quot;,
           &quot;Respiratory System Disease&quot;,
  &quot;Iron deficiency anemia (disorder)&quot;,
  &quot;Alzheimer disease, APOE carrier status&quot;,
  &quot;Alzheimer&#39;s disease biomarker measurement&quot;,
 &quot;Genital neoplasm, female&quot;,
 &quot;HIV-associated neurocognitive disorder&quot;,
 &quot;encephalopathy acute infection-induced&quot;
           )

disease_terms = c(mondo_descendants,
                  efo_descendants,
                  ncit_descendants,
                  orphanet_descendants,
                  age_of_onset_descendants,
                  disease_measurement_terms,
                  disease_typos,
                  other) |&gt;
                 unique()


disease_terms = stringr::str_trim(tolower(disease_terms))

print(&quot;Number of terms related to disease or disorder&quot;)</code></pre>
<pre><code>[1] &quot;Number of terms related to disease or disorder&quot;</code></pre>
<pre class="r"><code>length(disease_terms)</code></pre>
<pre><code>[1] 53397</code></pre>
<pre class="r"><code># Find GWAS traits that fall within disease or disorder terms
simple_disease_terms = all_gwas_terms[all_gwas_terms %in% disease_terms]

# Also search for cases where there are multiple terms separated by commas
# and one of them is a disease term
not_simple_disease_terms = all_gwas_terms[!all_gwas_terms %in% disease_terms]

# sometimes there&#39;s multiple terms - check if any disease term is in these gwas terms
multiple_terms = grep(&quot;,&quot;, not_simple_disease_terms, value = T)

disease_chunks &lt;- split(disease_terms, ceiling(seq_along(disease_terms) / 100))
disease_chunks  &lt;- lapply(disease_chunks, function(x) paste0(x, collapse = &quot;|&quot;))
mask &lt;- Reduce(`|`, lapply(disease_chunks, function(x) grepl(x, multiple_terms, ignore.case = T)))
additional_disease_gwas &lt;- multiple_terms[mask]

disease_gwas = c(all_gwas_terms[all_gwas_terms %in% disease_terms],
                 additional_disease_gwas)

not_disease_terms = not_simple_disease_terms[!not_simple_disease_terms %in% additional_disease_gwas]

print(&quot;Number of GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>length(all_gwas_terms) - length(not_disease_terms)</code></pre>
<pre><code>[1] 3501</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * (length(all_gwas_terms) - length(not_disease_terms)) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 15.3</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits not under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_disease_terms) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 84.7</code></pre>
<pre class="r"><code>not_accounted_for = not_disease_terms </code></pre>
</div>
</div>
<div id="phenotype-abnormality-overlap" class="section level2"
number="2.2">
<h2><span class="header-section-number">2.2</span> Phenotype abnormality
overlap</h2>
<pre class="r"><code>pheno_abnorm &lt;- readLines(here::here(&quot;output/trait_ontology/hp_0000118_descendants.txt&quot;))
pheno_abnorm = stringr::str_trim(tolower(pheno_abnorm))

# Find terms where all comma-split pieces are in measurement
pheno_abnorm_gwas &lt;- not_accounted_for[
  sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts) # remove extra spaces
    all(parts %in% pheno_abnorm)
  })
]

additional_pheno_abnorm &lt;- not_accounted_for[not_accounted_for %in% pheno_abnorm]

pheno_abnorm_gwas  = c(pheno_abnorm_gwas, additional_pheno_abnorm) |&gt; unique()

print(&quot;Percentage of GWAS traits under phenotype abnormality terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under phenotype abnormality terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(pheno_abnorm_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.5</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% pheno_abnorm_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 83.2</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 18990</code></pre>
</div>
<div
id="add-disease-phenotypic-abnormality-terms-to-gwas-study-info-dataset"
class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Add disease &amp;
phenotypic abnormality terms to GWAS study info dataset</h2>
<pre class="r"><code>find_disease_terms  &lt;-   function(MAPPED_TRAIT) {
        # find all disease terms that appear in the trait
        split_mapped_traits &lt;- stringr::str_split(MAPPED_TRAIT, &quot;, &quot;) |&gt; 
                               unlist()
        
        mapped_disease_terms &lt;- split_mapped_traits[split_mapped_traits %in% disease_terms]
        mapped_pheno_abnorm_terms &lt;- split_mapped_traits[split_mapped_traits %in% pheno_abnorm]
        
        mapped_disease_terms = unique(c(mapped_disease_terms, 
                                        mapped_pheno_abnorm_terms
                                        )
        )
                                      
        return(paste0(mapped_disease_terms, collapse = &quot;, &quot;))  # combine multiple matches
        
    }

gwas_study_info &lt;- 
  gwas_study_info |&gt; 
  dplyr::rowwise() |&gt;
  dplyr::mutate(
    disease_terms = 
      ifelse(stringr::str_trim(tolower(MAPPED_TRAIT)) %in% c(disease_gwas,pheno_abnorm_gwas),
             find_disease_terms(stringr::str_trim(tolower(MAPPED_TRAIT))),
             NA)
  )

gwas_study_info &lt;- 
  gwas_study_info |&gt;
  rowwise() |&gt;
  dplyr::mutate(
   disease_terms = 
      ifelse(MAPPED_TRAIT == &quot;&quot;,
             NA,
            disease_terms)
  ) |&gt;
  ungroup()</code></pre>
</div>
<div
id="measurement-overlap-how-many-gwas-traits-fall-within-measurement-terms"
class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Measurement Overlap
(how many GWAS traits fall within measurement terms?)</h2>
<div id="combine-measurement" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Combine
measurement</h3>
<pre class="r"><code>measurement &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0001444_descendants.txt&quot;))
total_choles &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0004574_descendants.txt&quot;))

measurement &lt;- c(total_choles,
                 measurement) 

measurement &lt;- unique(measurement)

measurement &lt;- c(&quot;cerebrospinal fluid composition attribute&quot;,
                 &quot;blood protein amount&quot;,
                 measurement)

measurement = stringr::str_trim(tolower(measurement))</code></pre>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
measurement_gwas &lt;- not_accounted_for[
  sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts)
    all(parts %in% measurement)
  })
]
additional_measurement &lt;- not_accounted_for[not_accounted_for %in% measurement]

measurement_gwas  = c(measurement_gwas, additional_measurement) |&gt; unique()

print(&quot;Number of GWAS traits under measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under measurement terms&quot;</code></pre>
<pre class="r"><code>length(measurement_gwas)</code></pre>
<pre><code>[1] 18068</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(measurement_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 79.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% measurement_gwas]

print(&quot;Percentage of GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 921</code></pre>
</div>
</div>
<div id="response-to-stimulus" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Response to
stimulus</h2>
<div id="combine-response-terms" class="section level3" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Combine response
terms</h3>
<pre class="r"><code>go_response = readLines(here::here(&quot;output/trait_ontology/go_0050896_descendants.txt&quot;))

efo_response &lt;- readLines(here::here(&quot;output/trait_ontology/efo_go_0050896_descendants.txt&quot;))

response &lt;- c(go_response,
              efo_response,
              &quot;response to stimulus&quot;)

response &lt;- unique(response)

response = stringr::str_trim(tolower(response))</code></pre>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
response_gwas &lt;- not_accounted_for[
  sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts)
    all(parts %in% response)
  })
]
additional_response &lt;- not_accounted_for[not_accounted_for %in% response]

measurement_gwas  = c(measurement_gwas, additional_response) |&gt; unique()

print(&quot;Percentage of GWAS traits under response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(response_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.7</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% response_gwas]

print(&quot;Percentage of GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.4</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 768</code></pre>
</div>
</div>
<div id="mental-process" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Mental process</h2>
<pre class="r"><code>mental &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0004323_descendants.txt&quot;))
mental = stringr::str_trim(tolower(mental))

mental_gwas = not_accounted_for[not_accounted_for %in% mental]

print(&quot;Percentage of GWAS traits under mental process terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under mental process terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(mental_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% mental_gwas]

print(&quot;Percentage of GWAS traits not accounted for thus far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for thus far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for thus far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for thus far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 750</code></pre>
</div>
<div id="behavior" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Behavior</h2>
<pre class="r"><code>behavior &lt;- readLines(here::here(&quot;output/trait_ontology/go_0007610_descendants.txt&quot;))
behavior = stringr::str_trim(tolower(behavior))

behavior_gwas = not_accounted_for[not_accounted_for %in% behavior]

print(&quot;Percentage of GWAS traits under behavouir terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under behavouir terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(behavior_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% behavior_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.2</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 727</code></pre>
</div>
<div id="injury" class="section level2" number="2.8">
<h2><span class="header-section-number">2.8</span> Injury</h2>
<pre class="r"><code>injury &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000546_descendants.txt&quot;))

injury = stringr::str_trim(tolower(injury))

injury_gwas = not_accounted_for[not_accounted_for %in% injury]

print(&quot;Percentage of GWAS traits under injury terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under injury terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(injury_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% injury_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.1</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 708</code></pre>
</div>
<div id="phenotype" class="section level2" number="2.9">
<h2><span class="header-section-number">2.9</span> Phenotype</h2>
<pre class="r"><code>phenotype &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000651_descendants.txt&quot;))

phenotype = stringr::str_trim(tolower(phenotype))

phenotype_gwas = not_accounted_for[not_accounted_for %in% phenotype]

print(&quot;Percentage of GWAS traits under phenotype terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under phenotype terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(phenotype_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.2</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% phenotype_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 2.9</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 661</code></pre>
</div>
</div>
<div id="add-categories-to-gwas-info" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Add Categories to GWAS
Info</h1>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::mutate(MAPPED_TRAIT_CATEGORY = dplyr::case_when(is.na(MAPPED_TRAIT) ~ NA,
                                                         tolower(MAPPED_TRAIT) %in% disease_gwas ~ &quot;Disease/Disorder&quot;,
                                                         tolower(MAPPED_TRAIT) %in% pheno_abnorm_gwas ~ &quot;Phenotypic Abnormality&quot;,
                                                         tolower(MAPPED_TRAIT) %in% measurement_gwas ~ &quot;Measurement&quot;,
                                                         tolower(MAPPED_TRAIT) %in% response_gwas ~ &quot;Response&quot;,
                                                         tolower(MAPPED_TRAIT) %in% mental_gwas ~ &quot;Mental Process&quot;,
                                                         tolower(MAPPED_TRAIT) %in% behavior_gwas ~ &quot;Behavior&quot;,
                                                         tolower(MAPPED_TRAIT) %in% injury_gwas ~ &quot;Injury&quot;,
                                                         tolower(MAPPED_TRAIT) %in% phenotype_gwas ~ &quot;Phenotype&quot;,
                                                          TRUE ~ &quot;Other&quot;
                                                          )
                )</code></pre>
</div>
<div id="background-traits" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Background traits</h1>
<pre class="r"><code>gwas_study_info$MAPPED_BACKGROUND_TRAIT |&gt; unique() -&gt; gwas_background

gwas_background = stringr::str_trim(tolower(gwas_background))

length(gwas_background)</code></pre>
<pre><code>[1] 314</code></pre>
<div id="overlap-with-diseasedisorder-traits" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> Overlap with
disease/disorder traits</h2>
<pre class="r"><code>multiple_terms = grep(&quot;,&quot;, gwas_background, value = T)
mask &lt;- Reduce(`|`, lapply(disease_terms, function(x) grepl(x, multiple_terms)))
additional_disease_gwas &lt;- multiple_terms[mask]

disease_gwas = c(gwas_background[gwas_background %in% disease_terms],
                 additional_disease_gwas)

print(&quot;Number of background GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>length(disease_gwas)</code></pre>
<pre><code>[1] 229</code></pre>
<pre class="r"><code>print(&quot;Percentage of background GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(disease_gwas) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 72.9</code></pre>
<pre class="r"><code>not_accounted_for = gwas_background[!gwas_background %in% disease_gwas]</code></pre>
<pre class="r"><code>gwas_study_info &lt;- 
  gwas_study_info |&gt;
  rowwise() |&gt;
  dplyr::mutate(
    background_disease_terms = 
      ifelse(stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% c(disease_gwas, pheno_abnorm_gwas),
             find_disease_terms(stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT))),
             NA)
  ) |&gt;
  ungroup()


gwas_study_info &lt;- 
  gwas_study_info |&gt;
  rowwise() |&gt;
  dplyr::mutate(
    background_disease_terms = 
      ifelse(MAPPED_BACKGROUND_TRAIT == &quot;&quot;,
             NA,
             background_disease_terms)
  ) |&gt;
  ungroup()</code></pre>
</div>
<div id="phenotype-abnormality-overlap-1" class="section level2"
number="4.2">
<h2><span class="header-section-number">4.2</span> Phenotype abnormality
overlap</h2>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
pheno_abnorm_gwas &lt;- not_accounted_for[
  sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts) # remove extra spaces
    all(parts %in% pheno_abnorm)
  })
]

additional_pheno_abnorm &lt;- not_accounted_for[not_accounted_for %in% pheno_abnorm]

pheno_abnorm_gwas  = c(pheno_abnorm_gwas, additional_pheno_abnorm) |&gt; unique()

print(&quot;Percentage of background GWAS traits under phenotype abnormality terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits under phenotype abnormality terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(pheno_abnorm_gwas) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 2.2</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% pheno_abnorm_gwas]

print(&quot;Percentage of background GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 25.2</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits not accounted for by so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits not accounted for by so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 79</code></pre>
</div>
<div id="measurement-traits" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Measurement
traits</h2>
<pre class="r"><code>measurement_gwas &lt;- not_accounted_for[
  sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts) # remove extra spaces
    all(parts %in% measurement)
  })
]

measurement_gwas  = c(measurement_gwas, additional_measurement) |&gt; unique()
additional_measurement &lt;- not_accounted_for[not_accounted_for %in% measurement]

measurement_gwas  = c(measurement_gwas, additional_measurement) |&gt; unique()

not_accounted_for = not_accounted_for[!not_accounted_for %in% measurement_gwas]

print(&quot;Percentage of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 7.6</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 24</code></pre>
</div>
<div id="response-traits" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Response traits</h2>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
response_gwas &lt;- not_accounted_for[
  sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts)
    all(parts %in% response)
  })
]
additional_response &lt;- not_accounted_for[not_accounted_for %in% response]

response_gwas  = c(response_gwas, additional_response) |&gt; unique()

not_accounted_for = not_accounted_for[!not_accounted_for %in% response_gwas]

print(&quot;Percentage of background GWAS traits under response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits under response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(response_gwas) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 1.6</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits under response terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits under response terms&quot;</code></pre>
<pre class="r"><code>length(response_gwas)</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 19</code></pre>
<pre class="r"><code>print(&quot;Percentage of background GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 6.1</code></pre>
</div>
<div id="background-trait-categories" class="section level2"
number="4.5">
<h2><span class="header-section-number">4.5</span> Background trait
categories</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::mutate(BACKGROUND_TRAIT_CATEGORY = 
                   dplyr::case_when(
                                      MAPPED_BACKGROUND_TRAIT == &quot;&quot; ~ NA,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% disease_gwas ~ &quot;Disease/Disorder&quot;,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% pheno_abnorm_gwas ~ &quot;Phenotypic Abnormality&quot;,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% measurement_gwas ~ &quot;Measurement&quot;,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% response_gwas ~ &quot;Response&quot;,
                                      TRUE ~ &quot;Other&quot;)
                )</code></pre>
</div>
</div>
<div id="saving-gwas-trait-categories" class="section level1"
number="5">
<h1><span class="header-section-number">5</span> Saving GWAS trait
categories</h1>
<pre class="r"><code>gwas_study_info |&gt;
  group_by(MAPPED_TRAIT_CATEGORY, BACKGROUND_TRAIT_CATEGORY) |&gt;
  summarise(n_studies = n()) |&gt; 
  arrange(desc(n_studies))</code></pre>
<pre><code># A tibble: 33 × 3
# Groups:   MAPPED_TRAIT_CATEGORY [9]
   MAPPED_TRAIT_CATEGORY  BACKGROUND_TRAIT_CATEGORY n_studies
   &lt;chr&gt;                  &lt;chr&gt;                         &lt;int&gt;
 1 Measurement            &lt;NA&gt;                         101836
 2 Disease/Disorder       &lt;NA&gt;                          20314
 3 Measurement            Disease/Disorder              12223
 4 Other                  &lt;NA&gt;                           2725
 5 Phenotypic Abnormality &lt;NA&gt;                           2328
 6 Measurement            Measurement                     969
 7 Disease/Disorder       Disease/Disorder                549
 8 Injury                 &lt;NA&gt;                            508
 9 Phenotype              &lt;NA&gt;                            340
10 Other                  Disease/Disorder                335
# ℹ 23 more rows</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::rowwise() |&gt;
  dplyr::mutate(DISEASE_STUDY = 
                   ifelse(MAPPED_TRAIT_CATEGORY == &quot;Disease/Disorder&quot; | 
                          MAPPED_TRAIT_CATEGORY == &quot;Phenotypic Abnormality&quot; |  
                          BACKGROUND_TRAIT_CATEGORY == &quot;Disease/Disorder&quot; | 
                          BACKGROUND_TRAIT_CATEGORY == &quot;Phenotypic Abnormality&quot;,
                          T, F )
                ) |&gt;
  dplyr::ungroup() </code></pre>
<div id="number-of-disease-studies" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Number of disease
studies</h2>
<pre class="r"><code>gwas_study_info |&gt;
  filter(DISEASE_STUDY == T) |&gt;
  nrow()</code></pre>
<pre><code>[1] 36022</code></pre>
</div>
<div id="giving-the-disease-labels" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Giving the disease
labels</h2>
<pre class="r"><code>combined_disease_terms = function(MAPPED_TRAIT_1, MAPPED_TRAIT_2){
  
  
  MAPPED_TRAIT_1 = stringr::str_split(MAPPED_TRAIT_1, &quot;, &quot;) |&gt; unlist()
  MAPPED_TRAIT_2  = stringr::str_split(MAPPED_TRAIT_2, &quot;, &quot;) |&gt; unlist()
  
  all_mapped_disease_terms = 
    c(MAPPED_TRAIT_1, MAPPED_TRAIT_2) |&gt;
    unique()
  
  combined_mapped_disease_terms = paste0(all_mapped_disease_terms, 
                                         collapse = &quot;, &quot;)
  
  return(combined_mapped_disease_terms)
  
}



gwas_study_info &lt;- 
  gwas_study_info |&gt;
  dplyr::rowwise() |&gt;
  dplyr::mutate(all_disease_terms = 
                case_when(is.na(background_disease_terms) &amp; is.na(disease_terms) ~ NA,
                          is.na(background_disease_terms) &amp; !is.na(disease_terms) ~ disease_terms,
                          !is.na(background_disease_terms) &amp; is.na(disease_terms) ~ background_disease_terms,
                          !is.na(background_disease_terms) &amp; !is.na(disease_terms) ~
                            combined_disease_terms(background_disease_terms,
                                                   disease_terms)) 

  ) |&gt;
  dplyr::ungroup()</code></pre>
</div>
</div>
<div id="saving" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Saving:</h1>
<pre class="r"><code>data.table::fwrite(gwas_study_info,
                  here::here(&quot;output/gwas_study_info_trait_ontology_info.csv&quot;), 
                  sep = &quot;,&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] stringr_1.5.1     ggplot2_3.5.2     dplyr_1.1.4       data.table_1.17.8
[5] workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       jsonlite_2.0.0     compiler_4.3.1     renv_1.0.3        
 [5] promises_1.3.3     tidyselect_1.2.1   Rcpp_1.1.0         git2r_0.36.2      
 [9] callr_3.7.6        later_1.4.2        jquerylib_0.1.4    scales_1.4.0      
[13] yaml_2.3.10        fastmap_1.2.0      here_1.0.1         R6_2.6.1          
[17] generics_0.1.4     knitr_1.50         tibble_3.3.0       rprojroot_2.1.0   
[21] RColorBrewer_1.1-3 bslib_0.9.0        pillar_1.11.0      rlang_1.1.6       
[25] utf8_1.2.6         cachem_1.1.0       stringi_1.8.7      httpuv_1.6.16     
[29] xfun_0.52          getPass_0.2-4      fs_1.6.6           sass_0.4.10       
[33] cli_3.6.5          withr_3.0.2        magrittr_2.0.3     ps_1.9.1          
[37] grid_4.3.1         digest_0.6.37      processx_3.8.6     rstudioapi_0.17.1 
[41] lifecycle_1.0.4    vctrs_0.6.5        evaluate_1.0.4     glue_1.8.0        
[45] farver_2.1.2       whisker_0.4.1      rmarkdown_2.29     httr_1.4.7        
[49] tools_4.3.1        pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
