<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-08-24" />

<title>GWAS Trait Categorisation</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GWAS Trait Categorisation</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-08-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreeb710a4db85001c23f15255b79a1fade1dccab8d2targetblankb710a4da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/b710a4db85001c23f15255b79a1fade1dccab8d2" target="_blank">b710a4d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreeb710a4db85001c23f15255b79a1fade1dccab8d2targetblankb710a4da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/b710a4db85001c23f15255b79a1fade1dccab8d2" target="_blank">b710a4d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    data/gwas_catalog/
    Ignored:    output/gwas_study_info_cohort_corrected.csv
    Ignored:    output/gwas_study_info_trait_corrected.csv
    Ignored:    output/gwas_study_info_trait_ontology_info.csv
    Ignored:    output/trait_ontology/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  data/.DS_Store
    Untracked:  renv/

Unstaged changes:
    Modified:   .Rprofile
    Modified:   analysis/collapse_traits.Rmd
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/replication_ancestry_bias.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/trait_ontology_collapse.Rmd</code>) and HTML
(<code>docs/trait_ontology_collapse.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/b710a4db85001c23f15255b79a1fade1dccab8d2/analysis/trait_ontology_collapse.Rmd" target="_blank">b710a4d</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-25
</td>
<td>
Update trait ontology investigation to group disease terms
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/243218e2a8094d602e56712012c41902986dbe84/docs/trait_ontology_collapse.html" target="_blank">243218e</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/6ceba65f7523468737192025be09f5703416d3b5/analysis/trait_ontology_collapse.Rmd" target="_blank">6ceba65</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-08-24
</td>
<td>
workflowr::wflow_publish("analysis/trait_ontology_collapse.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Set up</h1>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, 
                      message = FALSE, 
                      warning = FALSE
                      )

library(data.table)
library(dplyr)
library(ggplot2)
library(stringr)</code></pre>
</div>
<div id="get-trait-ontology-terms-relating-to-disease-etc."
class="section level1" number="2">
<h1><span class="header-section-number">2</span> Get trait ontology
terms relating to ‘Disease’ (etc.)</h1>
<pre class="r"><code>library(httr)
library(jsonlite)</code></pre>
<div id="mondo-human-disease-httppurl.obolibrary.orgobomondo_0700096"
class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Mondo ‘Human Disease’
(<a href="http://purl.obolibrary.org/obo/MONDO_0700096"
class="uri">http://purl.obolibrary.org/obo/MONDO_0700096</a>)</h2>
<pre class="r"><code>url &lt;-
&quot;http://www.ebi.ac.uk/ols4/api/ontologies/mondo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0700096/descendants&quot;

mondo_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  mondo_descendants &lt;- c(mondo_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(mondo_descendants)
mondo_descendants[1:5]

writeLines(mondo_descendants, here::here(&quot;output/trait_ontology/mondo_0700096_descendants.txt&quot;))</code></pre>
</div>
<div id="efo-disease-descendants-httpwww.ebi.ac.ukefoefo_0000408"
class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> EFO ‘Disease’
Descendants (<a href="http://www.ebi.ac.uk/efo/EFO_0000408"
class="uri">http://www.ebi.ac.uk/efo/EFO_0000408</a>)</h2>
<pre class="r"><code>url &lt;-
&quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000408/descendants&quot;

efo_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  efo_descendants &lt;- c(efo_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(efo_descendants)
efo_descendants[1:5]

writeLines(efo_descendants, here::here(&quot;output/trait_ontology/efo_0000408_descendants.txt&quot;))</code></pre>
</div>
<div id="ncit-disease-or-disorder-httppurl.obolibrary.orgoboncit_c2991"
class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> NCIT (Disease or
Disorder): <a href="http://purl.obolibrary.org/obo/NCIT_C2991"
class="uri">http://purl.obolibrary.org/obo/NCIT_C2991</a></h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/ncit/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FNCIT_C2991/descendants&quot;


ncit_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  ncit_descendants &lt;- c(ncit_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(ncit_descendants)
ncit_descendants[1:5]

writeLines(ncit_descendants, here::here(&quot;output/trait_ontology/ncit_C2991_descendants.txt&quot;))</code></pre>
</div>
<div id="orphanet-disorder-httpwww.orpha.netordoorphanet_557493"
class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Orphanet Disorder (<a
href="http://www.orpha.net/ORDO/Orphanet_557493"
class="uri">http://www.orpha.net/ORDO/Orphanet_557493</a>)</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/ordo/terms/http%253A%252F%252Fwww.orpha.net%252FORDO%252FOrphanet_557493/descendants&quot;

orphanet_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  orphanet_descendants &lt;- c(orphanet_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(orphanet_descendants)
orphanet_descendants[1:5]

writeLines(orphanet_descendants, here::here(&quot;output/trait_ontology/orphanet_557493_descendants.txt&quot;))</code></pre>
</div>
<div id="pathologic-process-httppurl.obolibrary.orgoboobi_1110122"
class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Pathologic process
(<a href="http://purl.obolibrary.org/obo/OBI_1110122"
class="uri">http://purl.obolibrary.org/obo/OBI_1110122</a>)</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FOBI_1110122/descendants&quot;

path_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  path_descendants &lt;- c(path_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(path_descendants)
path_descendants[1:5]

writeLines(path_descendants, here::here(&quot;output/trait_ontology/obi_1110122_descendants.txt&quot;))</code></pre>
</div>
<div id="disease-course-httppurl.obolibrary.orgoboogms_0000063"
class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Disease course <a
href="http://purl.obolibrary.org/obo/OGMS_0000063"
class="uri">http://purl.obolibrary.org/obo/OGMS_0000063</a></h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FOGMS_0000063/descendants&quot;

disease_course_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  disease_course_descendants &lt;- c(disease_course_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(disease_course_descendants)
disease_course_descendants[1:5]

writeLines(disease_course_descendants, here::here(&quot;output/trait_ontology/ogms_0000063_descendants.txt&quot;))</code></pre>
</div>
</div>
<div
id="get-trait-ontology-terms-relating-to-phenotypic-abnormality-httppurl.obolibrary.orgobohp_0000118"
class="section level1" number="3">
<h1><span class="header-section-number">3</span> Get trait ontology
terms relating to ‘Phenotypic abnormality’ (<a
href="http://purl.obolibrary.org/obo/HP_0000118"
class="uri">http://purl.obolibrary.org/obo/HP_0000118</a>)</h1>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0000118/descendants&quot;


hp_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  hp_descendants &lt;- c(hp_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(hp_descendants)</code></pre>
<pre><code>[1] 2625</code></pre>
<pre class="r"><code>hp_descendants[1:5]</code></pre>
<pre><code>[1] &quot;hypoxia&quot;                                  
[2] &quot;necrosis&quot;                                 
[3] &quot;Abnormality of limbs&quot;                     
[4] &quot;Abnormality of the musculoskeletal system&quot;
[5] &quot;Vocal cord dysfunction&quot;                   </code></pre>
<pre class="r"><code>writeLines(hp_descendants, here::here(&quot;output/trait_ontology/hp_0000118_descendants.txt&quot;))</code></pre>
</div>
<div id="get-trait-ontology-terms-relating-to-measurement"
class="section level1" number="4">
<h1><span class="header-section-number">4</span> Get trait ontology
terms relating to ‘Measurement’</h1>
<div id="measurement-terms-httpwww.ebi.ac.ukefoefo_0001444"
class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Measurement terms (<a
href="http://www.ebi.ac.uk/efo/EFO_0001444"
class="uri">http://www.ebi.ac.uk/efo/EFO_0001444</a>)</h2>
<pre class="r"><code># Define the API endpoint
url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0001444/descendants&quot;

measurement_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  measurement_descendants &lt;- c(measurement_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(measurement_descendants)
measurement_descendants[1:5]


writeLines(measurement_descendants, here::here(&quot;output/trait_ontology/efo_0001444_descendants.txt&quot;))</code></pre>
</div>
<div id="total-cholesterol-measurements-httpwww.ebi.ac.ukefoefo_0004574"
class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Total cholesterol
measurements (<a href="http://www.ebi.ac.uk/efo/EFO_0004574"
class="uri">http://www.ebi.ac.uk/efo/EFO_0004574</a>)</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0004574/descendants&quot;


total_choles_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  total_choles_descendants &lt;- c(total_choles_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

writeLines(total_choles_descendants, here::here(&quot;output/trait_ontology/efo_0004574_descendants.txt&quot;))


length(total_choles_descendants)
total_choles_descendants[1:5]</code></pre>
</div>
</div>
<div
id="get-ontology-terms---response-to-stimulus-httppurl.obolibrary.orgobogo_0050896"
class="section level1" number="5">
<h1><span class="header-section-number">5</span> Get ontology terms -
response to stimulus (<a
href="http://purl.obolibrary.org/obo/GO_0050896"
class="uri">http://purl.obolibrary.org/obo/GO_0050896</a>)</h1>
<p>Includes:</p>
<ul>
<li>Response to chemical (e.g. drug) (<a
href="http://purl.obolibrary.org/obo/GO_0042221"
class="uri">http://purl.obolibrary.org/obo/GO_0042221</a>)</li>
<li>Immune response (<a href="http://purl.obolibrary.org/obo/GO_0006955"
class="uri">http://purl.obolibrary.org/obo/GO_0006955</a>)</li>
</ul>
<div id="from-go" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> From GO</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/go/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FGO_0050896/descendants&quot;

go_response_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  go_response_descendants &lt;- c(go_response_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(go_response_descendants)
go_response_descendants[1:5]

writeLines(go_response_descendants, here::here(&quot;output/trait_ontology/go_0050896_descendants.txt&quot;))</code></pre>
</div>
<div id="from-efo" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> From EFO</h2>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FGO_0050896/descendants&quot;

efo_response_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  efo_response_descendants &lt;- c(efo_response_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(efo_response_descendants)
efo_response_descendants[1:5]

writeLines(efo_response_descendants, here::here(&quot;output/trait_ontology/efo_go_0050896_descendants.txt&quot;))</code></pre>
</div>
</div>
<div
id="get-ontology-terms-relating-to-phenotype-httpwww.ebi.ac.ukefoefo_0000651"
class="section level1" number="6">
<h1><span class="header-section-number">6</span> Get ontology terms
relating to ‘Phenotype’ (<a href="http://www.ebi.ac.uk/efo/EFO_0000651"
class="uri">http://www.ebi.ac.uk/efo/EFO_0000651</a>)</h1>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000651/descendants&quot;

phenotype_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  phenotype_descendants &lt;- c(phenotype_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(phenotype_descendants)
phenotype_descendants[1:5]


writeLines(phenotype_descendants, here::here(&quot;output/trait_ontology/efo_0000651_descendants.txt&quot;))</code></pre>
</div>
<div id="mental-process-httpwww.ebi.ac.ukefoefo_0004323"
class="section level1" number="7">
<h1><span class="header-section-number">7</span> Mental process (<a
href="http://www.ebi.ac.uk/efo/EFO_0004323"
class="uri">http://www.ebi.ac.uk/efo/EFO_0004323</a>)</h1>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0004323/descendants&quot;

mental_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  mental_descendants &lt;- c(mental_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(mental_descendants)
mental_descendants[1:5]
mental_descendants = c(&quot;mental process&quot;, 
                       &quot;cognitive function measurement&quot;,
                       mental_descendants)

writeLines(mental_descendants, here::here(&quot;output/trait_ontology/efo_0004323_descendants.txt&quot;))</code></pre>
</div>
<div id="behavior-httppurl.obolibrary.orgobogo_0007610"
class="section level1" number="8">
<h1><span class="header-section-number">8</span> Behavior (<a
href="http://purl.obolibrary.org/obo/GO_0007610"
class="uri">http://purl.obolibrary.org/obo/GO_0007610</a>)</h1>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FGO_0007610/descendants&quot;

behavior_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  behavior_descendants &lt;- c(behavior_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(behavior_descendants)
behavior_descendants[1:5]

behavior_descendants = c(&quot;behavior&quot;, behavior_descendants)
writeLines(behavior_descendants, here::here(&quot;output/trait_ontology/go_0007610_descendants.txt&quot;))</code></pre>
</div>
<div id="injury-httpwww.ebi.ac.ukefoefo_0000546" class="section level1"
number="9">
<h1><span class="header-section-number">9</span> Injury (<a
href="http://www.ebi.ac.uk/efo/EFO_0000546"
class="uri">http://www.ebi.ac.uk/efo/EFO_0000546</a>)</h1>
<pre class="r"><code>url &lt;- &quot;http://www.ebi.ac.uk/ols4/api/ontologies/efo/terms/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0000546/descendants&quot;

injury_descendants &lt;- c()
repeat {
  res &lt;- GET(url)
  stop_for_status(res)
  data &lt;- fromJSON(content(res, as = &quot;text&quot;, encoding = &quot;UTF-8&quot;))
  
  injury_descendants &lt;- c(injury_descendants, data$`_embedded`$terms$label)
  
  # check if there is a next page
  if (!is.null(data$`_links`$`next`$href)) {
    url &lt;- data$`_links`$`next`$href
  } else {
    break
  }
}

length(injury_descendants)
injury_descendants[1:5]

injury_descendants = c(&quot;injury&quot;, injury_descendants)
writeLines(injury_descendants, here::here(&quot;output/trait_ontology/efo_0000546_descendants.txt&quot;))</code></pre>
</div>
<div id="sociological-environmental-factors" class="section level1"
number="10">
<h1><span class="header-section-number">10</span> Sociological /
environmental factors</h1>
<pre class="r"><code>enviro_factors &lt;- c(
  &quot;income&quot;,
&quot;socioeconomic status&quot;,
&quot;encounter with health service related to socioeconomic and psychosocial circumstances&quot; ,
&quot;townsend deprivation index&quot;,
&quot;household income&quot;,
&quot;social deprivation&quot;,
&quot;economic and social preference&quot;,
&quot;social deprivation&quot;,
&quot;physical activity&quot;,
&quot;exercise&quot;,
&quot;family relationship&quot;
)</code></pre>
</div>
<div id="overlap-ontology-terms-and-gwas-traits" class="section level1"
number="11">
<h1><span class="header-section-number">11</span> Overlap ontology terms
and GWAS traits</h1>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;output/gwas_study_info_cohort_corrected.csv&quot;))</code></pre>
<pre class="r"><code>all_gwas_terms = unique(gwas_study_info$MAPPED_TRAIT)

all_gwas_terms = stringr::str_trim(tolower(all_gwas_terms))</code></pre>
<div
id="disease-overlap-how-many-gwas-traits-fall-within-disease-or-disorder-terms"
class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Disease Overlap (How
many GWAS traits fall within disease or disorder terms?)</h2>
<div id="combine-disease-terms" class="section level3" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Combine disease
terms</h3>
<pre class="r"><code>efo_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000408_descendants.txt&quot;))

mondo_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/mondo_0700096_descendants.txt&quot;))

ncit_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/ncit_C2991_descendants.txt&quot;))

orphanet_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/orphanet_557493_descendants.txt&quot;))

disease_terms = c(mondo_descendants,
                  efo_descendants,
                  ncit_descendants,
                  orphanet_descendants) |&gt;
                 unique()

disease_terms = stringr::str_trim(tolower(disease_terms))

print(&quot;Number of terms related to disease or disorder&quot;)</code></pre>
<pre><code>[1] &quot;Number of terms related to disease or disorder&quot;</code></pre>
<pre class="r"><code>length(disease_terms)</code></pre>
<pre><code>[1] 53251</code></pre>
<pre class="r"><code>not_simple_disease_terms = all_gwas_terms[!all_gwas_terms %in% disease_terms]

# sometimes there&#39;s multiple terms - check if any disease term is in these gwas terms
multiple_terms = grep(&quot;,&quot;, not_simple_disease_terms, value = T)

mask &lt;- Reduce(`|`, lapply(disease_terms, function(x) grepl(x, multiple_terms)))
additional_disease_gwas &lt;- multiple_terms[mask]

disease_gwas = c(all_gwas_terms[all_gwas_terms %in% disease_terms],
                 additional_disease_gwas)

not_disease_terms = not_simple_disease_terms[!not_simple_disease_terms %in% additional_disease_gwas]

print(&quot;Number of GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>length(all_gwas_terms) - length(not_disease_terms)</code></pre>
<pre><code>[1] 3377</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * (length(all_gwas_terms) - length(not_disease_terms)) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 14.8</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits not under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_disease_terms) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 85.2</code></pre>
</div>
</div>
<div id="add-disease-terms-to-gwas-study-info-dataset"
class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Add disease terms to
GWAS study info dataset</h2>
<pre class="r"><code>find_disease_terms  =   function(MAPPED_TRAIT) {
        # find all disease terms that appear in the trait
        split_mapped_traits &lt;- stringr::str_split(MAPPED_TRAIT, &quot;, &quot;) |&gt; 
                               unlist()
        
        mapped_disease_terms = disease_terms[disease_terms %in% split_mapped_traits]
        mapped_disease_terms = unique(mapped_disease_terms)
        return(paste0(mapped_disease_terms, collapse = &quot;, &quot;))  # combine multiple matches
        
    }

gwas_study_info &lt;- 
  gwas_study_info |&gt; 
  dplyr::rowwise() |&gt;
  dplyr::mutate(
    disease_terms = 
      ifelse(MAPPED_TRAIT %in% disease_gwas,
             find_disease_terms(MAPPED_TRAIT),
             NA)
  )</code></pre>
</div>
<div
id="measurement-overlap-how-many-gwas-traits-fall-within-measurement-terms"
class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> Measurement Overlap
(how many GWAS traits fall within measurement terms?)</h2>
<div id="combine-measurement" class="section level3" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Combine
measurement</h3>
<pre class="r"><code>measurement &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0001444_descendants.txt&quot;))
total_choles &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0004574_descendants.txt&quot;))

measurement &lt;- c(total_choles,
                 measurement) 

measurement &lt;- unique(measurement)

measurement = stringr::str_trim(tolower(measurement))</code></pre>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
measurement_gwas &lt;- not_disease_terms[
  sapply(strsplit(not_disease_terms, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts) # remove extra spaces
    all(parts %in% measurement)
  })
]

additional_measurement &lt;- not_disease_terms[not_disease_terms %in% measurement]

measurement_gwas  = c(measurement_gwas, additional_measurement) |&gt; unique()

print(&quot;Number of GWAS traits under measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under measurement terms&quot;</code></pre>
<pre class="r"><code>length(measurement_gwas)</code></pre>
<pre><code>[1] 18226</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(measurement_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 79.7</code></pre>
<pre class="r"><code>not_accounted_for = not_disease_terms[!not_disease_terms %in% measurement_gwas]

print(&quot;Percentage of GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 5.6</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 1274</code></pre>
</div>
</div>
<div id="response-to-stimulus" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> Response to
stimulus</h2>
<div id="combine-response-terms" class="section level3" number="11.4.1">
<h3><span class="header-section-number">11.4.1</span> Combine response
terms</h3>
<pre class="r"><code>go_response = readLines(here::here(&quot;output/trait_ontology/go_0050896_descendants.txt&quot;))

efo_response &lt;- readLines(here::here(&quot;output/trait_ontology/efo_go_0050896_descendants.txt&quot;))

response &lt;- c(go_response,
              efo_response,
              &quot;response to stimulus&quot;)

response &lt;- unique(response)

response = stringr::str_trim(tolower(response))</code></pre>
<pre class="r"><code>response_gwas = not_accounted_for[not_accounted_for %in% response]

print(&quot;Percentage of GWAS traits under response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(response_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.6</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% response_gwas]

print(&quot;Percentage of GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 1141</code></pre>
</div>
</div>
<div id="phenotype-abnormality-overlap" class="section level2"
number="11.5">
<h2><span class="header-section-number">11.5</span> Phenotype
abnormality overlap</h2>
<pre class="r"><code>pheno_abnorm &lt;- readLines(here::here(&quot;output/trait_ontology/hp_0000118_descendants.txt&quot;))
pheno_abnorm = stringr::str_trim(tolower(pheno_abnorm))

pheno_abnorm_gwas = not_accounted_for[not_accounted_for %in% pheno_abnorm]

print(&quot;Percentage of GWAS traits under phenotype abnormality terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under phenotype abnormality terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(pheno_abnorm_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.5</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% pheno_abnorm_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.5</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 804</code></pre>
</div>
<div id="mental-process" class="section level2" number="11.6">
<h2><span class="header-section-number">11.6</span> Mental process</h2>
<pre class="r"><code>mental &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0004323_descendants.txt&quot;))
mental = stringr::str_trim(tolower(mental))

mental_gwas = not_accounted_for[not_accounted_for %in% mental]

print(&quot;Percentage of GWAS traits under mental process terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under mental process terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(mental_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% mental_gwas]

print(&quot;Percentage of GWAS traits not accounted for thus far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for thus far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.4</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for thus far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for thus far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 786</code></pre>
</div>
<div id="behavior" class="section level2" number="11.7">
<h2><span class="header-section-number">11.7</span> Behavior</h2>
<pre class="r"><code>behavior &lt;- readLines(here::here(&quot;output/trait_ontology/go_0007610_descendants.txt&quot;))
behavior = stringr::str_trim(tolower(behavior))

behavior_gwas = not_accounted_for[not_accounted_for %in% behavior]

print(&quot;Percentage of GWAS traits under behavouir terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under behavouir terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(behavior_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% behavior_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 763</code></pre>
</div>
<div id="injury" class="section level2" number="11.8">
<h2><span class="header-section-number">11.8</span> Injury</h2>
<pre class="r"><code>injury &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000546_descendants.txt&quot;))

injury = stringr::str_trim(tolower(injury))

injury_gwas = not_accounted_for[not_accounted_for %in% injury]

print(&quot;Percentage of GWAS traits under injury terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under injury terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(injury_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% injury_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3.3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 744</code></pre>
</div>
<div id="phenotype" class="section level2" number="11.9">
<h2><span class="header-section-number">11.9</span> Phenotype</h2>
<pre class="r"><code>phenotype &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000651_descendants.txt&quot;))

phenotype = stringr::str_trim(tolower(phenotype))

phenotype_gwas = not_accounted_for[not_accounted_for %in% phenotype]

print(&quot;Percentage of GWAS traits under phenotype terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under phenotype terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(phenotype_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.2</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% phenotype_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 695</code></pre>
</div>
</div>
<div id="add-categories-to-gwas-info" class="section level1"
number="12">
<h1><span class="header-section-number">12</span> Add Categories to GWAS
Info</h1>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::mutate(MAPPED_TRAIT_CATEGORY = dplyr::case_when(MAPPED_TRAIT %in% disease_gwas ~ &quot;Disease/Disorder&quot;,
                                                          MAPPED_TRAIT %in% pheno_abnorm_gwas ~ &quot;Phenotypic Abornmality&quot;,
                                                          MAPPED_TRAIT %in% measurement_gwas ~ &quot;Measurement&quot;,
                                                          MAPPED_TRAIT %in% response_gwas ~ &quot;Response&quot;,
                                                          TRUE ~ &quot;Other&quot;
                                                          )
                )</code></pre>
</div>
<div id="background-traits" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Background traits</h1>
<pre class="r"><code>gwas_study_info$MAPPED_BACKGROUND_TRAIT |&gt; unique() -&gt; gwas_background

gwas_background = stringr::str_trim(tolower(gwas_background))</code></pre>
<div id="overlap-with-diseasedisorder-traits" class="section level2"
number="13.1">
<h2><span class="header-section-number">13.1</span> Overlap with
disease/disorder traits</h2>
<pre class="r"><code>multiple_terms = grep(&quot;,&quot;, gwas_background, value = T)
mask &lt;- Reduce(`|`, lapply(disease_terms, function(x) grepl(x, multiple_terms)))
additional_disease_gwas &lt;- multiple_terms[mask]

disease_gwas = c(gwas_background[gwas_background %in% disease_terms],
                 additional_disease_gwas)</code></pre>
<pre class="r"><code>gwas_study_info &lt;- 
  gwas_study_info |&gt;
  rowwise() |&gt;
  dplyr::mutate(
    background_disease_terms = 
      ifelse(MAPPED_BACKGROUND_TRAIT %in% disease_gwas,
             find_disease_terms(MAPPED_BACKGROUND_TRAIT),
             NA)
  )</code></pre>
</div>
<div id="overlap-with-measurement-traits" class="section level2"
number="13.2">
<h2><span class="header-section-number">13.2</span> Overlap with
measurement traits</h2>
<pre class="r"><code>not_accounted_background = gwas_background[!gwas_background %in% disease_gwas]

measurement_gwas &lt;- not_disease_terms[
  sapply(strsplit(not_accounted_background, &quot;, &quot;), function(parts) {
    parts &lt;- trimws(parts) # remove extra spaces
    all(parts %in% measurement)
  })
]

measurement_gwas  = c(measurement_gwas, additional_measurement) |&gt; unique()
additional_measurement &lt;- not_accounted_background[not_accounted_background %in% measurement]</code></pre>
</div>
<div id="overlap-with-response-traits" class="section level2"
number="13.3">
<h2><span class="header-section-number">13.3</span> Overlap with
response traits</h2>
<pre class="r"><code>not_accounted_background = not_accounted_background[!not_accounted_background %in% measurement_gwas]

response_gwas = not_accounted_background[not_accounted_background %in% response]

gwas_study_info = 
gwas_study_info |&gt;
  dplyr::mutate(BACKGROUND_TRAIT_CATEGORY = dplyr::case_when(MAPPED_BACKGROUND_TRAIT %in% disease_gwas ~ &quot;Disease/Disorder&quot;,
                                                          MAPPED_BACKGROUND_TRAIT %in% measurement_gwas ~ &quot;Measurement&quot;,
                                                          MAPPED_BACKGROUND_TRAIT %in% response_gwas ~ &quot;Response&quot;,
                                                          TRUE ~ &quot;Other&quot;
                                                          ))</code></pre>
</div>
</div>
<div id="saving-gwas-trait-categories" class="section level1"
number="14">
<h1><span class="header-section-number">14</span> Saving GWAS trait
categories</h1>
<pre class="r"><code>gwas_study_info |&gt;
  group_by(MAPPED_TRAIT_CATEGORY, BACKGROUND_TRAIT_CATEGORY) |&gt;
  summarise(n_studies = n()) |&gt; 
  arrange(desc(n_studies))</code></pre>
<pre><code># A tibble: 18 × 3
# Groups:   MAPPED_TRAIT_CATEGORY [5]
   MAPPED_TRAIT_CATEGORY  BACKGROUND_TRAIT_CATEGORY n_studies
   &lt;chr&gt;                  &lt;chr&gt;                         &lt;int&gt;
 1 Measurement            Measurement                   84977
 2 Other                  Measurement                   27173
 3 Disease/Disorder       Measurement                   16585
 4 Measurement            Disease/Disorder               7252
 5 Other                  Disease/Disorder               5109
 6 Phenotypic Abornmality Measurement                     436
 7 Disease/Disorder       Disease/Disorder                365
 8 Response               Measurement                     274
 9 Measurement            Other                           256
10 Response               Disease/Disorder                155
11 Other                  Other                           108
12 Disease/Disorder       Other                            97
13 Phenotypic Abornmality Disease/Disorder                 37
14 Response               Other                            13
15 Phenotypic Abornmality Other                            11
16 Measurement            Response                          5
17 Disease/Disorder       Response                          1
18 Other                  Response                          1</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::rowwise() |&gt;
  dplyr::mutate(DISEASE_STUDY = ifelse(MAPPED_TRAIT_CATEGORY == &quot;Disease/Disorder&quot; | BACKGROUND_TRAIT_CATEGORY == &quot;Disease/Disorer&quot;, T, F )) |&gt;
  dplyr::ungroup()</code></pre>
<pre class="r"><code>combined_disease_terms = function(MAPPED_TRAIT_1, MAPPED_TRAIT_2){
  
  
  MAPPED_TRAIT_1 = stringr::str_split(MAPPED_TRAIT_1, &quot;, &quot;) |&gt; unlist()
  MAPPED_TRAIT_2  = stringr::str_split(MAPPED_TRAIT_2, &quot;, &quot;) |&gt; unlist()
  
  all_mapped_disease_terms = 
    c(MAPPED_TRAIT_1, MAPPED_TRAIT_2) |&gt;
    unique()
  
  combined_mapped_disease_terms = paste0(all_mapped_disease_terms, 
                                         collapse = &quot;, &quot;)
  
  return(combined_mapped_disease_terms)
  
}



gwas_study_info &lt;- 
  gwas_study_info |&gt;
  dplyr::rowwise() |&gt;
  dplyr::mutate(all_disease_terms = 
                case_when(is.na(background_disease_terms) &amp; is.na(disease_terms) ~ NA,
                          is.na(background_disease_terms) &amp; !is.na(disease_terms) ~ disease_terms,
                          !is.na(background_disease_terms) &amp; is.na(disease_terms) ~ background_disease_terms,
                          !is.na(background_disease_terms) &amp; !is.na(disease_terms) ~
                            combined_disease_terms(background_disease_terms,
                                                   disease_terms)) 

  ) |&gt;
  dplyr::ungroup()</code></pre>
</div>
<div id="saving" class="section level1" number="15">
<h1><span class="header-section-number">15</span> Saving:</h1>
<pre class="r"><code>data.table::fwrite(gwas_study_info,
                  here::here(&quot;output/gwas_study_info_trait_ontology_info.csv&quot;), 
                  sep = &quot;,&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] jsonlite_2.0.0    httr_1.4.7        stringr_1.5.1     ggplot2_3.5.2    
[5] dplyr_1.1.4       data.table_1.17.8 workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       compiler_4.3.1     renv_1.0.3         promises_1.3.3    
 [5] tidyselect_1.2.1   Rcpp_1.1.0         git2r_0.36.2       callr_3.7.6       
 [9] later_1.4.2        jquerylib_0.1.4    scales_1.4.0       yaml_2.3.10       
[13] fastmap_1.2.0      here_1.0.1         R6_2.6.1           generics_0.1.4    
[17] curl_6.4.0         knitr_1.50         tibble_3.3.0       rprojroot_2.1.0   
[21] RColorBrewer_1.1-3 bslib_0.9.0        pillar_1.11.0      rlang_1.1.6       
[25] utf8_1.2.6         cachem_1.1.0       stringi_1.8.7      httpuv_1.6.16     
[29] xfun_0.52          getPass_0.2-4      fs_1.6.6           sass_0.4.10       
[33] cli_3.6.5          withr_3.0.2        magrittr_2.0.3     ps_1.9.1          
[37] grid_4.3.1         digest_0.6.37      processx_3.8.6     rstudioapi_0.17.1 
[41] lifecycle_1.0.4    vctrs_0.6.5        evaluate_1.0.4     glue_1.8.0        
[45] farver_2.1.2       whisker_0.4.1      rmarkdown_2.29     tools_4.3.1       
[49] pkgconfig_2.0.3    htmltools_0.5.8.1 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
