<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Isobel Beasley" />

<meta name="date" content="2025-08-24" />

<title>GWAS Trait Categorisation</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">genomics_ancest_disease_dispar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GWAS Trait Categorisation</h1>
<h4 class="author">Isobel Beasley</h4>
<h4 class="date">2025-08-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-12-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>genomics_ancest_disease_dispar/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220216code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220216code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220216)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreef1b9ff4adf7472a7be06f8147af2f3531f2ed2f1targetblankf1b9ff4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/f1b9ff4adf7472a7be06f8147af2f3531f2ed2f1" target="_blank">f1b9ff4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomIJbeasleygenomicsancestdiseasedispartreef1b9ff4adf7472a7be06f8147af2f3531f2ed2f1targetblankf1b9ff4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/tree/f1b9ff4adf7472a7be06f8147af2f3531f2ed2f1" target="_blank">f1b9ff4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    .venv/
    Ignored:    analysis/.DS_Store
    Ignored:    ancestry_dispar_env/
    Ignored:    data/.DS_Store
    Ignored:    data/cohort/
    Ignored:    data/gbd/.DS_Store
    Ignored:    data/gbd/IHME-GBD_2021_DATA-d8cf695e-1.csv
    Ignored:    data/gbd/IHME-GBD_2023_DATA-73cc01fd-1.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2019_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_disease_burden_rate_all_ages.csv
    Ignored:    data/gbd/ihme_gbd_2021_global_paf_rate_percent_all_ages.csv
    Ignored:    data/gwas_catalog/
    Ignored:    data/icd/.DS_Store
    Ignored:    data/icd/IHME_GBD_2019_COD_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2019_NONFATAL_CAUSE_ICD_CODE_MAP_Y2020M10D15.XLSX
    Ignored:    data/icd/IHME_GBD_2021_COD_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/IHME_GBD_2021_NONFATAL_CAUSE_ICD_CODE_MAP_Y2024M05D16.XLSX
    Ignored:    data/icd/UK_Biobank_master_file.tsv
    Ignored:    data/icd/cdc_valid_icd10_Sep_23_2025.xlsx
    Ignored:    data/icd/cdc_valid_icd9_Sep_23_2025.xlsx
    Ignored:    data/icd/hp_umls_mapping.csv
    Ignored:    data/icd/lancet_conditions_icd10.xlsx
    Ignored:    data/icd/manual_disease_icd10_mappings.xlsx
    Ignored:    data/icd/mondo_umls_mapping.csv
    Ignored:    data/icd/phecode_international_version_unrolled.csv
    Ignored:    data/icd/phecode_to_icd10_manual_mapping.xlsx
    Ignored:    data/icd/semiautomatic_ICD-pheno.txt
    Ignored:    data/icd/semiautomatic_ICD-pheno_UKB_subset.txt
    Ignored:    human_dictionary/
    Ignored:    igsr_populations.tsv
    Ignored:    output/.DS_Store
    Ignored:    output/abstracts/
    Ignored:    output/doccano/
    Ignored:    output/fulltexts/
    Ignored:    output/gwas_cat/
    Ignored:    output/gwas_cohorts/
    Ignored:    output/icd_map/
    Ignored:    output/trait_ontology/
    Ignored:    pubmedbert-cohort-ner-model/
    Ignored:    pubmedbert-cohort-ner/
    Ignored:    r-spacyr/
    Ignored:    renv/
    Ignored:    venv/

Untracked files:
    Untracked:  code/extract_cdc_meta.R
    Untracked:  code/figure_4a.R
    Untracked:  code/poster_figures.R
    Untracked:  code/umls_ontology.R
    Untracked:  data/cdc/
    Untracked:  data/icd/2025AA/
    Untracked:  data/icd/umls-2025AA-mrconso.zip
    Untracked:  figures/
    Untracked:  visualization.Rdata

Unstaged changes:
    Modified:   analysis/disease_inves_by_ancest.Rmd
    Modified:   analysis/get_full_text.Rmd
    Modified:   analysis/group_cancer_diseases.Rmd
    Modified:   analysis/gwas_to_gbd.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/level_1_disease_group_non_cancer.Rmd
    Modified:   analysis/level_2_disease_group.Rmd
    Modified:   analysis/manual_trait_map_icd10.Rmd
    Modified:   analysis/map_trait_to_icd10.Rmd
    Modified:   analysis/missing_cohort_info.Rmd
    Modified:   analysis/replication_ancestry_bias.Rmd
    Modified:   analysis/text_for_cohort_labels.Rmd
    Modified:   code/custom_plotting.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/trait_ontology_categorization.Rmd</code>) and HTML
(<code>docs/trait_ontology_categorization.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/f1b9ff4adf7472a7be06f8147af2f3531f2ed2f1/analysis/trait_ontology_categorization.Rmd" target="_blank">f1b9ff4</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-12-29
</td>
<td>
Updating identifying disease studies step
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/5e4dc041dda296a22c8abd96f03d05126cfaf9d6/docs/trait_ontology_categorization.html" target="_blank">5e4dc04</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/6846cca0e94059acff207915f9b1fececf3de33a/analysis/trait_ontology_categorization.Rmd" target="_blank">6846cca</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-14
</td>
<td>
Fixing typos on intital trait categorization .. again
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/cba6936bfabc7672a4d12bb38bbaea15312c8b75/docs/trait_ontology_categorization.html" target="_blank">cba6936</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/14b700b64bce6eb38a5af2acd470e4e5d84ad9c8/analysis/trait_ontology_categorization.Rmd" target="_blank">14b700b</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-14
</td>
<td>
Fixing typos on intital trait categorization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/6a8f9cce3290795fe2b0a149dfad4be589af38dc/docs/trait_ontology_categorization.html" target="_blank">6a8f9cc</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/63bd79a326caf7470b7a4c0cc47e6284115d7037/analysis/trait_ontology_categorization.Rmd" target="_blank">63bd79a</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Update cancer grouping
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/IJbeasley/genomics_ancest_disease_dispar/2853e611c797d6cce1e58a676db71ce7d2627f89/docs/trait_ontology_categorization.html" target="_blank">2853e61</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/IJbeasley/genomics_ancest_disease_dispar/blob/a7e2f7c2475c635908aa5c2b8af83c3088f54ac3/analysis/trait_ontology_categorization.Rmd" target="_blank">a7e2f7c</a>
</td>
<td>
IJbeasley
</td>
<td>
2025-09-10
</td>
<td>
Fixing / re-formatting of initial trait categorization
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Set up</h1>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, 
                      message = FALSE, 
                      warning = FALSE
                      )

library(data.table)
library(dplyr)
library(ggplot2)
library(stringr)</code></pre>
<div id="get-data.frame-of-gwas-traits" class="section level2"
number="1.1">
<h2><span class="header-section-number">1.1</span> Get data.frame of
GWAS traits</h2>
<pre class="r"><code>gwas_study_info &lt;- fread(here::here(&quot;data/gwas_catalog/gwas-catalog-v1.0.3.1-studies-r2025-07-21.tsv&quot;))

gwas_study_info = 
  gwas_study_info |&gt;
  rename_all(~gsub(&quot; &quot;, &quot;_&quot;, .x))

gwas_study_info &lt;-
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = tolower(MAPPED_TRAIT),
         MAPPED_BACKGROUND_TRAIT = tolower(MAPPED_BACKGROUND_TRAIT)
         )

gwas_study_info  &lt;-
  gwas_study_info |&gt;
  mutate(YEAR = lubridate::year(DATE))</code></pre>
</div>
<div id="mapping-terms-for-studies-with-unmapped-traits"
class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Mapping terms for
studies with unmapped traits:</h2>
<pre class="r"><code># some traits are not mapped:
print(&quot;Before fixing, how many unmapped traits are there?&quot;)</code></pre>
<pre><code>[1] &quot;Before fixing, how many unmapped traits are there?&quot;</code></pre>
<pre class="r"><code>gwas_study_info |&gt;
  filter(is.na(MAPPED_TRAIT) | MAPPED_TRAIT == &quot;&quot;) |&gt;
  nrow()</code></pre>
<pre><code>[1] 17</code></pre>
<pre class="r"><code># by Zoom, Anomalous atrioventricular excitation
# -&gt; Anomalous atrioventricular excitation (disorder)
# http://snomed.info/id/17869006
unmapped_traits &lt;-
  data.frame(&quot;DISEASE/TRAIT&quot; = &quot;Anomalous atrioventricular excitation (PheCode 426.4)&quot;,
             MAPPED_TRAIT = &quot;anomalous atrioventricular excitation (disorder)&quot;,
             MAPPED_TRAIT_URI = &quot;http://snomed.info/id/17869006&quot;,
             stringsAsFactors = FALSE
             ) |&gt;
  rename(`DISEASE/TRAIT` = &quot;DISEASE.TRAIT&quot;)

# by Zooma,  Pilocytic astrocytoma -&gt; MONDO_0016691 (http://purl.obolibrary.org/obo/MONDO_0016691)
unmapped_traits &lt;-
  unmapped_traits |&gt;
  add_row(`DISEASE/TRAIT` = &quot;Pilocytic astrocytoma&quot;,
             MAPPED_TRAIT = &quot;pilocytic astrocytoma&quot;,
             MAPPED_TRAIT_URI = &quot;http://purl.obolibrary.org/obo/MONDO_0016691&quot;
             )

# by Zooma, Pilocytic astrocytoma and optic pathway glioma
# -&gt; http://purl.obolibrary.org/obo/MONDO_0016167, http://purl.obolibrary.org/obo/MONDO_0016691
unmapped_traits =
  unmapped_traits |&gt;
  add_row(`DISEASE/TRAIT` = &quot;Pilocytic astrocytoma and optic pathway glioma&quot;,
          MAPPED_TRAIT = &quot;optic pathway glioma, pilocytic astrocytoma&quot;,
          MAPPED_TRAIT_URI = &quot;http://purl.obolibrary.org/obo/MONDO_0016167, http://purl.obolibrary.org/obo/MONDO_0016691&quot;
          )

# by searching ontology lookup service:
# Leukotriene levels (480.2454_0.351) &amp; Leukotriene levels (337.1632_0.339)
# -&gt; Fatty Acid Measurement
# http://purl.obolibrary.org/obo/NCIT_C80157
unmapped_traits =
  unmapped_traits |&gt;
  add_row(`DISEASE/TRAIT` = c(&quot;Leukotriene levels (480.2454_0.351)&quot;,
                               &quot;Leukotriene levels (337.1632_0.339)&quot;
                               ),
          MAPPED_TRAIT = c(&quot;fatty acid measurement&quot;,
                           &quot;fatty acid measurement&quot;
                           ),
          MAPPED_TRAIT_URI = c(&quot;http://purl.obolibrary.org/obo/NCIT_C80157&quot;,
                               &quot;http://purl.obolibrary.org/obo/NCIT_C80157&quot;
                               )
          )

# by searching ontology lookup service:
# X-11244 levels
# X-11255 levels 
# to be mapped to: http://www.ebi.ac.uk/efo/EFO_0004725
# metabolite measurement
unmapped_traits =
  unmapped_traits |&gt;
  add_row(`DISEASE/TRAIT` = c(&quot;X-11244 levels&quot;,
                               &quot;X-11255 levels&quot;
                               ),
          MAPPED_TRAIT = c(&quot;metabolite measurement&quot;,
                           &quot;metabolite measurement&quot;
                           ),
          MAPPED_TRAIT_URI = c(&quot;http://www.ebi.ac.uk/efo/EFO_0004725&quot;,
                               &quot;http://www.ebi.ac.uk/efo/EFO_0004725&quot;
                               )
          )

# by searching ontology lookup service:
# N-acetylornithine levels, &amp; N-acetylornithine levels in chronic kidney disease
# to be mapped to http://www.ebi.ac.uk/efo/EFO_0021538
# N-acetylornithine measurement
unmapped_traits =
  unmapped_traits |&gt;
  add_row(`DISEASE/TRAIT` = c(&quot;N-acetylornithine levels&quot;,
                               &quot;N-acetylornithine levels in chronic kidney disease&quot;
                               ),
          MAPPED_TRAIT = c(&quot;n-acetylornithine measurement&quot;,
                           &quot;n-acetylornithine measurement&quot;
                           ),
          MAPPED_TRAIT_URI = c(&quot;http://www.ebi.ac.uk/efo/EFO_0021538&quot;,
                               &quot;http://www.ebi.ac.uk/efo/EFO_0021538&quot;
                               )
          )

# by searching ontology lookup service:
# Scleritis and episcleritis (PheCode 379.1)
# map to: Scleritis and episcleritis (disorder)
# http://snomed.info/id/267659002
unmapped_traits =
  unmapped_traits |&gt;
  add_row(`DISEASE/TRAIT` = &quot;Scleritis and episcleritis (PheCode 379.1)&quot;,
          MAPPED_TRAIT = &quot;scleritis and episcleritis (disorder)&quot;,
          MAPPED_TRAIT_URI = &quot;http://snomed.info/id/267659002&quot;
          )

gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(MAPPED_TRAIT == &quot;&quot;,
                               NA,
                               MAPPED_TRAIT))

gwas_study_info =
  gwas_study_info |&gt;
  rows_patch(unmapped_traits,
             by = c(&quot;DISEASE/TRAIT&quot;),
             unmatched = &quot;ignore&quot;
             )

# yay all mapped now
print(&quot;After fixing, how many unmapped traits remain?&quot;)</code></pre>
<pre><code>[1] &quot;After fixing, how many unmapped traits remain?&quot;</code></pre>
<pre class="r"><code>gwas_study_info |&gt;
  filter(is.na(MAPPED_TRAIT) | MAPPED_TRAIT == &quot;&quot;) |&gt;
  nrow()</code></pre>
<pre><code>[1] 0</code></pre>
</div>
<div id="correcting-some-mapped_trait-mapped-by-gwas-catalog-terms"
class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Correcting some
MAPPED_TRAIT (MAPPED by GWAS Catalog) terms</h2>
<div id="re-map-some-traits-based-on-diseasetrait"
class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Re-map some traits
based on DISEASE/TRAIT</h3>
<pre class="r"><code># Periodontal disease related phenotype
# set trait to periodontal disorder
# rather than periodontal measurement
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;Periodontal disease related phenotype&quot;, `DISEASE/TRAIT`),
                               str_replace_all(pattern = &quot;periodontal measurement&quot;,
                                               replacement = &quot;periodontal disorder&quot;,
                                               MAPPED_TRAIT
                                               ),
                               MAPPED_TRAIT)
         ) |&gt;
    mutate(MAPPED_TRAIT_URI = ifelse(grepl(&quot;Periodontal disease related phenotype&quot;, `DISEASE/TRAIT`),
                               str_replace_all(pattern = &quot;http://www.ebi.ac.uk/efo/EFO_0007780&quot;,
                                               replacement = &quot;http://purl.obolibrary.org/obo/MONDO_0002635&quot;,
                                               MAPPED_TRAIT_URI
                                               ),
                               MAPPED_TRAIT_URI)
    )


# DISEASE/TRAIT == &quot;Spontaneous preterm birth (preterm delivery)&quot;
# MAPPED_TRAIT == &quot;delivery measurement&quot;
# MAPPED_TRAIT_URI == http://www.ebi.ac.uk/efo/EFO_0006922

# make MAPPED_TRAIT to &quot;premature birth&quot;
# MAPPED_TRAIT_URI to http://www.ebi.ac.uk/efo/EFO_0003917
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;Spontaneous preterm birth \\(preterm delivery\\)&quot;, 
                                     `DISEASE/TRAIT`) &amp; 
                                MAPPED_TRAIT == &quot;delivery measurement&quot;,
                               &quot;premature birth&quot;,
                               MAPPED_TRAIT)
         ) |&gt;
    mutate(MAPPED_TRAIT_URI = ifelse(grepl(&quot;Spontaneous preterm birth \\(preterm delivery\\)&quot;, `DISEASE/TRAIT`) &amp; 
                                      MAPPED_TRAIT_URI == &quot;http://www.ebi.ac.uk/efo/EFO_0006922&quot;,
                                     &quot;http://www.ebi.ac.uk/efo/EFO_0003917&quot;,
                                     MAPPED_TRAIT_URI)
           )

## DISEASE/TRAIT == &quot;Sporadic miscarriage&quot;
## MAPPED_TRAIT == &quot;fertility trait&quot;
## MAPPED_TRAIT_URI == &quot;http://purl.obolibrary.org/obo/OBA_VT0010464&quot;

# make MAPPED_TRAIT to &quot;spontaneous abortion&quot;
# MAPPED_TRAIT_URI to &quot;http://www.ebi.ac.uk/efo/EFO_1001255&quot;

gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;Sporadic miscarriage&quot;, `DISEASE/TRAIT`) &amp; 
                                MAPPED_TRAIT == &quot;fertility trait&quot;,
                               &quot;spontaneous abortion&quot;,
                               MAPPED_TRAIT)
         ) |&gt;
    mutate(MAPPED_TRAIT_URI = ifelse(grepl(&quot;Sporadic miscarriage&quot;, `DISEASE/TRAIT`) &amp; 
                                      MAPPED_TRAIT_URI == &quot;http://purl.obolibrary.org/obo/OBA_VT0010464&quot;,
                                     &quot;http://www.ebi.ac.uk/efo/EFO_1001255&quot;,
                                     MAPPED_TRAIT_URI)
           )

# Multiple consecutive miscarriage
## MAPPED_TRAIT == &quot;fertility trait&quot;
## MAPPED_TRAIT_URI == &quot;http://purl.obolibrary.org/obo/OBA_VT0010464&quot;

# make MAPPED_TRAIT to &quot;spontaneous abortion&quot;
# MAPPED_TRAIT_URI to &quot;http://www.ebi.ac.uk/efo/EFO_1001255&quot;

gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;Multiple consecutive miscarriage&quot;, `DISEASE/TRAIT`) &amp; 
                                MAPPED_TRAIT == &quot;fertility trait&quot;,
                               &quot;spontaneous abortion&quot;,
                               MAPPED_TRAIT)
         ) |&gt;
    mutate(MAPPED_TRAIT_URI = ifelse(grepl(&quot;Multiple consecutive miscarriage&quot;, `DISEASE/TRAIT`) &amp; 
                                      MAPPED_TRAIT_URI == &quot;http://purl.obolibrary.org/obo/OBA_VT0010464&quot;,
                                     &quot;http://www.ebi.ac.uk/efo/EFO_1001255&quot;,
                                     MAPPED_TRAIT_URI)
           )


## DISEASE/TRAIT == &quot;Post-term birth&quot;
## MAPPED_TRAIT == &quot;gestational age&quot;
## MAPPED_TRAIT_URI == &quot;http://www.ebi.ac.uk/efo/EFO_0005112&quot;

# make MAPPED_TRAIT to &quot;post term pregnancy&quot;
# MAPPED_TRAIT_URI to &quot;http://www.ebi.ac.uk/efo/EFO_0009681&quot;

gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;Post-term birth&quot;, `DISEASE/TRAIT`) &amp; 
                                MAPPED_TRAIT == &quot;gestational age&quot;,
                               &quot;post term pregnancy&quot;,
                               MAPPED_TRAIT)
         ) |&gt;
    mutate(MAPPED_TRAIT_URI = ifelse(grepl(&quot;Post-term birth&quot;, `DISEASE/TRAIT`) &amp; 
                                      MAPPED_TRAIT_URI == &quot;http://www.ebi.ac.uk/efo/EFO_0005112&quot;,
                                     &quot;http://www.ebi.ac.uk/efo/EFO_0009681&quot;,
                                     MAPPED_TRAIT_URI)
           )</code></pre>
</div>
<div id="add-some-missing-mapped_background_trait-terms"
class="section level3" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Add some missing
MAPPED_BACKGROUND_TRAIT terms</h3>
<pre class="r"><code># for MAPPED_TRAIT contains &quot;sars-cov-2&quot;
# set background trait to covid-19
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(grepl(&quot;sars-cov-2&quot;, MAPPED_TRAIT),
                                         &quot;covid-19&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(grepl(&quot;sars-cov-2&quot;, MAPPED_TRAIT),
                                                &quot;http://purl.obolibrary.org/obo/MONDO_0100096&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# for pubmed id: 32247823
# set background trait to non-alcoholic steatohepatitis
# and background trait uri to EFO_1001249
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(PUBMED_ID == 32247823,
                                         &quot;non-alcoholic steatohepatitis&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(PUBMED_ID == 32247823,
                                                &quot;http://www.ebi.ac.uk/efo/EFO_1001249&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# Exploratory eye movement dysfunction in schizophrenia
# set background trait to schizophrenia
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(grepl(&quot;Exploratory eye movement dysfunction in schizophrenia&quot;, `DISEASE/TRAIT`),
                                         &quot;schizophrenia&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(grepl(&quot;Exploratory eye movement dysfunction in schizophrenia&quot;, `DISEASE/TRAIT`),
                                                &quot;http://purl.obolibrary.org/obo/MONDO_0005090&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# for pubmed_id: 21107309
# set background trait to schizophrenia
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(PUBMED_ID == 21107309,
                                         &quot;schizophrenia&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(PUBMED_ID == 21107309,
                                                &quot;http://purl.obolibrary.org/obo/MONDO_0005090&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# if `DISEASE/TRAIT` contains Adverse response to chemotherapy in breast cancer
# set MAPPED_BACKGROUND_TRAIT to breast cancer
# http://purl.obolibrary.org/obo/MONDO_0007254
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(grepl(&quot;Adverse response to chemotherapy in breast cancer&quot;, `DISEASE/TRAIT`),
                                         &quot;breast cancer&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(grepl(&quot;Adverse response to chemotherapy in breast cancer&quot;, `DISEASE/TRAIT`),
                                                &quot;http://purl.obolibrary.org/obo/MONDO_0007254&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# for pubmed id: 30188897
# and DISEASE/TRAIT contains &quot;miscarriages&quot;
# add spontaneous abortion, http://www.ebi.ac.uk/efo/EFO_1001255
# as MAPPED_BACKGROUND_TRAIT
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(PUBMED_ID == 30188897 &amp; 
                                           grepl(&quot;miscarriage&quot;, `DISEASE/TRAIT`),
                                         &quot;spontaneous abortion&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(PUBMED_ID == 30188897 &amp; 
                                                  grepl(&quot;miscarriage&quot;, `DISEASE/TRAIT`),
                                                &quot;http://www.ebi.ac.uk/efo/EFO_1001255&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# for pubmed id: 30188897
# and DISEASE/TRAIT contains &quot;stillbirth&quot;
# add stillbirth, http://purl.obolibrary.org/obo/NCIT_C49151
# as MAPPED_BACKGROUND_TRAIT
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(PUBMED_ID == 30188897 &amp; 
                                           grepl(&quot;stillbirth&quot;, `DISEASE/TRAIT`),
                                         &quot;stillbirth&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(PUBMED_ID == 30188897 &amp; 
                                                  grepl(&quot;stillbirth&quot;, `DISEASE/TRAIT`),
                                                &quot;http://purl.obolibrary.org/obo/NCIT_C49151&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# if MAPPED_TRAIT == &quot;response to covid-19 vaccine&quot;
# set MAPPED_BACKGROUND_TRAIT to covid-19
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(MAPPED_TRAIT == &quot;response to covid-19 vaccine&quot;,
                                         &quot;covid-19&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(MAPPED_TRAIT == &quot;response to covid-19 vaccine&quot;,
                                                &quot;http://purl.obolibrary.org/obo/MONDO_0100096&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )


    
# if MAPPED_TRAIT == &quot;response to vaccine&quot;
# and DISEASE/TRAIT contains &quot;smallpox&quot; 
# set MAPPED_BACKGROUND_TRAIT to smallpox
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                           grepl(&quot;smallpox&quot;, `DISEASE/TRAIT`),
                                         &quot;smallpox&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                                  grepl(&quot;smallpox&quot;, `DISEASE/TRAIT`),
                                                &quot;http://purl.obolibrary.org/obo/DOID_8736&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# if MAPPED_TRAIT == &quot;response to vaccine&quot;
# and DISEASE/TRAIT contains &quot;influenza&quot;
# set MAPPED_BACKGROUND_TRAIT to influenza
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                           grepl(&quot;influenza&quot;, `DISEASE/TRAIT`),
                                         &quot;influenza&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                                  grepl(&quot;influenza&quot;, `DISEASE/TRAIT`),
                                                &quot;http://www.ebi.ac.uk/efo/EFO_0007328&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# if. MAPPED_TRAIT ==  &quot;response to vaccine&quot;
# and DISEASE/TRAIT contains &quot;hepatitis B&quot;
# set MAPPED_BACKGROUND_TRAIT to hepatitis B
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                           grepl(&quot;hepatitis B|Hepatitis B&quot;, `DISEASE/TRAIT`),
                                         &quot;hepatitis b&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                                  grepl(&quot;hepatitis B|Hepatitis B&quot;, `DISEASE/TRAIT`),
                                                &quot;http://purl.obolibrary.org/obo/DOID_2043&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )


# if MAPPED_TRAIT == &quot;response to covid-19 vaccine&quot;
# and MAPPED_BACKGROUND_TRAIT == &quot;&quot;
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(MAPPED_TRAIT == &quot;response to covid-19 vaccine&quot; &amp;
                                           (is.na(MAPPED_BACKGROUND_TRAIT) | MAPPED_BACKGROUND_TRAIT == &quot;&quot;),
                                         &quot;covid-19&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(MAPPED_TRAIT == &quot;response to covid-19 vaccine&quot; &amp;
                                                  (is.na(MAPPED_BACKGROUND_TRAIT_URI) | MAPPED_BACKGROUND_TRAIT_URI == &quot;&quot;),
                                                &quot;http://purl.obolibrary.org/obo/MONDO_0100096&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# if MAPPED_TRAIT == &quot;response to vaccine&quot;
# and MAPPED_BACKGROUND_TRAIT == &quot;&quot;
# and DISEASE/TRAIT contains &quot;Immune response to smallpox&quot;
# set MAPPED_BACKGROUND_TRAIT to smallpox

gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                           (is.na(MAPPED_BACKGROUND_TRAIT) | MAPPED_BACKGROUND_TRAIT == &quot;&quot;) &amp;
                                           grepl(&quot;Immune response to smallpox&quot;, `DISEASE/TRAIT`),
                                         &quot;smallpox&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                                  (is.na(MAPPED_BACKGROUND_TRAIT_URI) | MAPPED_BACKGROUND_TRAIT_URI == &quot;&quot;) &amp;
                                                  grepl(&quot;Immune response to smallpox&quot;, `DISEASE/TRAIT`),
                                                &quot;http://purl.obolibrary.org/obo/OMIT_0013787&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )

# if MAPPED_TRAIT == &quot;response to vaccine&quot;
# and MAPPED_BACKGROUND_TRAIT == &quot;&quot;
# and DISEASE/TRAIT contains &quot;Immune response to measles vaccine&quot;
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                           (is.na(MAPPED_BACKGROUND_TRAIT) | MAPPED_BACKGROUND_TRAIT == &quot;&quot;) &amp;
                                           grepl(&quot;Immune response to measles vaccine&quot;, `DISEASE/TRAIT`),
                                         &quot;measles&quot;,
                                         MAPPED_BACKGROUND_TRAIT)
         ) |&gt;
    mutate(MAPPED_BACKGROUND_TRAIT_URI = ifelse(MAPPED_TRAIT == &quot;response to vaccine&quot; &amp;
                                                  (is.na(MAPPED_BACKGROUND_TRAIT_URI) | MAPPED_BACKGROUND_TRAIT_URI == &quot;&quot;) &amp;
                                                  grepl(&quot;Immune response to measles vaccine&quot;, `DISEASE/TRAIT`),
                                                &quot;http://purl.obolibrary.org/obo/DOID_8622&quot;,
                                                MAPPED_BACKGROUND_TRAIT_URI)
           )</code></pre>
</div>
</div>
<div id="replacing-for-inside-a-mapped_term" class="section level2"
number="1.4">
<h2><span class="header-section-number">1.4</span> Replacing * for ,
inside a MAPPED_TERM</h2>
<p>To ensure splitting MAPPED_TERM column by commas would split distinct
traits</p>
<pre class="r"><code># in MAPPED_BACKGROUND_TRAIT, replace commas with &quot;*&quot; in:
# migraine without aura, susceptibility to, 4
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = ifelse(grepl(&quot;migraine without aura, susceptibility to, 4&quot;, MAPPED_BACKGROUND_TRAIT),
                                         stringr::str_replace_all(MAPPED_BACKGROUND_TRAIT, 
                                                                  pattern = &quot;migraine without aura, susceptibility to, 4&quot;, 
                                                                  &quot;migraine without aura* susceptibility to* 4&quot;),
                                         MAPPED_BACKGROUND_TRAIT)
         )

# in MAPPED_TRAIT, replace commas with &quot;*&quot; in:
# migraine without aura, susceptibility to, 4
gwas_study_info =
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;migraine without aura, susceptibility to, 4&quot;, MAPPED_TRAIT),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;migraine without aura, susceptibility to, 4&quot;, 
                                                        &quot;migraine without aura* susceptibility to* 4&quot;),
                               MAPPED_TRAIT)
         )

# Other MAPPED_TRAIT fixes
gwas_study_info =
gwas_study_info |&gt; 
  mutate(MAPPED_TRAIT = case_when(
                         # osteoarthritis, hip ... http://www.ebi.ac.uk/efo/EFO_1000786
                         grepl(&quot;EFO_1000786&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;osteoarthritis, hip&quot;, 
                                                                                           &quot;osteoarthritis* hip&quot;),
                         
                         #  osteoarthritis, hand ... http://www.ebi.ac.uk/efo/EFO_1000789
                         grepl(&quot;EFO_1000789&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;osteoarthritis, hand&quot;, 
                                                                                           &quot;osteoarthritis* hand&quot;
                                                                                           ),
                         # osteoarthritis, knee ... http://www.ebi.ac.uk/efo/EFO_0004616
                         grepl(&quot;EFO_0004616&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;osteoarthritis, knee&quot;, 
                                                                                           &quot;osteoarthritis* knee&quot;),
                         
                         #  osteoarthritis, spine ... http://www.ebi.ac.uk/efo/EFO_1000787
                         grepl(&quot;EFO_1000787&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;osteoarthritis, spine&quot;, 
                                                                                           &quot;osteoarthritis* spine&quot;),
                         
                         #  Hepatitis, Alcoholic, http://www.ebi.ac.uk/efo/EFO_1001345
                         grepl(&quot;EFO_1001345&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;hepatitis, alcoholic&quot;, 
                                                                                           &quot;hepatitis* alcoholic&quot;),
                         
                         # psoriasis 14, pustular http://purl.obolibrary.org/obo/MONDO_0013626
                         grepl(&quot;MONDO_0013626&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                             pattern = &quot;psoriasis 14, pustular&quot;, 
                                                                                             &quot;psoriasis 14* pustular&quot;),
                         
                         # hypertension, pregnancy-induced http://purl.obolibrary.org/obo/MONDO_0024664
                         grepl(&quot;MONDO_0024664&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                             pattern = &quot;hypertension, pregnancy-induced&quot;, 
                                                                                             &quot;hypertension* pregnancy-induced&quot;),
                         
                         # renal agenesis, unilateral http://purl.obolibrary.org/obo/MONDO_0019636
                         grepl(&quot;MONDO_0019636&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                             pattern = &quot;renal agenesis, unilateral&quot;, 
                                                                                             &quot;renal agenesis* unilateral&quot;),
                         
                         #  Cholecystitis, Acute http://www.ebi.ac.uk/efo/EFO_1001289
                         grepl(&quot;EFO_1001289&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;cholecystitis, acute&quot;, 
                                                                                           &quot;cholecystitis* acute&quot;),
                         #  Genital neoplasm, female http://www.ebi.ac.uk/efo/EFO_1001331
                         grepl(&quot;EFO_1001331&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;genital neoplasm, female&quot;, 
                                                                                           &quot;genital neoplasm* female&quot;),
                         #  Anemia, Hemolytic, Autoimmune http://www.ebi.ac.uk/efo/EFO_1001264
                          grepl(&quot;EFO_1001264&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                            pattern = &quot;anemia, hemolytic, autoimmune&quot;,
                                                                                            &quot;anemia* hemolytic* autoimmune&quot;),
                         grepl(&quot;EFO_1002020&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                          pattern = &quot;polyarticular juvenile idiopathic arthritis, rheumatoid factor negative&quot;, 
                                                                                          &quot;polyarticular juvenile idiopathic arthritis* rheumatoid factor negative&quot;),
                         # http://www.ebi.ac.uk/efo/EFO_0007294, hand, foot and mouth disease, 
                         grepl(&quot;EFO_0007294&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                          pattern = &quot;hand, foot and mouth disease&quot;, 
                                                                                          &quot;hand* foot and mouth disease&quot;),
                         # neural tube defects, susceptibility to, http://purl.obolibrary.org/obo/MONDO_0020705
                         grepl(&quot;MONDO_0020705&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                             pattern = &quot;neural tube defects, susceptibility to&quot;, 
                                                                                             &quot;neural tube defects* susceptibility to&quot;),
                         # self-reported traits
                         grepl(&quot;EFO_0009803|EFO_0009822|EFO_0009803|EFO_0009817|EFO_0009822|EFO_0009819|EFO_0009823|EFO_0009824&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                          pattern = &quot;, self-reported$&quot;, 
                                                                                          &quot;* self-reported&quot;),
                         # Hodgkins lymphoma, mixed cellularity http://www.ebi.ac.uk/efo/EFO_1002031
                         grepl(&quot;EFO_1002031&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                          pattern = &quot;hodgkins lymphoma, mixed cellularity&quot;, 
                                                                                          &quot;hodgkins lymphoma* mixed cellularity&quot;),
                         # encephalopathy, acute, infection-induced, http://purl.obolibrary.org/obo/MONDO_0000166
                         grepl(&quot;MONDO_0000166&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                             pattern = &quot;encephalopathy, acute, infection-induced&quot;, 
                                                                                             &quot;encephalopathy* acute* infection-induced&quot;),
                         # Diarrhea, Infantile http://www.ebi.ac.uk/efo/EFO_1001306
                         grepl(&quot;EFO_1001306&quot;, MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, 
                                                                                           pattern = &quot;diarrhea, infantile&quot;, 
                                                                                           &quot;diarrhea* infantile&quot;),
                         
                         TRUE ~ MAPPED_TRAIT
                         )
  ) 


# in MAPPED_TRAIT, replace commas with &quot;*&quot; in:
# chromosome, telomeric region length
# fractures, ununited
# osteoarthritis, knee ... http://www.ebi.ac.uk/efo/EFO_0004616
# localized superficial swelling, mass, or lump
# cys-gly, oxidized measurement
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;chromosome, telomeric region length&quot;, MAPPED_TRAIT),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;chromosome, telomeric region length&quot;, 
                                                        &quot;chromosome* telomeric region length&quot;),
                               MAPPED_TRAIT)
         ) |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;fractures, ununited&quot;, MAPPED_TRAIT),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;fractures, ununited&quot;, 
                                                        &quot;fractures* ununited&quot;),
                               MAPPED_TRAIT)
         ) |&gt;
  mutate(MAPPED_TRAIT =                          # osteoarthritis, knee ... http://www.ebi.ac.uk/efo/EFO_0004616
                         ifelse(grepl(&quot;EFO_0004616&quot;, MAPPED_TRAIT_URI),
                                stringr::str_replace_all(MAPPED_TRAIT, 
                                                         pattern = &quot;osteoarthritis, knee&quot;, 
                                                         &quot;osteoarthritis* knee&quot;),
                                MAPPED_TRAIT)
         ) |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;localized superficial swelling, mass, or lump&quot;, MAPPED_TRAIT),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;localized superficial swelling, mass, or lump&quot;, 
                                                        &quot;localized superficial swelling* mass* or lump&quot;),
                               MAPPED_TRAIT)
         ) |&gt;
  mutate(MAPPED_TRAIT = ifelse(grepl(&quot;cys-gly, oxidized measurement&quot;, MAPPED_TRAIT),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = &quot;cys-gly, oxidized measurement&quot;, 
                                                        &quot;cys-gly* oxidized measurement&quot;),
                               MAPPED_TRAIT)
         )</code></pre>
<div id="fix-by-the-number-of-commas" class="section level3"
number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Fix by the number
of commas</h3>
<pre class="r"><code># fixing weird terms- where comma is in the term

# count number of separating commas in 
# MAPPED_TRAIT, MAPPED_TRAIT_URI, MAPPED_BACKGROUND_TRAIT, MAPPED_BACKGROUND_TRAIT_URI
gwas_study_info =
gwas_study_info |&gt;
  mutate(n_commas_trait = str_count(MAPPED_TRAIT, &quot;, (?![^()]*\\))&quot;),
                                    #&quot;, &quot;),
         n_commas_trait_uri = str_count(MAPPED_TRAIT_URI, &quot;, (?![^()]*\\))&quot;),
         n_commas_bg_trait = str_count(MAPPED_BACKGROUND_TRAIT, &quot;, (?![^()]*\\))&quot;),
         n_commas_bg_trait_uri = str_count(MAPPED_BACKGROUND_TRAIT_URI, &quot;, (?![^()]*\\))&quot;)
                                          # &quot;, &quot;)
         ) |&gt;
  # select(contains(&quot;n_commas&quot;), 
  #        MAPPED_TRAIT, MAPPED_TRAIT_URI, 
  #        MAPPED_BACKGROUND_TRAIT, MAPPED_BACKGROUND_TRAIT_URI
  #        ) |&gt;
  distinct()

# if n_commas_trait &gt;= 1, n_commas_trait_uri == 0,
# replace comma in MAPPED_TRAIT with &quot;*&quot;
gwas_study_info = 
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = ifelse(n_commas_trait &gt;= 1 &amp; 
                                  n_commas_trait_uri == 0,
                                stringr::str_replace_all(MAPPED_TRAIT, 
                                                         pattern = &quot;, &quot;, 
                                                         &quot;* &quot;),
                                MAPPED_TRAIT)
         )

# now that&#39;s been correct, recalculate number of commas
gwas_study_info = 
  gwas_study_info |&gt;
    mutate(n_commas_trait = str_count(MAPPED_TRAIT, &quot;, (?![^()]*\\))&quot;),
                                    #&quot;, &quot;),
         n_commas_trait_uri = str_count(MAPPED_TRAIT_URI, &quot;, (?![^()]*\\))&quot;),
         n_commas_bg_trait = str_count(MAPPED_BACKGROUND_TRAIT, &quot;, (?![^()]*\\))&quot;),
         n_commas_bg_trait_uri = str_count(MAPPED_BACKGROUND_TRAIT_URI, &quot;, (?![^()]*\\))&quot;)
                                          # &quot;, &quot;)
         ) 


# check the number of commas in MAPPED_TRAIT is always equal to number of commas in MAPPED_TRAIT_URI
# and the number of commas in MAPPED_BACKGROUND_TRAIT is not equal to number of commas in MAPPED_BACKGROUND_TRAIT_URI
gwas_study_info = 
  gwas_study_info |&gt;
  mutate(match_comma_trait = ifelse(n_commas_trait != n_commas_trait_uri,
                                   FALSE, 
                                   TRUE),
         match_comma_bg_trait = ifelse(n_commas_bg_trait != n_commas_bg_trait_uri,
                                      FALSE, 
                                      TRUE)
         )

gwas_study_info  |&gt;
  filter(match_comma_trait == FALSE |
         match_comma_bg_trait == FALSE)</code></pre>
<pre><code>Empty data.table (0 rows and 32 cols): DATE_ADDED_TO_CATALOG,PUBMED_ID,FIRST_AUTHOR,DATE,JOURNAL,LINK...</code></pre>
<pre class="r"><code># yay! all match now</code></pre>
</div>
</div>
</div>
<div id="overlap-ontology-terms-and-gwas-traits" class="section level1"
number="2">
<h1><span class="header-section-number">2</span> Overlap ontology terms
and GWAS traits</h1>
<div id="create-data.frame-of-gwas-traits-with-one-trait-per-row"
class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Create data.frame of
GWAS traits with one trait per row</h2>
<pre class="r"><code>gwas_study_info =
  gwas_study_info |&gt;
  select(
         `DISEASE/TRAIT`,
         PUBMED_ID,
         YEAR, 
         STUDY_ACCESSION,
         contains(&quot;MAPPED&quot;)
         )

# now split by commas to get each MAPPED_TRAIT on an individual row
gwas_study_info &lt;-
gwas_study_info |&gt;
  tidyr::separate_longer_delim(cols = c(&quot;MAPPED_TRAIT&quot;, 
                                        &quot;MAPPED_TRAIT_URI&quot;
                                        ),
                               delim = stringr::regex(&quot;, (?![^()]*\\))&quot;)
                               ) |&gt;
  tidyr::separate_longer_delim(cols = c(&quot;MAPPED_BACKGROUND_TRAIT&quot;, 
                                        &quot;MAPPED_BACKGROUND_TRAIT_URI&quot;
                                        ),
                               delim = stringr::regex(&quot;, (?![^()]*\\))&quot;)
  ) |&gt;
  distinct()

# now replace &#39;*&#39; back to commas
gwas_study_info =
  gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = stringr::str_replace_all(MAPPED_TRAIT, 
                                                 pattern = &quot;\\* &quot;, 
                                                 &quot;, &quot;),
         MAPPED_BACKGROUND_TRAIT = stringr::str_replace_all(MAPPED_BACKGROUND_TRAIT, 
                                                          pattern = &quot;\\* &quot;, 
                                                          &quot;, &quot;)
         )


gwas_study_info &lt;-
gwas_study_info |&gt;
  mutate(MAPPED_TRAIT = stringr::str_trim(tolower(MAPPED_TRAIT))) |&gt;
  mutate(MAPPED_BACKGROUND_TRAIT = stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT))) </code></pre>
<pre class="r"><code>all_gwas_terms = gwas_study_info$MAPPED_TRAIT
all_gwas_terms = stringr::str_trim(tolower(all_gwas_terms))
all_gwas_terms = unique(all_gwas_terms)

print(&quot;Number of unique GWAS traits&quot;)</code></pre>
<pre><code>[1] &quot;Number of unique GWAS traits&quot;</code></pre>
<pre class="r"><code>length(all_gwas_terms)</code></pre>
<pre><code>[1] 19975</code></pre>
</div>
<div
id="disease-overlap-how-many-gwas-traits-fall-within-disease-or-disorder-terms"
class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Disease Overlap (How
many GWAS traits fall within disease or disorder terms?)</h2>
<div id="combine-disease-terms" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Combine disease
terms</h3>
<pre class="r"><code>efo_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000408_descendants.txt&quot;))

mondo_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/mondo_0700096_descendants.txt&quot;))

ncit_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/ncit_C2991_descendants.txt&quot;))

orphanet_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/orphanet_557493_descendants.txt&quot;))

age_of_onset_descendants &lt;- readLines(here::here(&quot;output/trait_ontology/oba_2020000_descendants.txt&quot;))

disease_measurement_terms &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0001444_disease_measurement_terms.txt&quot;))

disease_typos = c(&quot;Alzheimer disease&quot;,
                  &quot;late-onset Alzheimers disease&quot;,
                  &quot;Chagas cardiomyopathy&quot;,
                  &quot;Parkinson disease&quot;,
                  &quot;Iron deficiency anemia&quot;
                  )

biomarker_terms &lt;- c(&quot;cardiovascular disease biomarker measurement&quot;,
                     &quot;cancer biomarker measurement&quot;,
                     &quot;diabetes mellitus biomarker&quot;,
                     &quot;osteoarthritis biomarker measurement&quot;,
                     &quot;liver disease biomarker&quot;,
                     &quot;alzheimer&#39;s disease biomarker measurement&quot;,
                     &quot;iron deficiency anemia (disorder)&quot;
                     )

other_disorders &lt;- c(
           &quot;Allergic disease&quot;, 
           &quot;Churg-Strauss syndrome&quot;,
           &quot;Iridocyclitis&quot;,
           &quot;Phlebitis&quot;,
           &quot;pregnancy induced alloimmunization&quot;,
           &quot;somnambulism&quot;,
           &quot;suicide&quot;,
           &quot;attempted suicide&quot;,
           &quot;suicide behaviour&quot;,
           &quot;suicide ideation measurement&quot;,
           &quot;suicide behaviour measurement&quot;,
           &quot;Lewy body dementia&quot;,
           &quot;Lewy body attribute&quot;,
           &quot;non-Hodgkins lymphoma&quot;,
           &quot;Ischemic Stroke&quot;,
           &quot;Lung disease&quot;,
           &quot;Respiratory System Disease&quot;,
           &quot;Alzheimer disease, APOE carrier status&quot;,
           &quot;Genital neoplasm, female&quot;,
           &quot;HIV-associated neurocognitive disorder&quot;,
           &quot;encephalopathy acute infection-induced&quot;,
           &quot;anomalous atrioventricular excitation (disorder)&quot;,
           &quot;scleritis and episcleritis (disorder)&quot;,
           &quot;atopic march&quot;,
           &quot;infection&quot;,
           &quot;neural tube defects, susceptibility to&quot;,
           &quot;migraine without aura, susceptibility to, 4&quot;,
           &quot;hiv mother to child transmission&quot;,
           &quot;hemolysis&quot;,
           &quot;chromosomal aberration&quot;,
           &quot;dna methylation&quot;,
           &quot;gata1 gene mutation&quot;,
           &quot;atropy&quot;,
           &quot;premature birth&quot;,
           &quot;growth delay&quot;,
           &quot;reduced left ventricular ejection fraction&quot;,
           &quot;hepatitis b&quot;,
           &quot;vascular brain injury measurement&quot;,
           &quot;borderline personality disorder symptom&quot;,
           &quot;miscarriage&quot;,
           &quot;emphysema pattern measurement&quot;,
           &quot;emphysema imaging measurement&quot;,
           &quot;persistent staphylococcus aureus carrier status&quot;,
           &quot;intermittent staphylococcus aureus carrier status&quot;,
           &quot;influenza a severity measurement&quot;,
           &quot;pneumonia severity measurement&quot;,
           &quot;hsv2 virologic severity measurement&quot;,
           &quot;opioid overdose severity measurement&quot;,
           &quot;nausea and vomiting of pregnancy severity measurement&quot;,
           &quot;myopic maculopathy severity measurement&quot;
)
 
 
 disease_status_terms &lt;- c(
 &quot;benign&quot;,
 &quot;remission&quot;,
 &quot;disease recurrence&quot;,
 &quot;complicated disease course&quot;,
 &quot;disease prognosis measurement&quot;,
 &quot;mild disease course&quot;,
 &quot;disease free survival&quot;,
 &quot;progression free survival&quot;,
 &quot;survival time&quot;,
 &quot;overall survival&quot;,
 &quot;illness severity status&quot;
 )

family_disease_terms &lt;- c(&quot;family history of breast cancer&quot;,
                          &quot;family history of cancer&quot;,
                          &quot;family history of prostate cancer&quot;,
                          &quot;family history of upper gastrointestinal cancer&quot;,
                          &quot;family history of uterine fibroids&quot;)

disease_terms = c(mondo_descendants,
                  efo_descendants,
                  ncit_descendants,
                  orphanet_descendants,
                  age_of_onset_descendants,
                  disease_measurement_terms,
                  family_disease_terms,
                  disease_typos,
                  biomarker_terms,
                  disease_status_terms,
                  other_disorders) |&gt;
                  unique()


disease_terms = stringr::str_trim(tolower(disease_terms))
disease_terms = unique(disease_terms)

print(&quot;Number of ontology terms found related to disease or disorder&quot;)</code></pre>
<pre><code>[1] &quot;Number of ontology terms found related to disease or disorder&quot;</code></pre>
<pre class="r"><code>length(disease_terms)</code></pre>
<pre><code>[1] 45146</code></pre>
</div>
<div id="find-disease-terms-in-gwas-traits" class="section level3"
number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Find disease terms
in GWAS traits</h3>
<pre class="r"><code># Find GWAS traits that fall within disease or disorder terms
#simple_disease_terms = all_gwas_terms[all_gwas_terms %in% disease_terms]
disease_gwas &lt;- all_gwas_terms[all_gwas_terms %in% disease_terms]

# # Also search for cases where there are multiple terms separated by commas
# # and one of them is a disease term
# not_simple_disease_terms = all_gwas_terms[!all_gwas_terms %in% disease_terms]
# 
# # sometimes there&#39;s multiple terms - check if any disease term is in these gwas terms
# multiple_terms = grep(&quot;,&quot;, 
#                       not_simple_disease_terms, value = T)
# 
# disease_chunks &lt;- split(disease_terms, ceiling(seq_along(disease_terms) / 100))
# disease_chunks  &lt;- lapply(disease_chunks, function(x) paste0(x, collapse = &quot;|&quot;))
# mask &lt;- Reduce(`|`, lapply(disease_chunks, function(x) grepl(x, multiple_terms, ignore.case = T)))
# additional_disease_gwas &lt;- multiple_terms[mask]

# disease_gwas = c(all_gwas_terms[all_gwas_terms %in% disease_terms],
#                  additional_disease_gwas)

#not_disease_terms = not_simple_disease_terms[!not_simple_disease_terms %in% additional_disease_gwas]
not_disease_terms = all_gwas_terms[!all_gwas_terms %in% disease_gwas]

print(&quot;Number of GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>length(disease_gwas)</code></pre>
<pre><code>[1] 1978</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * (length(disease_gwas)) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 9.9</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits not under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_disease_terms) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 90.1</code></pre>
<pre class="r"><code>not_accounted_for = not_disease_terms </code></pre>
</div>
</div>
<div id="phenotype-abnormality-overlap" class="section level2"
number="2.3">
<h2><span class="header-section-number">2.3</span> Phenotype abnormality
overlap</h2>
<pre class="r"><code>pheno_abnorm &lt;- readLines(here::here(&quot;output/trait_ontology/hp_0000118_descendants.txt&quot;))
pheno_abnorm = stringr::str_trim(tolower(pheno_abnorm))
pheno_abnorm = unique(pheno_abnorm)

pheno_abnorm &lt;- c(&quot;abnormal pap smear&quot;,
                  &quot;abnormal result of function studies&quot;,
                  &quot;abnormal result of diagnostic imaging&quot;,
                  pheno_abnorm)

# # Find terms where all comma-split pieces are in measurement
# pheno_abnorm_gwas &lt;- not_accounted_for[
#   sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
#     parts &lt;- trimws(parts) # remove extra spaces
#     all(parts %in% pheno_abnorm)
#   })
# ]

#additional_pheno_abnorm &lt;- not_accounted_for[not_accounted_for %in% pheno_abnorm]
pheno_abnorm_gwas &lt;- not_accounted_for[not_accounted_for %in% pheno_abnorm]

#pheno_abnorm_gwas  = c(pheno_abnorm_gwas, additional_pheno_abnorm) |&gt; unique()

print(&quot;Percentage of GWAS traits under phenotype abnormality terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under phenotype abnormality terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(pheno_abnorm_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.9</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% pheno_abnorm_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 88.2</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 17623</code></pre>
</div>
<div
id="measurement-overlap-how-many-gwas-traits-fall-within-measurement-terms"
class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Measurement Overlap
(how many GWAS traits fall within measurement terms?)</h2>
<div id="combine-measurement" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Combine
measurement</h3>
<pre class="r"><code>measurement &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0001444_descendants.txt&quot;))
total_choles &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0004574_descendants.txt&quot;))
measurement &lt;- c(total_choles,
                 measurement) 

measurement &lt;- unique(measurement)
measurement &lt;- c(&quot;cerebrospinal fluid composition attribute&quot;,
                 &quot;blood protein amount&quot;,
                 &quot;fatty acid measurement&quot;,
                 &quot;obsolete_3,3&#39;,5-triiodo-l-thyronine measurement&quot;,
                 &quot;1-(1-enyl-stearoyl)-2-linoleoyl-gpe (p-18:0/18:2), measurement&quot;,
                 &quot;microtubule-associated protein tau&quot;,
                 measurement)

measurement = stringr::str_trim(tolower(measurement))
measurement = unique(measurement)</code></pre>
</div>
<div id="bmi-weight-terms-body-fat-terms" class="section level3"
number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> BMI / weight terms
/ body fat terms</h3>
<pre class="r"><code>bmi_weight_terms &lt;- grep(&quot;bmi|body mass index|weight|bmi&quot;, measurement, value = T)
bmi_weight_terms &lt;- grep(&quot;fetal|birth|gestational&quot;, bmi_weight_terms, value = T, invert = T)
bmi_weight_terms &lt;- c(bmi_weight_terms, 
                      &quot;body composition measurement&quot;,
                      &quot;body fat percentage&quot;,
                      &quot;body fat distribution&quot;)

measurement &lt;- measurement[!(measurement %in% bmi_weight_terms)]

bmi_weight_gwas = not_accounted_for[not_accounted_for %in% bmi_weight_terms]
print(&quot;Number of GWAS traits under BMI / weight / body fat terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under BMI / weight / body fat terms&quot;</code></pre>
<pre class="r"><code>length(bmi_weight_gwas)</code></pre>
<pre><code>[1] 23</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under BMI / weight / body fat terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under BMI / weight / body fat terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(bmi_weight_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% bmi_weight_gwas]
print(&quot;Percentage of GWAS traits not accounted for by disease, disorder or BMI / weight / body fat terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, disorder or BMI / weight / body fat terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 88.1</code></pre>
</div>
<div id="lipid-cholesterol-measurement-terms" class="section level3"
number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> Lipid / cholesterol
measurement terms</h3>
<pre class="r"><code>lipid_cholesterol_terms &lt;- grep(&quot;cholesterol|lipid|triglyceride|ldl|hdl|apolipoprotein&quot;, 
                                measurement, 
                                value = T)

measurement &lt;- measurement[!(measurement %in% lipid_cholesterol_terms)]

lipid_cholesterol_gwas = not_accounted_for[not_accounted_for %in% lipid_cholesterol_terms]

print(&quot;Number of GWAS traits under lipid / cholesterol terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under lipid / cholesterol terms&quot;</code></pre>
<pre class="r"><code>length(lipid_cholesterol_gwas)</code></pre>
<pre><code>[1] 253</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under lipid / cholesterol terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under lipid / cholesterol terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(lipid_cholesterol_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.3</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% lipid_cholesterol_gwas]
print(&quot;Percentage of GWAS traits not accounted for by disease, disorder or lipid / cholesterol terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, disorder or lipid / cholesterol terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 86.8</code></pre>
</div>
<div id="brain-measurement-terms" class="section level3" number="2.4.4">
<h3><span class="header-section-number">2.4.4</span> Brain measurement
terms</h3>
<pre class="r"><code>brain_measurement_terms &lt;- grep(&quot;brain|volume&quot;, 
                                measurement, 
                                value = T)

brain_measurement_terms &lt;- grep(&quot;bone|muscle|reticulocyte|erythrocyte|expiratory|platelet|urinary|thyroid|pancreas|kidney|spleen|liver|ventricular|blood&quot;, 
                                brain_measurement_terms, 
                                value = T, 
                                invert = T)</code></pre>
</div>
<div id="blood-pressure-measurement-terms" class="section level3"
number="2.4.5">
<h3><span class="header-section-number">2.4.5</span> Blood pressure
measurement terms</h3>
<pre class="r"><code>blood_pressure_terms &lt;- grep(&quot;blood pressure&quot;, 
                                measurement, 
                                value = T)

measurement &lt;- measurement[!(measurement %in% blood_pressure_terms)]

blood_pressure_gwas = not_accounted_for[not_accounted_for %in% blood_pressure_terms]

print(&quot;Number of GWAS traits under blood pressure terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under blood pressure terms&quot;</code></pre>
<pre class="r"><code>length(blood_pressure_gwas)</code></pre>
<pre><code>[1] 7</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under blood pressure terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under blood pressure terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(blood_pressure_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% blood_pressure_gwas]

print(&quot;Percentage of GWAS traits not accounted for by disease, disorder or blood pressure terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, disorder or blood pressure terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 86.8</code></pre>
</div>
<div id="find-seropositivity-terms-in-gwas-traits"
class="section level3" number="2.4.6">
<h3><span class="header-section-number">2.4.6</span> Find seropositivity
terms in GWAS traits</h3>
<pre class="r"><code>seropositivity_terms &lt;- grep(&quot;seropositivity|antibody&quot;, measurement, value = T)
seropositivity_terms &lt;- c(seropositivity_terms,
                          &quot;foot-and-mouth disease virus seropositivity&quot;,
                          &quot;bacillus phage virus seropositivity&quot;)

measurement &lt;- measurement[!(measurement %in% seropositivity_terms)]

seropositivity_gwas = not_accounted_for[not_accounted_for %in% seropositivity_terms]

print(&quot;Number of GWAS traits under seropositivity terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under seropositivity terms&quot;</code></pre>
<pre class="r"><code>length(seropositivity_gwas)</code></pre>
<pre><code>[1] 143</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under seropositivity terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under seropositivity terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(seropositivity_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.7</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% seropositivity_gwas]
print(&quot;Percentage of GWAS traits not accounted for by disease, disorder or seropositivity terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, disorder or seropositivity terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 86.1</code></pre>
</div>
<div id="general-measurement-terms" class="section level3"
number="2.4.7">
<h3><span class="header-section-number">2.4.7</span> General measurement
terms</h3>
<pre class="r"><code>behavior_measurement &lt;- c(
                    &quot;smoking&quot;,
                    &quot;alcohol consumption&quot;,
                    &quot;alcoholic beverage consumption&quot;,
                    &quot;alcohol exposure&quot;,
                    &quot;behavior&quot;,
                    &quot;farm exposure&quot;,
                    &quot;tobacco&quot;,
                    &quot;cannabis&quot;,
                    &quot;physical activity&quot;,
                    &quot;cognitive function&quot;,
                    &quot;pack-years&quot;,
                    &quot;coffee&quot;,
                    &quot;opioid&quot;,
                    &quot;environment&quot;,
                    &quot;exercise&quot;
                    )

behavior_measurement&lt; grep(paste0(behavior_measurement, 
                                  collapse = &quot;|&quot;), 
                            measurement,
                            value = T
                           )</code></pre>
<pre><code> [1]  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE FALSE FALSE  TRUE FALSE  TRUE
[13]  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE
[25] FALSE FALSE FALSE FALSE  TRUE  TRUE FALSE  TRUE  TRUE</code></pre>
<pre class="r"><code>measurement &lt;- measurement[!(measurement %in% behavior_measurement)]

measurement &lt;- grep(&quot;emphysema|eye colour|lifestyle&quot;, 
                    measurement, 
                    value = T, 
                    invert = T)


# Find terms where all comma-split pieces are in measurement
# measurement_gwas &lt;- not_accounted_for[
#   sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
#     parts &lt;- trimws(parts)
#     all(parts %in% measurement)
#   })
# ]
measurement_gwas &lt;- not_accounted_for[not_accounted_for %in% measurement]
#additional_measurement &lt;- not_accounted_for[not_accounted_for %in% measurement]

#measurement_gwas  = c(measurement_gwas, additional_measurement) |&gt; unique()

print(&quot;Number of GWAS traits under measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits under measurement terms&quot;</code></pre>
<pre class="r"><code>length(measurement_gwas)</code></pre>
<pre><code>[1] 16704</code></pre>
<pre class="r"><code>print(&quot;Percentage of GWAS traits under measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(measurement_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 83.6</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% measurement_gwas]

print(&quot;Percentage of GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 2.5</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 493</code></pre>
</div>
</div>
<div id="response-to-stimulus" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Response to
stimulus</h2>
<div id="combine-response-terms" class="section level3" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Combine response
terms</h3>
<pre class="r"><code>go_response = readLines(here::here(&quot;output/trait_ontology/go_0050896_descendants.txt&quot;))
efo_response &lt;- readLines(here::here(&quot;output/trait_ontology/efo_go_0050896_descendants.txt&quot;))

response &lt;- c(go_response,
              efo_response,
              &quot;response to stimulus&quot;)

response &lt;- unique(response)

response = stringr::str_trim(tolower(response))
response = unique(response)</code></pre>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
# response_gwas &lt;- not_accounted_for[
#   sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
#     parts &lt;- trimws(parts)
#     all(parts %in% response)
#   })
# ]

response_gwas &lt;- not_accounted_for[not_accounted_for %in% response]

#additional_response &lt;- not_accounted_for[not_accounted_for %in% response]
#response_gwas  = c(response_gwas, additional_response) |&gt; unique()

print(&quot;Percentage of GWAS traits under response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(response_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.2</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% response_gwas]

print(&quot;Percentage of GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 253</code></pre>
</div>
</div>
<div id="mental-process" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Mental process</h2>
<pre class="r"><code>mental &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0004323_descendants.txt&quot;))
mental = stringr::str_trim(tolower(mental))
mental &lt;- unique(mental)
mental &lt;- c(mental,
          &quot;memory performance&quot;,
          &quot;visual memory process attribute&quot;,
          &quot;verbal memory measurement&quot;,
          &quot;executive function measurement&quot;,
          &quot;cognitive function measurement&quot;
          )

mental_gwas = not_accounted_for[not_accounted_for %in% mental]

print(&quot;Percentage of GWAS traits under mental process terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under mental process terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(mental_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% mental_gwas]

print(&quot;Percentage of GWAS traits not accounted for thus far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for thus far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.2</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for thus far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for thus far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 235</code></pre>
</div>
<div id="behavior" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Behavior</h2>
<pre class="r"><code>behavior &lt;- readLines(here::here(&quot;output/trait_ontology/go_0007610_descendants.txt&quot;))
behavior = stringr::str_trim(tolower(behavior))
behavior &lt;- unique(behavior)
behavior &lt;- c(behavior,
              behavior_measurement,
              &quot;physical activity&quot;)

behavior_gwas = not_accounted_for[not_accounted_for %in% behavior]

print(&quot;Percentage of GWAS traits under behavouir terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under behavouir terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(behavior_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% behavior_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1.1</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 213</code></pre>
</div>
<div id="injury" class="section level2" number="2.8">
<h2><span class="header-section-number">2.8</span> Injury</h2>
<pre class="r"><code>injury &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000546_descendants.txt&quot;))
injury = stringr::str_trim(tolower(injury))
injury &lt;- c(injury,
            &quot;fall&quot;)

injury_gwas = not_accounted_for[not_accounted_for %in% injury]

print(&quot;Percentage of GWAS traits under injury terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under injury terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(injury_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% injury_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 192</code></pre>
</div>
<div id="phenotype" class="section level2" number="2.9">
<h2><span class="header-section-number">2.9</span> Phenotype</h2>
<pre class="r"><code>phenotype &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0000651_descendants.txt&quot;))
phenotype = stringr::str_trim(tolower(phenotype))
phenotype &lt;- unique(c(phenotype,
                      &quot;aging&quot;,
                      &quot;biological sex&quot;,
                      &quot;comparative body size at age 10, self-reported&quot;,
                      &quot;complex trait&quot;,                      
                      &quot;eye colour measurement&quot;,
                      &quot;strand of hair color&quot;,
                      &quot;high altitude adaptation&quot;,
                      &quot;multiple gestation&quot;,
                      &quot;normal&quot;,
                      &quot;personality trait&quot;,
                      &quot;skin pigmentation&quot;,
                      &quot;personality&quot;,
                      &quot;growth delay&quot;,
                      &quot;sensory perception of taste&quot;,
                      &quot;sensory perception of bitter taste&quot;,
                      &quot;sensory perception of sweet taste&quot;,
                      &quot;sensory perception of smell&quot;,
                      &quot;sensory perception of sound&quot;,
                      &quot;size&quot;,
                      &quot;skin aging&quot;,
                      &quot;sexual dimorphism&quot;,
                      &quot;voice quality trait&quot;)
)

phenotype_gwas = not_accounted_for[not_accounted_for %in% phenotype]

print(&quot;Percentage of GWAS traits under phenotype terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under phenotype terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(phenotype_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.4</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% phenotype_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.6</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 122</code></pre>
</div>
<div id="medical-procedure" class="section level2" number="2.10">
<h2><span class="header-section-number">2.10</span> Medical
procedure</h2>
<pre class="r"><code>medical_procedure &lt;- readLines(here::here(&quot;output/trait_ontology/efo_0002571_descendants.txt&quot;))
surgical_procedure &lt;- readLines(here::here(&quot;output/trait_ontology/maxo_0000004_descendants.txt&quot;))
clinical_history &lt;- c(&quot;clinical history&quot;,
                      &quot;encounter with health service&quot;,
                      &quot;encounter with health service for adjustment and management of implanted device&quot;,
                      &quot;encounter with health service related to reproduction&quot;)



medical_procedure = stringr::str_trim(tolower(medical_procedure))
medical_procedure = unique(c(medical_procedure, 
                             surgical_procedure,
                             clinical_history,
                             &quot;braces&quot;,
                             &quot;vaccination&quot;,
                             &quot;hormone replacement therapy&quot;,
                             &quot;cognitive behavioural therapy&quot;,
                             &quot;organ extraction&quot;,
                             &quot;gastric bypass&quot;,
                             &quot;medical procedure&quot;,
                             &quot;number of treatments or medications taken, self-reported&quot;,
                             &quot;treatment&quot;,
                             &quot;test result&quot;,
                             &quot;hospitalisation&quot;,
                             &quot;clinical treatment&quot;)
                           )



medical_procedure_gwas = not_accounted_for[not_accounted_for %in% medical_procedure]

print(&quot;Percentage of GWAS traits under medical procedure terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under medical procedure terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(medical_procedure_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.3</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% medical_procedure_gwas]
print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 69</code></pre>
</div>
<div id="environmental-factors" class="section level2" number="2.11">
<h2><span class="header-section-number">2.11</span> Environmental
factors</h2>
<pre class="r"><code>enviro_factors &lt;- c(
&quot;diet measurement&quot;,
&quot;economic and social preference&quot;,
&quot;educational attainment&quot;,
&quot;encounter with health service related to socioeconomic and psychosocial circumstances&quot; ,
&quot;energy intake&quot;,
&quot;environmental factor&quot;,
&quot;family relationship&quot;,
&quot;household income&quot;,
&quot;income&quot;,
&quot;lifestyle measurement&quot;,
&quot;risk factor&quot;,
&quot;self reported educational attainment&quot;,
&quot;social deprivation&quot;,
&quot;social risk factor&quot;,
&quot;socioeconomic status&quot;,
&quot;townsend deprivation index&quot;
)


enviro_factors_gwas = not_accounted_for[not_accounted_for %in% enviro_factors]

print(&quot;Percentage of GWAS traits under environmental factor terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under environmental factor terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(enviro_factors_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.1</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% enviro_factors_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 56</code></pre>
</div>
<div id="other" class="section level2" number="2.12">
<h2><span class="header-section-number">2.12</span> Other</h2>
<div id="biological-process" class="section level3" number="2.12.1">
<h3><span class="header-section-number">2.12.1</span> Biological
process</h3>
<pre class="r"><code>bio_process &lt;- c(&quot;pregnancy&quot;,
                 &quot;puberty&quot;,
                 &quot;menopause&quot;,
                 &quot;ovulation&quot;,
                 &quot;positive regulation of ovulation&quot;)

bio_process = stringr::str_trim(tolower(bio_process))

bio_process_gwas = not_accounted_for[not_accounted_for %in% bio_process]

print(&quot;Percentage of GWAS traits under biological process terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under biological process terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(bio_process_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% bio_process_gwas]
print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.3</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 51</code></pre>
</div>
<div id="traditional-medicine-constitutional-types"
class="section level3" number="2.12.2">
<h3><span class="header-section-number">2.12.2</span> Traditional
medicine constitutional types</h3>
<pre class="r"><code>tm_constitution &lt;- c(&quot;yu-zhi constitution type&quot;,
                      &quot;sasang constitutional medicine&quot;,
                      &quot;sasang constitutional medicine type&quot;,
                      &quot;hepatonia constitution type&quot;,
                      &quot;pulmotonia constitution type&quot;,
                      &quot;tae-yang&quot;,
                      &quot;tae-eum&quot;,
                      &quot;so-eum&quot;,
                      &quot;so-yang&quot;
                      )

tm_constitution = stringr::str_trim(tolower(tm_constitution))

tm_constitution_gwas = not_accounted_for[not_accounted_for %in% tm_constitution]

print(&quot;Percentage of GWAS traits under traditional medicine constitutional type terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits under traditional medicine constitutional type terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(tm_constitution_gwas) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% tm_constitution_gwas]

print(&quot;Percentage of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)</code></pre>
<pre><code>[1] 0.2</code></pre>
<pre class="r"><code>print(&quot;Number of GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 42</code></pre>
</div>
<div id="ancestry-info" class="section level3" number="2.12.3">
<h3><span class="header-section-number">2.12.3</span> Ancestry-info</h3>
<pre class="r"><code>ancestry_gwas &lt;- &quot;latin or admixed american ancestry&quot;

not_accounted_for = not_accounted_for[!not_accounted_for %in% ancestry_gwas]</code></pre>
</div>
<div id="cell-lines" class="section level3" number="2.12.4">
<h3><span class="header-section-number">2.12.4</span> Cell lines</h3>
<pre class="r"><code>cell_line_gwas &lt;- &quot;gm11992&quot;

not_accounted_for = not_accounted_for[!not_accounted_for %in% cell_line_gwas]</code></pre>
</div>
</div>
</div>
<div id="add-categories-to-gwas-info" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Add Categories to GWAS
Info</h1>
<div
id="add-disease-phenotypic-abnormality-terms-to-gwas-study-info-dataset"
class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Add disease &amp;
phenotypic abnormality terms to GWAS study info dataset</h2>
<pre class="r"><code># find_disease_terms  &lt;-   function(MAPPED_TRAIT) {
#         # find all disease terms that appear in the trait
#         # split_mapped_traits &lt;- stringr::str_split(MAPPED_TRAIT, &quot;, &quot;) |&gt; 
#         #                        unlist()
#         
#         mapped_disease_terms &lt;- split_mapped_traits[split_mapped_traits %in% disease_terms]
#         mapped_pheno_abnorm_terms &lt;- split_mapped_traits[split_mapped_traits %in% pheno_abnorm]
#         
#         mapped_disease_terms = unique(c(mapped_disease_terms, 
#                                         mapped_pheno_abnorm_terms
#                                         )
#         )
#                                       
#         return(str_flatten(mapped_disease_terms, 
#                            collapse = &quot;, &quot;,
#                            na.rm = T))  # combine multiple matches
#         
# }
# 
disease_or_disorder &lt;- c(disease_gwas,
                          pheno_abnorm_gwas,
                          seropositivity_gwas
                          )

gwas_study_info &lt;- 
  gwas_study_info |&gt; 
  dplyr::rowwise() |&gt;
  dplyr::mutate(
    disease_terms = 
      ifelse(MAPPED_TRAIT %in% disease_or_disorder,
             MAPPED_TRAIT,
             NA)
  )</code></pre>
</div>
<div id="map-gwas-traits-to-high-level-categories"
class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Map GWAS traits to
high-level categories</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::mutate(MAPPED_TRAIT_CATEGORY = dplyr::case_when(is.na(MAPPED_TRAIT) ~ NA,
                                                         MAPPED_TRAIT == &quot;&quot; ~ NA,
                                                         tolower(MAPPED_TRAIT) %in% disease_gwas ~ &quot;Disease/Disorder&quot;,
                                                         tolower(MAPPED_TRAIT) %in% pheno_abnorm_gwas ~ &quot;Phenotypic Abnormality&quot;,
                                                         tolower(MAPPED_TRAIT) %in% seropositivity_gwas ~ &quot;Seropositivity&quot;,
                                                         tolower(MAPPED_TRAIT) %in% bmi_weight_gwas ~ &quot;BMI/Weight/Body Fat Measurement&quot;,
                                                         tolower(MAPPED_TRAIT) %in% lipid_cholesterol_gwas ~ &quot;Lipid/Cholesterol Measurement&quot;,
                                                         tolower(MAPPED_TRAIT) %in% brain_measurement_terms ~ &quot;Brain Measurement&quot;,
                                                         tolower(MAPPED_TRAIT) %in% blood_pressure_gwas ~ &quot;Blood Pressure Measurement&quot;,
                                                         tolower(MAPPED_TRAIT) %in% measurement_gwas ~ &quot;Measurement&quot;,
                                                         tolower(MAPPED_TRAIT) %in% response_gwas ~ &quot;Response&quot;,
                                                         tolower(MAPPED_TRAIT) %in% mental_gwas ~ &quot;Mental Process&quot;,
                                                         tolower(MAPPED_TRAIT) %in% behavior_gwas ~ &quot;Behavior&quot;,
                                                         tolower(MAPPED_TRAIT) %in% injury_gwas ~ &quot;Injury&quot;,
                                                         tolower(MAPPED_TRAIT) %in% phenotype_gwas ~ &quot;Phenotype&quot;,
                                                         tolower(MAPPED_TRAIT) %in% medical_procedure_gwas ~ &quot;Medical Procedure&quot;,
                                                         tolower(MAPPED_TRAIT) %in% enviro_factors_gwas ~ &quot;Environmental Factor&quot;,
                                                         
                                                          TRUE ~ &quot;Other&quot;
                                                          )
                )</code></pre>
</div>
</div>
<div id="background-traits" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Background traits</h1>
<pre class="r"><code>gwas_background &lt;- gwas_study_info$MAPPED_BACKGROUND_TRAIT

gwas_background = stringr::str_trim(tolower(gwas_background))
gwas_background &lt;- unique(gwas_background)
gwas_background &lt;- gwas_background[gwas_background != &quot;&quot;]


print(&quot;Number of unique background GWAS traits&quot;)</code></pre>
<pre><code>[1] &quot;Number of unique background GWAS traits&quot;</code></pre>
<pre class="r"><code>length(gwas_background)</code></pre>
<pre><code>[1] 290</code></pre>
<div id="overlap-with-diseasedisorder-traits" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> Overlap with
disease/disorder traits</h2>
<pre class="r"><code># multiple_terms = grep(&quot;,&quot;, gwas_background, value = T)
# mask &lt;- Reduce(`|`, lapply(disease_terms, function(x) grepl(x, multiple_terms)))
# additional_disease_gwas &lt;- multiple_terms[mask]

# disease_gwas = c(gwas_background[gwas_background %in% disease_terms],
#                  additional_disease_gwas)

disease_gwas = gwas_background[gwas_background %in% disease_terms]

print(&quot;Number of background GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>length(disease_gwas)</code></pre>
<pre><code>[1] 199</code></pre>
<pre class="r"><code>print(&quot;Percentage of background GWAS traits under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(disease_gwas) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 68.6</code></pre>
<pre class="r"><code>not_accounted_for = gwas_background[!gwas_background %in% disease_gwas]

print(&quot;Percentage of background GWAS traits not under disease or disorder terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not under disease or disorder terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 31.4</code></pre>
</div>
<div id="phenotype-abnormality-overlap-1" class="section level2"
number="4.2">
<h2><span class="header-section-number">4.2</span> Phenotype abnormality
overlap</h2>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
# pheno_abnorm_gwas &lt;- not_accounted_for[
#   sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
#     parts &lt;- trimws(parts) # remove extra spaces
#     all(parts %in% pheno_abnorm)
#   })
# ]

# additional_pheno_abnorm &lt;- not_accounted_for[not_accounted_for %in% pheno_abnorm]

# pheno_abnorm_gwas  = c(pheno_abnorm_gwas, additional_pheno_abnorm) |&gt; unique()

pheno_abnorm_gwas  = pheno_abnorm_gwas


print(&quot;Percentage of background GWAS traits under phenotype abnormality terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits under phenotype abnormality terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(pheno_abnorm_gwas) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 129</code></pre>
<pre class="r"><code>not_accounted_for = not_accounted_for[!not_accounted_for %in% pheno_abnorm_gwas]

print(&quot;Percentage of background GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 29.3</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits not accounted for so far&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits not accounted for so far&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 85</code></pre>
</div>
<div
id="add-disease-phenotypic-abnormality-terms-to-gwas-study-info-dataset-1"
class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Add disease &amp;
phenotypic abnormality terms to GWAS study info dataset</h2>
<pre class="r"><code>disease_or_disorder &lt;- c(disease_gwas,
                          pheno_abnorm_gwas
  )

gwas_study_info &lt;- 
  gwas_study_info |&gt;
  rowwise() |&gt;
  dplyr::mutate(
    background_disease_terms = 
      ifelse(MAPPED_BACKGROUND_TRAIT %in% disease_or_disorder,
             MAPPED_BACKGROUND_TRAIT,
             NA)
  ) |&gt;
  ungroup()

# gwas_study_info &lt;- 
#   gwas_study_info |&gt;
#   rowwise() |&gt;
#   dplyr::mutate(
#     background_disease_terms = 
#       ifelse(MAPPED_BACKGROUND_TRAIT == &quot;&quot;,
#              NA,
#              background_disease_terms)
#   ) |&gt;
#   ungroup()</code></pre>
</div>
<div id="measurement-traits" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Measurement
traits</h2>
<pre class="r"><code># measurement_gwas &lt;- not_accounted_for[
#   sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
#     parts &lt;- trimws(parts) # remove extra spaces
#     all(parts %in% measurement)
#   })
# ]

# additional_measurement &lt;- not_accounted_for[not_accounted_for %in% measurement]

# measurement_gwas  = c(measurement_gwas, additional_measurement) |&gt; unique()

measurement_gwas  = measurement[measurement %in% not_accounted_for]

not_accounted_for = not_accounted_for[!not_accounted_for %in% measurement_gwas]

print(&quot;Percentage of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 13.1</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits not accounted for by disease, disorder or measurement terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 38</code></pre>
</div>
<div id="response-traits" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Response traits</h2>
<pre class="r"><code># Find terms where all comma-split pieces are in measurement
# response_gwas &lt;- not_accounted_for[
#   sapply(strsplit(not_accounted_for, &quot;, &quot;), function(parts) {
#     parts &lt;- trimws(parts)
#     all(parts %in% response)
#   })
# ]

# additional_response &lt;- not_accounted_for[not_accounted_for %in% response]
# 
# response_gwas  = c(response_gwas, additional_response) |&gt; unique()

response_gwas  = response[response %in% not_accounted_for]

not_accounted_for = not_accounted_for[!not_accounted_for %in% response_gwas]

print(&quot;Percentage of background GWAS traits under response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits under response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(response_gwas) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 2.4</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits under response terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits under response terms&quot;</code></pre>
<pre class="r"><code>length(response_gwas)</code></pre>
<pre><code>[1] 7</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 31</code></pre>
<pre class="r"><code>print(&quot;Percentage of background GWAS traits not accounted for by disease, measurement or response terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not accounted for by disease, measurement or response terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 10.7</code></pre>
</div>
<div id="medical-procedure-traits" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Medical procedure
traits</h2>
<pre class="r"><code>medical_procedure_gwas = medical_procedure[medical_procedure %in% not_accounted_for]

not_accounted_for = not_accounted_for[!not_accounted_for %in% medical_procedure_gwas]

print(&quot;Percentage of background GWAS traits not accounted for by disease, measurement, response or medical procedure terms&quot;)</code></pre>
<pre><code>[1] &quot;Percentage of background GWAS traits not accounted for by disease, measurement, response or medical procedure terms&quot;</code></pre>
<pre class="r"><code>round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)</code></pre>
<pre><code>[1] 6.9</code></pre>
<pre class="r"><code>print(&quot;Number of background GWAS traits not accounted for by disease, measurement, response or medical procedure terms&quot;)</code></pre>
<pre><code>[1] &quot;Number of background GWAS traits not accounted for by disease, measurement, response or medical procedure terms&quot;</code></pre>
<pre class="r"><code>length(not_accounted_for)</code></pre>
<pre><code>[1] 20</code></pre>
</div>
<div id="background-trait-categories" class="section level2"
number="4.7">
<h2><span class="header-section-number">4.7</span> Background trait
categories</h2>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::mutate(BACKGROUND_TRAIT_CATEGORY = 
                   dplyr::case_when(
                                      MAPPED_BACKGROUND_TRAIT == &quot;&quot; ~ NA,
                                      is.na(MAPPED_BACKGROUND_TRAIT) ~ NA,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% disease_gwas ~ &quot;Disease/Disorder&quot;,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% pheno_abnorm_gwas ~ &quot;Phenotypic Abnormality&quot;,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% measurement_gwas ~ &quot;Measurement&quot;,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% response_gwas ~ &quot;Response&quot;,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% medical_procedure_gwas ~ &quot;Medical Procedure&quot;,
                                      TRUE ~ &quot;Other&quot;)
                )</code></pre>
</div>
</div>
<div
id="summary-of-number-of-disease-studies-and-studies-of-each-kind-of-trait"
class="section level1" number="5">
<h1><span class="header-section-number">5</span> Summary of number of
disease studies (and studies of each kind of trait)</h1>
<pre class="r"><code>gwas_study_info |&gt;
  group_by(MAPPED_TRAIT_CATEGORY, BACKGROUND_TRAIT_CATEGORY) |&gt;
  summarise(n_studies = n()) |&gt; 
  arrange(desc(n_studies))</code></pre>
<pre><code># A tibble: 60 × 3
# Groups:   MAPPED_TRAIT_CATEGORY [16]
   MAPPED_TRAIT_CATEGORY         BACKGROUND_TRAIT_CATEGORY n_studies
   &lt;chr&gt;                         &lt;chr&gt;                         &lt;int&gt;
 1 Measurement                   &lt;NA&gt;                          91966
 2 Disease/Disorder              &lt;NA&gt;                          22284
 3 Measurement                   Disease/Disorder              18816
 4 Brain Measurement             &lt;NA&gt;                           7143
 5 Lipid/Cholesterol Measurement &lt;NA&gt;                           4150
 6 Seropositivity                &lt;NA&gt;                           2972
 7 Phenotypic Abnormality        &lt;NA&gt;                           2702
 8 Medical Procedure             &lt;NA&gt;                           1266
 9 Response                      &lt;NA&gt;                            987
10 Response                      Disease/Disorder                780
# ℹ 50 more rows</code></pre>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::rowwise() |&gt;
  dplyr::mutate(DISEASE_STUDY = 
                   case_when(MAPPED_TRAIT_CATEGORY == &quot;Disease/Disorder&quot; | 
                             MAPPED_TRAIT_CATEGORY == &quot;Phenotypic Abnormality&quot; |  
                             MAPPED_TRAIT_CATEGORY == &quot;Seropositivity&quot; |
                             BACKGROUND_TRAIT_CATEGORY == &quot;Disease/Disorder&quot; | 
                             BACKGROUND_TRAIT_CATEGORY == &quot;Phenotypic Abnormality&quot; ~ T,
                             T ~ F )
                ) |&gt;
  dplyr::ungroup() 

gwas_study_info |&gt; 
  group_by(DISEASE_STUDY, 
           MAPPED_TRAIT_CATEGORY, 
           BACKGROUND_TRAIT_CATEGORY) |&gt; 
  summarise(n = n()) </code></pre>
<pre><code># A tibble: 60 × 4
# Groups:   DISEASE_STUDY, MAPPED_TRAIT_CATEGORY [28]
   DISEASE_STUDY MAPPED_TRAIT_CATEGORY           BACKGROUND_TRAIT_CATEGORY     n
   &lt;lgl&gt;         &lt;chr&gt;                           &lt;chr&gt;                     &lt;int&gt;
 1 FALSE         BMI/Weight/Body Fat Measurement Measurement                   7
 2 FALSE         BMI/Weight/Body Fat Measurement Other                         6
 3 FALSE         BMI/Weight/Body Fat Measurement &lt;NA&gt;                        726
 4 FALSE         Behavior                        Measurement                   2
 5 FALSE         Behavior                        &lt;NA&gt;                        249
 6 FALSE         Blood Pressure Measurement      &lt;NA&gt;                        531
 7 FALSE         Brain Measurement               Measurement                   1
 8 FALSE         Brain Measurement               Response                      6
 9 FALSE         Brain Measurement               &lt;NA&gt;                       7143
10 FALSE         Environmental Factor            &lt;NA&gt;                        179
# ℹ 50 more rows</code></pre>
<pre class="r"><code># Number of papers with at least one disease study
gwas_study_info |&gt; 
  group_by(PUBMED_ID) |&gt; 
  summarise(DISEASE_STUDY = any(DISEASE_STUDY == T)) |&gt; 
  group_by(DISEASE_STUDY) |&gt; 
  summarise(n = n())</code></pre>
<pre><code># A tibble: 2 × 2
  DISEASE_STUDY     n
  &lt;lgl&gt;         &lt;int&gt;
1 FALSE          2711
2 TRUE           4615</code></pre>
<pre class="r"><code># ~ 60% of papers have at least one disease study

gwas_study_info |&gt; 
  group_by(PUBMED_ID, YEAR) |&gt; 
  summarise(DISEASE_STUDY = any(DISEASE_STUDY == T)) |&gt; 
  group_by(YEAR) |&gt;
  summarise(n_disease_studies = sum(DISEASE_STUDY == T),
            n_total_studies = n()) |&gt;
  mutate(percentage_disease_studies = 100 * n_disease_studies / n_total_studies) |&gt;
  ggplot(aes(x= YEAR, 
             y= percentage_disease_studies)) +
  geom_line() +
  geom_point() +
  labs(title = &quot;Percentage of papers with at least one disease GWAS over time&quot;,
       x = &quot;Year&quot;,
       y = &quot;Percentage of papers with at least one disease GWAS&quot;) +
  theme_bw() + 
  lims(y = c(0,100))</code></pre>
<p><img src="figure/trait_ontology_categorization.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="number-of-disease-studies" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Number of disease
studies</h2>
<pre class="r"><code>gwas_study_info |&gt;
  filter(DISEASE_STUDY == T) |&gt;
  nrow()</code></pre>
<pre><code>[1] 49158</code></pre>
</div>
</div>
<div
id="creating-disease-labels-column-of-just-disease-or-phenotype-abnormality-terms-for-each-study---so-that-we-can-see-what-diseases-are-being-studied"
class="section level1" number="6">
<h1><span class="header-section-number">6</span> Creating disease labels
column of just disease or phenotype abnormality terms for each study -
so that we can see what diseases are being studied</h1>
<div
id="make-disease-label-column---combining-disease-terms-from-both-mapped-trait-and-background-trait"
class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Make disease label
column - combining disease terms from both mapped trait and background
trait</h2>
<pre class="r"><code>combined_disease_terms = function(MAPPED_TRAIT_1, MAPPED_TRAIT_2){
  
  
  MAPPED_TRAIT_1 = stringr::str_split(MAPPED_TRAIT_1, &quot;, &quot;) |&gt; unlist()
  MAPPED_TRAIT_2  = stringr::str_split(MAPPED_TRAIT_2, &quot;, &quot;) |&gt; unlist()
  
  all_mapped_disease_terms = 
    c(MAPPED_TRAIT_1, MAPPED_TRAIT_2) |&gt;
    unique()
  
  combined_mapped_disease_terms = str_flatten(all_mapped_disease_terms, 
                                         collapse = &quot;, &quot;,
                                         na.rm = T
                                         )
  
  return(combined_mapped_disease_terms)
  
}


gwas_study_info &lt;- 
  gwas_study_info |&gt;
  dplyr::rowwise() |&gt;
  dplyr::mutate(all_disease_terms = 
                case_when(is.na(background_disease_terms) &amp; is.na(disease_terms) ~ NA,
                          is.na(background_disease_terms) &amp; !is.na(disease_terms) ~ disease_terms,
                          !is.na(background_disease_terms) &amp; is.na(disease_terms) ~ background_disease_terms,
                          !is.na(background_disease_terms) &amp; !is.na(disease_terms) ~
                            combined_disease_terms(background_disease_terms,
                                                   disease_terms)) 

  ) |&gt;
  dplyr::ungroup()</code></pre>
</div>
<div id="minor-fixes-of-trait-categorisation-and-returning-traits"
class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Minor fixes of trait
categorisation and returning traits</h2>
<pre class="r"><code># What studies are disease studies but have no collected disease terms?
gwas_study_info |&gt; 
  filter(DISEASE_STUDY == T) |&gt; 
  filter(all_disease_terms == &quot;&quot;)  |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_CATEGORY) |&gt; 
  distinct() |&gt;
  nrow()</code></pre>
<pre><code>[1] 0</code></pre>
<div
id="fix-bug-where-mapped_traitbackground_mapped_trait-is-empty-string-but-trait_category-is-listed-as-diseasephenotypic-abnormality"
class="section level3" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Fix bug where
MAPPED_TRAIT/BACKGROUND_MAPPED_TRAIT is empty string but TRAIT_CATEGORY
is listed as disease/phenotypic abnormality</h3>
<pre class="r"><code>gwas_study_info = gwas_study_info |&gt;
  rowwise() |&gt;
  mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT == &quot;&quot;,
                                        &quot;Other&quot;,
                                        MAPPED_TRAIT_CATEGORY)) |&gt;
  mutate(BACKGROUND_TRAIT_CATEGORY = ifelse(MAPPED_BACKGROUND_TRAIT == &quot;&quot;,
                                        &quot;Other&quot;,
                                        BACKGROUND_TRAIT_CATEGORY)) </code></pre>
</div>
<div id="fix-some-specific-mis-categorizations" class="section level3"
number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Fix some specific
mis-categorizations</h3>
<pre class="r"><code># Fixing fractures, ununited as injury 
gwas_study_info = gwas_study_info |&gt;
  mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT == &quot;fractures, ununited&quot;,
                                        &quot;Injury&quot;,
                                        MAPPED_TRAIT_CATEGORY)
         )   

# Fixing some response terms mislabelled as disease/disorder
other_response_terms = c(&quot;response to COVID-19 vaccine, localized superficial swelling, mass, or lump&quot;,
                         &quot;response to COVID-19 vaccine, SARS-CoV-2 neutralizing antibody measurement&quot;,
                         &quot;Anti-hepatitis B virus surface antigen IgG measurement, response to vaccine&quot;,
                         &quot;anti-SARS-CoV-2 IgG measurement, response to COVID-19 vaccine&quot;,
                         &quot;anti-tetanus toxoid IgG measurement, response to vaccine&quot;,
                         &quot;height growth attribute, response to growth hormone&quot;,
                         &quot;adrenal suppression measurement, response to corticosteroid&quot;,
                         &quot;asthenia, response to COVID-19 vaccine&quot;,
                         &quot;SARS-CoV-2 antibody measurement, response to COVID-19 vaccine&quot;,
                         &quot;response to vaccine, anti-Haemophilus influenzae type b polyribosylribitol phosphate IgG measurement&quot;,
                         &quot;height growth attribute, response to growth hormone&quot;
                )
# Fixing some measurement terms mislabelled as disease/disorder
other_measurement_terms = c(&quot;amygdala volume, pallidum volume, nucleus accumbens volume, hippocampal volume, putamen volume, intracranial volume measurement, thalamus volume, caudate nucleus volume&quot;,
                            &quot;lamina-associated polypeptide 2, isoforms beta/gamma measurement&quot;,
                            &quot;apoptosis-inducing factor 1, mitochondrial measurement&quot;,
                            &quot;central corneal thickness, intraocular pressure measurement&quot;,
                            &quot;level of apoptosis-inducing factor 1, mitochondrial in blood serum&quot;,
                            &quot;level of MHC class I polypeptide-related sequence A in blood, level of MHC class I polypeptide-related sequence B in blood&quot;,
                            &quot;psychosocial stress measurement, intracranial volume measurement&quot;)

# Fixing some other terms mislabelled as disease/disorder
other_terms = c(&quot;insulin metabolic clearance rate measurement, disposition index measurement, insulin sensitivity measurement, glucose homeostasis trait, glucose effectiveness measurement, acute insulin response measurement&quot;,
                &quot;disease recurrence, response to allogeneic hematopoietic stem cell transplant&quot;,
                &quot;disease recurrence, response to allogeneic hematopoietic stem cell transplant, donor genotype effect measurement&quot;
)

gwas_study_info =
  gwas_study_info |&gt;
    mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT %in% other_response_terms,
                                        &quot;Response&quot;,
                                        MAPPED_TRAIT_CATEGORY)) |&gt;
    mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT %in% other_measurement_terms,
                                        &quot;Response&quot;,
                                        MAPPED_TRAIT_CATEGORY)) |&gt;
    mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT %in% other_terms,
                                        &quot;Other&quot;,
                                        MAPPED_TRAIT_CATEGORY))  </code></pre>
</div>
<div id="fixing-some-specific-missing-disease-terms"
class="section level3" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Fixing some
specific missing disease terms</h3>
<pre class="r"><code># spine osteoarthritis was not being picked up as a disease term
gwas_study_info = 
  gwas_study_info |&gt;
    mutate(disease_terms = ifelse(grepl(&quot;spine osteoarthritis&quot;, MAPPED_TRAIT),
                                        str_flatten(&quot;spine opsteoarthritis&quot;, 
                                               disease_terms, 
                                               collapse = &quot;,&quot;,
                                               na.rm = T),
                                        disease_terms)
    ) |&gt;
      mutate(all_disease_terms = ifelse(grepl(&quot;spine osteoarthritis&quot;, MAPPED_TRAIT),
                                        str_flatten(&quot;spine psteoarthritis&quot;, 
                                                    all_disease_terms, 
                                                    collapse = &quot;,&quot;,
                                                    na.rm = T),
                                        all_disease_terms)
    )

# liver disease biomarker was not being picked up as a disease term 
gwas_study_info = 
  gwas_study_info |&gt;
    mutate(disease_terms = ifelse(grepl(&quot;liver disease biomarker&quot;, MAPPED_TRAIT),
                                        str_flatten(&quot;liver disease&quot;, 
                                                    disease_terms, 
                                                    collapse = &quot;,&quot;,
                                                    na.rm = T),
                                        disease_terms)
    ) |&gt;
      mutate(all_disease_terms = ifelse(grepl(&quot;liver disease biomarker&quot;, MAPPED_TRAIT),
                                        str_flatten(&quot;liver disease&quot;, 
                                                    all_disease_terms, 
                                                    collapse = &quot;,&quot;,
                                                    na.rm = T),
                                        all_disease_terms)
    )


gwas_study_info = 
  gwas_study_info |&gt;
    mutate(disease_terms = ifelse(grepl(&quot;liver disease biomarker&quot;, MAPPED_TRAIT),
                                        str_flatten(&quot;liver disease&quot;, 
                                               disease_terms, 
                                               collapse = &quot;,&quot;,
                                               na.rm = T),
                                        disease_terms)
    ) |&gt;
      mutate(all_disease_terms = ifelse(grepl(&quot;liver disease biomarker&quot;, MAPPED_TRAIT),
                                        str_flatten(&quot;liver disease&quot;, 
                                                    all_disease_terms, 
                                                    collapse = &quot;,&quot;,
                                                    na.rm = T),
                                        all_disease_terms)
    )

gwas_study_info = gwas_study_info |&gt;
  mutate(all_disease_terms = str_replace_all(all_disease_terms, 
                                             &quot;liver diseaseliver diseaseNA&quot;, 
                                             &quot;liver disease&quot;)
  ) |&gt;  mutate(all_disease_terms = str_replace_all(all_disease_terms, 
                                             &quot;liver diseaseliver disease&quot;, 
                                             &quot;liver disease&quot;)
  ) 


# growth delay was not being picked up as a disease term
gwas_study_info = 
  gwas_study_info |&gt;
    mutate(background_disease_terms= ifelse(grepl(&quot;Growth delay&quot;, MAPPED_BACKGROUND_TRAIT),
                                        str_flatten(&quot;growth delay&quot;, 
                                                    background_disease_terms, 
                                                    collapse = &quot;,&quot;,
                                                    na.rm = T),
                                        background_disease_terms)
    ) |&gt; 
      mutate(all_disease_terms = ifelse(grepl(&quot;Growth delay&quot;, MAPPED_BACKGROUND_TRAIT),
                                        str_flatten(&quot;growth delay&quot;, 
                                                    all_disease_terms, 
                                                    collapse = &quot;,&quot;,
                                                    na.rm = T),
                                        all_disease_terms)
    )


gwas_study_info = gwas_study_info |&gt;
  mutate(all_disease_terms = str_replace_all(all_disease_terms, 
                                             &quot;growth delayNA&quot;, 
                                             &quot;growth delay&quot;)
  ) 


gwas_study_info = gwas_study_info |&gt;
  mutate(all_disease_terms = str_replace_all(all_disease_terms, 
                                             &quot;spine psteoarthritisNA&quot;, 
                                             &quot;spine psteoarthritis&quot;)
  ) </code></pre>
</div>
<div id="recalculate-disease-study-flag-is-disease-study-or-not"
class="section level3" number="6.2.4">
<h3><span class="header-section-number">6.2.4</span> Recalculate disease
study flag (is disease study or not?)</h3>
<p>now that I have corrected any mistakes in categorization and added
some missing disease terms</p>
<pre class="r"><code>gwas_study_info = 
gwas_study_info |&gt;
  dplyr::rowwise() |&gt;
  dplyr::mutate(DISEASE_STUDY = 
                   ifelse(MAPPED_TRAIT_CATEGORY == &quot;Disease/Disorder&quot; | 
                          MAPPED_TRAIT_CATEGORY == &quot;Phenotypic Abnormality&quot; |  
                          BACKGROUND_TRAIT_CATEGORY == &quot;Disease/Disorder&quot; | 
                          BACKGROUND_TRAIT_CATEGORY == &quot;Phenotypic Abnormality&quot;,
                          T, F )
                ) |&gt;
  dplyr::ungroup() 


gwas_study_info |&gt; 
  filter(DISEASE_STUDY == T) |&gt; 
  filter(all_disease_terms == &quot;&quot;)  |&gt; 
  select(MAPPED_TRAIT, MAPPED_TRAIT_CATEGORY) |&gt; 
  distinct() |&gt;
  nrow()</code></pre>
<pre><code>[1] 0</code></pre>
</div>
</div>
</div>
<div id="saving" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Saving:</h1>
<pre class="r"><code>data.table::fwrite(gwas_study_info,
                  here::here(&quot;output/gwas_cat/gwas_study_info_trait_cat.csv&quot;), 
                  sep = &quot;,&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] stringr_1.5.2     ggplot2_3.5.2     dplyr_1.1.4       data.table_1.17.8
[5] workflowr_1.7.1  

loaded via a namespace (and not attached):
 [1] sass_0.4.10        utf8_1.2.6         generics_0.1.4     tidyr_1.3.1       
 [5] renv_1.0.3         stringi_1.8.7      digest_0.6.37      magrittr_2.0.4    
 [9] evaluate_1.0.5     grid_4.3.1         timechange_0.3.0   RColorBrewer_1.1-3
[13] fastmap_1.2.0      rprojroot_2.1.0    jsonlite_2.0.0     processx_3.8.6    
[17] whisker_0.4.1      ps_1.9.1           promises_1.3.3     httr_1.4.7        
[21] purrr_1.1.0        scales_1.4.0       jquerylib_0.1.4    cli_3.6.5         
[25] rlang_1.1.6        withr_3.0.2        cachem_1.1.0       yaml_2.3.10       
[29] tools_4.3.1        httpuv_1.6.16      here_1.0.1         vctrs_0.6.5       
[33] R6_2.6.1           lifecycle_1.0.4    lubridate_1.9.4    git2r_0.36.2      
[37] fs_1.6.6           pkgconfig_2.0.3    callr_3.7.6        pillar_1.11.1     
[41] bslib_0.9.0        later_1.4.4        gtable_0.3.6       glue_1.8.0        
[45] Rcpp_1.1.0         xfun_0.53          tibble_3.3.0       tidyselect_1.2.1  
[49] rstudioapi_0.17.1  knitr_1.50         farver_2.1.2       htmltools_0.5.8.1 
[53] labeling_0.4.3     rmarkdown_2.30     compiler_4.3.1     getPass_0.2-4     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
