---
title: "GWAS Trait Categorisation"
author: "Isobel Beasley"
date: "2025-08-24"
output:
  workflowr::wflow_html:
    toc: true
    toc_float: true
    number_sections: true
editor_options: 
  markdown: 
    wrap: 72
---

# Set up

```{r setup, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      message = FALSE, 
                      warning = FALSE
                      )

library(data.table)
library(dplyr)
library(ggplot2)
library(stringr)

```


# Overlap ontology terms and GWAS traits

```{r}

gwas_study_info <- fread(here::here("output/gwas_study_info_cohort_corrected.csv"))

# fixing weird terms- where comma is in the term
# gwas_study_info |> 
#  mutate(n_commas_trait = stringr::str_count(MAPPED_TRAIT, ", "),
#         n_commas_uri = stringr::str_count(MAPPED_TRAIT_URI, ",")) |>
#   filter(n_commas_trait != n_commas_uri)


gwas_study_info =
gwas_study_info |> 
  rowwise() |>
  mutate(MAPPED_TRAIT = case_when(
                         # osteoarthritis, hip ... http://www.ebi.ac.uk/efo/EFO_1000786
                         grepl("EFO_1000786", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "osteoarthritis, hip", "hip osteoarthritis"),
                         # 	osteoarthritis, hand ... http://www.ebi.ac.uk/efo/EFO_1000789
                         grepl("EFO_1000789", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "osteoarthritis, hand", "hand osteoarthritis"),
                         # 	osteoarthritis, spine ... http://www.ebi.ac.uk/efo/EFO_1000787
                         grepl("EFO_1000787", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "osteoarthritis, spine", "spine osteoarthritis"),
                         # 	Hepatitis, Alcoholic, http://www.ebi.ac.uk/efo/EFO_1001345
                         grepl("EFO_1001345", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "Hepatitis, Alcoholic", "Alcoholic Hepatitis"),
                         # psoriasis 14, pustular http://purl.obolibrary.org/obo/MONDO_0013626
                         grepl("MONDO_0013626", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "psoriasis 14, pustula", "pustular psoriasis 14"),
                         # hypertension, pregnancy-induced http://purl.obolibrary.org/obo/MONDO_0024664
                         grepl("MONDO_0024664", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "hypertension, pregnancy-induced", "pregnancy-induced hypertension"),
                         # renal agenesis, unilateral http://purl.obolibrary.org/obo/MONDO_0019636
                         grepl("MONDO_0019636", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "renal agenesis, unilateral", "unilateral renal agenesis"),
                         # 	Cholecystitis, Acute http://www.ebi.ac.uk/efo/EFO_1001289
                         grepl("EFO_1001289", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "Cholecystitis, Acute", "Acute Cholecystitis"),
                         # 	Genital neoplasm, female http://www.ebi.ac.uk/efo/EFO_1001331
                         grepl("EFO_1001331", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "Genital neoplasm, female", "female reproductive organ cancer"),
                         # 	Anemia, Hemolytic, Autoimmune http://www.ebi.ac.uk/efo/EFO_1001264
                          grepl("EFO_1001264", MAPPED_TRAIT_URI) ~ stringr::str_replace_all(MAPPED_TRAIT, pattern = "Anemia, Hemolytic, Autoimmune", "autoimmune haemolytic anemia"),
                         TRUE ~ MAPPED_TRAIT
                         )
  ) |>
  ungroup()

# osteoarthritis, knee ... http://www.ebi.ac.uk/efo/EFO_0004616
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(
                         grepl("EFO_0004616", MAPPED_TRAIT_URI),
                         stringr::str_replace_all(MAPPED_TRAIT, pattern = "osteoarthritis, knee", "knee osteoarthritis"),
                         MAPPED_TRAIT
  )
  ) 

# rheumatoid factor-negative juvenile idiopathic arthritis
# polyarticular juvenile idiopathic arthritis, rheumatoid factor negative
# http://www.ebi.ac.uk/efo/EFO_1002020
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("EFO_1002020", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = "polyarticular juvenile idiopathic arthritis, rheumatoid factor negative", 
                                                        "rheumatoid factor-negative juvenile idiopathic arthritis"),
                               MAPPED_TRAIT)
         )

# http://www.ebi.ac.uk/efo/EFO_0007294, hand, foot and mouth disease,
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("EFO_0007294", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = "hand, foot and mouth disease", 
                                                        "hand foot and mouth disease"),
                               MAPPED_TRAIT)
         )

# susceptibility to migriane without aura http://purl.obolibrary.org/obo/MONDO_0011847 

gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("MONDO_0011847", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, pattern = "migraine without aura, susceptibility to, 4", "migraine without aura"),
                               MAPPED_TRAIT)
         )

# 	neural tube defects, susceptibility to, http://purl.obolibrary.org/obo/MONDO_0020705
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("MONDO_0020705", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = "neural tube defects, susceptibility to", 
                                                        "neural tube defects"),
                               MAPPED_TRAIT)
         )

# infantile diarrhea
# Diarrhea, Infantile http://www.ebi.ac.uk/efo/EFO_1001306
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("EFO_1001306", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = "Diarrhea, Infantile", 
                                                        "infantile diarrhea"),
                               MAPPED_TRAIT)
         )

# self reported traits 
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("EFO_0009803|EFO_0009822|EFO_0009803|EFO_0009817|EFO_0009822|EFO_0009819|EFO_0009823|EFO_0009824", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = ", self-reported$", 
                                                        " self-reported"),
                               MAPPED_TRAIT)
         )


# Hodgkins lymphoma, mixed cellularity http://www.ebi.ac.uk/efo/EFO_1002031
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("EFO_1002031", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, pattern = "Hodgkins lymphoma, mixed cellularity", "Hodgkins lymphoma mixed cellularity"),
                               MAPPED_TRAIT)
         )

# encephalopathy, acute, infection-induced, http://purl.obolibrary.org/obo/MONDO_0000166
gwas_study_info =
gwas_study_info |>
  mutate(MAPPED_TRAIT = ifelse(grepl("MONDO_0000166", MAPPED_TRAIT_URI),
                               stringr::str_replace_all(MAPPED_TRAIT, 
                                                        pattern = "encephalopathy, acute, infection-induced", 
                                                        "encephalopathy acute infection-induced"),
                               MAPPED_TRAIT)
         )


gwas_study_info = 
  gwas_study_info |>
  mutate(MAPPED_TRAIT = stringr::str_replace_all(MAPPED_TRAIT, 
                                                 "^level of .+, mitochondrial in blood$", "blood protein amount")
         )




```

```{r}
all_gwas_terms = unique(gwas_study_info$MAPPED_TRAIT)

all_gwas_terms = stringr::str_trim(tolower(all_gwas_terms))
```

## Disease Overlap (How many GWAS traits fall within disease or disorder terms?)

### Combine disease terms

```{r}

efo_descendants <- readLines(here::here("output/trait_ontology/efo_0000408_descendants.txt"))

mondo_descendants <- readLines(here::here("output/trait_ontology/mondo_0700096_descendants.txt"))

ncit_descendants <- readLines(here::here("output/trait_ontology/ncit_C2991_descendants.txt"))

orphanet_descendants <- readLines(here::here("output/trait_ontology/orphanet_557493_descendants.txt"))

age_of_onset_descendants <- readLines(here::here("output/trait_ontology/oba_2020000_descendants.txt"))

disease_measurement_terms <- readLines(here::here("output/trait_ontology/efo_0001444_disease_measurement_terms.txt"))

disease_typos = c("Alzheimer disease",
                  "late-onset Alzheimers disease",
                  "age of onset of Alzheimer disease",
                  "Chagas cardiomyopathy",
                  "Parkinson disease",
                  "Iron deficiency anemia",
                  "Churg-Strauss syndrome",
                  "Iridocyclitis",
                  "Phlebitis" 
                  )

other <- c("Allergic disease", 
  "Lewy body dementia",
  "Lewy body attribute",
 "non-Hodgkins lymphoma",
           "Ischemic Stroke",
           "Lung disease",
           "Respiratory System Disease",
  "Iron deficiency anemia (disorder)",
  "Alzheimer disease, APOE carrier status",
  "Alzheimer's disease biomarker measurement",
 "Genital neoplasm, female",
 "HIV-associated neurocognitive disorder",
 "encephalopathy acute infection-induced"
           )

disease_terms = c(mondo_descendants,
                  efo_descendants,
                  ncit_descendants,
                  orphanet_descendants,
                  age_of_onset_descendants,
                  disease_measurement_terms,
                  disease_typos,
                  other) |>
                 unique()


disease_terms = stringr::str_trim(tolower(disease_terms))

print("Number of terms related to disease or disorder")
length(disease_terms)

```



```{r}
# Find GWAS traits that fall within disease or disorder terms
simple_disease_terms = all_gwas_terms[all_gwas_terms %in% disease_terms]

# Also search for cases where there are multiple terms separated by commas
# and one of them is a disease term
not_simple_disease_terms = all_gwas_terms[!all_gwas_terms %in% disease_terms]

# sometimes there's multiple terms - check if any disease term is in these gwas terms
multiple_terms = grep(",", not_simple_disease_terms, value = T)

disease_chunks <- split(disease_terms, ceiling(seq_along(disease_terms) / 100))
disease_chunks  <- lapply(disease_chunks, function(x) paste0(x, collapse = "|"))
mask <- Reduce(`|`, lapply(disease_chunks, function(x) grepl(x, multiple_terms, ignore.case = T)))
additional_disease_gwas <- multiple_terms[mask]

disease_gwas = c(all_gwas_terms[all_gwas_terms %in% disease_terms],
                 additional_disease_gwas)

not_disease_terms = not_simple_disease_terms[!not_simple_disease_terms %in% additional_disease_gwas]

print("Number of GWAS traits under disease or disorder terms")
length(all_gwas_terms) - length(not_disease_terms)

print("Percentage of GWAS traits under disease or disorder terms")
round(100 * (length(all_gwas_terms) - length(not_disease_terms)) / length(all_gwas_terms),
      digits = 1)

print("Percentage of GWAS traits not under disease or disorder terms")
round(100 * length(not_disease_terms) / length(all_gwas_terms),
      digits = 1)


not_accounted_for = not_disease_terms 

```

## Phenotype abnormality overlap

```{r}


pheno_abnorm <- readLines(here::here("output/trait_ontology/hp_0000118_descendants.txt"))
pheno_abnorm = stringr::str_trim(tolower(pheno_abnorm))

# Find terms where all comma-split pieces are in measurement
pheno_abnorm_gwas <- not_accounted_for[
  sapply(strsplit(not_accounted_for, ", "), function(parts) {
    parts <- trimws(parts) # remove extra spaces
    all(parts %in% pheno_abnorm)
  })
]

additional_pheno_abnorm <- not_accounted_for[not_accounted_for %in% pheno_abnorm]

pheno_abnorm_gwas  = c(pheno_abnorm_gwas, additional_pheno_abnorm) |> unique()

print("Percentage of GWAS traits under phenotype abnormality terms")
round(100 * length(pheno_abnorm_gwas) / length(all_gwas_terms),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% pheno_abnorm_gwas]

print("Percentage of GWAS traits not accounted for so far")
round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)

print("Number of GWAS traits not accounted for by so far")
length(not_accounted_for)

```

## Add disease & phenotypic abnormality terms to GWAS study info dataset

```{r disease_terms}

find_disease_terms  <-   function(MAPPED_TRAIT) {
        # find all disease terms that appear in the trait
        split_mapped_traits <- stringr::str_split(MAPPED_TRAIT, ", ") |> 
                               unlist()
        
        mapped_disease_terms <- split_mapped_traits[split_mapped_traits %in% disease_terms]
        mapped_pheno_abnorm_terms <- split_mapped_traits[split_mapped_traits %in% pheno_abnorm]
        
        mapped_disease_terms = unique(c(mapped_disease_terms, 
                                        mapped_pheno_abnorm_terms
                                        )
        )
                                      
        return(paste0(mapped_disease_terms, collapse = ", "))  # combine multiple matches
        
    }

gwas_study_info <- 
  gwas_study_info |> 
  dplyr::rowwise() |>
  dplyr::mutate(
    disease_terms = 
      ifelse(stringr::str_trim(tolower(MAPPED_TRAIT)) %in% c(disease_gwas,pheno_abnorm_gwas),
             find_disease_terms(stringr::str_trim(tolower(MAPPED_TRAIT))),
             NA)
  )

gwas_study_info <- 
  gwas_study_info |>
  rowwise() |>
  dplyr::mutate(
   disease_terms = 
      ifelse(MAPPED_TRAIT == "",
             NA,
            disease_terms)
  ) |>
  ungroup()

```

## Measurement Overlap (how many GWAS traits fall within measurement terms?)

### Combine measurement

```{r}

measurement <- readLines(here::here("output/trait_ontology/efo_0001444_descendants.txt"))
total_choles <- readLines(here::here("output/trait_ontology/efo_0004574_descendants.txt"))

measurement <- c(total_choles,
                 measurement) 

measurement <- unique(measurement)

measurement <- c("cerebrospinal fluid composition attribute",
                 "blood protein amount",
                 measurement)

measurement = stringr::str_trim(tolower(measurement))
```

```{r}

# Find terms where all comma-split pieces are in measurement
measurement_gwas <- not_accounted_for[
  sapply(strsplit(not_accounted_for, ", "), function(parts) {
    parts <- trimws(parts)
    all(parts %in% measurement)
  })
]
additional_measurement <- not_accounted_for[not_accounted_for %in% measurement]

measurement_gwas  = c(measurement_gwas, additional_measurement) |> unique()

print("Number of GWAS traits under measurement terms")
length(measurement_gwas)

print("Percentage of GWAS traits under measurement terms")
round(100 * length(measurement_gwas) / length(all_gwas_terms),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% measurement_gwas]

print("Percentage of GWAS traits not accounted for by disease, disorder or measurement terms")
round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)

print("Number of GWAS traits not accounted for by disease, disorder or measurement terms")
length(not_accounted_for)
```

## Response to stimulus

### Combine response terms

```{r}

go_response = readLines(here::here("output/trait_ontology/go_0050896_descendants.txt"))

efo_response <- readLines(here::here("output/trait_ontology/efo_go_0050896_descendants.txt"))

response <- c(go_response,
              efo_response,
              "response to stimulus")

response <- unique(response)

response = stringr::str_trim(tolower(response))
```

```{r}

# Find terms where all comma-split pieces are in measurement
response_gwas <- not_accounted_for[
  sapply(strsplit(not_accounted_for, ", "), function(parts) {
    parts <- trimws(parts)
    all(parts %in% response)
  })
]
additional_response <- not_accounted_for[not_accounted_for %in% response]

measurement_gwas  = c(measurement_gwas, additional_response) |> unique()

print("Percentage of GWAS traits under response terms")
round(100 * length(response_gwas) / length(all_gwas_terms),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% response_gwas]

print("Percentage of GWAS traits not accounted for by disease, measurement or response terms")
round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)

print("Number of GWAS traits not accounted for by disease, measurement or response terms")
length(not_accounted_for)

```



## Mental process

```{r}

mental <- readLines(here::here("output/trait_ontology/efo_0004323_descendants.txt"))
mental = stringr::str_trim(tolower(mental))

mental_gwas = not_accounted_for[not_accounted_for %in% mental]

print("Percentage of GWAS traits under mental process terms")
round(100 * length(mental_gwas) / length(all_gwas_terms),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% mental_gwas]

print("Percentage of GWAS traits not accounted for thus far")
round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)

print("Number of GWAS traits not accounted for thus far")
length(not_accounted_for)

```

## Behavior

```{r}

behavior <- readLines(here::here("output/trait_ontology/go_0007610_descendants.txt"))
behavior = stringr::str_trim(tolower(behavior))

behavior_gwas = not_accounted_for[not_accounted_for %in% behavior]

print("Percentage of GWAS traits under behavouir terms")
round(100 * length(behavior_gwas) / length(all_gwas_terms),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% behavior_gwas]

print("Percentage of GWAS traits not accounted for so far")
round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)

print("Number of GWAS traits not accounted for so far")
length(not_accounted_for)

```

## Injury

```{r}

injury <- readLines(here::here("output/trait_ontology/efo_0000546_descendants.txt"))

injury = stringr::str_trim(tolower(injury))

injury_gwas = not_accounted_for[not_accounted_for %in% injury]

print("Percentage of GWAS traits under injury terms")
round(100 * length(injury_gwas) / length(all_gwas_terms),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% injury_gwas]

print("Percentage of GWAS traits not accounted for so far")
round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)

print("Number of GWAS traits not accounted for so far")
length(not_accounted_for)


```

## Phenotype

```{r}

phenotype <- readLines(here::here("output/trait_ontology/efo_0000651_descendants.txt"))

phenotype = stringr::str_trim(tolower(phenotype))

phenotype_gwas = not_accounted_for[not_accounted_for %in% phenotype]

print("Percentage of GWAS traits under phenotype terms")
round(100 * length(phenotype_gwas) / length(all_gwas_terms),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% phenotype_gwas]

print("Percentage of GWAS traits not accounted for so far")
round(100 * length(not_accounted_for) / length(all_gwas_terms),
      digits = 1)

print("Number of GWAS traits not accounted for so far")
length(not_accounted_for)

```

# Add Categories to GWAS Info

```{r}

gwas_study_info = 
gwas_study_info |>
  dplyr::mutate(MAPPED_TRAIT_CATEGORY = dplyr::case_when(is.na(MAPPED_TRAIT) ~ NA,
                                                         tolower(MAPPED_TRAIT) %in% disease_gwas ~ "Disease/Disorder",
                                                         tolower(MAPPED_TRAIT) %in% pheno_abnorm_gwas ~ "Phenotypic Abnormality",
                                                         tolower(MAPPED_TRAIT) %in% measurement_gwas ~ "Measurement",
                                                         tolower(MAPPED_TRAIT) %in% response_gwas ~ "Response",
                                                         tolower(MAPPED_TRAIT) %in% mental_gwas ~ "Mental Process",
                                                         tolower(MAPPED_TRAIT) %in% behavior_gwas ~ "Behavior",
                                                         tolower(MAPPED_TRAIT) %in% injury_gwas ~ "Injury",
                                                         tolower(MAPPED_TRAIT) %in% phenotype_gwas ~ "Phenotype",
                                                          TRUE ~ "Other"
                                                          )
                )

```

# Background traits

```{r}

gwas_study_info$MAPPED_BACKGROUND_TRAIT |> unique() -> gwas_background

gwas_background = stringr::str_trim(tolower(gwas_background))

length(gwas_background)

```

## Overlap with disease/disorder traits

```{r}

multiple_terms = grep(",", gwas_background, value = T)
mask <- Reduce(`|`, lapply(disease_terms, function(x) grepl(x, multiple_terms)))
additional_disease_gwas <- multiple_terms[mask]

disease_gwas = c(gwas_background[gwas_background %in% disease_terms],
                 additional_disease_gwas)

print("Number of background GWAS traits under disease or disorder terms")
length(disease_gwas)

print("Percentage of background GWAS traits under disease or disorder terms")
round(100 * length(disease_gwas) / length(gwas_background),
      digits = 1)

not_accounted_for = gwas_background[!gwas_background %in% disease_gwas]

```

```{r background_trait_disease_terms}

gwas_study_info <- 
  gwas_study_info |>
  rowwise() |>
  dplyr::mutate(
    background_disease_terms = 
      ifelse(stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% c(disease_gwas, pheno_abnorm_gwas),
             find_disease_terms(stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT))),
             NA)
  ) |>
  ungroup()


gwas_study_info <- 
  gwas_study_info |>
  rowwise() |>
  dplyr::mutate(
    background_disease_terms = 
      ifelse(MAPPED_BACKGROUND_TRAIT == "",
             NA,
             background_disease_terms)
  ) |>
  ungroup()

```

## Phenotype abnormality overlap

```{r}

# Find terms where all comma-split pieces are in measurement
pheno_abnorm_gwas <- not_accounted_for[
  sapply(strsplit(not_accounted_for, ", "), function(parts) {
    parts <- trimws(parts) # remove extra spaces
    all(parts %in% pheno_abnorm)
  })
]

additional_pheno_abnorm <- not_accounted_for[not_accounted_for %in% pheno_abnorm]

pheno_abnorm_gwas  = c(pheno_abnorm_gwas, additional_pheno_abnorm) |> unique()

print("Percentage of background GWAS traits under phenotype abnormality terms")
round(100 * length(pheno_abnorm_gwas) / length(gwas_background),
      digits = 1)

not_accounted_for = not_accounted_for[!not_accounted_for %in% pheno_abnorm_gwas]

print("Percentage of background GWAS traits not accounted for so far")
round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)

print("Number of background GWAS traits not accounted for by so far")
length(not_accounted_for)

```

## Measurement traits

```{r}

measurement_gwas <- not_accounted_for[
  sapply(strsplit(not_accounted_for, ", "), function(parts) {
    parts <- trimws(parts) # remove extra spaces
    all(parts %in% measurement)
  })
]

measurement_gwas  = c(measurement_gwas, additional_measurement) |> unique()
additional_measurement <- not_accounted_for[not_accounted_for %in% measurement]

measurement_gwas  = c(measurement_gwas, additional_measurement) |> unique()

not_accounted_for = not_accounted_for[!not_accounted_for %in% measurement_gwas]

print("Percentage of background GWAS traits not accounted for by disease, disorder or measurement terms")
round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)

print("Number of background GWAS traits not accounted for by disease, disorder or measurement terms")
length(not_accounted_for)


```

## Response traits

```{r}

# Find terms where all comma-split pieces are in measurement
response_gwas <- not_accounted_for[
  sapply(strsplit(not_accounted_for, ", "), function(parts) {
    parts <- trimws(parts)
    all(parts %in% response)
  })
]
additional_response <- not_accounted_for[not_accounted_for %in% response]

response_gwas  = c(response_gwas, additional_response) |> unique()

not_accounted_for = not_accounted_for[!not_accounted_for %in% response_gwas]

print("Percentage of background GWAS traits under response terms")
round(100 * length(response_gwas) / length(gwas_background),
      digits = 1)

print("Number of background GWAS traits under response terms")
length(response_gwas)

print("Number of background GWAS traits not accounted for by disease, measurement or response terms")
length(not_accounted_for)

print("Percentage of background GWAS traits not accounted for by disease, measurement or response terms")
round(100 * length(not_accounted_for) / length(gwas_background),
      digits = 1)


```

## Background trait categories 


```{r}

gwas_study_info = 
gwas_study_info |>
  dplyr::mutate(BACKGROUND_TRAIT_CATEGORY = 
                   dplyr::case_when(
                                      MAPPED_BACKGROUND_TRAIT == "" ~ NA,
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% disease_gwas ~ "Disease/Disorder",
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% pheno_abnorm_gwas ~ "Phenotypic Abnormality",
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% measurement_gwas ~ "Measurement",
                                      stringr::str_trim(tolower(MAPPED_BACKGROUND_TRAIT)) %in% response_gwas ~ "Response",
                                      TRUE ~ "Other")
                )


```



# Summary of number of disease studies (and studies of each kind of trait)

```{r}

gwas_study_info |>
  group_by(MAPPED_TRAIT_CATEGORY, BACKGROUND_TRAIT_CATEGORY) |>
  summarise(n_studies = n()) |> 
  arrange(desc(n_studies))

gwas_study_info = 
gwas_study_info |>
  dplyr::rowwise() |>
  dplyr::mutate(DISEASE_STUDY = 
                   ifelse(MAPPED_TRAIT_CATEGORY == "Disease/Disorder" | 
                          MAPPED_TRAIT_CATEGORY == "Phenotypic Abnormality" |  
                          BACKGROUND_TRAIT_CATEGORY == "Disease/Disorder" | 
                          BACKGROUND_TRAIT_CATEGORY == "Phenotypic Abnormality",
                          T, F )
                ) |>
  dplyr::ungroup() 

```

## Number of disease studies 

```{r}

gwas_study_info |>
  filter(DISEASE_STUDY == T) |>
  nrow()

```


# Creating disease labels column of just disease or phenotype abnormality terms for each study - so that we can see what diseases are being studied

## Make disease label column - combining disease terms from both mapped trait and background trait

```{r combined_disease_terms}

combined_disease_terms = function(MAPPED_TRAIT_1, MAPPED_TRAIT_2){
  
  
  MAPPED_TRAIT_1 = stringr::str_split(MAPPED_TRAIT_1, ", ") |> unlist()
  MAPPED_TRAIT_2  = stringr::str_split(MAPPED_TRAIT_2, ", ") |> unlist()
  
  all_mapped_disease_terms = 
    c(MAPPED_TRAIT_1, MAPPED_TRAIT_2) |>
    unique()
  
  combined_mapped_disease_terms = paste0(all_mapped_disease_terms, 
                                         collapse = ", ")
  
  return(combined_mapped_disease_terms)
  
}


gwas_study_info <- 
  gwas_study_info |>
  dplyr::rowwise() |>
  dplyr::mutate(all_disease_terms = 
                case_when(is.na(background_disease_terms) & is.na(disease_terms) ~ NA,
                          is.na(background_disease_terms) & !is.na(disease_terms) ~ disease_terms,
                          !is.na(background_disease_terms) & is.na(disease_terms) ~ background_disease_terms,
                          !is.na(background_disease_terms) & !is.na(disease_terms) ~
                            combined_disease_terms(background_disease_terms,
                                                   disease_terms)) 

  ) |>
  dplyr::ungroup()

```

## Minor fixes of trait categorisation and returning traits 

```{r}

# What studies are disease studies but have no collected disease terms?
gwas_study_info |> 
  filter(DISEASE_STUDY == T) |> 
  filter(all_disease_terms == "")  |> 
  select(MAPPED_TRAIT, MAPPED_TRAIT_CATEGORY) |> 
  distinct() |>
  nrow()

```

### Fix bug where MAPPED_TRAIT/BACKGROUND_MAPPED_TRAIT is empty string but TRAIT_CATEGORY is listed as disease/phenotypic abnormality

```{r}

gwas_study_info = gwas_study_info |>
  rowwise() |>
  mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT == "",
                                        "Other",
                                        MAPPED_TRAIT_CATEGORY)) |>
  mutate(BACKGROUND_TRAIT_CATEGORY = ifelse(MAPPED_BACKGROUND_TRAIT == "",
                                        "Other",
                                        BACKGROUND_TRAIT_CATEGORY)) 

```

### Fix some specific mis-categorizations

```{r}

# Fixing fractures, ununited as injury 
gwas_study_info = gwas_study_info |>
  mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT == "fractures, ununited",
                                        "Injury",
                                        MAPPED_TRAIT_CATEGORY)
         )   

# Fixing some response terms mislabelled as disease/disorder
other_response_terms = c("response to COVID-19 vaccine, localized superficial swelling, mass, or lump",
                         "response to COVID-19 vaccine, SARS-CoV-2 neutralizing antibody measurement",
                         "Anti-hepatitis B virus surface antigen IgG measurement, response to vaccine",
                         "anti-SARS-CoV-2 IgG measurement, response to COVID-19 vaccine",
                         "anti-tetanus toxoid IgG measurement, response to vaccine",
                         "height growth attribute, response to growth hormone",
                         "adrenal suppression measurement, response to corticosteroid",
                         "asthenia, response to COVID-19 vaccine",
                         "SARS-CoV-2 antibody measurement, response to COVID-19 vaccine",
                         "response to vaccine, anti-Haemophilus influenzae type b polyribosylribitol phosphate IgG measurement",
                         "height growth attribute, response to growth hormone"
                )
# Fixing some measurement terms mislabelled as disease/disorder
other_measurement_terms = c("amygdala volume, pallidum volume, nucleus accumbens volume, hippocampal volume, putamen volume, intracranial volume measurement, thalamus volume, caudate nucleus volume",
                            "lamina-associated polypeptide 2, isoforms beta/gamma measurement",
                            "apoptosis-inducing factor 1, mitochondrial measurement",
                            "central corneal thickness, intraocular pressure measurement",
                            "level of apoptosis-inducing factor 1, mitochondrial in blood serum",
                            "level of MHC class I polypeptide-related sequence A in blood, level of MHC class I polypeptide-related sequence B in blood",
                            "psychosocial stress measurement, intracranial volume measurement")

# Fixing some other terms mislabelled as disease/disorder
other_terms = c("insulin metabolic clearance rate measurement, disposition index measurement, insulin sensitivity measurement, glucose homeostasis trait, glucose effectiveness measurement, acute insulin response measurement",
                "disease recurrence, response to allogeneic hematopoietic stem cell transplant",
                "disease recurrence, response to allogeneic hematopoietic stem cell transplant, donor genotype effect measurement"
)

gwas_study_info =
  gwas_study_info |>
    mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT %in% other_response_terms,
                                        "Response",
                                        MAPPED_TRAIT_CATEGORY)) |>
    mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT %in% other_measurement_terms,
                                        "Response",
                                        MAPPED_TRAIT_CATEGORY)) |>
    mutate(MAPPED_TRAIT_CATEGORY = ifelse(MAPPED_TRAIT %in% other_terms,
                                        "Other",
                                        MAPPED_TRAIT_CATEGORY))  

```


### Fixing some specific missing disease terms

```{r}
# spine osteoarthritis was not being picked up as a disease term
gwas_study_info = 
  gwas_study_info |>
    mutate(disease_terms = ifelse(grepl("spine osteoarthritis", MAPPED_TRAIT),
                                        paste0("spine psteoarthritis", disease_terms, collapse = ","),
                                        disease_terms)
    ) |>
      mutate(all_disease_terms = ifelse(grepl("spine osteoarthritis", MAPPED_TRAIT),
                                        paste0("spine psteoarthritis", all_disease_terms, collapse = ","),
                                        all_disease_terms)
    )

# liver disease biomarker was not being picked up as a disease term 
gwas_study_info = 
  gwas_study_info |>
    mutate(disease_terms = ifelse(grepl("liver disease biomarker", MAPPED_TRAIT),
                                        paste0("liver disease", disease_terms, collapse = ","),
                                        disease_terms)
    ) |>
      mutate(all_disease_terms = ifelse(grepl("liver disease biomarker", MAPPED_TRAIT),
                                        paste0("liver disease", all_disease_terms, collapse = ","),
                                        all_disease_terms)
    )


gwas_study_info = 
  gwas_study_info |>
    mutate(disease_terms = ifelse(grepl("liver disease biomarker", MAPPED_TRAIT),
                                        paste0("liver disease", disease_terms, collapse = ","),
                                        disease_terms)
    ) |>
      mutate(all_disease_terms = ifelse(grepl("liver disease biomarker", MAPPED_TRAIT),
                                        paste0("liver disease", all_disease_terms, collapse = ","),
                                        all_disease_terms)
    )

gwas_study_info = gwas_study_info |>
  mutate(all_disease_terms = str_replace_all(all_disease_terms, 
                                             "liver diseaseliver diseaseNA", 
                                             "liver disease")
  ) |>  mutate(all_disease_terms = str_replace_all(all_disease_terms, 
                                             "liver diseaseliver disease", 
                                             "liver disease")
  ) 


# growth delay was not being picked up as a disease term
gwas_study_info = 
  gwas_study_info |>
    mutate(background_disease_terms= ifelse(grepl("Growth delay", MAPPED_BACKGROUND_TRAIT),
                                        paste0("growth delay", background_disease_terms, collapse = ","),
                                        background_disease_terms)
    ) |> 
      mutate(all_disease_terms = ifelse(grepl("Growth delay", MAPPED_BACKGROUND_TRAIT),
                                        paste0("growth delay", all_disease_terms, collapse = ","),
                                        all_disease_terms)
    )


gwas_study_info = gwas_study_info |>
  mutate(all_disease_terms = str_replace_all(all_disease_terms, 
                                             "growth delayNA", 
                                             "growth delay")
  ) 


```

### Recalculate disease study flag (is disease study or not?)

now that I have corrected any mistakes in categorization and added some missing disease terms

```{r}

gwas_study_info = 
gwas_study_info |>
  dplyr::rowwise() |>
  dplyr::mutate(DISEASE_STUDY = 
                   ifelse(MAPPED_TRAIT_CATEGORY == "Disease/Disorder" | 
                          MAPPED_TRAIT_CATEGORY == "Phenotypic Abnormality" |  
                          BACKGROUND_TRAIT_CATEGORY == "Disease/Disorder" | 
                          BACKGROUND_TRAIT_CATEGORY == "Phenotypic Abnormality",
                          T, F )
                ) |>
  dplyr::ungroup() 


gwas_study_info |> 
  filter(DISEASE_STUDY == T) |> 
  filter(all_disease_terms == "")  |> 
  select(MAPPED_TRAIT, MAPPED_TRAIT_CATEGORY) |> 
  distinct() |>
  nrow()

```

# Saving:

```{r}

data.table::fwrite(gwas_study_info,
                  here::here("output/gwas_cat/gwas_study_info_trait_cat.csv"), 
                  sep = ",")

```
